{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pymongo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pubcrawler.article import Article"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "from reporter import MongoQueryReporter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "client = pymongo.MongoClient(\"localhost\", 27017)\n",
    "articles = client.pmc.articles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To take the first one\n",
    "record = articles.find_one()\n",
    "article = Article(record[\"xml\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To sample a random article:\n",
    "record = articles.aggregate([\n",
    "    { \"$sample\": { \"size\": 1 } }\n",
    "]).next()\n",
    "article = Article(record[\"xml\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# An article with tables:\n",
    "record = articles.find(\n",
    "    { \"_id\": \"6112272\" }\n",
    ").next()\n",
    "article = Article(record[\"xml\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'pmid': '30159487', 'pmc': '6112272', 'publisher-id': 'S2468-2659(16)30016-6', 'doi': '10.1016/j.pld.2016.09.005'}\n"
     ]
    }
   ],
   "source": [
    "print(article.pub_ids())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "\n",
      "\n",
      "Introduction\n",
      "\n",
      "Loss of continuous natural areas and conversion of forests to smaller areas have drastically reduced the size of plant and animal populations, increasing the isolation between populations of the same species (Young and Clarke, 2000, Lowe et al., 2005). Plant reproduction may decrease in fragmented populations because plants may receive fewer flower visitors due to a decline in the richness and abundance of pollinators, modification in species composition and limitation in movement among landscapes (Young and Clarke, 2000, Cascante et al., 2002, Goverde et al., 2002, Frankham, 2010). Disturbance also changes the abundance, richness or behavior of animal seed dispersers affecting seed dispersal (Ghazoul, 2005, Lowe et al., 2005), recruitment and establishing of seedlings (Ewers and Didham, 2006), and the spatial genetic structure of plant populations (Hamrick, 2004). Shortly following a disturbance, levels of genetic homozygosity in affected plant populations are predicted to increase, and this may lead to the expression of recessive alleles deleterious to individual fitness (Hart and Clark, 1997, Reed and Frankham, 2003). In the long term, the loss of genetic variation may induce severe population bottlenecks due to genetic drift, high population divergence and low effective population size (Zhang et al., 2004, Moreira et al., 2009).\n",
      "\n",
      "Dry ecosystems around the world are experiencing major environmental impacts due to climate change since water is the major limiting factor for plant productivity in such environments (Fischer and Turner, 1978, Anderson et al., 1994, Hamrick et al., 2002). Other factors, such as strong drying winds, extreme temperatures, limited nutrients, high light intensity, overexploitation of natural resources, and desertification have caused habitat degradation, fragmentation and loss of biodiversity (McNeely et al., 1990, Ayyad, 2003).\n",
      "\n",
      "The Cactaceae family includes approximately 1600 species distributed in arid and semi-arid habitats (Gibson and Nobel, 1986, Dávila et al., 2002). Mexico is the most important center of diversification with almost 850 species and 54 genera (Bravo-Hollis, 1978, Arias, 1993, Anderson et al., 1994, Hernandez and Godinez, 1994). The Balsas River Basin and the Tehuacán Valley in central Mexico have a great number of cacti with approximately 81 species, of which 25% are endemic, constituting one of the principal structural and floristic elements of the dry forests (Arias et al., 1997). The species of Mammillaria predominate in this area and have biological characteristics that make them more vulnerable to the effects of disturbance in their populations such as low growth rates and long life cycles (Gibson and Nobel, 1986). Particularly, Mammillaria pectinifera is an extremely narrowly distributed species with only 18 fragmented populations remaining in the Tehuacán-Cuicatlán Valley in the states of Puebla and Oaxaca in Central Mexico (Martorell and Peters, 2005, Valverde et al., 2009). This species grows on relatively deep alkaline calcareous soils and the plants are localized on small isolated patches in xerophyllous scrub and grasslands (Arias et al., 1997). The populations of M. pectinifera fae serious risk of extinction, mainly due to massive collection of plants for commercial purposes and the degradation of their habitats (Glass, 1998, Valverde and Zavala-Hurtado, 2006). Grazing also causes the mortality of plants of M. pectinifera in its natural habitats (Meyrán, 1973, Bravo-Hollis, 1978, Bravo-Hollis and Sánchez-Mejorada, 1991). Thus, M. pectinifera is included in the category of risk of extinction in international (IUCN; CITES, 2013) and national (NOM 059 ECO1 2001; SEMARNAT, 2010) regulations.\n",
      "\n",
      "Most of the previous population genetic studies within the Cactaceae family have been mainly conducted on columnar and short globose cacti using allozymes (Nassar et al., 2001, Hamrick et al., 2002, Moraes et al., 2005, Parra et al., 2008), RAPDs (Clark-Tapia et al., 2005) and microsatellites (Figueredo et al., 2010, Castro-Felix et al., 2013, Solórzano et al., 2014). Cornejo-Romero et al. (2014) found that populations of species of Mammillaria were genetically structured indicating that geographic isolation and limited dispersal were the primary causes of genetic population differentiation.\n",
      "\n",
      "Ecological studies on Mammillaria have assessed the risk of extinction of its species due to the high specificity of the habitats where they occur and the degree of disturbance of the natural environments (Contreras and Valverde, 2002, Martorell and Peters, 2005, Valverde and Zavala-Hurtado, 2006, Valverde et al., 2009, Peters et al., 2014). Many aspects about genetic structure of the populations of M. pectinifera are unknown. Therefore, it is important to conduct studies to assess the genetic diversity of their populations and provide some guidelines for conservation. The main objective of this study was to examine the levels of genetic diversity of populations of M. pectinifera and to characterize its population genetic structure, inbreeding, recent bottlenecks and fluctuations in the effective population size to identify the factors that are determining the evolution and maintenance of this endangered microendemic species, and to propose conservation strategies to preserve this species.\n",
      "\n",
      "2\n",
      "\n",
      "\n",
      "Materials and methods2.1\n",
      "\n",
      "\n",
      "Study species\n",
      "\n",
      "\n",
      "M. pectinifera F.A.C.Weber in Bois (Cactaceae) is a small cactus that in habitat appears above the ground as a squat stem completely covered by flattened spines, and usually solitary. Flowers are medium-sized in a ring around the sides of the globe, white to pink in color with darker mid-strip, 20–30 mm long and stigmas are green (Arias et al., 1997). This species has a blooming season from December to March. Reproduction starts at an age of eight years. Fruits are small, red at maturity, barely emerging above the spines. Seeds, usually black, are retained among the plant tubercles and are released gradually (Arias et al., 1997).\n",
      "\n",
      "We collected from 18 populations of M. pectinifera in the Tehuacan-Cuicatlan Biosphere Reserve in central Mexico; from 7 to 13 individuals were sampled randomly with at least 100 m of distance between each individual. In all the samples, we did cuts about 1 cm^2 to get photosynthetic tissue, subsequently, the tissue was frozen at −70 °C until used for DNA extraction. For the DNA extraction (184 individuals), we followed the protocols of Doyle and Dickson (1987) and Doyle and Doyle (1987). Six nuclear DNA (nSSR) microsatellite loci were selected and amplified in multiplex polymerase chain reactions (PCR). Two groups of primers were arranged according to allele size and fluorescent labels. The first group was formed by the primer pairs for Mam VTC2, Mam VTC8, Mam VTC9, Mam VTC11, previously designed for the species M. crucigera by Solórzano-Lujano et al. (2009), whereas the second group included the primer pairs for P44 and P50, previously designed for Polaskia chichipe (Otero-Arnaiz, 2005). PCR was performed using the QIAGEN Multiplex PCR kit (QIAGEN) in a volume of 5 μl containing 1× Multiplex PCR Master Mix, 2 μM each primer, dH2O, and 20 ng template DNA. The thermal cycling conditions consisted of 40 cycles, each at 95 °C for 1 min, annealing for 1 min (i.e. first and second primer groups 58 °C and 60 °C, respectively), extension for 2 min at 72 °C and a final extension at 72 °C for 10 min. Multiplex PCR products were combined with a GeneScan-500 LIZ size standard and sequenced by an ABI-PRISM 3100 Avant sequencer (Applied Biosystems). Fragments were analyzed and registered with the Peak Scanner program 1.0 (Applied Biosystems).\n",
      "\n",
      "2.2\n",
      "\n",
      "\n",
      "Genetic diversity\n",
      "\n",
      "To test whether there was evidence of null alleles, upper allele dropout, or small genotyping errors due to stutter in the SSR data that we accumulated in 18 populations of M. pectinifera, we used the Micro-Checker v 2.2.3 program with 10^2 bootstrap simulations and a 95% confidence interval (Van Oosterhout et al., 2005). We calculated the values and standard error of mean number of alleles per locus (N_a), mean effective number of alleles (N_e), mean observed heterozygosity (H_O), mean expected heterozygosity (H_E) and mean of fixation index (F_IS) in 18 populations of M. pectinifera using the GENETIX 4 program (Belkhir et al., 2004).\n",
      "\n",
      "2.3\n",
      "\n",
      "\n",
      "Population structure and cluster analysis\n",
      "\n",
      "Population genetic differentiation R_ST was estimated by the stepwise mutation model (SMM) performed with 10^4 permutations in the ARLEQUIN 3.5.1.2 software (Excoffier and Lischer, 2010). A hierarchical test of population structure was estimated using the stepwise mutation model (SMM) performed with AMOVA in ARLEQUIN 3.5. (Excoffier et al., 2005). We compared the variance distribution between groups, among populations between groups and within populations. The statistical significance was tested using 10^4 permutations utilizing the resulting two groups of genotypes obtained previously by STRUCTURE.\n",
      "\n",
      "A cluster analysis using a Bayesian approach was conducted using STRUCTURE version 2.3.3 (Pritchard et al., 2000, Falush et al., 2003, Hubisz et al., 2009). In this analysis individuals are probabilistically assigned to one of the predefined K populations (gene pools) to identify the optimal number of genetic groups (Evanno et al., 2005). The optimum number of groups (K) was determined by varying the value of K from 1 to 10 and running the analysis ten times per K value, in order to determine the maximum value of posterior likelihood [LnP (D)]. Each run was performed using 10^5 burn-in periods and 10^6 Markov Chain Monte Carlo (MCMC) repetitions after burn-in. We used a model allowing for admixture with correlated allelic frequencies without any prior information. Also, we determined the most probable value of K using the maximum value of ΔK according to Evanno et al. (2005) implemented in the program Structure Harvester 0.6.1 (Earl and Von Holdt, 2011).\n",
      "\n",
      "To identify probable geographic and genetic breaks among populations of M. pectinifera, we used the Monmonier's maximum difference algorithm with the software BARRIER version 2.2 (Manni et al., 2004) that creates a map of sampling locations from their geographical coordinates. Barriers are then characterized on the map by identifying the maximum values within the population pairwise genetic distance matrix. We employed an average square distances (ASD) matrix of (Goldstein et al., 1995, Slatkin, 1995) estimated for 18 populations of M. pectinifera. Resampling random subsets of individuals within populations provided 100 bootstrap replicate distances were constructed with the MSA program (Dieringer and Schlötterer, 2003) to accomplish statistical significance for the predicted barriers.\n",
      "\n",
      "Population connectivity among M. pectinifera was scrutinized with the POPGRAPH library (http://dyerlab.bio.vcu.edu/software.html), this approaches generate a network of population linkages and defines the degree of genetic variation within populations (Dyer and Nason, 2004). In the POPGRAPH context, the set of nodes represents sampled populations, and the edges represent the multivariate measures of genetic covariance among populations. The difference in node size reflects differences in within-population genetic variability, while the edge length represents the among-population component of genetic variation due to the connecting nodes (Dyer and Nason, 2004). POPGRAPH, recognize population pairs where long distance migration may have occurred by indicating population pairs with significantly greater inter-site distances (Dyer et al., 2010). It also identifies pairs of populations that are located significantly closer than predicted by inter-site separation, which suggests that a direct barrier might exist between the linear distances.\n",
      "\n",
      "2.4\n",
      "\n",
      "\n",
      "Population bottleneck and effective population size\n",
      "\n",
      "To detect evidence of recent bottlenecks in M. pectinifera populations, we used the BOTTLENECK 1.2 software (Piry et al., 1999) to examine the two genetic groups previously obtained by STRUCTURE. Due to small sample sizes, our analysis was limited to. Also, England et al. (2006) showed that the disequilibrium method leads to severe downward bias in the estimate of Ne when sample size (S) is small relative to effective size. Therefore, the estimation by population will result in bias due to the small sample size of the population. Recent bottlenecks could be defined as a population where the rare alleles are the first to be lost, diminishing the mean number of alleles per locus. In contrast, heterozygosity is less affected, generating a transient excess in heterozygosity compared to that expected given the resulting number of alleles (Cornuet and Luikart, 1996, Luikart and Cornuet, 1998). To analyze the data set, we used 90% stepwise and 10% multistep mutations 10^4 iterations with the Wilcoxon signed-rank test, and the stepwise mutation (SMM), the infinite allele (IAM) and two-phase mutation (TPM) models. Moreover, we also assessed the effective population size in populations of M. pectinifera with the program LDNe (Waples and Do, 2008), which implements the bias-correction method developed by Waples (2006) to obtain Ne from each sample of S individuals. For LDNe, we used the criterion Pcrit = 0.02 (alleles with frequency ˂ 0.02 are excluded), which generally provides a good balance among accuracy and bias (Waples and Do, 2008). Confidence intervals (CIs) for Ne were based in the chi-square approximation implemented by LDNe (Waples, 2006).3\n",
      "\n",
      "\n",
      "Results3.1\n",
      "\n",
      "\n",
      "Genetic diversity\n",
      "\n",
      "In general, we did not find evidence for null alleles over all sample-loci combinations, and tests for error due to stutter and upper allele dropout were negatives in all cases. The values of the genetic diversity parameters estimated in 18 populations of M. pectinifera are reported in Table 1. We observed a wide range of values from 4.50 to 7.00 in mean number of effective alleles per locus (Na) and 0.716 to 0.950 in mean observed heterozygosity (H_O) values (Table 1). Wright's inbreeding coefficients within populations (F_IS) were positive in almost all cases, indicating a deficit of heterozygotes ranging from F_IS = 0.071 to 0.621 and some populations with negative values ranging F_IS = −0.010 to 0.446 (Table 1).\n",
      "\n",
      "[Omitted Table: 'Table 1']\n",
      "\n",
      "3.2\n",
      "\n",
      "\n",
      "Population structure and cluster analysis\n",
      "\n",
      "Pairwise genetic differentiation among populations indicated low to moderate genetic differentiation based on R_ST (Table 2). Hierarchical analysis of molecular variance (AMOVA), performed for the SMM mutation models (i.e. R_ST), indicated that most of the genetic variation resided within populations (Φ_ST = 80.63%, P = 0.001) followed by variation among populations within groups (Φ_SC = 10.32%, P = 0.001), while the differentiation among groups only accounted for the remaining variation (Φ_CT = 9.04%, P = 0.001) (Table 2).\n",
      "\n",
      "[Omitted Table: 'Table 2']\n",
      "\n",
      "\n",
      "\n",
      "The highest posterior probability was obtained from Bayesian likelihood [LnP (D)] and the ΔK approach (Evanno et al., 2005) implemented in the program Structure Harvester (Earl and Von Holdt, 2011). This statistic determined that K = 2 is the optimum value for the number of genetic clusters included in the analysis (Fig. 1). Both genetic groups showed a very widespread distribution and also some of them exhibited a great amount of exchange among groups. For instance, the first genetic group (i.e. Cluster 1, green) of populations were consistently structured and very widespread over all the distribution of M. pectinifera. In addition, some of the populations had a greater ancestry coefficient Q from 0.669 to 0.847 such as the populations CO, CH, TE, SM, RE, SI, ZA and LA (Fig. 2). Populations from the second genetic group (i.e. Cluster 2, red), such as FR, SU, EC, CC, AC, ER and CQ had a widespread distribution and were very well structured with ancestry coefficient (Q) going from 0.710 to 0.904. The rest of populations TC SB and NO showed a greater amount of exchange among genetic groups with an ancestry ranged from Q 0.410–0.590 (Fig. 2). Table 3.\n",
      "\n",
      "Fig. 1 Mean and standard deviation values of ΔK for ten independent runs of STRUCTURE plotted against the number of genetic groups K. The peak indicates the most probable number of genetic groups given the data using Structure Harvester (Earl and Von Holdt, 2011).Fig. 1\n",
      "\n",
      "Fig. 2 Map shows sampling localities representing 18 populations of M. pectinifera in the Tehuacán-Cuicatlán Biosphere Reserve. Green genotype and red genotype represent the genetic ancestry groups corresponding to M. pectinifera populations obtained by the program STRUCTURE. Black bars show the barriers detected with the BARRRIERS program.Fig. 2\n",
      "\n",
      "[Omitted Table: 'Table 3']\n",
      "\n",
      "\n",
      "\n",
      "In total, four barriers were identified among the 18 populations of M. pectinifera using the Barriers analysis with 100 bootstrap replicates of ASD genetic distance matrixes (Fig. 2). The most significant barrier, with a bootstrap support of 98%, separated TC from SM in the northern part of the distribution. The second barrier, with 90% bootstrap value, indicated that there is a complex barrier that divides SM from the rest of the populations in the center and the south-western region. The third barrier, with 87% bootstrap support, separated populations in the center of the distribution, including, CH, EC, LA and SU. The fourth barrier, with a 79% bootstrap support, separated the populations from RE and NO located in the center-west of the distribution of M. pectinifera (Fig. 2).\n",
      "\n",
      "The POPGRAPH network of populations based on nSSR genotypes had 17 edges out of the possible 22, which indicates extensive historical connectivity among most populations (Fig. 3). The most common colonization patterns occurred in a north–south axis along the Tehuacán-Cuicatlán Biosphere Reserve, throughout southern portions of Puebla to Zapotitlán, but it was not uncommon to see east–west gene exchanges, especially across the north–east to the south–west part of the species range. Our data indicate occasional long-distance gene exchange across the southern portion of Tehuacán, FR, to the north toward TC and CH, where the genotypes are significantly more similar than expected based on spatial distance. Our analysis also identified a great amount of network connection between the center of the distribution towards the northwest, south, but not much connectivity with the southwest portion, which were more genetically dissimilar than predicted by spatial distance (Fig. 3).\n",
      "\n",
      "Fig. 3 Patterns of genetic connectivity among 18 populations of M. pectinifera in the Tehuacán-Cuicatlán Biosphere Reserve, Mexico performed with the program POPGRAPH.Fig. 33.3\n",
      "\n",
      "\n",
      "Population bottleneck and effective population size\n",
      "\n",
      "The results of the analysis to test the evidence of recent bottlenecks (excess of heterozygosity) using the infinite allele model (IAM), two-phase model (TPM) and stepwise mutation models (SMM) are presented in Table 4. Results were significant (P < 0.050) for the IAM P = 0.0003 and the SMM P = 0.006 models observed in the loci VTCMAM11, P50 and P44. Our analysis revealed that some populations of M. pectinifera departed from neutrality, exhibiting excess heterozygosity. This suggests that the populations have recently undergone a bottleneck. The results of estimation of effective population size (N_e), performed with the program LDNe, in populations of M. pectinifera showed that both genetic groups had moderated values in the first group (N_e = 455 individuals), followed by the second group (N_e = 89.8); in all cases, estimates had high Jackknife support and a good confidence interval (CIs) (Table 4).\n",
      "\n",
      "[Omitted Table: 'Table 4']\n",
      "\n",
      "4\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "Anthropogenic activities, such as massive extraction of plants and disturbance of natural habitats, may lead to a decrease in population size and changes in the genetic structure of species populations (Lowe et al., 2005). M. pectinifera has been subject to these pressures, particularly in reproductive stages, which has led to a significant reduction in their populations. Unexpectedly, the populations of M. pectinifera have been shown to have similar (López-Ortiz et al., 2013, Macías-Arrastio, 2013) or higher (Ibarra-Suárez, 2009, Tapia-Salcido, 2011) values of genetic diversity compared to other short-globose cacti species of the same genus. However, we found significant evidence of recent bottlenecks in some populations of M. pectinifera. Lowe et al. (2005) suggested that after an episode of habitat disruption, populations that remained small for many generations suffer loss of allelic diversity and random genetic drift (Hamrick, 2004, Ellstrand and Ellam, 1993). The destruction and removal of reproductive adult trees might result in fecundity variance, reduced effective population size and limited gene flow among the remnant populations (Ewers and Didham, 2006). Furthermore, M. pectinifera has life history traits that make it more vulnerable to the effects of disturbance over the course of generations such as low growth rates, long life cycles and serotiny (Gibson and Nobel, 1986, Bravo-Hollis and Sánchez-Mejorada, 1991, Valverde et al., 2009).\n",
      "\n",
      "High inbreeding values, moderate to high heterozygosity and low to moderate effective population size are positive indicators that predict short-term population viability in M. pectinifera. While only a relatively small number of individuals may be needed in the short term to prevent inbreeding, greater genetic variation is thought to increase the likelihood of survival over much longer time scales (Frankham, 2010, Jamieson and Allendorf, 2012). Adaptation or resilience to novel or changing habitats is the most important feature (Williams and Hoffman, 2009). Thus, management strategies should be implemented that maintain, as far as possible, the full complement of genetic diversity within the Tehuacán-Cuicatlán Biosphere Reserve.\n",
      "\n",
      "Hierarchical analysis of molecular variance revealed that most of the genetic variation is within populations, and there is low genetic differentiation among populations. This observation was confirmed by the POPGRAPH analysis that also indicated extensive historical connectivity. The connectivity values obtained for M. pectinifera were relatively high and it is likely the result of historical events of gene flow through pollen and seed dispersion by biotic agents. However, extant populations of M. pectinifera are distributed in isolated patches with few opportunities for seed dispersal across valleys and foothills (Valverde and Zavala-Hurtado, 2006, Valverde et al., 2009; and Peters et al., 2014). Furthermore, populations of Mammillaria in the Tehuacán Valley present a restricted spatial distribution, high habitat specificity in restricted altitudinal ranges, soil and vegetation type (Hernandez and Godinez, 1994, Hershkovitz and Zimmer, 1997). In particular, M. pectinifera plants occurred only at altitudinal gradients from 1778 to 2100 m in mild to moderate slopes, on relatively deep alkaline calcareous soils, with high water retention capacity on stony hills (Valverde and Zavala-Hurtado, 2006, Valverde et al., 2009, Peters et al., 2014).\n",
      "\n",
      "The massive extraction of plants, deforestation, changes in land use, and over-exploitation of areas by human activities have damaged the natural habitats and the populations of M. pectinifera (Zavala-Hurtado and Valverde, 2003, Martorell and Peters, 2005). Despite this, we observed that populations of M. pectinifera still show moderate to high levels of genetic diversity due to their strong capacity to adapt and survive in localities with high water deficits (Hernandez and Godinez, 1994, Hershkovitz and Zimmer, 1997). However, if not properly preserved, this species will become extinct in the future. For instance, according to Peters et al. (2014), 45% of all known populations of M. pectinifera are located outside of natural protected areas. For that reason, we suggest in situ conservation to prevent the decrease in population sizes and loss of genetic diversity. Natural reserves such as the Tehuacán-Cuicatlán Biosphere Reserve must play an essential role in protecting and restoring the habitat of M. pectinifera (Contreras and Valverde, 2002, Peters et al., 2014). Also, a long-term ex situ conservation program which includes collecting seed from different sources and optimizing seed germination and plant establishment protocols is needed to restore disturbed habitats. Also, a supply of living plants for trade would prevent further extraction of plants from nature.\n"
     ]
    }
   ],
   "source": [
    "print(record[\"extracted_text\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "cursor = articles.find({})\n",
    "reporter = MongoQueryReporter(1, articles, {})\n",
    "\n",
    "for idx, record in enumerate(cursor):\n",
    "    article = Article(record[\"xml\"])\n",
    "\n",
    "    record_update = {\n",
    "        \"extracted_text\": article.extract_text()\n",
    "    }\n",
    "\n",
    "    articles.update_one(\n",
    "        filter={\"_id\": record[\"_id\"]}, update={\"$set\": record_update}\n",
    "    )\n",
    "\n",
    "    reporter.report(idx)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['field work',\n",
       " 'fieldwork',\n",
       " 'field study',\n",
       " 'field site',\n",
       " 'field area',\n",
       " 'study site',\n",
       " 'study location',\n",
       " 'study area',\n",
       " 'research site',\n",
       " 'research location',\n",
       " 'sampling site',\n",
       " 'sampling location',\n",
       " 'sampling area']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def print_article(article):\n",
    "    print(\"\"\"\n",
    "PMC ID\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TITLE\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "METADATA\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TEXT MATCHES\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "EXTRACTED TEXT\n",
    "\n",
    "{}\"\"\".format(article.get(\"_id\"),\n",
    "             article.get(\"article_title\"),\n",
    "             article.get(\"article_meta\"),\n",
    "             article.get(\"text_matches\"),\n",
    "             article.get(\"extracted_text\").strip()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def view_one_article(query={ \"text_matches\": { \"$in\": terms } }):\n",
    "    query = [\n",
    "        { \"$match\": query },\n",
    "        { \"$sample\": { \"size\": 1 } }\n",
    "    ]\n",
    "    article = articles.aggregate(query).next()\n",
    "    print(\"\"\"\n",
    "PMC ID\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TITLE\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "METADATA\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TEXT MATCHES\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "EXTRACTED TEXT\n",
    "\n",
    "{}\"\"\".format(article.get(\"_id\"),\n",
    "             article.get(\"article_title\"),\n",
    "             article.get(\"article_meta\"),\n",
    "             article.get(\"text_matches\"),\n",
    "             article.get(\"extracted_text\").strip()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['field work',\n",
       " 'fieldwork',\n",
       " 'field study',\n",
       " 'field site',\n",
       " 'field area',\n",
       " 'study site',\n",
       " 'study location',\n",
       " 'study area',\n",
       " 'research site',\n",
       " 'research location',\n",
       " 'sampling site',\n",
       " 'sampling location',\n",
       " 'sampling area']"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "2824121\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "Phase II study of motesanib in Japanese patients with advanced gastrointestinal stromal tumors with prior exposure to imatinib mesylate\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "{'article_type': 'research-article', 'has_body': True, 'all_geospans': 15, 'geospan_density': 0.0007973633850733575, 'n_geospans': 15}\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['study site']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Introduction\n",
      "\n",
      "Gastrointestinal stromal tumor (GIST) is a rare stromal neoplasm that predominantly arises from the muscularis propria layers, representing the most common mesenchymal tumor of the gastrointestinal tract. Although the primary therapy for nonmetastatic GIST is surgical resection, there still remain unresectable cases of advanced or metastatic GIST. Unresectable GISTs are resistant to conventional chemotherapy and radiotherapy [1]. Before imatinib mesylate was introduced in clinical practice, the prognosis for patients with unresectable GIST was dismal, with a median survival period of 22 months [2].\n",
      "\n",
      "The critical transforming and oncogenic mechanisms of GISTs are activating mutations in the stem-cell factor receptor, KIT tyrosine kinase [3]. About 5% of GISTs are caused by activating mutations of the platelet-derived growth factor receptor alpha (PDGFRA), and are independent of c-kit mutations [4]. The c-kit and PDGFRA mutations appear to be alternative and mutually exclusive oncogenic mechanisms in GIST.\n",
      "\n",
      "Imatinib mesylate blocks the constitutively activated form of KIT in GISTs, and has dramatically improved the outcome for patients with unresectable GIST [5]. Treatment with imatinib mesylate results in partial response (PR) or stable disease (SD) in approximately 80% of patients with advanced or metastatic GIST [6], and the 2-year survival rate of these patients is reported to be 70% [7].\n",
      "\n",
      "However, approximately 10–15% of advanced GIST patients will suffer a progressive disease (PD) despite treatment with imatinib mesylate. Many of the patients who initially responded to imatinib mesylate therapy experience tumor progression after an average of 2 years of treatment [7, 8].\n",
      "\n",
      "Sunitinib is an orally administered receptor tyrosine kinase (RET) inhibitor that targets multiple kinases and is used as a second-line treatment for patients with imatinib-resistant or -intolerant GIST. A Phase III double-blind, placebo-controlled trial comparing sunitinib with placebo showed that the time to progression was significantly longer in the sunitinib group than in the placebo group (6.3 versus 1.5 months). Adverse reactions, though observed, were acceptable [9]. However, despite initial response or stabilization, the disease developed resistance in most patients after approximately 7 months. Because no therapies are available for patients with GIST once imatinib and sunitinib fail, there exists a need for alternative agents that block the signaling pathways in GIST cells.\n",
      "\n",
      "Motesanib is a novel, synthetic, small molecule that strongly and selectively inhibits vascular endothelial growth factor receptors 1, 2, and 3, as well as the cellular KIT, the platelet-derived growth factor receptor (PDGFR), and the glial-derived nerve growth factor family ligand RET. The safety and pharmacokinetic (PK) profile of motesanib were evaluated in a Phase I, monotherapy, open-label, dose-finding study [10]. In this study, motesanib showed clinical activity in patients with advanced refractory solid tumors; SD was observed in a significant proportion of the patients, although the overall tumor response rate was low.\n",
      "\n",
      "The above findings prompted us to conduct a Phase II study to evaluate the efficacy, safety, and PK of motesanib in Japanese patients with advanced GIST, after failure or withdrawal of imatinib mesylate due to resistance or intolerance.\n",
      "\n",
      "\n",
      "Materials and methods\n",
      "\n",
      "\n",
      "Patients\n",
      "\n",
      "Japanese patients with pathologically confirmed advanced or metastatic GIST were eligible for this study if they met the following criteria; age ≥ 20 years; a proven KIT positive or activating mutation of PDGFR; prior imatinib mesylate therapy of 400 mg/day or more for at least 8 weeks; disease progression or relapse while on previous treatment with imatinib mesylate; at least one tumor lesion measurable by a computed tomographic (CT) scan or magnetic resonance imaging (MRI); an Eastern Cooperative Oncology Group (ECOG) performance status (PS) of 0 to 2; a life expectancy of more than 3 months; adequate organ functions as defined by: neutrophils ≥1.5 × 103 cells/mm3, platelets ≥1.0 × 104 cells/mm3, hemoglobin ≥9.0 g/dl, serum creatinine ≤2.0× upper limit of normal (ULN), urine protein quantitative value of ≤1+ on dipstick or 30 mg/dl in urinalysis, aspartate aminotransferase ≤2.5 × ULN (5.0 × ULN in patients with liver metastasis), alanine aminotransferase ≤2.5 × ULN (5.0 × ULN in patients with liver metastasis), alkaline phosphatase ≤2.5 × ULN (5.0 × ULN in patients with bone or liver metastasis), and total bilirubin ≤2.0 × ULN. This protocol was approved by the Institutional Review Board at each study site. All patients provided written informed consent.\n",
      "\n",
      "\n",
      "Study design\n",
      "\n",
      "This study was an open-label and multicenter Phase II clinical study. The primary endpoint was the objective response rate to a once daily oral treatment with 125 mg motesanib in patients with advanced GISTs who experienced disease progression or relapse while on imatinib mesylate treatment. (Sunitinib was not approved for imatinib-resistant GIST until 2 years after their study was completed.) The secondary endpoints were duration of response, progression-free survival (PFS), time to response, overall survival, and PK profiles of motesanib in Japanese patients with advanced GISTs.\n",
      "\n",
      "The dose was firstly reduced to 100 mg, and if need be, to 75 mg in the second time. If the grade 3 adverse event (AE) is not adequately controlled with appropriate supportive care or a grade 4 AE occurs, motesanib was withheld. Once the grade 3 or 4 AE has resolved to baseline or grade ≤1 for nonhematologic toxicities or baseline or grade ≤2 for hematologic toxicities, the dose was reduced by 25 mg and treatment was resumed. If treatment with motesanib was withheld for >21 days, the patient should be withdrawn from the treatment period and complete the end of study procedures. If a patient was receiving 75 mg and requires a dose reduction, treatment with motesanib was stopped and the patient should complete the end of study procedures.\n",
      "\n",
      "Tumor evaluation was performed after 8 weeks and at every 8 weeks thereafter, by using the modified Response Evaluation Criteria in Solid Tumor (RECIST). A confirmation of tumor response was performed by using the modified RECIST at least 4 weeks after a complete response (CR) or PR was first documented. An appointed radiographic image reviewer who was independent of the study site or the study sponsor reviewed CT or MRI of all patients.\n",
      "\n",
      "The severity of AEs was graded according to Common Toxicity Criteria for Adverse Events (CTCAE, version 3). Special attention was paid to cardiac function, hypertension, hypothyroidism, and cholecystitis. Laboratory assessments (serum chemistry, hematology, thyroid hormones, blood pressure, and electrocardiogram) were performed every 2 weeks.\n",
      "\n",
      "Ten patients had the following PK parameters: maximum observed plasma concentration (Cmax), terminal elimination half-life (t1/2), area under the plasma concentration–time curve from time 0 to 24 h after dosing (AUC0–24), concentration at 24 h after dosing (C24), maximum plasma concentration time (tmax), the area under the plasma concentration versus time curve from 0 to infinity (AUC0–inf), and apparent plasma clearance (CL/F). These PK parameters of motesanib were calculated by the standard noncompartmental model using WinNonlin software version 4.1e (Pharsight Corporation, Mountain View, CA, USA) and summarized according to the study day and history of gastrectomy using descriptive statistics. Individual plasma concentration–time profiles were summarized by history of gastrectomy.\n",
      "\n",
      "\n",
      "Statistical analysis\n",
      "\n",
      "Descriptive statistics are provided for each endpoint. The safety analysis population consisted of all patients who received at least one dose of motesanib. The objective response rate and its two-sided 95% confidence interval (95% CI) were calculated. The CI was constructed by the exact method described by Collett [10]. For a PFS, calculated as the number of days between the first dose of motesanib and the date when radiological evidence of disease progression is determined (date of CT scan/MRI), or death (regardless of cause), whichever comes first (date of PD or death minus date of first dose of motesanib), Kaplan–Meier curve (with two-sided 95% CI) was generated and its standard error was calculated using Greenwood’s formula. Statistical analyses were performed using the SAS statistical software package (SAS Institute Inc., Cary, NC, USA) [11].\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "\n",
      "Patient population\n",
      "\n",
      "A total of 35 patients were enrolled and treated with motesanib between November 2005 and June 2006 at the following sites: Aichi Cancer Center Hospital, Osaka University Hospital, National Cancer Center Hospital, Hokkaido University Hospital, Niigata University Hospital, National Cancer Center Hospital East, National Hospital Organization Kure Medical Center, and Kumamoto University Hospital. One patient did not undergo baseline CT assessment. Hence, 34 patients were eligible for tumor response evaluation, and 35 for toxicity evaluation. Baseline demographic and clinical characteristics are summarized in Table 1. Of the 35 patients enrolled, 17 (49%) were female and the median age was 62.0 years (range 31–83 years). Every patient was diagnosed as having GIST with positive immunoreactivity for KIT protein. The most common primary sites of the tumor were the small intestine (n = 17) and the stomach (n = 10). The other sites of the tumor were the colon (n = 2) and the rectum (n = 2). All patients had received treatment with imatinib mesylate but not with other tyrosine kinase inhibitors. The mean time from the last imatinib treatment was 0.9 months (range 0.2–5.5 months).\n",
      "\n",
      "Table 1\n",
      "All patients (N = 35)\n",
      "Sex, n (%)\n",
      " Female\t17 (49)\n",
      " Male\t18 (51)\n",
      "Age\n",
      " Median\t62.0\n",
      " Min, max\t31, 83\n",
      "Age group, n (%)\n",
      " <65 years\t23 (66)\n",
      " ≥65 years\t12 (34)\n",
      " ≥75 years\t4 (11)\n",
      "ECOG PS, n (%)\n",
      " 0\t24 (69)\n",
      " 1\t9 (26)\n",
      " 2\t2 (6)\n",
      "Site of primary tumor at diagnosis, n (%)\n",
      " Small intestine\t17 (49)\n",
      " Stomach\t10 (29)\n",
      " Colon\t2 (6)\n",
      " Rectum\t2 (6)\n",
      "\n",
      "\n",
      "Outcome measures\n",
      "\n",
      "The tumor response as assessed by an independent radiographic image reviewer is shown in Table 2. No CR was observed among the 35 patients enrolled in this study. One patient (3%; 95% CI 0.1–14.9%) had a PR and seven patients (20%) demonstrated SD for at least 24 months, two of whom continued to have SD for more than 2 years. Twelve additional patients had SD lasting for 12 weeks or more. Thirteen patients experienced disease progression within 12 weeks. The patient with PR had a gastric GIST with spindle-cell type, exon 11 mutation, liver, and peritoneal metastases, and had initially responded to imatinib with SD as assessed by RECIST (Fig. 1). The median PFS time of motesanib was 16.1 weeks (95% CI 8.4–32.0 weeks; Fig. 2).\n",
      "\n",
      "Table 2\n",
      "All patients (N = 35)\n",
      "Patients with measurable disease at baseline\t34 (97)\n",
      "Response assessment, n (%)\n",
      "Confirmed CR\t0 (0)\n",
      "Confirmed PR\t1 (3)\n",
      "SDa\t19 (54)\n",
      "PD\t13 (37)\n",
      "Unevaluableb\t1 (3)\n",
      "Not done\t1 (3)\n",
      "Confirmed objective response (CR or PR)\t1 (3)\n",
      "95% CIc\t0.1–14.9\n",
      "Durable SDd\t7 (20)\n",
      "\n",
      "Fig. 1 A 68-year-old male with a primary GIST of the stomach and recurrent liver and peritoneal metastases. a Pre-treatment CT scan shows multiple low-density masses. b CT scan obtained after 3 months of treatment with once daily motesanib 125 mg shows that the multiple lesions have become significantly smaller and less dense\n",
      "\n",
      "Fig. 2 Kaplan–Meier estimates of PFS\n",
      "\n",
      "\n",
      "Safety and tolerability\n",
      "\n",
      "Table 3 summarizes treatment-related adverse events (TRAEs) (patient incidence ≥15%). The most frequent nonhematologic TRAEs included hypertension (63% of patients), diarrhea (51%), and fatigue (43%). Five patients (14%) experienced grade 3 hypertension and two patients (6%) experienced grade 3 fatigue. The only hematological toxicity was anemia (grade 2 in 3% of patients, grade 3 in 6% of patients, and grade 4 in 0% of patients). One patient (3%) experienced grade 4 hyperuricemia. No grade 5 TRAEs occurred.\n",
      "\n",
      "Table 3\n",
      "Preferred term\tNumber of patients reporting TRAE, n (%), 35 (100)\tAll patients (N = 35)\n",
      "Grade 1/2\tGrade 3\tGrade 4\tAll grades\n",
      "Anemia\t1 (3)\t2 (6)\t0 (0)\t3 (9)\n",
      "Hypertension\t17 (48)\t5 (14)\t0 (0)\t22 (63)\n",
      "Diarrhea\t18 (51)\t0 (0)0 (0)\t18 (51)\n",
      "Fatigue\t13 (38)\t2 (6)\t0 (0)\t15 (43)\n",
      "Headache\t11 (31)\t0 (0)0 (0)\t11 (31)\n",
      "Weight decreased\t11 (31)\t0 (0)0 (0)\t11 (31)\n",
      "Rash\t10 (29)\t0 (0)0 (0)\t10 (29)\n",
      "Anorexia\t5 (14)\t4 (11)\t0 (0)\t9 (26)\n",
      "Nausea\t8 (23)\t1 (3)\t0 (0)\t9 (26)\n",
      "Blood thyroid-stimulating hormone increased\t8 (23)\t0 (0)0 (0)\t8 (23)\n",
      "Dysphonia\t8 (23)\t0 (0)0 (0)\t8 (23)\n",
      "Protein urine present\t6 (17)\t1 (3)\t0 (0)\t7 (20)\n",
      "Dry skin\t6 (17)\t0 (0)0 (0)\t6 (17)\n",
      "Vomiting\t5 (15)\t1 (3)\t0 (0)\t6 (17)\n",
      "\n",
      "Previous motesanib studies have reported an increased occurrence of cholecystitis in patients receiving motesanib, specifically at a dose of 75 mg twice daily continuously. The etiology of cholecystitis observed in patients receiving motesanib is unknown. Cholecystitis was not reported in this study. Gallbladder disorder was reported in three patients, specifically extended gallbladder or wall thickening, which was incidentally discovered in these patients on ultrasound sonography (US). The patients had not undergone US before starting motesanib treatment, nor were these disorders detected on routine CT scanning.\n",
      "\n",
      "Figure 3 and Table 4 summarize the results of the intensive PK analyses. After a single-dose oral administration of 125 mg on day 1, motesanib was rapidly absorbed, with an overall median tmax of 0.75 h; a similar median tmax value (0.79 h) was observed after daily administration of motesanib on day 29. The mean Cmax, AUC0–24, and C24 were slightly lower on day 29 than on day 1, indicating that there was no accumulation after daily administration. The day 29 to day 1 mean ratios were 0.62, 0.71, and 0.80 for Cmax, AUC0–24, and C24, respectively, for all evaluable patients.\n",
      "\n",
      "Fig. 3 Mean concentration–time profiles after oral administration of 125 mg of motesanib on day 1 in patients without gastrectomy and in patients with partial or full gastrectomy\n",
      "\n",
      "Table 4\n",
      "PK parameter\tN\tDay 1Mean ± SD\tDay 29Mean ± SD\tDay 29:Day 1 ratioMean ± SD\n",
      "All evaluable patients\n",
      " tmax (h)a\t10\t0.75 (0.25–2.0)\t0.79 (0.50–4.0)\tNA\n",
      " Cmax (ng/ml)\t10\t800 ± 439\t488 ± 363\t0.62 ± 0.20\n",
      " AUC0–24 (μg h/ml)\t9\t3.87 ± 2.28\t2.51 ± 2.10\t0.71 ± 0.32\n",
      " AUC0–inf (μg h/ml)\t9\t4.14 ± 2.47\tNANA\n",
      " t1/2,z (h)\t8\t5.42 ± 1.51\t4.27 ± 1.26\tNA\n",
      " CL/F (l/h)\t9\t41.1 ± 22.3\t69.3 ± 31.8\tNA\n",
      " C24 (ng/ml)\t9\t27.6 ± 23.8\t12.9 ± 15.4\t0.80b ± 1.17\n",
      "Evaluable patients with no prior gastrectomy\n",
      " tmax (h)a\t6\t1.0 (0.25–2.0)\t1.0 (0.50–4.0)\tNA\n",
      " Cmax (ng/ml)\t6\t692 ± 312\t354 ± 193\t0.53 ± 0.16\n",
      " AUC0–24 (μg h/ml)\t5\t3.91 ± 2.43\t1.93 ± 0.67\t0.67 ± 0.39\n",
      " AUC0–inf (μg h/ml)\t5\t4.27 ± 2.73\tNANA\n",
      " t1/2,z (h)\t4\t5.20 ± 1.79\t4.32 ± 1.89\tNA\n",
      " CL/F (l/h)\t5\t40.7 ± 24.5\t71.5 ± 26.0\tNA\n",
      " C24 (ng/ml)\t5\t33.5 ± 31.1\t16.6 ± 20.3\t1.11b ± 1.57\n",
      "Evaluable patients with partial or full gastrectomy\n",
      " tmax (h)a\t4\t0.50 (0.25–2.0)\t0.50 (0.50–1.0)\tNA\n",
      " Cmax (ng/ml)\t4\t962 ± 599\t689 ± 492\t0.75 ± 0.21\n",
      " AUC0–24 (μg h/ml)\t4\t3.82 ± 2.45\t3.23 ± 3.16\t0.75 ± 0.25\n",
      " AUC0–inf (μg h/ml)\t4\t3.99 ± 2.49\tNANA\n",
      " t1/2,z (h)\t4\t5.63 ± 1.42\t4.22 ± 0.38\tNA\n",
      " CL/F (l/h)\t4\t41.6 ± 22.9\t66.4 ± 42.2\tNA\n",
      " C24 (ng/ml)\t4\t20.2 ± 9.2\t8.37 ± 6.24\t0.40 ± 0.16\n",
      "\n",
      "For the patients who had partial or full gastrectomy (n = 4), day 1 Cmax values were slightly higher (<2-fold) and AUC values were similar to those who had no gastrectomy. Means for Cmax and AUC values on day 29 were higher compared with those who had no gastrectomy but not significant. Median tmax values occurred earlier in patients with gastrectomy on both days 1 and 29 (median tmax = 0.50 h with gastrectomy versus 1.0 h with no gastrectomy). C24 values on days 1 and 29 were lower in patients with gastrectomy compared with those who had no gastrectomy, though the mean t1/2 values were similar (mean t1/2,z value = 4.22 versus 4.32 h, respectively).\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "Although regression of thyroid cancer, renal cell carcinoma, and leiomyosarcoma was observed in the Phase I study of motesanib [12], objective tumor regression was observed in only one patient (3%) with GIST in this study. Motesanib administered as a single-agent was well tolerated, and a number of patients experienced prolonged stabilization of the disease. Seven (20%) did not exhibit disease progression for a minimum of 24 weeks, and the median PFS was 16.1 weeks. Serious hematological AEs (grade 3/4) were observed after sunitinib treatment in the Phase III trial. The incidence of grade 3/4 anemia, leukocytopenia, and neutrocytopenia was 4, 4, and 10%, respectively [9]. Anemia, leukocytopenia, and neutropenia were also reported as AEs for patients treated with imatinib; the incidence of grade 3/4 anemia, leukocytopenia, and neutropenia in a large Phase III study was 2.0, 1.4, and 4.8%, respectively [6]. The only hematological toxicity of motesanib was anemia (grade 2, 3%; grade 3, 6%). Despite long-term exposure to motesanib, hematological toxicities were mild. Motesanib may therefore present an alternative treatment option for patients who experienced neutrocytopenia or thrombocytopenia after treatment with imatinib or sunitinib.\n",
      "\n",
      "To evaluate the potential effect of gastrectomy on motesanib disposition, the motesanib PK profiles from patients with a history of gastrectomy were compared with those who did not have prior gastrectomy. Median tmax values occurred earlier in patients with gastrectomy on both days 1 and 29, suggesting faster absorption in patients with a history of gastrectomy. Since gastrectomy impacts the gastric emptying rate and the absorption rate resulting in increase in Cmax, these findings should be considered in the following clinical trials.\n",
      "\n",
      "The current treatment options for patients with GIST after treatment failure with imatinib and sunitinib are limited to best supportive care and investigative therapies. This study shows that in Japanese patients with advanced GIST motesanib is well tolerated, and, although an objective tumor response was observed in only one patient, motesanib may have an impact on survival in a retrospective analysis. However, focusing on other clinically meaningful measures, such as the Choi criteria [13] that incorporating tumor density and small changes in tumor size as revealed by CT scanning, is more important than focusing on the tumor response rate, which may fail to identify a potentially effective therapy [14, 15]. Randomized, well-controlled studies with time to progression or survival as the primary endpoints of efficacy will be needed to identify agents for which a tumor regression effect is not anticipated. Results from such studies will help in making an informed decision of whether or not to continue the clinical development of such agents in GIST.\n"
     ]
    }
   ],
   "source": [
    "view_one_article()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## View Articles at Different GeoName Densities"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "query = {\n",
    "    \"text_matches\": {\"$in\": terms},\n",
    "    \"article_meta.article_type\": \"research-article\",\n",
    "    \"article_meta.has_body\": True,\n",
    "    \"article_meta.n_geospans\": {\"$exists\": True}\n",
    "}\n",
    "\n",
    "cursor = articles.aggregate([\n",
    "    {\"$match\": query},\n",
    "    {\"$project\": {\n",
    "        \"n_geospans\": \"$article_meta.n_geospans\",\n",
    "        \"length\": {\"$strLenCP\": \"$extracted_text\"}\n",
    "    }}\n",
    "])\n",
    "\n",
    "subset_df = pd.DataFrame(cursor)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "deciles = subset_df.quantile(q=np.arange(0, 1.1, 0.1))\n",
    "quantiles = subset_df.quantile(q=[0.05, 0.25, 0.5, 0.75, 0.95])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>length</th>\n",
       "      <th>n_geospans</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0.0</th>\n",
       "      <td>3048.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.1</th>\n",
       "      <td>18614.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.2</th>\n",
       "      <td>22789.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.3</th>\n",
       "      <td>25835.0</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.4</th>\n",
       "      <td>28323.0</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.5</th>\n",
       "      <td>31158.0</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.6</th>\n",
       "      <td>34320.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.7</th>\n",
       "      <td>37611.0</td>\n",
       "      <td>43.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.8</th>\n",
       "      <td>42225.0</td>\n",
       "      <td>58.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.9</th>\n",
       "      <td>50307.0</td>\n",
       "      <td>90.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1.0</th>\n",
       "      <td>453774.0</td>\n",
       "      <td>2448.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       length  n_geospans\n",
       "0.0    3048.0         0.0\n",
       "0.1   18614.0         7.0\n",
       "0.2   22789.0        11.0\n",
       "0.3   25835.0        16.0\n",
       "0.4   28323.0        21.0\n",
       "0.5   31158.0        26.0\n",
       "0.6   34320.0        34.0\n",
       "0.7   37611.0        43.0\n",
       "0.8   42225.0        58.0\n",
       "0.9   50307.0        90.0\n",
       "1.0  453774.0      2448.0"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "deciles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>length</th>\n",
       "      <th>n_geospans</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0.05</th>\n",
       "      <td>15334.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.25</th>\n",
       "      <td>24279.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.50</th>\n",
       "      <td>31158.0</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.75</th>\n",
       "      <td>39655.0</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.95</th>\n",
       "      <td>56676.0</td>\n",
       "      <td>128.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       length  n_geospans\n",
       "0.05  15334.0         4.0\n",
       "0.25  24279.0        14.0\n",
       "0.50  31158.0        26.0\n",
       "0.75  39655.0        50.0\n",
       "0.95  56676.0       128.0"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "quantiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>length</th>\n",
       "      <th>n_geospans</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0.01</th>\n",
       "      <td>10097.2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.25</th>\n",
       "      <td>24279.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.95</th>\n",
       "      <td>56676.0</td>\n",
       "      <td>128.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.99</th>\n",
       "      <td>77687.0</td>\n",
       "      <td>234.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       length  n_geospans\n",
       "0.01  10097.2         1.0\n",
       "0.25  24279.0        14.0\n",
       "0.95  56676.0       128.0\n",
       "0.99  77687.0       234.2"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "quantiles = subset_df.quantile(q=[0.01, 0.25, 0.95, 0.99])\n",
    "quantiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "lower_break = 9\n",
    "upper_break = 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "128.0"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "quantiles.n_geospans.iloc[4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "geospan_query = {\n",
    "    \"text_matches\": {\"$in\": terms},\n",
    "    \"article_meta.article_type\": \"research-article\",\n",
    "    \"article_meta.has_body\": True,\n",
    "    \"article_meta.n_geospans\": {\"$gte\": quantiles.n_geospans.iloc[4],\n",
    "                                \"$lte\": deciles.n_geospans.iloc[upper_break], }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "5678892\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "Phylogeny of xerophilic aspergilli (subgenus Aspergillus) and taxonomic revision of section Restricti\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "{'all_geospans': 695, 'n_geospans': 695, 'article_type': 'research-article', 'has_body': True}\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['sampling site']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Introduction\n",
      "\n",
      "\n",
      "Aspergillus section Restricti species occurs in environments with low water activity (aw). They are commonly found on building materials, in house dust (Kaarakainen et al., 2009, Visagie et al., 2014) and dried, salted or high sugar content foods (Pitt and Hocking, 2009, Frasz and Miller, 2015). Much attention is being paid to the indoor air quality (Kasuga, 2012, Flannigan et al., 2016) and these fungi are repeatedly reported to be present in this environment (Samson et al., 2002, Meklin et al., 2004, Meklin et al., 2007), where they are considered a potential agent responsible for sick building syndrome, respiratory problems and allergies (Terr, 2009, Saijo et al., 2011, Abe, 2012). Species may cause post-harvest rot in improperly dried commodities such as maize or wheat (Christensen & Kaufmann 1965), cotton goods are susceptible to A. restrictus rot (Smith 1931), while A. vitricola can even damage optical instruments (Ohtsuki 1962). Recently, fatal disseminated aspergillosis in an infant was proved to be caused by A. penicillioides (Gupta et al. 2016) and A. conicus was detected as the causal agent of an intraocular infection (Smith et al. 2013). While there are reports in the literature with infection cases from A. restrictus-like fungi, the patient nearly always has a documented underlying disease state, suggesting opportunistic infections (de Hoog et al. 2009).\n",
      "\n",
      "Secondary metabolite production in these species has not been studied extensively. According to Micheluz et al. (2016), the homothallic species A. halophilicus produces many metabolites (chaetoviridin A, deoxybrevianamid E, pseurotin A, pseurotin D, rugulusovin, stachybotryamide and tryprostatin B) compared to anamorphic species producing a much narrower spectrum of substances with only asperglaucide detected in A. penicillioides and some unspecified metabolites in A. vitricola (Micheluz et al. 2016). There are no known mycotoxins produced by members of sect. Restricti and they do not pose a direct threat to consumers, but they cause significant losses for food and agricultural industries (Deschuyffeleer et al. 2015). More worrying is their potential for creating more favourable conditions for less xerophilic fungal species that may produce hazardous mycotoxins. They do this by producing metabolic water, thereby increasing water activity of the substrate. From a biotechnological perspective, polyextremophilic α-amylase produced by A. penicillioides has significant potential for use as a detergent (Ali et al. 2015).\n",
      "\n",
      "The first species described from this section was A. penicillioides observed from Argentinian sugar cane (Spegazzini 1896). Aspergillus caesiellus was subsequently described from air in Tokyo (Saito 1904), A. gracilis from a Monilinia fructigena fruiting body and at the time thought to resemble A. fumigatus (Bainier 1907). Blochwitz described A. conicus from chalky soil (Dale 1914); Smith (1931) described A. restrictus causing degradation of cotton in the manufacturing proces; and finally, A. vitricola was described from binocular lens (Ohtsuki 1962). The taxonomy of Aspergillus was advanced greatly by the designation of type specimens and ex-type cultures by Samson & Gams (1985). Types for all aforementioned species in sect. Restricti were summarised or newly designated with the exception of A. vitricola (was not accepted by the authors). Pitt & Samson (1990) reduced the section to three species on the basis of physiology and morphology. Peterson (2000) used a phylogenetic analysis of LSU rDNA (28S) sequences to provide genetic evidence that there are seven species in the section, including A. halophilicus, which was described as Eurotium halophilicum from corn seeds by Christensen et al. (1959). Peterson (2008) used multilocus DNA sequence data, phylogenetic and concordance analysis to produce a statistically supported analysis of sect. Restricti containing seven species.\n",
      "\n",
      "The morphology of sect. Restricti species is very simple and the number of taxonomically relevant morphological characters is low. Correct identification based solely on morphology is therefore challenging if not impossible and sequence comparisons represent the best method for fast and robust identifications. Phylogenetic analysis based on multiple loci has become an indispensable part of taxonomic studies. A polyphasic approach to species delimitation is currently standard in Aspergillus (Samson and Varga, 2009, Samson et al., 2014) with the phylogenetic component usually relying on the genealogical concordance phylogenetic species recognition (GCPSR) approach proposed by Taylor et al. (2000).\n",
      "\n",
      "New and advanced multi-locus methods for species delimitation were introduced recently (Fujita et al., 2012, Tang et al., 2014, Fontaneto et al., 2015). The majority of new approaches and associated software is based on coalescent theory and multispecies coalescent model (Flot 2015). Simultaneously because of the possible incongruence between gene trees, the focus is also shifting from gene tree to species tree inference using methods, that take into account incomplete lineage sorting (ILS), the most common cause of locus incongruence (Edwards 2009). We followed the suggestion of Carstens et al. (2013) that unites the two mentioned tasks into one analysis, i.e., species delimitation and species tree estimation. Firstly, species are delimited from a set of individuals by several species delimitation methods, possibly based on different approaches (based on trees, genetic distances, haplowebs, etc.). Some methods or some loci may be more prone to over delimitation (i.e., the method splits the dataset into more potential species compared with other methods or loci) of species than others, so it is recommended to compare more methods with as much loci as possible (Carstens et al. 2013). The species tree is subsequently inferred with individuals assigned into putative species based on the results of species delimitation. These methods are currently being applied to many different groups of organisms (Flot 2015) but infrequently in fungi. In this study, we applied them for the first time to delimit species boundaries in Aspergillus.\n",
      "\n",
      "During identifications of isolates of sect. Restricti from various substrates and locations, we encountered many isolates that could not credibly be placed in the seven species accepted by Peterson (2008). In order to substantiate the initial finding we assembled a set of 193 isolates from sect. Restricti including type isolates, conducted DNA sequencing of four genetic loci, coalescence analysis, physiological testing (temperature and aw gradients), micro- and macromorphology and SEM (scanning electron microscopy) in order to describe the biodiversity found in this section. Additionally, the monophyly of section Restricti within subgenus Aspergillus was addressed.\n",
      "\n",
      "\n",
      "Materials and methods\n",
      "\n",
      "\n",
      "Fungal strains\n",
      "\n",
      "Strains used in this study were obtained from and deposited into various culture collections: 1) ATCC, American Type culture collection, Manassas, Virginia, USA (https://www.lgcstandards-atcc.org/en/Products/ATCC_Genuine_Cultures.aspx); 2) BCCM/IHEM, Biomedical Fungi and Yeasts Collection, Scientific Institute of Public Health, Brussels, Belgium (http://bccm.belspo.be/about-us/bccm-ihem); 3) CCF, Culture Collection of Fungi, Charles University, Prague, Czech Republic (https://www.natur.cuni.cz/biology/botany/structure/culture-collection-of-fungi-ccf); 4) CBS, culture collection of the Westerdijk Fungal Biodiversity Institute, Utrecht, the Netherlands (http://www.westerdijkinstitute.nl/Collections); 5) DAOMC, Canadian Collection of Fungal Cultures, Agriculture and Agri-Food Canada, Ottawa, Canada; 6) DTO, working collection of the Applied and Industrial Mycology department housed at the Westerdijk Fungal Biodiversity Institute, Utrecht, the Netherlands; 7) EMSL, EMSL Analytical Inc., New Jersey, USA (http://www.emsl.com); 8) EXF, Culture Collection of Extremophilic Fungi, University of Ljubljana, Slovenia (http://www.ex-genebank.com/index.php/en/fungi-2); 9) IBT, Culture Collection at Department of Biotechnology and beimedicine, Lyngby, Denmark; 10) IMI/CABI, International Mycological Institute, Kew, England (http://www.cabi.org/services/microbial-services); 11) KAS, fungal collection of Keith A. Seifert, Ottawa, Canada; 12) NRRL, Agricultural Research Service Culture Collection, Peoria, Illinois, USA (https://nrrl.ncaur.usda.gov); 13) UBOCC, Université de Bretagne Occidentale Culture Collection, Brest, France (https://www.univ-brest.fr/plateformes-technologiques/menu/nos-plates-formes/UBOCC); 14) MUT, Mycotheca dell'Università degli Studi di Torino, Turin, Italy, http://www.mut.unito.it/en/Collezione). Dried holotype and isotype specimens were deposited into the herbarium of the Mycological Department, National Museum, Prague, Czech Republic (PRM) or Canadian National Mycological Herbarium, Ottawa, Canada (DAOM).\n",
      "\n",
      "Many strains were specifically isolated for the purposes of this study from the indoor environment in the USA, Bermuda, Puerto Rico, Trinidad and Tobago (Ž. Jurjević), house dust from Canada (C.M. Visagie), bakery products and deteriorated paintings in France (M. Coton, F. Déniel), deteriorated paintings from Slovenia (P. Zalar, D.D. Graf) and hypersaline water from Slovenia (P. Zalar, N. Gunde-Cimerman) and Israel (R. Tkavc).\n",
      "\n",
      "Different isolation techniques were used for species isolation. Samples from indoor environments across the USA were collected using the following techniques: air samples were collected as detailed previously (Peterson & Jurjević 2013) using malt extract agar (MEA) as isolation medium. Dilution plates were used to isolate fungi that were taken by swabs as described previously (Jurjević et al. 2015), using MEA supplemented with chloramphenicol and dichloran-glycerol (DG18) agar as isolation medium. Sedimentation plate samples were taken for a one-hour exposure time using Potato Dextrose agar (PDA) as isolation medium. Isolations from Canadian dust were made using a modified dilution-to-extinction method (Collado et al. 2007) as described in (Visagie et al., 2014, Visagie et al., 2017) using DG18, Malt extract Yeast extract 10 % glucose 12 % NaCl (MY10-12) and Malt extract Yeast extract 50 % glucose agar (MY50G) as isolation media.\n",
      "\n",
      "The samples from paintings in Slovenia were collected as follows. The sampling was performed in 2014 on old (from 300 to 400 years) oil canvas paintings originating from various Slovene churches and at the time of sampling stored in the Restoration Centre for the Protection of Cultural Heritage of Slovenia (ZVKDS). The sampled paintings were partly visibly overgrown with fungi, on the front painted side or on the back of the canvas. Samples were collected by rubbing overgrown areas with sterile cotton swabs soaked in physiological solution [0.9 % (w/v) NaCl]. Inoculum was subsequently spread onto plates containing DG18, MY10-12 or MY50G, all media amended with chloramphenicol (50 mg/l). The plates were incubated at 25 °C for up to 21 d. Pure cultures of the fungi were obtained from the primary isolation plates by further subculturing. Direct isolations were made from fungal growth on deteriorated paintings and bakery products in France using salt malt medium [5 % malt extract, 5 % NaCl, 1.5 % agar, (w/v)] as isolation media. Dilution series were sometimes also used. The paintings were visibly damaged and had been stored in the painting storage area at the Musée des beaux Arts in Brest, France. Plates were incubated at 25 °C for up to 21 d.\n",
      "\n",
      "Several strains were isolated from different salterns in Slovenia and Israel using filtration of hypersaline water through membrane filters (pore diam 0.45 μm), followed by incubation of the filters on different cultivation media with lowered water activity, as reported by (Gunde-Cimerman et al. 2000).\n",
      "\n",
      "\n",
      "Molecular studies\n",
      "\n",
      "All isolates included in this study were identified using sequence data, but amplification of four genetic loci (see below) was performed only in 193 isolates selected for phylogenetic analyses (Table 1).\n",
      "\n",
      "Table 1\n",
      "Species\tStrain no.\tSource\tGenBank accession nos.\n",
      "ITS\tbenA\tCaM\tRPB2\n",
      "A. caesiellus\tNRRL 5061\tUnknown, Japan\tEF652044\tEF651884\tEF652030\tEF651981\n",
      "NRRL 25810 = CCF 5662\tCloth, Panama\tKY087751\tKY117814\tKY068301\tKY117992\n",
      "CCF 5450 = EMSL No. 1614\tAir, outside, Delaware, USA\tKY087598\tKY117667\tKY068151\tKY117844\n",
      "CCF 5448 = EMSL No. 1383 = IBT 34621\tAir, home, Pennsylvania, USA\tKY117665\tKY068149\tKY117842\n",
      "DTO 026-C7\tIndoor environment, Germany\tKY087684\tKY117748\tKY068232\tKY117925\n",
      "DTO 025-I4 = IBT 34538\tIndoor environment, Germany\tKY087683\tKY117747\tKY068231\tKY117924\n",
      "CCF 5451 = EMSL No. 1650 = IBT 34622\tAir, pineapple room, warehouse, Delaware, USA\tKY087599\tKY117668\tKY068153\tKY117845\n",
      "CCF 5449 = EMSL No. 1499\tAir, home, Delaware, USA\tKY087597\tKY117666\tKY068150\tKY117843\n",
      "NRRL 25861\tUnknown, Gorakpur, India\tKY087766\tKY117829\tKY068316\tKY118007\n",
      "NRRL 25815 = DTO 356-D1 = CCF 5663\tHobnail shoes, Florida, USA\tKY087753\tKY117816\tKY068303\tKY117994\n",
      "A. canadensis\tCCF 5548\tHouse dust, Nova Scotia, Wolfville, Canada\tKY087667\tKY117731\tKY068215\tKY117909\n",
      "KAS 7705 = DTO 357-A8\tHouse dust, Ottawa, Ontario, Canada\tKY087672\tKY117736\tKY068220\tKY117914\n",
      "KAS 7707 = DTO 357-B1\tHouse dust, Ottawa, Ontario, Canada\tKY087673\tKY117737\tKY068221\tKY117915\n",
      "KAS 7708 = DTO 357-B2 = CCF 5550 = IBT 34637\tHouse dust, Ottawa, Ontario, Canada\tKY087674\tKY117738\tKY068222\tKY117916\n",
      "KAS 7710 = DTO 357-B4 = CCF 5552 = IBT 34636\tHouse dust, Ottawa, Ontario, Canada\tKY087675\tKY117739\tKY068223\tKY117917\n",
      "KAS 7711 = DTO 357-B5\tHouse dust, Ottawa, Ontario, Canada\tKY087676\tKY117740\tKY068224\tKY117918\n",
      "KAS 7716 = DTO 357-B8\tHouse dust, Ottawa, Ontario, Canada\tKY087677\tKY117741\tKY068225\tKY117919\n",
      "KAS 7717 = DTO 357-B9\tHouse dust, Ottawa, Ontario, Canada\tKY087678\tKY117742\tKY068226\tKY117920\n",
      "KAS 7718 = DTO 357-C1\tHouse dust, Ottawa, Ontario, Canada\tKY087679\tKY117743\tKY068227\tKY117921\n",
      "KAS 7719 = DTO 357-C2 = CCF 5553 = IBT 34638\tHouse dust, Ottawa, Ontario, Canada\tKY087680\tKY117744\tKY068228\tKY117922\n",
      "KAS 7704 = DTO 357-A7\tHouse dust, Ottawa, Ontario, Canada\tKY087671\tKY117735\tKY068219\tKY117913\n",
      "KAS 7721 = DTO 357-C4\tHouse dust, Ottawa, Ontario, Canada\tKY087681\tKY117745\tKY068229\tKY117923\n",
      "A. clavatophorus\tNRRL 25874\tMouldy paper, Athens, Georgia, USA\tKY087772\tKY117836\tKY068323\tKY118014\n",
      "NRRL 25873 = CCF 5453 = IBT 34632\tMouldy paper, Athens, Georgia, USA\tKY117835\tKY068322\tKY118013\n",
      "DTO 257-G5 = IBT 34561 = CCF 5669\tPuerh tea, China\tKY087703\tKY117764\tKY068251\tKY117943\n",
      "A. conicus\tNRRL 149\tUnknown\tEF652039\tEF651881\tEF652033\tEF651975\n",
      "CCF 5458 = EMSL No. 1490\tAir, home, California, USA\tKY087601\tKY117670\tKY068155\tKY117846\n",
      "CCF 5457 = EMSL No. 1318 = NRRL 62007\tAir, home, Idaho, USA\tKY087600\tKY117669\tKY068154\n",
      "NRRL 25881\tUnknown, New York, USA\tKY087775\tKY117839\tKY068326\tKY118017\n",
      "CCF 5459 = EMSL No. 1649\tAir, pineapple room, warehouse, Delaware, USA\tKY087602\tKY117671\tKY068156\tKY117847\n",
      "CCF 5461 = EMSL No. 2549\tAir, office, Bayamon, Puerto Rico\tKY087604\tKY117673\tKY068157\tKY117849\n",
      "CCF 5460 = EMSL No. 2217\tAir, living room, West Chester, Pennsylvania, USA\tKY087603\tKY117672\tKY068152\tKY117848\n",
      "EXF-7663 = IBT 34267 = IBT 33574\tOil painting on canvas, Ljubljana, Slovenia\tKY087715\tKY117778\tKY068265\tKY117956\n",
      "DTO 077-H5\tIndoor air, Witten, The Netherlands\tKY087687\tKY117751\tKY068235\n",
      "IHEM 16531\tWooden statue, Braine-l'Alleud, Belgium\tKY087735\tKY117798\tKY068285\tKY117976\n",
      "CCF 4042\tKernel of \tKY087659\tKY117723\tKY068207\tKY117903\n",
      "EXF-7660 = IBT 34263 = IBT 33577\tOil painting on canvas, Ljubljana, Slovenia\tKY087713\tKY117776\tKY068263\tKY117954\n",
      "EXF-5015 = IBT 34273 = CCF 5650\tMicrobial mat, Eliat, Israel\tKY117774\tKY068261\tKY117952\n",
      "DTO 231-C3\tMuseum piece, Zwartewaal, The Netherlands\tKY087701\tKY117763\tKY068249\tKY117941\n",
      "NRRL 25848\tAsphalt roof shingle, Chicago, Illinois, USA\tKY087764\tKY117827\tKY068314\tKY118005\n",
      "IHEM 20709\tCandy, Belgium\tKY087736\tKY117799\tKY068286\tKY117977\n",
      "DTO 110-C5\tAir in bathroom, near Copenhagen, Denmark\tKY087690\tKY117754\tKY068238\tKY117930\n",
      "DTO 110-F5 = IBT 34534 = CCF 5667\tAir in living room, near Copenhagen, Denmark\tKY087691\tKY117755\tKY068239\tKY117931\n",
      "DTO 017-B9\tIndoor air, Eindhoven, The Netherlands\tKY087682\tKY117746\tKY068230\n",
      "A. destruens\tNRRL 145\tMaize seed, Maryland, USA\tKY087748\tKY117811\tKY068298\tKY117989\n",
      "EXF-7699 = IBT 34262\tOil painting on canvas, Ljubljana, Slovenia\tKY087717\tKY117780\tKY068267\tKY117958\n",
      "EXF-7651 = IBT 34258 = CCF 5653\tOil painting on canvas, Ljubljana, Slovenia\tKY087712\tKY117775\tKY068262\tKY117953\n",
      "EXF-7661 = IBT 34271 = IBT 33573\tOil painting on canvas, Ljubljana, Slovenia\tKY087714\tKY117777\tKY068264\tKY117955\n",
      "DTO 254-B2 = IBT 34522\tAir in villa, Utrecht, The Netherlands\tKY087702\tKY068250\tKY117942\n",
      "EXF-10411 = IBT 34265\tOil painting on canvas, Ljubljana, Slovenia\tKY087724\tKY117787\tKY068274\tKY117965\n",
      "EXF-7667 = IBT 34288\tOil painting on canvas, Ljubljana, Slovenia\tKY087716\tKY117779\tKY068266\tKY117957\n",
      "DTO 220-B2\tAir in bakery, Tilburg, The Netherlands\tKY087698\tKY117760\tKY068246\tKY117938\n",
      "DTO 161-B7 = CCF 5671\tSurface of cheese, The Netherlands\tKY087697\tKY117759\tKY068245\tKY117937\n",
      "NRRL 157 = CCF 5463\tUnknown, USA\tKY087749\tKY117812\tKY068299\tKY117990\n",
      "EXF-10407 = IBT 34285 = CCF 5652\tOil painting on canvas, Ljubljana, Slovenia\tKY087723\tKY117786\tKY068273\tKY117964\n",
      "EXF-7703 = IBT 34259\tOil painting on canvas, Ljubljana, Slovenia\tKY087718\tKY117781\tKY068268\tKY117959\n",
      "DTO 147-E2\tIndoor air, Hungary\tKY087696\tKY117758\tKY068244\tKY117936\n",
      "DTO 113-E7 = CCF 5668\tAir in bakery, Tilburg, The Netherlands\tKY087692\tKY117756\tKY068240\tKY117932\n",
      "A. domesticus\tDTO 079-F2\tWallpaper, Tiel, The Netherlands\tKY087688\tKY117752\tKY068236\tKY117928\n",
      "DTO 231-C1 = NRRL 66617 = CCF 5665\tMuseum piece, Zwartewaal, The Netherlands\tKY087700\tKY117762\tKY068248\tKY117940\n",
      "DTO 231-B9\tMuseum piece (mouldy chair backrest), Zwartewaal, The Netherlands\tKY087699\tKY117761\tKY068247\tKY117939\n",
      "DTO 086-D1 = CCF 5670\tArchive material, Gorinchem, The Netherlands\tKY087689\tKY117753\tKY068237\tKY117929\n",
      "IHEM 6549\tDust from mattress, Brussels, Belgium\tKY087734\tKY117797\tKY068284\tKY117975\n",
      "EXF-10012 = IBT 34274\tStatue made of wood, Textile and sea shells, Ljubljana, Slovenia\tKY087719\tKY117782\tKY068269\tKY117960\n",
      "UBOCC-A-116022 = CCF 5465\tPainting, Brittany, France\tKY087605\tKY117674\tKY068158\tKY117850\n",
      "CCF 5466 = EMSL No. 1316\tAir, home, Idaho, USA\tKY087606\tKY117675\tKY068159\tKY117851\n",
      "A. glabripes\tCCF 5474\tOffice folder, Macoya, Trinidad & Tobago\tKY087614\tKY117683\tKY068166\tKY117859\n",
      "CCF 5473 = EMSL No. 2442 = IBT 34626\tGreen fabric covered binders, import from China, New Jersey, USA\tKY087613\tKY117682\tKY068165\tKY117858\n",
      "CCF 5475 = EMSL No. 2463\tOffice folder, Macoya, Trinidad & Tobago\tKY087615\tKY117684\tKY068167\tKY117860\n",
      "CCF 5476 = EMSL No. 2464\tOffice folder, Macoya, Trinidad & Tobago\tKY087616\tKY117685\tKY068168\tKY117861\n",
      "CCF 5477 = EMSL No. 2465\tOffice folder, Macoya, Trinidad & Tobago\tKY087617\tKY117686\tKY068169\tKY117862\n",
      "CCF 5469 = EMSL No. 1483 = DTO 356-E5 = IBT 34519 = IBT 34821 = NRRL 66619\tAir, home, California, USA\tKY087609\tKY117678\tKY068161\tKY117854\n",
      "EMSL No. 1812 = CCF 5470\tFront cover of log book, library, Louisiana, USA\tKY087610\tKY117679\tKY068162\tKY117855\n",
      "EMSL No. 1813 = CCF 5471\tBook, library, Louisiana, USA\tKY087611\tKY117680\tKY068163\tKY117856\n",
      "EMSL No. 1317 = CCF 5468\tAir, home, Idaho, USA\tKY087608\tKY117677\tKY068160\tKY117853\n",
      "EMSL No. 2305 = CCF 5472\tAir, kitchen, Summerville, South Carolina, USA\tKY087612\tKY117681\tKY068164\tKY117857\n",
      "UBOCC-A-116021 = CCF 5467 = IBT 34625\tPainting, Brittany, France\tKY087607\tKY117676\tKY117852\n",
      "A. gracilis\tNRRL 4962\tGun-firing mechanism, South Pacific\tEF652045\tEF651883\tEF652031\tEF651980\n",
      "CCF 5479 = EMSL No. 2775 = DTO 356-F4 = IBT 34559\tChild carrier, San Diego, California, USA\tKY087618\tKY117687\tKY068170\tKY117863\n",
      "CCF 5480 = EMSL No. 2920\tChild carrier, San Diego, California, USA\tKY087619\tKY117688\tKY068171\tKY117864\n",
      "CCF 5481 = EMSL No. 2922\tChild carrier, San Diego, California, USA\tKY087620\tKY117689\tKY068172\tKY117865\n",
      "CCF 5482 = EMSL No. 2923 = IBT 34623\tChild carrier, San Diego, California, USA\tKY087621\tKY117690\tKY068173\tKY117866\n",
      "A. halophilicus\tNRRL 2739\tDried corn, St. Paul, Minnesota, USA\tEF652088\tEF651926\tEF652034\tEF651982\n",
      "DTO 271-F4 = CCF 5825 = IBT 34881\tTextile, imported into the Netherlands\tKY087705\tKY117766\tKY068253\tKY117945\n",
      "A. hordei\tNRRL 25825\tBarley, St. Paul, Minnesota, USA\tKY087759\tKY117822\tKY068309\tKY118000\n",
      "NRRL 25826 = CCF 5484\tBarley, St. Paul, Minnesota, USA\tKY087760\tKY117823\tKY068310\tKY118001\n",
      "NRRL 25830 = CCF 5485 = IBT 34631\tInsulating board, St. Paul, Minnesota, USA\tKY087761\tKY117824\tKY068311\tKY118002\n",
      "A. infrequens\tNRRL 25868\tWheat, Peoria, Illionois, USA\tKY087770\tKY117833\tKY068320\tKY118011\n",
      "A. magnivesiculatus\tNRRL 25866\tKatsuobushi, Tokyo, Japan\tKY087768\tKY117831\tKY068318\tKY118009\n",
      "CCF 5491 = EMSL No. 2918\tChild carrier, San Diego, California, USA\tKY087624\tKY117692\tKY068176\tKY117869\n",
      "CCF 5489 = EMSL No. 1315 = DTO 356-E2 = IBT 34516\tAir, home, Idaho, USA\tKY087622\tKY117691\tKY068174\tKY117867\n",
      "CCF 5490 = EMSL No. 2741\tChild carrier, San Diego, California, USA\tKY087623\tKY068175\tKY117868\n",
      "NRRL 25867 = CCF 5660\tKatsuobushi, Tokyo, Japan\tKY087769\tKY117832\tKY068319\tKY118010\n",
      "NRRL 25821 = CCF 5487\tDried corn, St. Paul, Minnesota, USA\tKY087756\tKY117819\tKY068306\tKY117997\n",
      "EXF-10353 = IBT 34284\tOil painting on canvas, Ljubljana, Slovenia\tKY087720\tKY117783\tKY068270\tKY117961\n",
      "KAS 5655 = DTO 356-G8\tHouse dust, Ottawa, Ontario, Canada\tKY087661\tKY117725\tKY068209\tKY117905\n",
      "KAS 5754 = DTO 356-G9\tHouse dust, Ottawa, Ontario, Canada\tKY087662\tKY117726\tKY068210\n",
      "EXF-10377\tOil painting on canvas, Ljubljana, Slovenia\tKY087721\tKY117784\tKY068271\tKY117962\n",
      "KAS 5623 = DTO 356-G7\tHouse dust, Stittsville, Ontario, Canada\tKY087660\tKY117724\tKY068208\tKY117904\n",
      "KAS 6089 = DTO 356-H3\tHouse dust, Wolfville, Nova Scotia, Canada\tKY087664\tKY117728\tKY068212\tKY117907\n",
      "A. pachycaulis\tNRRL 25824\tUnknown, Washington, District of Columbia, USA\tKY087758\tKY117821\tKY068308\tKY117999\n",
      "CCF 5493 = EMSL No. 2310 = DTO 356-E6 = IBT 34536\tAir, home, California, USA\tKY087625\tKY117693\tKY068177\tKY117870\n",
      "A. penicillioides\tNRRL 4548\tHuman skin, Recife, Brazil\tEF652036\tEF651928\tEF652024\tEF651930\n",
      "CCF 5497 = EMSL No. 2430 = IBT 34628\tGreen fabric covered binders, import from China, New Jersey, USA\tKY087626\tKY117694\tKY068178\tKY117871\n",
      "IHEM 2330\tSeeds of cereal, France\tKY087730\tKY117793\tKY068280\tKY117971\n",
      "IHEM 2476\tIndoor air, Brussels, Belgium\tKY087732\tKY117795\tKY068282\tKY117973\n",
      "DTO 281-A7\tLeather, imported into the Netherlands\tKY087706\tKY117767\tKY068254\tKY117946\n",
      "CCF 3282\tSweet roll with chocolate glaze, Prague, Czech Republic\tKY087657\tFR775347\tHE578103\tKY117902\n",
      "CBS 140430 = UBOCC-A-115042 = DTO 334-E1\tFrench madeleines, France\tKY087596\tKY117664\tKY068148\tKY117841\n",
      "CCF 5500 = EMSL No. 2651 = IBT 34630\tBaseball gloves, store, O'fallon, Illinois, USA\tKY087629\tKY117697\tKY068181\tKY117874\n",
      "CCF 5503 = EMSL No. 2909\tChild carrier, San Diego, California, USA\tKY087632\tKY117700\tKY068184\tKY117877\n",
      "NRRL 25816 = CCF 5661\tUnknown, Durham, North Carolina, USA\tKY087754\tKY117817\tKY068304\tKY117995\n",
      "NRRL 25834 = CCF 5659\tPeas, St. Paul, Minnesota, USA\tKY087762\tKY117825\tKY068312\tKY118003\n",
      "NRRL 25835\tWheat, St. Paul, Minnesota, USA\tKY087763\tKY117826\tKY068313\tKY118004\n",
      "KAS 7745\tHouse dust, Ottawa, Ontario, Canada\tKY087746\tKY117809\tKY068296\tKY117987\n",
      "KAS 7746\tHouse dust, Ottawa, Ontario, Canada\tKY087747\tKY117810\tKY068297\tKY117988\n",
      "DTO 267-A9 = CCF 5664\tHouse dust, Micronesia\tKY087704\tKY117765\tKY068252\tKY117944\n",
      "CCF 2666\tLeather shoe, Zlín, Czech republic\tKY087655\tHE578081\tHE578102\tKY117900\n",
      "CCF 5501 = EMSL No. 2749 = IBT 34629\tChild carrier, San Diego, California, USA\tKY087630\tKY117698\tKY068182\tKY117875\n",
      "CCF 5504 = EMSL No. 3264\tArchival cabinet, Bethesda, Maryland, USA\tKY087633\tKY117701\tKY068185\tKY117878\n",
      "NRRL 25870\tUnknown\tKY087771\tKY117834\tKY068321\tKY118012\n",
      "NRRL 25879\tBlood sample, New York, USA\tKY087774\tKY117838\tKY068325\tKY118016\n",
      "CCF 5499 = EMSL No. 2441\tGreen fabric covered binders, import from China, New Jersey, USA\tKY087628\tKY117696\tKY068180\tKY117873\n",
      "NRRL 4550 = CCF 5495\tHuman skin, Recife, Brazil\tEF652037\tEF651929\tEF652025\tEF651931\n",
      "NRRL 4553 = CCF 5496\tHuman skin, Recife, Brazil\tKY087750\tKY117813\tKY068300\tKY117991\n",
      "CCF 5498 = EMSL No. 2440 = DTO 356-E7 = IBT 34815\tGreen fabric covered binders, import from China, New Jersey, USA\tKY087627\tKY117695\tKY068179\tKY117872\n",
      "CCF 5502 = EMSL No. 2900\tChild carrier, San Diego, California, USA\tKY087631\tKY117699\tKY068183\tKY117876\n",
      "NRRL 25820\tDried corn, St. Paul, Minnesota, USA\tKY087755\tKY117818\tKY068305\tKY117996\n",
      "NRRL 25822\tDried corn, St. Paul, Minnesota, USA\tKY087757\tKY117820\tKY068307\tKY117998\n",
      "A. pseudogracilis\tCCF 5505\tChild carrier, San Diego, California, USA\tKY087634\tKY117702\tKY068186\tKY117879\n",
      "A. restrictus\tNRRL 154\tCloth, United Kingdom\tEF652042\tEF651880\tEF652029\tEF651978\n",
      "CCF 5511 = EMSL No. 1675 = IBT 34616\tPacking material, Maryland, USA\tKY087639\tKY117707\tKY068191\tKY117884\n",
      "NRRL 25862\tCulture contaminant, Peoria, Illinois, USA\tKY087767\tKY117830\tKY068317\tKY118008\n",
      "IHEM 2121\tDust from mattress, Antwerp, Belgium\tKY087729\tKY117792\tKY068279\tKY117970\n",
      "CCF 5512 = EMSL No. 2206 = IBT 34617\tAir, auditorium, school, Sicklerville, New Jersey, USA\tKY087640\tKY117708\tKY068192\tKY117885\n",
      "CCF 5513 = EMSL No. 2429\tGreen fabric covered binders, import from China, New Jersey, USA\tKY087641\tKY117709\tKY068193\tKY117886\n",
      "CCF 5509 = EMSL No. 1611\tMattress cover, North Carolina, USA\tKY087637\tKY117705\tKY068189\tKY117882\n",
      "DTO 065-C7 = IBT 34541\tCorn kernels, Indonesia\tKY087685\tKY117749\tKY068233\tKY117926\n",
      "CCF 3364 = IBT 34619\tSclerotium of fungus \tKY087658\tFR775348\tHE578101\tHE578109\n",
      "CCF 5514 = EMSL No. 2652\tBaseball gloves, store, O'fallon, Illinois, USA\tKY087642\tKY117710\tKY068194\tKY117887\n",
      "NRRL 25882\tCattle feed, USA\tKY087776\tKY117840\tKY068327\tKY118018\n",
      "CCF 5515 = EMSL No. 2906\tChild carrier, San Diego, California, USA\tKY087643\tKY117711\tKY068195\tKY117888\n",
      "IHEM 818\tIndoor air, Estinnes-au-Mont, Belgium\tKY087728\tKY117791\tKY068278\tKY117969\n",
      "CCF 5510 = EMSL No. 1633\tAir, hospital, New Jersey, USA\tKY087638\tKY117706\tKY068190\tKY117883\n",
      "CCF 5508 = EMSL No. 1416\tAir, home, Alabama, USA\tKY087636\tKY117704\tKY068188\tKY117881\n",
      "CCF 5507 = EMSL No. 1379 = IBT 34618\tAir, home, Bermuda\tKY087635\tKY117703\tKY068187\tKY117880\n",
      "IHEM 2477\tIndoor air, Brussels, Belgium\tKY087733\tKY117796\tKY068283\tKY117974\n",
      "IHEM 2373\tIndoor air, Brussels, Belgium\tKY087731\tKY117794\tKY068281\tKY117972\n",
      "A. reticulatus\tNRRL 25852\tLung biopsy, Charleston, South Carolina, USA\tKY087765\tKY117828\tKY068315\tKY118006\n",
      "CCF 5523 = EMSL No. 2526\tAir, administrative area, Bayamon, Puerto Rico\tKY087649\tKY117717\tKY068201\tKY117894\n",
      "CCF 5524 = EMSL No. 2548 = IBT 34637\tAir, office, Bayamon, Puerto Rico\tKY087650\tKY117718\tKY068202\tKY117895\n",
      "CCF 5518 = EMSL No. 1272 = NRRL 58903 = IBT 34819\tAir, home, Idaho, USA\tKY087644\tKY117712\tKY068196\tKY117889\n",
      "CCF 5519 = EMSL No. 1313 = NRRL 62004\tAir, home, Idaho, USA\tKY087645\tKY117713\tKY068197\tKY117890\n",
      "CCF 5520 = EMSL No. 1314 = NRRL 62005\tAir, home, Idaho, USA\tKY087646\tKY117714\tKY068198\tKY117891\n",
      "CCF 5521 = EMSL No. 1362 = DTO 356-E4 = IBT 34518\tAir, outside, Idaho, USA\tKY087647\tKY117715\tKY068199\tKY117892\n",
      "CCF 5525 = EMSL No. 885 = NRRL 58572 = IBT 34880\tAir, home, Florida, USA\tKY087651\tKY117719\tKY068203\tKY117896\n",
      "CCF 5522 = EMSL No. 2525\tAir, administrative area, Bayamon, Puerto Rico\tKY087648\tKY117716\tKY068200\tKY117893\n",
      "IHEM 22696\tDust from carpet, Brussels, Belgium\tKY087737\tKY117800\tKY068287\tKY117978\n",
      "EXF-10429 = CCF 5656\tOil painting on canvas, Ljubljana, Slovenia\tKY087725\tKY117788\tKY068275\tKY117966\n",
      "CCF 3112 = IBT 34634 = NRRL 62490\tLeather shoe, Zlín, Czech Republic\tKY087656\tFR775323\tFR751451\tKY117901\n",
      "NRRL 25878 = CCF 5517\tLung biopsy, Chamblee, Georgia, USA\tKY087773\tKY117837\tKY068324\tKY118015\n",
      "A. salinicola\tEXF-10401\tOil painting on canvas, Ljubljana, Slovenia\tKY087722\tKY117785\tKY068272\tKY117963\n",
      "KAS 6054\tHouse dust, Wolfville, Nova Scotia, Canada\tKY087738\tKY117801\tKY068288\tKY117979\n",
      "UBOCC-A-116019 = CCF 5528 = IBT 34635\tPainting, Brittany, France\tKY087652\tKY117720\tKY068204\tKY117897\n",
      "EXF-226 = CCF 5527 = IBT 34277 = NRRL 66622\tHypersaline water from salterns, Sečovlje salterns, Slovenia\tKY087711\tKY117773\tKY068260\n",
      "A. tardicrescens\tDTO 316-B5\tMuseum piece (measuring table), Alphen aan den Rijn, The Netherlands\tKY087710\tKY117772\tKY068259\tKY117951\n",
      "DTO 316-A7\tMuseum piece (dentist chair), Alphen aan den Rijn, The Netherlands\tKY087707\tKY117768\tKY068255\tKY117947\n",
      "DTO 316-A8 = IBT 34562\tMuseum piece (rubber tyre of brancard), Alphen aan den Rijn, The Netherlands\tKY087708\tKY117769\tKY068256\tKY117948\n",
      "DTO 316-A9\tMuseum piece (x-ray table), Alphen aan den Rijn, The Netherlands\tKY117770\tKY068257\tKY117949\n",
      "DTO 316-B4\tMuseum piece (vitrine), Alphen aan den Rijn, The Netherlands\tKY087709\tKY117771\tKY068258\tKY117950\n",
      "EXF-10456 = IBT 34286\tAir in depot of Conservation Centre of the Institute for the Protection of Cultrural Heritage of Slovenia, Ljubljana, Slovenia\tKY087727\tKY117790\tKY068277\tKY117968\n",
      "KAS 6252 = DTO 356-I5\tHouse dust, Wolfville, Nova Scotia, Canada\tKY087669\tKY117733\tKY068217\tKY117911\n",
      "EXF-10454 = IBT 34281 = CCF 5530 = NRRL 66624\tOil painting on canvas, Ljubljana, Slovenia\tKY087726\tKY117789\tKY068276\tKY117967\n",
      "DTO 073-H6\tMoist wall of archive, Tilburg, The Netherlands\tKY087686\tKY117750\tKY068234\tKY117927\n",
      "A. villosus\tNRRL 25813\tUnknown, Kirkhill, Scotland, United Kingdom\tKY087752\tKY117815\tKY068302\tKY117993\n",
      "UBOCC-A-116020 = CCF 5532 = IBT 34624\tPainting, Brittany, France\tKY087653\tKY117721\tKY068205\tKY117898\n",
      "A. vitricola\tNRRL 5125\tBinocular lens, Japan\tEF652046\tEF651927\tEF652035\tEF651973\n",
      "KAS 6086\tHouse dust, Little Lepreau, New Brunswick, Canada\tKY087739\tKY117802\tKY068289\tKY117980\n",
      "KAS 6281\tHouse dust, Victoria, British Columbia, Canada\tKY087744\tKY117807\tKY068294\tKY117985\n",
      "KAS 6087 = DTO 356-H2 = IBT 34532\tHouse dust, Victoria, British Columbia, Canada\tKY087663\tKY117727\tKY068211\tKY117906\n",
      "KAS 6238\tHouse dust, Victoria, British Columbia, Canada\tKY087743\tKY117806\tKY068293\tKY117984\n",
      "KAS 6199\tHouse dust, Victoria, British Columbia, Canada\tKY087742\tKY117805\tKY068292\tKY117983\n",
      "KAS 6288\tHouse dust, Wolfville, Nova Scotia, Canada\tKY087745\tKY117808\tKY068295\tKY117986\n",
      "KAS 6237 = DTO 356-I2\tHouse dust, Victoria, British Columbia, Canada\tKY087668\tKY117732\tKY068216\tKY117910\n",
      "KAS 6133 = DAOMC 251500\tHouse dust, Little Lepreau, New Brunswick, Canada\tKY087741\tKY117804\tKY068291\tKY117982\n",
      "KAS 6093\tHouse dust, Victoria, British Columbia, Canada\tKY087740\tKY117803\tKY068290\tKY117981\n",
      "DTO 122-I4\tArchive material, Gorinchem, The Netherlands\tKY087693\tKY117757\tKY068241\tKY117933\n",
      "DTO 123-B2\tArchive material, Gorinchem, The Netherlands\tKY087695\tKY068243\tKY117935\n",
      "DTO 122-I5\tArchive material, Gorinchem, The Netherlands\tKY087694\tKY068242\tKY117934\n",
      "KAS 6278 = DTO 356-I8\tHouse dust, Wolfville, Nova Scotia, Canada\tKY087670\tKY117734\tKY068218\tKY117912\n",
      "KAS 6150 = DTO 356-H5 = IBT 34531\tHouse dust, Wolfville, Nova Scotia, Canada\tKY087665\tKY117729\tKY068213\n",
      "KAS 6178 = DTO 356-H8\tHouse dust, Little Lepreau, New Brunswick, Canada\tKY087666\tKY117730\tKY068214\tKY117908\n",
      "CCF 5534 = EMSL No. 2785\tChild carrier, San Diego, California, USA\tKY087654\tKY117722\tKY068206\tKY117899\n",
      "\n",
      "1\n",
      "\n",
      "2\n",
      "\n",
      "ArchivePure DNA yeast and Gram2+ kit (5 PRIME Inc., Gaithersburg, MD) were used for DNA isolation from 14 d old cultures according to manufacturer instructions as updated by Hubka et al. (2015). Target genetic loci, including ITS + LSU rDNA (ID region), partial genes encoding calmodulin (CaM), β-tubulin (benA) and the second largest subunit of RNA polymerase II (RPB2), were amplified using primer combinations listed in Table 2. Amplification of RPB2 with the widely used primers (fRPB2-5F, fRPB2-7CR) designed by Liu et al. (1999) was problematic for many sect. Restricti isolates. Hence, new primers specific for section Restricti (fRPB2ResF100, fRPB2ResR950, Table 2) were designed based on the alignment of available sequences obtained with the Liu et al. (1999) primer set. Quality control (hairpin, self-dimer or hetero-dimer formation, melting temperature mismatch) was performed in OligoAnalyzer v. 3.1 (available online http://eu.idtdna.com/calc/analyzer). Standard and touchdown PCR protocols were described previously (Hubka et al., 2014, Hubka et al., 2016). PCR product purification followed Réblová et al. (2016). Automated sequencing was performed at Macrogen Sequencing Service (Amsterdam, The Netherlands) using both terminal primers. Sequences were deposited into GenBank with accession numbers shown in Table 1. All alignments are available from the Dryad Digital Repository: https://doi.org/10.5061/dryad.3t423.\n",
      "\n",
      "Table 2\n",
      "Locus\tPrimer\tOrientation\tSequence (from 5′ to 3′)\tReferences\n",
      "benA\tBt2a\tForward\tGGTAACCAAATCGGTGCTGCTTTC\tGlass & Donaldson (1995)\n",
      "T10\tForward\tACGATAGGTTCACCTCCAGAC\tO'Donnell & Cigelnik (1997)\n",
      "Bt2b\tReverse\tACCCTCAGTGTAGTGACCCTTGGC\tGlass & Donaldson (1995)\n",
      "CaM\tCF1L\tForward\tGCCGACTCTTTGACYGARGAR\tPeterson (2008)\n",
      "CF1M\tForward\tAGGCCGAYTCTYTGACYGA\tPeterson (2008)\n",
      "cmd5\tForward\tCCGAGTACAAGGAGGCCTTC\tHong \n",
      "CF4\tReverse\tTTTYTGCATCATRAGYTGGAC\tPeterson (2008)\n",
      "cmd6\tReverse\tTTTYTGCATCATRAGYTGGAC\tHong \n",
      "RPB2\tfRPB2-5F\tForward\tGAYGAYMGWGATCAYTTYGG\tLiu \n",
      "fRPB2ResF100\tForward\tTGAARTAYGCICTTGCYAC\tNewly designed\n",
      "fRPB2-7CR\tReverse\tCCCATRGCTTGYTTRCCCAT\tLiu \n",
      "fRPB2ResR950\tReverse\tCARTGYGTCCADGTRTGKGC\tNewly designed\n",
      "ITS\tITS1\tForward\tTCCGTAGGTGAACCTGCGG\tWhite \n",
      "NL4\tReverse\tGGTCCGTGTTTCAAGACGG\tO'Donnell (1993)\n",
      "ITS4\tReverse\tTCCTCCGCTTATTGATATGC\tWhite \n",
      "\n",
      "\n",
      "Phylogenetic analysis\n",
      "\n",
      "Sequences were inspected in FinchTV (available online http://www.geospiza.com/Products/finchtv.shtml) and assembled in Bioedit v. 7.2.5 (Hall 1999). Alignments were performed using the G-INS-i strategy, as implemented in MAFFT v. 7 (Katoh & Standley 2013). The benA alignment contained 431 characters with 227 variable and 217 parsimony informative sites, CaM 652 characters with 288 variable and 268 parsimony informative sites, RPB2 819 characters with 272 variable and 249 parsimony informative sites, ID 1 123 characters with 231 variable and 175 parsimony informative sites. The concatenated alignment contained 3 025 characters, with 1 018 variable and 909 parsimony informative sites.\n",
      "\n",
      "Phylogenetic trees based on the concatenated dataset were inferred with both Maximum likelihood (ML) and Bayesian inference (BI) analysis. Partitioning scheme and substitution models for analyses were selected using PartitionFinder v. 1.1.1 (Lanfear et al. 2012) with settings allowing introns, exons and codon positions to be independent datasets. Proposed partitioning schemes for each dataset are listed in Table 3. Hamigera avellanea NRRL 1938 was used as outgroup.\n",
      "\n",
      "Table 3\n",
      "Dataset\tPhylogenetic method\tPartitioning scheme (substitution model)\n",
      "Sect. \tML\tbenA\n",
      "BI\tbenA\n",
      "Subg. \tML\tbenA\n",
      "BI\tbenA\n",
      "\n",
      "1\n",
      "\n",
      "The ML trees were constructed with IQ-TREE v. 1.4.4 (Nguyen et al. 2015) with branch support values obtained from 1 000 bootstrap replicates. Bayesian posterior probabilities (PP) were calculated using MrBayes v. 3.2.6 (Ronquist et al. 2012). Optimal partitioning scheme and substitution models were selected using PartitionFinder v. 1.1.1 as described above. The analyses ran for 5 × 106 generations, two parallel runs with four chains each were used, every 1 000th tree was retained, and the first 25 % of trees were discarded as burn-in.\n",
      "\n",
      "For inferring relationships within subg. Aspergillus, phylogenies were calculated from a benA, CaM and RPB2 concatenated dataset containing 102 individuals and 1 902 characters, of which 929 characters were variable and 864 were parsimony informative. ML and BI analyses were inferred as described above. Suitable partitioning schemes selected using PartitionFinder v. 1.1.1 are listed in Table 3.\n",
      "\n",
      "\n",
      "Species delimitation and species tree inference\n",
      "\n",
      "Nucleotide substitution models for each locus were determined using jModeltest v. 2.1.7 (Posada 2008) based on the Bayesian information criterion (BIC) and are listed in Table 4.\n",
      "\n",
      "Table 4\n",
      "Clade\tLocus\tSelected substitution model\n",
      "A. restrictus\tbenA\tSYM + G\n",
      "CaM\tTrNef + G\n",
      "RPB2\tK80 + G\n",
      "ITS + LSU\tTrN + I\n",
      "A. penicillioides\tbenA\tTrNef + G\n",
      "CaM\tTrNef + I\n",
      "RPB2\tTrNef + G\n",
      "ITS + LSU\tSYM + I + G\n",
      "Whole dataset\tbenA\tTrNef + G\n",
      "CaM\tTrNef + G\n",
      "RPB2\tTrNef + I + G\n",
      "ITS + LSU\tTrNef + I + G\n",
      "\n",
      "To assign individuals into species, several species delimitation methods were employed. In all cases the alignment was split into two parts. The first part contained the A. restrictus, A. conicus and A. vitricola clades, the second part the A. penicillioides clade. The Bayesian version of the general mixed yule-coalescent model (bgmyc) was performed in R v. 3.3.1 (R Core Team 2015) with the bGMYC package (Reid & Carstens 2012). The general mixed yule-coalescent method (GMYC) was performed in R v. 3.3.1 using the splits package (Fujisawa & Barraclough 2013). Single-locus ultrametric trees created in BEAST v. 2.4.2 (Bouckaert et al. 2014) were used as an input for both methods. Chain length for each tree was 1 × 107 generations with 25 % burn-in. The highest credibility tree was used for the GMYC method and 100 trees sampled throughout the analysis were used for the bGMYC method. These trees were obtained by equal sampling of all the trees from the analysis after discarding the first 60 % of trees. The analysis according Poisson tree processes (PTP) model was performed on species delimitation server (Zhang et al. 2013). The method does not require an ultrametric tree, so the single-locus input trees were calculated using ML analysis in IQ-TREE web server (Trifinopoulos et al. 2016). Species delimitation using the Automatic barcode gap discovery (ABGD) was performed on ABGD web (Puillandre et al. 2012). Finally multilocus species delimitation (STACEY) was performed with the BEAST v. 2.4.2 add-on STACEY v. 1.2.2 (Jones 2017). The chain length was set to 5 × 108 generations, priors were set as follows: the species tree prior was set to the Yule model, growth rate prior was set to lognormal distribution (M = 5, S = 2), clock rate priors for all loci were set to lognormal distribution (M = 0, S = 1), PopPriorScale prior was set to lognormal distribution (M = -7, S = 2) and relativeDeathRate prior was set to beta distribution (α = 1, β = 1 000). The output was processed with SpeciesDelimitationAnalyzer (Jones 2017).\n",
      "\n",
      "Species trees were inferred using *BEAST (Heled & Drummond 2010) implemented in BEAST v. 2.4.2. (Bouckaert et al. 2014). Individuals were assigned into species based on the consensual results from the above-mentioned species delimitation methods. The MCMC analysis was run for 1 × 108 of generations, 25 % of trees was discarded as burn-in. Strict molecular clock was chosen for all loci and population function was set as constant. Convergence was assessed by examining the likelihood plots in Tracer v. 1.6 (Rambaut et al. 2014).\n",
      "\n",
      "Species delimitation hypotheses were tested by a coalescent-based approach implemented in BP&P v. 3.1 (Bayesian phylogenetics and phylogeography) (Yang & Rannala 2010). Species delimitation using rjMCMC (reversible jump MCMC algorithm allows to inspect different models with given species tree) was performed with similar isolates allocation to species as during the species tree inference and the tree topology created according to the results from *BEAST. We analysed three combinations on the prior distributions of the parameters ϑ (ancestral population size) and τ0 (root age) as proposed by Leaché & Fujita (2010), i.e. large ancestral population sizes and deep divergence: ϑ ∼ G (1, 10) and τ0 ∼ G (1, 10); small ancestral population sizes and shallow divergences among species: ϑ ∼ G (2, 2 000) and τ0 ∼ G (2, 2 000); large ancestral populations sizes and shallow divergences among species: ϑ ∼ G (1, 10) and τ0 ∼ G (2, 2 000).\n",
      "\n",
      "Species boundaries were further validated by calculating the genealogical sorting index (GSI) (Cummings et al. 2008) which quantifies the degree of exclusive ancestry of hypothetical species. In order to perform the analysis, 100 trees inferred from each locus were created using RAxML (Stamatakis et al. 2008) with the bootstrap option. Calculation of gsi statistics was performed at http://www.molecularevolution.org, with 1 × 104 permutations for evaluation of statistical significance.\n",
      "\n",
      "\n",
      "Comparison with 454 sequence data\n",
      "\n",
      "Reference sequences generated in this study were compared to 454-pyrosequences obtained from house dust collected during a world-wide survey (Amend et al. 2010). Information with regards to dust collection and metagenomic analyses methods, readers are refered to Amend et al. (2010). For our comparisons, 454-sequences belonging to sect. Restricti were harvested by firstly doing a BLAST search of ITS barcodes from sect. Restricti against the main 454 database and retaining all sequences with at least 90 % similarity. This dataset was aligned in MAFFT v. 7 using the G-INS-i algorithm and subsequent neighbour-joining tree calculated in MEGA v. 7 (Kumar et al. 2016). This tree was used to remove all sequences that do not belong in sect. Restricti. The dataset was subsequently re-aligned and neighbour-joining tree calculated using Hamigera avellanea as outgroup.\n",
      "\n",
      "\n",
      "Morphology\n",
      "\n",
      "Macromorphological characters of colonies were observed on Harrold’s agar (M40Y) (Harrold 1950), Czapek yeast extract agar (CYA) (Pitt 1979), CYA supplemented with 20 % sucrose (CY20S) (Klich 2002), dichloran 18 % glycerol agar (DG18) (Hocking & Pitt 1980), malt extract agar (MEA; Oxoid) (Samson et al. 2010), Harrold's agar supplemented with 60 % Sucrose (M60Y) (Raper & Fennell 1965) and MEA supplemented with 10 % NaCl (MEA + 10 % NaCl). The isolates were inoculated in three points on 90 mm Petri dishes and incubated for 14 d at 25 °C in darkness. In addition, CY20S and M60Y plates were incubated at 30 °C and 37 °C. Colony diameters were measured after 14 d of incubation. The colony shape and texture, degree of sporulation, obverse and reverse colony colours, the production of soluble pigments and exudates were determined. The isolates of A. halophilicus were cultivated on Czapek–Dox agar (Thom & Church 1926) supplemented with 70 % sucrose (CZA70S) for 30 d. Colour names and codes used in descriptions refer to Kornerup & Wanscher (1967).\n",
      "\n",
      "Light microscope preparations were made from 14 d old colonies grown on M40Y. Lactic acid (60 %) was used as mounting fluid and ethanol (96 %) used to remove excess conidia and prevent air bubble formation. Microphotography was done using an Olympus BX-51 microscope with an Olympus DP72 camera and Zeiss AX10 Imager A2 light microscope equipped with a Nikon DS-Ri2 camera. Macromorphology of the colonies was observed and captured on a Zeiss Stereo Discovery V20 dissecting microscope equipped with a Nikon DS-Ri2 camera. Pictures were processed and photographic plates prepared in Adobe Photoshop CS6. Micromorphological characters (length and width of conidia, width of stipes and vesicles and length of phialides) were measured from at least five isolates of each species (when available). Slides were prepared from both the colony centre and margins. Each character was recorded at least forty times for each character and isolate. Linear discriminant analysis was performed with measured data in R 3.3.1 (R Core Team 2015) with packages MASS (Venables & Ripley 2002) and ggplot2 (Wickham 2009). The isolates were assigned to groups based on the results of molecular phylogenetic analyses (see above).\n",
      "\n",
      "Scanning electron microscopy (SEM) was performed using a JEOL-6380 LV microscope (JEOL Ltd. Tokyo, Japan) as described previously (Hubka et al. 2013).\n",
      "\n",
      "In brief, plugs from colonies (5 × 5 mm) grown 14 d on M40Y containing conidiophores, and ascomata in the case of A. halophilicus (longer incubation on CZA70S was necessary) were fixed in osmium tetroxide vapours for 2 wk at 5–10 °C and gold-coated in a Bal-Tec SCD 050 sputter coater. The specimens were observed with spot size 40–42 μm and accelerating voltage 25 kV. Terminology of the surface ornamentation of the conidia was adopted from Kozakiewicz (1989).\n",
      "\n",
      "\n",
      "Physiology\n",
      "\n",
      "At least five isolates from each species, when available, were selected for determining species growth rates in an osmotic gradient. Each strain was cultivated at 25 °C on MEA with six different concentrations of NaCl ranging from 0 to 25 %. After 14 d, colony sizes were recorded and growth curves for each species calculated using local regression (LOESS) in R v. 3.3.1. (R Core Team 2015) using the ggplot2 package (Wickham 2009).\n",
      "\n",
      "\n",
      "Exometabolite analysis\n",
      "\n",
      "The isolates of Aspergillus sect. Restricti were incubated on DG18, CY20S and yeast extract sucrose agar (YES) agar, for 2 wk at 25 °C. Two agar plugs from each medium (6 plugs in total) were combined in one vial and extrolites extracted with ethylacetate / isopropanol (3:1) with added 1 % formic acid, and ultrasonication (50 min). In the case of A. halophilicus, six CZA70S plugs containing ascomata (ca 1–2 mo old colonies) were extracted. After ultrasonication, the plugs were removed and the organic solvent evaporated. The remaining extract was re-dissolved in methanol, centrifuged at 13 300 rpm and transferred to a small vial with a V-formed insert. HPLC analysis was done according to Frisvad & Thrane (1987) as modified by Nielsen et al. (2011).\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "\n",
      "Phylogeny of subgenus Aspergillus\n",
      "\n",
      "ML and BI analysis of 102 concatenated sequences of benA, CaM and RPB2 contained 31 species from sect. Aspergillus as recognised by Chen et al. (2017) and 21 species from sect. Restricti recognised here (see below). Tree topologies between ML and BI did not differ and the ML tree was used with both bootstrap and pp values included (Fig. 1, Fig. 2). Both analyses supported the monophyly of both sections. Each section contains several highly supported clades, but the exact position of species within the clades is often unresolved. Despite producing a eurotium-like sexual state common in sect. Aspergillus, A. halophilicus is resolved with high statistical support in sect. Restricti, but its exact position remains unclear. It is apparent from the radial representation of the tree (Fig. 2), that there are large genetic distances between the different clades of sect. Restricti, but these gaps may represent only hidden variability that has not been discovered during our study due to insufficient sampling or the use of inappropriate isolation media. The retaining of the current classification scheme of subg. Aspergilllus with two sections seems currently the best solution until more data on species diversity in sect. Restricti are collected. Additionally, sects. Restricti and Aspergillus are well supported by phenotypic data (see discussion).\n",
      "\n",
      "Fig. 1 A 90 % majority consensus tree of the subgenus Aspergillus inferred with Maximum likelihood analysis based on benA, CaM and RPB2 loci (partitioning scheme and substitution models are listed in Table 3). The data set contained 102 strains and 1902 characters, of which 929 characters were variable and 864 were parsimony informative. Support values represent maximum likelihood bootstrap/ Bayesian posterior probability values, 100 % bootstrap values and 1.00 posterior probability are designated by asterisk *. The ex-type isolates are designated by a superscript T. Hamigera avellanea (NRRL 1938) was used as outgroup.Fig. 1\n",
      "\n",
      "Fig. 2 Maximum likelihood phylogenetic tree of the subgenus Aspergillus inferred from partitioned analysis of concatenated dataset (benA, CaM and RPB2) with Maximum likelihood method and presented in radial format.Fig. 2\n",
      "\n",
      "\n",
      "Species delimitation and validation in sect. Restricti\n",
      "\n",
      "For species delimitation, the alignment was divided into two parts as discussed earlier. Eleven species were delimited within the first part that consisted of the A. restrictus, A. conicus and A. vitricola clades using STACEY and similarly, nine species were delimited within the second part of the data set (A. penicillioides clade). Aspergillus halophilicus was exluded from species delimitation analyses because it clearly represents a distantly related clade within the section. Results are summarised in Fig. 3, Fig. 4. Tree topologies in the Fig. 3, Fig. 4 were inferred in STACEY and used solely for the comprehensive presentation of the results from different methods; the evolutionary relationships in the section inferred by *BEAST are presented as the most robust (Fig. 5). In the first step, we compared results from four single-locus species delimitation methods with those derived from STACEY, that is currently one of the most advanced species delimitation methods because it processes multiple loci simultaneously during a single analysis (Jones 2017). Although the results vary across the methods and loci, the consensual results from single-locus species delimitation methods are generally in agreement with the results of STACEY. Single-locus method bGMYC was the most computationally intensive method among those used in this study and its results were most similar to STACEY. The method with greatest variability across the four loci was GMYC with 14 to 39 delimited species for the first part of the analysed data set (A. restrictus, A. conicus and A. vitricola clades) and eight to 38 species for A. penicillioides clade. A significant over delimitation was observed when analysing the ID region of the first part of the data set and also CaM and RPB2 loci using GMYC method (Fig. 3). A similar problem was observed in the case of CaM when analysing the second part of the data set and also in RPB2 locus when using GMYC method (Fig. 4). Although ABGD is a quite simplistic method compared to other used methods it yielded similar results to STACEY and bGMYC but over delimitation was observed when analysing the CaM locus in the A. penicillioides clade (Fig. 3, Fig. 4). The number of species delimited by PTP was slightly higher compared to bGMYC, but lower than in the case of GMYC. Single-locus methods often delimited additional species within A. conicus and A. vitricola, but the results were not consistent and in some cases even contradictory, suggesting recombination within the clade (Fig. 3). These tentative species had no or very limited phenotypic support, which is the reason for adopting a broader species concept. In contrast, the majority of single-locus methods did not support delimitation of A. clavatophorus and A. pachycaulis based on the ID region; the methods also did not support recognition of A. destruens and A. domesticus when analysing ID-region and RPB2 in contrast to benA and CaM. All these species were supported by STACEY and phenotype analysis, resulting in us proposing them as new species.\n",
      "\n",
      "Fig. 3 Schematic representation of results of species delimitation methods in A. restrictus, A. conicus and A. vitricola clades (108 isolates). The results of multilocus method (STACEY) are compared to results of single-locus methods (bGMYC, PTP, GMYC, ABGD). Results from different methods are depicted with coloured bars highlighting congruence across methods. Displayed tree comes from STACEY analysis and is used solely for the comprehensive presentation of the results from different methods.Fig. 3\n",
      "\n",
      "Fig. 4 Schematic representation of results of species delimitation methods in A. penicillioides clade (86 isolates). The results of multilocus method (STACEY) are compared to results of single-locus methods (bGMYC, PTP, GMYC, ABGD). Results from different methods are depicted with coloured bars highlighting congruence across methods. Displayed tree comes from STACEY analysis and is used only used solely for the comprehensive presentation of the results from different methods.Fig. 4\n",
      "\n",
      "Fig. 5 Bayesian species tree based on sequence data from four loci of 193 isolates inferred by *BEAST with posterior probabilities appended to nodes. Terminal branches represent delimited species (each comprises all isolates of respective species).Fig. 5\n",
      "\n",
      "Based on consensus results of species delimitation methods and after reflection of phenotypic data in ambiguous species, we recognise 21 species within the sect. Restricti. This number comprises seven previously recognised and 14 new species proposed here (see section Taxonomy). All four loci have sufficient variability for reliable species identification and can be used as DNA barcodes. ID region has the lowest discriminative power but it is still sufficient for differentiation of all species. The locus with the highest ratio of variable positions to the sequence length was benA.\n",
      "\n",
      "The species validation analysis results are listed in Table 5. All species were supported by the posterior probability 1.00 based on the analysis in BP&P v. 3.1 (Yang & Rannala 2010) under all three scenarios simulated by different prior distributions of parameters ϑ (ancestral population size) and τ0 (root age). The gsi calculations and significance testing performed using the genealogical sorting index software also confirmed that all delimited species can be considered separate evolutionary lineages. The ensemble statistic gsiT (weighted sum of gsi across genealogies) for each dataset (100 bootstrap trees for each locus and concensus trees) are listed in Table 5. Relatively low gsi value was obtained from the analysis of β-tubulin dataset in A. hordei and ID dataset in A. domesticus, however even in these cases the p-value of the permutation test is nearly 0. The gsi statistic was not calculated for A. infrequens and A. pseudogracilis because they were both represented by only one isolate.\n",
      "\n",
      "Table 5\n",
      "Species\tbenA\tCaM\tITS\tRPB2\tConsensus trees\n",
      "gsi\tp-value\tgsi\tp-value\tgsi\tp-value\tgsi\tp-value\tgsi\tp-value\n",
      "A. restrictus\n",
      "A. restrictus\t0.997\t0.0001\t0.989\t0.0001\t0.898\t0.0001\t0.999\t0.0001\t0.983\t0.0001\n",
      "A. caesiellus\t1\t0.0001\t0.959\t0.0001\t0.947\t0.0001\t0.959\t0.0001\t1\t0.0001\n",
      "A. pachycaulis\t1\t0.0003\t1\t0.0003\t0.942\t0.0001\t1\t0.0009\t1\t0.0005\n",
      "A. conicus\n",
      "A. conicus\t0.999\t0.0001\t0.994\t0.0001\t0.914\t0.0001\t0.979\t0.0001\t1\t0.0001\n",
      "A. destruens\t0.997\t0.0001\t1\t0.0001\t0.894\t0.0001\t0.991\t0.0001\t1\t0.0001\n",
      "A. domesticus\t0.98\t0.0001\t0.897\t0.0001\t0.707\t0.0001\t0.891\t0.0001\t0.933\t0.0001\n",
      "A. gracilis\t1\t0.0001\t1\t0.0001\t0.946\t0.0001\t1\t0.0001\t1\t0.0001\n",
      "A. villosus\t1\t0.0005\t1\t0.0007\t0.952\t0.0001\t1\t0.0006\t1\t0.0007\n",
      "A. vitricola\n",
      "A. vitricola\t0.942\t0.0001\t1\t0.0001\t0.958\t0.0001\t0.995\t0.0001\t0.981\t0.0001\n",
      "A. glabripes\t0.999\t0.0001\t1\t0.0001\t0.913\t0.0001\t0.979\t0.0001\t0.975\t0.0001\n",
      "A. penicillioides\n",
      "A. penicillioides\t0.871\t0.0001\t0.983\t0.0001\t0.948\t0.0001\t0.995\t0.0001\t0.960\t0.0001\n",
      "A. canadensis\t0.987\t0.0001\t1\t0.0001\t1\t0.0001\t1\t0.0001\t1\t0.0001\n",
      "A. clavatophorus\t1\t0.0001\t0.980\t0.0001\t1\t0.0005\t1\t0.0001\t1\t0.0003\n",
      "A. hordei\t0.675\t0.0001\t1\t0.0001\t0.948\t0.0001\t0.997\t0.0001\t0.915\t0.0001\n",
      "A. magnivesiculatus\t0.997\t0.0001\t0.974\t0.0001\t0.898\t0.0001\t0.998\t0.0001\t1\t0.0001\n",
      "A. reticulatus\t0.995\t0.0001\t0.995\t0.0001\t0.930\t0.0001\t1\t0.0001\t0.977\t0.0001\n",
      "A. salinicola\t0.971\t0.0001\t0.997\t0.0001\t0.987\t0.0001\t1\t0.0003\t1\t0.0001\n",
      "A. tardicrescens\t1\t0.0001\t0.933\t0.0001\t1\t0.0001\t0.985\t0.0001\t1\t0.0001\n",
      "\n",
      "\n",
      "Species tree\n",
      "\n",
      "The species tree topology was inferred with *BEAST (Heled & Drummond 2010) and is depicted on Fig. 5, Fig. 6. It was used as a guide tree for species validation in BP&P but it also represents the most probable evolutionary relationships between the species. Four clusters of species designated as the A. restrictus, A. conicus, A. vitricola and A. penicillioides clades are denoted in Fig. 5. We observed only slight differences in the topology of the species tree and statistical support of some nodes inferred with *BEAST when compared to the results of ML and BI analysis of the concatenated and partitioned dataset (see below). Several nodes had only limited support in all types of analyses and there is definitely some degree of uncertainty. These uncertainties can be visualized in DensiTree (Bouckaert 2010) (Fig. 6), that displays all trees created during the analysis except burn-in phase and trees with one of the three most common topologies are differently coloured. The most obvious problem is the position of A. halophilicus. Based on the available sequence data it is not clear whether it is situated basally within sect. Restricti or if it is rather phylogenetically related to some clade within the section. In addition, *BEAST analysis supported (PP = 1.00) an A. conicus clade consisting of six species, A. conicus, A. domesticus, A. destruens, A. villosus, A. gracilis and A. pseudogracilis, however this arrangement was not supported by the concatenated dataset analysis (see Fig. 7) where A. gracilis and A. pseudogracilis formed a distinct clade from the remaining four species. The position of A. villosus within the clade is also not fully resolved and differs between phylogenetic methods (compare Figs 5 and 7). The topology of the A. penicillioides clade was almost identical when using all methods. The exception is an unresolved position of A. clavatophorus and A. tardicrescens. The respective nodes gained only limited support while all other nodes are suported by PP = 1.00 in *BEAST analysis.\n",
      "\n",
      "Fig. 6 Species tree inferred with *BEAST visualized by using DensiTree (Bouckaert 2010). All trees created in the analysis (except burn-in phase) are displayed. Trees with the most common topology are highlighted by blue colour, trees with the second most common topology by red colour, trees with the third most common topology by pale green and all other trees by dark green.Fig. 6\n",
      "\n",
      "Fig. 7 Bayesian tree depicting the relationship of species from Aspergillus section Restricti based on four loci (partitioning scheme and substitution models are listed in Table 3). Total length of the concatenated alignment was 2093 characters, with 837 variable and 668 parsimony informative sites. Support values represent Bayesian posterior probability/maximum likelihood bootstrap values, 100 % bootstrap values and 1.00 posterior probability are designated by asterisk *. The ex-type isolates are designated by bold font and superscript T. Hamigera avellanea (NRRL 1938) was used as outgroup.Fig. 7\n",
      "\n",
      "\n",
      "Bayesian and Maximum likelihood analysis of the partitioned sequence data\n",
      "\n",
      "All species delimited by methods based on coalescent model were clearly supported by BI and ML analyses (Fig. 7). All clades representing species were supported by PP = 1.00 in BI analysis and at least 93 % bootstrap support in ML analysis. Aspergillus halophilicus was in a polytomy on the base of the section in agreement with the *BEAST analysis. Slight differences were observed in the interspecies relationships between core species of A. penicillioides as mentioned above. The ML and BI analysis of partitioned dataset did not separate A. restrictus and A. conicus clades as did the *BEAST analysis (compare Figs 5 and 7).\n",
      "\n",
      "\n",
      "Comparisons with 454 data\n",
      "\n",
      "A total of 1061 454-pyrosequences were found to belong to sect. Restricti and included in a dataset with 188 ITS barcodes generated during this study. The aligned dataset was 402 bp long and the calculated neighbour-joining tree is shown in Fig. 8. Many species were detected from dust samples, while A. pseudogracilis, A. villosus, A. destruens, A. glabripes, A. salinicola and A. infrequens were not detected. The phylogeny revealed 3–5 unnamed lineages from dust collected from Indonesia, Mexico, Micronesia, the Netherlands, South Africa, Thailand and Uruguay. Two singletons were also detected across the tree. Interestingly, each lineage is represented by OTU's from multiple countries, meaning they are probably common species.\n",
      "\n",
      "Fig. 8 Neighbour Joining tree created in MEGA 7 with 1000 bootstrap replicates based on ITS from 454-pyrosequences obtained from house dust samples (n = 1061; Amend et al. 2010) and from this study (n = 188). The tree is rooted with Hamigera avellanea. Monophyletic groups/species are collapsed and shown as proportional triangles. The information about country of origin pertains to and is given only for species found in house dust.Fig. 8\n",
      "\n",
      "\n",
      "Phenotype analysis\n",
      "\n",
      "\n",
      "Micromorphology\n",
      "\n",
      "Results of linear discriminant analysis (LDA) based on measurements of micromorphological characters (length and width of conidia, width of stipe and vesicle from the centre and edge of the colony and length of phialides) are shown on Fig. 9, Fig. 10, Fig. 11. The values of individual phenotypic characters (average value ± standard deviation) for each species can be found in Table 6. The results of the analysis with individuals assigned to the species complexes are depicted on Fig. 9. While species from A. penicillioides complex and A. restrictcus/conicus complex form well-defined and relatively separated clusters, individuals from the A. vitricola complex formed two distinct clusters. These two clusters correspond to the two micromorphologically dissimilar species within the complex, i.e., A. vitricola and A. glabripes. It is also apparent from the analysis, that the most important character discriminating species complexes is the conidial size.\n",
      "\n",
      "Fig. 9 Results of linear discriminant analysis (LDA) performed in R 3.3.1 based on micromorphological measurements of all individuals assigned into species complexes. Ellipses represent 95 % confidence interval and arrows represent the contribution of each character to the axes. The length (l) and width (w) of conidia, diameter of stipes and vesicles were measured separately from the center and the edge of colonies.Fig. 9\n",
      "\n",
      "Fig. 10 Results of linear discriminant analysis (LDA) performed in R 3.3.1 based on micromorphological measurements of species from A. restrictus, A. conicus and A. vitricola clade. Dashed ellipses represent 95 % confidence interval, full circles euclidean distances. Analyses involving only two species (B, D, F) are represented by probability density function. 1 – A. restrictus; 2 – A. caesiellus; 3 – A. pachycaulis; 4 – A. conicus; 5 – A. domesticus; 6 – A. destruens; 7 – A. villosus; 8 – A. gracilis; 9 – A. pseudogracilis; 10 – A. vitricola; 11 – A. glabripes.Fig. 10\n",
      "\n",
      "Fig. 11 Results of linear discriminant analysis (LDA) performed in R 3.3.1 based on micromorphological measurements of species from A. penicillioides clade. Dashed ellipses represent 95 % confidence interval, full circles euclidean distances. Analyses involving only two species (B, D, F) are represented by probability density function. 1 – A. penicillioides; 2 – A. tardicrescens; 3 – A. clavatophorus; 4 – A. magnivesiculatus; 5 – A. hordei; 6 – A. reticulatus; 7 – A. salinicola; 8 – A. canadensis; 9 – A. infrequens.Fig. 11\n",
      "\n",
      "Table 6\n",
      "Species\tConidia\tConidiophores\n",
      "Length\tWidth\tOrnamentation (SEM)\tMature conidia shape\tStipe width (μm)\tVesicle width (μm)\tPhialides length (μm)\tStipe surface (SEM)\tShape of vesicle\n",
      "A. restrictus\n",
      "A. restrictus\t3.5–4.5 (4.1±0.4)\t2.5–3 (2.9±0.4)\tAculeate\tEllipsoidal to ovate\t4–5.5 (4.8±1)\t9–12 (11.5±2)\t8–9 (8.4±0.9)\tDensely covered by long hairs\tPyriform to clavate\n",
      "A. caesiellus\t4.5–6 (5±0.8)\t3–4.5 (3.6±0.5)\tAculeate\tEllipsoidal to ovate\t5–7.5 (5.5±0.8)\t11–16 (13.2±2)\t8–9 (8.2±2)\tDensely covered by long hairs\tPyriform to clavate\n",
      "A. pachycaulis\t4–5 (4.4±0.2)\t2.5–3.5 (3±0.2)\tAculeate\tEllipsoidal to ovate\t7–8 (7.3±0.8)\t16–20 (18.6±1.8)\t8–9 (8.7±0.5)\tDensely covered by short hairs\tPyriform to subclavate\n",
      "A. conicus\n",
      "A. conicus\t4–4.5 (4.1±0.4)\t2.5–3 (2.9±0.3)\tAculeate\tGlobose to subglobose or barrel-shaped\t4–5.5 (4.5±0.9)\t8–13 (10.5±2.5)\t8.5–10 (9.2±1.5)\tSparsely covered by short hairs\tPyriform, spatulate to clavate\n",
      "A. destruens\t3.5–5.5 (4.3±0.6)\t2.5–3.5 (3.1±0.4)\tAculeate\tSubglobose to ovate\t4–6 (4.2±0.9)\t8–14 (10.2±2.1)\t8.5–11.5 (8.6±1.7)\tDensely covered by long hairs\tPredominantly ellipsoidal to pyriform\n",
      "A. domesticus\t3.5–4.5 (4.2±0.5)\t2.5–3.5 (2.8±0.3)\tAculeate\tEllipsoidal or barrel-shaped\t4–6 (4.8±0.8)\t7–13 (10.6±2.7)\t8.5–11 (9.7±1)\tDensely covered by long hairs\tPyriform, spatulate to clavate\n",
      "A. gracilis\t3–4 (3.4±0.2)\t2–3 (2.5±0.2)\tAculeate\tSubglobose or barrel-shaped\t3–5 (4.1±0.5)\t6.5–9 (8±0.7)\t7.5–9 (8.7±0.6)\tDensely covered by very long hairs\tSpatulate to clavate\n",
      "A. pseudogracilis\t3–4 (3.6±0.3)\t2–3 (2.7±0.2)\tAculeate\tSubglobose or barrel-shaped\t4–6 (5±0.6)\t9–13 (10.3±1.6)\t9–10.5 (10±0.4)\tDensely covered by long hairs\tSpatulate to clavate\n",
      "A. villosus\t3.5–4.5 (3.9±0.4)\t2.5–3.5 (2.9±0.2)\tTuberculate\tEllipsoidal or barrel-shaped\t5–6 (5.5±0.5)\t10–14 (12.6±1.4)\t9–12 (10.8±1.1)\tDensely covered by long hairs\tPyriform, spatulate to clavate\n",
      "A. vitricola\n",
      "A. vitricola\t4.5–5.5 (4.8±0.4)\t3–4 (3.4±0.3)\tAculeate\tSubglobose to ellipsoidal\t4.5–6 (5.1±0.6)\t7–12 (9.6±1.7)\t8–10 (9.1±0.7)\tSmooth\tSpatulate, pyriform to clavate\n",
      "A. glabripes\t3.5–4.5 (4.1±0.2)\t2–3 (2.9±0.2)\tAculeate\tSubglobose\t6–8 (6.9±0.8)\t16–22 (19.9±2.3)\t7–9 (8.5±1)\tSmooth\tGlobose, pyriform to subclavate\n",
      "A. penicillioides\n",
      "A. penicillioides\t3.5–4.5 (4±0.6)\t2.5–3.5 (3±0.5)\tTuberculate\tSubglobose or barrel-shaped\t4–6 (5.1±0.9)\t10–18 (13.5±2.3)\t8–10 (9.3±1.7)\tSparsely covered by short hairs\tPyriform, spatulate to clavate\n",
      "A. canadensis\t3–4 (3.3±0.4)\t2.5–3 (2.8±0.4)\tLobate-reticulate\tSubglobose or barrel-shaped\t3.5–5 (4.2±0.7)\t(9–)13–16(–20) (13.9±2.5)\t6–8 (6.9±0.4)\tSparsely covered by bundles of long hairs\tPyriform to clavate\n",
      "A. clavatophorus\t4–5 (4.8±0.2)\t3–4 (3.5±0.2)\tTuberculate\tSubglobose, ovate or barrel-shaped\t6–9 (6.6±0.5)\t14–19 (16.5±1.3)\t8–10 (9.3±0.7)\tDensely covered by short hairs\tSpatulate to clavate\n",
      "A. hordei\t3.5–5 (4.1±0.4)\t2.5–4 (3.2±0.4)\tLobate-reticulate\tGlobose or barrel-shaped\t4.5–6 (5.5±0.6)\t12–14.5 (13.2±1.4)\t8.5–11 (9.5±1)\tDensely covered by short hairs\tSpatulate, ellipsoidal, pyriform to subclavate\n",
      "A. infrequens\t4–5 (4.8±0.3)\t3–4 (3.6±0.3)\tAculeate\tSubglobose, ovate or barrel-shaped\t6–7 (6.7±0.8)\t(10–)13–17(–20) (15.9±2.6)\t8–10 (9.5±1)\tSparsely covered by bundles of long hairs\tPyriform, spatulate to clavate\n",
      "A. magnivesiculatus\t4.5–5.5 (5.2±0.3)\t3.5–4.5 (3.9±0.3)\tTuberculate\tOvate to barrel-shaped\t5–7 (5.9±1.2)\t17–22 (21.4±3.2)\t9–11 (9.6±0.8)\tDensely covered by short hairs\tPyriform, subclavate to clavate\n",
      "A. reticulatus\t3.5–4.5 (3.9±0.3)\t2.5–3.5 (3±0.3)\tTuberculate to lobate-reticulate\tGlobose or barrel-shaped\t5–7 (6±1)\t14–20 (16.3±3.2)\t9–11 (9.3±0.8)\tDensely covered by short hairs\tGlobose or pyriform\n",
      "A. salinicola\t3–4 (3.6±0.4)\t2–3 (2.6±0.2)\tTuberculate to lobate-reticulate\tGlobose to subglobose\t4–6 (3.6±0.4)\t11–14 (12.6±1.3)\t8–9 (8.7±0.4)\tSparsely covered by short hairs\tPyriform, spatulate to subclavate\n",
      "A. tardicrescens\t3.5–4.5 (4±0.4)\t2.5–3.5 (2.9±0.3)\tTuberculate\tSubglobose or barrel-shaped\t5–8 (6±1.2)\t14–19 (16.4±3.3)\t8–11 (8.9±1.2)\tDensely covered by short hairs\tSubglobose, spatulate, pyriform to subclavate\n",
      "\n",
      "\n",
      "Fig. 10, Fig. 11 show results of the analysis separately for particular species within species complexes/clades. The LDA clearly discriminated A. pachycaulis within the A. restrictus clade (Fig. 10A); the discrimation between A. restrictus and A. caesiellus was not so obvious because some isolates have similar micromorphologies. The discriminatory power of micromorphological characters within the A. conicus clade is low except for A. gracilis and A. pseudogracilis that were relatively well separated from other species (see Fig. 10C). The two species from the A. vitricola clade vary greatly in their micromorphological characters and consequently it resulted in clear separation of species in LDA.\n",
      "\n",
      "The micromorphology of core species of the A. penicillioides clade is nearly identical, except A. magnivesiculatus that has much larger vesicles. It is more meaningful to compare micromorphology only in pairs of related species and even then some species are not well separated as in the case of A. penicillioides and A. hordei (see Fig. 11C). On the other hand two pairs of closely related species, distant from the core group (A. reticulatus – A. salinicola and A. canadensis – A. infrequens) can be distinguished very well by the LDA (Fig. 11E, F).\n",
      "\n",
      "\n",
      "Growth in an osmotic gradient\n",
      "\n",
      "The ability to grow in an osmotic gradient proved to be very consistent during repeated testing of the same isolate and similarly, there was only low infraspecific variability. In general, it can be considered a useful supplementary taxonomic character that can distinguish some species. Growth curves created from measurements of the extension rate after 14 d are displayed on Fig. 12, Fig. 13. The A. restrictus clade contains less xerophilic species compared to others and grow faster on media with higher water activity (MEA or MEA + 5 % NaCl); A. caesiellus differs from the two remaining species of A. restrictus clade by faster growth on MEA + 10 % NaCl and 15 % NaCl. Species from the A. conicus clade show relatively similar growth curves; A. domesticus, A. gracilis and A. pseudogracilis grow slower than the three remaining species on MEA + 5 % and 10 % NaCl; A. conicus grows fastest on MEA + 5 % and 10 % NaCl; A. villosus grows faster on MEA and MEA + 5 % NaCl but with decreasing water activity the curve quickly declines and the species grows poorly on MEA + 15 % and 20 % NaCl. Aspergillus vitricola and A. glabripes differ significantly in their growth on MEA and MEA + 5 % NaCl.\n",
      "\n",
      "Fig. 12 Osmotic gradient growth curves of species from A. restrictus, A. conicus and A. vitricola clades. Growth curves were created using local regression (LOESS) in R 3.3.1, grey zones represent 95 % confidence intervals. 1 – A. restrictus; 2 – A. caesiellus; 3 – A. pachycaulis; 4 – A. conicus; 5 – A. domesticus; 6 – A. destruens; 7 – A. villosus; 8 – A. gracilis; 9 – A. pseudogracilis; 10 – A. vitricola; 11 – A. glabripes.Fig. 12\n",
      "\n",
      "Fig. 13 Osmotic gradient growth curves of species from A. penicillioides clade. Growth curves were created using local regression (LOESS) in R 3.3.1, grey zones represent 95 % confidence intervals. 1 – A. penicillioides; 2 – A. tardicrescens; 3 – A. clavatophorus; 4 – A. magnivesiculatus; 5 – A. hordei; 6 – A. reticulatus; 7 – A. salinicola; 8 – A. canadensis; 9 – A. infrequens.Fig. 13\n",
      "\n",
      "Growth curves of species belonging to the core group of the A. penicillioides clade are nearly identical but differences can be found when comparing some species pairs (see Fig. 13), e.g. A. penicillioides and A. tardicrescens (Fig. 13B). Aspergillus reticulatus and A. salinicola grow faster than all other species on MEA + 5 % NaCl and slower on MEA + 20 % NaCl. Aspergillus canadensis and A. infrequens grow slowest on MEA + 5 % NaCl. The growth curves of the last mentioned species pairs are also different (see Fig. 13F).\n",
      "\n",
      "\n",
      "Other phenotypic characters\n",
      "\n",
      "Some other phenotypic characters are taxonomically relevant and useful for differentiation of species complexes or particular species. An overview of recorded characters is given in Table 6, Table 7 and their usability for differentiation of species complexes is detailed in the Taxonomy section; possibilities of distinguishing species within complexes are detailed in Distinguishing characters that follows the descriptions of species. The shape of conidial heads can be used to distinguish between clades. Conidial heads of A. restrictus clade form compact columns, heads of A. conicus clade are loosely columnar, those of A. vitricola clade are radiate and A. penicillioides clade have mainly globose conidial heads.\n",
      "\n",
      "Table 7\n",
      "Species\tSporulation colour\tReverse colour in the center\tShape of conidial heads\n",
      "A. restrictus\n",
      "A. restrictus\tGreyish turquoise (24C4)\tCinnamon brown (6D6)\tColumnar\n",
      "A. caesiellus\tGreyish turquoise (24E4)\tSunburn light brown (6D5)\tColumnar\n",
      "A. pachycaulis\tGreyish turquoise (24C4)\tBrownish yellow (5C8)\tColumnar\n",
      "A. conicus\n",
      "A. conicus\tPale turquoise (24A3)\tOlive grey (2F2)\tLoosely columnar\n",
      "A. destruens\tMyrtle green (25F7)\tApricot greyish orange (5B6)\tLoosely columnar\n",
      "A. domesticus\tAquamarine greyish turquoise (24B3)\tOlive grey (2F2)\tLoosely columnar\n",
      "A. gracilis\tAquamarine greyish turquoise (24B3) to turquoise white (24A2)\tOlive grey (2F2)\tLoosely columnar\n",
      "A. pseudogracilis\tLight turquoise (24A4)\tCinnamon brown (6D6)\tLoosely columnar\n",
      "A. villosus\tGreyish green (25E7) to bluish green (25A6)\tOlive green (2F6)\tLoosely columnar\n",
      "A. vitricola\n",
      "A. vitricola\tGreyish turquoise (water blue) (24C5)\tOlive green (2F6)\tRadiate\n",
      "A. glabripes\tYellowish green (29C7)\tDark brown (6F7)\tRadiate\n",
      "A. penicillioides\n",
      "A. penicillioides\tDeep green (25E8)\tCoffee brown (5F7)\tFirst globose, later becoming loosely columnar\n",
      "A. canadensis\tJungle green (25F5) to pale turquoise (24A3)\tOlive grey (2F2)\tRadiate\n",
      "A. clavatophorus\tGreyish green (28D6)\tCoffee brown (5F7)\tFirst globose, later becoming radiate\n",
      "A. hordei\tDark turquoise (24F8)\tCuba reddish brown (9E8)\tFirst globose, later becoming loosely columnar\n",
      "A. infrequens\tPale turquoise (24A3) to dark turquoise (24F8)\tDark green (30F8)\tRadiate\n",
      "A. magnivesiculatus\tGreyish green (26D5)\tMahogany red (8E7)\tFirst globose, later becoming loosely columnar to radiate\n",
      "A. reticulatus\tGreyish turquoise (24E6)\tGolden brown (5D7)\tGlobose to radiate\n",
      "A. salinicola\tGreyish green (25B5)\tGolden brown (5D7)\tFirst globose, later becoming loosely radiate\n",
      "A. tardicrescens\tGreyish turquoise (water blue) (24C5)\tOlive brown (4F8)\tFirst globose, later becoming radiate\n",
      "\n",
      "The extension rates of each species are given in Table 8. Aspergillus pachycaulis is the only member of the section Restricti that was able to grow on CY20S at 37 °C, but several species were able to grow at 37 °C on a more sucrose rich medium like M60Y. The extension rates of the species on optimal media (M40Y, M60Y, MEA + 10 % NaCl) was generally the same, but the species exhibit a greater degree of variability in more extreme conditions (high water activity, high temperature) and the ability or inability to grow on these suboptimal media can be used as taxonomic characters to distinguish between closely related species (e.g. A. conicus, A. domesticus and A. destruens).\n",
      "\n",
      "Table 8\n",
      "Species\tCYA, 25 °C\tMEA, 25 °C\tMEA+10 % NaCl, 25 °C\tDG18, 25 °C\tM40Y, 25 °C\tCY20S\tM60Y\n",
      "25 °C\t30 °C\t37 °C\t25 °C\t30 °C\t37 °C\n",
      "A. restrictus\n",
      "A. restrictus\t12–16\t5–13\t33–40\t28–33\t42–53\t24–29\t12–25\t0\t38–50\t45–55\t20–40\n",
      "A. caesiellus\t11–22\t17–22\t36–45\t32–38\t58–60\t21–35\t12–40\t0\t48–60\t45–60\t32–55\n",
      "A. pachycaulis\t11–26\t21–31\t27–43\t35–48\t20–60\t15–40\t22–36\t(0) 8–10\t25–60\t36–60\t38–60\n",
      "A. conicus\n",
      "A. conicus\t(0) 10–12\t(0) 8–10\t16–36\t18–31\t32–40\t5–18\t0\t26–40\t26–37\t0\n",
      "A. destruens\t5–10\t(0) 4–7\t20–30\t14–30\t19–36\t6–13\t0\t34–41\t16–41\t0\n",
      "A. domesticus\t0–8\t0\t23–31\t20–24\t36–38\t5–17\t0\t29–38\t15–30\t0\n",
      "A. gracilis\t0–8\t0\t12–24\t9–16\t30–42\t7–13\t0\t38–44\t39–45\t0\n",
      "A. pseudogracilis\t0\t25–26\t20–22\t42–45\t7–8\t0\t52–53\t43–45\t0\n",
      "A. villosus\t8–10\t5–6\t22–23\t16–20\t21–23\t17–19\t7–8\t0\t26–28\t28–30\t6–8\n",
      "A. vitricola\n",
      "A. vitricola\t0\t9–26\t8–18\t19–36\t(0) 4–7\t0\t25–42\t16–42\t0\n",
      "A. glabripes\t3–7\t4–11\t22–28\t15–25\t38–52\t8–18\t5–16\t0\t35–40\t41–56\t(0) 7–10\n",
      "A. penicillioides\n",
      "A. penicillioides\t7–9\t5–9\t24–37\t17–36\t27–43\t11–20\t11–15\t0\t28–59\t40–60\t17–45\n",
      "A. canadensis\t0\t17–33\t7–15\t25–29\t(0) 10–11\t0\t23–42\t2–20\t0\n",
      "A. clavatophorus\t0\t18–28\t18–25\t27–33\t7–16\t0\t35–44\t29–54\t(0) 5–17\n",
      "A. hordei\t0–8\t0–5\t26–29\t13–26\t25–29\t(0) 13–16\t(0)8–15\t0\t31–47\t25–48\t22–33\n",
      "A. infrequens\t4–5\t0\t18–19\t25–26\t21–22\t7–8\t0\t40–43\t8–10\t0\n",
      "A. magnivesiculatus\t0\t27–34\t20–26\t27–35\t11–17\t0\t47–60\t40–50\t0\n",
      "A. reticulatus\t8–14\t2–7\t32–40\t26–33\t46–50\t21–30\t8–18\t0\t55–60\t50–55\t3–20\n",
      "A. salinicola\t(0) 8–9\t0–5\t32–35\t21–30\t32–40\t9–20\t0–5\t0\t42–55\t27–43\t0\n",
      "A. tardicrescens\t0\t19–29\t12–17\t17–29\t0–5\t0\t31–50\t15–35\t0\n",
      "\n",
      "\n",
      "Exometabolite analysis\n",
      "\n",
      "The overview of metabolites produced by sect. Restricti species is given in Table 9 and their absorption maxima and bracketed retention indexes are listed in Table S1. Six species had isolates that could produce mycophenolic acid: A. glabripes, A. gracilis, A. halophilicus, A. pachycaulis, A. penicillioides and A. tardicrescens (Table 9). We detected asperglaucide, its precursor aurantiamide and/or asperphenamate in all species in sect. Restricti (Table 9). Other extrolites detected include antarone A, asperentins, clavatols, cyclopaldic acid, orthosporins, naphtho-gamma-pyrones and chrysogine. Several other extrolites were detected including indole alkaloids, phthalides, pyrones and some compounds with yellow chromophores (Table 9). Members of the echinulin/neoechinulin/isoechinulin/cristatine biosynthetic family were detected in A. halophilicus, A. hordei, A. magnivesiculatus, A. penicillioides and A. tardicrescens, indicating a strong chemical relationship with Aspergillus section Aspergillus. However, while nearly all isolates in species of section Aspergillus produce the yellow auroglaucin and flaviglaucin-type metabolites (colouring the ascomata), these yellow metabolites were not found in any species of section Restricti.\n",
      "\n",
      "Table 9\n",
      "Species\tStrain number\tExometabolites\n",
      "A. caesiellus\tNRRL 5061\tAsperphenamate, indole alkaloid “A” & “B”, CADA1, CADA2\n",
      "DTO 025-I4 = IBT 34538\tAsperphenamate, CAS1-CAS5\n",
      "CCF 5448 = EMSL No. 1383 = IBT 34621\tAsperphenamate, indole alkaloid “A”, CA1, CA2, CAD1, CADA2, CAS1-CAS4\n",
      "CCF 5451 = EMSL No. 1650 = IBT 34622\tAsperphenamate, indole alkaloid “A” & “B”, CA1, CAS1-CAS5\n",
      "A. canadensis\tCCF 5548\tAsperglaucide, aurantiamide\n",
      "KAS 7710 = DTO 357-B4 = CCF 5552 = IBT 34636\tAsperglaucide, aurantiamide, indole alkaloid “A”\n",
      "KAS 7708 = DTO 357-B2 = CCF 5550 = IBT 34637\tasperglaucide, indole alkaloid “A”, ASPHAM1\n",
      "KAS 7719 = DTO 357-C2 = CCF 5553 = IBT 34638\tAsperglaucide, aurantiamide , indole alkaloid “A”, ASPHAM1, CANA1-CANA4\n",
      "KAS 7706 = DTO 357-A9 = CCF 5549 = IBT 34639 = NRRL 66615\tAsperglaucide, aurantiamide , indole alkaloid “A”, ASPHAM1\n",
      "KAS 7709 = DTO 357-B3 = CCF 5551 = IBT 34640\tAsperglaucide, asperphenamate, indole alkaloid “A”\n",
      "KAS 7720 = DTO 357-C3 = IBT 34641\tAsperglaucide, aurantiamide\n",
      "A. clavatophorus\tNRRL 25874\tAsperglaucide, aurantiamide, indole alkaloid “A”\n",
      "DTO 257-G5 = CCF 5669 = IBT 34561\tAsperglaucide, PRO1, Raistrick phenol “A”\n",
      "NRRL 25873 = CCF 5453 = IBT 34632\tAsperglaucide, aurantiamide, PRO1, Raistrick phenol “B”\n",
      "A. conicus\tNRRL 149\tAsperglaucide, asperphenamate, indole alkaloid “A”, phthalide\n",
      "DTO 110-F5 = CCF 5667 = IBT 34534\tAsperphenamate, phthalide, PRO2\n",
      "EXF-7660 = IBT 34263 = IBT 33577\tAsperphenamate, phthalide, WOVI, PITAL\n",
      "EXF-7663 = IBT 34267 = IBT 33574\tAsperphenamate, phthalide, clavatol “A”, FOLS, indole alkaloid “B”, PITAL\n",
      "EXF-5015 = CCF 5650 = IBT 34273\tAsperphenamate\n",
      "EXF-10387 = CCF 5651 = IBT 34289\tAsperglaucide, PI-4 (fulvic acid biosynthetic family)\n",
      "A. destruens\tNRRL 145\tAsperglaucide, chromanol “X”, naphtho-gamma-pyrone “A”, YELLO\n",
      "DTO 254-B2 = IBT 34522\tAsperphenamate, clavatol “B” & “C”, pyrone “A”, “B”, “C”, “D”, “E”, “F”, “G”, “H” & “I”, WOVI\n",
      "EXF-7657 = IBT 34264\tAsperphenamate, pyrone “A”, pyrone “G”, WOVI\n",
      "EXF-7703 = IBT 34259\tAsperphenamate, clavatol “D”\n",
      "EXF-7651 = CCF 5653 = IBT 34258\tAsperphenamate, pyrone “B”, “D” & “G”, clavatol “A”\n",
      "EXF-7699 = IBT 34262\tAsperphenamate, pyrone “A”, pyrone “B” & “G”, clavatol “A”\n",
      "EXF-7666 = IBT 34268\tAsperphenamate\n",
      "EXF-7661 = IBT 34271 = IBT 33573\tAsperphenamate, clavatol “A”, ASPHAM1, CAS1, CAS2, naphtho-gamma-pyrone “A”\n",
      "EXF-7667 = IBT 34288\tAsperphenamate, pyrone “G”\n",
      "EXF-10431 = IBT 34257\tAsperphenamate, pyrone “A”, “B” & “G”\n",
      "EXF-10411 = IBT 34265\tAsperphenamate, pyrone “A”, “B”, “E” & “G”\n",
      "EXF-10350 = IBT 34270\tAsperphenamate, WOVI\n",
      "EXF-10407 = CCF 5652 = IBT 34285\tAsperglaucide\n",
      "EXF-10017 = IBT 33576\tAsperphenamate, ASPHAM1, naphtho-gamma-pyrone “A”, WOVI\n",
      "EXF-10390 = IBT 34276\tAsperphenamate, clavatol “A”, SKIZ1, SKIZ2, CAS5\n",
      "A. domesticus\tDTO 079-F2\tAsperphenamate, clavatol “D”, FOLS\n",
      "CBS 233.90 = DTO 079-A7 = IBT 34515\tAsperphenamate, WOVI\n",
      "CBS 116419 = DTO 077-G6 = IBT 34535\tAsperphenamate, indole alkaloid “B”, FOLS\n",
      "EXF-10012 = IBT 34274\tAsperphenamate, clavatol “D”, FOLS, indole alkaloid “B”\n",
      "EXF-10583 = IBT 34275\tAsperphenamate, indole alkaloid “B”\n",
      "A. glabripes\tCCF 5474\tSCUB, STALD1, STALD2, FJOS, indole alkaloid “A”, PRO1\n",
      "CCF 5469 = EMSL No. 1483 = DTO 356-E5 = IBT 34519 = IBT 34821 = NRRL 66619\tMycophenolic acid, FAESI, GUOY, GUOY2, PRO1, SCUB, SPOJN1, SPOJN2, STALD1-STALD3, TERRIT1-TERRIT3, XANS\n",
      "UBOCC-A-116021 = CCF 5467 = IBT 34625\tIndole alkaloid “A”, PITAL\n",
      "CCF 5473 = EMSL No. 2442 = IBT 34626\tAsperphenamate, indole alkaloid “A”, PRO1, SCUB\n",
      "IBT 26603\tSCUB, STALD1, STALD2, FJOS, indole alkaloid “A”, PRO1\n",
      "A. gracilis\tNRRL 4962\tAsperphenamate, CHE1-CHE3, chromanol “X”, clavatol “D”, indole alkaloid “B”, FOL1, FOL2, FOLS, SCAB\n",
      "CCF 5479 = EMSL No. 2775 = DTO 356-F4 = IBT 34559\tAsperphenamate, mycophenolic acid, FOL2, FOLS, indole alkaloid “B”, SCAB\n",
      "CCF 5482 = EMSL No. 2923 = IBT 34623\tIndole alkaloid “A”, SCAB\n",
      "A. halophilicus\tNRRL 2739\tAsperphenamate, echinulin, preechinulin, cristatin A, pyrone “R”, “U”, “V”, “W”, “X”, “Z”, cristatin “X”, DYLD1-DYLD4, PITAL\n",
      "DTO 271-F4 = CCF 5825 = IBT 34881\tAsperphenamate, echinulin, preechinulin, cristatin A, trace of mycophenolic acid, pyrone “K”, “L”, “M”, “N”, “O”, “P”, “R”, “S” & “T”, indole alkaloid “A”, cristatine “X”, PITAL\n",
      "MUT 562\tAsperphenamate, cristatin A, echinulin, pyrone “K”, “N”, “U”, “V”, “W”, “X”, “R”, DYLD1-DYLD4, PITAL\n",
      "MUT 1305\tAsperphenamate, cristatin A, echinulin, pyrone “K”, “N”, “R”, “V”, “W”, “X”, DYLD1-DYLD4, PITAL\n",
      "MUT 798\tAsperphenamate, cristatin A, echinulin, pyrone “K”, “M”, “N”, “O”, “P”, “R”, DYLD1-DYLD4, PITAL\n",
      "MUT 1307\tAsperphenamate, cristatin A, echinulin, pyrone “K”, “N”, “R”, “U”, “V”, “W”, “X”, “Y”, “Z”, DYLD1-DYLD4, PITAL\n",
      "A. hordei\tNRRL 25825\tAsperglaucide, aurantiamide, aurantiamide “B”, cristatine A, echinulin, indole alkaloid “C”, “D” & “E”\n",
      "DTO 281-A4 = IBT 34517\tAsperglaucide, aurantiamide, chromanol “X”, cristatin A, echinulin, YEY\n",
      "NRRL 25830 = CCF 5485 = IBT 34631\tAsperglaucide, aurantiamide, aurantiamide “B”, echinulin, cristatin A, KOS3, PENIT\n",
      "A. infrequens\tNRRL 25868\tAsperphenamate\n",
      "A. magnivesiculatus\tNRRL 25866\tAsperglaucide, cristatin A, echinulin, indole alkaloid “A”, KOS1-KOS3\n",
      "CCF 5489 = EMSL No. 1315 = DTO 356-E2 = IBT 34516\tAntarone A, asperglaucide, cristatin A, echinulin, FU1, FU2, KOS1-KOS3, FUSM1, FUSM2\n",
      "EXF-10353 = IBT 34284\tAsperglaucide, pyrone “A”, “B”, “D” & “G”\n",
      "EXF-10356 = IBT 34279\tAsperglaucide, pyrone “G” & “I”\n",
      "EXF-10347 = IBT 34283\tAsperglaucide\n",
      "A. pachycaulis\tNRRL 25824\tAsperphenamate, indole alkaloid “A” & “B”, phthalide, mycophenolic acid, ASPHAM2, CACCA, OSM1, OSM2, clavatol “D”, FIMUI, naphtho-gamma-pyrone “C” & “D”\n",
      "CCF 5493 = EMSL No. 2310 = DTO 356-E6 = IBT 34536\tAsperphenamate, phthalide, indole alkaloid “A” and “B”\n",
      "A. penicillioides\tNRRL 4548\tAsperglaucide, aurantiamide\n",
      "DTO 267-A4 = IBT 34614\tAsperglaucide, cristatine A, trace of mycophenolic acid, YEY, echinulin\n",
      "CCF 5498 = EMSL No. 2440 = DTO 356-E7 = IBT 34815\tAsperglaucide\n",
      "CCF 5497 = EMSL No. 2430 = IBT 34628\tAsperglaucide, aurantiamide, cristatine A, YEY\n",
      "CCF 5501 = EMSL No. 2749 = IBT 34629\tAsperglaucide, YEY, echinulin, met k\n",
      "CCF 5500 = EMSL No. 2651 = IBT 34630\tAsperglaucide, YEY\n",
      "A. pseudogracilis\tCCF 5505\tAsperphenamate, indole alkaloid “B”, FOLS\n",
      "A. restrictus\tNRRL 154\tAsperphenamate, CACCA, indole alkaloid “A” & “B”\n",
      "DTO 065-C7 = IBT 34541\tAsperphenamate, indole alkaloid “B”, CRYPT1, CRYPT2, FNOVL1, FNOVL2\n",
      "CCF 5511 = EMSL No. 1675 = IBT 34616\tAsperphenamate, FNOVL2, FOLS, indole alkaloid “B”\n",
      "CCF 3364 = IBT 34619\tAsperphenamate, CRYPT1, CRYPT2, CRYPT4, FNOVL1, FNOVL2, FOLS, indole alkaloid “B”\n",
      "CCF 5507 = EMSL No. 1379 = IBT 34618\tAsperphenamate, indole alkaloid “B”, FOLS\n",
      "CCF 5512 = EMSL No. 2206 = IBT 34617\tAsperphenamate, CRYPT1-CRYPT5, FNOVL1-FNOVL3, PALO1-PALO3\n",
      "A. reticulatus\tNRRL 25852\tAsperglaucide, aurantiamide, indole alkaloid “A”, clavatol “D”\n",
      "CCF 5521 = EMSL No. 1362 = DTO 356-E4 = IBT 34518\tAsperglaucide, aurantiamide, indole alkaloid “A”, PITAL\n",
      "CCF 5524 = EMSL No. 2548 = IBT 34637\tAsperglaucide, aurantiamide, clavatol “D”\n",
      "CCF 3112 = IBT 34634 = NRRL 62490\tAsperglaucide, aurantiamide, indole alkaloid “A”, cyclopaldic acid chromanols\n",
      "CCF 5525 = NRRL 58572 = EMSL No. 885 = IBT 34880\tAsperglaucide, cyclopaldic acid, cyclopaldic acid chromanols\n",
      "CCF 5518 = NRRL 58903 = EMSL No. 1272 = IBT 34819\tAsperglaucide, aurantiamide, indole alkaloid “A”, cyclopaldic acid chromanols\n",
      "A. salinicola\tEXF-10401\tAntarone A, asperentin “A”, “B”, “C”, “D”, “E”, “F”, “G”, “H” & “I”, asperglaucide, aurantiamide, met k, SOSO\n",
      "EXF-226 = CCF 5527 = IBT 34277 = NRRL 66622\tAsperglaucide, aurantiamide, indole alkaloid “A”\n",
      "UBOCC-A-116019 = CCF 5528 = IBT 34635\tAsperglaucide, aurantiamide\n",
      "A. tardicrescens\tDTO 316-B5\tAsperglaucide, mycophenolic acid, clavatol “D”, FOL2, indole alkaloid “A”\n",
      "DTO 316-A8 = IBT 34562\tAsperglaucide, mycophenolic acid, FOL2, KOS3, indole alkaloid “A”, indole alkaloid “F”\n",
      "EXF-10454 = IBT 34281 = CCF 5530 = NRRL 66624\tAsperfuran, asperglaucide, echinulin, indole alkaloid “D”, FOL2, KOS3\n",
      "EXF-10456 = IBT 34286\tAsperglaucide\n",
      "A. villosus\tNRRL 25813\tAsperphenamate, orthosporin “A”, “A1”, “D”, “F”, “G”, “H”, “I”, “J”, “K”, “L”, “M”, “N”, “O”, “P”, “Q”, “R” & “S”, MOL\n",
      "UBOCC-A-116020 = CCF 5532 = IBT 34624\tIndol alkaloid “A”\n",
      "IBT 22529, IBT 22530, IBT 22546\tAsperphenamate, orthosporin “A”, “F”, “G”, “O”, “R” & “S”\n",
      "A. vitricola\tNRRL 5125\tNo exometabolites detected\n",
      "KAS 6150 = DTO 356-H5 = IBT 34531\tOrthosporin “A”, “B”, “C”, “D” & “E”\n",
      "KAS 6087 = DTO 356-H2 = IBT 34532\tIndole alkaloid “G”\n",
      "EXF-7700 = CCF 5657 = IBT 34282 = IBT 33575\tAsperglaucide, YELL\n",
      "EXF-10383 = CCF 5655 = IBT 34272\tAsperglaucide, Raistrick phenol “A”\n",
      "EXF-10301 = CCF 5654 = IBT 34290\tDOKU1, DOKU2\n",
      "\n",
      "1\n",
      "\n",
      "2\n",
      "\n",
      "3\n",
      "\n",
      "4\n",
      "\n",
      "5\n",
      "\n",
      "6\n",
      "\n",
      "\n",
      "Taxonomy\n",
      "\n",
      "\n",
      "Aspergillus section Restricti (Gams et al. 1985), Adv. Penicillium Aspergillus Syst.: 56. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: Aspergillus restrictus G. Sm., J. Textile Inst. 22: T115. 1931. ",
      " ",
      "\n",
      "\n",
      "The section includes 20 anamorphic species and one homothallic species. Xerophilic, growing optimally either on substrates containing high concentrations of sugar or salt or with low water content. Young conidial heads columnar (A. restrictus and A. conicus clade), radiate (A. vitricola clade) or globose (A. penicillioides clade). Conidiophores uniseriate with hyaline stipes, usually villose in SEM. Vesicles pyriform, subglobose or clavate, fertile over one to two thirds of the surface. Phialides flask-shaped, usually villose in SEM. Conidia ellipsoidal to ovate, barrel-shaped or subglobose, rough-walled; aculeate, tuberculate or lobate-reticulate in SEM. Ascomata present only in A. halophilicus, cleistothecial, hyaline to pale yellow, globose to subglobose. Asci globose to subglobose. Ascospores hyaline, lenticular with two equatorial crests, convex surface finely roughened in the area neighbouring the equatorial region, dimpled in SEM.\n",
      "\n",
      "\n",
      "Aspergillus restrictus clade\n",
      "\n",
      "\n",
      "Aspergillus restrictus, A. caesiellus and A. pachycaulis are distinguished from all other members of sect. Restricti by compact columnar conidial heads, different growth parameters in an osmotic gradient (compared to other clades, they grow faster on media with high water activity, e.g. MEA and MEA + 5 % NaCl and slower on MEA + 20 % NaCl and MEA + 25 % NaCl) and rapid growth and deep green sporulation on CY20S. ",
      " ",
      "\n",
      "\n",
      "Accepted species:\n",
      "\n",
      "\n",
      "Aspergillus caesiellus Saito, J. Fac. Sci. Coll. Imper. Univ. Tokyo 18: 49. 1904. [MB205025].\n",
      "\n",
      "\n",
      "Aspergillus pachycaulis F. Sklenar, S.W. Peterson, Ž. Jurjević & Hubka, this study. [MB823048].\n",
      "\n",
      "\n",
      "Aspergillus restrictus G. Sm., J. Textile Inst. 22: T115. 1931. [MB276290].\n",
      "\n",
      "\n",
      "Aspergillus conicus clade\n",
      "\n",
      "This clade contains six species (A. conicus, A. domesticus, A. destruens, A. villosus, A. gracilis and A. pseudogracilis). Members of the A. conicus clade have loosely columnar conidial heads, while species from the closely related A. vitricola clade have radiate heads. Several characters are useful for distinguishing between members of the A. conicus and A. penicillioides clade. The colour of sporulation of A. conicus clade species is usually paler green, colony reverse colour on M40Y is usually olive grey compared to the brown or red reverse of A. penicillioides clade species. Further, the members of A. penicillioides clade produce white cottony secondary mycelium on M60Y in contrast to A. conicus clade and conidial heads are usually globose. Conidia of A. conicus clade species in SEM are usually aculeate, whereas conidia of A. penicillioides clade species are tuberculate or lobate-reticulate. The differences from A. restrictus clade were described above (see distinguishing characters of A. restrictus). ",
      " ",
      "\n",
      "\n",
      "Accepted species:\n",
      "\n",
      "\n",
      "Aspergillus conicus Blochwitz in Dale, Ann. Mycol. 12: 38. 1914. [MB120214].\n",
      "\n",
      "\n",
      "Aspergillus destruens Zalar, F. Sklenar, S.W. Peterson & Hubka, this study. [MB818930].\n",
      "\n",
      "\n",
      "Aspergillus domesticus F. Sklenar, Houbraken, Zalar & Hubka, this study. [MB818931].\n",
      "\n",
      "\n",
      "Aspergillus gracilis Bainier, Bull. Soc. Mycol. Fr. 23: 92. 1907. [MB167554].\n",
      "\n",
      "\n",
      "Aspergillus pseudogracilis F. Sklenar, Ž. Jurjević & Hubka, this study. [MB818932].\n",
      "\n",
      "\n",
      "Aspergillus villosus F. Sklenar, S.W. Peterson & Hubka, this study. [MB818933].\n",
      "\n",
      "\n",
      "Aspergillus vitricola clade\n",
      "\n",
      "This clade contains two species, A. vitricola and A. glabripes, and is defined by their radiate conidial heads, smooth stipes and phialides in SEM (unique through the whole section), and aculeate conidia in SEM. ",
      " ",
      "\n",
      "\n",
      "Accepted species:\n",
      "\n",
      "\n",
      "Aspergillus vitricola Ohtsuki, Bot. Mag. (Tokyo) 75: 436. 1962. [MB326665].\n",
      "\n",
      "\n",
      "Aspergillus glabripes F. Sklenar, Ž. Jurjević & Hubka, this study. [MB818934].\n",
      "\n",
      "\n",
      "Aspergillus penicillioides clade\n",
      "\n",
      "This clade encompasses nine species (Fig. 7). Aspegillus tardicrescens, A. clavatophorus, A. magnivesiculatus and A. hordei are closely related to A. penicillioides, while the remaining four species are relatively distant. The core species can be considered the most xerophilic members of the section, and generally some of the most xerophilic fungi. Conidial heads of these species are always globose when young and sporulation colour is usually darker compared to species from other clades. Stipes are covered by short hairs observed with SEM and conidia are usually born as ellipsoidal (not cylindrical) and have tuberculate or lobate-reticulate ornamentation in SEM. ",
      " ",
      "\n",
      "\n",
      "Accepted species:\n",
      "\n",
      "\n",
      "Aspergillus canadensis Visagie, Yilmaz, F. Sklenar & Seifert, this study. [MB818935].\n",
      "\n",
      "\n",
      "Aspergillus clavatophorus F. Sklenar, S.W. Peterson & Hubka, this study. [MB818936].\n",
      "\n",
      "\n",
      "Aspergillus hordei F. Sklenar, S.W. Peterson & Hubka, this study. [MB818937].\n",
      "\n",
      "\n",
      "Aspergillus infrequens F. Sklenar, S.W. Peterson & Hubka, this study. [MB818938].\n",
      "\n",
      "\n",
      "Aspergillus magnivesiculatus F. Sklenar, Zalar, Ž. Jurjević & Hubka, this study. [MB818939].\n",
      "\n",
      "\n",
      "Aspergillus reticulatus F. Sklenar, Ž. Jurjević, S.W. Peterson & Hubka, this study. [MB818940].\n",
      "\n",
      "\n",
      "Aspergillus salinicola Zalar, F. Sklenar, Visagie & Hubka, this study. [MB818941].\n",
      "\n",
      "\n",
      "Aspergillus tardicrescens F. Sklenar, Houbraken, Zalar & Hubka, this study. [MB818942].\n",
      "\n",
      "\n",
      "Aspergillus penicillioides Speg., Revista Fac. Agron. Univ. Nac. La Plata 2: 245. 1896. [MB309234].\n",
      "\n",
      "\n",
      "Aspergillus halophilicus clade\n",
      "\n",
      "This clade contains only one species, Aspergillus halophilicus. It is easily recognisable, because it is the only species from the section that produces sexual reproductive state. ",
      " ",
      "\n",
      "\n",
      "Accepted species:\n",
      "\n",
      "\n",
      "Aspergillus halophilicus C.M. Chr., Papav. & C.R. Benj., Mycologia 51: 636. 1959. [MB326633].\n",
      "\n",
      "\n",
      "Species descriptions\n",
      "\n",
      "\n",
      "Aspergillus caesiellus Saito, J. Fac. Sci. Coll. Imper. Univ. Tokyo 18: 49. 1904, emended description. MycoBank MB205025. Fig. 14.\n",
      "\n",
      "Fig. 14 Aspergillus caesiellus. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 14\n",
      "\n",
      "\n",
      "Synonyms: Aspergillus gracilis var. sartoryi Bat., O.G. Lima & A.F. Vital, Mycopath. Mycol. Appl. 8: 96. 1957.\n",
      "\n",
      "\n",
      "Aspergillus restrictus mut. eborinus G. Sm., Trans. Brit. Mycol. Soc. 44: 45. 1961. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: Japan, unknown substrate, unknown year of isolation, isolated by K. Saito [neotype Herb. IMI 172278 designated by Samson & Gams (1985) and modified by Pitt & Samson (1993), culture ex-type NRRL 5061 = CBS 470.65 = DTO 093-H3 = ATCC 11905 = IMI 172278 = CCF 5447 = IBT 34620]. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 58–60; CYA 11–22; CY20S 21–35; CY20S 30 °C 12–40; CY20S 37 °C No growth; DG18 32–38; MEA 17–22; M60Y 48–60; M60Y 30 °C 45–60; M60Y 37 °C 32–55; MEA + 10 % NaCl 36–45. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: flat; colony surface floccose; margin entire; mycelial areas white; sporulation greyish turquoise (24E4); soluble pigment absent; exudate absent; reverse centrally sunburn light brown (6D5) to greyish orange (5B4) in margins. CYA 25 °C, 14 d: with central depression; colony surface floccose, margin entire; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally dull blue (23E4) to orange white (5A2) in margins. CY20S 25 °C, 14 d: raised; colony surface floccose; margin entire to undulate; mycelial areas white; sporulation jungle green (25F5) to dull green (25E4); soluble pigment absent; exudate absent; reverse centrally baby blue (bluish grey) (23B3) to orange grey (5B2) in margins. DG18 25 °C, 14 d: centrally raised; colony surface velvety to floccose in the centre, margin entire to delicately filiform; mycelial areas white; sporulation dull green (25D4); soluble pigment absent; exudate absent; reverse centrally birch grey (5C2) to orange grey (5B2) in margins. MEA 25 °C, 14 d: with central depression, irregulary wrinkled; colony surface floccose, margin delicately undulate; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally dark blonde (5D4) to greyish orange (5B3) in margins. M60Y 25 °C, 14 d: flat; colony surface filamentous; margin filiform; mycelial areas white; sporulation dull green (27E3); soluble pigment absent; exudate absent; reverse pale yellow (3A3). MEA + 10 % NaCl 25 °C, 14 d: centrally raised; colony surface floccose; margin irregular; mycelial areas white; sporulation dull green (26D3); soluble pigment absent; exudate absent; reverse centrally light yellow (4A4) to pastel yellow (3A4) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads columnar, frequently sinuous. Conidiophores uniseriate. Stipes smooth, densely covered by long hairs in SEM, occasionally with septa, 5–7.5 μm wide. Vesicles pyriform to clavate, 11–16 μm in diameter. Phialides flask-shaped, densely covered by long hairs in SEM, 8–9 μm long, covering one third to one half of the vesicle. Conidia borne as smooth cylinders, at maturity ellipsoidal to ovate, connectives evident, echinulate, aculeate in SEM, 4.5–6 × 3–4.5 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: See distinguishing characters of A. restrictus. The stipes of A. caesiellus are usually wider than those of A. restrictus and narrower than those of A. pachycaulis, vesicles of A. caesiellus are usually larger than those of A. restrictus and smaller than those of A. pachycaulis. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Taxonomic notes: The LSU region of the ex-type strain of A. gracilis var. sartoryi (NRRL 4745; GenBank U29642) is identical with the ex-type strain of A. caesiellus and confirms conclusions of Raper and Fennell (1965) based on morphology.\n",
      "\n",
      "Authentic strain of A. restrictus mut. eborinus NRRL 4756 is conspecific with A. caesiellus based on molecular data. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Germany, air, unknown year of isolation, collector unknown, DTO 077-H6 = CBS 117330. Germany, indoor environment, 2006, collector unknown, DTO 026-C7, DTO 025-I4 = IBT 34538. India, Gorakpur, unknown substrate, before 1974, isolated by T. N. Bhargana, NRRL 25861. Indonesia, corn kernels, unknown year of isolation, collector unknown, DTO 065-D7. Panama, cloth, 1945, isolated by W. H. Weston, NRRL 25810 = CCF 5662. The Netherlands, wood of crate, imported from China, unknown year of isolation, isolated by J. Houbraken, DTO 060-H9. USA, Delaware, outside air, 2012, isolated by Ž. Jurjević, CCF 5450 = EMSL No. 1614. USA, Pennsylvania, home air, 2010, isolated by Ž. Jurjević, CCF 5448 = EMSL No. 1383 = IBT 34621. USA, Delaware, air, pineapple room, warehouse, 2012, isolated by Ž. Jurjević, CCF 5451 = EMSL No. 1650 = IBT 34622. USA, Delaware, home air, 2011, isolated by Ž. Jurjević, CCF 5449 = EMSL No. 1499. USA, Florida, hobnail shoes, 1944, isolated by W. H. Weston, NRRL 25815 = CCF 5663 = DTO 356-D1. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus canadensis Visagie, Yilmaz, F. Sklenar & Seifert, sp. nov. Mycobank MB818935. Fig. 15. ",
      " ",
      "\n",
      "\n",
      "Fig. 15 Aspergillus canadensis. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 15\n",
      "\n",
      "\n",
      "Etymology: Named after country of origin. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: Canada, Nova Scotia, Wolfville, house dust, 2015, collected by A. Walker, isolated by C.M. Visagie, (holotype DAOM 740109, culture ex-type CCF 5548 = KAS 6194 = DTO 356-H9 = IBT 34520 = IBT 34642 = NRRL 66614). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 25–29; CYA No growth; CY20S (0)10–11; CY20S 30 °C No growth; CY20S 37 °C No growth; DG18 7–15; MEA No growth; M60Y 23–42; M60Y 30 °C 2–20; M60Y 37 °C No growth; MEA + 10 % NaCl 17–33. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies umbonate, irregularly wrinkled, surface velutinous; margin delicately filiform; mycelial areas white; sporulation jungle green (25F5) to pale turquoise (24A3); soluble pigment absent; exudate absent; reverse centrally olive grey (2F2), corn greyish yellow (4B5) to champagne greyish yellow (4B4) in margins. CYA 25 °C, 14 d: no growth. CY20S 25 °C, 14 d: cerebriform, wrinkled; colony surface velutinous, margin undulate; mycelial areas white; sporulation pale turquoise (24A3); soluble pigment absent; exudate absent; reverse centrally dark green (27F7) to greenish grey (25D2), orange white (5A2) in margins. DG18 25 °C, 14 d: centrally raised, radially wrinkled, colony surface velutinous, margin undulate; mycelial areas white; sporulation turquoise grey (24B2); soluble pigment absent; exudate absent; reverse centrally orange grey (5B2) to orange white (5A2) in margins. MEA 25 °C, 14 d: no growth. M60Y 25 °C, 14 d: umbonate; colony surface floccose; margin delicately filiform; mycelial areas white; sporulation greyish green (25C5); soluble pigment absent; exudate absent; reverse centrally olive green (2F8) to mustard yellow (3B6), yellowish white (4A2) in the margins. MEA + 10 % NaCl 25 °C, 14 d: flat to umbonate, irregularly wrinkled; colony surface velutinous; margin entire to delicately filiform; mycelial areas white; sporulation dull green (25D4) to greyish green (25B4); soluble pigment absent; exudate absent; reverse centrally brownish grey (5F2) to mouse grey (5E3), champagne greyish yellow (4B4) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads radiate. Conidiophores uniseriate. Stipes smooth, sparsely covered by bundles of long hairs in SEM, 3.5–5 μm wide in the middle part, non-septate. Vesicles pyriform to clavate, (9–)13–16(–20) μm wide. Phialides flask-shaped, sparsely covered by long hairs in SEM, 6–8 μm long, covering two thirds of the vesicle. Conidia borne smooth and ellipsoidal, becoming rough-walled, subglobose or barrel-shaped at maturity, lobate-reticulate in SEM, commonly remaining in chains, connectives evident, 3–4 × 2.5–3 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: Aspergillus canadensis and A. infrequens are phylogenetically distinct from the core species in the A. penicillioides clade. Aspergillus canadensis is unable to grow on CYA and MEA at 25 °C and grows much slower on M60Y at 30 °C than remaining species from the A. penicillioides clade. The phialides of this species are the shortest across the whole section. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Canada, Ontario, Ottawa, house dust, 2015, collected by J. Mack, isolated by C.M. Visagie, DAOMC 251505 = KAS 6095, DAOMC 251506 = KAS 6269, KAS 7704 = DTO 357-A7, KAS 7705 = DTO 357-A8, CCF 5549 = KAS 7706 = DTO 357-A9 = IBT 34639 = NRRL 66615, KAS 7707 = DTO 357-B1, CCF 5550 = KAS 7708 = DTO 357-B2 = IBT 34637, CCF 5551 = KAS 7709 = DTO 357-B3 = IBT 34640, CCF 5552 = KAS 7710 = DTO 357-B4 = IBT 34636, KAS 7711 = DTO 357-B5, KAS 7716 = DTO 357-B8, KAS 7717 = DTO 357-B9, KAS 7718 = DTO 357-C1, CCF 5553 = KAS 7719 = DTO 357-C2 = IBT 34638, KAS 7720 = DTO 357-C3 = IBT 34641, KAS 7721 = DTO 357-C4. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Note: The majority of strains of A. canadensis quickly degenerated in culture and changed significantly their phenotype (restricted growth, waxy colonies with poor or no sporulation). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus clavatophorus F. Sklenar, S.W. Peterson & Hubka, sp. nov. MycoBank MB818936. Fig. 16. ",
      " ",
      "\n",
      "\n",
      "Fig. 16 Aspergillus clavatophorus. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 16\n",
      "\n",
      "\n",
      "Etymology: Refers to the clavate vesicles of the conidiophores ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: USA, Georgia, Athens, mouldy paper, 1987, isolated by R. T. Hanlin (holotype PRM 944440, isotype PRM 944441, culture ex-type NRRL 25874 = CCF 5454 = IBT 34560 = IBT 34823 = DTO 356-D8). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 27–33; CYA No growth; CY20S 7–16; CY20S 30 °C No growth; CY20S 37 °C No growth; DG18 18–25; MEA No growth; M60Y 35–44; M60Y 30 °C 29–54; M60Y 37 °C (0)5–17; MEA + 10 % NaCl 18–28. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies raised to umbonate, surface floccose to velutinous; margin entire to delicately filiform; mycelial areas white; sporulation greyish green (28D6); soluble pigment absent; exudate absent; reverse centrally coffee brown (5F7), golden brown (5D7) to pale yellow (4A3) in margins. CYA 25 °C, 14 d: no growth to microcolonies. CY20S 25 °C, 14 d: convex, radially wrinkled; colony surface floccose, margin undulate; mycelial areas white; sporulation rare, sand greyish yellow (4B3); soluble pigment absent; exudate absent; reverse centrally yellowish grey (4B2) to orange white (5A2) in margins. DG18 25 °C, 14 d: centrally raised; colony surface floccose to cottony, margin entire to delicately filiform; mycelial areas white; sporulation aquamarine greyish green (25B4) to pale green (27A3); soluble pigment absent; exudate absent; reverse centrally greyish turquoise (24C3) to orange white (5A2) in margins. MEA 25 °C, 14 d: no growth. M60Y 25 °C, 14 d: raised; colony surface floccose to lanose; margin entire to delicately filiform; mycelial areas white; sporulation greyish green (25D6); soluble pigment absent; exudate absent; reverse centrally dark brown (6F6) to golden brownish orange (5C6), pale yellow (4A3) in margins. MEA + 10 % NaCl 25 °C, 14 d: flat to umbonate; colony surface powdery to floccose; margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (water blue) (24C5); soluble pigment absent; exudate absent; reverse centrally coffee brown (5F7), olive brown (4E5) to yellowish orange (4B7), cream pale yellow (4A3) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads at first globose, later becoming radiate. Conidiophores uniseriate. Stipes smooth, densely covered by short hairs in SEM, 6–9 μm wide in the middle part, widening gradually toward the vesicle, non-septate. Vesicles spatulate to clavate, 14–19 μm wide. Phialides flask-shaped, densely covered by short hairs in SEM, 8–10 μm long, covering two thirds to three quarters of the vesicle. Conidia borne smooth, ellipsoidal, becoming rough-walled, tuberculate in SEM, subglobose, ovate or barrel-shaped at maturity, connectives evident, 4–5 × 3–4 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: See distinguishing characters of A. penicillioides. Isolates of A. clavatophorus have larger conidia (on average 4.8 × 3.5 μm) compared to A. tardicrescens (on average 4 × 2.9 μm) and smaller vesicles (on average 16.5 μm) compared to A. magnivesiculatus (on average 21.4 μm). The shape of the vesicles is usually spatulate to clavate in contrast to all other members of the A. penicillioides clade that have at least in part pyriform vesicles. Isolates of this species can grow on M60Y at 37 °C. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: China, puerh tea, unknown year of isolation, isolated by J. Houbraken, DTO 257-G5 = CCF 5669 = IBT 34561. The Netherlands, unknown substrate, 2014, isolated by M. Meijer, DTO 316-B3. The Netherlands, painting, 2014, isolated by J. Houbraken, DTO 324-G2. USA, Georgia, Athens, mouldy paper, 1987, isolated by R. T. Hanlin, NRRL 25873 = CCF 5453 = IBT 34632. USA, Illinois, Peoria, mouldy cardboard, 1989, collector unknown, NRRL 25875 = CCF 5455. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus conicus Blochwitz in Dale, Ann. Mycol. 12: 38. 1914, emended description. MycoBank MB120214. Fig. 17. ",
      " ",
      "\n",
      "\n",
      "Fig. 17 Aspergillus conicus. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 17\n",
      "\n",
      "\n",
      "Typus: Unknown locality of isolation, unknown substrate, 1924, isolated by P. Biourge [neotype Herb. IMI 172281 designated by Samson & Gams (1985), culture ex-type NRRL 149 = CBS 475.65 = IBT 33667 = DTO 096-H6 = ATCC 16908 = IMI 172281 = CCF 5456]. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 32–40; CYA (0)10–12; CY20S 5–18; CY20S 30 °C No growth; CY20S 37 °C No growth; DG18 18–31; MEA (0)8–10; M60Y 26–40; M60Y 30 °C 26–37; M60Y 37 °C No growth; MEA + 10 % NaCl 16–36. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies flat, wrinkled, surface velutinous; margin filiform; mycelial areas white; sporulation pale turquoise (24A3); soluble pigment absent; exudate absent; reverse centrally olive grey (2F2) to champagne greyish yellow (4B4) in margins. CYA 25 °C, 14 d: crateriform, wrinkled; colony surface velutinous, margin undulate; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally Persian blue (23A3) to yellowish white (4A2) in margins. CY20S 25 °C, 14 d: cerebriform; colony surface velutinous, margin undulate; mycelial areas white; sporulation light turquoise (24A4); soluble pigment absent; exudate absent; reverse dark turquoise (23F7) to greyish brown (7E3), orange white (5A2) in margins. DG18 25 °C, 14 d: flat, wrinkled; colony surface velutinous to filamentous, margin filiform; mycelial areas white; sporulation bluish grey (23B3); soluble pigment absent; exudate absent; reverse centrally greyish turquoise (24D3) to orange white (5A2) in margins. MEA 25 °C, 14 d: cerebriform; colony surface velutinous, margin undulate; mycelial areas white; sporulation Persian blue (23A3); soluble pigment absent; exudate absent; reverse centrally camel light brown (6D4) to sand greyish yellow (4B3) in margins. M60Y 25 °C, 14 d: flat; colony surface velutinous; margin filiform; mycelial areas white; sporulation pale turquoise (24A3); soluble pigment absent; exudate absent; reverse olive grey (2F2) to lead grey (2D2), champagne greyish yellow (4B4) in margins. MEA + 10 % NaCl 25 °C, 14 d: flat; colony surface velutinous; margin filiform; mycelial areas white; sporulation pale turquoise (24A3); soluble pigment absent; exudate absent; reverse centrally greenish grey (27D2) to sand greyish yellow (4B3) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads loosely columnar, frequently twisted. Conidiophores uniseriate. Stipes smooth, sparsely covered by short hairs in SEM, 4–5.5 μm wide in the middle part, widening gradually toward the vesicle, with occasional septa. Vesicles pyriform, spatulate to clavate, 8–13 μm wide. Phialides flask-shaped, sparsely covered by short hairs in SEM, 8.5–10 μm long, covering one third to one half of the vesicle. Conidia borne as smooth long cylinders, then becoming delicately rough to definitely aculeate in SEM, globose to subglobose or barrel-shaped, commonly remaining in chains, connectives evident, 4–4.5 × 2.5–3 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: Aspergillus conicus is most closely related to A. domesticus and A. destruens. The colour of sporulating colonies on M40Y is pale turquoise in A. conicus, while myrtle green in A. destruens; reverse of A. conicus is olive grey in the centre, but apricot greyish orange in A. destruens. These three species share almost identical micromorphology, but differences can be found in their response to the osmotic stress. Aspergillus conicus grows fastest on MEA + 5 % NaCl and MEA + 10 % NaCl, while A. domesticus grows slowest (see Fig. 12E). Aspergillus gracilis and A. pseudogracilis differ from these three species by their inability to grow on MEA and CYA at 25 °C (several strains of A. gracilis exhibited very restricted growth on CYA) and more rapid growth on M60Y at 25 °C and 30 °C. In contrast to A. domesticus, isolates of A. conicus can grow on MEA and grow slightly faster on CYA. Aspergillus villosus can be distinguished from A. conicus, A. domesticus and A. destruens by its faster growth on CY20S, ability to grow on CY20S at 30 °C and on M60Y at 37 °C. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Belgium, candy, 2004, collector unknown, IHEM 20709. Belgium, Braine-l'Alleud, wooden statue, 2000, collector unknown, IHEM 16531. Czech Republic, kernel of Bertholletia excelsa (Brazil nut), unknown year of isolation, isolated by L. Marvanová, CCF 4042. Denmark, near Copenhagen, air in bathroom, 2009, isolated by J. C. Frisvad, DTO 110-C5. Denmark, near Copenhagen, air in living room, 2009, isolated by J. C. Frisvad, DTO 110-F5 = CCF 5667 = IBT 34534. France, French madeleines duchesse, 2013, isolated by F. Déniel, CBS 140429 = UBOCC-A-115041 = DTO 334-D9. Germany, Koln, air, unknown year of isolation, collector unknown, DTO 077-H2 = CBS 117332. Germany, indoor air, 2014, isolated by U. Hack, DTO 321-H2, DTO 321-H5. India, Tamil Nadu, Tiruchirapally, Musa acuminate, unknown year of isolation, DTO 079-A9 = CBS 122.453. Israel, Eliat, microbial mat, 2008, isolated by R. Tkavc, EXF-5015 = CCF 5650 = IBT 34273. Puerto Rico, Bayamon, office air, 2014, isolated by Ž. Jurjević, CCF 5461 = EMSL No. 2549. Slovenia, Ljubljana, oil painting on canvas, 2012, isolated by P. Zalar, EXF-7663 = IBT 34267 = IBT 33574, EXF-7660 = IBT 34263 = IBT 33577. Slovenia, Ljubljana, oil painting on canvas, 2015, isolated by P. Zalar and D.D. Graf, EXF-10387 = CCF 5651 = IBT 34289. The Netherlands, Witten, indoor air, unknown year of isolation, isolated by C. Trautmann and I. Dill, DTO 077-H5. The Netherlands, Zwartewaal, museum piece, 2012, isolated by M. Meijer, DTO 231-C2, DTO 231-C3 = CCF 5665. The Netherlands, Eindhoven, indoor air, 2006, isolated by J. Houbraken, DTO 017-B9. USA, Illinois, Chicago, asphalt roof shingle, 1959, isolated by H. G. Hedrick, NRRL 25848. USA, Delaware, air, pineapple room, warehouse, 2012, isolated by Ž. Jurjević, CCF 5459 = EMSL No. 1649. USA, California, home air, 2011, isolated by Ž. Jurjević, CCF 5458 = EMSL No. 1490. USA, Idaho, home air, 2010, isolated by Ž. Jurjević, CCF 5457 = NRRL 62007 = EMSL No. 1318. USA, New York, unknown substrate, 1975, isolated by I. Weitzman, NRRL 25881. USA, Pennsylvania, West Chester, living room air, 2013, isolated by Ž. Jurjević, CCF 5460 = EMSL No. 2217. USA, Georgia, Kennesaw, basement, 2015, isolated by Ž. Jurjević, EMSL No. 3245. USA, Missouri, Saint Louis, bedroom, wood base, 2015, isolated by Ž. Jurjević, EMSL No. 3209. Unknown locality of isolation, unknown substrate, 2007, isolated by T. van Doorn, DTO 039-G7. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus destruens Zalar, F. Sklenar, S.W. Peterson & Hubka, sp. nov. MycoBank MB818930. Fig. 18. ",
      " ",
      "\n",
      "\n",
      "Fig. 18 Aspergillus destruens. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 18\n",
      "\n",
      "\n",
      "Etymology: Named after the common occurrence of this species on old deteriorated paintings. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: USA, Maryland, maize seed, 1917, isolated by C. V. Piper (holotype PRM 944428, isotype PRM 944429, culture ex-type NRRL 145 = IMI 358691 = CCF 5462 = CBS 593.91 = DTO 079-A8 = IBT 34818). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 19–36; CYA 5–10; CY20S 6–13; CY20S 30 °C No growth; CY20S 37 °C No growth; DG18 14–30; MEA (0) 4–7; M60Y 34–41; M60Y 30 °C 16–41; M60Y 37 °C No growth; MEA + 10 % NaCl 20–30. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies centrally raised and winkled, surface velutinous; margin irregular and delicately filiform; mycelial areas white; sporulation myrtle green (25F7); soluble pigment absent; exudate absent; reverse centrally apricot greyish orange (5B6) to light yellow (4A4) in margins. CYA 25 °C, 14 d: cerebriform; colony surface velutinous, margin undulate; mycelial areas white; sporulation turquoise white (24A2); soluble pigment absent; exudate absent; reverse centrally sky grey (23B2) to cream pale yellow (4A3) in margins. CY20S 25 °C, 14 d: cerebriform to crateriform; colony surface velutinous, margin entire to delicately filiform; mycelial areas white; sporulation light turquoise (24A4); soluble pigment absent; exudate absent; reverse greyish turquoise (24E7) to pale orange (5A3) in margins. DG18 25 °C, 14 d: craterifiorm; colony surface floccose, margin entire to undulate; mycelial areas white; sporulation aquamarine greyish turquoise (23B3); soluble pigment greyish yellow (2B3); exudate absent; reverse centrally dull green (27E3) to yellowish white (4A2) in margins. MEA 25 °C, 14 d: cerebriform; colony surface velutinous, margin entire; sporulation greenish grey (25B2); soluble pigment absent; exudate absent; reverse centrally camel olive grey (2F2) to sand dark yellow (4C8) in margins. M60Y 25 °C, 14 d: umbonate; colony surface floccose to filamentous; margin entire to delicately filiform; mycelial areas white; sporulation pastel green (25B4); soluble pigment absent; exudate absent; reverse cinnamon brown (6D6) to champagne greyish yellow (4B4). MEA + 10 % NaCl 25 °C, 14 d: raised, irregularly wrinkled; colony surface velutinous; margin delicately filiform; mycelial areas white; sporulation bluish grey (23B3); soluble pigment absent; exudate absent; reverse centrally greenish grey (27D2) to cream pale yellow (4A3) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads loosely columnar. Conidiophores uniseriate. Stipes smooth, densely covered by long hairs in SEM, 4–6 μm wide in the middle part, widening gradually toward the vesicle, with occasional septa. Vesicles predominantly ellipsoidal to pyriform, less commonly spatulate to clavate, 8–14 μm wide. Phialides flask-shaped, densely covered by long hairs in SEM, 8.5–11.5 μm long, covering one to two thirds of the vesicle. Conidia borne as smooth, long cylinders, becoming delicately rough and definitely echinulate at maturity, aculeate in SEM, subglobose to ovate, commonly remaining in chains, connectives evident, 3.5–5.5 × 2.5–3.5 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: See distinguishing characters of A. conicus. Aspergillus destruens is most closely related to A. conicus and A. domesticus, but even though the latter two are morphologically similar, they are different from A. destruens. The colour of sporulating colonies of A. destruens on M40Y and M60Y is darker, the reverse colour is apricot greyish orange in the centre in contrast to olive grey in the other two species. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Hungary, indoor air, 2009, collector unknown, DTO 147-E2. Slovenia, Ljubljana, oil painting on canvas, 2015, isolated by P. Zalar and D.D. Graf, EXF-10350 = IBT 34270, EXF-10407 = CCF 5652 = IBT 34285, EXF-10411 = IBT 34265, EXF-10390 = IBT 34276, EXF-10017 = IBT 33576. Slovenia, Ljubljana, oil painting on canvas, 2012, isolated by P. Zalar and D.D. Graf, EXF-7699 = IBT 34262, EXF-7651 = CCF 5653 = IBT 34258, EXF-7661 = IBT 34271 = IBT 33573, EXF-7667 = IBT 34288, EXF-7703 = IBT 34259, EXF-7657 = IBT 34264, EXF-10431 = IBT 34257, EXF-7666 = IBT 34268. The Netherlands, Utrecht, air in villa, unknown year of isolation, collector unknown, DTO 254-B2 = IBT 34522. The Netherlands, Tilburg, air in bakery, 2012, collector unknown, DTO 220-B2. The Netherlands, surface of cheese, 2011, isolated by J. Houbraken, DTO 161-B7 = CCF 5671. The Netherlands, Tilburg, air in bakery, 2009, isolated by J. Houbraken, DTO 113-E7 = CCF 5668. The Netherlands, Leerdam, unknown substrate, unknown year of isolation, isolated by M. Meijer, DTO 303-H2. USA, unknown substrate, 1939, isolated by C.W. Emmons, NRRL 157 = CCF 5463. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus domesticus F. Sklenar, Houbraken, Zalar & Hubka, sp. nov. MycoBank MB818931. Fig. 19. ",
      " ",
      "\n",
      "\n",
      "Fig. 19 Aspergillus domesticus. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 19\n",
      "\n",
      "\n",
      "Etymology: Referring to the common occurrence of this species in indoor environments. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: The Netherlands, Tiel, wallpaper, 2008, collector unknown (holotype PRM 944426, isotype PRM 944427, culture ex-type DTO 079-F2 = CCF 5464 = NRRL 66616 = IBT 34814). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 36–38; CYA 0–8; CY20S 5–17; CY20S 30 °C No growth; CY20S 37 °C No growth; DG18 20–24; MEA No growth; M60Y 29–38; M60Y 30 °C 15–30; M60Y 37 °C No growth; MEA + 10 % NaCl 23–31. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies flat, wrinkled in the centre, surface velutinous; margin filiform; mycelial areas white; sporulation aquamarine greyish turquoise (24B3); soluble pigment absent; exudate absent; reverse centrally olive grey (2F2) to champagne greyish yellow (4B4) in margins. CYA 25 °C, 14 d: crateriform; colony surface velutinous, margin undulate; mycelial areas white; sporulation turquoise white (24A2); soluble pigment absent; exudate absent; reverse centrally fog blue (23C3) to orange white (5A2) in margins. CY20S 25 °C, 14 d: cerebriform; colony surface velutinous, margin entire to delicately filiform; mycelial areas white; sporulation light turquoise (24A5) to turquoise white (24A2); soluble pigment absent; exudate absent; reverse dark turquoise (23F7) to fog blue (23C3), pale orange (5A3) in margins. DG18 25 °C, 14 d: umbonate, delicately wrinkled; colony surface powdery to velutinous, margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (24C3); soluble pigment absent; exudate absent; reverse centrally peacock blue (24D7) to orange white (5A2) in margins. MEA 25 °C, 14 d: no growth. M60Y 25 °C, 14 d: flat; colony surface velutinous to powdery; margin filiform; mycelial areas white; sporulation turquoise grey (24B2); soluble pigment absent; exudate absent; reverse light blond greyish yellow (4C3). MEA + 10 % NaCl 25 °C, 14 d: flat, wrinkled; colony surface velutinous; margin entire to delicately filiform; mycelial areas white; sporulation light turquoise (24A4); soluble pigment absent; exudate absent; reverse centrally nougat greyish brown (5D3) to sand greyish yellow (4B3) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads loosely columnar, commonly twisted. Conidiophores uniseriate. Stipes smooth, densely covered by long hairs in SEM, 4–6 μm wide in the middle part, widening gradually toward the vesicle, with occasional septa. Vesicles pyriform, spatulate to clavate, 7–13 μm wide. Phialides flask-shaped, densely covered by long hairs in SEM, 8.5–11 μm long, covering the upper half of the vesicle. Conidia borne as smooth, long cylinders, becoming echinulate at maturity, aculeate in SEM, ellipsoidal or barrel-shaped, commonly remaining in chains, connectives evident, 3.5–4.5 × 2.5–3.5 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: See distinguishing characters of A. conicus. Aspergillus domesticus is most closely related to A. conicus and A. destruens (Fig. 5). Unlike the two other species, A. domesticus does not grow on MEA and also grows slower on CYA. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Belgium, Brussels, dust from mattress, 1991, collector unknown, IHEM 6549. France, French almond madeleines, 2014, isolated by F. Déniel, UBOCC-A-115040 = CBS 140428 = DTO 334-D8, UBOCC-A-115039 = CBS 140427 = DTO 334-D7. France, air from the bakery oven room, 2014, isolated by F. Déniel, UBOCC-A-115038. France, French mini cakes, 2013, isolated by F. Déniel, UBOCC-A-115048 = CBS 140436 = DTO 334-E7. France, Brest, painting from Brittany (painter: E.J. Damery; painting title: Portrait de femme; painting technique: oil on canvas; and painter: J. Gamelin; painting title: Sainte en extase; painting technique: oil on canvas), 2015, isolated by F. Déniel, UBOCC-A-116022 = CCF 5465. Korea, Phragmidium sp. (rust) on Rubus coreanus, 2004, collected by H.D. Shin, isolated by P. Crous, CBS 116419 = DTO 077-G6 = IBT 34535. Slovenia, Ljubljana, oil painting on canvas, 2015, isolated by P. Zalar, EXF-10583 = IBT 34275. Slovenia, Ljubljana, statue made of wood, textile and sea shells, 2015, isolated by P. Zalar, EXF-10012 = IBT 34274. The Netherlands, Zwartewaal, museum piece, 2012, isolated by M. Meijer, DTO 231-C1 = CCF 5666 = NRRL 66617, DTO 231-B9. The Netherlands, Gorinchem, archive material, 2008, isolated by J. Houbraken and M. Meijer, DTO 086-D1 = CCF 5670. The Netherlands, Paleis het Loo, wallpaper, unknown year of isolation, isolated by J. Varga, CBS 233.90 = DTO 079-A7 = IBT 34515. USA, Idaho, home air, 2010, isolated by Ž. Jurjević, CCF 5466 = EMSL No. 1316. USA, New Jersey, Freehold, breakfast room, settle plates, 2015, isolated by Ž. Jurjević, EMSL No. 2780. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus glabripes F. Sklenar, Ž. Jurjević & Hubka, sp. nov. MycoBank MB818934. Fig. 20. ",
      " ",
      "\n",
      "\n",
      "Fig. 20 Aspergillus glabripes. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 20\n",
      "\n",
      "\n",
      "Etymology: Referring to the absence of fine hairs on the stipe which are visible in majority of sect. Restricti species by SEM. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: Trinidad & Tobago, Macoya, office folder, 2014, isolated by Ž. Jurjević (holotype PRM 944436, isotype PRM 944437, culture ex-type CCF 5474 = DTO 356-E8 = EMSL No. 2462 = NRRL 66618 = IBT 34820). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 38–52; CYA 3–7; CY20S 8–18; CY20S 30 °C 5–16; CY20S 37 °C No growth; DG18 15–25; MEA 4–11; M60Y 35–40; M60Y 30 °C 41–56; M60Y 37 °C (0)7–10; MEA + 10 % NaCl 22–28. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies raised, surface floccose; margin entire to delicately filiform; mycelial areas white; sporulation yellowish green (29C7); soluble pigment absent; exudate absent; reverse centrally dark brown (6F7) to champagne greyish yellow (4B4) in margins. CYA 25 °C, 14 d: cerebriform; colony surface floccose, margin undulate; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse pale orange (5A3). CY20S 25 °C, 14 d: centrally slightly raised; colony surface floccose, margin entire to delicately filiform; mycelial areas white; sporulation yellowish green (29B7); soluble pigment absent; exudate absent; reverse centrally yellowish grey (4D3) to yellowish white (4A2) in margins. DG18 25 °C, 14 d: umbonate; colony surface floccose, margin entire; mycelial areas yellowish white (4A2); sporulation pale green (29A3); soluble pigment absent; exudate absent; reverse centrally yellowish orange (4B7) to yellowish white (4A2) in margins. MEA 25 °C, 14 d: cerebriform; colony surface floccose, margin filiform; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally clay brownish orange (5C5) to pale orange (5A3) in margins. M60Y 25 °C, 14 d: flat; colony surface floccose; margin entire to delicately filiform; mycelial areas white; sporulation deep green (26D8) to yellowish green (29A8); soluble pigment absent; exudate absent; reverse centrally olive brown (4E5) to corn greyish yellow (4B5) in margins. MEA + 10 % NaCl 25 °C, 14 d: flat; colony surface floccose, margin entire to delicately filiform; mycelial areas white; sporulation opal greyish green (25C6) to light green (28A5); soluble pigment absent; exudate absent; reverse centrally olive brown (2F6) to champagne greyish yellow (4B4) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads radiate. Conidiophores uniseriate. Stipes smooth in SEM, 6–8 μm wide in the middle part, widening gradually toward the vesicle, with occasional septa. Vesicles globose, pyriform to subclavate, 16–22 μm wide. Phialides flask-shaped, 7–9 μm long, covering three quarters of the vesicle. Conidia borne as smooth cylinders, echinulate at maturity, aculeate in SEM, subglobose, connectives evident, 3.5–4.5 × 2–3 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: See distinguishing characters of A. vitricola. Aspergillus glabripes can easily be differentiated from A. vitricola by its large vesicles that are most commonly globose or pyriform. Most A. glabripes isolates have a yellowish green colour of sporulation on M40Y and M60Y. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Canada, New Brunswick, Little Lepreau, house dust, 2015, collected by A. Walker, isolated by C.M. Visagie, KAS 6253. France, Brest, painting from Brittany (painter: A. Nozal; painting title: Saint-Briac, grève à marée basse; painting technique: pastel; and painter: B.M. Agüro; painting title: La vision de Saint Antoine de Padoue; painting technique: oil on canvas), 2015, isolated by F. Déniel, UBOCC-A-116021 = CCF 5467 = IBT 34625. France, brownies, 2014, isolated by F. Déniel, UBOCC-A-115046. Trinidad & Tobago, Macoya, office folder, 2014, isolated by Ž. Jurjević, CCF 5475 = EMSL No. 2463, CCF 5476 = EMSL No. 2464, CCF 5477 = EMSL No. 2465. USA, New Jersey, green fabric covered binders, import from China, 2014, isolated by Ž. Jurjević, CCF 5473 = EMSL No. 2442 = IBT 34626. USA, California, home air, 2011, isolated by Ž. Jurjević, CCF 5469 = EMSL No. 1483 = DTO 356-E5 = IBT 34519 = IBT 34821 = NRRL 66619. USA, Louisiana, library, front cover of log book, 2012, isolated by Ž. Jurjević, CCF 5470 = EMSL No. 1812. USA, Louisiana, library, book, 2012, isolated by Ž. Jurjević, CCF 5471 = EMSL No. 1813. USA, Idaho, home air, 2010, isolated by Ž. Jurjević, CCF 5468 = EMSL No. 1317. USA, South Carolina, Summerville, kitchen air, 2014, isolated by Ž. Jurjević, CCF 5472 = EMSL No. 2305. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus gracilis Bainier, Bull. Soc. Mycol. Fr. 23: 92. 1907, emended description. MycoBank MB167554. Fig. 21. ",
      " ",
      "\n",
      "\n",
      "Fig. 21 Aspergillus gracilis. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 21\n",
      "\n",
      "\n",
      "Synonyms: Aspergillus gracilis var. exiguus Bainier & Sartory, Bull. Soc. Mycol. Fr. 28: 47. 1912. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: South Pacific, gun-firing mechanism, unknown year of isolation, R. Emerson (neotype Herb. IMI 211393 designated by Samson & Gams (1985), culture ex-type NRRL 4962 = CBS 539.65 = DTO 351-H7 = CCF 5478 = ATCC 16906 = IMI 211393 = IBT 34817). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 30–42; CYA 0–8; CY20S 7–13; CY20S 30 °C No growth; CY20S 37 °C No growth; DG18 9–16; MEA No growth; M60Y 38–44; M60Y 30 °C 39–45; M60Y 37 °C No growth; MEA + 10 % NaCl 12–24. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies flat, wrinkled in the centre, surface velutinous; margin entire to delicately filiform; mycelial areas white; sporulation aquamarine greyish turquoise (24B3) to turquoise white (24A2); soluble pigment absent; exudate absent; reverse centrally olive grey (2F2) to straw yellow (3B4), cream pale yellow (4A3) in margins. CYA 25 °C, 14 d: no growth to microcolonies. CY20S 25 °C, 14 d: crateriform; colony surface velutinous, margin filiform; mycelial areas white; sporulation turquoise grey (24C2); soluble pigment absent; exudate absent; reverse dull green (25D3) to pale orange (5A3) in margins. DG18 25 °C, 14 d: convex; colony surface floccose to velutinous, margin filiform; mycelial areas white; sporulation greyish turquoise (24D5); soluble pigment absent; exudate absent; reverse centrally yellowish grey (4B2) to orange white (5A2) in margins. MEA 25 °C, 14 d: no growth. M60Y 25 °C, 14 d: raised; colony surface granular to floccose in the centre; margin irregular, filiform; mycelial areas white; sporulation pale turquoise (24A3); soluble pigment absent; exudate absent; reverse centrally golden yellow (5B7) to champagne greyish yellow (4B4) in margins. MEA + 10 % NaCl 25 °C, 14 d: raised; colony surface velutinous, margin entire to irregular; mycelial areas white; sporulation pale turquoise (24A3); soluble pigment absent; exudate absent; reverse centrally cinnamon tan brown (6E6) to champagne greyish yellow (4B4) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads loosely columnar, frequently twisted. Conidiophores uniseriate. Stipes smooth, densely covered by very long hairs in SEM, 3–5 μm wide in the middle part, with occasional septa. Vesicles spatulate to clavate, 6.5–9 μm wide. Phialides flask-shaped, densely covered by very long hairs in SEM, 7.5–9 μm long, covering up to two thirds of the vesicle. Conidia borne as smooth cylinders, echinulate at maturity, aculeate in SEM, subglobose or barrel-shaped, connectives evident, 3–4 × 2–3 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: Aspergillus gracilis is most closely related to A. pseudogracilis. These two species can be distinguished from other species belonging to the A. conicus clade by their inability to grow on MEA and CYA at 25 °C (several strains of A. gracilis exhibit very restricted growth on CYA) and more rapid growth on M60Y at 25 °C and 30 °C. Aspergillus pseudogracilis can be distinguished from A. gracilis by its wider stipes, larger vesicles and longer phialides. Growth parameters of both species in the osmotic gradient are identical (see Fig. 12D). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Notes: Aspergillus gracilis var. exiguus was proposed based on slight differences in physiological characters and treated as synonym of A. gracilis by Raper & Fennell (1965). We could not obtain a living culture and no herbarium material is extant, resulting in us following conclusions of Raper & Fennell (1965). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: USA, California, San Diego, child carrier, 2015, isolated by Ž. Jurjević, CCF 5479 = EMSL No. 2775 = DTO 356-F4 = IBT 34559, CCF 5480 = EMSL No. 2920, CCF 5481 = EMSL No. 2922, CCF 5482 = EMSL No. 2923 = IBT 34623, EMSL No. 2763, EMSL No. 2774. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus halophilicus C.M. Chr., Papav. & C.R. Benj., Mycologia 51: 636. 1959, emended description. MycoBank MB326633. Fig. 22. ",
      " ",
      "\n",
      "\n",
      "Fig. 22 Aspergillus halophilicus. A. Colonies after 14 d at 25 °C: left to right, obverse CZA70S, reverse CZA70S, M40Y and MEA + 10 % NaCl. B–E. Cleistothecia. F, G. Asci. H, I. Ascospores. J, K. SEM (ascospores). Scale bars: D–I = 10 μm; J, K = 2 μm.Fig. 22\n",
      "\n",
      "\n",
      "Typus: USA, Minnesota, St. Paul, dried corn, 1957, isolated by C. M. Christensen [lectotype BPI 566153 designated by Samson & Gams (1985), culture ex-type NRRL 2739 = CCF 5687 = ATCC 16401 = CBS 12262 = IMI 211802 = IBT 34878]. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 30 d (mm): CZA70S 10–12; no growth on all other tested media. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: CZA70S 25 °C, 30 d: Colonies flat, surface floccose; margin filiform; mycelial areas hyaline to white; sporulation absent; ascomata produced after months of incubation, eurotium-like, hyaline to pale yellow; soluble pigment absent; exudate absent; reverse centrally greyish orange (6C6) to orange yellow (5A2) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Ascomata maturing slowly within 1–2 mo on CZA70S at 25 °C, 100–150 μm diam, hyaline to pale yellow (3A3), asci globose to subglobose, 9–15 × 7–11 μm, ascospores hyaline, lenticular with two equatorial crests, convex surface finely roughened in the area neighbouring the equatorial region, with holes, 4.5–6.5 × 3.5–5 μm including crests.\n",
      "\n",
      "Conidial state not observed in our cultures but (Christensen et al. 1959) observed it after 45 d on PDA + 20 % NaCl. Conidial heads at first globose, later becoming loosely radiate. Conidiophores uniseriate. Vesicles globose to subglobose or club-shaped, occasionally with one or two enlargements below the vesicle, 6–10 μm wide. Phialides flask-shaped, 6–14 μm long, covering only a portion of the top but occasionally up to three quarters of the vesicle, sometimes growing from the swellings below the vesicle. Conidia globose, elliptical, pyriform or irregular, spiny, hyaline to light blue-green, 6–11 × 4–6.5 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: Aspergillus halophilicus can easily be distinguished from all other members of section Restricti by its eurotium-like sexual reproductive states and inability to grow on majority of tested media, including those with high sugar (M60Y) or salt (MEA + 10 % NaCl) content after 14 d. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Italy, Venice, book cover (swab), 2013, isolated by A. Micheluz, MUT 562. Italy, Lazio, Rome, Archivio Stampati del Vaticano, book cover (swab), 2010, isolated by M. Montanari, MUT 798. Italy, Marche, Loreto, Archivio della Casa Madre, book cover (swab), 2012, isolated by M. Montanari, MUT 1305. Italy, Piedmont, Torre Pellice, Archivio della Tavola Valdese, book cover (swab), 2012, isolated by M. Montanari, MUT 1307. The Netherlands, textile imported from unknown country, 2013, isolated by M. Meijer, DTO 271-F4 = CCF 5825 = IBT 34881. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus hordei F. Sklenar, S.W. Peterson & Hubka, sp. nov. MycoBank MB818937. Fig. 23. ",
      " ",
      "\n",
      "\n",
      "Fig. 23 Aspergillus hordei. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 23\n",
      "\n",
      "\n",
      "Etymology: Named after the isolation of species from barley (Hordeum vulgare). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: USA, Minnesota, St. Paul, barley, 1957, isolated by G. C. Papavizas (holotype PRM 944446, isotype PRM 944447, culture ex-type NRRL 25825 = CCF 5483 = DTO 356-D3 = IBT 34539). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 25–29; CYA 0–8; CY20S (0)13–16; CY20S 30 °C (0)8–15; CY20S 37 °C No growth; DG18 13–26; MEA 0–5; M60Y 31–47; M60Y 30 °C 25–48; M60Y 37 °C 22–33; MEA + 10 % NaCl 26–29. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies umbonate, surface velutinous; margin entire; mycelial areas white; sporulation dark turquoise (24F8); soluble pigment absent; exudate absent; reverse centrally Cuba reddish brown (9E8), less commonly brownish red (8C7) to olive yellow (3C7), pale yellow (4A3) in margins. CYA 25 °C, 14 d: cerebriform; colony surface floccose, margin undulate; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally orange grey (5B2) to orange white (5A2) in margins. CY20S 25 °C, 14 d: cerebriform to crateriform; colony surface velutinous, margin entire to delicately filiform; mycelial areas white; sporulation aquamarine greyish turquoise (24B3) to greyish turquoise (24D4); soluble pigment absent; exudate absent; reverse centrally aquamarine turquoise grey (24E2) to turquoise grey (24C2), orange white (5A2) in margins. DG18 25 °C, 14 d: crateriform; colony surface velutinous, margin entire to delicately filiform; mycelial areas white; sporulation aquamarine dark turquoise (24F8); soluble pigment greyish yellow (2B3); exudate absent; reverse centrally yellowish brown (5E8) to olive grey (2E2), champagne greyish yellow (4B4) to yellowish white (4A2) in margins. MEA 25 °C, 14 d: no growth to microcolonies. M60Y 25 °C, 14 d: flat; colony surface velutinous; margin entire; mycelial areas white; sporulation dark turquoise (24F7); soluble pigment absent; exudate absent; reverse centrally cinnamon brown (6D6) to pale yellow (4A3), greyish yellow (4C5) to pale yellow (4A3) in margins. MEA + 10 % NaCl 25 °C, 14 d: flat to slightly centrally raised; colony surface floccose to velutinose; margin entire; mycelial areas white; sporulation deep turquoise (24E8) to greyish turquoise (24B4); soluble pigment absent; exudate absent; reverse centrally olive grey (2F2), greyish orange (5B4) to cream pale yellow (4A3) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads at first globose, later becoming loosely columnar. Conidiophores uniseriate. Stipes smooth, densely covered by short hairs in SEM, 4.5–6 μm wide in the middle part, with occasional septa. Vesicles spatulate, ellipsoidal, pyriform to subclavate, 12–14.5 μm wide. Phialides flask-shaped, densely covered by short hairs in SEM, 8.5–11 μm long, covering one to two thirds of the vesicle. Conidia borne smooth and ellipsoidal, becoming coarsely roughned, globose or barrel-shaped at maturity, lobate-reticulate in SEM, connectives evident, 3.5–5 × 2.5–4 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: See distinguishing characters of A. penicillioides. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Panama, telescope reticle, 1945, collector unknown, NRRL 25812 = CCF 5658. The Netherlands, cigars, unknown year of isolation, isolated by M. Meijer, DTO 318-F5, DTO 318-F6, DTO 318-F7, DTO 321-F9, DTO 321-G1. The Netherlands, leather, imported from unknown country, 2013, isolated by J. Houbraken, DTO 281-A4 = IBT 34517. USA, Minnesota, St. Paul, barley, 1957, isolated by G. C. Papavizas, NRRL 25826 = CCF 5484. USA, Minnesota, St. Paul, insulating board, 1957, isolated by G. C. Papavizas, NRRL 25830 = CCF 5485 = IBT 34631. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus infrequens F. Sklenar, S.W. Peterson & Hubka, sp. nov. Mycobank MB818938. Fig. 24. ",
      " ",
      "\n",
      "\n",
      "Fig. 24 Aspergillus infrequens. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 24\n",
      "\n",
      "\n",
      "Etymology: Referring to its rare occurrence. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: USA, Illinois, Peoria, wheat, 1971, isolated by D. I. Fennell (holotype PRM 944449, isotype PRM 944450, culture ex-type NRRL 25868 = CCF 5486 = DTO 356-D6 = IBT 34524). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 21–22; CYA 4–5; CY20S 7–8; CY20S 30 °C No growth; CY20S 37 °C No growth; DG18 25–26; MEA No growth; M60Y 40–43; M60Y 30 °C 8–10; M60Y 37 °C No growth; MEA + 10 % NaCl 18–19. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies crateriform, wrinkled, surface floccose; margin filiform; mycelial areas white; sporulation pale turquoise (24A3) to dark turquoise (24F8); soluble pigment absent; exudate absent; reverse centrally dark green (30F8), olive grey (2F2) to dark green (30F8), champagne greyish yellow (4B4) in margins. CYA 25 °C, 14 d: no growth to microcolonies. CY20S 25 °C, 14 d: crateriform, wrinkled; colony surface floccose, margin undulate; mycelial areas white; sporulation pale green (26A3) to pastel green (25A4); soluble pigment absent; exudate absent; reverse centrally turquoise grey (24D2) to greyish orange (5B3), orange white (5A2) in margins. DG18 25 °C, 14 d: umbonate, wrinkled; colony surface velutinous, margin filiform; mycelial areas white; sporulation pastel green (25A4) to greyish turquoise (water blue) (24C5); soluble pigment absent; exudate absent; reverse centrally dark turquoise (24F8) to greyish turquoise (24D5), orange white (5A2) in margins. MEA 25 °C, 14 d: no growth. M60Y 25 °C, 14 d: raised; colony surface powdery to cottony; margin filiform; mycelial areas white; sporulation greyish green (26B4); soluble pigment absent; exudate absent; reverse centrally olive (2E7) to mustard yellow (3B6), cream pale yellow (4A3) in margins. MEA + 10 % NaCl 25 °C, 14 d: crateriform, wrinkled; colony surface velutinous; margin filiform; mycelial areas white; sporulation light turquoise (24A4); soluble pigment absent; exudate absent; reverse centrally slate grey (3F2) to olive (3E7), champagne greyish yellow (4B4) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads radiate. Conidiophores uniseriate. Stipes smooth, sparsely covered by bundles of long hairs in SEM, 6–7 μm wide in the middle part, non-septate. Vesicles pyriform, spatulate to clavate, (10–)13–17(–20) μm wide. Phialides flask-shaped, sparsely covered by long hairs in SEM, 8–10 μm long, covering one third to two thirds of the vesicle. Conidia borne smooth and ellipsoidal, becoming echinulate, subglobose, ovate or barrel-shaped at maturity, aculeate in SEM, commonly remaining in chains, connectives evident, 4–5 × 3–4 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: Aspergillus infrequens is represented only by the ex-type isolate. It is phylogenetically distinct from other species in the A. penicillioides clade and resolves in the basal position. Although A. infrequens phenotypically resembles species from the A. penicillioides clade, the ornamentation of conidia in SEM is aculeate and atypical for the clade (this ornamentation is found in the A. restrictus, A. conicus and A. vitricola clades). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus magnivesiculatus F. Sklenar, Zalar, Ž. Jurjević & Hubka, sp. nov. Mycobank MB818939. Fig. 25. ",
      " ",
      "\n",
      "\n",
      "Fig. 25 Aspergillus magnivesiculatus. A. Colonies: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 25\n",
      "\n",
      "\n",
      "Etymology: Referring to its large vesicles. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: Japan, Tokyo, katsuobushi, 1967, isolated by K. Yamada (holotype PRM 944444, isotype PRM 944445, culture ex-type NRRL 25866 = CCF 5488 = IBT 34816). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 27–35; CYA No growth; CY20S 11–17; CY20S 30 °C No growth; CY20S 37 °C No growth; DG18 20–26; MEA No growth; M60Y 47–60; M60Y 30 °C 40–50; M60Y 37 °C No growth; MEA + 10 % NaCl 27–34. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies raised to umbonate, surface floccose; margin entire to delicately filiform; mycelial areas centrally yellow (3A6) to white in margins; sporulation greyish green (26D5); soluble pigment absent; exudate absent; reverse centrally mahogany red (8E7), yellowish red (8B8) to pale yellow (4A3) in margins. CYA 25 °C, 14 d: no growth. CY20S 25 °C, 14 d: crateriform; colony surface velutinous, margin entire; mycelial areas white; sporulation greyish green (25B4) to greyish turquoise (24B4); soluble pigment absent; exudate absent; reverse centrally aquamarine turquoise grey (24D2) to orange white (5A2) in margins. DG18 25 °C, 14 d: umbonate; colony surface velutinous to floccose, margin entire to delicately filiform; mycelial areas centrally yellowish white (4A2) to white in margins; sporulation aquamarine greenish white (27A2); soluble pigment absent; exudate absent; reverse centrally corn greyish yellow (4B5) to yellowish white (4A2) in margins. MEA 25 °C, 14 d: no growth. M60Y 25 °C, 14 d: raised; colony surface floccose to cottony; margin filiform; mycelial areas pastel yellow (3A4), pale yellow (4A3) or white; sporulation pale green (27A3) to white; soluble pigment absent; exudate absent; reverse centrally caramel brown (6C6), apricot yellow (5B6) to corn greyish yellow (4B5), pale yellow (4A3) in margins. MEA + 10 % NaCl 25 °C, 14 d: flat or centrally slightly raised; colony surface floccose to powdery; margin entire to delicately filiform; mycelial areas centraly light yellow (4A4) to white in margins; sporulation greyish green (26D5); soluble pigment absent; exudate absent; reverse centrally olive grey (2F2) to golden brown (5D7), cream pale yellow (4A3) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads at first globose, later becoming loosely columnar to radiate. Conidiophores uniseriate. Stipes smooth, densely covered by short hairs in SEM, 5–7 μm wide in the middle part, non-septate. Vesicles pyriform, subclavate to clavate, 17–22 μm wide. Phialides flask-shaped, densely covered by short hairs in SEM, 9–11 μm long, covering two thirds to three quarters of the vesicle. Conidia borne smooth, ellipsoidal, coarsely roughned, ovate to barrel-shaped at maturity, tuberculate in SEM, connectives evident, 4.5–5.5 × 3.5–4.5 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: Isolates of A. magnivesiculatus have the largest vesicles among species of the A. penicillioides clade; reverse colour on M40Y is usually mahogany red. See also distinguishing characters of A. penicillioides. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Canada, Nova Scotia, Wolfville, house dust, 2015, collected by A. Walker, isolated by C.M. Visagie, KAS 6089 = DTO 356-H3. Canada, Ontario, Ottawa, house dust, 2015, collected and isolated by C. M. Visagie, KAS 5655 = DTO 356-G8, KAS 5754 = DTO 356-G9. Canada, Ontario, Stittsville, house dust, 2014, collected by K. A. Seifert, isolated by C. M. Visagie, KAS 5623 = DTO 356-G7. Japan, Tokyo, katsuobushi, 1967, isolated by K. Yamada, NRRL 25867 = CCF 5660. Slovenia, Ljubljana, oil painting on canvas, 2015, isolated by P. Zalar and D.D. Graf, EXF-10353 = IBT 34284. Slovenia, Ljubljana, oil painting on canvas, 2015, isolated by P. Zalar and D.D. Graf, EXF-10377, EXF-10347 = IBT 34283, EXF-10356 = IBT 34279. USA, California, San Diego, child carrier, 2014, isolated by Ž. Jurjević, CCF 5490 = EMSL No. 2741. USA, California, San Diego, child carrier, 2015, isolated by Ž. Jurjević, CCF 5491 = EMSL No. 2918. USA, Idaho, home air, 2010, isolated by Ž. Jurjević, CCF 5489 = EMSL No. 1315 = DTO 356-E2 = IBT 34516. USA, Minnesota, St. Paul, dried corn, 1954, isolated by C. M. Christensen, NRRL 25821 = CCF 5487. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus pachycaulis F. Sklenar, S.W. Peterson, Ž. Jurjević & Hubka, sp. nov. MycoBank MB823048. Fig. 26. ",
      " ",
      "\n",
      "\n",
      "Fig. 26 Aspergillus pachycaulis. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 26\n",
      "\n",
      "\n",
      "Etymology: Referring to the wide conidiophore stipes, especially compared to the closely related A. caesiellus. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: USA, District of Columbia, Washington, unknown substrate, 1956, isolated by H. E. Wheeler (holotype PRM 944432, isotype PRM 944433, culture ex-type NRRL 25824 = CCF 5492 = DTO 356-D2 = IBT 34521 = IBT 34812). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 20–60; CYA 11–26; CY20S 15–40; CY20S 30 °C 22–36; CY20S 37 °C (0)8–10; DG18 35–48; MEA 21–31; M60Y 25–60; M60Y 30 °C 36–60; M60Y 37 °C 38–60; MEA + 10 % NaCl 27–43. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies raised, surface powdery to floccose; margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (24C4); soluble pigment absent; exudate absent; reverse centrally brownish yellow (5C8) to light yellow (4A4) in margins. CYA 25 °C, 14 d: centrally raised, radially wrinkled; colony surface lanose to floccose, margin entire; mycelial areas white; sporulation greyish green (25C3); soluble pigment absent; exudate absent; reverse centrally deep blue (21D8) to yellowish white (2A2) in margins. CY20S 25 °C, 14 d: with central depression; colony surface lanose to floccose; margin entire; mycelial areas white; sporulation dull green (25D3); soluble pigment absent; exudate absent; reverse centrally greyish brown nougat (5D3) to orange white (5A2) in margins. DG18 25 °C, 14 d: flat or umbonate; colony surface powdery to floccose, margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (water blue) (24D5); soluble pigment absent; exudate absent; reverse milk white (1A2). MEA 25 °C, 14 d: centrally raised; colony surface lanose to floccose, margin entire; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally reddish golden (6C7) to pale orange (5A3) in margins. M60Y 25 °C, 14 d: raised; colony surface lanose to filamentous; margin entire to delicately filiform; mycelial areas white; sporulation dull green (27D4); soluble pigment absent; exudate absent; reverse centrally Pompeian yellow (5C6) to pastel yellow (3A4) in margins. MEA + 10 % NaCl 25 °C, 14 d: umbonate; colony surface powdery; margin irregularly filiform; mycelial areas white; sporulation greyish blue (23C5); soluble pigment absent; exudate absent; reverse centrally greyish orange (5B4) to light yellow (4A4) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads columnar, frequently sinuous. Conidiophores uniseriate. Stipes smooth, densely covered by short hairs in SEM, occasionally with septa, 7–8 μm wide. Vesicles pyriform to subclavate, 16–20 μm wide. Phialides flask-shaped, densely covered by short hairs in SEM, 8–9 μm long, covering one to two thirds of the vesicle. Conidia borne as smooth cylinders, at maturity ellipsoidal to ovate, aculeate in SEM, connectives evident, 4–5 × 2.5–3.5 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: See distinguishing characters of A. restrictus. Aspergillus pachycaulis has wider stipes and larger vesicles compared to A. caesiellus and A. restrictus. It also grows slightly faster on MEA and some isolates can grow on CY20S at 37 °C (exception throughout the whole sect. Restricti). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Canada, Ontario, Ottawa, unknown substrate, isolated by J. Bissett, DAOMC 226750 = KAS 7748. USA, California, home air, 2005, isolated by Ž. Jurjević, CCF 5493 = EMSL No. 2310 = DTO 356-E6 = IBT 34536. Unknown locality of isolation, unknown substrate, unknown year of isolation, collector unknown, DTO 066-I6. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus penicillioides Speg., Revista Fac. Agron. Univ. Nac. La Plata 2: 245. 1896, emended description. MycoBank MB309234. Fig. 27. ",
      " ",
      "\n",
      "\n",
      "Fig. 27 Aspergillus penicillioides. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 27\n",
      "\n",
      "\n",
      "Typus: Brazil, Recife, isolated from human cutaneous disease, 1958, D. Borelli (neotype Herb. IMI 211392 designated by Samson & Gams (1985), culture ex-type NRRL 4548 = CBS 540.65 = ATCC 16910 = IMI 211342 = DTO 207-I7 = CCF 5494 = IBT 34627). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 27–43; CYA 7–9; CY20S 11–20; CY20S 30 °C 11–15; CY20S 37 °C No growth; DG18 17–36; MEA 5–9; M60Y 28–59; M60Y 30 °C 40–60; M60Y 37 °C 17–45; MEA + 10 % NaCl 24–37. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies raised to umbonate, surface powdery; margin entire to delicately filiform; mycelial areas white; sporulation deep green (25E8); soluble pigment absent; exudate absent; reverse centrally coffee brown (5F7) to yellowish brown (5E8), golden greyish yellow (4C6) to orange white (5A2) in margins. CYA 25 °C, 14 d: convex, wrinkled; colony surface velutinous, margin undulate; mycelial areas white; no sporulation; soluble pigment absent; exudate absent; reverse centrally greyish orange (5B3) to cream orange white (5A2) in margins. CY20S 25 °C, 14 d: crateriform; colony surface velutinous, margin filiform; mycelial areas white; sporulation pale green (25A3); soluble pigment absent; exudate absent; reverse centrally greyish yellow (2B5) to olive grey (2F2), greyish orange (5B3) to orange white (5A2) in margins. DG18 25 °C, 14 d: umbonate to convex; colony surface floccose, margin filiform; mycelial areas white; sporulation aquamarine light turquoise (24A4); soluble pigment greyish yellow (2B3); exudate absent; reverse centrally olive yellow (2D6) to yellowish white (4A2) in margins. MEA 25 °C, 14 d: crateriform; colony surface velutinous, margin entire; no sporulation; soluble pigment absent; exudate absent; reverse centrally caramel brown (6C6) to pale orange (5A3) in margins. M60Y 25 °C, 14 d: raised; colony surface cottony; margin entire to delicately filiform; mycelial areas white; sporulation greyish green (25D7); soluble pigment absent; exudate absent; reverse centrally dark brown (6F6) to golden greyish yellow (4C6), pale yellow (4A3) in margins. MEA + 10 % NaCl 25 °C, 14 d: centrally raised; colony surface floccose; margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (24C6); soluble pigment absent; exudate absent; reverse centrally olive brown (4F5) to yellowish orange (4B7), cream pale yellow (4A3) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads at first globose, later becoming loosely columnar. Conidiophores uniseriate. Stipes smooth, sparsely covered by short hairs in SEM, 4–6 μm wide in the middle part, widening gradually toward the vesicle, non-septate. Vesicles pyriform, spatulate to clavate, 10–18 μm wide. Phialides flask-shaped, sparsely covered by short hairs in SEM, 8–10 μm long, covering upper half to two thirds of the vesicle. Conidia borne smooth and ellipsoidal, later becoming coarsely roughened, subglobose or barrel-shaped, tuberculate in SEM, commonly remaining in chains, connectives evident, 3.5–4.5 × 2.5–3.5 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: All species belonging to the A. penicillioides clade are morphologically very similar and particularly A. penicillioides and A. hordei share almost identical morphologies. The latter two species can be distinguished from others in the clade by their ability to grow on CY20S at 30 °C, production of greyish yellow soluble pigment when cultivated on DG18, narrow stipes, small vesicles and loosely columnar rather then loosely radiate conidial heads. In contrast to most A. hordei isolates, A. penicillioides always grows on MEA and CYA. Conidia of A. penicillioides are tuberculate in SEM, but lobate-reticulate in A. hordei. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Belgium, Brussels, indoor air (air-conditioned office), 1985, collector unknown, IHEM 2476. Brazil, Recife, human cutaneous disease isolate, 1958, collector unknown, NRRL 4550 = CCF 5495, NRRL 4553 = CCF 5496. Canada, Ontario, Ottawa, house dust, 2015, collected by N. Yilmaz, isolated by C. M. Visagie, KAS 7745, KAS 7746. Czech Republic, Prague, sweet roll with chocolate glaze, 2002, isolated by H. Růžičková, CCF 3282. Czech Republic, Zlín, leather shoe, 1990, isolated by T. Skálová, CCF 2666. France, seeds of cereal, 1984, collector unknown, IHEM 2330. France, French madeleines, 2013, isolated by F. Déniel, UBOCC-A-115042 = CBS 140430 = DTO 334-E1. Germany, unknown substrate, unknown year of isolation, collector unknown, DTO 334-A9. Indonesia, peanuts, 2008, isolated by E.S. Rahayu and J. Houbraken, DTO 063-G8. Micronesia, house dust, unknown year of isolation, isolated by C. M. Visagie, DTO 267-A4 = IBT 34614. Micronesia, house dust, 2009, collected by W. Law, isolated by E. Whitfield and K. Mwange, DTO 267-A9 = CCF 5664. The Netherlands, leather imported from unknown country, 2013, isolated by J. Houbraken, DTO 281-A7. The Netherlands, Zwartewaal, air, 2012, isolated by M. Meijer, 231-A7. The Netherlands, flour imported from unknown country, 2005, isolated by J. Houbraken, DTO 011-C4. USA, Illinois, O'fallon, baseball gloves in store, 2014, isolated by Ž. Jurjević, CCF 5500 = EMSL No. 2651 = IBT 34630. USA, New Jersey, green fabric covered binders, import from China, 2014, isolated by Ž. Jurjević, CCF 5499 = EMSL No. 2441, CCF 5497 = EMSL No. 2430 = IBT 34628, CCF 5498 = EMSL No. 2440 = DTO 356-E7 = IBT 34815. USA, North Carolina, Durham, unknown substrate, 1952, isolated by J. L. Miranda, NRRL 25816 = CCF 5661. USA, Minnesota, St. Paul, peas, 1957, isolated by G. C. Papavizas, NRRL 25834 = CCF 5659. USA, Minnesota, St. Paul, wheat, 1957, isolated by G. C. Papavizas, NRRL 25835. USA, California, San Diego, child carrier, 2015, isolated by Ž. Jurjević, CCF 5501 = EMSL No. 2749 = IBT 34629, CCF 5502 = EMSL No. 2900, CCF 5503 = EMSL No. 2909, EMSL No. 2747, EMSL No. 2777, EMSL No. 2766, EMSL No. 2786, EMSL No. 2760, EMSL No. 2757, EMSL No. 2904, EMSL No. 2898. USA, Maryland, Bethesda, archival cabinet, 2015, isolated by Ž. Jurjević, CCF 5504 = EMSL No. 3264. USA, New York, blood sample, 1951, isolated by A. Howell, NRRL 25879. USA, Minnesota, St. Paul, dried corn, 1954, isolated by C. M. Christensen, NRRL 25820, NRRL 25822. Unknown locality of isolation, unknown substrate, unknown year of isolation, collector unknown, DTO 008-H4. Unknown locality of isolation, unknown substrate, unknown year of isolation, collector unknown, NRRL 25870. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus pseudogracilis F. Sklenar, Ž. Jurjević & Hubka, sp. nov. MycoBank MB818932. Fig. 28. ",
      " ",
      "\n",
      "\n",
      "Fig. 28 Aspergillus pseudogracilis. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 28\n",
      "\n",
      "\n",
      "Etymology: Referring to its close phylogenetic and phenotypic relatedness to A. gracillis. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: USA, California, San Diego, child carrier, 2015, isolated by Ž. Jurjević (holotype PRM 944434, isotype PRM 944435, culture ex-type CCF 5505 = EMSL No. 2765 = DTO 356-F3 = NRRL 66620 = IBT 34813). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 42–45; CYA No growth; CY20S 7–8; CY20S 30 °C No growth; CY20S 37 °C No growth; DG18 20–22; MEA No growth; M60Y 52–53; M60Y 30 °C 43–45; M60Y 37 °C No growth; MEA + 10 % NaCl 25–26. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies raised, surface floccose; margin entire to delicately filiform; mycelial areas white; sporulation light turquoise (24A4); soluble pigment absent; exudate absent; reverse centrally cinnamon brown (6D6) to champagne greyish yellow (4B4) in margins. CYA 25 °C, 14 d: no growth to microcolonies. CY20S 25 °C, 14 d: convex; colony surface floccose to cottony, margin entire to delicately filiform; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally greyish orange (5B3) to yellowish white (4A2) in margins. DG18 25 °C, 14 d: centrally raised; colony surface floccose, margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (24B4); soluble pigment absent; exudate absent; reverse centrally turquoise grey (24C2) to orange white (5A2) in margins. MEA 25 °C, 14 d: no growth. M60Y 25 °C, 14 d: raised; colony surface floccose; margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (24B4); soluble pigment absent; exudate absent; reverse champagne greyish yellow (4B4). MEA + 10 % NaCl 25 °C, 14 d: flat, wrinkled; colony surface floccose, margin entire to delicately filiform; mycelial areas white; sporulation pale turquoise (24A3); soluble pigment absent; exudate absent; reverse centrally light brown (5D6) to champagne greyish yellow (4B4) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads loosely columnar, frequently twisted. Conidiophores uniseriate. Stipes smooth, densely covered by long hairs in SEM, 4–6 μm wide in the middle part, widening gradually toward the vesicle, septate. Vesicles spatulate to clavate, 9–13 μm wide. Phialides flask-shaped, densely covered by long hairs in SEM, 9–10.5 μm long, covering two thirds of the vesicle. Conidia borne as smooth cylinders, echinulate at maturity, aculeate in SEM, subglobose or barrel-shaped, connectives evident, 3–4 × 2–3 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: See distinguishing characters of A. conicus and A. gracilis. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus restrictus G. Sm., J. Textile Inst. 22: T115. 1931, emended description. MycoBank MB276290. Fig. 29.\n",
      "\n",
      "Fig. 29 Aspergillus restrictus. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 29\n",
      "\n",
      "\n",
      "Synonyms: Penicillium fuscoflavum S. Abe, J. Gen. Appl. Microbiol., Tokyo 2: 64. 1956.\n",
      "\n",
      "\n",
      "Aspergillus restrictus var. B G. Sm., J. Textile Inst. 22: T115. 1931. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: United Kingdom, cloth, 1928, isolated by G. Smith [neotype Herb. IMI 16267 designated by Samson & Gams (1985) and modified by Pitt & Samson (1993), culture ex-type NRRL 154 = CBS 117.33 = CBS 541.65 = DTO 079-B2 = ATCC 16912 = IHEM 3920 = IMI 16267= CCF 5506 = IBT 34615]. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 42–53; CYA 12–16; CY20S 24–29; CY20S 30 °C 12–25; CY20S 37 °C No growth; DG18 28–33; MEA 5–13; M60Y 38–50; M60Y 30 °C 45–55; M60Y 37 °C 20–40; MEA + 10 % NaCl 33–40. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies flat, surface powdery, filamentous in margins; margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (24C4); soluble pigment absent; exudate absent; reverse centrally cinnamon brown (6D6) to sand greyish yellow (4B3) in margins. CYA 25 °C, 14 d: centrally raised, irregulary wrinkled; colony surface floccose, margin entire; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally dull blue (23E5) to orange white (5A2) in margins. CY20S 25 °C, 14 d: umbonate with concentric ring pattern; colony surface coarsely granular in the centre, velvety in margins; margin entire to delicately filiform; mycelial areas white; sporulation deep green (29E8) to light turquoise (25A4); soluble pigment absent; exudate absent; reverse centrally dull blue (23E5) to orange white (5A2) in margins. DG18 25 °C, 14 d: umbonate; colony surface powdery, margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (24D4); soluble pigment absent; exudate absent; reverse centrally brownish orange (6C3) to orange white (5A2) in margins. MEA 25 °C, 14 d: convex, radially wrinkled; colony surface floccose, margin entire; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally light brown (7D4) to flesh greyish orange (6B3) in margins. M60Y 25 °C, 14 d: flat; colony surface powdery; margin filiform; mycelial areas white; sporulation greyish turquoise (24C4); soluble pigment absent; exudate absent; reverse light yellow (4C4). MEA + 10 % NaCl 25 °C, 14 d: flat; colony surface powdery, filamentous in margins; margin entire to delicately filamentous; mycelial areas white; sporulation greyish turquoise (24C4); soluble pigment absent; exudate absent; reverse centrally brownish grey (8E2) to champagne greyish yellow (4B4) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads columnar, frequently sinuous. Conidiophores uniseriate. Stipes smooth, densely covered by long hairs in SEM, non-septate, 4–5.5 μm wide. Vesicles pyriform to clavate, 9–12 μm in diameter. Phialides flask-shaped, densely covered by long hairs in SEM, 8–9 μm long, covering one third to one half of the vesicle. Conidia borne as smooth cylinders in long appressed columns, at maturity ellipsoidal to ovate, commonly remaining in chains, connectives evident, coarsely roughned to echinulate, aculeate in SEM, 3.5–4.5 × 2.5–3 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: Aspergillus restrictus can be distinguished from A. caesiellus and A. pachycaulis by smaller vesicles and narrower stipes, slower growth on M40Y and M60Y and no sporulation on CYA. Osmotic gradient growth curves of these three species differ mainly on MEA, where A. pachycaulis grows slowest on MEA + 10 % NaCl and MEA + 15 % NaCl where A. restrictus grows fastest (see Fig. 12A). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Taxonomic notes: Penicillium fuscoflavum (ex-type strain NRRL 4783) and A. restrictus var. B (represented by NRRL 154) are taxa conspecific with A. restrictus, based on phylogenetic analysis (Peterson 2008, this study – see Fig. 7). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Belgium, Antwerp, dust from mattress, 1984, collector unknown, IHEM 2121. Belgium, Estinnes-au-Mont, indoor air, 1981, collector unknown, IHEM 818. Belgium, Brussels, indoor air (air-conditioned office), 1985, collector unknown, IHEM 2477. Belgium, Brussels, indoor air (air-conditioned office), 1984, collector unknown, IHEM 2373. Bermuda, home air, 2010, isolated by Ž. Jurjević, CCF 5507 = EMSL No. 1379 = IBT 34618. Canada, Ontario, Oshawa, maple syrup, unknown year of isolation, isolated by Samantha Frasz, DAOMC 250090 = KAS 7744. Czech Republic, Prague, sclerotium of fungus Corallocytostroma ornicopreoides imported from Australia, 2003, isolated by A. Kubátová, CCF 3364 = IBT 34619. Indonesia, corn, unknown year of isolation, collector unknown, DTO 065-C7 = IBT 34541. Indonesia, dust, unknown year of isolation, collector unknown, DTO 236-I7, DTO 237-A2, DTO 236-I8. USA, Maryland, packing material, 2012, isolated by Ž. Jurjević, CCF 5511 = EMSL No. 1675 = IBT 34616. USA, Illinois, Peoria, culture contaminant, 1975, isolated by T. G. Pridham, NRRL 25862. USA, New Jersey, Sicklerville, air in school auditorium, 2013, isolated by Ž. Jurjević, CCF 5512 = EMSL No. 2206 = IBT 34617. USA, New Jersey, green fabric covered binders, import from China, 2014, isolated by Ž. Jurjević, CCF 5513 = EMSL No. 2429. USA, North Carolina, mattress cover, 2012, isolated by Ž. Jurjević, CCF 5509 = EMSL No. 1611. USA, Illinois, O'fallon, baseball gloves in store, 2014, isolated by Ž. Jurjević, CCF 5514 = EMSL No. 2652. USA, cattle feed, 1984, isolated by B. W. Horn, NRRL 25882. USA, New Jersey, hospital air, 2012, isolated by Ž. Jurjević, CCF 5510 = EMSL No. 1633. USA, Alabama, home air, 2010, isolated by Ž. Jurjević, CCF 5508 = EMSL No. 1416. USA, New York, Upton, work station, indoor air, 2015, isolated by Ž. Jurjević, EMSL No. 2892. USA, California, San Diego, child carrier, 2015, isolated by Ž. Jurjević, CCF 5515 = EMSL No. 2906, EMSL No. 2746, EMSL No. 2897, EMSL No. 2748. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus reticulatus F. Sklenar, Ž. Jurjević, S.W. Peterson & Hubka, sp. nov. MycoBank MB818940. Fig. 30. ",
      " ",
      "\n",
      "\n",
      "Fig. 30 Aspergillus reticulatus. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 30\n",
      "\n",
      "\n",
      "Etymology: Referring to the lobate-reticulate ornamentation of the conidia. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: USA, South Carolina, Charleston, lung biopsy, 1965, isolated by L. Jaynes (holotype PRM 944442, isotype PRM 944443, culture ex-type NRRL 25852 = CCF 5516 = DTO 356-D4 = IBT 34540). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 46–50; CYA 8–14; CY20S 21–30; CY20S 30 °C 8–18; CY20S 37 °C No growth; DG18 26–33; MEA 2–7; M60Y 55–60; M60Y 30 °C 50–55; M60Y 37 °C 3–20; MEA + 10 % NaCl 32–40. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies raised, surface floccose to lanose; margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (24E6); soluble pigment absent; exudate absent; reverse centrally golden brown (5D7) to greyish yellow (4C4), pale yellow (4A3) in margins. CYA 25 °C, 14 d: crateriform, radially wrinkled; colony surface velutinous, margin undulate; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally greyish orange (5B3) to orange white (5A2) in margins. CY20S 25 °C, 14 d: centrally raised; colony surface velutinous to floccose, margin entire to delicately filiform; mycelial areas white; sporulation pale turquoise (24A3) to deep turquoise (24E8); soluble pigment absent; exudate absent; reverse centrally aquamarine greyish turquoise (24B3) to yellowish white (4A2) in margins. DG18 25 °C, 14 d: umbonate; colony surface floccose, margin entire to delicately filiform; mycelial areas white; sporulation aquamarine deep green (25E8); soluble pigment absent; exudate absent; reverse centrally dull green (25D3) to orange grey (5B2), yellowish white (4A2) in margins. MEA 25 °C, 14 d: cerebriform; colony surface velutinous, margin undulate; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally greyish orange (5C5) to orange white (5A2) in margins. M60Y 25 °C, 14 d: raised; colony surface floccose to lanose; margin entire to delicately filiform; mycelial areas white; sporulation greyish green (25E6); soluble pigment absent; exudate absent; reverse centrally cinnamon brown (6D6), pale yellow (4A3) to greyish yellow (4C5), pale yellow (4A3) in margins. MEA + 10 % NaCl 25 °C, 14 d: flat to umbonate; colony surface floccose; margin entire to delicately filiform; mycelial areas white; sporulation deep turquoise (24E8); soluble pigment absent; exudate absent; reverse centrally olive grey (2F2) to brownish orange (5C5), cream pale yellow (4A3) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads globose to radiate. Conidiophores uniseriate. Stipes smooth, densely covered by short hairs in SEM, 5–7 μm wide in the middle part, non-septate. Vesicles globose or pyriform, 14–20 μm wide. Phialides flask-shaped, densely covered by short hairs in SEM, 9–11 μm long, covering two thirds to three quarters of the vesicle. Conidia borne smooth, ellipsoidal, becoming rough-walled, globose or barrel-shaped at maturity, tuberculate to lobate-reticulate in SEM, sometimes remaining in chains, connectives evident, 3.5–4.5 × 2.5–3.5 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: Aspergillus reticulatus is phylogenetically closely related to A. salinicola (Fig. 7). These two species are less xerophilic than species closely related to A. penicillioides, because they grow slightly faster in conditions with higher water activity and slower in conditions with lower water activity (higher concentration of salt). In contrast to A. salinicola, isolates of A. reticulatus can grow on M60Y at 37 °C, grow faster on CY20S at 25 °C and 30 °C and usually have wider stipes and larger vesicles. See also distinguishing characters of A. penicillioides. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Belgium, Brussels, dust from carpet, 2008, collector unknown, IHEM 22696. Czech Republic, Zlín, leather shoe, 1998, isolated by T. Skálová, CCF 3112 = IBT 34634 = NRRL 62490. France, French madeleines, 2013, F. Déniel, UBOCC-A-115044, UBOCC-A-115045. Puerto Rico, Bayamon, administrative area, air, 2014, isolated by Ž. Jurjević, CCF 5522 = EMSL No. 2525, CCF 5523 = EMSL No. 2526. Puerto Rico, Bayamon, office air, 2014, isolated by Ž. Jurjević, CCF 5524 = EMSL No. 2548 = IBT 34637. Slovenia, Ljubljana, oil painting on canvas, 2015, isolated by P. Zalar and D.D. Graf, EXF-10429 = CCF 5656. USA, Idaho, home air, 2009, isolated by Ž. Jurjević, CCF 5518 = NRRL 58903 = EMSL No. 1272 = IBT 34819. USA, Idaho, home air, 2010, isolated by Ž. Jurjević, CCF 5519 = NRRL 62004 = EMSL No. 1313, CCF 5520 = NRRL 62005 = EMSL No. 1314. USA, Idaho, outside air, 2010, isolated by Ž. Jurjević, CCF 5521 = EMSL No. 1362 = DTO 356-E4 = IBT 34518. USA, Florida, home air, 2008, isolated by Ž. Jurjević, CCF 5525 = NRRL 58572 = EMSL No. 885 = IBT 34880. USA, Georgia, Chamblee, lung biopsy, 1950, isolated by L. Ajello, NRRL 25878 = CCF 5517. USA, California, San Diego, child carrier, 2015, isolated by Ž. Jurjević, EMSL No. 2778, EMSL No. 2767, EMSL No. 2750, EMSL No. 2760, EMSL No. 2770, EMSL No. 2771, EMSL No. 2772, EMSL No. 2745, EMSL No. 2756. USA, Hawaii, Honolulu, drawer, unknown year of isolation, isolated by J. Varga, CBS 114325 = DTO 077-H3. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus salinicola Zalar, F. Sklenar, Visagie & Hubka, sp. nov. MycoBank MB818941. Fig. 31. ",
      " ",
      "\n",
      "\n",
      "Fig. 31 Aspergillus salinicola. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 31\n",
      "\n",
      "\n",
      "Etymology: Referring to salterns, a source of isolation of some isolates. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: Slovenia, Ljubljana, oil painting on canvas, 2015, isolated by P. Zalar (holotype PRM 944448, culture ex-type EXF-10401 = IBT 34266 = CCF 5526 = NRRL 66621). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 32–40; CYA (0)8–9; CY20S 9–20; CY20S 30 °C 0–5; CY20S 37 °C No growth; DG18 21–30; MEA 0–5; M60Y 42–55; M60Y 30 °C 27–43; M60Y 37 °C No growth; MEA + 10 % NaCl 32–35. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies flat to umbonate, surface velutinous; margin delicately filiform; mycelial areas pinkish white (7A2) to white; sporulation greyish green (25B5); soluble pigment absent; exudate absent; reverse centrally golden brown (5D7) to pale yellow (4A3) in margins. CYA 25 °C, 14 d: cerebriform; colony surface floccose, margin undulate; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally greyish orange (5B3) to orange white (5A2) in margins. CY20S 25 °C, 14 d: centrally raised, radially wrinkled; colony surface velutinous, margin entire to delicately filiform; mycelial areas white; sporulation turquoise grey (24D2); soluble pigment absent; exudate absent; reverse centrally nougat dull yellow (3D4) to pale yellow (4A3) in margins. DG18 25 °C, 14 d: flat, slightly centrally raised; colony surface floccose, sometimes forming sectors, margin entire; mycelial areas white; sporulation greyish green (25D6); soluble pigment absent; exudate absent; reverse centrally greyish green (25C3) to greyish orange (5B3), orange white (5A2) in margins. MEA 25 °C, 14 d: no growth to microcolonies. M60Y 25 °C, 14 d: raised; colony surface floccose to lanose; margin filiform; mycelial areas white; sporulation greyish green (25D7); soluble pigment absent; exudate absent; reverse centrally olive green (2F6), pale yellow (4A3) to greyish yellow (4C5), pale yellow (4A3) in margins. MEA + 10 % NaCl 25 °C, 14 d: centrally raised; colony surface granular to filamentous, floccose in the centre; margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (24C4); soluble pigment absent; exudate absent; reverse centrally light brown (5D6) to amber yellow (4D6), champagne greyish yellow (4B4) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads at first globose, later becoming loosely radiate. Conidiophores uniseriate. Stipes smooth, sparsely covered by short hairs in SEM, 4–6 μm wide in the middle part, non-septate. Vesicles pyriform, spatulate to subclavate, 11–14 μm wide. Phialides flask-shaped, sparsely covered by short hairs in SEM, 8–9 μm long, covering one third to one half of the vesicle. Conidia borne smooth and ellipsoidal, becoming irregularly rough-walled, globose to subglobose, tuberculate to lobate-reticulate in SEM, connectives evident, 3–4 × 2–3 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: See distinguishing characters of A. reticulatus. A relatively unique character of A. salinicola is the pale colour of sporulation in comparison with other species belonging to the A. penicillioides clade. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Canada, Nova Scotia, Wolfville, house dust, 2015, collected by A. Walker, isolated by C.M. Visagie, KAS 6054. France, Brest, painting from Brittany (painter: J.M. Vien; painting title: Jésus et le fils du centenier de Capharnaüm; painting technique: oil on paper mounted on canvas), 2015, isolated by F. Déniel, UBOCC-A-116019 = CCF 5528 = IBT 34635. France, French savaroise, 2013, F. Déniel, UBOCC-A-115043 = CBS 140431 = DTO 334-E2. Germany, Essen, kindergarten, unknown year of isolation, isolated by J. Varga, CBS 117344 = DTO 077-I1. Slovenia, Sečovlje salterns, 1996, isolated by N. Gunde-Cimerman and P. Zalar, EXF-226 = CCF 5527 = IBT 34277 = NRRL 66622. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus tardicrescens F. Sklenar, Houbraken, Zalar & Hubka, sp. nov. MycoBank MB818942. Fig. 32. ",
      " ",
      "\n",
      "\n",
      "Fig. 32 Aspergillus tardicrescens. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 32\n",
      "\n",
      "\n",
      "Etymology: Referring to the slow growth of the species. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: The Netherlands, Alphen aan den Rijn, museum piece (measuring table), 2014, collector unknown (holotype PRM 944438, isotype PRM 944439, culture ex-type DTO 316-B5 = CCF 5529 = IBT 34558 = NRRL 66623). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 17–29; CYA No growth; CY20S 0–5; CY20S 30 °C No growth; CY20S 37 °C No growth; DG18 12–17; MEA No growth; M60Y 31–50; M60Y 30 °C 15–35; M60Y 37 °C No growth; MEA + 10 % NaCl 19–29. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies umbonate, surface powdery; margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (water blue) (24C5); soluble pigment absent; exudate absent; reverse centrally olive brown (4F8) to greyish yellow (4C7), pale orange (5A3) in margins. CYA 25 °C, 14 d: no growth. CY20S 25 °C, 14 d: convex; colony surface velutinous, margin undulate; mycelial areas white; sporulation pale green (25A3); soluble pigment absent; exudate absent; reverse centrally greenish grey (25D2) to orange white (5A2) in margins. DG18 25 °C, 14 d: convex; colony surface floccose, margin entire; mycelial areas white; sporulation aquamarine turquoise blue (24A6); soluble pigment absent; exudate absent; reverse centrally turquoise grey (24C2) to orange white (5A2) in margins. MEA 25 °C, 14 d: no growth. M60Y 25 °C, 14 d: raised; colony surface floccose to cottony; margin entire to delicately filiform; mycelial areas white; sporulation aquamarine greyish turquoise (24B3) to pastel green (25A4); soluble pigment absent; exudate absent; reverse centrally brownish orange (5C5) to pale yellow (4A3) in margins. MEA + 10 % NaCl 25 °C, 14 d: flat to umbonate; colony surface powdery to floccose; margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (24C3); soluble pigment absent; exudate absent; reverse centrally bronze brown (5E5) to greyish yellow (4C7), cream pale yellow (4A3) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads at first globose, later becoming radiate. Conidiophores uniseriate. Stipes smooth, densely covered by short hairs in SEM, 5–8 μm wide in the middle part, with occasional septa. Vesicles subglobose, spatulate, pyriform to subclavate, 14–19 μm wide. Phialides flask-shaped, densely covered by short hairs in SEM, 8–11 μm long, covering upper half to two thirds of the vesicle. Conidia borne irregularly rough and ellipsoidal, becoming coarsely rough-walled, subglobose or barrel-shaped at maturity, tuberculate in SEM, connectives evident, 3.5–4.5 × 2.5–3.5 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: See distinguishing characters of A. penicillioides. Conidia of A. tardicrescens are smaller than those of A. clavatophorus and A. magnivesiculatus; its vesicles are smaller than of A. magnivesiculatus. The majority of isolates is not able to grow on CY20S at 30 °C or even at 25 °C. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Canada, Nova Scotia, Wolfville, house dust, 2015, collected by A. Walker, isolated by C. M. Visagie, KAS 6252 = DTO 356-I5. Slovenia, Ljubljana, oil painting on canvas, 2015, isolated by P. Zalar and D.D. Graf, EXF-10454 = IBT 34281 = CCF 5530 = NRRL 66624. Slovenia, Ljubljana, air in depot of Conservation Centre of the Institute for the Protection of Cultrural Heritage of Slovenia, 2015, isolated by P. Zalar and D.D. Graf, EXF-10456 = IBT 34286. The Netherlands, Alphen aan den Rijn, museum piece (dentist chair), 2014, collector unknown, DTO 316-A7. The Netherlands, Alphen aan den Rijn, museum piece (rubber tyre of brancard), 2014, collector unknown, DTO 316-A8 = IBT 34562. The Netherlands, Alphen aan den Rijn, museum piece (x-ray table), 2014, collector unknown, DTO 316-A9. The Netherlands, Alphen aan den Rijn, museum piece (vitrine), 2014, collector unknown, DTO 316-B4. The Netherlands, Tilburg, moist wall of archive, 2008, isolated by J. Houbraken, DTO 073-H6. The Netherlands, Zwartewaal, unknown substrate, 2012, isolated by M. Meijer, DTO 231-B8. The Netherlands, archive material, unknown year of isolation, isolated by J. Houbraken, DTO 014-A3. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus villosus F. Sklenar, S.W. Peterson & Hubka, sp. nov. MycoBank MB818933. Fig. 33. ",
      " ",
      "\n",
      "\n",
      "Fig. 33 Aspergillus villosus. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 33\n",
      "\n",
      "\n",
      "Etymology: Referring to stipes and phialides, which are villose when observed using SEM. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Typus: United Kingdom, Scotland, Kirkhill, unknown substrate, 1945, isolated by D. Snow (holotype PRM 944430, isotype PRM 944431, culture ex-type NRRL 25813 = CCF 5531 = DTO 356-C9 = IBT 34822). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 21–23; CYA 8–10; CY20S 17–19; CY20S 30 °C 7–8; CY20S 37 °C No growth; DG18 16–20; MEA 5–6; M60Y 26–28; M60Y 30 °C 28–30; M60Y 37 °C 6–8; MEA + 10 % NaCl 22–23. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies flat, surface floccose; margin irregular, filiform; mycelial areas white; sporulation greyish green (25E7) to bluish green (25A6); soluble pigment absent; exudate absent; reverse centrally olive green (2F6) to champagne greyish yellow (4B4) in margins. CYA 25 °C, 14 d: convex; colony surface floccose, margin entire to undulate; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally yellowish grey (4B2) to yellowish white (4A2) in margins. CY20S 25 °C, 14 d: umbonate, slightly radially wrinkled; colony surface velutinous to floccose, margin entire; mycelial areas white; sporulation light turquoise (24A4) to pastel blue (23A4); soluble pigment absent; exudate absent; reverse dull green (25D3) to pale orange (5A3) in margins. DG18 25 °C, 14 d: centrally raised; colony surface floccose, margin filiform; mycelial areas pinkish white (7A2); sporulation greyish turquoise (24D5); soluble pigment absent; exudate absent; reverse centrally orange white (5A2) to yellowish white (4A2) in margins. MEA 25 °C, 14 d: convex; colony surface floccose, margin undulate; mycelial areas white; sporulation sparse; soluble pigment absent; exudate absent; reverse centrally clay brownish orange (5C5) to pale orange (5A3) in margins. M60Y 25 °C, 14 d: raised; colony surface floccose; margin irregular, filiform; mycelial areas white; sporulation pastel green (25B4); soluble pigment absent; exudate absent; reverse centrally olive grey (2F2) to champagne greyish yellow (4B4) in margins. MEA + 10 % NaCl 25 °C, 14 d: flat; colony surface floccose, margin filiform; mycelial areas pinkish white (7A2); sporulation greyish turquoise (24D5); soluble pigment absent; exudate absent; reverse centrally cinnamon brown (6D6) to champagne greyish yellow (4B4) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads loosely columnar. Conidiophores uniseriate. Stipes smooth, densely covered by long hairs in SEM, 5–6 μm wide in the middle part, widening gradually toward the vesicle, with occasional septa. Vesicles pyriform, spatulate to clavate, 10–14 μm wide. Phialides flask-shaped, densely covered by long hairs in SEM, 9–12 μm long, covering the upper half of the vesicle. Conidia borne as smooth, long cylinders, echinulate at maturity, tuberculate in SEM, ellipsoidal or barrel-shaped, connectives evident, 3.5–4.5 × 2.5–3.5 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: See the distinguishing characters of A. conicus. The phylogenetic position of A. villosus in the A. conicus clade is not fully resolved. It is a member of the A. conicus clade based on the *BEAST analysis, but the concatenated dataset analysis resolves it outside the clade with absolute support. Probably its closest relatives are A. conicus, A. domesticus and A. destruens. Aspergillus villosus can be distinguished by its faster growth on CY20S, ability to grow on CY20S at 30 °C and on M60Y at 37 °C. Additionally, A. villosus has rather spatulate vesicles in contrast to pyriform or ellipsoidal vesicles in mentioned relatives; conidia ornamentation in SEM is tuberculate, rather than aculeate common in the A. conicus clade. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: France, Brest, painting from Brittany (painter: A.D. Hondius; painting title: Jésus à la piscine de Bethsaïda ou La guérison du paralytique; painting technique: oil on canvas), 2015, isolated by F. Déniel, UBOCC-A-116020 = CCF 5532 = IBT 34624. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Aspergillus vitricola Ohtsuki, Bot. Mag. (Tokyo) 75: 436. 1962, emended description. MycoBank MB326665. Fig. 34. ",
      " ",
      "\n",
      "\n",
      "Fig. 34 Aspergillus vitricola. A. Colonies after 14 d at 25 °C: top row left to right, obverse M40Y, CYA, CY20S and DG18; bottom row left to right, reverse M40Y, MEA, M60Y and MEA + 10 % NaCl. B, C. Conidiophores. D, E. Conidial heads. F. Conidia. G. Conidia in SEM. H. Stipe in SEM. Scale bars: B, C, F = 10 μm; G, H = 2 μm.Fig. 34\n",
      "\n",
      "\n",
      "Typus: Japan, binocular lens, 1962, collector unknown (holotype in the Herbarium of the Nagao Institute, culture ex-type NRRL 5125 = DTO 356-F7 = ATCC 16905 = ATCC 36505 = IMI 108298 = CCF 5533 = IBT 34530). ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony diam, 14 d (mm): M40Y 19–36; CYA No growth; CY20S (0)4–7; CY20S 30 °C No growth; CY20S 37 °C No growth; DG18 8–18; MEA No growth; M60Y 25–42; M60Y 30 °C 16–42; M60Y 37 °C No growth; MEA + 10 % NaCl 9–26. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Colony characters: M40Y 25 °C, 14 d: Colonies flat, wrinkled, surface velutinous; margin entire to delicately filiform; mycelial areas white; sporulation greyish turquoise (water blue) (24C5); soluble pigment absent; exudate absent; reverse centrally olive green (2F6) to olive yellow (3D6), champagne greyish yellow (4B4) in margins. CYA 25 °C, 14 d: no growth. CY20S 25 °C, 14 d: no growth to microcolonies. DG18 25 °C, 14 d: centrally raised; colony surface velutinous, margin entire to delicately filiform; mycelial areas white; sporulation sparse, turquoise white (24A2); soluble pigment absent; exudate absent; reverse centrally turquoise white (24D2) to orange white (5A2) in margins. MEA 25 °C, 14 d: no growth. M60Y 25 °C, 14 d: flat to slightly centrally raised; colony surface velutinous to powdery; margin entire to delicately filiform; mycelial areas white; sporulation deep turquoise (25E8); soluble pigment absent; exudate absent; reverse centrally dark green (29F3) to champagne greyish yellow (4B4) in margins. MEA + 10 % NaCl 25 °C, 14 d: flat, wrinkled; colony surface velutinous, margin entire to delicately filiform; mycelial areas white; sporulation aquamarine greyish turquoise (24B3); soluble pigment absent; exudate absent; reverse centrally cinnamon brown (6D6) to olive yellow (3D6), champagne greyish yellow (4B4) in margins. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Micromorphology: Conidial heads radiate. Conidiophores uniseriate. Stipes smooth in SEM, 4.5–6 μm wide in the middle part, widening gradually toward the vesicle, with occasional septa. Vesicles spatulate, pyriform to clavate, 7–12 μm wide. Phialides flask-shaped, 8–10 μm long, covering the upper half to two thirds of the vesicle. Conidia borne as rough-walled long cylinders, echinulate at maturity, aculeate in SEM, subglobose to ellipsoidal, connectives evident, commonly remaining in chains, 4.5–5.5 × 3–4 μm. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Distinguishing characters: Isolates of A. vitricola have larger conidia, narrower stipes and much smaller vesicles compared to A. glabripes. Aspergillus vitricola does not grow on MEA and CYA at 25 °C and CY20S at 30 °C and exhibit only very restricted growth on CY20S at 25 °C. Both species are also different micromorphologically and in growth parameters on most media. ",
      " ",
      "\n",
      "\n",
      "\n",
      "Additional materials examined: Canada, British Columbia, Victoria, house dust, 2015, collected by B. Kendrick, isolated by C.M. Visagie, KAS 6093, KAS 6087 = DTO 356-H2 = IBT 34532, KAS 6199, KAS 6237 = DTO 356-I2, KAS 6238, KAS 6281. Canada, New Brunswick, Little Lepreau, house dust, 2015, collected by A. Walker, isolated by C.M. Visagie, DAOMC 251500 = KAS 6133, KAS 6086, KAS 6178 = DTO 356-H8, KAS 6279. Canada, Nova Scotia, Wolfville, house dust, 2015, collected by A. Walker, isolated by C.M. Visagie, KAS 6150 = DTO 356-H5 = IBT 34531, KAS 6235, KAS 6236, KAS 6250, KAS 6251, KAS 6263, KAS 6264, KAS 6278 = DTO 356-I8, KAS 6285, KAS 6287, KAS 6288, KAS 6290. Canada, Ontario, Stittsville, house dust, collected by K.A. Seifert, isolated by C.M. Visagie, KAS 5687. Slovenia, Ljubljana, mouldy hat of a national costume of a folklore group (stored in basement depot), 2015, isolated by P. Zalar, EXF-10301 = CCF 5654 = IBT 34290. Slovenia, Ljubljana, oil painting on canvas, 2015, isolated by P. Zalar and D.D. Graf, EXF-10383 = CCF 5655 = IBT 34272. Slovenia, Ljubljana, oil painting on canvas, 2012, isolated by P. Zalar, EXF-7700 = CCF 5657 = IBT 34282 = IBT 33575. France, raisin madeleines, 2014, isolated by F. Déniel, CBS 140435 = UBOCC-A-115047 = DTO 334-E6. The Netherlands, archive material, 2006, isolated by J. Houbraken, DTO 014-A4. The Netherlands, Noordwijk, carpet “zeegrastapijt” in house, 2015, J. Houbraken, DTO 331-D6. The Netherlands, Gorinchem, archive material, 2010, isolated by M. Meijer and O. Terhoeven, DTO 122-I4, DTO 123-B2, DTO 122-I5. USA, California, San Diego, child carrier, 2015, isolated by Ž. Jurjević, CCF 5534 = EMSL No. 2785.\n",
      "\n",
      "\n",
      "Dichotomous key to species from section Restricti\n",
      "\n",
      "\n",
      "1a)\n",
      "\n",
      "Homothallic, no growth on M40Y at 25 °C …………………………………………….… A. halophilicus1b)\n",
      "\n",
      "Anamorphic, good growth on M40Y at 25 °C ………… 2 ",
      " ",
      "2a)\n",
      "\n",
      "Conidial heads from colony edge radiate or globose when young ……………………………………………… 112b)\n",
      "\n",
      "Conidial heads columnar or loosely columnar when young ……………………………………………………… 3\n",
      "\n",
      "\n",
      "Aspergillus restrictus/conicus clades\n",
      "\n",
      "\n",
      "3a)\n",
      "\n",
      "No growth on CY20S at 30 °C and M60Y at 37 °C after 14 d ………………………………………………………. 43b)\n",
      "\n",
      "Growth on CY20S at 30 °C and M60Y at 37 °C after 14 d ………………………………………………………. 8 ",
      " ",
      "4a)\n",
      "\n",
      "Colonies on M60Y at 25 and 30 °C exceed 40 mm after 14 d ………………………………………………………. 54b)\n",
      "\n",
      "Colonies on M60Y at 25 and 30 °C do not exceed 40 mm after 14 d ……………….………………………. 6 ",
      " ",
      "5a)\n",
      "\n",
      "Stipes 3–5 μm wide, vesicles 6.5–9 μm wide, phialides 7.5–9 μm long, reverse colour on M40Y olive grey in the centre ……………………………………… A. gracilis5b)\n",
      "\n",
      "Stipes 4–6 μm wide, vesicles 9–13 μm wide, phialides 9–10.5 μm long, reverse colour on M40Y cinnamon brown in the centre …………………… A. pseudogracilis ",
      " ",
      "6a)\n",
      "\n",
      "Sporulation colour on M40Y myrtle green, reverse apricot greyish orange in the centre ……... A. destruens6b)\n",
      "\n",
      "Sporulation colour on M40Y turquoise, reverse olive grey in the centre ……………………………………..… 7 ",
      " ",
      "7a)\n",
      "\n",
      "Sporulation colour on M40Y pale turquoise, conidia predominantly globose to subglobose (or barrel-shaped) ……………………………………………. A. conicus7b)\n",
      "\n",
      "Sporulation colour on M40Y greyish turquoise, conidia predominantly ellipsoidal (or barrel-shaped) ……………………………………………... A. domesticus ",
      " ",
      "8a)\n",
      "\n",
      "Length of phialides ≥ 9 μm, colonies on CY20S at 30 °C and CYA at 25 °C and M60Y at 37 °C ≤ 10 mm after 14 d, conidia tuberculate in SEM ……… A. villosus8b)\n",
      "\n",
      "Length of phialides ≤ 9 μm, colonies on CY20S at 30 °C and CYA at 25 °C and M60Y at 37 °C > 10 mm after 14 d, conidia aculeate in SEM …………………… 9 ",
      " ",
      "9a)\n",
      "\n",
      "Diameter of vesicle ≥ 16 μm …………... A. pachycaulis9a)\n",
      "\n",
      "Diameter of vesicle ≤ 16 μm ………………………… 10 ",
      " ",
      "10a)\n",
      "\n",
      "Conidia 3.5–4.5 × 2.5–3 μm, vesicle diameter 9–12 μm ……………………………………. A. restrictus10b)\n",
      "\n",
      "Conidia 4.5–6 × 3–4.5 μm, vesicle diameter 11–16 μm ……………………………………………….. A. caesiellus\n",
      "\n",
      "\n",
      "Aspergillus penicillioides/vitricola clades\n",
      "\n",
      "\n",
      "11a)\n",
      "\n",
      "Average diameter of vesicles ≥ 19 μm ……………... 1211b)\n",
      "\n",
      "Average diameter of vesicles < 17 μm ………………. 13 ",
      " ",
      "12a)\n",
      "\n",
      "Conidia 3.5–4.5 × 2–3 μm, length of phialides 7–9 μm ………………………………………………... A. glabripes12b)\n",
      "\n",
      "Conidia 4.5–5.5 × 3.5–4.5 μm, length of phialides 9–11 μm …………………………… A. magnivesiculatus ",
      " ",
      "13a)\n",
      "\n",
      "Length of phialides ≤ 8 μm, average length ∼ 7 μm …………………………………………..…. A. canadensis13b)\n",
      "\n",
      "Length of phialides ≥ 8 μm, average length ≥ 8.5 μm …………………………………………………………… 14 ",
      " ",
      "14a)\n",
      "\n",
      "Conidia do not exceed 4 μm in long axis, average length ca 3.6 μm ……………………………. A. salinicola14b)\n",
      "\n",
      "Conidia commonly exceed 4 μm in long axis, average length ca 4–5 μm ……………………………………… 15 ",
      " ",
      "15a)\n",
      "\n",
      "Colony diameter on CY20S and M40Y at 25 °C after 14 d > 20 and 45 mm, respectively ……… A. reticulatus15b)\n",
      "\n",
      "Colony diameter smaller on both media …………….. 16 ",
      " ",
      "16a)\n",
      "\n",
      "Average width of stipes ≤ 5.5 μm ……………………………………………………………1716b)\n",
      "\n",
      "Average width of stipes frequently ≥ 6 μm ………………………………………………………..…. 19 ",
      " ",
      "17a)\n",
      "\n",
      "No production of soluble pigment on DG18, average length of conidia > 4.5 μm, stipes smooth in SEM ………………………………………………….. A. vitricola17b)\n",
      "\n",
      "Greyish yellow soluble pigment on DG18, average length of conidia ca 4 μm, stipes covered by hairs in SEM …………………………………………………….. 18 ",
      " ",
      "18a)\n",
      "\n",
      "Sporulation colour on M40Y deep green, conidia in SEM tuberculate ……………………….. A. penicillioides18b)\n",
      "\n",
      "Sporulation colour on M40Y dark turquoise, conidia in SEM lobate-reticulate …………………………. A. hordei ",
      " ",
      "19a)\n",
      "\n",
      "No growth on CYA at 25 °C after 14 d, colony reverse on M40Y in shades of yellow or brown, conidia tuberculate in SEM ………………………………………….. 2019b)\n",
      "\n",
      "Growth on CYA at 25 °C after 14 d, colony reverse on M40Y dark green, conidia aculeate in SEM ………………………………………………. A. infrequens ",
      " ",
      "20a)\n",
      "\n",
      "Sporulation colour on M40Y greyish green, average dimensions of conidia 4.8 × 3.5 μm, colony diameter on CY20S at 25 °C after 14 d > 6 mm ….. A. clavatophorus20b)\n",
      "\n",
      "Sporulation colour on M40Y greyish turquoise (water blue), average dimensions of conidia 4 × 2.9 μm, colony diameter on CY20S at 25 °C after 14 d < 6 mm …………………………………………... A. tardicrescens\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "\n",
      "Past and current concept of section Restricti\n",
      "\n",
      "\n",
      "Thom & Church (1926) were the first who studied the classification of the species of sect. Restricti. They placed A. penicilliodies, A. gracilis and A. conicus into a transitional group between A. glaucus and A. fumigatus. Thom & Raper (1945) included also A. restrictus and considered the members of sect. Restricti as a series within the A. glaucus group. The A. restrictus group was subsequently introduced by Raper & Fennell (1965) who added also A. caesiellus and treated A. vitricola as a synonym of A. penicillioides. Although the first described species was A. penicillioides, the authors decided to name the group after the species (A. restrictus) they considered most important. The authors also observed variability within A. restrictus and mentioned unique growth characterictics of isolate NRRL 145. This isolate is in the present study designated as the ex-type of a novel species A. destruens, a member of the A. conicus clade. This species resembles A. conicus rather than A. restrictus in almost all phenotypic characters.\n",
      "\n",
      "The taxonomic rank of section was later introduced by Samson & Gams (1985). The number of species forming the section was subsequently reduced to three by Pitt & Samson (1990), rejecting A. conicus and A. gracilis. Even though the morphological variability within the section is rather small, at least the clades (A. restrictus, A. conicus, A. vitricola and A. penicillioides) can be clearly distinguished solely based on morphology.\n",
      "\n",
      "Molecular studies conducted by Peterson (2000) confirmed that re-expansion of the section was necessary, supported recognition of at least six anamorphic species and placed A. halophilicus, which produces a homothallic teleomorph similar to those seen in sect. Aspergillus, into the sect. Restricti.\n",
      "\n",
      "In this study, we performed a new revision of sect. Restricti that involved a large number of isolates and used both phenotypic and molecular genetic data for species delimitation. The number of accepted species was tripled compared to previous studies and 14 new species were described. The genetic distances between the species are greater (similarity between sequences of sister species within the clades varies from 95 % to 97 %, sequence similarity between the clades is only slightly above 90 %) than typically seen in other sections of the genus Aspergillus, including sister sect. Aspergillus. Our study could not sufficiently cover all common isolation sources of sect. Restricti members and description of new taxa in future can be expected if studies with extensive sampling and focused on particular substrates or environments are conducted. This assumption is supported by a comparison of our generated ITS barcodes with 1061 sect. Restricti 454-pyrosequences from Amend et al. (2010). A phylogenetic tree (Fig. 8) confirms that there is still uncovered species diversity within section Restricti and discovery of new species is expected in future. The data showed that the house dust conceals high number of species that belong to all species complexes recognized in this study including A. halophilicus. Aspergillus penicillioides, A. vitricola, A. hordei, A. tardicrescens and A. magnivesiculatus belonged to the most abundantly detected species. Several undescribed phylogenetic lineages are present in A. restrictus/conicus clade and one unnamed lineage sister to A. tardicrescens is present in A. penicillioides clade (Fig. 8).\n",
      "\n",
      "Two of the 21 species accepted in this work were represented only by one isolate and three species by only two isolates. This situation, although not often discussed, is very common in the taxonomy of fungi (Lim et al. 2012). Even though working with species with no infraspecific variability is problematic, it is a general practice to include also these species into the analyses because the authors always want to cover all known variability (Ahrens et al. 2016). The results of some phylogenetic methods may be biased due to the presence of species with low or no variability, and population genetic methods cannot be used at all with datasets containing species without infraspecific variability. Since the aim of taxonomic revisions and monographs is to describe all current knowledge we included all species irrespective of the number of isolates and presence or absence of infraspecific variability. When we re-inspect the results of our phylogenetic analyses, it is common that the unresolved nodes are occupied by those species represented by a few isolates. Better resolution of phylogenetic relationships can be expected when the inevitable additional strains of rare species and discovery of novel taxa are made.\n",
      "\n",
      "\n",
      "Phylogeny of subgenus Aspergillus\n",
      "\n",
      "In contrast to sect. Restricti, sect. Aspergillus forms a compact clade. The phylogenetic analysis across species in subg. Aspergillus strongly supported the monophyly of both, sect. Restricti and sect. Aspergillus (Fig. 1, Fig. 2). This topology is in agreement with trees published by Peterson (2008) and Peterson et al. (2008), however, especially in the latter study the sect. Restricti received only limited statistical support. In the more recent studies of Houbraken and Samson, 2011, Houbraken et al., 2014 and Kocsubé et al. (2016), members of section Restricti did not form a separate monophyletic clade with respect to sister sect. Aspergillus. This was probably caused by the inadequate taxon sampling (low number of species included). Large genetic distances between particular clades within sect. Restricti could warrant proposal of several new sections (e.g., A. restrictus+A. conicus+A. vitricola clades = section Restricti s. str.; A. penicillioides clade; and A. halophilicus) solely based on phylogenetic distances (Fig. 1, Fig. 2) and their comparison with those between sections in other subgenera. Several factors prevent us doing so. Most importantly, the section forms a coherent group in terms of morphology (except A. halophilicus) and is strikingly different from homothallic species of sister sect. Aspergillus that produce abundant ascomata in vitro. When viewing the phylogenetic tree of the subgenus in the radial form (Fig. 2), the real relationships between sections and clades seem clearer. There are at least three clades in sect. Restricti separated by large genetic distances that are similar to distances of these clades from basal taxa of sect. Aspergillus. One could even argue that there are enough clades on the tree spread along the transition between sections Restricti and Aspergillus to merge both sections into one (Fig. 2). Another problem is the unresolved position of some taxa belonging to sect. Restricti, especially A. halophilicus. Considering the uniqueness of A. halophilicus in terms of morphology, physiology, reproductive strategy in addition to phylogenetic isolation, it might seem appropriate to exclude this species from sect. Restricti and create a separate, single-species section. However, such action would probably result in a paraphyletic sect. Restricti due to the unclear position of A. halophilicus. Isolation of additional taxa related to A. halophilicus could improve results of the phylogenetic analyses and provide new phenotypic data supporting or contradicting this scenario. Furthermore, the sexual reproductive strategy of anamorphic species from sect. Restricti is unknown but discovery of a sexual state would not be unexpected in light of recent developments in this field (Horn et al., 2009, O'Gorman et al., 2009, Horn et al., 2011, Nováková et al., 2014).\n",
      "\n",
      "\n",
      "Exometabolites\n",
      "\n",
      "All species in Aspergillus sect. Restricti produced asperglaucide, asperphenamate or both. This is in contrast to species of Aspergillus sect. Aspergillus that do not produce these compounds (Chen et al. 2017). On the other hand the species in both sections do produce some extrolites in common. Mycophenolic acid is produced by several species from both sections, similarly to echinulin and derivatives such as neoechinulin-type metabolites (Chen et al. 2017). In sect. Restricti no isolates were found to produce auroglaucins or epiheveadride, compounds which are very common in the ascomata of Aspergillus sect. Aspergillus. The ascomata of A. halophilicus do not contain the yellow auroglaucin or flavoglaucin compounds. It is interesting to note that antarone A and asperentins produced by A. salinicola are shared with a halotolerant Penicillium antarcticum (Vansteelandt et al., 2012, Geiger et al., 2013).\n",
      "\n",
      "The asperglaucides and asperphenamate are simple phenylalanine derived compounds with anticancer activity (Bladt et al. 2013). These compounds have been found in many plants (Frisvad et al. 2013), and one can speculate that these medicinal compounds are not produced by the plants but rather by its endophytic fungi, including possibly endophytic isolates of Aspergillus sect. Restricti.\n",
      "\n",
      "A very high chemical diversity was observed in section Restricti, but a large number of extrolites have until now only been found in this section. The chemoconsistency on the species level appeared to be less pronounced than in for example section Aspergillus, but the absence of metabolites may be caused by poor growth in some cases. For example A. villosus NRRL 25813T, IBT 22529, IBT 22530 and IBT 22546 produced a large number of orthosporins, that were not detected in the strain UBOCC-A-116020. In such cases other media than those used here may be more suitable for production of a broader spectrum of metabolites. Addition of a high concentration of salts or sugars strongly increased the growth rate of isolates from section Restricti, and also secondary metabolite production. More media, such as potato dextrose agar and malt agars may be more suitable for secondary metabolite production if they are supplemented with more salt or sugar. The overview of metabolites produced by species from section Restricti reported from previous studies is given in Table 10. Most secondary metabolites reported from A. halophilicus by Micheluz et al. (2016), such as chaetoviridin A, deoxybrevianamide E, pseurotin A and D, rugulosuvine, stachybotryamide, tenellin and tryprostatin B could not be recovered from any strain of A. halophilicus examined in this study. However these extrolites may have been detected because of contamination of the YES 15 % medium (YES agar with 15 % NaCl) they used. Several extrolites detected by Micheluz et al. (2016) are products of Aspergillus fumigatus, including pseurotins A, D, tenellin and tryprostatin (Frisvad and Larsen, 2016), and were only detected in one of three examined isolates. However, YES agar with 15 % NaCl has not been used in this study, so a final conclusion on the production of these A. fumigatus metabolites by A. halophilicus must await further testing.\n",
      "\n",
      "Table 10\n",
      "Exometabolite\tProducer\tReference\n",
      "Arestrictin A and B\tA. restrictus\tItabashi \n",
      "Arestrictin A and B\tA. penicillioides\tItabashi \n",
      "Cristatin A\tA. restrictus\tItabashi \n",
      "Neoechinulin A\tA. halophilicus\tMicheluz \n",
      "Asperglaucide (= aurantiamide acetate)\tA. penicillioides\tIsshiki et al., 2001\n",
      "Asperglaucide (= aurantiamide acetate)\tA. restrictus\tMicheluz \n",
      "Rugulusovin\tA. halophilicus\tMicheluz \n",
      "Citreorosein\tA. penicillioides\tMicheluz \n",
      "Deoxybrevianamide A\tA. penicillioides\tMicheluz \n",
      "Deoxybrevianamide E\tA. halophilicus\tMicheluz \n",
      "Exometabolites incorrectly ascribed to sect. \n",
      "Restrictocin\tA. restrictus\tLamy and Davies, 1991\n",
      "Chaetoviridin A\tA. halophilicus\tMicheluz \n",
      "Pseurotin A & D\tA. halophilicus\tMicheluz \n",
      "Tenellin\tA. halophilicus\tMicheluz \n",
      "Tryprostatin B\tA. halophilicus\tMicheluz \n",
      "\n",
      "1\n",
      "\n",
      "2\n",
      "\n",
      "\n",
      "Ecology and pathogenicity\n",
      "\n",
      "Species of sect. Restricti occur worldwide and they used to be most frequently isolated from food, feed and indoor environments. Pitt & Hocking (2009) summarised spoilage of different foods (e.g., flour, wheat, maize, rice, dried, cured or salted meat, dried fruit, spices, beans, nuts, coffee beans and salted dried fish) by A. restrictus and A. penicillioides. Aspergillus penicillioides is probably the most frequently isolated species from the section although it may be partly due to the broad concept adopted in the past. Members of sect. Restricti are commonly isolated from grain, such as corn, wheat and barley, and many species were found on grain during this study, namely A. penicillioides, A. restrictus, A. caesiellus, A. magnivesiculatus, A. hordei, A. halophilicus, A. infrequens. The isolates from food examined in this study originated mostly from candy and bakery products (A. penicillioides, A. restrictus, A. glabripes, A. destruens, A. domesticus, A. conicus, A. vitricola, A. salinicola), rarely from other foods such as cheese (A. destruens), nuts (A. penicillioides, A. conicus), tea (A. clavatophorus), katsuobushi (A. magnivesiculatus), etc. Some examples of spoiled foods examined in this study are shown in Fig. 35. The species are also commonly isolated from leather and textile goods, e.g. shoes, baseball gloves (see Fig. 36 G, H) or mattress covers (A. penicillioides, A. restrictus, A. reticulatus, A. caesiellus). The presence of these species in the indoor environment and house dust is well-known from past studies (Samson and Lustgraaf, 1978, Frisvad and Gravesen, 1994, Meklin et al., 2004, Kaarakainen et al., 2009, Amend et al., 2010, Visagie et al., 2014). Eleven species which were isolated from indoor air are included in our study, namely A. caesiellus, A. conicus, A. destruens, A. domesticus, A. glabripes, A. magnivesiculatus, A. pachycaulis, A. penicillioides, A. restrictus, A. reticulatus and A. tardicrescens. Examples of sect. Restricti members growing in various conditions, including indoor environment and textile can be found in Fig. 36. In a world-wide survey of fungi occurring in house dust using a portion of the ID locus as used here (Amend et al. 2010), A. penicillioides was a common OTU identified. In these analyses a 97 % cutoff is most commonly used, however, the low amount of sequence data available for sect. Restricti before this study makes the analysis of species diversity complicated. This is clear from our sect. Restricti phylogeny (Fig. 8) incorporating newly generated ITS barcodes and 454-pyrosequences from Amend et al. (2010), where we can now positively identify 14 of the OTU's down to species level, in contrast to the at most seven previously possible in theory. One of the main problems is the high sequence divergence between sect. Restricti species where culture independent methods had almost no chance of getting to a correct species name. In this study, we release more than 180 ITS rDNA barcode sequences covering both infra- and interspecies variation. This will greatly aid these modern techniques to identify common indoor fungi. A significant portion of strains and new species examined in this study was isolated from air, especially from indoor air of various regions in the USA. It seems that this enviroment might still be an important source of undiscovered genetic variability and new species. Many specimens examined in this work also come from house dust; for instance isolates of A. canadensis thus far originate exclusively from house dust in Canada.\n",
      "\n",
      "Fig. 35 Different kinds of food and commodities on which members of section Restricti are to be found. A–F. Various bakery products: A. Chocolate chip cookies (“A. penicillioides” by morphology – isolate not included in this study; isolated together with A. cibarius). B. Mouldy bread. C. Madeleines (isolate UBOCC-A-115045 – A. reticulatus). D. French savaroise (isolate UBOCC-A-115043 – A. salinicola). E. French mini cakes (isolate UBOCC-A-115048 – A. domesticus). F. Mouldy cake (mixed spoilage with Penicillium and Eurotium spp.). G. Grain containing diverse fungal populations similar to nuts, spices and seeds. H. Cigars (“A. penicillioides” by morphology – isolate not included in this study).Fig. 35\n",
      "\n",
      "Fig. 36 Overview of some substrates and habitats associated with members of section Restricti. A. Solar salterns at the Adriatic coast, Sečovlje, Slovenia (isolate EXF 226 – A. salinicola). B. Indoor environment, a source of numerous species: one of the sampling sites in Canada where house dust was collected. C. Water damaged wall surface, dry at the time of sampling (isolate DTO 073-H6 – A. tardicrescens). D. Mouldy chair backrest (isolate DTO 231-B9 – A. domesticus). E. Textiles (“A. penicillioides” by morphology – isolate not included in this study). F. Textile water hose imported from China (“A. penicillioides” and Cladosporium sp. by morphology – isolates not included in this study). G. Leather baseball gloves (isolate CCF 5500 – A. penicillioides and CCF 5514 – A. restrictus). H. Leather armrest of a dentist chair (isolate DTO 316-A7 – A. tardicrescens). I. Skeleton of Elephas maximus (Asian elephant) (“A. penicillioides” by morphology – isolate not included in this study). J. Mouldy sclerotium of Corallocytostroma ornicopreoides imported from Australia (isolate CCF 3364 – A. restrictus).Fig. 36\n",
      "\n",
      "The species can also cause significant damage to specimens deposited in herbaria (Zhang et al. 2016) and also to works of art, especially paintings (Fig. 37). Examination of fungi isolated from deteriorated paintings from Musée des beaux Arts de Brest (France) and Institute for the Protection of Cultural Heritage of Slovenia during this study showed that species A. destruens, A. conicus, A. domesticus, A. vilosus, A. vitricola, A. glabripes, A. tardicrescens, A. reticulatus, A. magnivesiculatus and A. salinicola can deteriorate paintings and generate cultural and economic losses. Aspergillus clavatophorus was also isolated from a painting in the Netherlands, while A. halophilicus was recently identified as a causative agent of book biodeterioration in archives and libraries from Micheluz et al. (2015).\n",
      "\n",
      "Fig. 37 Deteriorated paintings and books from archives. A–H. Deteriorated painting from Capuchin monastery of St. Francis, Ajdovščina, Slovenia (A) and Musée des beaux Arts in Brest, France (B–H); isolated species: A. destruens, A. domesticus, A. conicus, A. glabripes, A. magnivesiculatus, A. reticulatus, A. salinicola, A. tardicrescens, A. villosus, A. vitricola. I–M. Deteriorated books from archives; isolated species: A. vitricola and A. domesticus.Fig. 37\n",
      "\n",
      "Hypersaline water has proved to be a rich source of xerophilic fungi (Gunde-Cimerman et al. 2000), including members of sect. Restricti (Cantrell et al., 2006, Nazareth and Gonsalves, 2014). The novel species A. salinicola described in this study is a saltern inhabitant (Fig. 36. A).\n",
      "\n",
      "Some species can be isolated from very uncommon substrates as in the case of A. vitricola that was isolated from glass in Japan and is able to damage optical lenses (Ohtsuki 1962). As we found here, this species is also very common in house dust in Canada.\n",
      "\n",
      "The pathogenic potential of sect. Restricti seems to be low considering the small number of reported cases of opportunistic human infections. Several records of human infection caused by A. penicillioides and A. restrictus is summarised by de Hoog et al. (2009), but the identification of the pathogens was performed exclusively by morphological examination. It is very likely that some records represent confusion with A. fumigatus due to similar sporulation colour and micromorphology. However, a confirmed case of invasive aspergillosis in human has been described recently that was probably caused by an undescribed species from the A. penicillioides clade (ITS GenBank: KP316198) (Gupta et al. 2016). This case demonstrated that athough the species of the sect. Restricti are rarely found as pathogens, they have at least limited pathogenic potential.\n"
     ]
    }
   ],
   "source": [
    "view_one_article(geospan_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "length_query = {\n",
    "    \"text_matches\": {\"$in\": terms},\n",
    "    \"article_meta.article_type\": \"research-article\",\n",
    "    \"article_meta.has_body\": True,\n",
    "    \"article_meta.n_geospans\": {\"$exists\": True},\n",
    "    \"$expr\": {\"$gte\": [{\"$strLenCP\": \"$extracted_text\"}, quantiles.length.iloc[lower_break]]},\n",
    "    \"$expr\": {\"$lte\": [{\"$strLenCP\": \"$extracted_text\"}, quantiles.length.iloc[upper_break]]}\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "6132394\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "Assessment of seafood contamination under the marine strategy framework directive: contributions of the German environmental specimen bank\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "{'all_geospans': 113, 'n_geospans': 113, 'article_type': 'research-article', 'has_body': True}\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['sampling site', 'sampling area']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Introduction\n",
      "\n",
      "Marine environments are under massive pressure caused by anthropogenic exploitation and pollution. This includes pollution with chemical substances and marine litter, extensive fishing activities, deterioration of the sea floor, e.g., by construction measures, extraction of minerals, and fishing with ground nets, and introduction of noise, e.g., by ships, construction, renewable energy, and tourism (Ban and Alder 2008; Boldt et al. 2014; Halpern et al. 2008; IOC/UNESCO et al. 2011). As a consequence, the biodiversity has decreased, fish stocks have declined, and the resilience of marine ecosystems has dwindled in many parts of the world (e.g., Caddy and Seijo 2005; Costello et al. 2010; Johnston and Roberts 2009; Lotze et al. 2011; Pauly et al. 2005). Acknowledging this, the European Union has adopted the Marine Strategy Framework Directive 2008/56/EC (MSFD) (EC 2008a, and its amendment Commission Decision (EU) 2017/848 (EC 2017a)) that aims at the conservation and protection of the EU marine waters.\n",
      "\n",
      "By 2020, a Good Environmental Status (GES) of the marine environment shall be achieved whereby the GES is defined as “the environmental status of marine waters where these provide ecologically diverse and dynamic oceans and seas which are clean, healthy and productive within their intrinsic conditions, and the use of the marine environment is at a level that is sustainable, thus safeguarding the potential for uses and activities by current and future generations…” (EC 2008a). This definition implies an ecosystem-based approach of assessment that considers structure and function of marine ecosystems (Borja et al. 2013; Walmsley et al. 2017).\n",
      "\n",
      "To assess the status of the ecosystems, the MSFD requires 11 descriptors (D) that address different elements of the marine environment. D1 and D4 relate to the state of the ecosystem, i.e., its biodiversity and food webs, while the other descriptors are pressure-related addressing issues like non-indigenous species (D2), commercial fishing (D3), eutrophication (D5), physical loss and disturbance of the seafloor (D6), hydrographical changes (D7), contamination (D8, D9), marine litter (D10), and energy and underwater noise (D11). Each of these qualitative descriptors is broken down to quantifiable criteria. In total, 42 criteria have been specified and standardized methods for monitoring and assessment have been defined (Borja et al. 2011; EC 2017a).\n",
      "\n",
      "Descriptors D8 and D9 both deal with contaminants. D8 refers to contaminants in marine water, sediment, or biota which are assessed against threshold values (i.e., values set in accordance with Water Framework Directive WFD (EC 2000) and its daughter directives (EC 2008b, 2013a) or, if not applicable or no value is set under the WFD, values set by Member States through regional or subregional cooperation) (EC 2017a).\n",
      "\n",
      "Descriptor 9 focuses on contaminants in fish and other seafood for human consumption. The number of contaminants assessed under D9 is lower compared to D8 and comprises mainly those for which regulatory levels for foodstuffs are set under Regulation (EC) No 1881/2006 and its amendments (EC 2006a, 2008c, 2011a, b, c). However, on the basis of risk assessments, Member States can choose to not consider contaminants and/or include additional contaminants, for which threshold values must then be established by the Member States through regional or subregional cooperation (EC 2017a).\n",
      "\n",
      "There is a link between Descriptors 8 and 9 (Gago et al. 2014; Law et al. 2010; Maggi et al. 2014; Swartenbroux et al. 2010; Walmsley et al. 2017; Zampoukas et al. 2012, 2014): Because many contaminants are transferred along the food web those of concern to marine fish will likely also be of concern to humans (Fleming et al. 2006). On the other hand, concentrations exceeding the regulatory levels for food will probably also affect the ecosystem because food regulatory levels are usually higher than thresholds for assessing environmental pollution (Swartenbroux et al. 2010).\n",
      "\n",
      "Monitoring seafood related to human health is different from monitoring biota for environmental purposes: For the latter, a high degree of standardization and geographical traceability of the samples are crucial to the derivation of temporal trends and the assessment of compliance with reference values. In contrast, monitoring of seafood contamination for human consumption relies on the edible fraction of a wide variety of commercially relevant species for which the precise origin is not relevant and often unknown (Gago et al. 2014; Swartenbroux et al. 2010). The MSFD, however, requires that the GES has to be achieved or maintained for a specified region or subregion. The species monitored in the context of D9 shall be “relevant to the marine region or subregion concerned” (EC 2017a) implying that the geographical origin of the samples should be known.\n",
      "\n",
      "The MSFD relies on the institutional structures and programs of existing Regional Sea Conventions to minimize monitoring costs and efforts, namely on the Convention for the Protection of the Marine Environment in the North-East Atlantic (OSPAR 1992), the Convention on the Protection of the Marine Environment in the Baltic Sea Area (HELCOM 1992), the Convention for the Protection of Marine Environment and the Coastal Region of the Mediterranean (UNEP-MAP 1995), and the Convention for the Protection of the Black Sea (Bucharest Convention 1992).\n",
      "\n",
      "However, neither OSPAR nor HELCOM have so far developed indicators for the assessment of the status of fish and shellfish contamination in relation to human health as required by D9 (HELCOM 2017a; OSPAR 2017). In Germany, no marine monitoring programs exist that routinely collect georeferenced samples of edible fish in coastal waters and the German Exclusive Economic Zone (EEZ). A systematic assessment of fish addressing the specific requirements of MSFD is thus not possible to date. The recent assessment of the status of the marine environment in German waters under the MSFD mainly relies on samples from mussels in national waters and consequently does not fully address the requirements of D9 (BMUB 2018a, b).\n",
      "\n",
      "Against this background, the question arises of whether other monitoring programs can step in and provide data that can help assess D9. In Germany, the German Environmental Specimen Bank (ESB) collects and archives environmental specimens from defined terrestrial, limnetic, and marine ecosystems since the mid-1980s (www.umweltprobenbank.de/en). All samples are analyzed for a set of classical contaminants (Rüdel et al. 2010). Meanwhile, long time series are available highlighting, e.g., effects of regulatory measures. Furthermore, the archived samples allow to retrospectively analyze additional chemicals of concern that come into focus (e.g., Rüdel et al. 2003, 2011). In the North and the Baltic Seas the ESB routinely collects bladder wrack (Fucus vesiculosus), blue mussels (Mytilus edulis), eelpout (Zoarces viviparous), and eggs from sea gulls (Larus argentatus).\n",
      "\n",
      "With respect to D9 especially blue mussels and eelpout may be of interest. All samples are georeferenced and the archive allows the integration of new contaminants into the assessment or when time trends are required. However, regardless of the high operating standards of the ESB, it remains to be proven whether the nature of the samples and their handling and analysis are compatible with the requirements of the D9 monitoring under the MSFD.\n",
      "\n",
      "The aim of the present study is to evaluate the suitability of the ESB samples for D9 assessment and, based on ESB data, to exemplarily assess D9 in the coastal marine regions of Germany whereby not only those contaminants are considered that are currently assessed under D9 but also substances that may be relevant for D9 assessment in the future.\n",
      "\n",
      "\n",
      "Material and methods\n",
      "\n",
      "\n",
      "Sampling sites\n",
      "\n",
      "The marine sampling sites of the ESB (Fig. 1) are located in the coastal areas of the Central North Sea (FAO/ICES Division 27.4.b) and the Baltic Sea West of Bornholm (FAO/ICES Subdivision 27.3d.24). The sites are clearly defined and georeferenced and partly extend beyond the one nautical mile zone. More details are given in Rüdel et al. (2003) and at www.umweltprobenbank.de/en.\n",
      "\n",
      "Fig. 1 Marine sampling sites of the German Environmental Specimen Bank\n",
      "\n",
      "The two North Sea (NS) sampling areas are part of the National Park Wadden Sea, more precisely of the National Parks and Biosphere Reserves “Lower Saxony Wadden Sea,” and “Schleswig-Holstein Wadden Sea” that are listed as UNESCO World Heritage sites. The areas are extensively exploited by fishery and subject to riverine inputs of hazardous substances via the rivers Elbe, Weser, and Ems.\n",
      "\n",
      "In the Lower Saxony Wadden Sea (referred to as NS 1), the sampling site for mussels is located at the outlet of the Jade bight. Eelpout are fished in the Varel-Mellum Transect which extends over the entire tideway of the Jade Bight. Both sampling sites are influenced by emissions from agriculture and industries. Further contaminations stem from dumping and injection of dredged material from the Weser.\n",
      "\n",
      "In the Schleswig-Holstein Wadden Sea (referred to as NS 2), mussels are collected in a bay at the north end of the island Sylt which is part of the Sylt-Römö Wadden Sea. At low tide, only a small tideway connects the bay with the Sylt-Römo Wadden Sea, while about three quarters of the bay are intertidal. Contamination in this area originates mainly from atmospheric deposition. In contrast to the rather protected mussel sampling site, the sampling area for eelpout is located further south in the Meldorf Bay that is part of the Germany Bight and opens to the North Sea. It lies directly north of the Elbe estuary and is exposed to a large spectrum of contaminants carried by the Elbe. Eelpouts are fished in the sublitoral areas of gat Piep that drains the Meldorf Bay.\n",
      "\n",
      "The Baltic Sea sampling site (referred to as BS) for mussels and eelpout is located in the Bodden National Park of Western Pomerania which extends along the coast of Western Pomerania and includes several islands and the peninsula Fischland/Darß/Zingst. The national park is part of the world’s largest brackish water habitat and is extensively used by fishery and tourism.\n",
      "\n",
      "Both mussels and eelpout are sampled at Darßer Ort, the northernmost point of the Fischland/Darß/Zingst peninsula. The sampling area covers the coastal area up to two nautical miles off Darßer Ort. Contaminations in this area stem mainly from agriculture, atmospheric deposition, and ship traffic.\n",
      "\n",
      "\n",
      "Sampling and processing\n",
      "\n",
      "Sampling and processing under the German ESB program is highly standardized and follows the Standard Operating Procedures (SOPs) of the ESB (www.umweltprobenbank.de/en). Sampling and biometric characterization is performed by the ESB project team of Trier University. The entire process is accredited according to EN ISO/IEC 17025.\n",
      "\n",
      "Blue mussels are collected every 2 months at both North Sea sites and twice per year in the Baltic Sea (June and November). Directly after sampling, the mussels are weighed and deep-frozen at < − 130 °C. In the lab, the soft bodies are carefully removed without allowing the samples to thaw. The respiratory water and intestinal content remain in the mussel as part of the sample, whereby the amount of respiration water is recorded (Paulus et al. 2018) (Table S1, Supplementary material).\n",
      "\n",
      "Eelpout is sampled once per year in early summer (May–June) before the mating season to avoid any bias caused by reproductive activity, e.g., through transfer of lipid-associated contaminants to the gonads and eggs (Greenfield et al. 2005). Immediately after sampling, the fish are biometrically characterized and dissected under clean air conditions. The liver and both skinless fillets are separately deep-frozen at < − 130 °C (Klein et al. 2018) (Table S1, Supplementary material).\n",
      "\n",
      "In the laboratory, the tissues are cryo-milled to a homogenous powder and pool samples are prepared. All eelpout fillets (or livers) and all blue mussel soft bodies sampled in 1 year at one site are pooled resulting in one annual pool sample per specimen and sampling site. Aliquots of the samples are stored in the ESB archive at temperatures below − 150 °C in an inert atmosphere (from evaporating liquid nitrogen coolant) to minimize chemical alterations (for details see Rüdel et al. 2003, 2010). Since the pollutants covered here are assessed as persistent, no concentration changes even after long storage periods are expected under these conditions.\n",
      "\n",
      "\n",
      "Chemical analysis\n",
      "\n",
      "Chemical analysis of metals and tributyltin (TBT) compounds was performed at Fraunhofer IME, Schmallenberg. Organic compounds were analyzed by Eurofins GfA Lab Service, Hamburg. All methods are documented in laboratory-specific SOPs. Both laboratories have implemented quality assurance/quality control (QA/QC) measures. Table S2a–e (Supplementary material) summarizes the relevant QA/QC parameters.\n",
      "\n",
      "Metal analyses follow the SOPs of the ESB (www.umweltprobenbank.de). Detailed descriptions are also given in Rüdel et al. (2010). Lead (Pb) and cadmium (Cd) were measured by ICP-MS, mercury (Hg) was determined using a Direct Mercury Analyzer (DMA-80, MLS, Leutkirch, Germany). Metal data were not corrected for recovery (no extraction but digestion of sample material; digestion efficiency confirmed by recovery data of appropriate certified reference materials).\n",
      "\n",
      "Trace levels of TBT in current biota samples were analyzed after digestion, derivatization, and extraction by a validated sensitive SID-GC/ICP-MS method (Species Specific Isotope Dilution-Gas Chromatography-Inductively Coupled Plasma-Mass Spectrometry). Details are described in Radermacher (2015). Samples from years before 2009 were analyzed by gas chromatography applying an atomic emission detector (Rüdel et al. 2009). TBT data were not corrected for recovery.\n",
      "\n",
      "Dioxins and furans (PCDD/F) and polychlorinated biphenyls (PCB) were analyzed by high-resolution gas chromatography followed by high-resolution mass spectrometry (HRGC/HRMS) using the isotope dilution method. Polybrominated diphenylethers (PBDE), benzo[a]pyrene (B[a]P), benzo[a]anthracene, (B[a]A), benzo[b,j,k]fluoaranthene, and chrysene (analyzed together with triphenylene) were determined by gas chromatography coupled to mass spectrometry (GC/MS). Quantification was carried out using isotope-labeled internal standards. Response factors were taken into account thus considering a correction of data for extraction efficiency. For more details, see Fliedner et al. (2016).\n",
      "\n",
      "Hexabromocyclododecane (HBCDD, reported here as sum of α-, β-, and γ-diastereomers) and perfluorooctane sulfonic acid (PFOS) were analyzed by means of liquid chromatography coupled to mass spectrometry (LC/MS-MS). Quantification was carried out using isotope-labeled internal standards. Response factors were taken into account thus considering a correction of data for extraction efficiency. For more details, see Fliedner et al. (2016).\n",
      "\n",
      "Quantification of the target compounds included the use of either internal or external standards. If available, isotope-labeled standards were applied. During extraction, clean-up and measuring method blanks were analyzed in parallel to each batch of samples (at least one method blank per 12 samples). Precision and accuracy were monitored along with each batch of samples by analyzing in-house QA matrix samples, sample material of previous interlaboratory proficiency studies, or certified reference material.\n",
      "\n",
      "\n",
      "Data treatment\n",
      "\n",
      "In accordance with the MSFD (EC 2008a) and Reg. (EC) No.1881/2006 (EC 2006a), all data are expressed on a wet weight (ww) basis. Concentrations of PCDD/Fs and dl-PCBs are given as “WHO toxicity equivalents” (TEQ) that combine the toxicity of a defined set of dioxins, furans, and dioxin-like PCB (Van den Berg et al. 2006). Data are expressed as upper bound concentrations, i.e., the calculation includes the respective concentrations of the limits of quantification (LOQs) for congeners below or equal to the limit of quantification (LOQ).\n",
      "\n",
      "Differences between contaminant concentrations at different sites were compared using the Mann-Whitney test (VassarStats 2018, http://vassarstats.net/index.html). Temporal trends were analyzed using the MS-EXCEL-based software tool “LOESS Trend” (Version 1.1) developed by J. Wellmitz (German Environment Agency). The tool fits a locally weighted scatterplot smoother (LOESS; fixed window width of 7 years) through the annual contaminant data followed by tests for significance of linear and non-linear trend components by means of an analysis of variance (ANOVA) (Fryer and Nicholson 1999). Based on the LOESS function and a t test, the tool also allows to determine the significance of differences between selected time points and the end of the time series provided that the time span in between is at least 7 years (contrast test).\n",
      "\n",
      "\n",
      "Results and discussion\n",
      "\n",
      "\n",
      "Consistencies and differences between MSFD requirements and ESB standards\n",
      "\n",
      "The suitability of the ESB samples for D9 assessment was evaluated with respect to the MSFD requirements. Table 1 summarizes the main results. More details are given in Tables S1 and S2 (Supporting Information).\n",
      "\n",
      "Table 1\n",
      "MSFD requirements regarding\tCompliance with the MSFD requirements\tRemarks\n",
      "ESB mussels\tESB eelpout\n",
      "Geographical scope\t(x)\tEEZ not covered\n",
      "Temporal scope\tx\t(x)\tWorst case for eelpout\n",
      "Species\tx\t(x)\tEelpout are not among the most caught or consumed species\n",
      "Sampling\t(x)\tPersonnel not authorized according to Reg. (EC) No. 333/2007\n",
      "Sample processing\t(x)\tx\tBreathing water included in mussel samples\n",
      "Analyzed contaminants\tx\n",
      "Chemical analysis\t(x)\tESB method analyzes benzo[b,j,k]fluoranthene and chrysene / triphenylene as co-elutions\n",
      "\n",
      "\n",
      "Geographical and temporal scope\n",
      "\n",
      "D9 monitoring requires that “the temporal and geographical scope of sampling is adequate to provide a representative sample of the specified contaminants in fish and other seafood in the marine region or subregion” (EC 2017a). The ESB samples cover three coastal regions that are considered representative of German coasts. However, the fishing grounds in the open seas of the EEZ are not covered. With respect to the temporal scope, the ESB sampling of blue mussels meet the D9 requirements. In the case of eelpout, the samples are not representative for the whole year but may be considered as worst-case scenario with respect to lipophilic contaminants in fillets because sampling takes place prior to mating season and thus before body lipids are transferred to the reproductive tissue (Greenfield et al. 2005).\n",
      "\n",
      "\n",
      "Species\n",
      "\n",
      "The MSFD does not predefine the fish or mussel species and sizes to be analyzed for D9 assessment. The only specifications are that the species has to (1) be relevant to the marine region, (2) fall under the scope of Reg. (EC) No. 1881/2006, (3) be suitable for the contaminants, and (4) be among the most consumed in the Member States or the most caught or harvested for consumption. An indicative list with species is included in Swartenbroux et al. (2010). Zampoukas et al. (2014) also recommends to consider the ability of the species to biomagnify specific classes of contaminants and to ensure that different trophic levels and habitats are represented.\n",
      "\n",
      "Blue mussels meet all the above mentioned criteria: They are frequent inhabitants of the German coastal regions and are commercially exploited for human consumption. The mussels live attached to rocks, poles, and other hard substrates in intertidal areas. As filter feeders they are exposed to both water soluble and particle bound contaminants and are therefore widely used as bioindicators and chemical pollution monitoring species in coastal waters (review: Beyer et al. 2017).\n",
      "\n",
      "The suitability of eelpout for D9 assessment is less obvious. Eelpouts are abundant demersal fish in the German coastal regions. They have low migratory behavior and are often used as coastal bioindicators for biological and chemical effect assessments (Hedman et al. 2011; HELCOM 2017b; OSPAR 2013). However, while they are a welcome by-catch in commercial fisheries, they are not specifically fished upon and are not among the most caught or consumed species in Germany (listed in Centenera 2014).\n",
      "\n",
      "Nevertheless, the contamination of eelpout can give an indication of the contamination of some of the most consumed fish because of similarity of exposure and habitat. Eelpout feed mainly on bottom-dwelling organisms like snails, insect larvae, crustaceans, and eggs and fry of fish. Their trophic level (TL) is around 3.5 which makes them comparable to, e.g., plaice (Pleuronectes platessa, TL 3.2) and flounder (Platichthys flesus, TL 3.3; all TL data according to Froese and Pauly 2017). Like plaice and flounder, eelpouts live in close contact to the sediment and are exposed not only to contaminants in the water phase and to bioaccumulated contaminants via trophic transfer but also to substances bound to the sediment.\n",
      "\n",
      "The suitability of eelpout for D9 assessment is supported by data from Karl and Lahrssen-Wiederholt (2009) and Karl et al. (2010) who analyzed PCDD/Fs, dl-PCBs, and ndl-PCBs in cod (Gadus morhua) and herring (Clupea harengus) from georeferenced sites in the North and Baltic Seas: After lipid normalization levels in eelpout, cod and herring differ by no more than a factor of 2.\n",
      "\n",
      "\n",
      "Sampling and sample processing\n",
      "\n",
      "According to Commission Decision (EU) 2017/848, sampling for the assessment of the maximum levels of contaminants (D9) shall be performed in accordance with the quality standards required in food legislation laid down in Reg. (EC) No. 882/2004 (regarding the performance of controls to ensure compliance with feed and food law, animal health and animal welfare; EC 2004), Reg. (EU) No. 644/2017 (regarding sampling and analysis for the control of dioxins, dl-PCBs and ndl-PCBs; EC 2017b), and Reg. (EC) No. 333/2007 (regarding sampling and analysis of metals and B[a]P; EC 2007 amended by EC 2011d).\n",
      "\n",
      "The ESB meets most requirements regarding sampling and analysis for the control of dioxins, dl-PCBs, ndl-PCBs, and metals or has even stricter standards. Details are given in Tables S1 and S2 (Supporting material).\n",
      "\n",
      "However, sampling under the ESB is not performed by officially authorized personnel according to Reg. (EC) No. 333/2007 and Reg. (EC) No. 644/2017, and no controls according to Reg. (EU) No. 882/2004 are performed during sampling and processing.\n",
      "\n",
      "In the case of blue mussels, it should also be kept in mind that ESB mussel samples include the breathing water which accounts for in average 58% of the mussel wet weight at NS 2 and 67% at NS 1 and BS. Accordingly, respective factors of 3 (NS 1), 2.4 (NS 2), and 3 (BS) should be considered when comparing mussel data with maximum levels (MLs).\n",
      "\n",
      "\n",
      "Analyzed contaminants and chemical analyses\n",
      "\n",
      "D9 refers to contaminants in fish and other seafood for which regulatory levels have been set to protect human consumers. These are heavy metals (Pb, Cd, Hg), PAHs (benzo[a]pyrene, benzo[a]anthracene, benzo[b]fluoranthene, and chrysene), dioxins, furans and dioxin-like PCBs (PCDD/Fs + dl-PCBs), and non-dioxin-like PCBs (ndl-PCBs).\n",
      "\n",
      "Currently, all D9-relevant contaminants are covered by the ESB. The substances are analyzed either in blue mussel or eelpout samples (or in both). For PAHs, however, the analytical method used within the ESB program might lead to an overestimation regarding benzo[b]fluoranthene and chrysene to which the regulatory levels apply (EC 2011c) because the ESB’s analytical method is not able to distinguish between benzo[b]fluoranthene, benzo[j]fluoranthene, or benzo[k]fluoranthene and between chrysene and triphenylene but can only provide information on benzo[b,j,k]fluoranthene and chrysene/triphenylene as co-elutions. This should be taken into account when evaluating the data.\n",
      "\n",
      "The available ESB data are summarized in Table S3 (Supplementary material). The table also includes information on additional contaminants that might be of interest for future assessments.\n",
      "\n",
      "With respect to chemical analyses, the ESB meets all relevant requirements regarding sensitivity, accuracy, repeatability, and reproducibility (Table S2, Supplementary material). Requirements regarding recovery rates are not necessarily fulfilled for the analysis of organic contaminants. Nevertheless, the reliability of data sets concerning the pertained organic contaminants is checked and ensured within each batch of samples by analyzing in-house QA matrix samples, sample material of previous interlaboratory proficiency studies, or certified reference material.\n",
      "\n",
      "For PAHs, the analytical method used within the ESB-program might lead to an overestimation regarding benzo[b]fluoranthene and chrysene because additional substances are determined (i.e., benzo[j]fluoranthene, benzo[k]fluoranthene, and triphenylene).\n",
      "\n",
      "\n",
      "D9 assessment based on ESB data\n",
      "\n",
      "Annual pool samples of blue mussels and eelpout are routinely analyzed for the D9 relevant contaminants. Fairly long time series dating back to the 1990s are available for heavy metals, ∑4 PAHs and B[a]P in mussels and for Pb, Hg, and ndl-PCBs in eelpout (Table S3, Supplementary material). PCDD/Fs + dl-PCBs are currently only determined in eelpout, and time series are relatively short comprising every second year between 2003 and 2015 as well as 2016 and 2017.\n",
      "\n",
      "With respect to contaminants potentially subject to D9 assessment, TBT, and the WFD priority substances PFOS, PBDE, and HBCDD are included in the present study. For TBT in eelpout and mussels, the available time series end in 2013 (Radermacher 2015). PFOS, PBDE, and HBCDD data have so far been analyzed in archived eelpout samples from every second year between 2003 and 2015 as well as in samples from 2016 and 2017 (Table S3, Supplementary material).\n",
      "\n",
      "Tables 2, 3, and 4 summarize the contaminant levels in blue mussels and eelpout fillet.\n",
      "\n",
      "Table 2\n",
      "\n",
      "Table 3\n",
      "\n",
      "Table 4\n",
      "\n",
      "Concentrations in the last year of sampling are given as calculated values (derived from the trend line) and as measured values. In some cases, these values differ significantly (e.g., for TBT by factors of 3.6 and 4.8 in mussels from NS 1 and NS 2, respectively, and for B[a]P by a factor of 3.2 in mussels from BS) indicating that the trend function may not adequately reflect the last year. In these cases, assessing compliance for the last sampling year should consider also the measured concentrations. Note that for ∑4 PAHs in blue mussels from NS 2, TBT in mussels from BS, and HBCDD in eelpout from NS 1 and NS 2, the calculated concentrations are negative and are treated as “not detected.”\n",
      "\n",
      "\n",
      "North Sea/sampling sites NS 1 (lower Saxony Wadden Sea) and NS 2 (Schleswig-Holstein Wadden Sea)—FAO/ICES division 27.4.b\n",
      "\n",
      "\n",
      "D9 compliance assessment\n",
      "\n",
      "Blue mussels from both ESB North Sea sites had Pb, Cd, Hg, ∑4 PAH, and B[a]P levels well below the maximum allowed concentrations in fishery products laid down in Reg. (EC) No. 1881/2006 (amended by Reg. (EC) No. 629/2008, Reg. (EU) No. 1259/2011, Reg. (EU) No. 420/2011, and Reg. (EU) No. 835/2011; EC 2006a, 2008c, 2011a, b, c) (Tables 2, and 3, Figs. 2, and 3). This was still true when the dilution effect caused by breathing water was considered (i.e., multiplying the wet weight concentrations by 3 at NS 1, and 2.4 at NS 2) with the notable exception of ∑4 PAHs in the early assessment years (1985–1995 and 2000 at NS 1, and 1986 and 1990 at NS 2).\n",
      "\n",
      "Comparing both North Sea sites reveals that concentrations in mussels were always higher at sampling site NS 1 in the Lower Saxony Wadden Sea (Mann-Whitney U test, at least p = 0.02) (Tables 2 and 3, Fig. S1, Supplementary material).\n",
      "\n",
      "Comparative data are available only for NS 1: The Lower Saxony State Office for Consumer Protection and Food Safety report mean metal concentrations of 0.23 mg kg−1 Pb, 0.11 mg kg−1 Cd, and 0.033 mg kg−1 Hg in blue mussels form the Lower Saxony Wadden Sea in 2016 (LAVES 2017; BMUB 2018a).\n",
      "\n",
      "The Pb and Cd data fit well to the measured ESB mussel data from 2016 (i.e., 0.17 mg kg−1 Pb and 0.12 mg kg−1 Cd), whereas the Hg concentrations in ESB mussels from NS 1 were lower (i.e., 0.002 mg kg−1). When considering the dilution effect of breathing water, mussels at NS 1 had higher Pb and Cd burdens compared to the LAVES mussels (i.e., 0.52 mg kg−1 Pb and 0.37 mg kg−1 Cd), whereas Hg is still lower (0.006 mg kg−1).\n",
      "\n",
      "In eelpout fillets, levels of Pb, Hg, PCDD/Fs + dl-PCBs, and ndl-PCBs were below the respective maximum levels allowed in edible fish (EC 2006a, 2008c; EC 2011a) (Figs. 2 and 3).\n",
      "\n",
      "Since monitoring started in the mid-1990s, levels of Pb and Hg were similar in eelpout from both North Sea sites. Only in 1995 and 2001, higher Pb concentrations were observed at NS 1 (Tables 2 and 3, Figs. 2 and 3, and Fig. S2, Supplementary material).\n",
      "\n",
      "The eelpout data are in line with the initial assessment of the German North Sea according to Article 8 of the MSFD: Concentrations of PCDD/Fs + dl-PCBs in fillet of cod (Gadus morhua) were below 1 ng kg−1 ww in 2007 (i.e., 0.380 ng kg−1; Karl and Lahrssen-Wiederholt 2009; BMU 2012a) and thus met the threshold value for edible fish. Hg in North Sea fish was also below the maximum allowed concentration (BMU 2012a).\n",
      "\n",
      "\n",
      "Temporal trends\n",
      "\n",
      "Trend analysis revealed significant decreases (p < 0.01) in metals and PAH contamination of blue mussels from both North Sea sites (Figs. 2 and 3). Decreases were more pronounced in mussels from NS 1 indicating that pollution has declined at this site although it was still higher compared to NS 2 in the last year of sampling (Tables 2 and 3). For PAH, strong decreases were observed at both North Sea sites where PAH levels had been high in the late 1980s and early 1990s. PAH are ubiquitous pollutants in the marine environment originating, e.g., from atmospheric deposition, offshore activities, and operational or accidental spills from ships (Brockmeyer and Theobald 2016). Possibly, the stricter regulations concerning emission and dumping of oil and oily waters that came into force in 1983 (International Convention for the Prevention of Pollution from Ships, MARPOL (1973/1978) and its amendments) resulted in the pronounced decrease of PAHs.\n",
      "\n",
      "Fig. 2 Trends of D9 relevant contaminants in ESB samples of blue mussels (soft body) and eelpout (fillet) from the North Sea/Lower Saxony Wadden Sea (coastal region of FAO/ICES Division 27.4.b). The lines represent the results of the linear regression (solid for significant linear trend, dashed for not significant). ML maximum level allowed in fish and fishery products (Reg. (EC) No. 1881/2006 and amendments; EC 2006a, 2008c, 2011a, b, c)\n",
      "\n",
      "Fig. 3 Trends of D9 relevant contaminants in ESB samples of blue mussels (soft body) and eelpout (fillet) from the North Sea/Schleswig-Holstein Wadden Sea (coastal region of FAO/ICES Division 27.4.b). The lines represent the results of the linear regression (solid for significant linear trend, dashed for not significant). ML maximum level allowed in fish and fishery products (Reg. (EC) No. 1881/2006 and amendments; EC 2006a, 2008c, 2011a, b, c))\n",
      "\n",
      "Cd and Hg concentrations in mussels from NS 2 decreased slightly but steadily during the monitoring period. Concentrations between the first and last year differed by only − 23% for Cd and − 19% for Hg. Even though these decreases may not have a direct environmental impact, the steadiness of the decreases nevertheless indicates that pollution is declining in the area.\n",
      "\n",
      "For eelpout, significant decreases were detected for Hg and ndl-PCBs at NS 1 and for Pb and ndl-PCBs at NS 2 (at least p = 0.01). Pb has also decreased in eelpout from NS 1. This trend, however, was not significant.\n",
      "\n",
      "In contrast, PCDD/Fs + dl-PCBs increased slightly in eelpout from both North Sea sites (significant trend, p = 0.01, only at NS 2) (Figs. 2 and 3). So far, we have no conclusive explanation for these increases, all the more as no respective trends are observed in sea gull eggs from nearby sites (i.e., island Mellum at NS 1 and island Trischen at NS 2) sampled by the ESB between 2008 and 2015/2016 (data not published yet).\n",
      "\n",
      "\n",
      "Baltic Sea/sampling site BS (Bodden National Park of Western Pomerania)—FAO/ICES subdivision 27.3d.24\n",
      "\n",
      "\n",
      "D9 compliance assessment\n",
      "\n",
      "In mussel samples from the Baltic Sea concentrations of Pb, Cd, Hg, ∑4 PAHs, and B[a]P were well below the maximum levels allowed in fishery products as laid down in Reg. (EC) No. 1881/2006 (amended by Reg. (EC) No. 629/2008, Reg. (EU) No. 1259/2011, Reg. (EU) No. 420/2011, and Reg. (EU) No. 835/2011; EC 2006a, 2008c, 2011a, b, c) (Table 4; Fig. 4). This was, except for ∑4 PAHs in 1993, still true when considering the dilution effect caused by breathing water (i.e., multiplying the wet weight concentrations by 3).\n",
      "\n",
      "Likewise, compliance with the threshold values was observed for eelpout fillet: During the entire monitoring period, concentrations of the D9 relevant contaminants Pb, Hg, PCDD/Fs + dl-PCBs, and ndl-PCBs in eelpout fillet were below the respective maximum levels allowed in foodstuffs (EC 2006a, 2008b, 2011a) (Table 4, Fig. 4).\n",
      "\n",
      "These findings correspond to the initial assessment of the German Baltic Sea according to Article 8 of the MSFD: ML compliance was reported for PCDD/F + dl-PCB in fillet of cod (Gadus morhua) and herring (Clupea harengus) from the Baltic Sea (i.e., 0.228–0.631 pg g−1 WHO-TEQ in cod and 2.15–6.28 pg g−1 in herring in 2006; BMU 2012b; Karl et al. 2010) and also for ndl-PCB in herring fillets (14.7–30.4 ng g−1; Karl et al. 2010, ndl-PCB levels in cod were not determined). A separate study analyzed Hg in Baltic Sea fish and found no exceedance of the ML (BMU 2012b).\n",
      "\n",
      "Similarly, the State Office for Agriculture, Food Safety and Fisheries of Mecklenburg-Western Pomerania reports Pb, Cd, Hg, and PCDD/F + dl-PCB levels in herring fillets from the Western Baltic Sea (ICES Boxes 22 and 24) sampled between 2012 to 2016 that met the respective MLs (BMUB 2018b).\n",
      "\n",
      "\n",
      "Temporal trends\n",
      "\n",
      "Since monitoring started, concentrations of metals and ∑4 PAHs decreased steadily in blue mussels from the ESB site in the Baltic Sea (p < 0.01). Again, the stricter regulation for ships (e.g., ban of rinsing of tanks and dumping the oily water into the sea, MARPOL (1973/1978) is probably responsible for the decreases. For eelpout, significant decreasing trends (p < 0.01) were detected for Pb, PCDD/Fs + dl-PCBs, and ndl-PCBs, whereas Hg remained more or less constant (Fig. 4).\n",
      "\n",
      "Fig. 4 Trends of D9 relevant contaminants in ESB samples of blue mussels (soft body) and eelpout (fillet) from the Baltic Sea/Bodden National Park of Western Pomerania (coastal region of FAO/ICES Subdivision 27.3d.24). The lines represent the results of the linear regression (solid for significant linear trend, dashed for not significant). ML maximum level allowed in fish and fishery products (Reg. (EC) No. 1881/2006 and amendments; EC 2006a, 2008c, 2011a, b, c)\n",
      "\n",
      "\n",
      "Evaluation of additional contaminants\n",
      "\n",
      "Member States may choose to monitor additional contaminants not listed in Reg. (EC) No. 1881/2006 (EC 2006a, 2017a). Potentially problematic contaminants are, e.g., those listed as priority substances under the WFD (EC 2000) and assessed under Descriptor 8-C1 of the MSFD (EC 2017a). For 11 WFD priority substances, wet weight-based EQSs were derived for biota of which nine refer to fish (EC 2013b).\n",
      "\n",
      "The biota-EQSs represent substance concentrations that are expected to be safe not only for the organisms themselves but also for human consumers (protection goal “human health”) and piscivorous predators (protection goal “secondary poisoning”; EC 2014) because the most sensitive protection goal is decisive for the EQS.\n",
      "\n",
      "One possible trigger for inclusion in D9 assessment can be the exceedance of the biota-EQS derived for the protection of human health. An inclusion should also be considered if for any of these contaminants increasing trends are detected (Swartenbroux et al. 2010; Zampoukas et al. 2014).\n",
      "\n",
      "In the present study, four priority substances are exemplarily analyzed that might be of relevance as additional contaminants included in D9 assessment, i.e., PBDE, HBCDD, PFOS, and TBT compounds.\n",
      "\n",
      "The brominated flame retardants PBDE and HBCDD were widely used, e.g., in electronic products, building material, textiles, and upholstery. Technical Penta-BDE and Octa-BDE mixtures were phased out in the 1990s and are banned in the EU since 2003 (EC 2003a). The Biota-EQS of 0.0085 μg kg−1 fish refers to the sum of the congeners BDE-28, -47, -99, -100, -153, and -154. HBCDD was banned in 2013 (UNEP 2013) but exemptions allowed usage for building insulation materials until 2017. In the years before, it was increasingly used as substitute for PBDE. The biota-EQS for HBCDD is 167 μg kg−1 fish and refers to the sum of α-, β-, and γ- HBCDD.\n",
      "\n",
      "The fluorosurfactant PFOS was used, e.g., as fabric protector and impregnation agent. Since 2008, PFOS is restricted EU-wide to only a few applications (EC 2006b). Its biota-EQS is 9.1 μg kg−1 fish.\n",
      "\n",
      "TBT has long been used in antifouling paints on ships and boats. High environmental concentrations were typically associated with marinas and shipyards. In 2003, TBT was finally banned EU-wide (EC 2003b). No biota-EQS exists for TBT, but OSPAR and HELCOM have set an environmental assessment criterion (EAC) of 12 μg kg−1 dry weight for TBT in bivalves that relates to toxic effects on bivalves and the protection goal secondary poisoning (OSPAR 2004). Beyer et al. (2017) converted the original OSPAR EAC value to a wet weight-based EAC of 2 μg kg−1 ww for formulation-based TBT (using an average dry mass content for mussels of 17.38%). Blue mussels from the ESB sites had lower average dry mass contents, and accordingly, the respective wet weight-based EACs are lower (i.e., 1.13 μg kg−1 ww and 0.88 μg kg−1 ww for mussels from NS 1 and NS 2, respectively, and 0.63 μg kg−1 ww for mussels from BS; Tables 2, 3, and 4). Norway has derived a biota quality standard of 150 μg kg−1 ww for TBT compounds (NEA 2016).\n",
      "\n",
      "Based on retrospectively analyzed ESB samples of eelpout and blue mussel, temporal trends for TBT, PFOS, PBDE, and HBCDD have been calculated (Table S4, Fig. S4, S5, Supplementary material). Figure 5 summarizes the data from all three sampling sites.\n",
      "\n",
      "Fig. 5 Concentrations of TBT (μg kg−1 dry weight, as cation) in blue mussels (soft body) (a) and PFOS, PBDE, and HBCDD (μg kg−1 wet weight) in eelpout (fillet) (b–d) from the ESB sampling sites in the North and Baltic Seas. BS Baltic Sea/Bodden National Park of Western Pomerania, NS 1 North Sea/Lower Saxony Wadden Sea, NS 2 Schleswig-Holstein Wadden Sea, EAC OSPAR environmental assessment criterion, EQS environmental quality standard\n",
      "\n",
      "The retrospective analysis revealed high TBT concentrations in blue mussels from the Lower Saxony Wadden Sea (NS 1) in the 1980s and 1990s that exceeded the EAC value by far (factor up to 16 based on dry weight concentrations) (Fig. 5a). Since the EU-wide ban in 2003, concentrations have declined significantly (p < 0.01, Table S4, Fig. S4, Supplementary material) and complied with the EAC of 12 μg kg−1 dw in 2010–2013. For NS 2 and BS TBT, data are available since 2004 and 2005, respectively. Lowest contamination was detected at NS 2 where mussels are not directly exposed to emissions from marinas or shipping traffic. Similar to NS 1 TBT, levels have decreased significantly in mussels from BS and NS 2 (p < 0.01, Table S4, Fig. S4 and S5, Supplementary material) and met the EAC in 2008 (NS 2) and 2011 (BS). Time series for TBT in mussels end in 2013 because new increases are not expected (for more details see Rüdel et al. 2003, 2009 and Radermacher 2015).\n",
      "\n",
      "TBT levels in eelpout fillets were mostly higher than in mussels (Tables 2, 3, and 4, Fig. S3, Supplementary material). This is at least partly due to different exposure conditions as sampling sites for mussels and eelpout are not identical.\n",
      "\n",
      "PFOS, PBDE, and HBCDD were analyzed only in eelpout because their EQSs refer to fish.\n",
      "\n",
      "PFOS concentrations in eelpout fillet were similar at all three ESB sites and well below the respective EQS of 9.1 μg kg−1 (Fig. 5b). Significant changes since 2003 were only detected in fish from NS 2 (decreasing trend, p = 0.01; Table S4, Fig. S4, Supplementary material). For PFOS, the critical protection goal behind the EQS is human health. Accordingly, the results indicate that PFOS in marine fish from the three coastal sampling sites pose no risk to human consumers.\n",
      "\n",
      "The picture is different for PBDE with 100% EQS exceedance at all three sites and in all years (Fig. 5c). Despite significant decreases (p < 0.01, Table S4, Fig. S4 and S5, Supplementary material), PBDE concentrations in eelpout fillets from all three sites still exceeded the EQS in 2016 and 2017 by factors of 4–10. The critical protection goal behind the EQS is human health implying that, despite decreasing trends, PBDE in marine fish from the three coastal sites can still pose a risk for human consumers.\n",
      "\n",
      "HBCDD concentrations in eelpout fillet were always way below the EQS of 167 μg kg−1 (Fig. 5d). In most years, concentrations ranged between 0.11 and 1.5 μg kg−1 with higher levels detected only in 3 years (where sample handling or measurement errors cannot be excluded). Significant decreases were detected at NS 1 (p = 0.03) and BS (p < 0.01; Table S4, Fig. S4 and S5, Supplementary material). The EQS for HBCDD is based on the protection goal secondary poisoning. The quality standard derived for human health is about 36 times higher (i.e., 6100 μg kg−1, EC 2014). According to these data, HBCDD in marine fish from all three ESB sites pose no risk for human consumers.\n",
      "\n",
      "\n",
      "Summary and conclusions\n",
      "\n",
      "Evaluating the suitability of the ESB samples for D9 assessment gives a mixed picture:\n",
      "\n",
      "The requirements of the MSFD regarding the geographical scale of D9 monitoring are only partly fulfilled: Of the catch or production area according to Art. 38 of Reg. (EU) No. 1379/2013 (EC 2013b), only a section of the German coastal regions of the Baltic Sea and the North Sea are covered. The EEZ of the Greater North Sea and the Baltic Sea are not covered at all. This is a clear limitation with respect to D9 monitoring.\n",
      "\n",
      "Even though the comparison of the ESB data with fish data from the open seas (Karl and Lahrssen-Wiederholt 2009; Karl et al. 2010) suggests that contamination is comparable, there are many factors that can lead to differences. Coastal areas are mainly subject to pollution originating from land, like inputs from rivers, run-off from agricultural sites, and wastewaters from industrial and municipal wastewater treatment plants, while ship traffic, dumping, off-shore activities, and atmospheric deposition are the major sources of pollution in the open sea (Davis 1993). Differences pollution-wise between coastal regions and the open sea can be further enhanced by processes like dilution, degradation, and transformation of contaminants.\n",
      "\n",
      "The marine specimens archived by the ESB, i.e., blue mussel soft bodies and eelpout fillets, are considered suitable for D9 monitoring although with limitations in the case of eelpout as it is not one of the most consumed fish in Germany. It has to be emphasized, however, that two species can only give an indication of the contamination of fishery products and may be representative for some of the most consumed species but by far not for all. It is well-known that fish contamination depends on biological factors like fish age, size, gender, migration behavior, trophic position, and lipid levels (e.g., Brázová et al. 2012; Burger et al. 2001; Dušek et al. 2005; Gewurtz et al. 2011; McIntyre and Beauchamp 2007; Pulkrabová et al. 2007).\n",
      "\n",
      "Sampling and processing of the ESB meet the requirements of the MSFD with the limitation that the personnel involved in sampling is not authorized according to Reg. (EC) No. 333/2007 and Reg. (EC) No. 644/2017 and no controls according to Reg. (EU) No 882/2004 are performed during sampling and processing.\n",
      "\n",
      "The requirements of the MSFD regarding analytical methods are mainly fulfilled. An exemption is the PAH analyses where the ESB-analytical method does not distinguish between benzo[b]fluoranthene, benzo[j]fluoranthene, and benzo[k]fluoranthene and between chrysene and triphenylene.\n",
      "\n",
      "Taken together, it can be concluded that the ESB samples of blue mussels and eelpout fillet are basically suitable for D9 assessment but should be complemented by additional georeferenced samples of other species also from the EEZ.\n",
      "\n",
      "As long as such samples are not available, an indicative assessment of D9 could be based on ESB samples.\n",
      "\n",
      "The big advantage of the ESB samples is their clear assignment to a specific marine region. The samples can be used for D8-C1 and D9 assessment which can help to save costs. Furthermore, the archived samples of the ESB offer the opportunity of retrospective analysis to assess time series also of contaminants that may be included in D9 assessment in the future.\n",
      "\n",
      "The limitations regarding official controls and authorized personnel should be noted, but it should also be considered that the ESB works according to strict standard operating procedures and the whole process from sampling to chemical analyses follows the EN ISO/IEC 17025 standard.\n",
      "\n",
      "The deviation between the MSFD requirements and the ESB data regarding ∑4 PAHs are not viewed as limitation but rather as a worst-case scenario because additional substances are included (i.e., if the ESB data comply with the ML for ∑4 PAHs, this will also be true when considering only the four required PAHs). Nevertheless, in the future, a more selective method for PAHs could be applied to analyze benzo[b]fluoranthene and chrysene.\n",
      "\n",
      "The exemplary D9 assessment at the ESB sites showed that 100% of the blue mussel and eelpout samples from the North Sea (coastal areas of FAO/ICES Division 27.4.b) and the Baltic Sea (coastal area of FAO/ICES Subdivision 27.3d.24) sampled since the 1990s met the threshold values for Pb, Cd, Hg, ∑4 PAHs, PCDD/Fs + dl-PCBs, and ndl-PCBs in foodstuffs listed in Regulation (EC) No. 1881/2006 and its amendments (EC 2006a, 2008c, 2011a, b, c). When considering the dilution caused by breathing water in blue mussels, the thresholds were met at the latest since 2001. The additionally assessed contaminants PFOS and HBCDD were below the respective WFD biota-EQS in eelpout fillets and are considered as safe for human consumers. In contrast, PBDE levels in eelpout fillet exceeded the respective EQS indicating that a risk to human consumers cannot be excluded. TBT analysis ended in 2013 after concentrations in blue mussels had met the EAC for at least three consecutive years.\n",
      "\n",
      "\n",
      "Electronic supplementary material\n",
      "\n",
      "\n",
      "ESM 1\n",
      "\n",
      "(DOCX 566 kb)\n"
     ]
    }
   ],
   "source": [
    "view_one_article(length_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "quantiles = subset_df.quantile(q=[0.01, 0.25, 0.95, 0.99])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>length</th>\n",
       "      <th>n_geospans</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0.01</th>\n",
       "      <td>10097.2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.25</th>\n",
       "      <td>24279.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.95</th>\n",
       "      <td>56676.0</td>\n",
       "      <td>128.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0.99</th>\n",
       "      <td>77687.0</td>\n",
       "      <td>234.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       length  n_geospans\n",
       "0.01  10097.2         1.0\n",
       "0.25  24279.0        14.0\n",
       "0.95  56676.0       128.0\n",
       "0.99  77687.0       234.2"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "quantiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "simple_query = {\n",
    "    \"text_matches\": {\"$in\": terms},\n",
    "    \"article_meta.article_type\": \"research-article\",\n",
    "    \"article_meta.has_body\": True,\n",
    "    \"article_meta.n_geospans\": {\"$exists\": True},\n",
    "    \"article_meta.geospan_density\": {\"$exists\": True},\n",
    "}\n",
    "\n",
    "cursor = articles.aggregate([\n",
    "    {\"$match\": simple_query},\n",
    "    {\"$project\": {\n",
    "        \"n_geospans\": \"$article_meta.n_geospans\",\n",
    "        \"geospan_density\": \"$article_meta.geospan_density\",\n",
    "        \"length\": {\"$strLenCP\": \"$extracted_text\"}\n",
    "    }}\n",
    "])\n",
    "\n",
    "subset_df = pd.DataFrame(cursor)\n",
    "quantiles = subset_df.quantile(q=[0.01, 0.25, 0.95, 0.99])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.00013150962697391707"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "density = subset_df[\"geospan_density\"]\n",
    "density = density[density!=0]\n",
    "log_density = np.log(density)\n",
    "threshold_density = np.exp(np.mean(log_density) - np.std(log_density))\n",
    "\n",
    "threshold_density"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    0.000193\n",
       "1    0.000772\n",
       "2    0.000238\n",
       "5    0.001409\n",
       "6    0.000143\n",
       "7    0.000825\n",
       "8    0.000601\n",
       "Name: geospan_density, dtype: float64"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "density[density>threshold_density]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/toph/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/ipykernel_launcher.py:1: RuntimeWarning: divide by zero encountered in log\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "-inf"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.log(density)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'log' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-141-6e529ffcb90a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mlog_density\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlog\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdensity\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'log' is not defined"
     ]
    }
   ],
   "source": [
    "log_density = log(density)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_query = {\n",
    "    \"text_matches\": {\"$in\": terms},\n",
    "    \"article_meta.article_type\": \"research-article\",\n",
    "    \"article_meta.has_body\": True,\n",
    "    \"article_meta.n_geospans\": {\"$gt\": quantiles.n_geospans.iloc[1],\n",
    "                                \"$lt\": quantiles.n_geospans.iloc[3]},\n",
    "    \"$expr\": {\"$gte\": [{\"$strLenCP\": \"$extracted_text\"}, quantiles.length.iloc[0]]},\n",
    "    \"$expr\": {\"$lte\": [{\"$strLenCP\": \"$extracted_text\"}, quantiles.length.iloc[2]]}\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_query = {\n",
    "    \"text_matches\": {\"$in\": terms},\n",
    "    \"article_meta.article_type\": \"research-article\",\n",
    "    \"article_meta.has_body\": True,\n",
    "    \"article_meta.n_geospans\": {\"$gt\": quantiles.n_geospans.iloc[1],\n",
    "                                \"$lte\": quantiles.n_geospans.iloc[3]},\n",
    "    \"$and\": [\n",
    "        {\"$expr\": {\"$gt\": [{\"$strLenCP\": \"$extracted_text\"}, quantiles.length.iloc[0]]}},\n",
    "        {\"$expr\": {\"$lte\": [{\"$strLenCP\": \"$extracted_text\"}, quantiles.length.iloc[2]]}}\n",
    "    ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'text_matches': {'$in': ['field work',\n",
       "   'fieldwork',\n",
       "   'field study',\n",
       "   'field site',\n",
       "   'field area',\n",
       "   'study site',\n",
       "   'study location',\n",
       "   'study area',\n",
       "   'research site',\n",
       "   'research location',\n",
       "   'sampling site',\n",
       "   'sampling location',\n",
       "   'sampling area']},\n",
       " 'article_meta.article_type': 'research-article',\n",
       " 'article_meta.has_body': True,\n",
       " 'article_meta.n_geospans': {'$gt': 14.0, '$lte': 234.20000000000027},\n",
       " '$and': [{'$expr': {'$gt': [{'$strLenCP': '$extracted_text'}, 10097.2]}},\n",
       "  {'$expr': {'$lte': [{'$strLenCP': '$extracted_text'}, 56676.0]}}]}"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "4829693\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "Groundwater–surface water mixing shifts ecological assembly processes and stimulates organic carbon turnover\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "{'all_geospans': 29, 'n_geospans': 29, 'article_type': 'research-article', 'has_body': True}\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['sampling location']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Understanding the processes that underlie places and times of elevated biogeochemical activity is needed to construct predictive models of elemental cycling and contaminant fate and transport under future environmental scenarios. Without the knowledge of governing processes, predictions are unlikely to be accurate, owing to emergent properties that arise through the behaviour of complex systems1. Environmental transition zones often exhibit elevated rates of biogeochemical activity due to the mixing of complementary electron donors and terminal electron acceptors234. The hyporheic zone is a classic transition zone567 important to the functioning of lotic systems891011121314, where biogeochemical activity can be locally elevated during times of groundwater–surface water mixing151617.\n",
      "\n",
      "Shifts in the hyporheic zone microbial communities are a key, but understudied, component of hydrologic mixing-induced increases in biogeochemical reaction rates. Hyporheic zone microbial ecology has primarily been studied indirectly by characterizing spatiotemporal patterns in electron donors, terminal electron acceptors and microbial biomass or activity4181920. Although connections have been made between the community composition and metabolic function of hyporheic zone microbial communities21, there are few studies that directly investigate hyporheic zone microbial composition822. This represents a significant knowledge gap, especially considering that up to 96% of lotic ecosystem respiration occurs within the hyporheic zone12 and up to 95% of hyporheic zone respiration is microbial23. Although few in number, studies that directly evaluate hyporheic zone microbial communities have made important progress, revealing significant influences of sediment porosity21, temporal variation in physicochemical conditions242526, heavy metal contamination272829, groundwater–surface water exchange30 and dissolved organic carbon composition31.\n",
      "\n",
      "Temporal fluctuations in river stage cause variation in the degree of groundwater–surface water mixing3233, which in turn changes the resource environment experienced by hyporheic zone microbial communities. River stage fluctuations are therefore likely to have an indirect impact on hyporheic zone microbial community composition, mediated through changes in electron donor and/or terminal electron acceptor profiles. Although outside the spatial domain traditionally considered the hyporheic zone, Lin et al.34 provide evidence consistent with this hypothesis, revealing significant changes in community composition associated with the intrusion of surface water into a subsurface aquifer. Stegen et al.35 further showed that these changes in composition were due to intrusion-driven shifts in ecological selection pressures. These studies did not sample surface water or evaluate changes in the resource environment, making it difficult to understand linkages among groundwater–surface water mixing, biogeochemical conditions and community composition. Iribar et al.36 also sampled across groundwater–surface water mixing conditions in the hyporheic zone, but did not relate changes in mixing to changes in microbial community composition. Sliva and Williams37 used a field manipulation to study the connection between groundwater–surface water exchange and microbial composition, but found equivocal patterns due to methodological limitations.\n",
      "\n",
      "Here we evaluate the hypothesis that changes in groundwater–surface water mixing drive a coupled response in hyporheic zone biogeochemistry and microbial community composition due to shifting ecological selection pressures tied to the resource environment. To do so, we take advantage of short-term fluctuations in river stage that cause broad changes in the degree of groundwater–surface water mixing—and thus resource environment—within the Columbia River hyporheic zone and link the resulting hyporheic zone dynamics to system end members (surface water and groundwater). Although longer-term stage fluctuations are clearly important, we focus on short-term fluctuations to isolate the impacts of shifts in groundwater–surface water mixing; longer-term (for example, seasonal) sampling may confound changes in mixing with changes in other environmental variables (for example, temperature). Our analyses also leverage and combine next-generation DNA sequencing, ultra-high-resolution organic carbon profiling and ecological theory to track microbial community dynamics and reveal underlying drivers of change. We show that groundwater–surface water mixing shifts the coupled microbiome-biogeochemical system from a mixing-driven system characterized by stochastic ecological processes to a system with elevated biogeochemical rates and deterministic ecological processes.\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "\n",
      "Hydrologic mixing and stimulated respiration\n",
      "\n",
      "A sampling scheme was employed adjacent to and within the Columbia River (Fig. 1), to capture a range of river-stage conditions across ∼1 week. Although large, long-term fluctuations in river stage (Fig. 2a) are principally driven by snowpack melt and runoff across the Columbia River watershed, short-term fluctuations (Fig. 2b) are caused by up-river flow regulation at the Priest Rapids dam. Sampling across different river-stage conditions and across different spatial locations (vertically and horizontally distributed; Fig. 1) translated into a range of groundwater–surface water mixing conditions in the hyporheic zone as shown by substantial variation in Cl− concentration within hyporheic zone samples (Fig. 3); Cl− concentration was used as a conservative tracer. The three non-conservative (that is, reactive) analytes showed remarkable differences between groundwater and surface water, which allowed for evaluations of mixing-model deviations (Fig. 3); reactive analytes included non-purgable organic carbon (NPOC), dissolved inorganic carbon (DIC) and nitrate. Specific conductance—an alternative tracer of groundwater–surface water mixing conditions38—was logged in river water and groundwater wells. The relatively high and stable specific conductance values in groundwater wells—combined with low and stable values in the river—during the sampling period (Supplementary Fig. 1) provide evidence that water sampled from groundwater wells and from the river represented distinct end members.\n",
      "\n",
      "Consistent with previous studies420, we observed patterns in aqueous biogeochemistry that suggested elevated microbial metabolism in response to surface water intrusion into the hyporheic zone. Two groundwater–surface water mixing models were generated for each reactive analyte, revealing a region of concentrations consistent with mixing; NPOC (Fig. 3a) and DIC (Fig. 3b) concentrations commonly fell outside this space, whereas nitrate (Fig. 3c) concentrations did not. Positive DIC deviations were, however, only found below a Cl− concentration of ∼7 mg l−1. NPOC concentrations appeared to be a single continuous function of Cl− concentrations (Fig. 3a), which suggests relatively consistent quality of labile carbon entering the hyporheic zone across the week of sampling; a significant decline in quality would have shifted NPOC concentrations to values more consistent with the mixing model expectation.\n",
      "\n",
      "Taken alone, the loss of NPOC—relative to the mixing expectation—could be interpreted as a signal of either enhanced microbial metabolism of organic carbon or sorption onto hyporheic zone sediments. Enhanced microbial heterotrophic respiration could increase DIC by increasing the rate of CO2 production, whereby positive deviations would be expected from the DIC mixing model. Furthermore, assuming that the magnitude of the NPOC deviation indicates the degree to which respiration is enhanced, increasingly negative NPOC deviations would be associated with increasingly positive DIC deviations. This is indeed the pattern observed (Fig. 3d) and, in turn, our analyses are consistent with the hypothesis that the intrusion of surface water transports labile organic carbon into the hyporheic zone that, in turn, stimulates microbial respiration2039.\n",
      "\n",
      "It is important to note that there is a larger DIC increase than expected from the magnitude of NPOC loss (that is, the stoichiometry is not 1:1) (Fig. 3d). Two factors that may contribute to the excess DIC are calcite dissolution and organic carbon ‘priming'; a recent simulation study40 of our field system revealed that intrusion of river water into the hyporheic zone can lead to calcite dissolution and, in turn, elevated DIC; priming41 could occur if dissolved organic carbon transported to the hyporheic zone stimulated decomposition of sediment-bound organic carbon, thereby elevating DIC. Future work that leverages isotopic signatures of abiotic and biotic sources of DIC should enable evaluation of these hypotheses.\n",
      "\n",
      "Our results indicate that the transported labile carbon is dissolved—as opposed to particulate—given that particulate organic carbon (POC) was highest in the river and did not vary significantly between the hyporheic zone and groundwater (Supplementary Fig. 2). These patterns are consistent with previous work showing that POC is retained in surface sediments via filtration, thereby limiting its transport into the hyporheic zone4243. We did observe greater variation in hyporheic zone POC concentration—relative to groundwater (Supplementary Fig. 2)—and the cell density of planktonic organisms was found to increase with hyporheic zone POC (Supplementary Fig. 3); however, inclusion of POC from all water sources resulted in this relationship becoming nonsignificant (P>0.05). Although the POC-cell density relationship in the hyporheic zone was relatively weak (R2=0.32), it suggests that there may be times and/or places within the hyporheic zone where microbial growth due to elevated POC penetration resulted in higher biomass.\n",
      "\n",
      "Residence time of carbon-rich water in the hyporheic zone influences metabolic pathways used by hyporheic zone microbial communities4445; longer residence times can be associated with elevated nitrate reduction45 due to localized depletion of O2 (that is, anaerobic microsites)464748. Although we do not have estimates of surface water residence time, it was probably short due to the rapid rise and fall in river elevation across the sampling timeframe (Fig. 2b). This is supported by previous investigations in the study system showing that—during river-stage dynamics similar to those sampled here—the direction (and magnitude) of water flux changes on a 12-h time scale due to the 12-h cycle of high/low river stage38. Combining estimated flux rates from Fritz and Arntzen38—maximum influx of river water was 0.4 l m−2 min−1—with porosity of the hyporheic zone alluvium (0.18; ref. 49) provides an estimated maximum water velocity of ∼1.5 m per 12 h, with the vast majority of velocities across a 12-h period much slower than this38. The regular reversal of the hydrologic gradient—and low river elevation minima during the sampling period (Fig. 2)—suggests that the residence time of river water in the hyporheic zone was on the order of 1 day. The consistency between hyporheic zone  concentrations and mixing model expectations are therefore not surprising. It is likely that denitrification was not stimulated by surface water intrusion, because modest residence times, high O2 concentrations in intruding surface water and low NPOC concentrations collectively resulted in severe reaction-rate limitation50. In addition, significant O2 can be maintained in the hyporheic zone—and therefore inhibit denitrification2—under organic carbon concentrations similar to those observed here, even when residence times are two orders of magnitude longer than estimated here51. We further note that these conditions (high O2 and low NPOC) may be more conductive to nitrification than denitrification252, which could lead to a positive deviation from the  mixing model, but such a pattern was not apparent (Fig. 3).\n",
      "\n",
      "The significant loss of NPOC during times of surface water intrusion suggests that changes in the organic carbon profile may drive shifts in microbial community composition31. If there is such an effect, it could be mediated by the quantity and/or composition of organic carbon. To distinguish between these possibilities, we complemented the NPOC quantification with ultra-high-resolution carbon profiling, using Fourier transform ion cyclotron resonance mass spectrometry (FT-ICR MS).\n",
      "\n",
      "\n",
      "Shifts in organic carbon and microbial community composition\n",
      "\n",
      "Considering all water sources together, the composition of organic carbon—as characterized by FT-ICR MS—varied significantly (P=0.001) with changes in NPOC concentration (Fig. 4a) and this relationship was stronger (r=0.66) than one based on changes in Cl− concentration (r=0.55). In addition, the median mass-to-charge ratio within a given sample varied significantly across water sources (P<0.05), with larger values in groundwater (Fig. 4b). Low-molecular-weight organic carbon in the hyporheic zone, relative to groundwater, is consistent with increased microbial activity53. This further supports our inference that labile organic carbon—transported by intruding river water—stimulates microbial respiration in the hyporheic zone.\n",
      "\n",
      "Although organic carbon composition varied significantly with changes in NPOC concentration and water source (Fig. 4), it was not strongly related to microbial community composition (see below). However, we consider this evaluation preliminary, because our use of the FT-ICR MS data was limited to the presence/absence of unique m/z peaks. This approach does not capture important shifts in the relative abundance of labile carbon species, which may obscure the relationship between carbon composition and microbial community composition. We consider this probable given that previous work has shown a strong influence of carbon composition over hyporheic zone community composition31.\n",
      "\n",
      "Preliminary data exploration indicated that changes in community composition were most strongly related to shifts in Cl− and NPOC concentrations, with weaker relationships observed for changes in FT-ICR MS-based carbon composition and changes in temperature. In addition, the number of unique operational taxonomic units (OTUs) increased significantly with Cl− concentration and declined significantly with NPOC concentration (Supplementary Fig. 4); these relationships were effectively equivalent in terms of variation explained (R2≈0.45 and P<<0.0001 for both). In turn, we focus on microbial composition as related to Cl− and NPOC concentrations.\n",
      "\n",
      "Relating three metrics of community turnover—Bray–Curtis, β-mean nearest taxon distance (βMNTD) and β-nearest taxon index (βNTI)—to shifts in Cl− and NPOC concentrations revealed important influences of both groundwater–surface water mixing and changes in the resource environment. The Bray–Curtis and βMNTD metrics respectively quantify turnover in the relative abundances of OTUs and in community phylogenetic structure. These are ‘raw' metrics and do not, by themselves, carry information about the ecological processes that govern community turnover545556. That is, variation in the magnitudes of these two metrics can be driven by stochastic ecological processes such as organismal dispersal and more deterministic ecological processes associated with environmental or habitat selection.\n",
      "\n",
      "Bray–Curtis was nonlinearly related to changes in Cl− concentration (Fig. 5a), but a weak relationship was observed with respect to changes in NPOC concentration (Fig. 5b). βMNTD was significantly (P=0.001 for both) related to changes in both Cl− and NPOC concentrations (Fig. 5c,d); the relationship with Cl− (r=0.70) was stronger than with NPOC (r=0.52). Although a change in NPOC concentration probably reflects alteration of the resource environment, a change in Cl− concentration arises due to the transport of surface water into (or out of) the subsurface domain. The Bray–Curtis and βMNTD patterns therefore suggest that the transport of surface water may directly influence OTU relative abundances and community phylogenetic structure. One parsimonious interpretation is that the transport of surface water into the subsurface enhances organismal dispersal across system compartments (that is, from surface water to the hyporheic zone to the broader aquifer). Importantly, this indicates that organismal dispersal can influence community composition, but it does not rule out a strong influence of the resource environment. An alternative interpretation is that uranium contamination—which closely tracks groundwater–surface water mixing conditions in the study system38—may influence community composition. Konopka et al.57 showed, however, that microbial activity in the study system is not altered by prevailing uranium concentrations due to formation of carbonate uranyl complexes.\n",
      "\n",
      "\n",
      "Ecological processes\n",
      "\n",
      "The βNTI metric provides deeper insights into the potential role of the resource environment, because variation in the magnitude of βNTI is driven primarily by variation in the selective environment and, to a much lesser extent, by organismal dispersal56. The reason is that βNTI is based on a null model that factors out influences of stochastic ecological processes such as organismal dispersal and ecological drift. In turn, if the resource environment does have an impact on the structure of hyporheic zone microbial communities, we expect three patterns: (i) a significant relationship between βNTI and changes in NPOC; (ii) this relationship should be stronger than the relationship between βNTI and changes in Cl−; and (iii) with small changes in NPOC, βNTI values should be below −2 and βNTI should increase to above +2 with increasingly large NPOC changes. The reason for (iii) is that small changes in NPOC indicate there is a consistent resource environment and βNTI<−2 indicates that a consistent selective environment causes high similarity in community composition. Likewise, larger changes in NPOC indicate there is a large change in the resource environment and βNTI>+2 indicates that a change in the selective environment causes a significant shift in community composition5658.\n",
      "\n",
      "As shown in Fig. 5e,f, βNTI was significantly (P=0.001 for both) related to changes in both Cl− and NPOC concentrations (Fig. 5e,f); the relationship with NPOC (r=0.53) was stronger than with Cl− (r=0.21) and transitioned from βNTI<−2 (with small changes in NPOC) to βNTI >+2 (with large changes in NPOC) (Fig. 5f). All three expectations were therefore met, which provides evidence that changes in the resource environment deterministically cause shifts in microbial community composition. It is important to recognize that these resource-based influences worked alongside influences of organismal dispersal, and that neither process dominated the system.\n",
      "\n",
      "Coupling NPOC deviations with βNTI further revealed that homogeneous ecological selection565859 emerged in places and times of mixing. More specifically, median βNTI values for a given sample were significantly (P=0.0009, R2=0.19) and positively related to the deviation from the NPOC mixing model (Fig. 6a); this relationship was stronger when considering only hyporheic zone samples (P=0.0001, R2=0.37). Median βNTI values associated with negative NPOC deviations were also significantly (P<<0.0001) more negative than βNTI values associated with effectively no deviation (Fig. 6b). Negative NPOC deviations—which we equate with groundwater–surface water mixing-induced increases in microbial activity—were therefore associated with βNTI values centred near −2, which indicates a shift to a selective environment dominated by a small number of consistently imposed selective pressures (that is, homogeneous ecological selection). This aligns with experimental results of Findlay et al.31 but contrasts sharply with microbial communities from the current study associated with no effective deviation from the NPOC mixing model; these communities were associated with βNTI values (centred near −1) consistent with stochastic ecological processes58.\n",
      "\n",
      "\n",
      "Microbial taxa connected to biogeochemical shifts\n",
      "\n",
      "To assess possible microbial drivers of NPOC consumption in the hyporheic zone, community structure within samples exhibiting different magnitudes of deviation from the NPOC mixing model were compared. To remove effects of groundwater–surface water mixing, only samples with similar Cl− concentrations were compared against each other; we did not constrain the spatial and temporal relationships between pairs of samples selected for this analysis. Six pairs of samples were compared and 11 OTUs increased in relative abundance as NPOC deviations grew larger. In some instances, significant enrichments were measured between samples (>6% relative abundance increase) (Fig. 7).\n",
      "\n",
      "Limitations of 16S ribosomal RNA analyses prevent species-level identifications and inferences of exact metabolic potential, but taxonomic placings may be used to broadly assess physiological roles; most broadly, microbial groups assigned to the 11 OTUs all contain heterotrophic members. OTUs affiliated with the Actinomycetales ACK-M1 lineage exhibited a particularly strong increase in relative abundance concomitant with a greater loss of NPOC, relative to the mixing model (Fig. 7). Environmental strains within the Actinomycetales have been shown to contain increased numbers of ABC-type mce transporters, which may enable metabolic versatility in environments containing varied carbon substrates such as NPOC60. Likewise, members of the genus Limnohabitans isolated from aquatic environments have previously shown high rates of substrate uptake and the ability to grow on simple organic acids and sugars61. The largest relative abundance increase between two samples exhibiting the greatest NPOC deviation was for an OTU affiliated with the Pelagibacteraceae (∼7% relative abundance increase). Species within this family—members of the SAR11 cluster—are thought to be the most abundant microorganisms on Earth and recent studies have focused on the ecophysiology of a SAR11 freshwater sister group, termed LD12 (ref. 62). Studies have demonstrated that members of this group exhibit both a pronounced preference for uptake of glutamine and glutamate over other amino acids and higher uptake rates for these substrates relative to other freshwater community members62. These microorganisms also use other simple organic compounds such as acetate, glucose and fructose.\n",
      "\n",
      "Considering our results collectively, we infer that when surface water intrusion delivers labile organic carbon to the hyporheic zone, it stimulates microbial activity, but does so non-randomly through imposition of a homogeneous ecological selective environment565859. We further infer that in places and times in which there is no (or insufficient) groundwater–surface water mixing, microbial community composition is governed principally by organismal dispersal. Evaluations of these inferences are needed beyond our study system, the timespan it was sampled and suspended microbes.\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "Biogeochemical activity is often enhanced at environmental transitions due to reciprocal provisioning of resources that limit activity in one environment but are abundant in the other3. The hyporheic zone is a classic example of this dynamic15 and here we aimed to evaluate coupled dynamics between the biogeochemical environment and microbial composition, in response to changes in groundwater–surface water mixing within the hyporheic zone. Our results suggest that—across the spatial and temporal scales studied here—microbial communities within groundwater or surface water are governed by stochastic ecological processes, but that surface water intrusion into the hyporheic zone delivers labile organic carbon that stimulates microbial respiration and causes (ecologically) deterministic shifts in microbial community composition.\n",
      "\n",
      "We have revealed a collection of field-scale patterns that are consistent with a strong impact of surface water intrusion on hyporheic zone microbiomes and biogeochemical function. There are numerous mechanisms underlying these coupled dynamics that will require additional investigation. To guide future efforts and connect our study to a broader range of systems, we collate the patterns observed here into a conceptual model (Fig. 8). It is important to note that the conceptual model developed below is speculative and represents a collection of hypotheses that emerge from our results; it is intended to be a point of departure for future work. The conceptual model is most relevant to systems that transition between saturated and unsaturated conditions, such as sections of rivers that are influenced by dam operations and/or tides. We frame the conceptual model around a key question: what is the source (or sources) of labile organic carbon that is transported by surface water and why is it not consumed before entering the hyporheic zone?\n",
      "\n",
      "To develop a potential answer to this question, we note that the rise and fall of river stage results in saturated/unsaturated dynamics in near-surface hyporheic zone sediments that are similar to—although not as extreme as—wetting/drying dynamics in semi-arid soils. It is well known that rewetting of semi-arid soils causes a rapid increase in CO2 emissions (that is, the ‘Birch Effect')636465. Mechanisms underlying this phenomenon probably vary across systems, but are grouped into two categories66: (i) the ‘supply hypothesis' posits that rewetting increases the supply of labile organic carbon through physical process such as aggregate dissolution and hydrologic redistribution of spatially isolated organic matter67; and (ii) the ‘stress hypothesis' posits that rewetting causes drought adapted microorganisms to either lyse due to rapid increases in turgor pressure or to rapidly excrete labile osmolytes to avoid lysis6869.\n",
      "\n",
      "The stress and supply hypotheses contain multiple potential mechanisms, but considering details of our study system allows refinement. Given the relatively rapid rise and fall of river stage (Fig. 2b), sediments would not have desiccated such that hyporheic zone microbes would not have been osmotically stressed; we thus reject the stress hypothesis as an explanation for dynamics in our study system. Aggregate dissolution—and the associated release of occulated organic matter—is one mechanism underlying the supply hypothesis66, but is unlikely in light of the repeated saturated/unsaturated cycles and the consistent biogeochemical response observed here; repeated high/low moisture cycles select for stable aggregates70 and aggregate dissolution should result in a progressively diminished biogeochemical response71, yet hyporheic zone NPOC concentration can be described as one continuous function of groundwater–surface water mixing conditions. We suggest that a more probable scenario is a combination of (i) physical isolation of sediment-associated microorganisms—the primary drivers of organic carbon cycling, but not directly studied here—from organic matter following the transition from advective to diffusion-based transport; a decline in river stage results in unsaturated conditions and, in turn, diffusion-based transport; and (ii) the rapid transport of dissolved organic carbon through the hyporheic zone as river stage rises (Fig. 8).\n",
      "\n",
      "We specifically hypothesize that extracellular enzymes continue to degrade large organic molecules—supplied as POC—in locations that microorganisms cannot access (temporarily) due to diffusion-based transport during low river stage (that is, during times of desaturation). Labile organic carbon would, in this case, accumulate in isolated microsites during low river stage and—following an increase in river stage—subsequently be dispersed and carried into the hyporheic zone at a rate that exceeds the ability of recently re-saturated communities to metabolize it. This mechanism is consistent with models that indicate accumulation of labile carbon during dry periods7273 and observations that enzyme activities are less impacted by soil moisture than is microbial biomass74, suggesting continued enzymatic activity in times of reduced microbial activity75. The patterns observed here suggest that transported labile carbon may be progressively consumed along hyporheic zone flow paths2039—thereby stimulating microbial activity—and may also select for particular microbial taxa that can efficiently use the transported substrates (Fig. 8).\n",
      "\n",
      "Extending these concepts leads to an additional hypothesis: longer times between surface water inundations lead to greater accumulation of labile carbon within microsites, which is consistent with observations in soil systems71. If this is true, then less frequent (and large) increases in river stage may disperse and transport larger quantities of labile organic carbon into the hyporheic zone and, in turn, may cause greater enhancement of microbial activity. Higher rates of microbial respiration can generate anaerobic microsites47 and promote removal and/or immobilization of nitrate and other redox-sensitive contaminants44. This suggests an important link between climate-change-driven shifts in watershed hydrology and the rates of elemental cycling and contaminant dynamics. If large pulses of water associated with Spring-time snowpack melt (for example, Fig. 2a) are diminished by warmer winter temperatures, microbial respiration rates may be too low to deplete O2, therefore having a significant impact on contaminant dynamics. To rigorously evaluate the potential for such outcomes (for example, through predictive hydro-biogeochemical models), the hypotheses that constitute our proposed conceptual model (see Fig. 8) will need to be directly tested and refined.\n",
      "\n",
      "\n",
      "Methods\n",
      "\n",
      "\n",
      "Field sampling\n",
      "\n",
      "Samples were collected from the Columbia River, its hyporheic zone (sediment porosity of ∼0.18 and hydraulic conductivity of ∼1–4 m per day, from ref. 49) and the adjacent unconfined aquifer underlying the Hanford 300 Area (sediment porosity of ∼0.20 and hydraulic conductivity of ∼7,000–8,000 m per day, from ref. 49) (Fig. 1). To capture a broad range of groundwater–surface water mixing conditions, we targeted different river stages across four time intervals in November 2013; river stage was monitored ∼300 m downstream of our field sampling locations using a pressure transducer coupled to a CR1000 datalogger (Campbell Scientific Inc., for both). The range of sampled mixing conditions was further enhanced by taking advantage of spatial heterogeneity within the hyporheic zone; the hyporheic zone was sampled at multiple depths at each of five locations (Fig. 1) using a custom-built multi-level sampling device (Fig. 1) installed within fully screened stainless steel piezometers; 1.2 m screen length and 5.25 cm inside diameter (MAAS Midwest, Huntley, IL). The most Northern piezometer (Fig. 1) was sampled within the depth intervals of ∼1–11 and ∼41–51 cm (it could not be installed to full depth) and the other four piezometers were sampled within the depth intervals of ∼10–20 and 110–120 cm. The five sampled piezometers were not meant to be replicates, but were instead assumed to capture different physical features of the hyporheic zone such that sampling across them provided access to a broader range of mixing conditions. Groundwater was sampled from four wells screened across the top of the subsurface aquifer (Fig. 1) and river water was sampled near the piezometers. Water from the hyporheic zone and the river were sampled using a peristaltic pump and the subsurface aquifer was sampled using submersible pumps; all systems were purged—the equivalent of at least three volumes—before sample collection.\n",
      "\n",
      "At each location and depth, pumped water passed through 0.22 μm polyethersulfone Sterivex filters (Millipore Co., Billerica, MA) for ∼30 min or until 10 l had passed through a filter, whichever came first; sampled volume ranged from 1.33 to 10 l. Sampling time was constrained instead of sampled volume because of the dynamic nature of groundwater–surface water mixing conditions within the hyporheic zone of our system; changing the filtration time across samples to achieve a consistent volume could have resulted in some filters collecting microbes across larger ranges of mixing conditions. In addition, we note that no relationships emerged in the hyporheic zone between filtered volume and (i) the concentrations of aqueous analytes (NPOC, DIC, Cl− and ; R2<0.05 for all) or (ii) the microbial community turnover metrics (Bray–Curtis, βMNTD and βNTI; Mantel r≤0.1 for all). Filters were collected, frozen on dry ice in the field and stored at −80 oC. Sterivex-filtered water was used to collect samples for water chemistry. Samples for anions and dissolved organic and inorganic carbon were collected into pre-cleaned 40 ml glass vials (Thermo Scientific), stored at 4 °C and split for ion chromatographic and carbon analysis in the lab. Samples for organic carbon profiling were also taken in 40 ml glass vials and frozen at −20 °C. Cation samples were taken in 15 ml polypropylene tubes (Becton Dickinson Co., Franklin, NJ) pre-acidified with 0.1 ml, 70% HNO3 and stored at 4 °C. Handheld meters were used to measure water temperature and conductivity (Ultrameter II, Myron L Co., Carlsbad, CA). Simultaneously, unfiltered water was preserved in glutaraldehyde (1% final) for cell counts in 15 ml polypropylene tubes (Corning Inc. Corning, NY) and stored at 4 °C, and was separately filtered through pre-combusted 47 mm glass-fibre filters (Whatman, GE Healthcare, UK) for quantification of POC. Water volume filtered for POC was quantified, with a target of 10 l, and filters were stored at −20 °C in plastic petri dishes.\n",
      "\n",
      "\n",
      "Analytical methods\n",
      "\n",
      "POC samples were removed from the freezer and air dried in a biological safety cabinet, then cut into pieces to fit into a folded cup of 40 × 40 mm silver foil (Elementar Co., Germany). Cut POC filters were acidified dropwise with 1.2 N HCl, to remove inorganic carbon, and dried in a fume hood. POC filters in silver foil were wrapped in 50 × 50 mm squares of pure tin foil (Elemental Microanalysis, UK) and compressed, to remove air, with a manual pressing tool (Elementar Co.). POC samples were analysed for C, N & S using an Elementar vario EL cube.\n",
      "\n",
      "DIC and NPOC were determined with a Shimadzu combustion carbon analyser TOC-Vcsh with ASI-V auto sampler. The DIC was determined by injection into 25% phosphoric acid at ambient temperature. Inorganic carbon standards were made from sodium bicarbonate and sodium carbonate solids (Nacalia Tesque). The calibration range for the DIC determinations was from 0.35 to 40 mg l−1 as C and measurement error was 0.5%. The NPOC was determined by acidification of the sample with 2 N HCl (Optima, Fisher) with a 5-min sparging time to remove DIC. The sample was then injected into the furnace set to 680 °C. The organic carbon standards were made from potassium hydrogen phthalate solid (Nacalia Tesque). The calibration range for the NPOC determinations was from 0.35 to 3.5 mg l−1 as C and measurement error was 0.4%. A Dionex ICS-2000 anion chromatograph with AS40 auto sampler was used to determine  and Cl− concentrations. The guard column was an IonPac AG18 guard, 4 × 50 mm (Dionex). The analytical column was an IonPac AS18, 4 × 250 mm (Dionex). The suppressor was an RFIC ASRS, 300 4-mm, self-regenerating (Dionex) and the suppressor current was set at 99 mA. A 25-min gradient method was used with 25 μl injection volumes and a 1 ml min−1 flow rate at 30 °C. KOH (22 mM) was used for the first 7 min. From 7 to 8 min, the KOH concentration increased from 22 to 40 mM. From 8 to 20 min, 40 mM KOH was used. From 20 to 25 min, the KOH concentration decreased from 40 to 22 mM. The suppressor current was set to 99 mA. Anion standards were prepared from Spex CertiPrep (Metuchen, NJ; 08840). Anion standards (1,000 mg l−1) were diluted for a calibration range from 0.60 to 120 p.p.m. and measurement error was 2.5% for Cl− and 1.4% for .\n",
      "\n",
      "Cell densities were estimated from glutaraldehyde-preserved samples using a BD Influx Flow Cytometer (Becton Dickinson, San Jose, CA) with 488-nm excitation from a Sapphire LP laser (Coherent Inc., Santa Clara, CA, USA) at 100 mW. Optimization and calibration of the fluorescence-activated cell sorting was performed using 3 μm Ultra Rainbow Fluorescent Particles (Spherotech, Lake Forest, IL, USA). Samples were vortexed and a 500-μl aliquot was stained with SYBR Gold (Life Technologies, Grand Island, NY) at a concentration of 1:1,000 in the dark for 10 min at room temperature; an unstained sample was acquired to determine background fluorescence. A population gate was set to count stained events and a 100-μl subsample was used to prime the sample line. When events were detected, the initial subsample was removed and a second 100-μl subsample was loaded onto the instrument and data acquisition was started; the entire sample was analysed. For normalization, the number of background events was subtracted from the number of events counted in stained subsamples.\n",
      "\n",
      "\n",
      "Organic carbon characterization\n",
      "\n",
      "Ultra-high-resolution characterization of dissolved organic carbon was carried out using a 15-T Bruker SolariX FT-ICR MS located at the Environmental Molecular Sciences Laboratory, a Department of Energy–Office of Biological and Environmental Research national user facility in Richland, WA. Samples were thawed overnight and then prepared by adding 0.5 ml methanol (LC/MS grade Optima, Sigma Aldrich, St Louis, MO) to 0.5 ml of sample water in an Eppendorf 96-well plate fitted with 2 ml glass sleeves. Suwannee River Fulvic Acid (SRFA) standard (International Humic Substances Society) was used as a control and injected every 15th sample, to assure instrument stability. SRFA is the most commonly used standard for characterization of organic matter and humic material, and its composition has been extensively characterized76777879. The carbon (C) content of SRFA is 52.44% (refs 76, 77). SRFA samples were prepared from the International Humic Substances Society as received powder at 1 mg ml−1 in MilliQ filtered water and diluted to 2 μg ml−1 to serve as a control. Lab blanks were prepared, 50/50 LC/MS grade methanol/MilliQ filtered water (v/v) were then injected after the control, to verify no sample carryover.\n",
      "\n",
      "Samples were injected directly into the instrument using an automated custom-built system and ion accumulation time was optimized for all samples. A standard Bruker electrospray ionization source was used to generate negatively charged molecular ions. Samples were introduced to the electrospray ionization source equipped with a fused silica tube (30 μm intradermally) through an Agilent 1200 series pump (Agilent Technologies) at a flow rate of 3.0 μl min−1. Experimental conditions were as follows: needle voltage +4.4 kV, Q1 set to 50 m/z and the heated resistively coated glass capillary operated at 180 °C. These were optimal parameters established in earlier dissolved organic carbon characterization experiments. One hundred and forty-four individual scans were averaged for each sample and internally calibrated using an organic matter homologous series separated by 14 Da (–CH2 groups). The mass measurement accuracy was typically within 1 p.p.m. for singly charged ions across a broad m/z range (100–1,100 m/z). The mass resolution was 350,000 at 339.112 Da. DataAnalysis software (BrukerDaltonik version 4.2) was used to convert raw spectra to list of m/z values applying Fourier transform mass spectrometry peak picker with S/N threshold set to 7 and absolute intensity threshold to the default value of 100. Change in organic carbon profile between samples was based on the shifts in the presence/absence of m/z peaks and was quantified using the Sorensen dissimilarity metric commonly used in ecological studies, to measure turnover in community composition; the distance function in the ecodist R package was used for Sorensen calculations.\n",
      "\n",
      "\n",
      "DNA extraction and processing\n",
      "\n",
      "For each Sterivex filter, the housing was cracked open and the filter was removed using sterile instruments. DNA was extracted using a modification of methods from Bostrom et al.80. Filter material was incubated at 85 oC for 15 min in lysis buffer, as defined in Bostrom et al.80 To avoid DNA fragmentation, the solution was slowly cooled and lysozyme was added to a final concentration of 1 mg ml−1. The resulting solution was incubated at 37 °C for 30 min. SDS was added to 1% final concentration and proteinase-K was added to 100 μl ml−1 final concentration. The resulting solution was incubated at 55 °C for ∼12 h. To facilitate cell lysis, samples were run through three freeze–thaw cycles using liquid N2 followed by warming to 55 °C. The sample was then vortexed followed by isopropanol precipitation and pellet elution in TE80. The sample was then treated with 10 μg ml−1 RNase at 37 °C for 30 min, followed by phenol–chloroform clean-up, isopropanol precipitation and elution in TE. Extracted DNA was stored at −20 °C until shipment (on dry ice) to the sequencing facility at Argonne National Laboratory.\n",
      "\n",
      "At Argonne National Laboratory, DNA was amplified using the Earth Microbiome Project barcoded primer set, adapted for the Illumina MiSeq by adding nine extra bases in the adapter region of the forward amplification primer that support paired-end sequencing. The V4 region (515F-806R) of the 16S rRNA gene was amplified with region-specific primers that included the Illumina flowcell adapter sequences. The reverse amplification primer contained a 12-base barcode sequence that supports pooling of up to 2,167 samples in the single MiSeq lane8182. Each 25 μl of PCR reaction contained 12 μl of MoBio PCR Water (Certified DNA-Free), 10 μl of 5 Prime HotMasterMix (1 ×), 1 μl of Forward Primer (5 μM concentration, 200 pM final), 1 μl Golay Barcode Tagged Reverse Primer (5 μM concentration, 200 pM final) and 1 μl of template DNA. The conditions for PCR were as follows: 94 °C for 3 min to denature DNA, with 35 cycles at 94 °C for 45 s, 50 °C for 60 s and 72 °C for 90 s; with a final extension of 10 min at 72 °C, to ensure complete amplification. Amplicons were quantified using PicoGreen (Invitrogen) and a plate reader. Appropriate volumes of PCR product from each sample were then pooled so that each sample was represented with an equal amount of product. This pool was cleaned using the UltraClean PCR Clean-Up Kit (MoBIO), and quantified using the Qubit (Invitrogen). After quantification, the molarity of the pool was determined and diluted to 2 nM, denatured and then diluted to a final concentration of 6.75 pM with a 10% PhiX spike for sequencing on the Illumina MiSeq.\n",
      "\n",
      "Resulting sequences were processed in QIIME 1.8.0 (ref. 83). First, the function split_libraries_fastq.py was used to demultiplex the fastq-formatted sequences, with a Phred84 quality cutoff of 20. To identify chimeras the function identify_chimeric_seqs.py was used to call USEARCH v6.1 (ref. 85) with ‘non_chimera_retention' set to ‘intersection.' After removing identified chimeras (with function filter_fasta.py), the function pick_open_reference_otus.py was used to call USEARCH v6.1, pick OTUs and generate the associated phylogeny using SILVA (97% similarity, release 111) (http://www.arb-silva.de/) as the reference; default parameters were used, except for the following modifications: ‘suppress_de_novo_chimera_detection', ‘suppress_reference_chimera_detection' and ‘derep_fullseq' were all set to ‘True' and ‘prefilter_percent_id' was set to 0.\n",
      "\n",
      "All additional analyses were carried out in R (http://cran.r-project.org/). OTUs identified as chloroplasts were removed and the OTU table was subsequently rarefied to the lowest number of sequences (4,899) found within an individual sample using the rrarefy function in the vegan package. OTUs that had zero abundance across all samples, following rarefaction, were removed from both the OTU table and the phylogeny using the match.phylo.data function in the picante library. The number of OTUs present in each sample (OTU richness) was enumerated as an estimate of microbial diversity. The Bray–Curtis metric was used to quantify turnover in community composition and was calculated with the distance function in the ecodist package. βMNTD was used to quantify turnover in the phylogenetic structure of communities; abundance weighted βMNTD was calculated using the comdistnt function in the picante package (see also refs 35, 55, 58, 86). To evaluate the potential for shifts in the relative balance between stochastic and deterministic ecological processes, randomizations were used to generate a null distribution (999 randomizations) for each βMNTD estimate; βMNTD is a pairwise distance metric such that a null βMNTD distribution was generated for each pairwise community comparison. The βNTI measures the difference between observed βMNTD and the mean of the null distribution in units of standard deviations (see also refs 35, 55, 58, 86):\n",
      "\n",
      "\n",
      "Statistical analyses\n",
      "\n",
      "To characterize groundwater–surface water mixing conditions within the hyporheic zone we assumed Cl− to be a conserved tracer and took advantage of large differences in Cl− concentration between groundwater and surface water. Low variation was observed within river water for Cl− and the reactive analytes of interest (that is, , NPOC and DIC), but greater variation was observed within groundwater. In turn, two linear mixing models were generated for each analyte using mean river water conditions as one end-member and extreme groundwater conditions as the other end-member; for example, one NPOC mixing model used mean river conditions with maximum NPOC and maximum Cl− within groundwater, whereas the other mixing model used mean river conditions with minimum NPOC and minimum Cl− within groundwater. This approach was taken because of uncertainty in the source of groundwater sampled in the hyporheic zone.\n",
      "\n",
      "To evaluate potential drivers of organic carbon composition we correlated the pairwise Sorensen dissimilarity estimates with changes in NPOC concentration and changes in groundwater–surface water mixing conditions as estimated by changes in Cl− concentration. The statistical significance of the resulting relationships were evaluated with permutation-based Mantel tests (1,000 permuations) using the mantel function in the ecodist package. In addition, the median m/z value was found for each sample and distributions of these estimates were examined within each of the three water types studied here: river water, groundwater and hyporheic zone water. To determine whether observed changes in the m/z distributions were statistically significant, a Kruskal–Wallis test and its associated post-hoc test were used (functions kruskal.test and kruskalmc from the pgirmess library). These same statistical tests were used to evaluate shifts in POC across the three types of water.\n",
      "\n",
      "Ordinary least squares regression was used to evaluate whether OTU richness was related to groundwater–surface water mixing conditions, NPOC concentrations or POC concentrations; analogous analyses were carried out with suspended cell density as the dependent variable. Distance matrix regression and permutation-based Mantel tests were used to evaluate relationships between each of the three microbial community turnover metrics and changes in either NPOC concentration or mixing conditions. For each microbial community, we also calculated its median βNTI value based on its pairwise comparisons to all other sampled communities. For example, if community A was compared (separately) with communities B, C and D, there would be three separate βNTI estimates associated with community A; we found the median of these βNTI values. Each community, therefore, is associated with one median βNTI estimate; these estimates were related to the NPOC mixing model deviations by linear regression. The median βNTI estimates were also pooled into two groups—negative deviation or no deviation from the NPOC mixing model—and the resulting distributions were compared with a Kruskal–Wallis test.\n",
      "\n",
      "To identify OTUs potentially involved in the consumption of NPOC, microbial community composition was compared between pairs of hyporheic zone samples, which both deviated from the NPOC mixing model and which exhibited similar Cl− (±0.5 mg l−1) concentrations (that is, compared samples had similar groundwater–surface water mixing conditions, but different levels of deviation from the NPOC mixing model). Six pairs of samples were identified that met these criteria. Only OTUs with >1% relative abundance in at least one sample were used for this analysis. Taxonomic information was retrieved for OTUs showing an increase in relative abundance of >1% in the sample with the larger (that is, more negative) deviation from the NPOC mixing model.\n",
      "\n",
      "\n",
      "Additional information\n",
      "\n",
      "\n",
      "How to cite this article: Stegen, J. C. et al. Groundwater–surface water mixing shifts ecological assembly processes and stimulates organic carbon turnover. Nat. Commun. 7:11237 doi: 10.1038/ncomms11237 (2016).\n",
      "\n",
      "\n",
      "Supplementary Material\n",
      "\n",
      "\n",
      "Supplementary Information\n",
      "\n",
      "Supplementary Figures 1-4\n"
     ]
    }
   ],
   "source": [
    "view_one_article(final_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pubcrawler.article.Article at 0x11181e240>"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "xml = articles.find_one({\"_id\": \"4546277\"})[\"extracted_text\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "30029"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(xml)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "soup = BeautifulSoup(xml)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD v1.0 20120330//EN\" \"JATS-archivearticle1.dtd\">\n",
      "<html>\n",
      " <body>\n",
      "  <article article-type=\"research-article\" xmlns:mml=\"http://www.w3.org/1998/Math/MathML\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
      "   <?properties open_access?>\n",
      "   <front>\n",
      "    <journal-meta>\n",
      "     <journal-id journal-id-type=\"nlm-ta\">\n",
      "      PLoS One\n",
      "     </journal-id>\n",
      "     <journal-id journal-id-type=\"iso-abbrev\">\n",
      "      PLoS ONE\n",
      "     </journal-id>\n",
      "     <journal-id journal-id-type=\"publisher-id\">\n",
      "      plos\n",
      "     </journal-id>\n",
      "     <journal-id journal-id-type=\"pmc\">\n",
      "      plosone\n",
      "     </journal-id>\n",
      "     <journal-title-group>\n",
      "      <journal-title>\n",
      "       PLoS ONE\n",
      "      </journal-title>\n",
      "     </journal-title-group>\n",
      "     <issn pub-type=\"epub\">\n",
      "      1932-6203\n",
      "     </issn>\n",
      "     <publisher>\n",
      "      <publisher-name>\n",
      "       Public Library of Science\n",
      "      </publisher-name>\n",
      "      <publisher-loc>\n",
      "       San Francisco, CA USA\n",
      "      </publisher-loc>\n",
      "     </publisher>\n",
      "    </journal-meta>\n",
      "    <article-meta>\n",
      "     <article-id pub-id-type=\"pmid\">\n",
      "      26291610\n",
      "     </article-id>\n",
      "     <article-id pub-id-type=\"pmc\">\n",
      "      4546277\n",
      "     </article-id>\n",
      "     <article-id pub-id-type=\"doi\">\n",
      "      10.1371/journal.pone.0135749\n",
      "     </article-id>\n",
      "     <article-id pub-id-type=\"publisher-id\">\n",
      "      PONE-D-15-12790\n",
      "     </article-id>\n",
      "     <article-categories>\n",
      "      <subj-group subj-group-type=\"heading\">\n",
      "       <subject>\n",
      "        Research Article\n",
      "       </subject>\n",
      "      </subj-group>\n",
      "     </article-categories>\n",
      "     <title-group>\n",
      "      <article-title>\n",
      "       Air Pollution in China: Mapping of Concentrations and Sources\n",
      "      </article-title>\n",
      "      <alt-title alt-title-type=\"running-head\">\n",
      "       Air Pollution in China: Mapping of Concentrations and Sources\n",
      "      </alt-title>\n",
      "     </title-group>\n",
      "     <contrib-group>\n",
      "      <contrib contrib-type=\"author\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Rohde\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Robert A.\n",
      "        </given-names>\n",
      "       </name>\n",
      "       <xref ref-type=\"aff\" rid=\"aff001\">\n",
      "        <sup>\n",
      "         1\n",
      "        </sup>\n",
      "       </xref>\n",
      "       <xref ref-type=\"corresp\" rid=\"cor001\">\n",
      "        *\n",
      "       </xref>\n",
      "      </contrib>\n",
      "      <contrib contrib-type=\"author\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Muller\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Richard A.\n",
      "        </given-names>\n",
      "       </name>\n",
      "       <xref ref-type=\"aff\" rid=\"aff001\">\n",
      "        <sup>\n",
      "         1\n",
      "        </sup>\n",
      "       </xref>\n",
      "       <xref ref-type=\"aff\" rid=\"aff002\">\n",
      "        <sup>\n",
      "         2\n",
      "        </sup>\n",
      "       </xref>\n",
      "       <xref ref-type=\"corresp\" rid=\"cor001\">\n",
      "        *\n",
      "       </xref>\n",
      "      </contrib>\n",
      "     </contrib-group>\n",
      "     <aff id=\"aff001\">\n",
      "      <label>\n",
      "       1\n",
      "      </label>\n",
      "      <addr-line>\n",
      "       Berkeley Earth, Berkeley, California, United States of America\n",
      "      </addr-line>\n",
      "     </aff>\n",
      "     <aff id=\"aff002\">\n",
      "      <label>\n",
      "       2\n",
      "      </label>\n",
      "      <addr-line>\n",
      "       Department of Physics, University of California, Berkeley, California, United States of America\n",
      "      </addr-line>\n",
      "     </aff>\n",
      "     <contrib-group>\n",
      "      <contrib contrib-type=\"editor\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Ding\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Aijun\n",
      "        </given-names>\n",
      "       </name>\n",
      "       <role>\n",
      "        Editor\n",
      "       </role>\n",
      "       <xref ref-type=\"aff\" rid=\"edit1\">\n",
      "       </xref>\n",
      "      </contrib>\n",
      "     </contrib-group>\n",
      "     <aff id=\"edit1\">\n",
      "      <addr-line>\n",
      "       Nanjing University, CHINA\n",
      "      </addr-line>\n",
      "     </aff>\n",
      "     <author-notes>\n",
      "      <fn fn-type=\"COI-statement\" id=\"coi001\">\n",
      "       <p>\n",
      "        <bold>\n",
      "         Competing Interests:\n",
      "        </bold>\n",
      "        The authors have declared that no competing interests exist.\n",
      "       </p>\n",
      "      </fn>\n",
      "      <fn fn-type=\"con\" id=\"contrib001\">\n",
      "       <p>\n",
      "        Conceived and designed the experiments: RAM RAR. Performed the experiments: RAR. Analyzed the data: RAR RAM. Contributed reagents/materials/analysis tools: RAR. Wrote the paper: RAM RAR.\n",
      "       </p>\n",
      "      </fn>\n",
      "      <corresp id=\"cor001\">\n",
      "       * E-mail:\n",
      "       <email>\n",
      "        robert@berkeleyearth.org\n",
      "       </email>\n",
      "       (RAR);\n",
      "       <email>\n",
      "        rich@berkeleyearth.org\n",
      "       </email>\n",
      "       (RAM)\n",
      "      </corresp>\n",
      "     </author-notes>\n",
      "     <pub-date pub-type=\"epub\">\n",
      "      <day>\n",
      "       20\n",
      "      </day>\n",
      "      <month>\n",
      "       8\n",
      "      </month>\n",
      "      <year>\n",
      "       2015\n",
      "      </year>\n",
      "     </pub-date>\n",
      "     <pub-date pub-type=\"collection\">\n",
      "      <year>\n",
      "       2015\n",
      "      </year>\n",
      "     </pub-date>\n",
      "     <volume>\n",
      "      10\n",
      "     </volume>\n",
      "     <issue>\n",
      "      8\n",
      "     </issue>\n",
      "     <elocation-id>\n",
      "      e0135749\n",
      "     </elocation-id>\n",
      "     <history>\n",
      "      <date date-type=\"received\">\n",
      "       <day>\n",
      "        24\n",
      "       </day>\n",
      "       <month>\n",
      "        3\n",
      "       </month>\n",
      "       <year>\n",
      "        2015\n",
      "       </year>\n",
      "      </date>\n",
      "      <date date-type=\"accepted\">\n",
      "       <day>\n",
      "        25\n",
      "       </day>\n",
      "       <month>\n",
      "        7\n",
      "       </month>\n",
      "       <year>\n",
      "        2015\n",
      "       </year>\n",
      "      </date>\n",
      "     </history>\n",
      "     <permissions>\n",
      "      <copyright-statement>\n",
      "       © 2015 Rohde, Muller\n",
      "      </copyright-statement>\n",
      "      <copyright-year>\n",
      "       2015\n",
      "      </copyright-year>\n",
      "      <copyright-holder>\n",
      "       Rohde, Muller\n",
      "      </copyright-holder>\n",
      "      <license xlink:href=\"http://creativecommons.org/licenses/by/4.0/\">\n",
      "       <license-p>\n",
      "        This is an open-access article distributed under the terms of the Creative Commons Attribution License, which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are properly credited.\n",
      "       </license-p>\n",
      "      </license>\n",
      "     </permissions>\n",
      "     <self-uri content-type=\"pdf\" xlink:href=\"pone.0135749.pdf\" xlink:type=\"simple\">\n",
      "     </self-uri>\n",
      "     <abstract>\n",
      "      <p>\n",
      "       China has recently made available hourly air pollution data from over 1500 sites, including airborne particulate matter (PM), SO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       , NO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       , and O\n",
      "       <sub>\n",
      "        3\n",
      "       </sub>\n",
      "       . We apply Kriging interpolation to four months of data to derive pollution maps for eastern China. Consistent with prior findings, the greatest pollution occurs in the east, but significant levels are widespread across northern and central China and are not limited to major cities or geologic basins. Sources of pollution are widespread, but are particularly intense in a northeast corridor that extends from near Shanghai to north of Beijing. During our analysis period, 92% of the population of China experienced &gt;120 hours of unhealthy air (US EPA standard), and 38% experienced average concentrations that were unhealthy. China’s population-weighted average exposure to PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       was 52 μg/m\n",
      "       <sup>\n",
      "        3\n",
      "       </sup>\n",
      "       . The observed air pollution is calculated to contribute to 1.6 million deaths/year in China [0.7–2.2 million deaths/year at 95% confidence], roughly 17% of all deaths in China.\n",
      "      </p>\n",
      "     </abstract>\n",
      "     <funding-group>\n",
      "      <funding-statement>\n",
      "       The authors received support from Berkeley Earth, a 501c3 non-profit, whose website is\n",
      "       <ext-link ext-link-type=\"uri\" xlink:href=\"http://www.berkeleyearth.org\">\n",
      "        http://www.berkeleyearth.org\n",
      "       </ext-link>\n",
      "       . There was no directed funding for this project, and no funders were involved in the design, execution, or reporting of this study.\n",
      "      </funding-statement>\n",
      "     </funding-group>\n",
      "     <counts>\n",
      "      <fig-count count=\"5\">\n",
      "      </fig-count>\n",
      "      <table-count count=\"0\">\n",
      "      </table-count>\n",
      "      <page-count count=\"14\">\n",
      "      </page-count>\n",
      "     </counts>\n",
      "     <custom-meta-group>\n",
      "      <custom-meta id=\"data-availability\">\n",
      "       <meta-name>\n",
      "        Data Availability\n",
      "       </meta-name>\n",
      "       <meta-value>\n",
      "        Detailed summary tables of results are included in the supplementary material. Observational input data and gridded spatial-temporal fields are provided at both the Berkeley Earth website (\n",
      "        <ext-link ext-link-type=\"uri\" xlink:href=\"http://berkeleyearth.lbl.gov/manual/china_air_quality/\">\n",
      "         http://berkeleyearth.lbl.gov/manual/china_air_quality/\n",
      "        </ext-link>\n",
      "        ) and at the Harvard Dataverse repository (\n",
      "        <ext-link ext-link-type=\"uri\" xlink:href=\"http://dx.doi.org/10.7910/DVN/D6UMJ0\">\n",
      "         http://dx.doi.org/10.7910/DVN/D6UMJ0\n",
      "        </ext-link>\n",
      "        ).\n",
      "       </meta-value>\n",
      "      </custom-meta>\n",
      "     </custom-meta-group>\n",
      "    </article-meta>\n",
      "    <notes>\n",
      "     <title>\n",
      "      Data Availability\n",
      "     </title>\n",
      "     <p>\n",
      "      Detailed summary tables of results are included in the supplementary material. Observational input data and gridded spatial-temporal fields are provided at both the Berkeley Earth website (\n",
      "      <ext-link ext-link-type=\"uri\" xlink:href=\"http://berkeleyearth.lbl.gov/manual/china_air_quality/\">\n",
      "       http://berkeleyearth.lbl.gov/manual/china_air_quality/\n",
      "      </ext-link>\n",
      "      ) and at the Harvard Dataverse repository (\n",
      "      <ext-link ext-link-type=\"uri\" xlink:href=\"http://dx.doi.org/10.7910/DVN/D6UMJ0\">\n",
      "       http://dx.doi.org/10.7910/DVN/D6UMJ0\n",
      "      </ext-link>\n",
      "      ).\n",
      "     </p>\n",
      "    </notes>\n",
      "   </front>\n",
      "   <sec id=\"sec001\" sec-type=\"intro\">\n",
      "    <title>\n",
      "     Introduction\n",
      "    </title>\n",
      "    <p>\n",
      "     Air pollution is a problem for much of the developing world and is believed to kill more people worldwide than AIDS, malaria, breast cancer, or tuberculosis [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref001\">\n",
      "      1\n",
      "     </xref>\n",
      "     –\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref004\">\n",
      "      4\n",
      "     </xref>\n",
      "     ]. Airborne particulate matter (PM) is especially detrimental to health [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref005\">\n",
      "      5\n",
      "     </xref>\n",
      "     –\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref008\">\n",
      "      8\n",
      "     </xref>\n",
      "     ], and has previously been estimated to cause between 3 and 7 million deaths every year, primarily by creating or worsening cardiorespiratory disease [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref002\">\n",
      "      2\n",
      "     </xref>\n",
      "     –\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref004\">\n",
      "      4\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref006\">\n",
      "      6\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref007\">\n",
      "      7\n",
      "     </xref>\n",
      "     ]. Particulate sources include electric power plants, industrial facilities, automobiles, biomass burning, and fossil fuels used in homes and factories for heating. In China, air pollution was previously estimated to contribute to 1.2 to 2 million deaths annually [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref002\">\n",
      "      2\n",
      "     </xref>\n",
      "     –\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref004\">\n",
      "      4\n",
      "     </xref>\n",
      "     ].\n",
      "    </p>\n",
      "    <p>\n",
      "     In 2012, China adopted the Ambient Air Quality Standard [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref009\">\n",
      "      9\n",
      "     </xref>\n",
      "     ], and began development of a national Air Reporting System that now includes 945 sites in 190 cities. These automated stations report hourly via the internet, and focus on six pollutants: particulate matter &lt; 2.5 microns (PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     ), particulate matter &lt; 10 microns (PM\n",
      "     <sub>\n",
      "      10\n",
      "     </sub>\n",
      "     ), sulfur dioxide (SO\n",
      "     <sub>\n",
      "      2\n",
      "     </sub>\n",
      "     ), nitrogen dioxide (NO\n",
      "     <sub>\n",
      "      2\n",
      "     </sub>\n",
      "     ), ozone (O\n",
      "     <sub>\n",
      "      3\n",
      "     </sub>\n",
      "     ), and carbon monoxide (CO). Provincial governments perform air quality monitoring at 600 additional locations that are not yet integrated into the national system. Previous studies of regional scale air pollution have generally relied on satellite data [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref010\">\n",
      "      10\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref011\">\n",
      "      11\n",
      "     </xref>\n",
      "     ] or modeling [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref012\">\n",
      "      12\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref013\">\n",
      "      13\n",
      "     </xref>\n",
      "     ], but the high density of hourly data in China now allows regional patterns to be constructed directly from ground observations.\n",
      "    </p>\n",
      "   </sec>\n",
      "   <sec id=\"sec002\" sec-type=\"materials|methods\">\n",
      "    <title>\n",
      "     Materials and Methods\n",
      "    </title>\n",
      "    <p>\n",
      "     Though China deserves praise for its monitoring system and transparent communication, most archived observations are not publicly available. To compensate, real-time data was downloaded every hour during a four month interval from April 5, 2014 to August 5, 2014. Due to download restrictions on the official Chinese air quality reporting system, two different third-party sources were used:\n",
      "     <ext-link ext-link-type=\"uri\" xlink:href=\"http://PM25.in\">\n",
      "      PM25.in\n",
      "     </ext-link>\n",
      "     and\n",
      "     <ext-link ext-link-type=\"uri\" xlink:href=\"http://AQICN.org\">\n",
      "      AQICN.org\n",
      "     </ext-link>\n",
      "     .\n",
      "     <ext-link ext-link-type=\"uri\" xlink:href=\"http://PM25.in\">\n",
      "      PM25.in\n",
      "     </ext-link>\n",
      "     is a direct mirror of data from the 945 stations in China’s national network, while\n",
      "     <ext-link ext-link-type=\"uri\" xlink:href=\"http://AQICN.org\">\n",
      "      AQICN.org\n",
      "     </ext-link>\n",
      "     is the world’s largest aggregator of real-time air quality data and included many additional sites in China and surrounding areas. Nearly all of the additional data from within China originates from stations operated by provincial environmental agencies that have not yet been incorporated in China’s national network.\n",
      "    </p>\n",
      "    <p>\n",
      "     Consistency, quality control, and validation checks were applied to the raw data prior to further analysis in order to reduce the impact of outliers, badly calibrated instruments, and other problems. The most common quality problem was associated with stuck instruments that implausibly reported the same concentration continuously for many hours. A regional consistency check was also applied to verify that each station was reporting data similar to its neighboring stations. Approximately 8% of the data was removed as a result of the quality control review. Further details are described in the supplemental material (\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s001\">\n",
      "      S1 Text\n",
      "     </xref>\n",
      "     ).\n",
      "    </p>\n",
      "    <p>\n",
      "     As little monitoring is conducted in western China (\n",
      "     <xref ref-type=\"fig\" rid=\"pone.0135749.g001\">\n",
      "      Fig 1\n",
      "     </xref>\n",
      "     ); we will focus on China east of 95° E, which includes 97% of the population. After removing stations with a high percentage of missing values or with other quality control problems, this study used 880 national network sites, 640 other sites in China and Taiwan, and 236 sites in other countries within 500 km of China (mostly South Korea). The air quality network is skewed towards urban areas, often with several sites per city and fewer, if any, in rural areas. For the\n",
      "     <italic>\n",
      "      n\n",
      "     </italic>\n",
      "     -th site, we use\n",
      "     <italic>\n",
      "      p\n",
      "     </italic>\n",
      "     <sub>\n",
      "      <italic>\n",
      "       n\n",
      "      </italic>\n",
      "     </sub>\n",
      "     (\n",
      "     <italic>\n",
      "      t\n",
      "     </italic>\n",
      "     ) to denote the pollutant concentration time series and\n",
      "     <inline-formula id=\"pone.0135749.e001\">\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone.0135749.e001g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e001.jpg\">\n",
      "       </graphic>\n",
      "       <mml:math id=\"M1\">\n",
      "        <mml:mrow>\n",
      "         <mml:mover accent=\"true\">\n",
      "          <mml:mrow>\n",
      "           <mml:msub>\n",
      "            <mml:mi>\n",
      "             p\n",
      "            </mml:mi>\n",
      "            <mml:mi>\n",
      "             n\n",
      "            </mml:mi>\n",
      "           </mml:msub>\n",
      "          </mml:mrow>\n",
      "          <mml:mo stretchy=\"true\">\n",
      "           ¯\n",
      "          </mml:mo>\n",
      "         </mml:mover>\n",
      "        </mml:mrow>\n",
      "       </mml:math>\n",
      "      </alternatives>\n",
      "     </inline-formula>\n",
      "     to denote the mean pollutant concentration.\n",
      "    </p>\n",
      "    <fig id=\"pone.0135749.g001\" orientation=\"portrait\" position=\"float\">\n",
      "     <object-id pub-id-type=\"doi\">\n",
      "      10.1371/journal.pone.0135749.g001\n",
      "     </object-id>\n",
      "     <label>\n",
      "      Fig 1\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Map of real-time air pollution monitoring stations.\n",
      "      </title>\n",
      "      <p>\n",
      "       Map shows the locations of air quality monitoring sites in China and surrounding areas with sufficient hourly data to be included in this study. Selection criteria and data sources are described in\n",
      "       <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s001\">\n",
      "        S1 Text\n",
      "       </xref>\n",
      "       . The map was prepared in MATLAB using political boundaries from the Global Database of Administrative Areas (version 2;\n",
      "       <ext-link ext-link-type=\"uri\" xlink:href=\"http://gadm.org/\">\n",
      "        http://gadm.org/\n",
      "       </ext-link>\n",
      "       ).\n",
      "      </p>\n",
      "     </caption>\n",
      "     <graphic xlink:href=\"pone.0135749.g001\">\n",
      "     </graphic>\n",
      "    </fig>\n",
      "    <p>\n",
      "     For each pollutant, a correlation vs. distance function was estimated by computing all possible pairwise correlations between different stations and fitting the resulting correlations to a two part exponential decay as a function of distance. The resulting functional forms are stated in\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s003\">\n",
      "      S2 Table\n",
      "     </xref>\n",
      "     and shown in\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s006\">\n",
      "      S2\n",
      "     </xref>\n",
      "     and\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s007\">\n",
      "      S3\n",
      "     </xref>\n",
      "     Figs. The correlation functions are used to construct correlation matrices that in turn are used to compute Kriging coefficients [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref014\">\n",
      "      14\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref015\">\n",
      "      15\n",
      "     </xref>\n",
      "     ],\n",
      "     <inline-formula id=\"pone.0135749.e002\">\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone.0135749.e002g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e002.jpg\">\n",
      "       </graphic>\n",
      "       <mml:math id=\"M2\">\n",
      "        <mml:mrow>\n",
      "         <mml:msub>\n",
      "          <mml:mi>\n",
      "           K\n",
      "          </mml:mi>\n",
      "          <mml:mi>\n",
      "           n\n",
      "          </mml:mi>\n",
      "         </mml:msub>\n",
      "         <mml:mrow>\n",
      "          <mml:mo>\n",
      "           (\n",
      "          </mml:mo>\n",
      "          <mml:mover accent=\"true\">\n",
      "           <mml:mi>\n",
      "            x\n",
      "           </mml:mi>\n",
      "           <mml:mo>\n",
      "            →\n",
      "           </mml:mo>\n",
      "          </mml:mover>\n",
      "          <mml:mo>\n",
      "           )\n",
      "          </mml:mo>\n",
      "         </mml:mrow>\n",
      "        </mml:mrow>\n",
      "       </mml:math>\n",
      "      </alternatives>\n",
      "     </inline-formula>\n",
      "     .\n",
      "    </p>\n",
      "    <p>\n",
      "     The interpolated pollutant field,\n",
      "     <inline-formula id=\"pone.0135749.e003\">\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone.0135749.e003g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e003.jpg\">\n",
      "       </graphic>\n",
      "       <mml:math id=\"M3\">\n",
      "        <mml:mrow>\n",
      "         <mml:mi>\n",
      "          P\n",
      "         </mml:mi>\n",
      "         <mml:mrow>\n",
      "          <mml:mo>\n",
      "           (\n",
      "          </mml:mo>\n",
      "          <mml:mrow>\n",
      "           <mml:mover accent=\"true\">\n",
      "            <mml:mi>\n",
      "             x\n",
      "            </mml:mi>\n",
      "            <mml:mo>\n",
      "             →\n",
      "            </mml:mo>\n",
      "           </mml:mover>\n",
      "           <mml:mo>\n",
      "            ,\n",
      "           </mml:mo>\n",
      "           <mml:mi>\n",
      "            t\n",
      "           </mml:mi>\n",
      "          </mml:mrow>\n",
      "          <mml:mo>\n",
      "           )\n",
      "          </mml:mo>\n",
      "         </mml:mrow>\n",
      "        </mml:mrow>\n",
      "       </mml:math>\n",
      "      </alternatives>\n",
      "     </inline-formula>\n",
      "     , is then estimated in two parts.\n",
      "    </p>\n",
      "    <disp-formula id=\"pone.0135749.e004\">\n",
      "     <alternatives>\n",
      "      <graphic id=\"pone.0135749.e004g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e004.jpg\">\n",
      "      </graphic>\n",
      "      <mml:math id=\"M4\">\n",
      "       <mml:mrow>\n",
      "        <mml:mi>\n",
      "         P\n",
      "        </mml:mi>\n",
      "        <mml:mrow>\n",
      "         <mml:mo>\n",
      "          (\n",
      "         </mml:mo>\n",
      "         <mml:mrow>\n",
      "          <mml:mover accent=\"true\">\n",
      "           <mml:mi>\n",
      "            x\n",
      "           </mml:mi>\n",
      "           <mml:mo>\n",
      "            →\n",
      "           </mml:mo>\n",
      "          </mml:mover>\n",
      "          <mml:mo>\n",
      "           ,\n",
      "          </mml:mo>\n",
      "          <mml:mi>\n",
      "           t\n",
      "          </mml:mi>\n",
      "         </mml:mrow>\n",
      "         <mml:mo>\n",
      "          )\n",
      "         </mml:mo>\n",
      "        </mml:mrow>\n",
      "        <mml:mo>\n",
      "         =\n",
      "        </mml:mo>\n",
      "        <mml:mi>\n",
      "         S\n",
      "        </mml:mi>\n",
      "        <mml:mrow>\n",
      "         <mml:mo>\n",
      "          (\n",
      "         </mml:mo>\n",
      "         <mml:mover accent=\"true\">\n",
      "          <mml:mi>\n",
      "           x\n",
      "          </mml:mi>\n",
      "          <mml:mo>\n",
      "           →\n",
      "          </mml:mo>\n",
      "         </mml:mover>\n",
      "         <mml:mo>\n",
      "          )\n",
      "         </mml:mo>\n",
      "        </mml:mrow>\n",
      "        <mml:mo>\n",
      "         +\n",
      "        </mml:mo>\n",
      "        <mml:mi>\n",
      "         A\n",
      "        </mml:mi>\n",
      "        <mml:mrow>\n",
      "         <mml:mo>\n",
      "          (\n",
      "         </mml:mo>\n",
      "         <mml:mrow>\n",
      "          <mml:mover accent=\"true\">\n",
      "           <mml:mi>\n",
      "            x\n",
      "           </mml:mi>\n",
      "           <mml:mo>\n",
      "            →\n",
      "           </mml:mo>\n",
      "          </mml:mover>\n",
      "          <mml:mo>\n",
      "           ,\n",
      "          </mml:mo>\n",
      "          <mml:mi>\n",
      "           t\n",
      "          </mml:mi>\n",
      "         </mml:mrow>\n",
      "         <mml:mo>\n",
      "          )\n",
      "         </mml:mo>\n",
      "        </mml:mrow>\n",
      "       </mml:mrow>\n",
      "      </mml:math>\n",
      "     </alternatives>\n",
      "    </disp-formula>\n",
      "    <disp-formula id=\"pone.0135749.e005\">\n",
      "     <alternatives>\n",
      "      <graphic id=\"pone.0135749.e005g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e005.jpg\">\n",
      "      </graphic>\n",
      "      <mml:math id=\"M5\">\n",
      "       <mml:mrow>\n",
      "        <mml:mi>\n",
      "         S\n",
      "        </mml:mi>\n",
      "        <mml:mrow>\n",
      "         <mml:mo>\n",
      "          (\n",
      "         </mml:mo>\n",
      "         <mml:mover accent=\"true\">\n",
      "          <mml:mi>\n",
      "           x\n",
      "          </mml:mi>\n",
      "          <mml:mo>\n",
      "           →\n",
      "          </mml:mo>\n",
      "         </mml:mover>\n",
      "         <mml:mo>\n",
      "          )\n",
      "         </mml:mo>\n",
      "        </mml:mrow>\n",
      "        <mml:mo>\n",
      "         =\n",
      "        </mml:mo>\n",
      "        <mml:mrow>\n",
      "         <mml:mo>\n",
      "          (\n",
      "         </mml:mo>\n",
      "         <mml:mrow>\n",
      "          <mml:mstyle displaystyle=\"true\">\n",
      "           <mml:munder>\n",
      "            <mml:mo>\n",
      "             ∑\n",
      "            </mml:mo>\n",
      "            <mml:mi>\n",
      "             n\n",
      "            </mml:mi>\n",
      "           </mml:munder>\n",
      "           <mml:mrow>\n",
      "            <mml:msub>\n",
      "             <mml:mi>\n",
      "              K\n",
      "             </mml:mi>\n",
      "             <mml:mi>\n",
      "              n\n",
      "             </mml:mi>\n",
      "            </mml:msub>\n",
      "            <mml:mrow>\n",
      "             <mml:mo>\n",
      "              (\n",
      "             </mml:mo>\n",
      "             <mml:mover accent=\"true\">\n",
      "              <mml:mi>\n",
      "               x\n",
      "              </mml:mi>\n",
      "              <mml:mo>\n",
      "               →\n",
      "              </mml:mo>\n",
      "             </mml:mover>\n",
      "             <mml:mo>\n",
      "              )\n",
      "             </mml:mo>\n",
      "            </mml:mrow>\n",
      "            <mml:mrow>\n",
      "             <mml:mo>\n",
      "              (\n",
      "             </mml:mo>\n",
      "             <mml:mrow>\n",
      "              <mml:mover accent=\"true\">\n",
      "               <mml:mrow>\n",
      "                <mml:msub>\n",
      "                 <mml:mi>\n",
      "                  p\n",
      "                 </mml:mi>\n",
      "                 <mml:mi>\n",
      "                  n\n",
      "                 </mml:mi>\n",
      "                </mml:msub>\n",
      "               </mml:mrow>\n",
      "               <mml:mo stretchy=\"true\">\n",
      "                ¯\n",
      "               </mml:mo>\n",
      "              </mml:mover>\n",
      "              <mml:mo>\n",
      "               −\n",
      "              </mml:mo>\n",
      "              <mml:mi>\n",
      "               G\n",
      "              </mml:mi>\n",
      "              <mml:mrow>\n",
      "               <mml:mo stretchy=\"false\">\n",
      "                (\n",
      "               </mml:mo>\n",
      "               <mml:mrow>\n",
      "                <mml:msub>\n",
      "                 <mml:mover accent=\"true\">\n",
      "                  <mml:mi>\n",
      "                   x\n",
      "                  </mml:mi>\n",
      "                  <mml:mo>\n",
      "                   →\n",
      "                  </mml:mo>\n",
      "                 </mml:mover>\n",
      "                 <mml:mi>\n",
      "                  n\n",
      "                 </mml:mi>\n",
      "                </mml:msub>\n",
      "               </mml:mrow>\n",
      "               <mml:mo stretchy=\"false\">\n",
      "                )\n",
      "               </mml:mo>\n",
      "              </mml:mrow>\n",
      "             </mml:mrow>\n",
      "             <mml:mo>\n",
      "              )\n",
      "             </mml:mo>\n",
      "            </mml:mrow>\n",
      "           </mml:mrow>\n",
      "          </mml:mstyle>\n",
      "         </mml:mrow>\n",
      "         <mml:mo>\n",
      "          )\n",
      "         </mml:mo>\n",
      "        </mml:mrow>\n",
      "        <mml:mo>\n",
      "         +\n",
      "        </mml:mo>\n",
      "        <mml:mi>\n",
      "         G\n",
      "        </mml:mi>\n",
      "        <mml:mrow>\n",
      "         <mml:mo>\n",
      "          (\n",
      "         </mml:mo>\n",
      "         <mml:mover accent=\"true\">\n",
      "          <mml:mi>\n",
      "           x\n",
      "          </mml:mi>\n",
      "          <mml:mo>\n",
      "           →\n",
      "          </mml:mo>\n",
      "         </mml:mover>\n",
      "         <mml:mo>\n",
      "          )\n",
      "         </mml:mo>\n",
      "        </mml:mrow>\n",
      "       </mml:mrow>\n",
      "      </mml:math>\n",
      "     </alternatives>\n",
      "    </disp-formula>\n",
      "    <disp-formula id=\"pone.0135749.e006\">\n",
      "     <alternatives>\n",
      "      <graphic id=\"pone.0135749.e006g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e006.jpg\">\n",
      "      </graphic>\n",
      "      <mml:math id=\"M6\">\n",
      "       <mml:mrow>\n",
      "        <mml:mi>\n",
      "         A\n",
      "        </mml:mi>\n",
      "        <mml:mrow>\n",
      "         <mml:mo>\n",
      "          (\n",
      "         </mml:mo>\n",
      "         <mml:mrow>\n",
      "          <mml:mover accent=\"true\">\n",
      "           <mml:mi>\n",
      "            x\n",
      "           </mml:mi>\n",
      "           <mml:mo>\n",
      "            →\n",
      "           </mml:mo>\n",
      "          </mml:mover>\n",
      "          <mml:mo>\n",
      "           ,\n",
      "          </mml:mo>\n",
      "          <mml:mi>\n",
      "           t\n",
      "          </mml:mi>\n",
      "         </mml:mrow>\n",
      "         <mml:mo>\n",
      "          )\n",
      "         </mml:mo>\n",
      "        </mml:mrow>\n",
      "        <mml:mo>\n",
      "         =\n",
      "        </mml:mo>\n",
      "        <mml:mstyle displaystyle=\"true\">\n",
      "         <mml:munder>\n",
      "          <mml:mo>\n",
      "           ∑\n",
      "          </mml:mo>\n",
      "          <mml:mi>\n",
      "           n\n",
      "          </mml:mi>\n",
      "         </mml:munder>\n",
      "         <mml:mrow>\n",
      "          <mml:msubsup>\n",
      "           <mml:mi>\n",
      "            K\n",
      "           </mml:mi>\n",
      "           <mml:mi>\n",
      "            n\n",
      "           </mml:mi>\n",
      "           <mml:mo>\n",
      "            *\n",
      "           </mml:mo>\n",
      "          </mml:msubsup>\n",
      "          <mml:mrow>\n",
      "           <mml:mo>\n",
      "            (\n",
      "           </mml:mo>\n",
      "           <mml:mrow>\n",
      "            <mml:mover accent=\"true\">\n",
      "             <mml:mi>\n",
      "              x\n",
      "             </mml:mi>\n",
      "             <mml:mo>\n",
      "              →\n",
      "             </mml:mo>\n",
      "            </mml:mover>\n",
      "            <mml:mo>\n",
      "             ,\n",
      "            </mml:mo>\n",
      "            <mml:mi>\n",
      "             t\n",
      "            </mml:mi>\n",
      "           </mml:mrow>\n",
      "           <mml:mo>\n",
      "            )\n",
      "           </mml:mo>\n",
      "          </mml:mrow>\n",
      "          <mml:mrow>\n",
      "           <mml:mo>\n",
      "            (\n",
      "           </mml:mo>\n",
      "           <mml:mrow>\n",
      "            <mml:msub>\n",
      "             <mml:mi>\n",
      "              p\n",
      "             </mml:mi>\n",
      "             <mml:mi>\n",
      "              n\n",
      "             </mml:mi>\n",
      "            </mml:msub>\n",
      "            <mml:mrow>\n",
      "             <mml:mo>\n",
      "              (\n",
      "             </mml:mo>\n",
      "             <mml:mi>\n",
      "              t\n",
      "             </mml:mi>\n",
      "             <mml:mo>\n",
      "              )\n",
      "             </mml:mo>\n",
      "            </mml:mrow>\n",
      "            <mml:mo>\n",
      "             −\n",
      "            </mml:mo>\n",
      "            <mml:mover accent=\"true\">\n",
      "             <mml:mrow>\n",
      "              <mml:msub>\n",
      "               <mml:mi>\n",
      "                p\n",
      "               </mml:mi>\n",
      "               <mml:mi>\n",
      "                n\n",
      "               </mml:mi>\n",
      "              </mml:msub>\n",
      "             </mml:mrow>\n",
      "             <mml:mo stretchy=\"true\">\n",
      "              ¯\n",
      "             </mml:mo>\n",
      "            </mml:mover>\n",
      "           </mml:mrow>\n",
      "           <mml:mo>\n",
      "            )\n",
      "           </mml:mo>\n",
      "          </mml:mrow>\n",
      "         </mml:mrow>\n",
      "        </mml:mstyle>\n",
      "       </mml:mrow>\n",
      "      </mml:math>\n",
      "     </alternatives>\n",
      "    </disp-formula>\n",
      "    <p>\n",
      "     The stationary part,\n",
      "     <inline-formula id=\"pone.0135749.e007\">\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone.0135749.e007g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e007.jpg\">\n",
      "       </graphic>\n",
      "       <mml:math id=\"M7\">\n",
      "        <mml:mrow>\n",
      "         <mml:mi>\n",
      "          S\n",
      "         </mml:mi>\n",
      "         <mml:mrow>\n",
      "          <mml:mo>\n",
      "           (\n",
      "          </mml:mo>\n",
      "          <mml:mover accent=\"true\">\n",
      "           <mml:mi>\n",
      "            x\n",
      "           </mml:mi>\n",
      "           <mml:mo>\n",
      "            →\n",
      "           </mml:mo>\n",
      "          </mml:mover>\n",
      "          <mml:mo>\n",
      "           )\n",
      "          </mml:mo>\n",
      "         </mml:mrow>\n",
      "        </mml:mrow>\n",
      "       </mml:math>\n",
      "      </alternatives>\n",
      "     </inline-formula>\n",
      "     , is derived by applying Kriging interpolation to the mean pollutant concentrations and a global predictor,\n",
      "     <inline-formula id=\"pone.0135749.e008\">\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone.0135749.e008g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e008.jpg\">\n",
      "       </graphic>\n",
      "       <mml:math id=\"M8\">\n",
      "        <mml:mrow>\n",
      "         <mml:mi>\n",
      "          G\n",
      "         </mml:mi>\n",
      "         <mml:mrow>\n",
      "          <mml:mo>\n",
      "           (\n",
      "          </mml:mo>\n",
      "          <mml:mrow>\n",
      "           <mml:msub>\n",
      "            <mml:mover accent=\"true\">\n",
      "             <mml:mi>\n",
      "              x\n",
      "             </mml:mi>\n",
      "             <mml:mo>\n",
      "              →\n",
      "             </mml:mo>\n",
      "            </mml:mover>\n",
      "            <mml:mi>\n",
      "             n\n",
      "            </mml:mi>\n",
      "           </mml:msub>\n",
      "          </mml:mrow>\n",
      "          <mml:mo>\n",
      "           )\n",
      "          </mml:mo>\n",
      "         </mml:mrow>\n",
      "        </mml:mrow>\n",
      "       </mml:math>\n",
      "      </alternatives>\n",
      "     </inline-formula>\n",
      "     , that depends on latitude and longitude and contains free parameters that are adjusted to fit the observed means. The time-dependent anomaly part,\n",
      "     <inline-formula id=\"pone.0135749.e009\">\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone.0135749.e009g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e009.jpg\">\n",
      "       </graphic>\n",
      "       <mml:math id=\"M9\">\n",
      "        <mml:mrow>\n",
      "         <mml:mi>\n",
      "          A\n",
      "         </mml:mi>\n",
      "         <mml:mrow>\n",
      "          <mml:mo>\n",
      "           (\n",
      "          </mml:mo>\n",
      "          <mml:mrow>\n",
      "           <mml:mover accent=\"true\">\n",
      "            <mml:mi>\n",
      "             x\n",
      "            </mml:mi>\n",
      "            <mml:mo>\n",
      "             →\n",
      "            </mml:mo>\n",
      "           </mml:mover>\n",
      "           <mml:mo>\n",
      "            ,\n",
      "           </mml:mo>\n",
      "           <mml:mi>\n",
      "            t\n",
      "           </mml:mi>\n",
      "          </mml:mrow>\n",
      "          <mml:mo>\n",
      "           )\n",
      "          </mml:mo>\n",
      "         </mml:mrow>\n",
      "        </mml:mrow>\n",
      "       </mml:math>\n",
      "      </alternatives>\n",
      "     </inline-formula>\n",
      "     , depends only on the fluctuations at each station relative to the local mean, and its Kriging coefficients,\n",
      "     <inline-formula id=\"pone.0135749.e010\">\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone.0135749.e010g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e010.jpg\">\n",
      "       </graphic>\n",
      "       <mml:math id=\"M10\">\n",
      "        <mml:mrow>\n",
      "         <mml:msubsup>\n",
      "          <mml:mi>\n",
      "           K\n",
      "          </mml:mi>\n",
      "          <mml:mi>\n",
      "           n\n",
      "          </mml:mi>\n",
      "          <mml:mo>\n",
      "           *\n",
      "          </mml:mo>\n",
      "         </mml:msubsup>\n",
      "         <mml:mrow>\n",
      "          <mml:mo>\n",
      "           (\n",
      "          </mml:mo>\n",
      "          <mml:mrow>\n",
      "           <mml:msub>\n",
      "            <mml:mover accent=\"true\">\n",
      "             <mml:mi>\n",
      "              x\n",
      "             </mml:mi>\n",
      "             <mml:mo>\n",
      "              →\n",
      "             </mml:mo>\n",
      "            </mml:mover>\n",
      "            <mml:mi>\n",
      "             n\n",
      "            </mml:mi>\n",
      "           </mml:msub>\n",
      "           <mml:mo>\n",
      "            ,\n",
      "           </mml:mo>\n",
      "           <mml:mi>\n",
      "            t\n",
      "           </mml:mi>\n",
      "          </mml:mrow>\n",
      "          <mml:mo>\n",
      "           )\n",
      "          </mml:mo>\n",
      "         </mml:mrow>\n",
      "        </mml:mrow>\n",
      "       </mml:math>\n",
      "      </alternatives>\n",
      "     </inline-formula>\n",
      "     , are computed with restriction to stations that are active at time\n",
      "     <italic>\n",
      "      t\n",
      "     </italic>\n",
      "     . This two-step process reduces errors associated with stations that have intermittently missing data. This method is similar to that used by Berkeley Earth for its historical earth temperature analysis [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref016\">\n",
      "      16\n",
      "     </xref>\n",
      "     ]. Since the correlation vs. distance function has been constructed with the correlation at zero distance obtaining a value less than one, the resulting interpolated fields will be smoother than the original data. This design was chosen for its ability to compensate for noise in the underlying station measurements. Additional details of the interpolation process are provided in the supplement methods (\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s001\">\n",
      "      S1 Text\n",
      "     </xref>\n",
      "     ).\n",
      "    </p>\n",
      "    <p>\n",
      "     For mapping and computation, this continuous field was sampled with an approximately 6 km resolution, though in practice, the characteristic size of resolvable features is often larger (e.g. 30 km) and varies with station density and noise.\n",
      "    </p>\n",
      "    <p>\n",
      "     A simple estimate of pollutant fluxes,\n",
      "     <inline-formula id=\"pone.0135749.e011\">\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone.0135749.e011g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e011.jpg\">\n",
      "       </graphic>\n",
      "       <mml:math id=\"M11\">\n",
      "        <mml:mrow>\n",
      "         <mml:mi>\n",
      "          F\n",
      "         </mml:mi>\n",
      "         <mml:mrow>\n",
      "          <mml:mo>\n",
      "           (\n",
      "          </mml:mo>\n",
      "          <mml:mrow>\n",
      "           <mml:mover accent=\"true\">\n",
      "            <mml:mi>\n",
      "             x\n",
      "            </mml:mi>\n",
      "            <mml:mo>\n",
      "             →\n",
      "            </mml:mo>\n",
      "           </mml:mover>\n",
      "           <mml:mo>\n",
      "            ,\n",
      "           </mml:mo>\n",
      "           <mml:mi>\n",
      "            t\n",
      "           </mml:mi>\n",
      "          </mml:mrow>\n",
      "          <mml:mo>\n",
      "           )\n",
      "          </mml:mo>\n",
      "         </mml:mrow>\n",
      "        </mml:mrow>\n",
      "       </mml:math>\n",
      "      </alternatives>\n",
      "     </inline-formula>\n",
      "     , was computed by comparing observed changes in the hourly pollutant concentration to the concentrations expected due to short-term wind transport\n",
      "     <inline-formula id=\"pone.0135749.e012\">\n",
      "      <alternatives>\n",
      "       <graphic id=\"pone.0135749.e012g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e012.jpg\">\n",
      "       </graphic>\n",
      "       <mml:math id=\"M12\">\n",
      "        <mml:mrow>\n",
      "         <mml:mover accent=\"true\">\n",
      "          <mml:mi>\n",
      "           v\n",
      "          </mml:mi>\n",
      "          <mml:mo>\n",
      "           →\n",
      "          </mml:mo>\n",
      "         </mml:mover>\n",
      "         <mml:mrow>\n",
      "          <mml:mo>\n",
      "           (\n",
      "          </mml:mo>\n",
      "          <mml:mrow>\n",
      "           <mml:mover accent=\"true\">\n",
      "            <mml:mi>\n",
      "             x\n",
      "            </mml:mi>\n",
      "            <mml:mo>\n",
      "             →\n",
      "            </mml:mo>\n",
      "           </mml:mover>\n",
      "           <mml:mo>\n",
      "            ,\n",
      "           </mml:mo>\n",
      "           <mml:mi>\n",
      "            t\n",
      "           </mml:mi>\n",
      "          </mml:mrow>\n",
      "          <mml:mo>\n",
      "           )\n",
      "          </mml:mo>\n",
      "         </mml:mrow>\n",
      "        </mml:mrow>\n",
      "       </mml:math>\n",
      "      </alternatives>\n",
      "     </inline-formula>\n",
      "     and an exponential decay with lifetime\n",
      "     <italic>\n",
      "      τ\n",
      "     </italic>\n",
      "     . Differences from the simple transport and decay model are assumed to represent source fluxes.\n",
      "    </p>\n",
      "    <disp-formula id=\"pone.0135749.e013\">\n",
      "     <alternatives>\n",
      "      <graphic id=\"pone.0135749.e013g\" mimetype=\"image\" orientation=\"portrait\" position=\"anchor\" xlink:href=\"pone.0135749.e013.jpg\">\n",
      "      </graphic>\n",
      "      <mml:math id=\"M13\">\n",
      "       <mml:mrow>\n",
      "        <mml:mi>\n",
      "         F\n",
      "        </mml:mi>\n",
      "        <mml:mrow>\n",
      "         <mml:mo>\n",
      "          (\n",
      "         </mml:mo>\n",
      "         <mml:mrow>\n",
      "          <mml:mover accent=\"true\">\n",
      "           <mml:mi>\n",
      "            x\n",
      "           </mml:mi>\n",
      "           <mml:mo>\n",
      "            →\n",
      "           </mml:mo>\n",
      "          </mml:mover>\n",
      "          <mml:mo>\n",
      "           ,\n",
      "          </mml:mo>\n",
      "          <mml:mi>\n",
      "           t\n",
      "          </mml:mi>\n",
      "         </mml:mrow>\n",
      "         <mml:mo>\n",
      "          )\n",
      "         </mml:mo>\n",
      "        </mml:mrow>\n",
      "        <mml:mo>\n",
      "         =\n",
      "        </mml:mo>\n",
      "        <mml:mfrac>\n",
      "         <mml:mrow>\n",
      "          <mml:mi>\n",
      "           P\n",
      "          </mml:mi>\n",
      "          <mml:mrow>\n",
      "           <mml:mo>\n",
      "            (\n",
      "           </mml:mo>\n",
      "           <mml:mrow>\n",
      "            <mml:mover accent=\"true\">\n",
      "             <mml:mi>\n",
      "              x\n",
      "             </mml:mi>\n",
      "             <mml:mo>\n",
      "              →\n",
      "             </mml:mo>\n",
      "            </mml:mover>\n",
      "            <mml:mo>\n",
      "             +\n",
      "            </mml:mo>\n",
      "            <mml:mover accent=\"true\">\n",
      "             <mml:mi>\n",
      "              v\n",
      "             </mml:mi>\n",
      "             <mml:mo>\n",
      "              →\n",
      "             </mml:mo>\n",
      "            </mml:mover>\n",
      "            <mml:mrow>\n",
      "             <mml:mo>\n",
      "              (\n",
      "             </mml:mo>\n",
      "             <mml:mrow>\n",
      "              <mml:mover accent=\"true\">\n",
      "               <mml:mi>\n",
      "                x\n",
      "               </mml:mi>\n",
      "               <mml:mo>\n",
      "                →\n",
      "               </mml:mo>\n",
      "              </mml:mover>\n",
      "              <mml:mo>\n",
      "               ,\n",
      "              </mml:mo>\n",
      "              <mml:mi>\n",
      "               t\n",
      "              </mml:mi>\n",
      "             </mml:mrow>\n",
      "             <mml:mo>\n",
      "              )\n",
      "             </mml:mo>\n",
      "            </mml:mrow>\n",
      "            <mml:mspace width=\"0.2em\">\n",
      "            </mml:mspace>\n",
      "            <mml:mi mathvariant=\"normal\">\n",
      "             Δ\n",
      "            </mml:mi>\n",
      "            <mml:mi>\n",
      "             t\n",
      "            </mml:mi>\n",
      "            <mml:mo>\n",
      "             ,\n",
      "            </mml:mo>\n",
      "            <mml:mi>\n",
      "             t\n",
      "            </mml:mi>\n",
      "            <mml:mo>\n",
      "             +\n",
      "            </mml:mo>\n",
      "            <mml:mi mathvariant=\"normal\">\n",
      "             Δ\n",
      "            </mml:mi>\n",
      "            <mml:mi>\n",
      "             t\n",
      "            </mml:mi>\n",
      "           </mml:mrow>\n",
      "           <mml:mo>\n",
      "            )\n",
      "           </mml:mo>\n",
      "          </mml:mrow>\n",
      "          <mml:mo>\n",
      "           −\n",
      "          </mml:mo>\n",
      "          <mml:msup>\n",
      "           <mml:mi>\n",
      "            e\n",
      "           </mml:mi>\n",
      "           <mml:mrow>\n",
      "            <mml:mo>\n",
      "             −\n",
      "            </mml:mo>\n",
      "            <mml:mrow>\n",
      "             <mml:mrow>\n",
      "              <mml:mi mathvariant=\"normal\">\n",
      "               Δ\n",
      "              </mml:mi>\n",
      "              <mml:mi>\n",
      "               t\n",
      "              </mml:mi>\n",
      "             </mml:mrow>\n",
      "             <mml:mo>\n",
      "              /\n",
      "             </mml:mo>\n",
      "             <mml:mi>\n",
      "              τ\n",
      "             </mml:mi>\n",
      "            </mml:mrow>\n",
      "           </mml:mrow>\n",
      "          </mml:msup>\n",
      "          <mml:mi>\n",
      "           P\n",
      "          </mml:mi>\n",
      "          <mml:mrow>\n",
      "           <mml:mo>\n",
      "            (\n",
      "           </mml:mo>\n",
      "           <mml:mrow>\n",
      "            <mml:mover accent=\"true\">\n",
      "             <mml:mi>\n",
      "              x\n",
      "             </mml:mi>\n",
      "             <mml:mo>\n",
      "              →\n",
      "             </mml:mo>\n",
      "            </mml:mover>\n",
      "            <mml:mo>\n",
      "             ,\n",
      "            </mml:mo>\n",
      "            <mml:mi>\n",
      "             t\n",
      "            </mml:mi>\n",
      "           </mml:mrow>\n",
      "           <mml:mo>\n",
      "            )\n",
      "           </mml:mo>\n",
      "          </mml:mrow>\n",
      "         </mml:mrow>\n",
      "         <mml:mrow>\n",
      "          <mml:mn>\n",
      "           2\n",
      "          </mml:mn>\n",
      "          <mml:mi mathvariant=\"normal\">\n",
      "           Δ\n",
      "          </mml:mi>\n",
      "          <mml:mi>\n",
      "           t\n",
      "          </mml:mi>\n",
      "         </mml:mrow>\n",
      "        </mml:mfrac>\n",
      "        <mml:mo>\n",
      "         +\n",
      "        </mml:mo>\n",
      "        <mml:mfrac>\n",
      "         <mml:mrow>\n",
      "          <mml:mi>\n",
      "           P\n",
      "          </mml:mi>\n",
      "          <mml:mrow>\n",
      "           <mml:mo>\n",
      "            (\n",
      "           </mml:mo>\n",
      "           <mml:mrow>\n",
      "            <mml:mover accent=\"true\">\n",
      "             <mml:mi>\n",
      "              x\n",
      "             </mml:mi>\n",
      "             <mml:mo>\n",
      "              →\n",
      "             </mml:mo>\n",
      "            </mml:mover>\n",
      "            <mml:mo>\n",
      "             ,\n",
      "            </mml:mo>\n",
      "            <mml:mi>\n",
      "             t\n",
      "            </mml:mi>\n",
      "           </mml:mrow>\n",
      "           <mml:mo>\n",
      "            )\n",
      "           </mml:mo>\n",
      "          </mml:mrow>\n",
      "          <mml:mo>\n",
      "           −\n",
      "          </mml:mo>\n",
      "          <mml:msup>\n",
      "           <mml:mi>\n",
      "            e\n",
      "           </mml:mi>\n",
      "           <mml:mrow>\n",
      "            <mml:mo>\n",
      "             −\n",
      "            </mml:mo>\n",
      "            <mml:mrow>\n",
      "             <mml:mrow>\n",
      "              <mml:mi mathvariant=\"normal\">\n",
      "               Δ\n",
      "              </mml:mi>\n",
      "              <mml:mi>\n",
      "               t\n",
      "              </mml:mi>\n",
      "             </mml:mrow>\n",
      "             <mml:mo>\n",
      "              /\n",
      "             </mml:mo>\n",
      "             <mml:mi>\n",
      "              τ\n",
      "             </mml:mi>\n",
      "            </mml:mrow>\n",
      "           </mml:mrow>\n",
      "          </mml:msup>\n",
      "          <mml:mi>\n",
      "           P\n",
      "          </mml:mi>\n",
      "          <mml:mrow>\n",
      "           <mml:mo>\n",
      "            (\n",
      "           </mml:mo>\n",
      "           <mml:mrow>\n",
      "            <mml:mover accent=\"true\">\n",
      "             <mml:mi>\n",
      "              x\n",
      "             </mml:mi>\n",
      "             <mml:mo>\n",
      "              →\n",
      "             </mml:mo>\n",
      "            </mml:mover>\n",
      "            <mml:mo>\n",
      "             −\n",
      "            </mml:mo>\n",
      "            <mml:mover accent=\"true\">\n",
      "             <mml:mi>\n",
      "              v\n",
      "             </mml:mi>\n",
      "             <mml:mo>\n",
      "              →\n",
      "             </mml:mo>\n",
      "            </mml:mover>\n",
      "            <mml:mrow>\n",
      "             <mml:mo>\n",
      "              (\n",
      "             </mml:mo>\n",
      "             <mml:mrow>\n",
      "              <mml:mover accent=\"true\">\n",
      "               <mml:mi>\n",
      "                x\n",
      "               </mml:mi>\n",
      "               <mml:mo>\n",
      "                →\n",
      "               </mml:mo>\n",
      "              </mml:mover>\n",
      "              <mml:mo>\n",
      "               ,\n",
      "              </mml:mo>\n",
      "              <mml:mi>\n",
      "               t\n",
      "              </mml:mi>\n",
      "             </mml:mrow>\n",
      "             <mml:mo>\n",
      "              )\n",
      "             </mml:mo>\n",
      "            </mml:mrow>\n",
      "            <mml:mspace width=\"0.2em\">\n",
      "            </mml:mspace>\n",
      "            <mml:mi mathvariant=\"normal\">\n",
      "             Δ\n",
      "            </mml:mi>\n",
      "            <mml:mi>\n",
      "             t\n",
      "            </mml:mi>\n",
      "            <mml:mo>\n",
      "             ,\n",
      "            </mml:mo>\n",
      "            <mml:mi>\n",
      "             t\n",
      "            </mml:mi>\n",
      "            <mml:mo>\n",
      "             −\n",
      "            </mml:mo>\n",
      "            <mml:mi mathvariant=\"normal\">\n",
      "             Δ\n",
      "            </mml:mi>\n",
      "            <mml:mi>\n",
      "             t\n",
      "            </mml:mi>\n",
      "           </mml:mrow>\n",
      "           <mml:mo>\n",
      "            )\n",
      "           </mml:mo>\n",
      "          </mml:mrow>\n",
      "         </mml:mrow>\n",
      "         <mml:mrow>\n",
      "          <mml:mn>\n",
      "           2\n",
      "          </mml:mn>\n",
      "          <mml:mi mathvariant=\"normal\">\n",
      "           Δ\n",
      "          </mml:mi>\n",
      "          <mml:mi>\n",
      "           t\n",
      "          </mml:mi>\n",
      "         </mml:mrow>\n",
      "        </mml:mfrac>\n",
      "       </mml:mrow>\n",
      "      </mml:math>\n",
      "     </alternatives>\n",
      "    </disp-formula>\n",
      "    <p>\n",
      "     The near-surface (80 m) wind field from the Global Forecast System [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref017\">\n",
      "      17\n",
      "     </xref>\n",
      "     ] was used for this calculation, and the effective pollutant lifetime was estimated as described in\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s001\">\n",
      "      S1 Text\n",
      "     </xref>\n",
      "     and reported in\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s003\">\n",
      "      S2 Table\n",
      "     </xref>\n",
      "     . Flux averages were computed by time-averaging the resulting field after excluding outlying values and cells affected by rain events as determined from Tropical Rainfall Measuring Mission data [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref018\">\n",
      "      18\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref019\">\n",
      "      19\n",
      "     </xref>\n",
      "     ].\n",
      "    </p>\n",
      "    <p>\n",
      "     The change in mortality due to PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     air pollution was calculated by adopting the integrated exposure response function approach [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref020\">\n",
      "      20\n",
      "     </xref>\n",
      "     ] which considers relative risk of death for five disease classes (stroke, ischemic heart disease, lung cancer, chronic obstructive pulmonary disease, and lower respiratory infection) and which was adopted by World Health Organization (WHO) for the Global Burden of Disease study [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref021\">\n",
      "      21\n",
      "     </xref>\n",
      "     ]. The model incorporates non-linear response versus concentration and provides an estimate of uncertainty. Relative risk was calculated at the prefecture level using local average PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     concentration. The data for different diseases and prefectures was then combined to construct national average mortality estimates.\n",
      "    </p>\n",
      "    <p>\n",
      "     Analysis and figure rendering was performed using original software written for this project on the MATLAB platform (version 2014a;\n",
      "     <ext-link ext-link-type=\"uri\" xlink:href=\"http://www.mathworks.com/\">\n",
      "      http://www.mathworks.com/\n",
      "     </ext-link>\n",
      "     ). Additional details of these calculations and associated background information is provided in the supplemental methods document (\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s001\">\n",
      "      S1 Text\n",
      "     </xref>\n",
      "     ).\n",
      "    </p>\n",
      "   </sec>\n",
      "   <sec id=\"sec003\" sec-type=\"results\">\n",
      "    <title>\n",
      "     Results\n",
      "    </title>\n",
      "    <p>\n",
      "     <xref ref-type=\"fig\" rid=\"pone.0135749.g002\">\n",
      "      Fig 2\n",
      "     </xref>\n",
      "     shows a time series of PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     concentration at Beijing and interpolated maps at three time points separated by 6 hours each. This shows the volatile nature of air pollution and the role of weather patterns in redistributing pollution on short timescales. Our approach creates a smooth field that approximates the data at each station, but allows a degree of difference attributable to noise. The pollution is extensive and rapidly evolves in response to winds and other atmospheric conditions. In the figure, fresh air from the North displaces a period of heavy pollution. Hourly data allows us to capture this evolution and ultimately estimate source fluxes.\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s004\">\n",
      "      S1 Movie\n",
      "     </xref>\n",
      "     shows the time evolution of PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     across the entire country.\n",
      "    </p>\n",
      "    <fig id=\"pone.0135749.g002\" orientation=\"portrait\" position=\"float\">\n",
      "     <object-id pub-id-type=\"doi\">\n",
      "      10.1371/journal.pone.0135749.g002\n",
      "     </object-id>\n",
      "     <label>\n",
      "      Fig 2\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Time evolution of PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       pollution in the vicinity of Beijing.\n",
      "      </title>\n",
      "      <p>\n",
      "       (Top) Time series of PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       concentration at Beijing extracted from the interpolated field. Red circles indicate times shown in bottom row. (Bottom) Maps of interpolated PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       concentration during a period of high pollution. Pollution concentrations were computed as described in the text from hourly data and maps were rendered in MATLAB. Concentrations are shown using color gradients and contour lines, where color tones (green, yellow, etc.) correspond to health impact categories defined by the US EPA. Bold circles show station locations with the observed value at each station indicated by the color within the circle.\n",
      "      </p>\n",
      "     </caption>\n",
      "     <graphic xlink:href=\"pone.0135749.g002\">\n",
      "     </graphic>\n",
      "    </fig>\n",
      "    <p>\n",
      "     <xref ref-type=\"fig\" rid=\"pone.0135749.g003\">\n",
      "      Fig 3\n",
      "     </xref>\n",
      "     presents averages of the interpolated data for PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     , PM\n",
      "     <sub>\n",
      "      10\n",
      "     </sub>\n",
      "     , and O\n",
      "     <sub>\n",
      "      3\n",
      "     </sub>\n",
      "     across the study duration. The maps are color-coded based on US EPA health categories for 24-hour exposure [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref022\">\n",
      "      22\n",
      "     </xref>\n",
      "     ]. Maps for SO\n",
      "     <sub>\n",
      "      2\n",
      "     </sub>\n",
      "     and NO\n",
      "     <sub>\n",
      "      2\n",
      "     </sub>\n",
      "     are included in the supplemental materials and show “good” levels nearly everywhere (\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s015\">\n",
      "      S11\n",
      "     </xref>\n",
      "     and\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s017\">\n",
      "      S13\n",
      "     </xref>\n",
      "     Figs).\n",
      "    </p>\n",
      "    <fig id=\"pone.0135749.g003\" orientation=\"portrait\" position=\"float\">\n",
      "     <object-id pub-id-type=\"doi\">\n",
      "      10.1371/journal.pone.0135749.g003\n",
      "     </object-id>\n",
      "     <label>\n",
      "      Fig 3\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Average air pollution maps.\n",
      "      </title>\n",
      "      <p>\n",
      "       Maps of average pollutant concentration for PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       , PM\n",
      "       <sub>\n",
      "        10\n",
      "       </sub>\n",
      "       , and O\n",
      "       <sub>\n",
      "        3\n",
      "       </sub>\n",
      "       for eastern China (top row) and the Beijing to Shanghai corridor (bottom row). Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Pollution concentrations were computed as described in the text using hourly data and then the hourly concentration fields were averaged over the four month study duration. The resulting maps were rendered using MATLAB.\n",
      "      </p>\n",
      "     </caption>\n",
      "     <graphic xlink:href=\"pone.0135749.g003\">\n",
      "     </graphic>\n",
      "    </fig>\n",
      "    <p>\n",
      "     Air pollution is extensive in China, with the highest particulate concentrations observed south of Beijing (e.g. Xingtai / Handan), but significant levels extend throughout the interior, which is consistent with previous satellite and modeling estimates [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref011\">\n",
      "      11\n",
      "     </xref>\n",
      "     –\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref013\">\n",
      "      13\n",
      "     </xref>\n",
      "     ]. Extensive pollution is not surprising since particulate matter can remain airborne for days to weeks and travel thousands of kilometers. The corridor south of Beijing contains the highest pollution concentrations and, as discussed below, many of the largest sources. During this study, the southern coastal area experienced somewhat better air quality, possibly linked to greater rainfall (\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s005\">\n",
      "      S1 Fig\n",
      "     </xref>\n",
      "     ).\n",
      "    </p>\n",
      "    <p>\n",
      "     For PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     , portions of China encompassing roughly 38% of the population are classified as “unhealthy” on average (&gt;55 μg/m\n",
      "     <sup>\n",
      "      3\n",
      "     </sup>\n",
      "     , red) with an additional 45% of the population averaging “unhealthy for sensitive groups” (&gt;35 μg/m\n",
      "     <sup>\n",
      "      3\n",
      "     </sup>\n",
      "     , orange). Almost none of the study area averaged below the US EPA’s 12 μg/m\n",
      "     <sup>\n",
      "      3\n",
      "     </sup>\n",
      "     standard for annual average PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     exposure (green). The area-weighted average was 46 μg/m\n",
      "     <sup>\n",
      "      3\n",
      "     </sup>\n",
      "     and the population-weighted average exposure to PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     was 52 μg/m\n",
      "     <sup>\n",
      "      3\n",
      "     </sup>\n",
      "     . 92% of China’s population experienced unhealthy PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     for at least 120 hours during the study period. 46% of China’s population experienced PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     above the highest EPA threshold (“hazardous”, &gt;250 μg/m\n",
      "     <sup>\n",
      "      3\n",
      "     </sup>\n",
      "     ), during at least one hour in the observation period.\n",
      "    </p>\n",
      "    <p>\n",
      "     Patterns for PM\n",
      "     <sub>\n",
      "      10\n",
      "     </sub>\n",
      "     are similar but less severe, with average PM\n",
      "     <sub>\n",
      "      10\n",
      "     </sub>\n",
      "     levels “moderate” for most of China. Ozone concentrations are modest across most of China, though higher levels occur in the Northwest desert area, and in a small number of Northeastern cities. Though the average levels of PM\n",
      "     <sub>\n",
      "      10\n",
      "     </sub>\n",
      "     and O\n",
      "     <sub>\n",
      "      3\n",
      "     </sub>\n",
      "     are “moderate” or “good” for much of China, intermittently high levels of these pollutants can occur in some areas.\n",
      "    </p>\n",
      "    <sec id=\"sec004\">\n",
      "     <title>\n",
      "      Source Regions for Air Pollution in China\n",
      "     </title>\n",
      "     <p>\n",
      "      <xref ref-type=\"fig\" rid=\"pone.0135749.g004\">\n",
      "       Fig 4\n",
      "      </xref>\n",
      "      shows estimated pollutant fluxes for PM\n",
      "      <sub>\n",
      "       2.5\n",
      "      </sub>\n",
      "      , PM\n",
      "      <sub>\n",
      "       10\n",
      "      </sub>\n",
      "      , SO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      , and NO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      . Pollution emission is often localized, especially in the Beijing to Shanghai corridor where many of the highest PM concentrations also occur (\n",
      "      <xref ref-type=\"fig\" rid=\"pone.0135749.g003\">\n",
      "       Fig 3\n",
      "      </xref>\n",
      "      ). Most of the largest emissions appear in or near urban areas (e.g. Handan, Shijazhuang, Zibo, Tangshan, Linyi, Hangshou), though not all major cities have high pollution fluxes (e.g. Chongqing, Chengdu, Wuhan). The source map presumably reflects patterns of industrial activity, though detailed differences will not be explored here.\n",
      "     </p>\n",
      "     <fig id=\"pone.0135749.g004\" orientation=\"portrait\" position=\"float\">\n",
      "      <object-id pub-id-type=\"doi\">\n",
      "       10.1371/journal.pone.0135749.g004\n",
      "      </object-id>\n",
      "      <label>\n",
      "       Fig 4\n",
      "      </label>\n",
      "      <caption>\n",
      "       <title>\n",
      "        Air pollution source maps.\n",
      "       </title>\n",
      "       <p>\n",
      "        Maps of average pollutant flux for PM\n",
      "        <sub>\n",
      "         2.5\n",
      "        </sub>\n",
      "        , PM\n",
      "        <sub>\n",
      "         10\n",
      "        </sub>\n",
      "        , SO\n",
      "        <sub>\n",
      "         2\n",
      "        </sub>\n",
      "        , and NO\n",
      "        <sub>\n",
      "         2\n",
      "        </sub>\n",
      "        for eastern China (top row) and the Beijing to Shanghai corridor (bottom row). Pollutant fluxes were computed as described in the text from changes in the interpolated hourly pollution fields along with contemporaneous wind and weather data. Due to sparse sampling and secondary transformations of pollutants in the atmosphere, apparent source fluxes are likely to appear more diffuse than the true emissions source.\n",
      "       </p>\n",
      "      </caption>\n",
      "      <graphic xlink:href=\"pone.0135749.g004\">\n",
      "      </graphic>\n",
      "     </fig>\n",
      "     <p>\n",
      "      NO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      and SO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      emissions help suggest the pollution source. Nitrogen oxides, including nitric oxide (NO) and NO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      are created when air is heated, and on average have been attributed to transportation fuels (15–25%), fossil fuel burning in power plants (30–50%), and to industrial facilities (25–35%) [\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0135749.ref023\">\n",
      "       23\n",
      "      </xref>\n",
      "      ,\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0135749.ref024\">\n",
      "       24\n",
      "      </xref>\n",
      "      ]. It is expected that NO dominates at the combustion source, but in the presence of sunlight NO and NO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      will equilibrate within a few minutes (as well as reacting with O\n",
      "      <sub>\n",
      "       3\n",
      "      </sub>\n",
      "      ), implying that NO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      measurements reflect a combination of NO and NO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      emissions. SO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      emissions have been previously associated with coal (~90%) in power plants and industrial facilities [\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0135749.ref025\">\n",
      "       25\n",
      "      </xref>\n",
      "      ]. Beijing has negligible SO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      flux, despite a large NO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      signal, possibly a result of policies that limit coal burning in the immediate vicinity of Beijing and more extensively apply mitigation technologies.\n",
      "     </p>\n",
      "     <p>\n",
      "      Many of the SO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      and NO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      sources are also sources of PM pollution. This is not surprising since fossil fuel burning is also a major source of PM\n",
      "      <sub>\n",
      "       2.5\n",
      "      </sub>\n",
      "      and PM\n",
      "      <sub>\n",
      "       10\n",
      "      </sub>\n",
      "      . However, the PM sources appear more diffuse than either the SO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      or NO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      sources. In part, this is caused by secondary particulate matter formed within the atmosphere from other pollutants, such as SO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      or NO\n",
      "      <sub>\n",
      "       2\n",
      "      </sub>\n",
      "      [\n",
      "      <xref ref-type=\"bibr\" rid=\"pone.0135749.ref026\">\n",
      "       26\n",
      "      </xref>\n",
      "      ]. Secondary particulate formation may cause PM fluxes to appear more widely distributed than the underlying emitters. Nonetheless, many strong PM sources are identified through this analysis. Within the study region, 10% of the area is responsible for 34% of the PM\n",
      "      <sub>\n",
      "       2.5\n",
      "      </sub>\n",
      "      emissions, and 5% of the area is responsible for 22% of emissions. However, small and moderate sources are also important. Approximately 37% of the study region had PM\n",
      "      <sub>\n",
      "       2.5\n",
      "      </sub>\n",
      "      fluxes &gt;0.5 μg/m\n",
      "      <sup>\n",
      "       3\n",
      "      </sup>\n",
      "      /hr, sufficient to exceed US EPA standards after only 3 days of stagnant air.\n",
      "     </p>\n",
      "    </sec>\n",
      "   </sec>\n",
      "   <sec id=\"sec005\" sec-type=\"conclusions\">\n",
      "    <title>\n",
      "     Discussion\n",
      "    </title>\n",
      "    <p>\n",
      "     We have presented a technique for mapping air pollution concentrations and sources using data from monitoring stations. As has been known from satellite and modeling studies, particulate pollution is an extensive problem affecting nearly all of China’s population, but the observed heterogeneity of source locations could help develop strategies to reduce pollution.\n",
      "    </p>\n",
      "    <p>\n",
      "     We examined a four month interval as long-term station data were not available for most of China. Previous studies of both in situ and satellite data have indicated that winter and early spring months in China have somewhat higher PM concentrations due to increased use of fossil fuels for seasonal heating, weather patterns that concentrate pollution at low altitudes, and increased desert dust fluxes [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref012\">\n",
      "      12\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref027\">\n",
      "      27\n",
      "     </xref>\n",
      "     ]. In contrast, the air in China is typically cleanest from late summer to early fall. The April 5 to August 5 study period is somewhat intermediate. A review of hourly PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     station data from Beijing (2009–2014), Shanghai (2012–2014), Guangzhou (2012–2014), Chengdu (2013–2014), and Shenyang (2014) indicates that the months studied in this paper averaged 91%, 84%, 89%, 72%, and 73% respectively of the annual averages (U.S. Air Quality Monitoring Program,\n",
      "     <ext-link ext-link-type=\"uri\" xlink:href=\"http://www.stateair.net/web/mission/1/\">\n",
      "      http://www.stateair.net/web/mission/1/\n",
      "     </ext-link>\n",
      "     ). Monthly-resolved PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     satellite data for the whole study region was not immediately available, but a monthly satellite history for Beijing reported that April-July averaged 99% of the annual mean during 2000 to 2012 [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref028\">\n",
      "      28\n",
      "     </xref>\n",
      "     ]. Hence, particulate pollution estimates drawn from the current short study period will likely be similar to or somewhat lower than long-term averages. Future work could explore seasonal variations and long-term trends.\n",
      "    </p>\n",
      "    <p>\n",
      "     During the four months studied, the population-weighted and area-weighted PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     averages were 52 and 46 μg/m\n",
      "     <sup>\n",
      "      3\n",
      "     </sup>\n",
      "     respectively. Satellite pollution datasets generally focus on annual or multi-year averages, which limits the ability to make direct comparisons. However, the available satellite estimates tend to be similar to or somewhat lower than the ground observations. An analysis of the larger East Asia region estimated a population-weighted PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     exposure of 50 μg/m\n",
      "     <sup>\n",
      "      3\n",
      "     </sup>\n",
      "     for 2001 to 2010 [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref028\">\n",
      "      28\n",
      "     </xref>\n",
      "     ]. A version of the same dataset masked to the current study region had an area-weighted average of 40 μg/m\n",
      "     <sup>\n",
      "      3\n",
      "     </sup>\n",
      "     from 2010 to 2012 [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref028\">\n",
      "      28\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref029\">\n",
      "      29\n",
      "     </xref>\n",
      "     ]. A different satellite estimate using similar observations but different calibrations and modeling gave 25 μg/m\n",
      "     <sup>\n",
      "      3\n",
      "     </sup>\n",
      "     for the 2008 to 2010 average over the present study region [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref030\">\n",
      "      30\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref031\">\n",
      "      31\n",
      "     </xref>\n",
      "     ]. Both of these datasets show similar spatial patterns to what we observe (\n",
      "     <xref ref-type=\"fig\" rid=\"pone.0135749.g005\">\n",
      "      Fig 5\n",
      "     </xref>\n",
      "     ), though the magnitudes in van Donkelaar et al.’s work [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref028\">\n",
      "      28\n",
      "     </xref>\n",
      "     ] are clearly more consistent with our ground data estimates. As noted in previous satellite to in situ comparisons, satellite data may be more likely to underestimate pollution concentrations during the most extreme pollution events [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref028\">\n",
      "      28\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref032\">\n",
      "      32\n",
      "     </xref>\n",
      "     ].\n",
      "    </p>\n",
      "    <fig id=\"pone.0135749.g005\" orientation=\"portrait\" position=\"float\">\n",
      "     <object-id pub-id-type=\"doi\">\n",
      "      10.1371/journal.pone.0135749.g005\n",
      "     </object-id>\n",
      "     <label>\n",
      "      Fig 5\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Comparison of PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       observations to satellite data.\n",
      "      </title>\n",
      "      <p>\n",
      "       Maps of average PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       concentration from this study (top) and two satellite-derived datasets restricted to the same region. The average over the 2008 to 2010 time interval was chosen for the satellite data due to the limitations of the available satellite data. Both the concentrations reported by van Donkelaar [\n",
      "       <xref ref-type=\"bibr\" rid=\"pone.0135749.ref028\">\n",
      "        28\n",
      "       </xref>\n",
      "       ] (middle) and those reported by de Sherbinin [\n",
      "       <xref ref-type=\"bibr\" rid=\"pone.0135749.ref030\">\n",
      "        30\n",
      "       </xref>\n",
      "       ] (bottom) rely on similar satellite observations of aerosol optical depth (obtained by NASA), but interpret those observations differently when determining pollutant concentration. The satellite-derived data was imported from concentration data files provided by their respected sources and rendered via MATLAB to use the same US EPA health category color scheme applied in Figs\n",
      "       <xref ref-type=\"fig\" rid=\"pone.0135749.g002\">\n",
      "        2\n",
      "       </xref>\n",
      "       and\n",
      "       <xref ref-type=\"fig\" rid=\"pone.0135749.g003\">\n",
      "        3\n",
      "       </xref>\n",
      "       .\n",
      "      </p>\n",
      "     </caption>\n",
      "     <graphic xlink:href=\"pone.0135749.g005\">\n",
      "     </graphic>\n",
      "    </fig>\n",
      "    <p>\n",
      "     The conversion of pollution concentrations to mortality is complicated. We adopt the framework [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref020\">\n",
      "      20\n",
      "     </xref>\n",
      "     ] developed for the WHO Global Burden of Disease study [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref021\">\n",
      "      21\n",
      "     </xref>\n",
      "     ] that considers PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     mortality due to impacts on five distinct diseases and accounts for nonlinearities as a function of concentration. Using prefecture level population and pollution data along with national average death rates for the five modeled diseases, we calculate that 1.6 million deaths / year can be attributed to PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     air pollution under the WHO model [95% confidence: 0.7 to 2.2 million deaths/year]. This is equivalent to 4 thousand deaths / day or 17% of all deaths in China. Additional details appear in\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s001\">\n",
      "      S1 Text\n",
      "     </xref>\n",
      "     and\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s002\">\n",
      "      S1 Table\n",
      "     </xref>\n",
      "     . For perspective, the categories of mortality events considered by the WHO model, e.g. cardiorespiratory deaths, account for roughly 55% of all Chinese deaths [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref021\">\n",
      "      21\n",
      "     </xref>\n",
      "     ]. This compares to only 42% of mortality in the United States in the same cardiorespiratory categories, despite much higher incidence of obesity in the United States [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref021\">\n",
      "      21\n",
      "     </xref>\n",
      "     ]. The calculated mortality is somewhat higher than the 1.2 million deaths/year previously estimated from the Huai River study using Chinese air pollution measurements and mortality data [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref003\">\n",
      "      3\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref004\">\n",
      "      4\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref033\">\n",
      "      33\n",
      "     </xref>\n",
      "     ].\n",
      "    </p>\n",
      "    <p>\n",
      "     Though most of China is subject to potentially harmful levels of PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     , some large population centers (Chongqing, Wuhan, Chengdu) emit less than half the PM\n",
      "     <sub>\n",
      "      2.5\n",
      "     </sub>\n",
      "     of others. Among northeastern cities, Beijing has relatively low emissions except for NO\n",
      "     <sub>\n",
      "      2\n",
      "     </sub>\n",
      "     . Low SO\n",
      "     <sub>\n",
      "      2\n",
      "     </sub>\n",
      "     fluxes may indicate cities that benefit from lower coal usage or better smokestack pollution controls. Compared to natural gas, coal produces 150 to 400 times more PM for the same energy delivered [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref034\">\n",
      "      34\n",
      "     </xref>\n",
      "     ,\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref035\">\n",
      "      35\n",
      "     </xref>\n",
      "     ]. China has plans for new coal plants in the next decade that could effectively double their coal consumption [\n",
      "     <xref ref-type=\"bibr\" rid=\"pone.0135749.ref036\">\n",
      "      36\n",
      "     </xref>\n",
      "     ], potentially exacerbating the problem of air pollution. A table of pollution concentrations and fluxes by province and prefecture is included in\n",
      "     <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s002\">\n",
      "      S1 Table\n",
      "     </xref>\n",
      "     .\n",
      "    </p>\n",
      "    <p>\n",
      "     The methods of this study should be applicable to air quality monitoring in other regions of the world. However, these techniques require an extensive air-quality monitoring network with frequent updates (e.g. hourly), and such networks presently exist in only a few places. We hope that other countries will follow China’s lead and provide both extensive and transparent real-time air quality monitoring.\n",
      "    </p>\n",
      "   </sec>\n",
      "   <sec id=\"sec006\" sec-type=\"supplementary-material\">\n",
      "    <title>\n",
      "     Supporting Information\n",
      "    </title>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s001\">\n",
      "     <label>\n",
      "      S1 Text\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Supplemental Methods.\n",
      "      </title>\n",
      "      <p>\n",
      "       Additional discussion of methods, data handling, and validation.\n",
      "      </p>\n",
      "      <p>\n",
      "       (DOCX)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s001.docx\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s002\">\n",
      "     <label>\n",
      "      S1 Table\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Province and prefecture pollution and mortality data.\n",
      "      </title>\n",
      "      <p>\n",
      "       Provides pollution concentration and flux data broken down by geographic regions within China. Also provides the details of the associated mortality calculation.\n",
      "      </p>\n",
      "      <p>\n",
      "       (XLSX)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s002.xlsx\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s003\">\n",
      "     <label>\n",
      "      S2 Table\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Correlation vs. distance and lifetime of air pollutants in China.\n",
      "      </title>\n",
      "      <p>\n",
      "       Summary of the empirically determined correlation vs. distance functions for the five pollutants discussed in this study and the effective residence time in the atmosphere of their pollutant plumes in the absence of rain or snow. The correlation functions generally consist of a short range component (influenced largely by source distributions) and a long-range component (influenced by weather patterns). The effective pollutant plume lifetime shows the value empirically estimated for this study. Ranges in brackets indicate alternative lifetimes making different assumptions as discussed in the Supplemental Methods (\n",
      "       <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s001\">\n",
      "        S1 Text\n",
      "       </xref>\n",
      "       ).\n",
      "      </p>\n",
      "      <p>\n",
      "       (DOCX)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s003.docx\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s004\">\n",
      "     <label>\n",
      "      S1 Movie\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Animation of PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       Concentrations.\n",
      "      </title>\n",
      "      <p>\n",
      "       Movie showing the evolution of PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       concentrations, as inferred by this study, across Eastern China during the four month study interval. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts based on 24-hour exposure. Green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups”, red is “Unhealthy”, purple is “Very Unhealthy”, and dark red is “Hazardous”. Concentrations higher than the upper limit of the maximum EPA classification are shown in gray.\n",
      "      </p>\n",
      "      <p>\n",
      "       (MP4)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s004.mp4\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s005\">\n",
      "     <label>\n",
      "      S1 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Average wind and rain patterns in Eastern China.\n",
      "      </title>\n",
      "      <p>\n",
      "       (Left) Average wind speed pattern across China and surrounding areas during the period of this study according to Global Forecast System data at 80 m altitude. Arrow lengths are proportional to wind speed and the legend indicates the size of arrow consistent with a 10 km/hr average wind speed. (Right) Percentage of hours for each grid cell where rain was experienced during the study period based on observations from the Tropical Rainfall Measurement Mission satellite data. Maps were prepared with MATLAB with political boundaries from the Database of Global Administrative Areas (GADM version 2).\n",
      "      </p>\n",
      "      <p>\n",
      "       (TIF)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s005.tif\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s006\">\n",
      "     <label>\n",
      "      S2 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Pollution concentration correlation vs. distance.\n",
      "      </title>\n",
      "      <p>\n",
      "       Summary of empirically estimated correlation vs. distance functions for the five pollutants studied. Same values as\n",
      "       <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s003\">\n",
      "        S2 Table\n",
      "       </xref>\n",
      "       .\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s006.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s007\">\n",
      "     <label>\n",
      "      S3 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Empirical fit of PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       correlation data.\n",
      "      </title>\n",
      "      <p>\n",
      "       Heat map of correlation vs. distance generated by all possible pairwise comparisons of PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       data from stations in this study. Brighter colors indicate more frequently occurring combinations of R-value and distance, and are normalized for the abundance of comparisons at each distance. The red curve indicates the correlation vs. distance model fit to this data and described in\n",
      "       <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s003\">\n",
      "        S2 Table\n",
      "       </xref>\n",
      "       .\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s007.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s008\">\n",
      "     <label>\n",
      "      S4 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Impact of assumed plume lifetime on PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       source distribution.\n",
      "      </title>\n",
      "      <p>\n",
      "       Maps comparing calculated PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       source fluxes for different assumed values of the effective plume lifetime. The upper right panel corresponds to the data reported in the main paper, and has a lifetime chosen such that 5% of the field area is allowed to be negative (same as\n",
      "       <xref ref-type=\"fig\" rid=\"pone.0135749.g004\">\n",
      "        Fig 4\n",
      "       </xref>\n",
      "       and\n",
      "       <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s012\">\n",
      "        S8 Fig\n",
      "       </xref>\n",
      "       ). The upper left panel, corresponds to choosing a decay lifetime such that 2% of the flux field is negative, the lower left panel has 15% negative area, and the lower right panel has no plume decay in the absence of rain (infinite lifetime). Rain events are explicitly excluded from consideration, and the reported flux aims to capture the man-made sources of pollution. In general, assuming a shorter plume lifetime implies higher and more diffuse fluxes in order to maintain the observed concentration patterns. By contrast, if the pollutant plumes are assumed to have a very long lifetime then the apparent man-made flux will implausibly turn negative over substantial regions. Though quantitatively important, the choice of effective lifetime has little qualitative impact on the distribution of large sources. The regions of locally higher flux remain similar under all scenarios.\n",
      "      </p>\n",
      "      <p>\n",
      "       (TIF)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s008.tif\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s009\">\n",
      "     <label>\n",
      "      S5 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Comparison of PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       patterns in early and late half of study.\n",
      "      </title>\n",
      "      <p>\n",
      "       Maps comparing PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       concentrations (top) and fluxes (bottom) during the first half of the sampling period (left column) and the second half of the sampling period (right). PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       concentrations are presented using the same health impact associated color scheme as\n",
      "       <xref ref-type=\"fig\" rid=\"pone.0135749.g003\">\n",
      "        Fig 3\n",
      "       </xref>\n",
      "       .\n",
      "      </p>\n",
      "      <p>\n",
      "       (TIF)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s009.tif\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s010\">\n",
      "     <label>\n",
      "      S6 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Comparison of PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       patterns from half-density reconstructions.\n",
      "      </title>\n",
      "      <p>\n",
      "       The collection of available stations was randomly assigned to two groups, and the analyses for PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       concentrations (top) and fluxes (bottom) were repeated in full for both groups. These independent subsamples result in similar concentration and flux patterns, implying the reconstructions are likely to be stable with respect to data selection.\n",
      "      </p>\n",
      "      <p>\n",
      "       (TIF)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s010.tif\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s011\">\n",
      "     <label>\n",
      "      S7 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Average PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       concentration across China.\n",
      "      </title>\n",
      "      <p>\n",
      "       Larger version of panel from\n",
      "       <xref ref-type=\"fig\" rid=\"pone.0135749.g003\">\n",
      "        Fig 3\n",
      "       </xref>\n",
      "       presenting PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       concentrations across China. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Based on 24-hour exposure green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups” and red is “Unhealthy”.\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s011.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s012\">\n",
      "     <label>\n",
      "      S8 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       sources across China.\n",
      "      </title>\n",
      "      <p>\n",
      "       Larger version of panel from\n",
      "       <xref ref-type=\"fig\" rid=\"pone.0135749.g004\">\n",
      "        Fig 4\n",
      "       </xref>\n",
      "       presenting PM\n",
      "       <sub>\n",
      "        2.5\n",
      "       </sub>\n",
      "       source fluxes across China.\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s012.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s013\">\n",
      "     <label>\n",
      "      S9 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Average PM\n",
      "       <sub>\n",
      "        10\n",
      "       </sub>\n",
      "       concentration across China.\n",
      "      </title>\n",
      "      <p>\n",
      "       Larger version of panel from\n",
      "       <xref ref-type=\"fig\" rid=\"pone.0135749.g003\">\n",
      "        Fig 3\n",
      "       </xref>\n",
      "       presenting PM\n",
      "       <sub>\n",
      "        10\n",
      "       </sub>\n",
      "       concentrations across China. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Based on 24-hour exposure green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups” and red is “Unhealthy”.\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s013.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s014\">\n",
      "     <label>\n",
      "      S10 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       PM\n",
      "       <sub>\n",
      "        10\n",
      "       </sub>\n",
      "       sources across China.\n",
      "      </title>\n",
      "      <p>\n",
      "       Larger version of panel from\n",
      "       <xref ref-type=\"fig\" rid=\"pone.0135749.g004\">\n",
      "        Fig 4\n",
      "       </xref>\n",
      "       presenting PM\n",
      "       <sub>\n",
      "        10\n",
      "       </sub>\n",
      "       source fluxes across China.\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s014.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s015\">\n",
      "     <label>\n",
      "      S11 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Average SO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       concentration across China.\n",
      "      </title>\n",
      "      <p>\n",
      "       Map of SO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       concentrations across China. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Based on 24-hour exposure green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups” and red is “Unhealthy”. The limited coverage of this map is a consequence of the short correlation length for SO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       (\n",
      "       <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s003\">\n",
      "        S2 Table\n",
      "       </xref>\n",
      "       ;\n",
      "       <xref ref-type=\"supplementary-material\" rid=\"pone.0135749.s006\">\n",
      "        S2 Fig\n",
      "       </xref>\n",
      "       ), which limits the distance over which one can usefully interpolate SO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       . In part, the short correlation length is a consequence of the short lifetime of SO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       in the atmosphere, which makes estimates far from measurement sites highly uncertain.\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s015.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s016\">\n",
      "     <label>\n",
      "      S12 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       SO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       sources across China.\n",
      "      </title>\n",
      "      <p>\n",
      "       Larger version of panel from\n",
      "       <xref ref-type=\"fig\" rid=\"pone.0135749.g004\">\n",
      "        Fig 4\n",
      "       </xref>\n",
      "       presenting SO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       source fluxes across China.\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s016.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s017\">\n",
      "     <label>\n",
      "      S13 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Average NO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       concentration across China.\n",
      "      </title>\n",
      "      <p>\n",
      "       Map of NO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       concentrations across China. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Based on 24-hour exposure green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups” and red is “Unhealthy”.\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s017.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s018\">\n",
      "     <label>\n",
      "      S14 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       NO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       sources across China.\n",
      "      </title>\n",
      "      <p>\n",
      "       Larger version of panel from\n",
      "       <xref ref-type=\"fig\" rid=\"pone.0135749.g004\">\n",
      "        Fig 4\n",
      "       </xref>\n",
      "       presenting NO\n",
      "       <sub>\n",
      "        2\n",
      "       </sub>\n",
      "       source fluxes across China.\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s018.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s019\">\n",
      "     <label>\n",
      "      S15 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       Average O\n",
      "       <sub>\n",
      "        3\n",
      "       </sub>\n",
      "       concentration across China.\n",
      "      </title>\n",
      "      <p>\n",
      "       Larger version of panel from\n",
      "       <xref ref-type=\"fig\" rid=\"pone.0135749.g003\">\n",
      "        Fig 3\n",
      "       </xref>\n",
      "       presenting O\n",
      "       <sub>\n",
      "        3\n",
      "       </sub>\n",
      "       concentrations across China. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Based on 24-hour exposure green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups” and red is “Unhealthy”.\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s019.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "    <supplementary-material content-type=\"local-data\" id=\"pone.0135749.s020\">\n",
      "     <label>\n",
      "      S16 Fig\n",
      "     </label>\n",
      "     <caption>\n",
      "      <title>\n",
      "       O\n",
      "       <sub>\n",
      "        3\n",
      "       </sub>\n",
      "       sources across China.\n",
      "      </title>\n",
      "      <p>\n",
      "       Map of O\n",
      "       <sub>\n",
      "        3\n",
      "       </sub>\n",
      "       source fluxes across China.\n",
      "      </p>\n",
      "      <p>\n",
      "       (EPS)\n",
      "      </p>\n",
      "     </caption>\n",
      "     <media xlink:href=\"pone.0135749.s020.eps\">\n",
      "      <caption>\n",
      "       <p>\n",
      "        Click here for additional data file.\n",
      "       </p>\n",
      "      </caption>\n",
      "     </media>\n",
      "    </supplementary-material>\n",
      "   </sec>\n",
      "   <back>\n",
      "    <ack>\n",
      "     <p>\n",
      "      We thank Elizabeth Muller for suggesting this study, Steve Mosher and Zeke Hausfather for discussion and feedback, and John Li and Xinyu Zhang for help with Chinese materials.\n",
      "     </p>\n",
      "    </ack>\n",
      "    <ref-list>\n",
      "     <title>\n",
      "      References\n",
      "     </title>\n",
      "     <ref id=\"pone.0135749.ref001\">\n",
      "      <label>\n",
      "       1\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"other\">\n",
      "       World Health Organization. WHO methods and data sources for global causes of death 2000–2012. Global Health Estimates Technical Paper WHO/HIS/HSI/GHE/2014.7. 2014. Available:\n",
      "       <ext-link ext-link-type=\"uri\" xlink:href=\"http://www.who.int/entity/healthinfo/global_burden_disease/GlobalCOD_method_2000_2012.pdf\">\n",
      "        http://www.who.int/entity/healthinfo/global_burden_disease/GlobalCOD_method_2000_2012.pdf\n",
      "       </ext-link>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref002\">\n",
      "      <label>\n",
      "       2\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"other\">\n",
      "       World Health Organization. Burden of disease from the joint effects of Household and Ambient Air Pollution for 2012. WHO Technical Report. 2012. Available:\n",
      "       <ext-link ext-link-type=\"uri\" xlink:href=\"http://www.who.int/phe/health_topics/outdoorair/databases/AP_jointeffect_BoD_results_March2014.pdf\">\n",
      "        http://www.who.int/phe/health_topics/outdoorair/databases/AP_jointeffect_BoD_results_March2014.pdf\n",
      "       </ext-link>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref003\">\n",
      "      <label>\n",
      "       3\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"other\">\n",
      "       O’Keefe B. Recent Trends in Air Quality Standards in Europe and Asia: What’s next? HEI Annual Conference 2012. 2013. Available:\n",
      "       <ext-link ext-link-type=\"uri\" xlink:href=\"http://www.healtheffects.org/Slides/AnnConf2013/OKeefe-Sun.pdf\">\n",
      "        http://www.healtheffects.org/Slides/AnnConf2013/OKeefe-Sun.pdf\n",
      "       </ext-link>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref004\">\n",
      "      <label>\n",
      "       4\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Yang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         G\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Wang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Y\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zeng\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Y\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Gao\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         GF\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Liang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         X\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhou\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         M\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <etal>\n",
      "        et al\n",
      "       </etal>\n",
      "       <article-title>\n",
      "        Rapid health transition in China, 1990–2010: findings from the Global Burden of Disease Study 2010\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Lancet\n",
      "       </source>\n",
      "       <year>\n",
      "        2013\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        381\n",
      "       </volume>\n",
      "       :\n",
      "       <fpage>\n",
      "        1987\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        2015\n",
      "       </lpage>\n",
      "       .\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.1016/S0140-6736(13)61097-1\n",
      "       </pub-id>\n",
      "       <pub-id pub-id-type=\"pmid\">\n",
      "        23746901\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref005\">\n",
      "      <label>\n",
      "       5\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Dockery\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         DW\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Pope\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         CA\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Xu\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         X\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Spengler\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         JD\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Ware\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         JH\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Fay\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         ME\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <etal>\n",
      "        et al\n",
      "       </etal>\n",
      "       <article-title>\n",
      "        An association between air pollution and mortality in six U.S. cities\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        N Engl J Med\n",
      "       </source>\n",
      "       <year>\n",
      "        1993\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        329\n",
      "       </volume>\n",
      "       (\n",
      "       <issue>\n",
      "        24\n",
      "       </issue>\n",
      "       ),\n",
      "       <fpage>\n",
      "        1753\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        1759\n",
      "       </lpage>\n",
      "       .\n",
      "       <pub-id pub-id-type=\"pmid\">\n",
      "        8179653\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref006\">\n",
      "      <label>\n",
      "       6\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Pope\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         CA\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Burnett\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         RT\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Thun\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         MJ\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Calle\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         EE\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Krewski\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         D\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Ito\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         K\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <etal>\n",
      "        et al\n",
      "       </etal>\n",
      "       <article-title>\n",
      "        Lung cancer, cardiopulmonary mortality, and long-term exposure to fine particulate air pollution\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        JAMA\n",
      "       </source>\n",
      "       <year>\n",
      "        2002\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        287\n",
      "       </volume>\n",
      "       (\n",
      "       <issue>\n",
      "        9\n",
      "       </issue>\n",
      "       ):\n",
      "       <fpage>\n",
      "        1132\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        1141\n",
      "       </lpage>\n",
      "       .\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.1001/jama.287.9.1132\n",
      "       </pub-id>\n",
      "       .\n",
      "       <pub-id pub-id-type=\"pmid\">\n",
      "        11879110\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref007\">\n",
      "      <label>\n",
      "       7\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Hoek\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         G\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Krishnan\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         RM\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Beelen\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         R\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Peters\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         A\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Ostro\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         B\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Brunekreef\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         B\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <etal>\n",
      "        et al\n",
      "       </etal>\n",
      "       <article-title>\n",
      "        Long-term air pollution exposure and cardiorespiratory mortality: a review\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Env Health\n",
      "       </source>\n",
      "       .\n",
      "       <year>\n",
      "        2012\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        12\n",
      "       </volume>\n",
      "       :\n",
      "       <fpage>\n",
      "        43\n",
      "       </fpage>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref008\">\n",
      "      <label>\n",
      "       8\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Beelen\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         R\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Raaschou-Nielsen\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         O\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Stafoggia\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         M\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Andersen\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         ZJ\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Weinmayr\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         G\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        Effects of long-term exposure to air pollution on natural-cause mortality: an analysis of 22 European cohorts within the multicentre ESCAPE project\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Lancet\n",
      "       </source>\n",
      "       <year>\n",
      "        2013\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        383\n",
      "       </volume>\n",
      "       (\n",
      "       <issue>\n",
      "        9919\n",
      "       </issue>\n",
      "       ):\n",
      "       <fpage>\n",
      "        785\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        795\n",
      "       </lpage>\n",
      "       .\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.1016/S0140-6736(13)62158-3\n",
      "       </pub-id>\n",
      "       <pub-id pub-id-type=\"pmid\">\n",
      "        24332274\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref009\">\n",
      "      <label>\n",
      "       9\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"other\">\n",
      "       Government of China. Ambient Air Quality Standards (in Chinese). GB 3095–2012. 2012. Available:\n",
      "       <ext-link ext-link-type=\"uri\" xlink:href=\"http://kjs.mep.gov.cn/hjbhbz/bzwb/dqhjbh/dqhjzlbz/201203/W020120410330232398521.pdf\">\n",
      "        http://kjs.mep.gov.cn/hjbhbz/bzwb/dqhjbh/dqhjzlbz/201203/W020120410330232398521.pdf\n",
      "       </ext-link>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref010\">\n",
      "      <label>\n",
      "       10\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Q\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Geng\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         GN\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Wang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         SW\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Richter\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         A\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         He\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         KB\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        Satellite remote sensing of changes in NO\n",
      "        <sub>\n",
      "         x\n",
      "        </sub>\n",
      "        emissions over China during 1996–2010\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Chinese Sci. Bull\n",
      "       </source>\n",
      "       .\n",
      "       <year>\n",
      "        2012\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        57\n",
      "       </volume>\n",
      "       (\n",
      "       <issue>\n",
      "        22\n",
      "       </issue>\n",
      "       ):\n",
      "       <fpage>\n",
      "        2857\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        2864\n",
      "       </lpage>\n",
      "       .\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.1007/s11434-012-5015-4\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref011\">\n",
      "      <label>\n",
      "       11\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Wang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         J\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Xua\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         X\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Spurr\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         R\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Wang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Y\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Drury\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         E\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        Improved algorithm for MODIS satellite retrievals of aerosol optical thickness over land in dusty atmosphere: Implications for air quality monitoring in China\n",
      "       </article-title>\n",
      "       <source>\n",
      "        . Remote Sensing Env\n",
      "       </source>\n",
      "       .\n",
      "       <year>\n",
      "        2010\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        114\n",
      "       </volume>\n",
      "       :\n",
      "       <fpage>\n",
      "        2575\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        2583\n",
      "       </lpage>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref012\">\n",
      "      <label>\n",
      "       12\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Liu\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         X-H\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Y\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Cheng\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         S-H\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Xing\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         J\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Q\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Streets\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         DG\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <etal>\n",
      "        et al\n",
      "       </etal>\n",
      "       <article-title>\n",
      "        Understanding of regional air pollution over China using CMAQ, part I: performance evaluation and seasonal variation\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Atmos Env\n",
      "       </source>\n",
      "       .\n",
      "       <year>\n",
      "        2010\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        44\n",
      "       </volume>\n",
      "       (\n",
      "       <issue>\n",
      "        20\n",
      "       </issue>\n",
      "       ):\n",
      "       <fpage>\n",
      "        2415\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        2426\n",
      "       </lpage>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref013\">\n",
      "      <label>\n",
      "       13\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Lei\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Y\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Q\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         He\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         K\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Streets\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         D\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        Primary anthropogenic aerosol emission trends for China, 1990–2005\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Atmos Chem Phys. 2011\n",
      "       </source>\n",
      "       ;\n",
      "       <volume>\n",
      "        11\n",
      "       </volume>\n",
      "       ,\n",
      "       <fpage>\n",
      "        931\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        954\n",
      "       </lpage>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref014\">\n",
      "      <label>\n",
      "       14\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"book\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Schabenberger\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         O\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Gotway\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         CA\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <source>\n",
      "        Statistical Methods for Spatial Data Analysis\n",
      "       </source>\n",
      "       .\n",
      "       <publisher-name>\n",
      "        CRC Press\n",
      "       </publisher-name>\n",
      "       <year>\n",
      "        2004\n",
      "       </year>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref015\">\n",
      "      <label>\n",
      "       15\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Krige\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         DG\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        A statistical approach to some basic mine valuation problems on the Witwatersrand\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        J Chem Metall Min Soc S Afr\n",
      "       </source>\n",
      "       .\n",
      "       <year>\n",
      "        1951\n",
      "       </year>\n",
      "       ;\n",
      "       <month>\n",
      "        12\n",
      "       </month>\n",
      "       :\n",
      "       <fpage>\n",
      "        119\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        159\n",
      "       </lpage>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref016\">\n",
      "      <label>\n",
      "       16\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Rohde\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         RA\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Muller\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         RA\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Jacobsen\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         R\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Perlmutter\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         S\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Rosenfeld\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         A\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Wurtele\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         J\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .,\n",
      "       <etal>\n",
      "        et al\n",
      "       </etal>\n",
      "       <article-title>\n",
      "        Berkeley Earth Temperature Averaging Process\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Geoinfor Geostat: An Overview\n",
      "       </source>\n",
      "       <year>\n",
      "        2013\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        1\n",
      "       </volume>\n",
      "       (\n",
      "       <issue>\n",
      "        2\n",
      "       </issue>\n",
      "       )\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.4172/gigs.1000103\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref017\">\n",
      "      <label>\n",
      "       17\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"other\">\n",
      "       NOAA/NCEP. Global Forecast System (GFS) Atmospheric Model. 2012. Available:\n",
      "       <ext-link ext-link-type=\"uri\" xlink:href=\"http://www.ncdc.noaa.gov/data-access/model-data/model-datasets/global-forcast-system-gfs\">\n",
      "        www.ncdc.noaa.gov/data-access/model-data/model-datasets/global-forcast-system-gfs\n",
      "       </ext-link>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref018\">\n",
      "      <label>\n",
      "       18\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Huffman\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         GJ\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Adler\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         RF\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Bolvin\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         DT\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Gu\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         G\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Nelkin\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         EJ\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Bowman\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         KP\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <etal>\n",
      "        et al\n",
      "       </etal>\n",
      "       <article-title>\n",
      "        The TRMM Multi-satellite Precipitation Analysis: Quasi-global, multi-year, combined-sensor precipitation estimates at fine scale\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        J. Hydrometeor\n",
      "       </source>\n",
      "       .\n",
      "       <year>\n",
      "        2007\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        8\n",
      "       </volume>\n",
      "       (\n",
      "       <issue>\n",
      "        1\n",
      "       </issue>\n",
      "       ):\n",
      "       <fpage>\n",
      "        38\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        55\n",
      "       </lpage>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref019\">\n",
      "      <label>\n",
      "       19\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"other\">\n",
      "       Huffman GJ, Bolvin DT. Real-Time TRMM Multi-Satellite Precipitation Analysis Data Set Documentation. NASA/GSFC Laboratory for Atmospheres, 43 pp. 2011. Available:\n",
      "       <ext-link ext-link-type=\"ftp\" xlink:href=\"ftp://meso.gsfc.nasa.gov/pub/trmmdocs/rt/3B4XRT_doc.pdf\">\n",
      "        ftp://meso.gsfc.nasa.gov/pub/trmmdocs/rt/3B4XRT_doc.pdf\n",
      "       </ext-link>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref020\">\n",
      "      <label>\n",
      "       20\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Burnett\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         RT\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Pope CA\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         III\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Ezzati\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         M\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Olives\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         C\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Lim\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         SS\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Mehta\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         S\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <etal>\n",
      "        et al\n",
      "       </etal>\n",
      "       <article-title>\n",
      "        An integrated risk function for estimating the global burden of disease attributable to ambient fine particulate matter exposure\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Env Health Perspect\n",
      "       </source>\n",
      "       <year>\n",
      "        2014\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        122\n",
      "       </volume>\n",
      "       :\n",
      "       <fpage>\n",
      "        397\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        403\n",
      "       </lpage>\n",
      "       .\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.1289/ehp.1307049\n",
      "       </pub-id>\n",
      "       <pub-id pub-id-type=\"pmid\">\n",
      "        24518036\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref021\">\n",
      "      <label>\n",
      "       21\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Naghavi\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         M\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Wang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         H\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Lozano\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         R\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Davis\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         A\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Liang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         X\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhou\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         M\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <etal>\n",
      "        et al\n",
      "       </etal>\n",
      "       <article-title>\n",
      "        Global, regional, and national age–sex specific all-cause and cause-specific mortality for 240 causes of death, 1990–2013: a systematic analysis for the Global Burden of Disease Study 2013\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Lancet\n",
      "       </source>\n",
      "       <year>\n",
      "        2015\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        385\n",
      "       </volume>\n",
      "       (\n",
      "       <issue>\n",
      "        9963\n",
      "       </issue>\n",
      "       ):\n",
      "       <fpage>\n",
      "        117\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        171\n",
      "       </lpage>\n",
      "       .\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.1016/S0140-6736(14)61682-2\n",
      "       </pub-id>\n",
      "       <pub-id pub-id-type=\"pmid\">\n",
      "        25530442\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref022\">\n",
      "      <label>\n",
      "       22\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"other\">\n",
      "       US EPA. Air Quality Index: A Guide to Air Quality and Your Health. U.S. EPA Report EPA-456/F-14-002. 2014. Available:\n",
      "       <ext-link ext-link-type=\"uri\" xlink:href=\"http://www.epa.gov/airnow/aqi_brochure_02_14.pdf\">\n",
      "        http://www.epa.gov/airnow/aqi_brochure_02_14.pdf\n",
      "       </ext-link>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref023\">\n",
      "      <label>\n",
      "       23\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"other\">\n",
      "       Emission Database for Global Atmospheric Research (EDGAR), release 4.2. Available:\n",
      "       <ext-link ext-link-type=\"uri\" xlink:href=\"http://edgar.jrc.ec.europa.eu\">\n",
      "        http://edgar.jrc.ec.europa.eu\n",
      "       </ext-link>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref024\">\n",
      "      <label>\n",
      "       24\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Shi\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Y\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Xia\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Y-F\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Lu\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         B\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Liu\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         N\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         L\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Li\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         S-J\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <etal>\n",
      "        et al\n",
      "       </etal>\n",
      "       <article-title>\n",
      "        Emission inventory and trends of NO\n",
      "        <sub>\n",
      "         x\n",
      "        </sub>\n",
      "        for China, 2000–2020\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        J Zhejiang Univ-SCI A\n",
      "       </source>\n",
      "       <year>\n",
      "        2014\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        15\n",
      "       </volume>\n",
      "       (\n",
      "       <issue>\n",
      "        6\n",
      "       </issue>\n",
      "       ):\n",
      "       <fpage>\n",
      "        454\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        464\n",
      "       </lpage>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref025\">\n",
      "      <label>\n",
      "       25\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Lu\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Z\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Q\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Streets\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         D\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        Sulfur dioxide and primary carbonaceous aerosol emissions in China and India, 1996–2010\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Atmos Chem Phys\n",
      "       </source>\n",
      "       .\n",
      "       <year>\n",
      "        2011\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        11\n",
      "       </volume>\n",
      "       :\n",
      "       <fpage>\n",
      "        9839\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        9864\n",
      "       </lpage>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref026\">\n",
      "      <label>\n",
      "       26\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Davidson\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         C\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Phalen\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         R\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Solomon\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         P\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        Airborne particular matter and human health: a review\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Aerosol Sci Technol\n",
      "       </source>\n",
      "       .\n",
      "       <year>\n",
      "        2005\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        39\n",
      "       </volume>\n",
      "       :\n",
      "       <fpage>\n",
      "        737\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        749\n",
      "       </lpage>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref027\">\n",
      "      <label>\n",
      "       27\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         W-J\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Sun\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Y-L\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhuang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         G-S\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Xu\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         D-Q\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        Characteristics and Seasonal Variations of PM2.5, PM10, and TSP Aerosol in Beijing\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Biomed Env Sci\n",
      "       </source>\n",
      "       <year>\n",
      "        2006\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        19\n",
      "       </volume>\n",
      "       :\n",
      "       <fpage>\n",
      "        461\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        468\n",
      "       </lpage>\n",
      "       .\n",
      "       <pub-id pub-id-type=\"pmid\">\n",
      "        17319272\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref028\">\n",
      "      <label>\n",
      "       28\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         van Donkelaar\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         A\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Martin\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         RV\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Brauer\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         M\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Boys\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         BL\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        Global fine particulate matter concentrations from satellite for long-term exposure assessment\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Env Health Perspectives\n",
      "       </source>\n",
      "       <year>\n",
      "        2015\n",
      "       </year>\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.1289/ehp.1408646\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref029\">\n",
      "      <label>\n",
      "       29\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Boys\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         BL\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Martin\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         RV\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         van Donkelaar\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         A\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         MacDonell\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         R\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Hsu\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         NC\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Cooper\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         MJ\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Yantosca\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         RM\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Lu\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Z\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Streets\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         DG\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Q\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Wang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         S\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        Fifteen-year global time series of satellite-derived fine particulate matter\n",
      "       </article-title>\n",
      "       ,\n",
      "       <source>\n",
      "        Env Sci Technol\n",
      "       </source>\n",
      "       .\n",
      "       <year>\n",
      "        2014\n",
      "       </year>\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.1021/es502113p\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref030\">\n",
      "      <label>\n",
      "       30\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         de Sherbinin\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         A\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Levy\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         M\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zell\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         E\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Weber\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         S\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Jaiteh\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         M\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        Using Satellite Data to Develop Environmental Indicators\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Environmental Research Letters\n",
      "       </source>\n",
      "       <year>\n",
      "        2014\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        9\n",
      "       </volume>\n",
      "       (\n",
      "       <issue>\n",
      "        8\n",
      "       </issue>\n",
      "       ):\n",
      "       <fpage>\n",
      "        084013\n",
      "       </fpage>\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.1088/1748-9326/9/8/084013\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref031\">\n",
      "      <label>\n",
      "       31\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"book\">\n",
      "       <collab>\n",
      "        Battelle Memorial Institute, Center for International Earth Science Information Network—CIESIN—Columbia University\n",
      "       </collab>\n",
      "       <source>\n",
      "        . Global Annual Average PM2.5 Grids from MODIS and MISR Aerosol Optical Depth (AOD)\n",
      "       </source>\n",
      "       .\n",
      "       <publisher-loc>\n",
      "        Palisades, NY\n",
      "       </publisher-loc>\n",
      "       :\n",
      "       <publisher-name>\n",
      "        NASA Socioeconomic Data and Applications Center (SEDAC)\n",
      "       </publisher-name>\n",
      "       <year>\n",
      "        2013\n",
      "       </year>\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.7927/H4H41PB4\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref032\">\n",
      "      <label>\n",
      "       32\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Song\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         W\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Jia\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         H\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Huang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         J\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Zhang\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Y\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        A satellite-based geographically weighted regression model for regional PM2.5 estimation over the Pearl River Delta region in China\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        Remote Sensing Env\n",
      "       </source>\n",
      "       .\n",
      "       <year>\n",
      "        2014\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        154\n",
      "       </volume>\n",
      "       :\n",
      "       <fpage>\n",
      "        1\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        7\n",
      "       </lpage>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref033\">\n",
      "      <label>\n",
      "       33\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"journal\">\n",
      "       <name>\n",
      "        <surname>\n",
      "         Chen\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         Y\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Ebenstein\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         A\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Greenstone\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         M\n",
      "        </given-names>\n",
      "       </name>\n",
      "       ,\n",
      "       <name>\n",
      "        <surname>\n",
      "         Lie\n",
      "        </surname>\n",
      "        <given-names>\n",
      "         H\n",
      "        </given-names>\n",
      "       </name>\n",
      "       .\n",
      "       <article-title>\n",
      "        Evidence on the impact of sustained exposure to air pollution on life expectancy from China’s Huai River policy\n",
      "       </article-title>\n",
      "       .\n",
      "       <source>\n",
      "        PNAS\n",
      "       </source>\n",
      "       <year>\n",
      "        2013\n",
      "       </year>\n",
      "       ;\n",
      "       <volume>\n",
      "        110\n",
      "       </volume>\n",
      "       (\n",
      "       <issue>\n",
      "        32\n",
      "       </issue>\n",
      "       ):\n",
      "       <fpage>\n",
      "        12936\n",
      "       </fpage>\n",
      "       –\n",
      "       <lpage>\n",
      "        12941\n",
      "       </lpage>\n",
      "       .\n",
      "       <pub-id pub-id-type=\"doi\">\n",
      "        10.1073/pnas.1300018110\n",
      "       </pub-id>\n",
      "       <pub-id pub-id-type=\"pmid\">\n",
      "        23836630\n",
      "       </pub-id>\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref034\">\n",
      "      <label>\n",
      "       34\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"other\">\n",
      "       Cai H, Wang M, Elgowainy A, Han J. Updated Greenhouse Gas and Criteria Air Pollutant Emission Factors and Their Probability Distribution Functions for Electric Generating Units. Argonne National Laboratory ANL/ESD/12-2. 2012.\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref035\">\n",
      "      <label>\n",
      "       35\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"other\">\n",
      "       US Energy Information Agency. Natural Gas 1998, Issues and Trends. US EIA Report DOE/EIA-0560(98). 1998. Available:\n",
      "       <ext-link ext-link-type=\"uri\" xlink:href=\"http://www.eia.doe.gov/oil_gas/natural_gas/analysis_publications/natural_gas_1998_issues_and_trends/it98.html\">\n",
      "        http://www.eia.doe.gov/oil_gas/natural_gas/analysis_publications/natural_gas_1998_issues_and_trends/it98.html\n",
      "       </ext-link>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "     <ref id=\"pone.0135749.ref036\">\n",
      "      <label>\n",
      "       36\n",
      "      </label>\n",
      "      <mixed-citation publication-type=\"other\">\n",
      "       Yang A, Cui Y. Global Coal Risk Assessment: Data Analysis and Market Research. World Resources Institute Working Paper. 2012. Available:\n",
      "       <ext-link ext-link-type=\"uri\" xlink:href=\"http://pdf.wri.org/global_coal_risk_assessment.pdf\">\n",
      "        http://pdf.wri.org/global_coal_risk_assessment.pdf\n",
      "       </ext-link>\n",
      "       .\n",
      "      </mixed-citation>\n",
      "     </ref>\n",
      "    </ref-list>\n",
      "   </back>\n",
      "  </article>\n",
      " </body>\n",
      "</html>\n"
     ]
    }
   ],
   "source": [
    "print(soup.prettify())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "art = Article(xml)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "Introduction\n",
      "\n",
      "Air pollution is a problem for much of the developing world and is believed to kill more people worldwide than AIDS, malaria, breast cancer, or tuberculosis [1–4]. Airborne particulate matter (PM) is especially detrimental to health [5–8], and has previously been estimated to cause between 3 and 7 million deaths every year, primarily by creating or worsening cardiorespiratory disease [2–4,6,7]. Particulate sources include electric power plants, industrial facilities, automobiles, biomass burning, and fossil fuels used in homes and factories for heating. In China, air pollution was previously estimated to contribute to 1.2 to 2 million deaths annually [2–4].\n",
      "\n",
      "In 2012, China adopted the Ambient Air Quality Standard [9], and began development of a national Air Reporting System that now includes 945 sites in 190 cities. These automated stations report hourly via the internet, and focus on six pollutants: particulate matter < 2.5 microns (PM2.5), particulate matter < 10 microns (PM10), sulfur dioxide (SO2), nitrogen dioxide (NO2), ozone (O3), and carbon monoxide (CO). Provincial governments perform air quality monitoring at 600 additional locations that are not yet integrated into the national system. Previous studies of regional scale air pollution have generally relied on satellite data [10,11] or modeling [12,13], but the high density of hourly data in China now allows regional patterns to be constructed directly from ground observations.\n",
      "\n",
      "\n",
      "Materials and Methods\n",
      "\n",
      "Though China deserves praise for its monitoring system and transparent communication, most archived observations are not publicly available. To compensate, real-time data was downloaded every hour during a four month interval from April 5, 2014 to August 5, 2014. Due to download restrictions on the official Chinese air quality reporting system, two different third-party sources were used: PM25.in and AQICN.org. PM25.in is a direct mirror of data from the 945 stations in China’s national network, while AQICN.org is the world’s largest aggregator of real-time air quality data and included many additional sites in China and surrounding areas. Nearly all of the additional data from within China originates from stations operated by provincial environmental agencies that have not yet been incorporated in China’s national network.\n",
      "\n",
      "Consistency, quality control, and validation checks were applied to the raw data prior to further analysis in order to reduce the impact of outliers, badly calibrated instruments, and other problems. The most common quality problem was associated with stuck instruments that implausibly reported the same concentration continuously for many hours. A regional consistency check was also applied to verify that each station was reporting data similar to its neighboring stations. Approximately 8% of the data was removed as a result of the quality control review. Further details are described in the supplemental material (S1 Text).\n",
      "\n",
      "As little monitoring is conducted in western China (Fig 1); we will focus on China east of 95° E, which includes 97% of the population. After removing stations with a high percentage of missing values or with other quality control problems, this study used 880 national network sites, 640 other sites in China and Taiwan, and 236 sites in other countries within 500 km of China (mostly South Korea). The air quality network is skewed towards urban areas, often with several sites per city and fewer, if any, in rural areas. For the n-th site, we use pn(t) to denote the pollutant concentration time series and pn¯ to denote the mean pollutant concentration.\n",
      "\n",
      "10.1371/journal.pone.0135749.g001Fig 1 Map of real-time air pollution monitoring stations.Map shows the locations of air quality monitoring sites in China and surrounding areas with sufficient hourly data to be included in this study. Selection criteria and data sources are described in S1 Text. The map was prepared in MATLAB using political boundaries from the Global Database of Administrative Areas (version 2; http://gadm.org/).\n",
      "\n",
      "For each pollutant, a correlation vs. distance function was estimated by computing all possible pairwise correlations between different stations and fitting the resulting correlations to a two part exponential decay as a function of distance. The resulting functional forms are stated in S2 Table and shown in S2 and S3 Figs. The correlation functions are used to construct correlation matrices that in turn are used to compute Kriging coefficients [14, 15], Kn(x→).\n",
      "\n",
      "The interpolated pollutant field, P(x→,t), is then estimated in two parts.\n",
      "\n",
      "P(x→,t)=S(x→)+A(x→,t)\n",
      "\n",
      "S(x→)=(∑nKn(x→)(pn¯−G(x→n)))+G(x→)\n",
      "\n",
      "A(x→,t)=∑nKn*(x→,t)(pn(t)−pn¯)\n",
      "\n",
      "The stationary part, S(x→), is derived by applying Kriging interpolation to the mean pollutant concentrations and a global predictor, G(x→n), that depends on latitude and longitude and contains free parameters that are adjusted to fit the observed means. The time-dependent anomaly part, A(x→,t), depends only on the fluctuations at each station relative to the local mean, and its Kriging coefficients, Kn*(x→n,t), are computed with restriction to stations that are active at time t. This two-step process reduces errors associated with stations that have intermittently missing data. This method is similar to that used by Berkeley Earth for its historical earth temperature analysis [16]. Since the correlation vs. distance function has been constructed with the correlation at zero distance obtaining a value less than one, the resulting interpolated fields will be smoother than the original data. This design was chosen for its ability to compensate for noise in the underlying station measurements. Additional details of the interpolation process are provided in the supplement methods (S1 Text).\n",
      "\n",
      "For mapping and computation, this continuous field was sampled with an approximately 6 km resolution, though in practice, the characteristic size of resolvable features is often larger (e.g. 30 km) and varies with station density and noise.\n",
      "\n",
      "A simple estimate of pollutant fluxes, F(x→,t), was computed by comparing observed changes in the hourly pollutant concentration to the concentrations expected due to short-term wind transport v→(x→,t) and an exponential decay with lifetime τ. Differences from the simple transport and decay model are assumed to represent source fluxes.\n",
      "\n",
      "F(x→,t)=P(x→+v→(x→,t)Δt,t+Δt)−e−Δt/τP(x→,t)2Δt+P(x→,t)−e−Δt/τP(x→−v→(x→,t)Δt,t−Δt)2Δt\n",
      "\n",
      "The near-surface (80 m) wind field from the Global Forecast System [17] was used for this calculation, and the effective pollutant lifetime was estimated as described in S1 Text and reported in S2 Table. Flux averages were computed by time-averaging the resulting field after excluding outlying values and cells affected by rain events as determined from Tropical Rainfall Measuring Mission data [18, 19].\n",
      "\n",
      "The change in mortality due to PM2.5 air pollution was calculated by adopting the integrated exposure response function approach [20] which considers relative risk of death for five disease classes (stroke, ischemic heart disease, lung cancer, chronic obstructive pulmonary disease, and lower respiratory infection) and which was adopted by World Health Organization (WHO) for the Global Burden of Disease study [21]. The model incorporates non-linear response versus concentration and provides an estimate of uncertainty. Relative risk was calculated at the prefecture level using local average PM2.5 concentration. The data for different diseases and prefectures was then combined to construct national average mortality estimates.\n",
      "\n",
      "Analysis and figure rendering was performed using original software written for this project on the MATLAB platform (version 2014a; http://www.mathworks.com/). Additional details of these calculations and associated background information is provided in the supplemental methods document (S1 Text).\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "\n",
      "Fig 2 shows a time series of PM2.5 concentration at Beijing and interpolated maps at three time points separated by 6 hours each. This shows the volatile nature of air pollution and the role of weather patterns in redistributing pollution on short timescales. Our approach creates a smooth field that approximates the data at each station, but allows a degree of difference attributable to noise. The pollution is extensive and rapidly evolves in response to winds and other atmospheric conditions. In the figure, fresh air from the North displaces a period of heavy pollution. Hourly data allows us to capture this evolution and ultimately estimate source fluxes. S1 Movie shows the time evolution of PM2.5 across the entire country.\n",
      "\n",
      "10.1371/journal.pone.0135749.g002Fig 2 Time evolution of PM2.5 pollution in the vicinity of Beijing.(Top) Time series of PM2.5 concentration at Beijing extracted from the interpolated field. Red circles indicate times shown in bottom row. (Bottom) Maps of interpolated PM2.5 concentration during a period of high pollution. Pollution concentrations were computed as described in the text from hourly data and maps were rendered in MATLAB. Concentrations are shown using color gradients and contour lines, where color tones (green, yellow, etc.) correspond to health impact categories defined by the US EPA. Bold circles show station locations with the observed value at each station indicated by the color within the circle.\n",
      "\n",
      "\n",
      "Fig 3 presents averages of the interpolated data for PM2.5, PM10, and O3 across the study duration. The maps are color-coded based on US EPA health categories for 24-hour exposure [22]. Maps for SO2 and NO2 are included in the supplemental materials and show “good” levels nearly everywhere (S11 and S13 Figs).\n",
      "\n",
      "10.1371/journal.pone.0135749.g003Fig 3 Average air pollution maps.Maps of average pollutant concentration for PM2.5, PM10, and O3 for eastern China (top row) and the Beijing to Shanghai corridor (bottom row). Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Pollution concentrations were computed as described in the text using hourly data and then the hourly concentration fields were averaged over the four month study duration. The resulting maps were rendered using MATLAB.\n",
      "\n",
      "Air pollution is extensive in China, with the highest particulate concentrations observed south of Beijing (e.g. Xingtai / Handan), but significant levels extend throughout the interior, which is consistent with previous satellite and modeling estimates [11–13]. Extensive pollution is not surprising since particulate matter can remain airborne for days to weeks and travel thousands of kilometers. The corridor south of Beijing contains the highest pollution concentrations and, as discussed below, many of the largest sources. During this study, the southern coastal area experienced somewhat better air quality, possibly linked to greater rainfall (S1 Fig).\n",
      "\n",
      "For PM2.5, portions of China encompassing roughly 38% of the population are classified as “unhealthy” on average (>55 μg/m3, red) with an additional 45% of the population averaging “unhealthy for sensitive groups” (>35 μg/m3, orange). Almost none of the study area averaged below the US EPA’s 12 μg/m3 standard for annual average PM2.5 exposure (green). The area-weighted average was 46 μg/m3 and the population-weighted average exposure to PM2.5 was 52 μg/m3. 92% of China’s population experienced unhealthy PM2.5 for at least 120 hours during the study period. 46% of China’s population experienced PM2.5 above the highest EPA threshold (“hazardous”, >250 μg/m3), during at least one hour in the observation period.\n",
      "\n",
      "Patterns for PM10 are similar but less severe, with average PM10 levels “moderate” for most of China. Ozone concentrations are modest across most of China, though higher levels occur in the Northwest desert area, and in a small number of Northeastern cities. Though the average levels of PM10 and O3 are “moderate” or “good” for much of China, intermittently high levels of these pollutants can occur in some areas.\n",
      "\n",
      "\n",
      "Source Regions for Air Pollution in China\n",
      "\n",
      "\n",
      "Fig 4 shows estimated pollutant fluxes for PM2.5, PM10, SO2, and NO2. Pollution emission is often localized, especially in the Beijing to Shanghai corridor where many of the highest PM concentrations also occur (Fig 3). Most of the largest emissions appear in or near urban areas (e.g. Handan, Shijazhuang, Zibo, Tangshan, Linyi, Hangshou), though not all major cities have high pollution fluxes (e.g. Chongqing, Chengdu, Wuhan). The source map presumably reflects patterns of industrial activity, though detailed differences will not be explored here.\n",
      "\n",
      "10.1371/journal.pone.0135749.g004Fig 4 Air pollution source maps.Maps of average pollutant flux for PM2.5, PM10, SO2, and NO2 for eastern China (top row) and the Beijing to Shanghai corridor (bottom row). Pollutant fluxes were computed as described in the text from changes in the interpolated hourly pollution fields along with contemporaneous wind and weather data. Due to sparse sampling and secondary transformations of pollutants in the atmosphere, apparent source fluxes are likely to appear more diffuse than the true emissions source.\n",
      "\n",
      "NO2 and SO2 emissions help suggest the pollution source. Nitrogen oxides, including nitric oxide (NO) and NO2 are created when air is heated, and on average have been attributed to transportation fuels (15–25%), fossil fuel burning in power plants (30–50%), and to industrial facilities (25–35%) [23, 24]. It is expected that NO dominates at the combustion source, but in the presence of sunlight NO and NO2 will equilibrate within a few minutes (as well as reacting with O3), implying that NO2 measurements reflect a combination of NO and NO2 emissions. SO2 emissions have been previously associated with coal (~90%) in power plants and industrial facilities [25]. Beijing has negligible SO2 flux, despite a large NO2 signal, possibly a result of policies that limit coal burning in the immediate vicinity of Beijing and more extensively apply mitigation technologies.\n",
      "\n",
      "Many of the SO2 and NO2 sources are also sources of PM pollution. This is not surprising since fossil fuel burning is also a major source of PM2.5 and PM10. However, the PM sources appear more diffuse than either the SO2 or NO2 sources. In part, this is caused by secondary particulate matter formed within the atmosphere from other pollutants, such as SO2 or NO2 [26]. Secondary particulate formation may cause PM fluxes to appear more widely distributed than the underlying emitters. Nonetheless, many strong PM sources are identified through this analysis. Within the study region, 10% of the area is responsible for 34% of the PM2.5 emissions, and 5% of the area is responsible for 22% of emissions. However, small and moderate sources are also important. Approximately 37% of the study region had PM2.5 fluxes >0.5 μg/m3/hr, sufficient to exceed US EPA standards after only 3 days of stagnant air.\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "We have presented a technique for mapping air pollution concentrations and sources using data from monitoring stations. As has been known from satellite and modeling studies, particulate pollution is an extensive problem affecting nearly all of China’s population, but the observed heterogeneity of source locations could help develop strategies to reduce pollution.\n",
      "\n",
      "We examined a four month interval as long-term station data were not available for most of China. Previous studies of both in situ and satellite data have indicated that winter and early spring months in China have somewhat higher PM concentrations due to increased use of fossil fuels for seasonal heating, weather patterns that concentrate pollution at low altitudes, and increased desert dust fluxes [12, 27]. In contrast, the air in China is typically cleanest from late summer to early fall. The April 5 to August 5 study period is somewhat intermediate. A review of hourly PM2.5 station data from Beijing (2009–2014), Shanghai (2012–2014), Guangzhou (2012–2014), Chengdu (2013–2014), and Shenyang (2014) indicates that the months studied in this paper averaged 91%, 84%, 89%, 72%, and 73% respectively of the annual averages (U.S. Air Quality Monitoring Program, http://www.stateair.net/web/mission/1/). Monthly-resolved PM2.5 satellite data for the whole study region was not immediately available, but a monthly satellite history for Beijing reported that April-July averaged 99% of the annual mean during 2000 to 2012 [28]. Hence, particulate pollution estimates drawn from the current short study period will likely be similar to or somewhat lower than long-term averages. Future work could explore seasonal variations and long-term trends.\n",
      "\n",
      "During the four months studied, the population-weighted and area-weighted PM2.5 averages were 52 and 46 μg/m3 respectively. Satellite pollution datasets generally focus on annual or multi-year averages, which limits the ability to make direct comparisons. However, the available satellite estimates tend to be similar to or somewhat lower than the ground observations. An analysis of the larger East Asia region estimated a population-weighted PM2.5 exposure of 50 μg/m3 for 2001 to 2010 [28]. A version of the same dataset masked to the current study region had an area-weighted average of 40 μg/m3 from 2010 to 2012 [28, 29]. A different satellite estimate using similar observations but different calibrations and modeling gave 25 μg/m3 for the 2008 to 2010 average over the present study region [30, 31]. Both of these datasets show similar spatial patterns to what we observe (Fig 5), though the magnitudes in van Donkelaar et al.’s work [28] are clearly more consistent with our ground data estimates. As noted in previous satellite to in situ comparisons, satellite data may be more likely to underestimate pollution concentrations during the most extreme pollution events [28, 32].\n",
      "\n",
      "10.1371/journal.pone.0135749.g005Fig 5 Comparison of PM2.5 observations to satellite data.Maps of average PM2.5 concentration from this study (top) and two satellite-derived datasets restricted to the same region. The average over the 2008 to 2010 time interval was chosen for the satellite data due to the limitations of the available satellite data. Both the concentrations reported by van Donkelaar [28] (middle) and those reported by de Sherbinin [30] (bottom) rely on similar satellite observations of aerosol optical depth (obtained by NASA), but interpret those observations differently when determining pollutant concentration. The satellite-derived data was imported from concentration data files provided by their respected sources and rendered via MATLAB to use the same US EPA health category color scheme applied in Figs 2 and 3.\n",
      "\n",
      "The conversion of pollution concentrations to mortality is complicated. We adopt the framework [20] developed for the WHO Global Burden of Disease study [21] that considers PM2.5 mortality due to impacts on five distinct diseases and accounts for nonlinearities as a function of concentration. Using prefecture level population and pollution data along with national average death rates for the five modeled diseases, we calculate that 1.6 million deaths / year can be attributed to PM2.5 air pollution under the WHO model [95% confidence: 0.7 to 2.2 million deaths/year]. This is equivalent to 4 thousand deaths / day or 17% of all deaths in China. Additional details appear in S1 Text and S1 Table. For perspective, the categories of mortality events considered by the WHO model, e.g. cardiorespiratory deaths, account for roughly 55% of all Chinese deaths [21]. This compares to only 42% of mortality in the United States in the same cardiorespiratory categories, despite much higher incidence of obesity in the United States [21]. The calculated mortality is somewhat higher than the 1.2 million deaths/year previously estimated from the Huai River study using Chinese air pollution measurements and mortality data [3,4,33].\n",
      "\n",
      "Though most of China is subject to potentially harmful levels of PM2.5, some large population centers (Chongqing, Wuhan, Chengdu) emit less than half the PM2.5 of others. Among northeastern cities, Beijing has relatively low emissions except for NO2. Low SO2 fluxes may indicate cities that benefit from lower coal usage or better smokestack pollution controls. Compared to natural gas, coal produces 150 to 400 times more PM for the same energy delivered [34,35]. China has plans for new coal plants in the next decade that could effectively double their coal consumption [36], potentially exacerbating the problem of air pollution. A table of pollution concentrations and fluxes by province and prefecture is included in S1 Table.\n",
      "\n",
      "The methods of this study should be applicable to air quality monitoring in other regions of the world. However, these techniques require an extensive air-quality monitoring network with frequent updates (e.g. hourly), and such networks presently exist in only a few places. We hope that other countries will follow China’s lead and provide both extensive and transparent real-time air quality monitoring.\n",
      "\n",
      "\n",
      "Supporting InformationS1 Text\n",
      "\n",
      "\n",
      "Supplemental Methods.\n",
      "\n",
      "Additional discussion of methods, data handling, and validation.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.S1 Table\n",
      "\n",
      "\n",
      "Province and prefecture pollution and mortality data.\n",
      "\n",
      "Provides pollution concentration and flux data broken down by geographic regions within China. Also provides the details of the associated mortality calculation.\n",
      "\n",
      "(XLSX)\n",
      "\n",
      "Click here for additional data file.S2 Table\n",
      "\n",
      "\n",
      "Correlation vs. distance and lifetime of air pollutants in China.\n",
      "\n",
      "Summary of the empirically determined correlation vs. distance functions for the five pollutants discussed in this study and the effective residence time in the atmosphere of their pollutant plumes in the absence of rain or snow. The correlation functions generally consist of a short range component (influenced largely by source distributions) and a long-range component (influenced by weather patterns). The effective pollutant plume lifetime shows the value empirically estimated for this study. Ranges in brackets indicate alternative lifetimes making different assumptions as discussed in the Supplemental Methods (S1 Text).\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.S1 Movie\n",
      "\n",
      "\n",
      "Animation of PM2.5 Concentrations.\n",
      "\n",
      "Movie showing the evolution of PM2.5 concentrations, as inferred by this study, across Eastern China during the four month study interval. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts based on 24-hour exposure. Green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups”, red is “Unhealthy”, purple is “Very Unhealthy”, and dark red is “Hazardous”. Concentrations higher than the upper limit of the maximum EPA classification are shown in gray.\n",
      "\n",
      "(MP4)\n",
      "\n",
      "Click here for additional data file.S1 Fig\n",
      "\n",
      "\n",
      "Average wind and rain patterns in Eastern China.\n",
      "\n",
      "(Left) Average wind speed pattern across China and surrounding areas during the period of this study according to Global Forecast System data at 80 m altitude. Arrow lengths are proportional to wind speed and the legend indicates the size of arrow consistent with a 10 km/hr average wind speed. (Right) Percentage of hours for each grid cell where rain was experienced during the study period based on observations from the Tropical Rainfall Measurement Mission satellite data. Maps were prepared with MATLAB with political boundaries from the Database of Global Administrative Areas (GADM version 2).\n",
      "\n",
      "(TIF)\n",
      "\n",
      "Click here for additional data file.S2 Fig\n",
      "\n",
      "\n",
      "Pollution concentration correlation vs. distance.\n",
      "\n",
      "Summary of empirically estimated correlation vs. distance functions for the five pollutants studied. Same values as S2 Table.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.S3 Fig\n",
      "\n",
      "\n",
      "Empirical fit of PM2.5 correlation data.\n",
      "\n",
      "Heat map of correlation vs. distance generated by all possible pairwise comparisons of PM2.5 data from stations in this study. Brighter colors indicate more frequently occurring combinations of R-value and distance, and are normalized for the abundance of comparisons at each distance. The red curve indicates the correlation vs. distance model fit to this data and described in S2 Table.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.S4 Fig\n",
      "\n",
      "\n",
      "Impact of assumed plume lifetime on PM2.5 source distribution.\n",
      "\n",
      "Maps comparing calculated PM2.5 source fluxes for different assumed values of the effective plume lifetime. The upper right panel corresponds to the data reported in the main paper, and has a lifetime chosen such that 5% of the field area is allowed to be negative (same as Fig 4 and S8 Fig). The upper left panel, corresponds to choosing a decay lifetime such that 2% of the flux field is negative, the lower left panel has 15% negative area, and the lower right panel has no plume decay in the absence of rain (infinite lifetime). Rain events are explicitly excluded from consideration, and the reported flux aims to capture the man-made sources of pollution. In general, assuming a shorter plume lifetime implies higher and more diffuse fluxes in order to maintain the observed concentration patterns. By contrast, if the pollutant plumes are assumed to have a very long lifetime then the apparent man-made flux will implausibly turn negative over substantial regions. Though quantitatively important, the choice of effective lifetime has little qualitative impact on the distribution of large sources. The regions of locally higher flux remain similar under all scenarios.\n",
      "\n",
      "(TIF)\n",
      "\n",
      "Click here for additional data file.S5 Fig\n",
      "\n",
      "\n",
      "Comparison of PM2.5 patterns in early and late half of study.\n",
      "\n",
      "Maps comparing PM2.5 concentrations (top) and fluxes (bottom) during the first half of the sampling period (left column) and the second half of the sampling period (right). PM2.5 concentrations are presented using the same health impact associated color scheme as Fig 3.\n",
      "\n",
      "(TIF)\n",
      "\n",
      "Click here for additional data file.S6 Fig\n",
      "\n",
      "\n",
      "Comparison of PM2.5 patterns from half-density reconstructions.\n",
      "\n",
      "The collection of available stations was randomly assigned to two groups, and the analyses for PM2.5 concentrations (top) and fluxes (bottom) were repeated in full for both groups. These independent subsamples result in similar concentration and flux patterns, implying the reconstructions are likely to be stable with respect to data selection.\n",
      "\n",
      "(TIF)\n",
      "\n",
      "Click here for additional data file.S7 Fig\n",
      "\n",
      "\n",
      "Average PM2.5 concentration across China.\n",
      "\n",
      "Larger version of panel from Fig 3 presenting PM2.5 concentrations across China. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Based on 24-hour exposure green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups” and red is “Unhealthy”.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.S8 Fig\n",
      "\n",
      "\n",
      "PM2.5 sources across China.\n",
      "\n",
      "Larger version of panel from Fig 4 presenting PM2.5 source fluxes across China.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.S9 Fig\n",
      "\n",
      "\n",
      "Average PM10 concentration across China.\n",
      "\n",
      "Larger version of panel from Fig 3 presenting PM10 concentrations across China. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Based on 24-hour exposure green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups” and red is “Unhealthy”.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.S10 Fig\n",
      "\n",
      "\n",
      "PM10 sources across China.\n",
      "\n",
      "Larger version of panel from Fig 4 presenting PM10 source fluxes across China.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.S11 Fig\n",
      "\n",
      "\n",
      "Average SO2 concentration across China.\n",
      "\n",
      "Map of SO2 concentrations across China. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Based on 24-hour exposure green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups” and red is “Unhealthy”. The limited coverage of this map is a consequence of the short correlation length for SO2 (S2 Table; S2 Fig), which limits the distance over which one can usefully interpolate SO2. In part, the short correlation length is a consequence of the short lifetime of SO2 in the atmosphere, which makes estimates far from measurement sites highly uncertain.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.S12 Fig\n",
      "\n",
      "\n",
      "SO2 sources across China.\n",
      "\n",
      "Larger version of panel from Fig 4 presenting SO2 source fluxes across China.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.S13 Fig\n",
      "\n",
      "\n",
      "Average NO2 concentration across China.\n",
      "\n",
      "Map of NO2 concentrations across China. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Based on 24-hour exposure green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups” and red is “Unhealthy”.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.S14 Fig\n",
      "\n",
      "\n",
      "NO2 sources across China.\n",
      "\n",
      "Larger version of panel from Fig 4 presenting NO2 source fluxes across China.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.S15 Fig\n",
      "\n",
      "\n",
      "Average O3 concentration across China.\n",
      "\n",
      "Larger version of panel from Fig 3 presenting O3 concentrations across China. Concentrations are shown using color gradients and contour lines; the colors (green, yellow, etc.) represent US EPA qualitative health impacts. Based on 24-hour exposure green is “Good”, yellow is “Moderate”, orange is “Unhealthy for Sensitive Groups” and red is “Unhealthy”.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.S16 Fig\n",
      "\n",
      "\n",
      "O3 sources across China.\n",
      "\n",
      "Map of O3 source fluxes across China.\n",
      "\n",
      "(EPS)\n",
      "\n",
      "Click here for additional data file.\n"
     ]
    }
   ],
   "source": [
    "print(art.extract_text())"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
