{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pymongo\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer, TfidfVectorizer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.display import clear_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pubcrawler.article import Article\n",
    "from bs4 import BeautifulSoup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "from reporter import Reporter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "client = pymongo.MongoClient(\"localhost\", 27017)\n",
    "articles = client.pmc.articles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"terms\") as f:\n",
    "    terms = [line.strip() for line in f.readlines()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "art = articles.find_one()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To sample a random article:\n",
    "art = articles.aggregate([\n",
    "    { \"$sample\": { \"size\": 1 } }\n",
    "]).next()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_articles = list(articles.find({}))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "corpus = [article[\"extracted_text\"] for article in all_articles]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['field work',\n",
       " 'fieldwork',\n",
       " 'field study',\n",
       " 'field site',\n",
       " 'field area',\n",
       " 'study site',\n",
       " 'study location',\n",
       " 'study area',\n",
       " 'research site',\n",
       " 'research location',\n",
       " 'sampling site',\n",
       " 'sampling location',\n",
       " 'sampling area']"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [],
   "source": [
    "vectorizer = CountVectorizer(stop_words=\"english\")\n",
    "counts = vectorizer.fit_transform(corpus)\n",
    "transformer = TfidfTransformer(smooth_idf=True, sublinear_tf=True)\n",
    "tfidf = transformer.fit_transform(counts)\n",
    "word_features = vectorizer.get_feature_names()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[322204 256677 233299 461838 561712 971313 610883 971311 778468 562046]\n",
      "citi\n",
      "botswana\n",
      "batswana\n",
      "fhb\n",
      "hrdc\n",
      "seminars\n",
      "jfm\n",
      "seminar\n",
      "nonissues\n",
      "hru\n",
      "[ 790005 1101959 1073438  809599  447313  282914 1122152  243340  252286\n",
      "  437990]\n",
      "ob\n",
      "uw\n",
      "treadmill\n",
      "ow\n",
      "exercise\n",
      "calories\n",
      "walking\n",
      "bicycle\n",
      "bmi\n",
      "ergometer\n",
      "[221560 327330 816278 351788 957664 661510 458111 837499 584310 816496]\n",
      "avb\n",
      "cmr\n",
      "pacemaker\n",
      "cs\n",
      "sarcoidosis\n",
      "lge\n",
      "fdg\n",
      "pet\n",
      "implantation\n",
      "pacing\n",
      "[980339 698890 585305 577284 312817 480478 469401 687615 497705 407725]\n",
      "shlomi\n",
      "mba\n",
      "inactivereactions\n",
      "iems\n",
      "checkmodelconsistency\n",
      "fva\n",
      "fluxes\n",
      "mahadevan\n",
      "generic\n",
      "duarte\n",
      "[660260 314954 371466 248354 519958 725121 217678 456655 809702 696136]\n",
      "leukemia\n",
      "childhood\n",
      "daycare\n",
      "birth\n",
      "greaves\n",
      "miscarriage\n",
      "attendance\n",
      "father\n",
      "ownership\n",
      "maternal\n",
      "[ 762901  627716  988483  799108 1100513  878864  479228 1070437  266491\n",
      " 1016587]\n",
      "nemesis\n",
      "ketcher\n",
      "sketch\n",
      "openbabel\n",
      "users\n",
      "project\n",
      "functionality\n",
      "trajectory\n",
      "build\n",
      "structure\n",
      "[ 953659  541212 1089785  932509  990020  740154  152392  220204  220195\n",
      "  188483]\n",
      "saids\n",
      "hce\n",
      "uctd\n",
      "ro60\n",
      "sle\n",
      "mspa\n",
      "abatacept\n",
      "autoantibody\n",
      "autoantibodies\n",
      "analyte\n",
      "[ 532795 1161713  216133  563662  790067  669361 1168400  827364  985918\n",
      "  584890]\n",
      "h2ax\n",
      "γh2ax\n",
      "atm\n",
      "hsmg\n",
      "obb\n",
      "lmp400\n",
      "λpp\n",
      "pbmcs\n",
      "singleplexed\n",
      "imr90\n",
      "[1100537  281962  107035  587693  117830  663172  282243  793678    4301\n",
      "   49013]\n",
      "usesharnett\n",
      "calcutta29010800\n",
      "5197960\n",
      "indmedgaz72107\n",
      "618http\n",
      "lieut\n",
      "calcuttam\n",
      "officiating\n",
      "0024original\n",
      "192762\n",
      "[1063184  778897  725003  701117  792744  829347  976057  790225  593547\n",
      "  865742]\n",
      "tlif\n",
      "nonobese\n",
      "mis\n",
      "mcs\n",
      "odi\n",
      "pcs\n",
      "sf\n",
      "obese\n",
      "interbody\n",
      "postoperative\n"
     ]
    }
   ],
   "source": [
    "for i in range(10):\n",
    "    word_indexes = tfidf.getrow(i).todense().A1.argsort()[-10:][::-1]\n",
    "    print(word_indexes)\n",
    "    for i in word_indexes:\n",
    "        print(word_features[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "x = \"background of infectious disease mosquito sick\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[736358 982377 588862 227412 394134 391707 391705 391736 391735 391734]\n",
      "mosquito\n",
      "sick\n",
      "infectious\n",
      "background\n",
      "disease\n",
      "dimiker\n",
      "dimidschstein\n",
      "diminutesished\n",
      "diminuted\n",
      "diminuta\n"
     ]
    }
   ],
   "source": [
    "y = vectorizer.transform([x])\n",
    "z = transformer.transform(y)\n",
    "z2 = z.todense().A1.argsort()[-10:][::-1]\n",
    "print(z2)\n",
    "for i in z2:\n",
    "    print(word_features[i])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get the tf-idf scores for documents matching a certain term"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "term = \"field area\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['study site']"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_articles[0][\"text_matches\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [],
   "source": [
    "match_index = []\n",
    "for i, article in enumerate(all_articles):\n",
    "    if article.get(\"text_matches\"):\n",
    "        if term in article.get(\"text_matches\"):\n",
    "            match_index.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "Introduction\n",
      "\n",
      "A growing body of research on the communicative behaviour of non-human primates has demonstrated that their vocalisations can convey a considerably rich amount of information that is meaningful to receivers (e.g. [1]). For instance, field experiments with various primate species have shown that acoustically distinct alarm calls can inform listeners about specific types of dangers (e.g. [2]–[5]). In some species, there is evidence that signallers produce strings of acoustically variable calls composed in context-specific ways (e.g. [6]–[8]). For example, black-and-white Colobus monkeys (Colobus polykomos, C. guereza) produce two types of vocalisations to predators, which are arranged in event-specific sequences that are seemingly meaningful to others [8].\n",
      "\n",
      "Food discovery is another event type during which some primates produce highly context-specific vocalisations. Since food is often patchily distributed and seasonally dispersed, food calls can provide listeners with a useful means to access foraging patches more effectively, while callers appear to gain mainly social benefits [9]–[10]. The production of food-associated calls is not restricted to primates but found in other mammals and some birds, e.g. Gallus gallus[11], although relatively little is still known about the type of information conveyed by the calls. At the simplest level, food calls are a basic physiological response indicating that the caller has found something desirable, as demonstrated by receivers approaching food calls more rapidly than other calls [12]–[13] or by triggering foraging behaviour [14]. In some species, food calls appear to provide more detailed information about the food item itself, such as its quality or divisibility, which can be encoded by changes in call rates [12], [15]–[17] or acoustic structure [18]–[19]. For example, Rhesus monkeys (Macaca mulatta) produce up to five different food-associated call types [20], which receivers discriminate based on differences in their referential features, rather than their acoustic properties alone [21].\n",
      "\n",
      "Among the great apes, chimpanzees (Pan troglodytes) produce specific calls when discovering food, the ‘rough grunts’ [22]–[23]. The morphology of this call type co-varies with the caller's personal food preference both in captivity and the wild [24]. In a naturalistic playback experiment, it was demonstrated that acoustic variation in this call influences the foraging decisions of receivers, suggesting that the acoustic structure of this graded signal provides meaningful information to other chimpanzees about the quality of food encountered by the caller [25].\n",
      "\n",
      "What exactly governs receiver responses, however, is a matter of ongoing debate. For instance it is not clear whether receivers respond directly to the calls' physical features or their referential nature, i.e. the causal relation between calls and contexts [26]–[27]. Similarly, signalling is often said to be non-cooperative with signallers merely producing ‘natural’ information in response to biologically relevant events, while any representational content is largely generated by the listeners [28]. These problems are unsolved because the psychological states experienced by primates during call production and perception are rarely investigated. In one recent study, however, food call production in wild chimpanzees was found to co-vary with the presence and arrival of long-term allies, suggesting these calls may be used as a flexible social strategy [10].\n",
      "\n",
      "Very little is known about how our other closest relative, the bonobo (Pan paniscus), naturally communicates about events in the external world. This is despite the fact that some individuals have been remarkably capable in mastering artificial language systems [29]–[30]. In one recent systematic study, Clay & Zuberbühler [31] demonstrated that bonobos also vocalize upon encountering food, but that there are important differences between the two Pan species. Whilst both species produce food ‘grunts’, bonobos additionally give four acoustically distinguishable tonal calls when finding food (barks, peeps, peep-yelps, and yelps), which although lying on a graded scale, can be statistically discriminated from one another [31]. Barks are longest in duration, characterised by a distinctive pointed shape and numerous visible harmonic bands. Whilst peeps are also high pitched, they are temporally shorter than barks (and all other calls), with few harmonic bands and had a flat frequency contour. Yelps and peep-yelps are lower in pitch and although share acoustically similarities, yelps possess a marked downward stroke frequency contour, in contrast to the arched contour of the peep-yelps. In terms of production, peeps and barks are most frequently, but not exclusively, given to preferred foods, while yelps and grunts are more often, but not exclusively, given to less preferred foods. Peep-yelps are produced broadly, although they also tend to occur more to mid- and lesser-preferred foods. In sum, the link between individual call types and perceived food quality is only probabilistic in bonobos. One important consequence of this is that different food calls themselves do not appear to allow listeners to make strong predictions about the type of food encountered by the caller. However, in contrast to chimpanzees, bonobos regularly combine different call types together into longer mixed sequences. The composition of these food call sequences related reliably to food quality, suggesting that listeners can gain information from attending to the call sequences. The hypothesis of meaningful call combinations has been put forward before for bonobos, but has never been tested formally [32]–[33].\n",
      "\n",
      "In the current study, we examined whether listeners were able to extract meaningful information relating to food quality by attending to the composition of these heterogeneous call sequences. To this end, we conducted naturalistic playback experiments in which subjects heard different types of food all sequences, and their subsequent foraging responses were analysed.\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "The study was conducted at Twycross Zoo, UK, between April and July 2009. Individuals were permanently separated into two subgroups that occupied different indoor facilities but shared the same outdoor area via two separate doors in the morning or afternoon, respectively (fig. 1; table 1). We conducted playback experiments in which we broadcast food-calls of an individual of the morning subgroup responding to either high-quality food (kiwi) or low-quality food (apple) to all members of the afternoon subgroup before releasing them into the outdoor enclosure. Our general aim was to simulate a morning group individual discovering food of high or low quality in the outdoor area shortly before the release of the afternoon subgroup. We also included a control condition in which no food calls were played. We then assessed the foraging behaviours of receivers at previously learned locations for high-quality and low-quality food in the outdoor enclosure (fig. 1).\n",
      "\n",
      "10.1371/journal.pone.0018786.g001Figure 1 Schematic layout of the bonobo facility at Twycross Zoo, including location of playback equipment and artificial food sites.\n",
      "\n",
      "Table 1\n",
      "Subgroup A (call producers)\tSubgroup B (call receivers)\n",
      "Name\tID\tSex\tAge\tName\tID\tSex\tAge\n",
      "Kakowet\tKT\tM\t07.06.1980\tDiatou\tDT\tF\t21.10.1977\n",
      "Banya\tBY\tF\t16.02.1990\tKichele\tJS\tM\t02.08.1980\n",
      "Keke\tKK\tM\t02.01.1994\tJasongo\tKH\tF\t19.04.1989\n",
      "Maringa\tMR\tF\t05.05.1998\tCheka\tCK\tF\t18.03.1996\n",
      "Bokela\tBK\tF\t14.10.2003\tLuo\tLU\tM\t01.12.2002\n",
      "Gemena\tGM\tF\t07.11.2005\n",
      "\n",
      "\n",
      "Hearing food-associated calling sequences influenced foraging behaviour\n",
      "\n",
      "Following release, there was a strong baseline preference for the highly preferred ‘kiwi’ field. This was particularly evident in the control condition (when no food was presented), in which individuals were more likely to visit the kiwi field first and more often, as well as devoting more foraging effort to it compared to the apple field (fig. 2). Despite this baseline bias, playbacks of food-associated calls had an overall significant effect on the individuals' first choice of fields (χ2 (2) = 16.347, p<.001; Pearson chi-square, two-tailed; fig. 2a). Playback of call sequences originally given to kiwi resulted in an increase in first visits to the kiwi field, compared to baseline or apple trials. However, this change failed to reach significance due to a ceiling effect caused by the strong baseline bias for kiwi (First arrivals to kiwi site per individual: median N trials: control condition = 3.0 (50% of trials); kiwi playback condition = 6.0 (86%); apple playback condition = 5.0 (50%), all one-way χ2 tests: p>.05). Playback of call sequences originally give to apple resulted in a significant increase in the number of first visits to the apple field, compared to baseline or kiwi trials (First arrivals to apple site, per individual: median N trials: control condition = 0.0; kiwi playback condition = 0.0; apple playback condition = 4.0 (40%); both control and kiwi vs. apple: χ2 (1, N = 17) = 13.235, p<.001, with Bonferroni corrected alpha .0169). Hearing food-associated call sequences, in other words, influenced the bonobos' foraging decisions against their pre-existing food preference biases. We conducted this first analysis at the group level because individuals almost always foraged as a cohesive unit and, when released, entered the outside enclosure almost simultaneously (table S1).\n",
      "\n",
      "10.1371/journal.pone.0018786.g002Figure 2 Box plots indicating foraging responses of bonobos (N = 4) following playbacks of food-associated calls given to high value (kiwi) or low value (apple) foods.(a) Site of first entry expressed as a median proportion of the individual's median choices per condition; (b) median number of visits per trial; (c) median time spent foraging following playback (s). Box plots illustrate medians, inter-quartile ranges, and highest and lowest values, excluding outliers.\n",
      "\n",
      "Next, we determined whether hearing playbacks influenced the number of visits the group made to the two fields (fig. 2b). Again, we found a significant effect of playback condition on the median number of visits made by the group to both the kiwi field (χ2 = 6.486, df = 2, p = .034; two-tailed exact Kruskal-Wallis test) and the apple field (χ2 = 10.532, df = 2, p = .002; two-tailed exact Kruskal-Wallis test). Post-hoc, pair-wise comparisons using a Bonferroni correction (corrected alpha = .0169) revealed that individuals visited the ‘kiwi’ field more often after hearing ‘kiwi’ call sequences compared to control trials (mediancontrol = 0.5; mediankiwi = 1.0; medianapple = 1.0; N visits to kiwi field, ‘kiwi’ playback vs. control: Mann-Whitney U = 4.5, p = .015). Conversely, we found that individuals visited the apple field more often after hearing playback of ‘apple’ call sequences compared to the control condition (mediancontrol = 0.0; mediankiwi = 0.5; medianapple = 1.0; N visits to apple field, ‘apple’ playback vs control: U = 3, p = .002).\n",
      "\n",
      "Finally, hearing playbacks of food-associated call sequences had a significant effect on the foraging time devoted by the group at both the kiwi site (χ2 = 6.902, df = 2, p = .026; two-tailed exact Kruskal-Wallis test) and the apple site (χ2 = 10.876, df = 2, p = .002; two-tailed exact Kruskal-Wallis test; fig. 2c). Pair-wise comparisons (Bonferroni corrected alpha = .0169) revealed that individuals spent more time at the kiwi location after hearing ‘kiwi’ call sequences compared to control condition (mediancontrol = 2.25s; mediankiwi = 16.50s; medianapple = 5.75s; Kiwi site, ‘kiwi’ playback vs. control: Mann-Whitney U = 5, p = .022,) or hearing ‘apple’ call sequences (‘kiwi’ playback vs. ‘apple’ playback: U = 15.5, p = .058). Likewise, individuals spent more time in the apple field after hearing playbacks of ‘apple’ call sequences compared to control trials (mediancontrol = 0.0s; median apple = 9.5s; mediankiwi = 1.5; apple field, ‘apple’ playback vs. control: Mann-Whitney U = 6, p = .015). Although there was a trend for spending more time foraging at apple after hearing apple playbacks compared to kiwi playbacks, the result did not reach significance (apple vs. kiwi playbacks: U = 20, p>.05).\n",
      "\n",
      "All previous analyses were based on non-parametric comparisons at the group level. We selected this analysis strategy to avoid problems with data interdependency and type-two clustering errors [34] at the cost of a substantial reduction in statistical power. Thus, in a second set of analyses we relied on Generalized Linear Mixed Models based on Poisson distributions and log link functions (GLMM), which allowed us to maximise the amount of data available by considering the contributions of individuals. We accounted for individual identity by entering it as a random factor, although it may be argued that combining individuals into one model will not completely eliminate the problem of interdependency, The GLMM procedure is an extension of the more widely known General Linear Model (GLM) and is particularly useful for this study because it relaxes assumptions of normal data distribution and an identity link [35].\n",
      "\n",
      "This GLMM analysis revealed the same main effects, with a significant interaction between playback type and the number of visits to the two fields (two-tailed GLMM, F (2,178) = 5.037, p = .007; fig. 2b; table S2). Pair-wise comparisons revealed that individuals visited the kiwi field significantly more often after hearing ‘kiwi’ call sequences compared to control trials (p = 0.028). They also visited the apple field significantly more often after hearing ‘apple’ call sequences compared to control and kiwi conditions (p<.001; p = 0.008, respectively).\n",
      "\n",
      "Finally, hearing playbacks of food-associated call sequences had a significant effect on the foraging time devoted by each of the four individuals in the two fields (two-tailed GLMM: F (2, 178) = 120.772, p<.001; fig. 2c, table S3). Pair-wise comparisons revealed that individuals spent more time at the kiwi location after hearing ‘kiwi’ call sequences than ‘apple’ call sequences or compared to control trials (both p<.001). Likewise, individuals spent more time in the apple field after hearing playbacks of ‘apple’ call sequences compared to ‘kiwi’ call sequences or compared to control trials (both p<.001).\n",
      "\n",
      "\n",
      "Foraging errors and integration across the call sequence\n",
      "\n",
      "A key indicator of representationally-based signal processing is that subjects sometimes make mistakes, particularly with signals that are ambiguous or only weakly correlated with specific external events [28]. In our sample, some call sequences were better indicators of high or low food quality than others in terms of call composition, suggesting that if subjects made mistakes then this should happen in response to the more ambiguous sequences (e.g. visiting the apple field after hearing a kiwi sequence). To address this, we assigned a cumulative value to each sequence, based on its call composition. Each call contributed with a value that reflected its association strength with high preference food (table 2). For instance, in natural calling sequences, ‘barks’ were given six times more frequently to high than low preference food (proportion of ‘barks’ in calling sequences to high preference foods = 0.24, vs. low-preference foods = 0.04). Similarly, peeps were given 1.86 times more frequently to high than low preference food (proportion of ‘peeps’ in calling sequences to high preference foods = 0.52 vs. low-preference food = 0.28), and so on. These relative differences resulted in the following cumulative values: B = 6.00, P = 1.86, PY = 0.52, Y = 0.12. We also assigned an ordinal scale, based on the strength of their relationship to high-preference foods, with ‘barks’ being given most frequently to high-preference foods and yelps most infrequently: B = 4, P = 3, PY = 2, Y = 1 (see fig. 3, table 3).\n",
      "\n",
      "10.1371/journal.pone.0018786.g003Figure 3 Scatter plot showing the relationship between the food type encountered (highly preferred = kiwi, less preferred = apple) and the cumulative value of the stimuli sequence.Calls were assigned a cardinal score based on how frequently they were produced in response to high vs. low value food (e.g. barks were six times more frequent to high than low preference foods, so that: B = 6.00, P = 1.86, PY = 0.52, Y = 0.12). The relationship between foraging responses and stimuli scores are indicated in table 3.\n",
      "\n",
      "Table 2\n",
      "Sequence\tFood type\tCall type\n",
      "Bark\tPeep\tPeep-yelp\tYelp\tGrunt\n",
      "Natural\tHigh\t0.24\t0.52\t0.22\t0.03\t0.00\n",
      "Low\t0.04\t0.28\t0.42\t0.26\t0.02\n",
      "Playback\tHigh\t0.29\t0.50\t0.18\t0.04\t0.00\n",
      "Low\t0.05\t0.13\t0.40\t0.43\t0.00\n",
      "\n",
      "Table 3\n",
      "Signaller behaviour\tReceiver foraging effort (s)\n",
      "Food\tSequence\tCVO\tCVC\tKiwi field\tApple field\tKiwi bias\n",
      "Kiwi\tB B P B\t15\t19.86\t21.0\t2.5\t9.4\n",
      "Kiwi\tB B P B\t15\t19.86\t6.5\t2.5\t3.6\n",
      "Kiwi\tB B P PY\t13\t14.38\t28.3\t5.0\t6.7\n",
      "Apple\tPY B B PY\t12\t13.05\t20.0\t12.0\t2.7\n",
      "Kiwi\tP P PY P\t11\t6.10\t79.0\t18.8\t5.2\n",
      "Kiwi\tPY P P P\t11\t6.10\t20.8\t1.8\t12.6\n",
      "Kiwi\tP PY PY P\t10\t4.76\t1.3\t2.5\t1.5\n",
      "Kiwi\tP P PY Y\t9\t4.35\t16.5\t6.5\t3.5\n",
      "Apple\tPY P PY PY\t9\t3.43\t14.8\t3.3\t5.5\n",
      "Apple\tY PY PY P\t8\t3.02\t0.0\t15.8\t1.0\n",
      "Apple\tY PY P Y\t7\t2.61\t9.3\t2.0\t5.7\n",
      "Apple\tPY PY Y PY\t7\t1.69\t5.8\t20.8\t1.3\n",
      "Apple\tY P Y Y\t6\t2.20\t6.5\t14.3\t1.5\n",
      "Apple\tY Y Y P\t6\t2.20\t3.8\t40.8\t1.1\n",
      "Apple\tPY PY Y Y\t6\t1.28\t9.5\t9.0\t2.1\n",
      "Apple\tPY Y PY Y\t6\t1.28\t6.5\t17.3\t1.4\n",
      "Apple\tPY Y Y Y\t5\t0.87\t2.5\t10.3\t1.2\n",
      "\n",
      "There was a significant positive correlation between subjects’ foraging effort in the kiwi field and the overall cumulative food value as assessed by the composition of the sequence (time spent: cardinal scale Spearman's rho: N = 17 rs = 0.585, p = .014, fig. 3; ordinal scale: N = 17, rs = 0.575, p = .016). Inspection at the level of individual trials indicated an almost perfect separation of sequences given to apples and kiwis by the cumulative sequence value generated by the constituent calls. One exception was a call sequence given to apples (PY-B-B-PY), which interestingly also triggered almost twice as much searching in the (wrong) kiwi compared to the apple field. Also interesting were two responses to kiwi sequences, which only triggered weak searching in the kiwi field. However, in both cases, search effort in the apple field was also unusually low, suggesting that subjects were generally unmotivated to forage (table 3). In sum, the foraging effort was a strong reflection of the cumulative ‘good food’ score encoded by the sequence.\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "Human-enculturated bonobos have long been known for their extraordinary representational and communication skills [29]–[30], [36]–[37], but their natural communication behaviour has hardly been investigated. Our study provides progress to this end in showing that bonobos can increase their foraging success by attending to each other's call sequences. Our key finding was that subjects were able to direct their foraging effort to specific locations according to the call sequence presented to them. Whilst we found an unsurprising baseline preference to the high-preference food site, we found that playbacks of high-preference food call sequences resulted in an even greater amount of foraging effort at this site, indicating the calls were meaningful to the receivers. Furthermore, although lack of interest at the low-food preference apple site was to be expected (see baseline trials) we found a significant increase in search effort at this site that only occurred after hearing sequences associated with low-preference food. This result further suggests that individuals incorporated information extracted from the food call sequences to optimise their foraging strategies, in some cases against pre-existing foraging biases.\n",
      "\n",
      "Although it would be interesting to investigate the mechanism of decision making within the group, we were not able to address this point in our study due to the constraints of working with subjects that were part of a public display (with Zoo regulations prohibiting separation). It is likely that there were individual differences in the ability to assign meaning to the different call sequences used as playback stimuli, either due to individual differences in knowledge or other more immediate factors, such as receiver attention or motivation during the trials. Despite these constraints, we are able to draw the conservative conclusion that at least one individual in our group was able to comprehend the information conveyed by the call sequences, though the results of some trials suggested that several or all group members made their own independent foraging decisions prior to being released into the enclosure. For example, we found that, across trials, different individuals arrived at the food sites first and, in some trials, individuals diverged in their first choices (table S1).\n",
      "\n",
      "Our results also show that, although chimpanzees and bonobos are phylogenetically closely related, they appear to communicate about food in considerably different ways. In contrast to chimpanzees, who produce an acoustically graded call type that co-varies with food quality [26], bonobos regularly mix several acoustically distinct food call types into heterogeneous strings of vocalisations. Rather than at the level of individual calls, food quality appears related to the probabilistic composition of heterogeneous call sequences [31]. Results from our playback experiment indicate that rather than attending to individual call types, receivers took into account the relative proportions of different calls within the sequence and extracted meaning by integrating information from across the call units.\n",
      "\n",
      "In addition, the generation of more foraging errors in structurally ambiguous call sequences (which were less strongly indicative of high or low preference foods) indicated that the information extracted from the stimuli sequences was influencing the foraging decisions of the receivers. In a recent discussion, Stegmann [28] argued that in contrast to ‘natural information’, which does not allow for errors, the generation of misrepresentations and errors is a defining feature of what we consider as ‘semantic information’ in animal signals.\n",
      "\n",
      "Whilst there is a growing body of evidence that numerous monkey species produce strings or sequences of acoustically variable calls composed in context-specific ways, evidence for meaningful signal combinations in apes has been poor (although see [38]). A recent study of gorilla gestural sequences failed to find evidence of syntactic organisation or corresponding semantic content [39]. Results from the current study provide the first empirical evidence that call combinations do play a role in bonobo communication in the foraging context. However, it is important to note that we also did not find any evidence for syntactic rules or that the sequencing structure itself was itself semantically relevant. Thus, although call combinations appear to represent a useful means of communicating information in bonobos, the manner in which bonobos use call combinations contrasts the way linguistic units are combined and structured in human language. This finding highlights the importance of studying non-human primate communication as a means to identify the features of the language faculty that are uniquely human.\n",
      "\n",
      "One recurrent topic in the animal communication literature is whether signals given in response to external events, such as in this study, should be conceptualised as ‘referential’ or a mere readout of a caller's motivational or internal state [26], [40]–[41]. Great apes, especially chimpanzees and bonobos, are often described as exceedingly ‘emotional’, suggesting that arousal-based explanations may be more in line with the nature of the phenomenon described here (e.g. [26]). In particular, sequences containing a greater amount of calls with presumably high emotional valence may lead receivers to search at the high-value food site. Although we do not discount the integral role motivation plays in animal communication [26], [41], gaining meaningful measurements of internal state or arousal have so far proved very challenging, and thus it has often proved more empirically fruitful to focus on the relation between receiver response and external variables that can be manipulated and measured experimentally [1], [42]. Furthermore, even calls with presumably high motivational content (as may be the case for food discovery) are still able to inform receivers about the external world. This has been demonstrated by studies showing that, regardless of the signaller's motivational state during call production, calls can provide listeners with representational information about external objects and events, in way that can be studied experimentally [1], [27], [41], [43]–[45]. Recent work on the alarm call responses of meerkats (Suricata suricata) for example, has demonstrated that both emotional and referential information are encoded into the same signal and develop on different ontogenetic time scales [45]. Meaningful progress will focus more specifically on the motivational experience of the caller and how this influences signal production.\n",
      "\n",
      "How do receivers extract information from these call sequences? Are they attending to individual calls or do they perceive the sequence as a holistic unit? For example, it could be argued that an increasing number of high-pitched calls, such as peeps and barks, increases the perceived gestalt of the sequence in a discrete way, enabling individuals to make foraging decisions without paying attention to individual calls. Further research will be necessary to address this issue in more detail.\n",
      "\n",
      "Another related question is whether receivers process vocalisations purely based on their acoustic properties, or whether they attach some semantic value to them. For example, subjects may have learned the contingencies of this particular experiment, e.g. that high-pitched vocalisations were associated with food at one specific location, without any understanding of more general relations between a caller's perceived food quality and the vocal signals produced in response. However, research on food-associated calls of rhesus macaques (Macaca mulatta), another primate with a graded vocal system, has demonstrated that individuals categorise calls based on their meaning, not just their acoustic structure alone [31]. Whether bonobos process their own calls in the same way remains open as a topic for future research.\n",
      "\n",
      "A final unresolved question concerns the function of food call production in bonobos. In capuchins (Cebus capucinus), food calls are thought to provide ecological benefits, functioning to announce food ownership and a willingness to defend, thereby resulting in reducing foraging competition from others [46]. In red-bellied tamarins (Saguinas labiatus), it has been suggested that food calls are not solely a function of arousal in the presence of highly desirable food patches, but may provide social benefits by attracting allies, even at the cost of increasing feeding competition [9]. A similar effect has recently been indicated in wild male chimpanzees who were found to call more in the presence of close allies and even recommenced calling upon their arrival [10]. For bonobos, it has also been suggested that individuals receive benefits from producing food calls, for instance by attracting mates or potential allies [47]. Further work investigating the interplay and influence of social and ecological variables on the production of food-associated calls in bonobos is required to explore the adaptive significance of these calls in this species.\n",
      "\n",
      "\n",
      "Materials and Methods\n",
      "\n",
      "\n",
      "Study site and subjects\n",
      "\n",
      "The study was conducted at Twycross Zoo, UK, between April and July 2009.Individuals were permanently separated into two subgroups that occupied separate indoor facilities but shared the same outdoor area via two separate doors (fig. 1). Subgroup A consisted of 5 individuals (2 adult males and females, 1 juvenile female; range 6–29 years); subgroup B consisted of 6 individuals (1 adult male, 3 adult females, 1 juvenile male, 1 juvenile female; range 4–32 years; table 1). Each subgroup was housed in one of two separated heated indoor halls (62 m2) with additional sleeping areas (22 m2) and both facilities were separately connected to an outdoor enclosure (588 m2). There was no visual contact between indoor and outdoor enclosures, although vocalizations produced outside could be heard indoors. Both subgroups were fed a range of fruits and vegetables twice per day in a scatter feed. Water was freely available. Bonobos were provided with regular enrichment materials and feeds (such as branches, seeds, grapes or frozen juice) as well as supplements such as yoghurt, egg and bread.\n",
      "\n",
      "\n",
      "Ethical statement\n",
      "\n",
      "The Twycross Zoo Ethics and Management Committee and the Zoo Research Coordinator and gave full ethical approval to this behavioural, non-invasive study, which complied with the ethical guidelines set out by the British and Irish Association of Zoos and Aquariums (BIAZA). During all stages of the study, we took steps to ensure that the welfare of all animals was not compromised. No individual showed distress during any part of this study and their participation throughout was voluntary. In order to reduce stress and to comply with Zoo guidelines, we did not separate any individual in any stage of this study.\n",
      "\n",
      "\n",
      "Experimental Design\n",
      "\n",
      "The basic design was to simulate a member of the morning subgroup A finding food shortly before the midday switchover, in order to investigate whether this influenced the subsequent foraging behaviour of subgroup B members. The study consisted of four stages: (1) food preference tests, (2) recording of food-specific calling sequences, (3) establishment of two feeding areas, and (4) playback experiments.\n",
      "\n",
      "\n",
      "1. Food preference tests\n",
      "\n",
      "First, we conducted food preference tests to identify a highly valued and a lesser-valued food. Equal sized piles of two foods were placed next to each other on the ground and the first choice was recorded for each individual, repeated across four days, once per day. For each individual, we created a preference matrix based on the percentage of trials in which each food was selected over the other food type across the four sessions. These percentages were then combined to give a cumulative preference score per food, and ranked accordingly (table S4). From eight familiar foods, kiwis and apples were consistently ranked as high and low by all individuals, while both still reliably triggering food calls. We thus selected kiwis and apples as our experimental foods.\n",
      "\n",
      "\n",
      "2. Recording calls\n",
      "\n",
      "From April to May 2009, we recorded food-associated call sequences given by all individuals feeding in the outdoor enclosure (fig. 4). This allowed us to build up a sound library of call sequences given to kiwi and apples by individuals of the ‘morning’ subgroup for the subsequent playback experiments and to compare their behaviour with a previous study on bonobos [31] (table 2). We recorded vocalizations using a Sennheiser MKH816T directional microphone and a Marantz PMD660 solid-state recorder. Verbal comments were given and later transcribed. We transferred recordings onto a Toshiba Laptop (Equium 1.8 GHz) at a sampling rate of 44.1 kHz with 16-bit accuracy. To control for hunger levels, novelty, and other factors, we only recorded calls produced during first morning feeds. We excluded calls produced by individuals interacting with more than one type of food or when caller identity was uncertain. Calls were recorded from a range of locations from a distance of 2–15 m.\n",
      "\n",
      "10.1371/journal.pone.0018786.g004Figure 4 Spectrographic illustrations of two playback stimuli.(a) High value sequence originally given to kiwi consisting of bark/peep/bark/peep and (b) low value sequence originally given to apple consisting of peep-yelp/peep-yelp/yelp/yelp. Recordings of the corresponding call sequences are available as audio S1 and audio S2.\n",
      "\n",
      "\n",
      "3. Foraging training\n",
      "\n",
      "During this same period, we established two outdoor foraging patches for the afternoon subgroup in a daily foraging task. Before their midday release, a caretaker hid finely cut pieces (1 cm2 pieces, total 300 g) of either apple or kiwi in the grass in one of two 30 m2 fields so that they were not visible from a distance. The two fields were on slopes, equidistant to the door (21 m); the distance between them was 8 m. Both field areas were equal (length top = 6.5 m; width = 4.0 m; length bottom = 8.5 m), starting with a flat descent and finishing at the concrete border of the enclosure wall. Kiwi and apple feeds were presented in random order so individuals could not predict which patch was baited so had to inspect the two areas separately. Only one food type was ever provided and no other food or enrichment was provided during training. The keeper always visited both areas, even if no food was placed, to prevent individuals from learning noises associated with scattering food. There were 16 training days for each food type, and 10 control days during which no food was provided. We filmed the individuals' foraging behaviour and kept a daily record of each individual's food encounters (text S1, table S5). Individuals soon learned the two feeding fields and quickly formed a preference for the kiwi field.\n",
      "\n",
      "\n",
      "4. Playback experiments\n",
      "\n",
      "In the final step, we conducted playback experiments in which call sequences of members of the morning subgroup were played to individuals in the afternoon subgroup (text S1). The experimental routine was as follows. Around midday, the morning subgroup was brought inside for a seed feed. Live radio broadcasting was played via an inside keeper door to prevent subjects from the morning group from hearing the stimuli (i.e. their own calls) in the subsequent experiment. This was effective as no vocal responses were elicited from any individual during playback trials (except for one apple trial, which was excluded from analysis). Meanwhile, individuals from the afternoon subgroup were waiting to be released through their own door. Beforehand, three key manipulations were carried out. First, a keeper entered the outdoor enclosure from a side door to mimic placing food (none was provisioned). Individuals were familiar with this routine from the previous foraging training. They could not see the event, but could hear the associated sounds. After the keeper's return, subjects heard the opening sounds of the door, which connected the morning subgroup to the outdoor enclosure (to suggest a re-entry of the morning subgroup), although, in reality, no subject was released. A trial was conducted only if (a) no vocalizations had been produced by the morning subgroup for at least 1 minute, (b) individuals of the afternoon subgroup were waiting close (<1–2 m) to the door and not distracted by social activities (play, agonistic, sex) for at least 1 minute; (c) there was no rain or excessive wind outdoors. Communication between a keeper who stayed indoors with the bonobos and the experimenter, who stayed outdoors, was maintained with two-way radios. If these conditions were not met, the trial was either delayed or, in some cases, abandoned. If conditions were met, the afternoon subgroup then heard a 4 s playback of a series of four equally spaced calls extracted from a natural call sequence to either apple or kiwi (to simulate a morning subgroup member finding apple or kiwi) played from their outdoor enclosure (fig. S1). During control trials, all features of the procedure remained the same except that no stimulus was played. One minute after the playback (a sufficient time period for morning subgroup to ‘return’ indoors), the afternoon subgroup was released and their foraging behaviour was monitored for up to 10 min using a camcorder with additional verbal comments. We simultaneously recorded all vocal responses with professional sound recording equipment as previously described. During experimental and control trials, no food was ever provided on either field (to rule out visually-based foraging). To reduce the possibility of extinction, we interspersed a number of refresher days between each trial, i.e. between 1–4 days during which we provided either kiwi or apple pieces on the corresponding fields in random order (N = 28 total).\n",
      "\n",
      "Zoo regulations prohibited separation of group members (due to stress provoked) so all individuals were released simultaneously into the outdoor enclosure and behavioural response measures were collected while individuals interacted with each other as a group. A total of 28 trials were conducted; three were discarded due to poor weather (preventing the bonobos from being released), one due to unexpected vocalizations (see before), and one due to a communication problem between keeper and experimenter. The remaining 23 trials consisted of N = 10 apple, N = 7 kiwi, and N = 6 control trials, which were completed by four individuals (GM, CK, KH, LU).\n",
      "\n",
      "The remaining two individuals (DT, JS) were excluded. JS did not complete the training phase and was not motivated to go outdoors due to his low social rank. DT showed no evidence of having learned the food locations during the training phase and showed little interest during the playback phase, only completing 5 of 23 trials, not enough for statistical analyses.\n",
      "\n",
      "We extracted systematic data on three dependent variables across the different conditions: (a) patch first visited (kiwi vs. apple), (b) time spent actively foraging in each patch (time trespassing, sitting, resting, or sleeping were subtracted), (c) total number of visits per patch (N times entering and exiting the patch areas interrupted by at least one bout of foraging). Because data from individuals were interdependent (we were unable to separate individuals so all individuals foraged together), we conducted our principal analyses at the group level using the median scores for individuals combined per trial. The nature of the data distribution for this method meant that only non-parametric statistics were employed.\n",
      "\n",
      "Whilst measuring the central tendency of the group across trials reduces the problems of interdependency of the data and type-two clustering errors [34], the cost is a substantial reduction in statistical power. Furthermore, rather than using the foraging behaviours of receivers (upon which our hypothesis is based) as the unit of analysis, the unit of analysis becomes the trials in which the responses of a groups of receivers were measured. We therefore conducted a second analysis using Generalized Linear Mixed Models (Poisson distribution-log link) in order to address the problem of statistical power. The Generalised Linear Mixed Model Procedure (GLMM) is an extension of the General Linear Mixed Model, characterised by a flexible generalization of ordinary least squares regression. The procedure relaxes the assumptions of normal distribution and identity link [35], a crucial prerequisite for the ordinal data of this study. The GLMM procedure enables individual identity to be accounted for (by entering it as a random factor), although this does not completely address the problem of potential inter-dependency in the individuals' foraging decisions.\n",
      "\n",
      "\n",
      "Supporting InformationFigure S1\n",
      "\n",
      "Images depicting (a) the playback speaker positioned during the experimental phase, (b) the view of the sloped outdoor enclosure from the bonobo exit door.\n",
      "\n",
      "(TIF)\n",
      "\n",
      "Click here for additional data file.Table S1\n",
      "\n",
      "\n",
      "Order that individuals first arrived to one of the two fields per trial, with their choice of field indicated in parentheses (k = kiwi field, a = apple field). If two or three individuals arrived simultaneously, each of these individuals was given their number plus 0.5 or 0.3, respectively.\n",
      "\n",
      "(DOC)\n",
      "\n",
      "Click here for additional data file.Tables S2\n",
      "\n",
      "\n",
      "Median number of visits by each individual to the two fields after playbacks of food-associated call sequences. The top number indicates the median value, with the bottom numbers, indicating the 25 and 75 percentiles. PB  =  Playback condition.\n",
      "\n",
      "(DOC)\n",
      "\n",
      "Click here for additional data file.Table S3\n",
      "\n",
      "\n",
      "Mean time spent (sec) by each individual at the apple and kiwi slope after hearing food- associated call playbacks. The top number indicates the median value with the ± indicating the standard errors.\n",
      "\n",
      "(DOC)\n",
      "\n",
      "Click here for additional data file.Table S4\n",
      "\n",
      "\n",
      "Results of food preference tests conducted on two groups of captive bonobos at Twycross Zoo, UK.\n",
      "\n",
      "(DOC)\n",
      "\n",
      "Click here for additional data file.Table S5\n",
      "\n",
      "\n",
      "Direct and indirect experiences by subgroup B individuals during foraging training phase. Direct experience indicates a foraging event where the individual had physical contact with a food item at the location; indirect experience indicates a foraging event where the individual witnessed another individual eating or on contact with a food item, but themselves did not.\n",
      "\n",
      "(DOC)\n",
      "\n",
      "Click here for additional data file.Audio S1\n",
      "\n",
      "Recording of a call sequence produced by a bonobo in response to finding kiwi, a high-ranked food.\n",
      "\n",
      "(WAV)\n",
      "\n",
      "Click here for additional data file.Audio S2\n",
      "\n",
      "Recording of a call sequence produced by a bonobo in response to finding apples, a low-ranked food.\n",
      "\n",
      "(WAV)\n",
      "\n",
      "Click here for additional data file.Text S1\n",
      "\n",
      "(DOC)\n",
      "\n",
      "Click here for additional data file.\n"
     ]
    }
   ],
   "source": [
    "print(corpus[match_index[0]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [],
   "source": [
    "i = 593"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [],
   "source": [
    "article_tfidf = tfidf.getrow(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "dense = article_tfidf.todense()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0.01425685, 0.        , 0.        , ..., 0.        , 0.        ,\n",
       "       0.        ])"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dense."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0.01425685 0.         0.         ... 0.         0.         0.        ]]\n",
      "[632189 198563 471176 855415 471461 255138 282795 892066 973154 587331]\n",
      "kiwi\n",
      "apple\n",
      "food\n",
      "playback\n",
      "foraging\n",
      "bonobos\n",
      "calls\n",
      "py\n",
      "sequences\n",
      "individuals\n",
      "[[0.        0.0027204 0.        ... 0.        0.        0.       ]]\n",
      "[ 711305  309924  429585  576144 1020736  694238  444644  995491  734764\n",
      "  414854]\n",
      "methanogens\n",
      "ch4\n",
      "enceladus\n",
      "icy\n",
      "subsurface\n",
      "mars\n",
      "europa\n",
      "solar\n",
      "moons\n",
      "earth\n",
      "[[0.01539038 0.02097378 0.         ... 0.         0.         0.        ]]\n",
      "[1013019  347431  251864 1030135  739649  350460  399969  739722 1013021\n",
      "  348150]\n",
      "stifle\n",
      "cr\n",
      "bm\n",
      "synovial\n",
      "msc\n",
      "crp\n",
      "dogs\n",
      "mscs\n",
      "stifles\n",
      "crcl\n",
      "[[0.         0.00298952 0.         ... 0.         0.         0.        ]]\n",
      "[1048546 1016386  888307  886893  879455  745420  255664  449177  594449\n",
      "  689388]\n",
      "tert\n",
      "stromal\n",
      "pts\n",
      "pt\n",
      "promoter\n",
      "mutation\n",
      "borderline\n",
      "expression\n",
      "intermediate\n",
      "malignant\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[ 748798  323378  462078 1038662  915717  748805 1152022  217778  972059\n",
      "  972220]\n",
      "mzi\n",
      "cladding\n",
      "fiber\n",
      "taper\n",
      "refractive\n",
      "mzis\n",
      "zeolite\n",
      "attenuation\n",
      "sensitivity\n",
      "sensors\n",
      "[[0.00195043 0.         0.         ... 0.         0.         0.        ]]\n",
      "[ 469805  960088  765725  716456  469839  406305 1115047  397581  912878\n",
      "  343313]\n",
      "fmr1\n",
      "sc\n",
      "neurons\n",
      "mice\n",
      "fmrp\n",
      "ds\n",
      "visual\n",
      "dlgn\n",
      "receptive\n",
      "corticocollicular\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[ 260619  401787  651925  191125  907872  485524  715508 1083273  369562\n",
      "  260660]\n",
      "breast\n",
      "dose\n",
      "lao\n",
      "angulations\n",
      "rao\n",
      "gafchromic\n",
      "mgy\n",
      "tube\n",
      "dap\n",
      "breasts\n",
      "[[0.         0.01224446 0.         ... 0.         0.         0.        ]]\n",
      "[ 457229  260619  284206 1132007  302356  825263  260411  163997  302654\n",
      " 1106429]\n",
      "fbc\n",
      "breast\n",
      "cancer\n",
      "women\n",
      "cell\n",
      "pathways\n",
      "brca1\n",
      "adhesion\n",
      "cells\n",
      "variants\n",
      "[[0.        0.0018099 0.        ... 0.        0.        0.       ]]\n",
      "[ 309924  969613  439524  836104  265719  469401 1020272  265727  969622\n",
      "  968049]\n",
      "ch4\n",
      "seep\n",
      "esas\n",
      "permafrost\n",
      "bubble\n",
      "fluxes\n",
      "subsea\n",
      "bubbles\n",
      "seeps\n",
      "sea\n",
      "[[0.         0.01513034 0.         ... 0.         0.         0.        ]]\n",
      "[ 662215  662213  613389  542304 1132579 1006223  958531  817724  878531\n",
      "  662214]\n",
      "lhws\n",
      "lhw\n",
      "job\n",
      "health\n",
      "workers\n",
      "srq\n",
      "satisfaction\n",
      "pakistan\n",
      "programme\n",
      "lhwp\n",
      "[[0.         0.00655473 0.         ... 0.         0.         0.        ]]\n",
      "[ 708304  307131  936417 1115047 1024699  437923  186919  825478  462809\n",
      "  933143]\n",
      "merg\n",
      "cga\n",
      "rp\n",
      "visual\n",
      "supplementation\n",
      "erg\n",
      "amplitude\n",
      "patients\n",
      "field\n",
      "rod\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[930951 880872 164202 879087 164205 302356 869740 284206 703693 336319]\n",
      "rm1\n",
      "prostate\n",
      "adipocyte\n",
      "proliferation\n",
      "adipocytes\n",
      "cell\n",
      "preadipocytes\n",
      "cancer\n",
      "media\n",
      "conditioned\n",
      "[[0.00145481 0.00771009 0.         ... 0.         0.         0.        ]]\n",
      "[554910 148574 432283 917155 703940 554922 238569 834283 454866 694297]\n",
      "hmo\n",
      "aapcc\n",
      "enrollees\n",
      "reimbursement\n",
      "medicare\n",
      "hmos\n",
      "beneficiaries\n",
      "percent\n",
      "fallon\n",
      "marshfield\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[ 275391  749368  275389  918928 1098123  588698  629998  716456  462517\n",
      "  944276]\n",
      "c5ar1\n",
      "mφs\n",
      "c5a\n",
      "renal\n",
      "upec\n",
      "infection\n",
      "kidney\n",
      "mice\n",
      "fibrosis\n",
      "rtecs\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[ 908910  908901  172765  653249  518384  955812  518411 1084260 1070345\n",
      "  966092]\n",
      "raters\n",
      "rater\n",
      "agreement\n",
      "latent\n",
      "grade\n",
      "samples\n",
      "grades\n",
      "tumour\n",
      "trait\n",
      "score\n",
      "[[0.         0.00605827 0.         ... 0.         0.         0.        ]]\n",
      "[ 729578  407519  554927  771971 1159542  227781  422109  846840   31944\n",
      " 1084034]\n",
      "mnt\n",
      "dtox\n",
      "hmp\n",
      "nls\n",
      "αmsh\n",
      "bacteriochlorin\n",
      "egf\n",
      "photosensitizer\n",
      "125i\n",
      "tumor\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[ 123304  222995  328160  925897 1145447  800131  925884  222969  764049\n",
      "  805271]\n",
      "6wpc\n",
      "axons\n",
      "cntf\n",
      "rgcs\n",
      "yfp\n",
      "optic\n",
      "rgc\n",
      "axon\n",
      "nerve\n",
      "ospn\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[ 594671  783961  890988 1019177  551783  909240  170856  161644  379858\n",
      " 1147712]\n",
      "interneurons\n",
      "npy\n",
      "pv\n",
      "subfield\n",
      "hippocampus\n",
      "rats\n",
      "aged\n",
      "activity\n",
      "dentate\n",
      "young\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[ 907738  516951 1001472  302654  733605 1086498  718752  351534 1131130\n",
      "  302356]\n",
      "rankl\n",
      "gp2\n",
      "spib\n",
      "cells\n",
      "monolayers\n",
      "typhimurium\n",
      "microparticles\n",
      "crypts\n",
      "wnt3a\n",
      "cell\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[1154402  518640  801726  995252 1127335 1145919  455897  455922  455937\n",
      "  449847]\n",
      "zn\n",
      "grain\n",
      "organic\n",
      "soil\n",
      "wheat\n",
      "yield\n",
      "farmers\n",
      "farms\n",
      "farmyard\n",
      "extractable\n",
      "[[0.         0.00380803 0.         ... 0.         0.         0.        ]]\n",
      "[ 853811  233807  464861  520544  462914 1093386  730168  462809  302654\n",
      "  592938]\n",
      "place\n",
      "bayesian\n",
      "firing\n",
      "grid\n",
      "fields\n",
      "uncertainty\n",
      "model\n",
      "field\n",
      "cells\n",
      "integration\n",
      "[[0.         0.00234082 0.         ... 0.         0.         0.        ]]\n",
      "[ 535033  792001  317651 1111163  834924  923716  379756  397801  372601\n",
      "  791973]\n",
      "haemodialysis\n",
      "octa\n",
      "choriocapillaris\n",
      "vessel\n",
      "perfused\n",
      "retinal\n",
      "density\n",
      "dm\n",
      "dcp\n",
      "oct\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[ 780542  149339  283559  850480 1063137  909240  435773  715315  551783\n",
      "  435747]\n",
      "norbin\n",
      "aav\n",
      "camkii\n",
      "pilocarpine\n",
      "tle\n",
      "rats\n",
      "epileptic\n",
      "mglur5\n",
      "hippocampus\n",
      "epilepsy\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[857408 860794 315285 994127 469401 775493 860776 175064 958449 857353]\n",
      "pm2\n",
      "pollution\n",
      "china\n",
      "so2\n",
      "fluxes\n",
      "no2\n",
      "pollutant\n",
      "air\n",
      "satellite\n",
      "pm10\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[ 981268  911019  954883  567763  977618  389003  458712 1016326  259449\n",
      "  918928]\n",
      "shrsp\n",
      "rd\n",
      "salt\n",
      "hydralazine\n",
      "sham\n",
      "diet\n",
      "fed\n",
      "stroke\n",
      "brain\n",
      "renal\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[    12 436607 357204 911527 392181 291193 341310 239555 357184 810080]\n",
      "0000000000000000000000000000000000\n",
      "epr\n",
      "cuii\n",
      "reaction\n",
      "dinuclear\n",
      "catalyst\n",
      "copper\n",
      "benzylpyridine\n",
      "cui\n",
      "oxidation\n",
      "[[0. 0. 0. ... 0. 0. 0.]]\n",
      "[258295 790974 790977 258311 410575 736720 591522 364701 369217  13066]\n",
      "bpm\n",
      "obstacle\n",
      "obstacles\n",
      "bpms\n",
      "dynamic\n",
      "motion\n",
      "input\n",
      "d1\n",
      "danger\n",
      "0185744\n"
     ]
    }
   ],
   "source": [
    "for i in match_index:\n",
    "    word_indexes = tfidf.getrow(i).todense().A1.argsort()[-10:][::-1]\n",
    "    article_tfidf = tfidf.getrow(i)\n",
    "    \n",
    "    print(word_indexes)\n",
    "    for i in word_indexes:\n",
    "        print(word_features[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "Y = vectorizer2.fit_transform(corpus)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[0.00484436, 0.0078168 , 0.00484436, ..., 0.        , 0.        ,\n",
       "        0.        ],\n",
       "       [0.        , 0.        , 0.        , ..., 0.00906395, 0.        ,\n",
       "        0.00906395],\n",
       "       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,\n",
       "        0.        ],\n",
       "       [0.        , 0.        , 0.        , ..., 0.        , 0.01458545,\n",
       "        0.        ],\n",
       "       [0.        , 0.01433153, 0.        , ..., 0.        , 0.        ,\n",
       "        0.        ]])"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Y.toarray()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['0001',\n",
       " '001',\n",
       " '003',\n",
       " '004',\n",
       " '01',\n",
       " '04',\n",
       " '040',\n",
       " '041',\n",
       " '043',\n",
       " '05',\n",
       " '051',\n",
       " '057',\n",
       " '060',\n",
       " '073',\n",
       " '076',\n",
       " '08',\n",
       " '080',\n",
       " '094',\n",
       " '10',\n",
       " '100',\n",
       " '1000',\n",
       " '101112',\n",
       " '1083',\n",
       " '11',\n",
       " '1187',\n",
       " '12',\n",
       " '121',\n",
       " '124',\n",
       " '13',\n",
       " '1360',\n",
       " '137',\n",
       " '14',\n",
       " '144',\n",
       " '1472',\n",
       " '15',\n",
       " '16',\n",
       " '163',\n",
       " '17',\n",
       " '177',\n",
       " '178',\n",
       " '18',\n",
       " '1827',\n",
       " '18f',\n",
       " '19',\n",
       " '190',\n",
       " '1905',\n",
       " '195',\n",
       " '1964',\n",
       " '1970',\n",
       " '1980',\n",
       " '1991',\n",
       " '1997',\n",
       " '1999',\n",
       " '20',\n",
       " '2000',\n",
       " '2001',\n",
       " '2002',\n",
       " '2003',\n",
       " '2004',\n",
       " '2005',\n",
       " '2006',\n",
       " '2007',\n",
       " '2008',\n",
       " '2009',\n",
       " '2010',\n",
       " '2013',\n",
       " '21',\n",
       " '211',\n",
       " '214',\n",
       " '22',\n",
       " '2235',\n",
       " '23',\n",
       " '24',\n",
       " '243',\n",
       " '25',\n",
       " '2500',\n",
       " '26',\n",
       " '27',\n",
       " '2732',\n",
       " '2766',\n",
       " '277',\n",
       " '28',\n",
       " '29',\n",
       " '30',\n",
       " '303',\n",
       " '304',\n",
       " '31',\n",
       " '314',\n",
       " '32',\n",
       " '33',\n",
       " '34',\n",
       " '342',\n",
       " '349',\n",
       " '35',\n",
       " '352',\n",
       " '358',\n",
       " '36',\n",
       " '37',\n",
       " '374',\n",
       " '3742',\n",
       " '38',\n",
       " '3844',\n",
       " '385',\n",
       " '39',\n",
       " '392',\n",
       " '40',\n",
       " '4000',\n",
       " '41',\n",
       " '42',\n",
       " '43',\n",
       " '431',\n",
       " '435',\n",
       " '44',\n",
       " '448',\n",
       " '454',\n",
       " '46',\n",
       " '460',\n",
       " '46073',\n",
       " '47',\n",
       " '471',\n",
       " '48',\n",
       " '4806',\n",
       " '484',\n",
       " '49',\n",
       " '50',\n",
       " '507',\n",
       " '5086',\n",
       " '51',\n",
       " '5103',\n",
       " '52',\n",
       " '5221',\n",
       " '53',\n",
       " '5308',\n",
       " '534',\n",
       " '538',\n",
       " '539',\n",
       " '54',\n",
       " '542',\n",
       " '551',\n",
       " '565',\n",
       " '57',\n",
       " '571',\n",
       " '57331',\n",
       " '58',\n",
       " '5868',\n",
       " '590',\n",
       " '60',\n",
       " '6000',\n",
       " '603',\n",
       " '6044',\n",
       " '606',\n",
       " '6061',\n",
       " '61',\n",
       " '6174',\n",
       " '62',\n",
       " '624',\n",
       " '63',\n",
       " '638',\n",
       " '64',\n",
       " '6433',\n",
       " '65',\n",
       " '653',\n",
       " '667',\n",
       " '6691',\n",
       " '671',\n",
       " '6745',\n",
       " '6920',\n",
       " '70',\n",
       " '708',\n",
       " '71',\n",
       " '717',\n",
       " '729',\n",
       " '76',\n",
       " '768',\n",
       " '77',\n",
       " '779',\n",
       " '78',\n",
       " '787',\n",
       " '789',\n",
       " '80',\n",
       " '815',\n",
       " '837',\n",
       " '84',\n",
       " '8469',\n",
       " '85',\n",
       " '861',\n",
       " '864',\n",
       " '868',\n",
       " '872',\n",
       " '873',\n",
       " '89',\n",
       " '890',\n",
       " '891',\n",
       " '907',\n",
       " '91',\n",
       " '92',\n",
       " '929',\n",
       " '937',\n",
       " '95',\n",
       " '957',\n",
       " '96',\n",
       " '97',\n",
       " 'abidin',\n",
       " 'abilities',\n",
       " 'ability',\n",
       " 'able',\n",
       " 'abnormal',\n",
       " 'abnormalities',\n",
       " 'about',\n",
       " 'above',\n",
       " 'absence',\n",
       " 'absent',\n",
       " 'abuoun',\n",
       " 'abuse',\n",
       " 'ac',\n",
       " 'academic',\n",
       " 'accelerate',\n",
       " 'access',\n",
       " 'accessed',\n",
       " 'accessibility',\n",
       " 'accessible',\n",
       " 'accessing',\n",
       " 'accidentally',\n",
       " 'accompanied',\n",
       " 'accomplished',\n",
       " 'accordance',\n",
       " 'according',\n",
       " 'account',\n",
       " 'accounting',\n",
       " 'accounts',\n",
       " 'accumulation',\n",
       " 'accuracy',\n",
       " 'accurate',\n",
       " 'accurately',\n",
       " 'achieva',\n",
       " 'achieve',\n",
       " 'achieved',\n",
       " 'acid',\n",
       " 'acids',\n",
       " 'acknowledge',\n",
       " 'across',\n",
       " 'action',\n",
       " 'activate',\n",
       " 'activated',\n",
       " 'activation',\n",
       " 'active',\n",
       " 'activities',\n",
       " 'activity',\n",
       " 'acute',\n",
       " 'adaptation',\n",
       " 'adapted',\n",
       " 'add',\n",
       " 'adddition',\n",
       " 'added',\n",
       " 'adding',\n",
       " 'addition',\n",
       " 'additional',\n",
       " 'address',\n",
       " 'addressing',\n",
       " 'adequate',\n",
       " 'adiposity',\n",
       " 'adjusting',\n",
       " 'administered',\n",
       " 'administration',\n",
       " 'administrative',\n",
       " 'adolescent',\n",
       " 'adolescents',\n",
       " 'advance',\n",
       " 'advanced',\n",
       " 'advancing',\n",
       " 'advantage',\n",
       " 'advantages',\n",
       " 'advisory',\n",
       " 'aerobic',\n",
       " 'affect',\n",
       " 'affiliated',\n",
       " 'after',\n",
       " 'again',\n",
       " 'age',\n",
       " 'agencies',\n",
       " 'agents',\n",
       " 'ages',\n",
       " 'aggregated',\n",
       " 'aggregative',\n",
       " 'aggressive',\n",
       " 'agricultural',\n",
       " 'aids',\n",
       " 'aim',\n",
       " 'aiming',\n",
       " 'aims',\n",
       " 'akin',\n",
       " 'al',\n",
       " 'alexander',\n",
       " 'algorithm',\n",
       " 'all',\n",
       " 'alleviated',\n",
       " 'alleviating',\n",
       " 'allow',\n",
       " 'almost',\n",
       " 'alone',\n",
       " 'along',\n",
       " 'already',\n",
       " 'also',\n",
       " 'alter',\n",
       " 'alteration',\n",
       " 'alterations',\n",
       " 'altered',\n",
       " 'alternative',\n",
       " 'alternatively',\n",
       " 'although',\n",
       " 'ambiguous',\n",
       " 'amenable',\n",
       " 'amino',\n",
       " 'aml',\n",
       " 'ammonia',\n",
       " 'among',\n",
       " 'amount',\n",
       " 'an',\n",
       " 'anaerobic',\n",
       " 'analyses',\n",
       " 'analysis',\n",
       " 'analyzed',\n",
       " 'analyzer',\n",
       " 'and',\n",
       " 'anemia',\n",
       " 'animals',\n",
       " 'anonymous',\n",
       " 'another',\n",
       " 'anova',\n",
       " 'answered',\n",
       " 'answers',\n",
       " 'any',\n",
       " 'apart',\n",
       " 'appear',\n",
       " 'appeared',\n",
       " 'appears',\n",
       " 'applicability',\n",
       " 'applications',\n",
       " 'applied',\n",
       " 'apply',\n",
       " 'applying',\n",
       " 'appointed',\n",
       " 'approach',\n",
       " 'approaches',\n",
       " 'appropriate',\n",
       " 'approval',\n",
       " 'approved',\n",
       " 'approximately',\n",
       " 'arabidopsis',\n",
       " 'are',\n",
       " 'area',\n",
       " 'areas',\n",
       " 'argininosuccinate',\n",
       " 'arise',\n",
       " 'arm',\n",
       " 'arms',\n",
       " 'arose',\n",
       " 'array',\n",
       " 'arrhythmias',\n",
       " 'arrived',\n",
       " 'artificial',\n",
       " 'as',\n",
       " 'asians',\n",
       " 'ask',\n",
       " 'asked',\n",
       " 'asl',\n",
       " 'aspect',\n",
       " 'aspects',\n",
       " 'aspiration',\n",
       " 'ass',\n",
       " 'assembled',\n",
       " 'assess',\n",
       " 'assessed',\n",
       " 'assessing',\n",
       " 'assessment',\n",
       " 'assign',\n",
       " 'assigned',\n",
       " 'assignment',\n",
       " 'assignments',\n",
       " 'assist',\n",
       " 'associated',\n",
       " 'association',\n",
       " 'associations',\n",
       " 'assumed',\n",
       " 'assumes',\n",
       " 'asymptomatic',\n",
       " 'at',\n",
       " 'ataxia',\n",
       " 'atmospheres',\n",
       " 'atrioventricular',\n",
       " 'attaching',\n",
       " 'attempted',\n",
       " 'attempts',\n",
       " 'attend',\n",
       " 'attendance',\n",
       " 'attended',\n",
       " 'attendees',\n",
       " 'attention',\n",
       " 'attracted',\n",
       " 'au400',\n",
       " 'auc',\n",
       " 'augment',\n",
       " 'authority',\n",
       " 'authors',\n",
       " 'auto',\n",
       " 'automated',\n",
       " 'automatically',\n",
       " 'autonomic',\n",
       " 'autonomous',\n",
       " 'autonomy',\n",
       " 'availability',\n",
       " 'available',\n",
       " 'availed',\n",
       " 'avb',\n",
       " 'average',\n",
       " 'avoid',\n",
       " 'avoided',\n",
       " 'away',\n",
       " 'background',\n",
       " 'bal',\n",
       " 'balance',\n",
       " 'basal',\n",
       " 'based',\n",
       " 'baseline',\n",
       " 'basis',\n",
       " 'batswana',\n",
       " 'battery',\n",
       " 'be',\n",
       " 'bearing',\n",
       " 'beat',\n",
       " 'because',\n",
       " 'become',\n",
       " 'becomes',\n",
       " 'been',\n",
       " 'before',\n",
       " 'began',\n",
       " 'begin',\n",
       " 'behavior',\n",
       " 'behaviors',\n",
       " 'behavioural',\n",
       " 'behind',\n",
       " 'believe',\n",
       " 'belmont',\n",
       " 'belong',\n",
       " 'below',\n",
       " 'benefits',\n",
       " 'better',\n",
       " 'between',\n",
       " 'beyond',\n",
       " 'biased',\n",
       " 'biases',\n",
       " 'bibliomic',\n",
       " 'bicycle',\n",
       " 'bile',\n",
       " 'binders',\n",
       " 'bioartificial',\n",
       " 'biochemical',\n",
       " 'bioelectric',\n",
       " 'bioengineering',\n",
       " 'bioethics',\n",
       " 'biofluid',\n",
       " 'biomarker',\n",
       " 'biomarkers',\n",
       " 'biomechanics',\n",
       " 'biomedcentral',\n",
       " 'biomedical',\n",
       " 'biostatistics',\n",
       " 'biosynthesis',\n",
       " 'birth',\n",
       " 'block',\n",
       " 'blogs',\n",
       " 'blood',\n",
       " 'bloom',\n",
       " 'bmi',\n",
       " 'board',\n",
       " 'boards',\n",
       " 'bock',\n",
       " 'bodies',\n",
       " 'body',\n",
       " 'bone',\n",
       " 'borne',\n",
       " 'both',\n",
       " 'botswana',\n",
       " 'bound',\n",
       " 'boundary',\n",
       " 'bradycardia',\n",
       " 'brain',\n",
       " 'brief',\n",
       " 'briefed',\n",
       " 'bross',\n",
       " 'build',\n",
       " 'building',\n",
       " 'built',\n",
       " 'busse',\n",
       " 'but',\n",
       " 'by',\n",
       " 'bypasses',\n",
       " 'bypassing',\n",
       " 'cafes',\n",
       " 'calculated',\n",
       " 'caliper',\n",
       " 'calories',\n",
       " 'campus',\n",
       " 'campuses',\n",
       " 'can',\n",
       " 'cancer',\n",
       " 'cancers',\n",
       " 'candidate',\n",
       " 'cannot',\n",
       " 'capacities',\n",
       " 'capacity',\n",
       " 'capital',\n",
       " 'carbohydrate',\n",
       " 'carbohydrates',\n",
       " 'carbon',\n",
       " 'carcinogenesis',\n",
       " 'cardiac',\n",
       " 'cardiorespiratory',\n",
       " 'cardiovascular',\n",
       " 'cards',\n",
       " 'care',\n",
       " 'carefully',\n",
       " 'carried',\n",
       " 'carry',\n",
       " 'carve',\n",
       " 'carves',\n",
       " 'case',\n",
       " 'casebooks',\n",
       " 'cases',\n",
       " 'category',\n",
       " 'cats',\n",
       " 'causally',\n",
       " 'causative',\n",
       " 'cause',\n",
       " 'caused',\n",
       " 'causes',\n",
       " 'cautiously',\n",
       " 'caveats',\n",
       " 'cbm',\n",
       " 'cd',\n",
       " 'cell',\n",
       " 'cells',\n",
       " 'cellular',\n",
       " 'center',\n",
       " 'centers',\n",
       " 'central',\n",
       " 'certain',\n",
       " 'certificates',\n",
       " 'certification',\n",
       " 'ch',\n",
       " 'challenge',\n",
       " 'challenges',\n",
       " 'chan',\n",
       " 'change',\n",
       " 'changed',\n",
       " 'changes',\n",
       " 'characteristics',\n",
       " 'characterized',\n",
       " 'characterizes',\n",
       " 'characterizing',\n",
       " 'chat',\n",
       " 'chatziioannou',\n",
       " 'check',\n",
       " 'checkmodelconsistency',\n",
       " 'chemical',\n",
       " 'chemotherapy',\n",
       " 'chi',\n",
       " 'chicago',\n",
       " 'child',\n",
       " 'childbirth',\n",
       " 'childhood',\n",
       " 'children',\n",
       " 'chisari',\n",
       " 'choice',\n",
       " 'cholesterol',\n",
       " 'choose',\n",
       " 'chosen',\n",
       " 'christian',\n",
       " 'chronic',\n",
       " 'ci',\n",
       " 'cigarette',\n",
       " 'cioms',\n",
       " 'circulatory',\n",
       " 'citi',\n",
       " 'city',\n",
       " 'cl',\n",
       " 'clarification',\n",
       " 'clarify',\n",
       " 'class',\n",
       " 'classification',\n",
       " 'classifier',\n",
       " 'classifiers',\n",
       " 'clearly',\n",
       " 'climate',\n",
       " 'cline',\n",
       " 'clinical',\n",
       " 'clonality',\n",
       " 'close',\n",
       " 'clustering',\n",
       " 'cm',\n",
       " 'cml',\n",
       " 'cmr',\n",
       " 'co',\n",
       " 'cobra',\n",
       " 'code',\n",
       " 'codes',\n",
       " 'coefficient',\n",
       " 'coils',\n",
       " 'collaborations',\n",
       " 'collaborative',\n",
       " 'colleagues',\n",
       " 'collection',\n",
       " 'collegial',\n",
       " 'com',\n",
       " 'combine',\n",
       " 'combining',\n",
       " 'comfortable',\n",
       " 'comment',\n",
       " 'comments',\n",
       " 'commit',\n",
       " 'commitment',\n",
       " 'committee',\n",
       " 'committees',\n",
       " 'committing',\n",
       " 'common',\n",
       " 'commonly',\n",
       " 'communal',\n",
       " 'communication',\n",
       " 'communities',\n",
       " 'community',\n",
       " 'comparable',\n",
       " 'compared',\n",
       " 'comparing',\n",
       " 'comparison',\n",
       " 'compartment',\n",
       " 'compartmentalization',\n",
       " 'compartments',\n",
       " 'compatible',\n",
       " 'competency',\n",
       " 'competing',\n",
       " 'competitive',\n",
       " 'complain',\n",
       " 'complement',\n",
       " 'complete',\n",
       " 'completed',\n",
       " 'completely',\n",
       " 'completing',\n",
       " 'completion',\n",
       " 'complex',\n",
       " 'complexity',\n",
       " 'compliance',\n",
       " 'component',\n",
       " 'components',\n",
       " 'composed',\n",
       " 'composition',\n",
       " 'comprehensive',\n",
       " 'comprises',\n",
       " 'comprising',\n",
       " 'computational',\n",
       " 'computationally',\n",
       " 'compute',\n",
       " 'computed',\n",
       " 'computerized',\n",
       " 'computers',\n",
       " 'computes',\n",
       " 'concentration',\n",
       " 'concentrations',\n",
       " 'concepts',\n",
       " 'conceptualization',\n",
       " 'concern',\n",
       " 'concerned',\n",
       " 'concerted',\n",
       " 'conclusion',\n",
       " 'conclusions',\n",
       " 'conclusive',\n",
       " 'concordance',\n",
       " 'concurrently',\n",
       " 'condition',\n",
       " 'conditional',\n",
       " 'conditions',\n",
       " 'conduct',\n",
       " 'conducted',\n",
       " 'conduction',\n",
       " 'conference',\n",
       " 'confidence',\n",
       " 'confidences',\n",
       " 'confidentiality',\n",
       " 'confirmation',\n",
       " 'confirmed',\n",
       " 'conflicts',\n",
       " 'congestive',\n",
       " 'connectivity',\n",
       " 'cons',\n",
       " 'consent',\n",
       " 'consider',\n",
       " 'considerable',\n",
       " 'consideration',\n",
       " 'considered',\n",
       " 'considering',\n",
       " 'consist',\n",
       " 'consistency',\n",
       " 'consistent',\n",
       " 'consistently',\n",
       " 'consisting',\n",
       " 'consists',\n",
       " 'constantly',\n",
       " 'constants',\n",
       " 'constitutional',\n",
       " 'constraint',\n",
       " 'constraints',\n",
       " 'construct',\n",
       " 'constructed',\n",
       " 'constructing',\n",
       " 'construction',\n",
       " 'consultation',\n",
       " 'consuming',\n",
       " 'consumption',\n",
       " 'contain',\n",
       " 'contained',\n",
       " 'containing',\n",
       " 'contains',\n",
       " 'content',\n",
       " 'context',\n",
       " 'contexts',\n",
       " 'contradiction',\n",
       " 'contradictory',\n",
       " 'contrast',\n",
       " 'contribute',\n",
       " 'contributed',\n",
       " 'contributions',\n",
       " 'control',\n",
       " 'controlled',\n",
       " 'controlling',\n",
       " 'controls',\n",
       " 'controversial',\n",
       " 'controversies',\n",
       " 'convenience',\n",
       " 'conversation',\n",
       " 'conversion',\n",
       " 'copies',\n",
       " 'cordier',\n",
       " 'core',\n",
       " 'cores',\n",
       " 'correct',\n",
       " 'corrected',\n",
       " 'correctly',\n",
       " 'correlated',\n",
       " 'corresponding',\n",
       " 'cost',\n",
       " 'could',\n",
       " 'council',\n",
       " 'counterparts',\n",
       " 'countries',\n",
       " 'country',\n",
       " 'course',\n",
       " 'courses',\n",
       " 'cover',\n",
       " 'covered',\n",
       " 'create',\n",
       " 'created',\n",
       " 'credits',\n",
       " 'criteria',\n",
       " 'critical',\n",
       " 'cross',\n",
       " 'crucial',\n",
       " 'cs',\n",
       " 'culture',\n",
       " 'cultured',\n",
       " 'curated',\n",
       " 'curation',\n",
       " 'current',\n",
       " 'currently',\n",
       " 'curricula',\n",
       " 'curriculum',\n",
       " 'curve',\n",
       " 'curves',\n",
       " 'cycle',\n",
       " 'cytoscape',\n",
       " 'data',\n",
       " 'database',\n",
       " 'dates',\n",
       " 'daughter',\n",
       " 'daughters',\n",
       " 'day',\n",
       " 'daycare',\n",
       " 'de',\n",
       " 'dead',\n",
       " 'debate',\n",
       " 'decade',\n",
       " 'decades',\n",
       " 'decelerate',\n",
       " 'december',\n",
       " 'decision',\n",
       " 'declaration',\n",
       " 'declare',\n",
       " 'decrease',\n",
       " 'decreased',\n",
       " 'decreases',\n",
       " 'decreasing',\n",
       " 'dedicated',\n",
       " 'deduced',\n",
       " 'deficiencies',\n",
       " 'deficiency',\n",
       " 'define',\n",
       " 'defined',\n",
       " 'defines',\n",
       " 'definition',\n",
       " 'degree',\n",
       " 'delayed',\n",
       " 'delivery',\n",
       " 'demand',\n",
       " 'demands',\n",
       " 'democracy',\n",
       " 'demographic',\n",
       " 'demonstrate',\n",
       " 'demonstrated',\n",
       " 'demonstrating',\n",
       " 'denoted',\n",
       " 'denotes',\n",
       " 'denoting',\n",
       " 'deoxyglucose',\n",
       " 'depend',\n",
       " 'dependency',\n",
       " 'dependent',\n",
       " 'depending',\n",
       " 'depends',\n",
       " 'depict',\n",
       " 'depicted',\n",
       " 'derive',\n",
       " 'derived',\n",
       " 'derives',\n",
       " 'deriving',\n",
       " 'described',\n",
       " 'describing',\n",
       " 'description',\n",
       " 'descriptive',\n",
       " 'design',\n",
       " 'designed',\n",
       " 'desirable',\n",
       " 'despite',\n",
       " 'destined',\n",
       " 'detail',\n",
       " 'detailed',\n",
       " 'details',\n",
       " 'detect',\n",
       " 'detected',\n",
       " 'determine',\n",
       " 'determining',\n",
       " 'develop',\n",
       " 'developed',\n",
       " 'developers',\n",
       " 'developing',\n",
       " 'development',\n",
       " 'devices',\n",
       " 'dhts',\n",
       " 'diabetes',\n",
       " 'diagnosed',\n",
       " 'diagnosis',\n",
       " 'diagnostic',\n",
       " 'dialog',\n",
       " 'did',\n",
       " 'diet',\n",
       " 'dietary',\n",
       " 'difference',\n",
       " 'differences',\n",
       " 'different',\n",
       " 'differentiate',\n",
       " 'differs',\n",
       " 'difficulty',\n",
       " 'dilemma',\n",
       " 'dilemmas',\n",
       " 'directed',\n",
       " 'directionality',\n",
       " 'disappointingly',\n",
       " 'discourage',\n",
       " 'discovered',\n",
       " 'discuss',\n",
       " 'discussion',\n",
       " 'discussions',\n",
       " 'disease',\n",
       " 'diseases',\n",
       " 'disorders',\n",
       " 'disproportionate',\n",
       " 'distance',\n",
       " 'distant',\n",
       " 'distinct',\n",
       " 'distinctions',\n",
       " 'distribution',\n",
       " 'distributions',\n",
       " 'district',\n",
       " 'diverse',\n",
       " 'divide',\n",
       " 'division',\n",
       " 'do',\n",
       " 'dockerty',\n",
       " 'documented',\n",
       " 'documents',\n",
       " 'does',\n",
       " 'doing',\n",
       " 'doll',\n",
       " 'done',\n",
       " 'draft',\n",
       " 'drafting',\n",
       " 'drawn',\n",
       " 'driven',\n",
       " 'dropout',\n",
       " 'drug',\n",
       " 'duarte',\n",
       " 'due',\n",
       " 'duration',\n",
       " 'during',\n",
       " 'durot',\n",
       " 'dysfunctional',\n",
       " 'each',\n",
       " 'earlier',\n",
       " 'early',\n",
       " 'easier',\n",
       " 'easy',\n",
       " 'edctp',\n",
       " 'edited',\n",
       " 'education',\n",
       " 'educational',\n",
       " 'educators',\n",
       " 'edwards',\n",
       " 'effect',\n",
       " 'effective',\n",
       " 'effectively',\n",
       " 'effectiveness',\n",
       " 'effects',\n",
       " 'efficiency',\n",
       " 'efficient',\n",
       " 'effort',\n",
       " 'efforts',\n",
       " 'eg',\n",
       " 'eh',\n",
       " 'eight',\n",
       " 'eighty',\n",
       " 'either',\n",
       " 'electromagnetic',\n",
       " 'element',\n",
       " 'elements',\n",
       " 'elicit',\n",
       " 'eligible',\n",
       " 'elimination',\n",
       " 'elucidated',\n",
       " 'em',\n",
       " 'email',\n",
       " 'emails',\n",
       " 'embedded',\n",
       " 'emission',\n",
       " 'emphasize',\n",
       " 'emphasized',\n",
       " 'empirical',\n",
       " 'employed',\n",
       " 'employee',\n",
       " 'emulate',\n",
       " 'enabled',\n",
       " 'enables',\n",
       " 'enabling',\n",
       " 'encourage',\n",
       " 'end',\n",
       " 'endeavour',\n",
       " 'endorsed',\n",
       " 'energy',\n",
       " 'enforcing',\n",
       " 'engage',\n",
       " 'engaged',\n",
       " 'engagement',\n",
       " 'english',\n",
       " 'enhance',\n",
       " 'enhanced',\n",
       " 'enhancement',\n",
       " 'enormous',\n",
       " 'enriched',\n",
       " 'enrolled',\n",
       " 'ensure',\n",
       " 'enterprise',\n",
       " 'entities',\n",
       " 'entry',\n",
       " 'environment',\n",
       " 'environmental',\n",
       " 'enzyme',\n",
       " 'enzymes',\n",
       " 'epidemiologic',\n",
       " 'epidemiological',\n",
       " 'ergometer',\n",
       " 'erroneous',\n",
       " 'error',\n",
       " 'errors',\n",
       " 'especially',\n",
       " 'essential',\n",
       " 'establish',\n",
       " 'established',\n",
       " 'estimated',\n",
       " 'et',\n",
       " 'ethical',\n",
       " 'ethics',\n",
       " 'etiology',\n",
       " 'european',\n",
       " 'evaluate',\n",
       " ...]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vectorizer2.get_feature_names()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'background': 418,\n",
       " 'many': 1611,\n",
       " 'countries': 752,\n",
       " 'in': 1366,\n",
       " 'the': 2526,\n",
       " 'global': 1210,\n",
       " 'south': 2386,\n",
       " 'are': 349,\n",
       " 'hard': 1250,\n",
       " 'pressed': 1984,\n",
       " 'to': 2558,\n",
       " 'identify': 1330,\n",
       " 'country': 753,\n",
       " 'personnel': 1901,\n",
       " 'with': 2742,\n",
       " 'adequate': 259,\n",
       " 'training': 2579,\n",
       " 'human': 1312,\n",
       " 'subjects': 2452,\n",
       " 'research': 2176,\n",
       " 'ethics': 996,\n",
       " 'participate': 1863,\n",
       " 'as': 361,\n",
       " 'investigators': 1469,\n",
       " 'staff': 2408,\n",
       " 'or': 1816,\n",
       " 'members': 1649,\n",
       " 'of': 1783,\n",
       " 'review': 2206,\n",
       " 'bodies': 477,\n",
       " 'international': 1452,\n",
       " 'partners': 1870,\n",
       " 'have': 1255,\n",
       " 'attempted': 391,\n",
       " 'address': 257,\n",
       " 'this': 2543,\n",
       " 'need': 1731,\n",
       " 'by': 495,\n",
       " 'incorporating': 1380,\n",
       " 'short': 2324,\n",
       " 'courses': 755,\n",
       " 'and': 323,\n",
       " 'workshops': 2755,\n",
       " 'into': 1460,\n",
       " 'their': 2527,\n",
       " 'capacity': 510,\n",
       " 'building': 491,\n",
       " 'programs': 2018,\n",
       " 'supporting': 2479,\n",
       " 'host': 1300,\n",
       " 'initiatives': 1416,\n",
       " 'implement': 1351,\n",
       " 'efforts': 932,\n",
       " 'own': 1844,\n",
       " 'while': 2729,\n",
       " 'such': 2459,\n",
       " 'generally': 1191,\n",
       " 'share': 2316,\n",
       " 'common': 624,\n",
       " 'goal': 1218,\n",
       " 'that': 2525,\n",
       " 'competency': 638,\n",
       " 'protections': 2031,\n",
       " 'they': 2538,\n",
       " 'can': 504,\n",
       " 'vary': 2676,\n",
       " 'significantly': 2341,\n",
       " 'instructional': 1428,\n",
       " 'approach': 342,\n",
       " 'format': 1146,\n",
       " 'some': 2383,\n",
       " 'focus': 1130,\n",
       " 'on': 1796,\n",
       " 'formal': 1145,\n",
       " 'guidelines': 1245,\n",
       " 'general': 1190,\n",
       " 'ethical': 995,\n",
       " 'principles': 1996,\n",
       " 'historically': 1290,\n",
       " 'noteworthy': 1757,\n",
       " 'cases': 528,\n",
       " 'abuse': 213,\n",
       " 'these': 2537,\n",
       " 'provide': 2038,\n",
       " 'researchers': 2177,\n",
       " 'information': 1406,\n",
       " 'needed': 1732,\n",
       " 'meet': 1648,\n",
       " 'regulatory': 2136,\n",
       " 'requirements': 2173,\n",
       " 'but': 494,\n",
       " 'do': 893,\n",
       " 'not': 1754,\n",
       " 'necessarily': 1728,\n",
       " 'prepare': 1979,\n",
       " 'them': 2528,\n",
       " 'apply': 339,\n",
       " 'reasoning': 2091,\n",
       " 'skills': 2367,\n",
       " 'design': 836,\n",
       " 'conduct': 680,\n",
       " 'which': 2728,\n",
       " 'emphasize': 950,\n",
       " 'compliance': 650,\n",
       " 'national': 1726,\n",
       " 'regulations': 2135,\n",
       " 'easier': 916,\n",
       " 'develop': 848,\n",
       " 'than': 2524,\n",
       " 'case': 526,\n",
       " 'based': 422,\n",
       " 'instruction': 1427,\n",
       " 'be': 427,\n",
       " 'taught': 2506,\n",
       " 'over': 1839,\n",
       " 'relatively': 2143,\n",
       " 'periods': 1895,\n",
       " 'time': 2554,\n",
       " 'readily': 2089,\n",
       " 'compared': 631,\n",
       " 'existing': 1039,\n",
       " 'ensure': 975,\n",
       " 'all': 294,\n",
       " 'required': 2171,\n",
       " 'elements': 940,\n",
       " 'been': 433,\n",
       " 'covered': 757,\n",
       " 'studies': 2445,\n",
       " 'however': 1302,\n",
       " 'suggested': 2465,\n",
       " 'more': 1701,\n",
       " 'likely': 1554,\n",
       " 'formulate': 1150,\n",
       " 'concepts': 667,\n",
       " 'scientific': 2262,\n",
       " 'integrity': 1434,\n",
       " 'responsibilities': 2192,\n",
       " 'towards': 2569,\n",
       " 'others': 1830,\n",
       " 'from': 1164,\n",
       " 'climate': 593,\n",
       " 'institutional': 1425,\n",
       " 'social': 2375,\n",
       " 'contexts': 723,\n",
       " 'norms': 1753,\n",
       " 'education': 920,\n",
       " 'combine': 614,\n",
       " 'lectures': 1535,\n",
       " 'inter': 1440,\n",
       " 'collegial': 612,\n",
       " 'discussion': 877,\n",
       " 'situations': 2358,\n",
       " 'may': 1636,\n",
       " 'effective': 925,\n",
       " 'fostering': 1153,\n",
       " 'recognition': 2100,\n",
       " 'issues': 1484,\n",
       " 'for': 1140,\n",
       " 'moral': 1699,\n",
       " 'intentionality': 1439,\n",
       " 'ability': 204,\n",
       " 'action': 241,\n",
       " 'also': 302,\n",
       " 'including': 1376,\n",
       " 'site': 2354,\n",
       " 'person': 1899,\n",
       " 'wholly': 2733,\n",
       " 'line': 1559,\n",
       " 'distance': 883,\n",
       " '14': 31,\n",
       " 'hybrid': 1316,\n",
       " 'combining': 615,\n",
       " 'face': 1068,\n",
       " 'supplementary': 2474,\n",
       " 'web': 2713,\n",
       " 'materials': 1625,\n",
       " 'addition': 255,\n",
       " 'number': 1760,\n",
       " 'collaborations': 608,\n",
       " 'produced': 2009,\n",
       " 'curricula': 772,\n",
       " 'modular': 1690,\n",
       " 'form': 1144,\n",
       " 'available': 411,\n",
       " 'print': 1997,\n",
       " 'online': 1802,\n",
       " 'cd': 539,\n",
       " 'roms': 2220,\n",
       " 'tailored': 2498,\n",
       " 'needs': 1733,\n",
       " 'resources': 2185,\n",
       " 'constraints': 709,\n",
       " 'individual': 1393,\n",
       " 'institutions': 1426,\n",
       " '15': 34,\n",
       " '16': 35,\n",
       " 'valuable': 2663,\n",
       " 'given': 1208,\n",
       " 'opportunity': 1808,\n",
       " 'debate': 787,\n",
       " 'exchange': 1026,\n",
       " 'ideas': 1326,\n",
       " 'resource': 2184,\n",
       " 'intensive': 1438,\n",
       " 'rely': 2152,\n",
       " 'availability': 410,\n",
       " 'participants': 1862,\n",
       " 'faculty': 1074,\n",
       " 'dedicated': 799,\n",
       " 'commit': 619,\n",
       " 'purposes': 2053,\n",
       " 'offer': 1785,\n",
       " 'maximal': 1632,\n",
       " 'flexibility': 1123,\n",
       " 'use': 2653,\n",
       " 'require': 2170,\n",
       " 'few': 1102,\n",
       " 'once': 1797,\n",
       " 'established': 992,\n",
       " 'added': 253,\n",
       " 'advantage': 270,\n",
       " 'accessibility': 219,\n",
       " 'large': 1520,\n",
       " 'numbers': 1761,\n",
       " 'people': 1883,\n",
       " 'different': 865,\n",
       " 'schedules': 2253,\n",
       " 'learning': 1531,\n",
       " 'speeds': 2397,\n",
       " '2009': 63,\n",
       " 'quantitative': 2059,\n",
       " 'meta': 1654,\n",
       " 'analysis': 320,\n",
       " 'program': 2016,\n",
       " 'evaluation': 1003,\n",
       " 'found': 1154,\n",
       " 'emphasized': 951,\n",
       " 'student': 2442,\n",
       " 'engagement': 967,\n",
       " 'through': 2550,\n",
       " 'highly': 1285,\n",
       " 'interactive': 1444,\n",
       " 'practice': 1956,\n",
       " 'activities': 246,\n",
       " 'promoted': 2027,\n",
       " 'effectiveness': 927,\n",
       " 'those': 2544,\n",
       " 'relied': 2150,\n",
       " 'self': 2285,\n",
       " 'directed': 871,\n",
       " '17': 37,\n",
       " 'other': 1829,\n",
       " 'scholars': 2257,\n",
       " 'extent': 1062,\n",
       " 'internet': 1453,\n",
       " 'contribute': 727,\n",
       " 'skill': 2365,\n",
       " 'development': 852,\n",
       " 'decision': 792,\n",
       " 'making': 1601,\n",
       " 'acknowledge': 239,\n",
       " 'interaction': 1443,\n",
       " 'students': 2443,\n",
       " 'ask': 363,\n",
       " 'trainees': 2578,\n",
       " 'tackle': 2497,\n",
       " 'problem': 2002,\n",
       " 'solving': 2382,\n",
       " 'writing': 2758,\n",
       " 'foster': 1152,\n",
       " '18': 40,\n",
       " 'recent': 2098,\n",
       " 'randomized': 2072,\n",
       " 'controlled': 731,\n",
       " 'trial': 2595,\n",
       " 'comparing': 632,\n",
       " 'biostatistics': 466,\n",
       " 'example': 1019,\n",
       " 'marked': 1614,\n",
       " 'similar': 2343,\n",
       " 'improvements': 1364,\n",
       " 'among': 315,\n",
       " 'volunteer': 2697,\n",
       " 'scientists': 2263,\n",
       " 'suggesting': 2466,\n",
       " 'cost': 748,\n",
       " 'scalable': 2248,\n",
       " 'alternative': 307,\n",
       " '19': 43,\n",
       " 'evaluators': 1004,\n",
       " 'importance': 1357,\n",
       " 'component': 651,\n",
       " 'establish': 991,\n",
       " 'collaborative': 609,\n",
       " 'atmospheres': 388,\n",
       " 'individuals': 1396,\n",
       " 'safely': 2233,\n",
       " 'engage': 965,\n",
       " 'dialog': 859,\n",
       " 'controversial': 734,\n",
       " '20': 53,\n",
       " 'botswana': 482,\n",
       " 'one': 1800,\n",
       " 'highest': 1283,\n",
       " 'rates': 2081,\n",
       " 'hiv': 1292,\n",
       " 'world': 2756,\n",
       " 'stable': 2407,\n",
       " 'democracy': 812,\n",
       " 'commitment': 620,\n",
       " 'public': 2048,\n",
       " 'health': 1260,\n",
       " 'has': 1254,\n",
       " 'attracted': 398,\n",
       " 'attention': 397,\n",
       " 'past': 1872,\n",
       " 'three': 2547,\n",
       " 'decades': 789,\n",
       " 'hub': 1310,\n",
       " 'behavioural': 439,\n",
       " 'epidemiological': 984,\n",
       " 'clinical': 595,\n",
       " 'related': 2139,\n",
       " 'aids': 286,\n",
       " 'co': 602,\n",
       " 'morbid': 1700,\n",
       " 'conditions': 679,\n",
       " 'foreign': 1142,\n",
       " 'academic': 215,\n",
       " 'universities': 2634,\n",
       " 'ngos': 1741,\n",
       " 'study': 2446,\n",
       " 'centers': 544,\n",
       " 'gaborone': 1180,\n",
       " 'capital': 511,\n",
       " 'city': 584,\n",
       " 'order': 1818,\n",
       " 'support': 2477,\n",
       " 'going': 1220,\n",
       " 'post': 1949,\n",
       " 'graduate': 1228,\n",
       " 'opportunities': 1807,\n",
       " 'tertiary': 2518,\n",
       " 'educational': 921,\n",
       " 'expanded': 1042,\n",
       " 'during': 910,\n",
       " 'decade': 788,\n",
       " 'resulting': 2198,\n",
       " 'growing': 1240,\n",
       " 'portfolio': 1941,\n",
       " 'rapid': 2075,\n",
       " 'increase': 1382,\n",
       " 'volume': 2696,\n",
       " 'complexity': 649,\n",
       " 'activity': 247,\n",
       " 'necessitated': 1730,\n",
       " 'strengthening': 2436,\n",
       " 'system': 2490,\n",
       " 'generated': 1193,\n",
       " 'demand': 810,\n",
       " 'trained': 2576,\n",
       " 'workforce': 2750,\n",
       " 'able': 205,\n",
       " 'reviews': 2208,\n",
       " 'serve': 2298,\n",
       " 'at': 386,\n",
       " 'levels': 1546,\n",
       " 'enterprise': 976,\n",
       " 'is': 1481,\n",
       " 'currently': 771,\n",
       " 'conducted': 681,\n",
       " 'government': 1225,\n",
       " 'appointed': 341,\n",
       " 'body': 478,\n",
       " 'committee': 621,\n",
       " 'hrdc': 1305,\n",
       " 'functions': 1172,\n",
       " 'an': 317,\n",
       " 'board': 474,\n",
       " 'irb': 1476,\n",
       " 'rec': 2093,\n",
       " 'administrative': 264,\n",
       " 'unit': 2630,\n",
       " 'ministry': 1674,\n",
       " 'hru': 1306,\n",
       " 'increasingly': 1386,\n",
       " 'relies': 2151,\n",
       " 'university': 2635,\n",
       " 'ub': 2614,\n",
       " 'its': 1491,\n",
       " 'retains': 2200,\n",
       " 'authority': 402,\n",
       " 'final': 1113,\n",
       " 'approval': 345,\n",
       " 'protocols': 2035,\n",
       " 'irbs': 1477,\n",
       " 'local': 1572,\n",
       " 'level': 1545,\n",
       " 'hospitals': 1299,\n",
       " 'like': 1552,\n",
       " 'suffer': 2460,\n",
       " 'paucity': 1877,\n",
       " 'knowledge': 1511,\n",
       " 'despite': 839,\n",
       " 'concerted': 671,\n",
       " 'moh': 1693,\n",
       " 'supported': 2478,\n",
       " 'part': 1858,\n",
       " 'various': 2675,\n",
       " 'agencies': 279,\n",
       " 'fogarty': 1133,\n",
       " 'center': 543,\n",
       " 'institutes': 1423,\n",
       " 'wellcome': 2721,\n",
       " 'trust': 2599,\n",
       " 'european': 998,\n",
       " 'developing': 851,\n",
       " 'trials': 2596,\n",
       " 'partnership': 1871,\n",
       " 'edctp': 918,\n",
       " 'remains': 2156,\n",
       " 'quite': 2064,\n",
       " 'small': 2370,\n",
       " '21': 66,\n",
       " '23': 71,\n",
       " 'same': 2236,\n",
       " 'limited': 1556,\n",
       " 'pool': 1936,\n",
       " 'often': 1789,\n",
       " 'service': 2301,\n",
       " 'multiple': 1715,\n",
       " 'projects': 2022,\n",
       " 'committees': 622,\n",
       " 'entities': 977,\n",
       " 'sponsored': 2401,\n",
       " 'endorsed': 962,\n",
       " 'participation': 1865,\n",
       " 'lose': 1580,\n",
       " 'completion': 647,\n",
       " 'lucrative': 1586,\n",
       " 'jobs': 1494,\n",
       " 'regional': 2130,\n",
       " 'organizations': 1823,\n",
       " 'empirical': 952,\n",
       " 'data': 778,\n",
       " 'approaches': 343,\n",
       " 'teaching': 2507,\n",
       " 'guide': 1244,\n",
       " 'expand': 1041,\n",
       " 'traditional': 2574,\n",
       " 'batswana': 425,\n",
       " 'discuss': 876,\n",
       " 'familial': 1078,\n",
       " 'communal': 626,\n",
       " 'settings': 2306,\n",
       " 'would': 2757,\n",
       " 'suggest': 2464,\n",
       " 'group': 1237,\n",
       " 'advantages': 271,\n",
       " 'low': 1582,\n",
       " 'flexible': 1124,\n",
       " 'medium': 1647,\n",
       " 'requires': 2174,\n",
       " 'reliable': 2148,\n",
       " 'access': 217,\n",
       " 'rarity': 2078,\n",
       " 'where': 2725,\n",
       " 'even': 1005,\n",
       " 'within': 2744,\n",
       " 'sporadic': 2404,\n",
       " 'email': 946,\n",
       " 'communication': 627,\n",
       " 'tool': 2564,\n",
       " 'fairly': 1076,\n",
       " 'particularly': 1867,\n",
       " 'professionals': 2012,\n",
       " 'although': 309,\n",
       " 'computers': 663,\n",
       " 'connectivity': 691,\n",
       " 'frequently': 1163,\n",
       " 'workplace': 2752,\n",
       " 'cafes': 498,\n",
       " '2010': 64,\n",
       " 'launched': 1526,\n",
       " 'multi': 1714,\n",
       " 'year': 2765,\n",
       " 'endeavour': 961,\n",
       " 'create': 758,\n",
       " 'wide': 2738,\n",
       " 'community': 629,\n",
       " 'advisory': 272,\n",
       " 'boards': 475,\n",
       " 'response': 2190,\n",
       " 'initiated': 1414,\n",
       " 'situated': 2356,\n",
       " 'outside': 1838,\n",
       " 'well': 2720,\n",
       " 'interest': 1446,\n",
       " 'new': 1739,\n",
       " 'sites': 2355,\n",
       " 'locations': 1573,\n",
       " 'seven': 2308,\n",
       " 'communities': 628,\n",
       " 'were': 2722,\n",
       " 'foci': 1129,\n",
       " 'proposed': 2028,\n",
       " 'selected': 2283,\n",
       " 'because': 430,\n",
       " 'satellite': 2242,\n",
       " 'campuses': 503,\n",
       " 'institute': 1422,\n",
       " 'sciences': 2261,\n",
       " 'ihs': 1335,\n",
       " 'institution': 1424,\n",
       " 'responsible': 2193,\n",
       " 'nurses': 1762,\n",
       " 'affiliated': 275,\n",
       " 'run': 2227,\n",
       " 'district': 889,\n",
       " 'teams': 2509,\n",
       " 'dhts': 854,\n",
       " 'semi': 2287,\n",
       " 'urban': 2648,\n",
       " 'settlements': 2307,\n",
       " 'jointly': 1495,\n",
       " 'recs': 2111,\n",
       " 'record': 2106,\n",
       " 'implementing': 1354,\n",
       " 'initiative': 1415,\n",
       " 'faced': 1069,\n",
       " 'challenges': 551,\n",
       " 'relating': 2140,\n",
       " 'course': 754,\n",
       " 'content': 721,\n",
       " 'appropriate': 344,\n",
       " 'methodologies': 1662,\n",
       " 'grant': 1229,\n",
       " 'penn': 1881,\n",
       " 'enabled': 956,\n",
       " 'authors': 403,\n",
       " 'fhb': 1104,\n",
       " 'jfm': 1493,\n",
       " 'work': 2748,\n",
       " 'professional': 2011,\n",
       " 'across': 240,\n",
       " 'resultant': 2197,\n",
       " 'was': 2707,\n",
       " 'used': 2654,\n",
       " 'environment': 979,\n",
       " 'population': 1940,\n",
       " 'methods': 1663,\n",
       " 'assess': 371,\n",
       " 'personalized': 1900,\n",
       " 'intervention': 1457,\n",
       " 'using': 2658,\n",
       " 'augment': 401,\n",
       " 'seminars': 2289,\n",
       " 'enrolled': 974,\n",
       " 'semester': 2286,\n",
       " 'long': 1576,\n",
       " 'included': 1374,\n",
       " 'two': 2610,\n",
       " 'day': 783,\n",
       " 'separated': 2292,\n",
       " 'six': 2359,\n",
       " 'week': 2715,\n",
       " 'assigned': 376,\n",
       " 'random': 2070,\n",
       " 'arms': 355,\n",
       " 'portion': 1942,\n",
       " 'control': 730,\n",
       " 'arm': 354,\n",
       " 'complete': 643,\n",
       " 'modules': 1692,\n",
       " 'citi': 583,\n",
       " '24': 72,\n",
       " 'asked': 364,\n",
       " 'comment': 617,\n",
       " 'sent': 2291,\n",
       " 'via': 2685,\n",
       " 'weekly': 2716,\n",
       " 'interim': 1449,\n",
       " 'mks': 1684,\n",
       " 'reviewed': 2207,\n",
       " 'respondents': 2189,\n",
       " 'analyses': 319,\n",
       " 'returned': 2203,\n",
       " 'individualized': 1394,\n",
       " 'comments': 618,\n",
       " 'each': 913,\n",
       " 'tested': 2520,\n",
       " 'start': 2413,\n",
       " 'again': 277,\n",
       " 'months': 1698,\n",
       " 'after': 276,\n",
       " 'both': 481,\n",
       " 'had': 1247,\n",
       " 'completed': 644,\n",
       " 'discussions': 878,\n",
       " 'team': 2508,\n",
       " 'test': 2519,\n",
       " 'scheduled': 2252,\n",
       " 'advance': 267,\n",
       " 'seminar': 2288,\n",
       " 'rather': 2082,\n",
       " 'overall': 1840,\n",
       " 'itself': 1492,\n",
       " 'illustrated': 1340,\n",
       " 'figure': 1108,\n",
       " 'designed': 837,\n",
       " 'explore': 1053,\n",
       " 'hypotheses': 1322,\n",
       " 'improve': 1361,\n",
       " 'who': 2731,\n",
       " 'participated': 1864,\n",
       " 'greater': 1232,\n",
       " 'gains': 1181,\n",
       " 'made': 1590,\n",
       " 'developed': 849,\n",
       " 'independently': 1389,\n",
       " 'feedback': 1094,\n",
       " 'weeks': 2717,\n",
       " 'examination': 1016,\n",
       " 'demonstrate': 814,\n",
       " 'correctly': 745,\n",
       " 'counterparts': 751,\n",
       " 'project': 2021,\n",
       " 'consultation': 714,\n",
       " 'office': 1787,\n",
       " 'ord': 1817,\n",
       " 'it': 1485,\n",
       " 'underwent': 2624,\n",
       " 'approved': 346,\n",
       " 'pennsylvania': 1882,\n",
       " 'documented': 895,\n",
       " 'consent': 693,\n",
       " 'signature': 2337,\n",
       " 'prior': 1998,\n",
       " 'randomization': 2071,\n",
       " 'first': 1117,\n",
       " 'bioethics': 459,\n",
       " 'input': 1418,\n",
       " 'mk': 1683,\n",
       " 'pk': 1922,\n",
       " 'topics': 2567,\n",
       " 'detailed': 842,\n",
       " 'below': 444,\n",
       " 'cover': 756,\n",
       " 'major': 1598,\n",
       " 'aspects': 367,\n",
       " 'values': 2665,\n",
       " 'systems': 2493,\n",
       " 'loose': 1579,\n",
       " 'leaf': 1528,\n",
       " 'binders': 454,\n",
       " 'containing': 719,\n",
       " 'copies': 739,\n",
       " 'declaration': 793,\n",
       " 'helsinki': 1266,\n",
       " 'belmont': 442,\n",
       " 'report': 2163,\n",
       " 'council': 750,\n",
       " 'medical': 1646,\n",
       " 'science': 2260,\n",
       " 'cioms': 581,\n",
       " 'biomedical': 465,\n",
       " 'involving': 1473,\n",
       " 'rule': 2226,\n",
       " '28': 81,\n",
       " 'mock': 1685,\n",
       " 'adapted': 250,\n",
       " 'created': 759,\n",
       " 'de': 785,\n",
       " 'novo': 1758,\n",
       " 'took': 2563,\n",
       " 'place': 1923,\n",
       " 'led': 1536,\n",
       " 'lecturers': 1534,\n",
       " 'drawn': 903,\n",
       " 'key': 1500,\n",
       " 'topic': 2566,\n",
       " 'areas': 351,\n",
       " 'segments': 2281,\n",
       " 'written': 2759,\n",
       " 'english': 968,\n",
       " 'official': 1788,\n",
       " 'language': 1519,\n",
       " 'standard': 2412,\n",
       " 'follows': 1139,\n",
       " 'value': 2664,\n",
       " 'driven': 904,\n",
       " 'goals': 1219,\n",
       " 'codes': 605,\n",
       " 'ich': 1325,\n",
       " 'good': 1221,\n",
       " 'structure': 2440,\n",
       " 'informed': 1407,\n",
       " 'what': 2723,\n",
       " 'we': 2711,\n",
       " 'know': 1510,\n",
       " 'about': 208,\n",
       " 'role': 2218,\n",
       " 'whose': 2734,\n",
       " 'autonomous': 408,\n",
       " 'impaired': 1347,\n",
       " 'autonomy': 409,\n",
       " 'children': 571,\n",
       " 'vulnerabilitysample': 2700,\n",
       " 'selection': 2284,\n",
       " 'recruitment': 2110,\n",
       " 'retention': 2202,\n",
       " 'drafting': 902,\n",
       " 'exercise': 1031,\n",
       " 'model': 1686,\n",
       " 'exercises': 1032,\n",
       " 'platform': 1928,\n",
       " 'free': 1160,\n",
       " 'involved': 1471,\n",
       " 'offers': 1786,\n",
       " 'tracks': 2572,\n",
       " 'targets': 2502,\n",
       " 'non': 1747,\n",
       " 'us': 2650,\n",
       " 'federally': 1093,\n",
       " 'funded': 1173,\n",
       " 'historical': 1289,\n",
       " 'evolution': 1012,\n",
       " 'process': 2006,\n",
       " 'guidance': 1243,\n",
       " 'documents': 896,\n",
       " 'pre': 1957,\n",
       " 'tests': 2522,\n",
       " 'family': 1080,\n",
       " 'fhi': 1105,\n",
       " 'curriculum': 773,\n",
       " 'retc': 2201,\n",
       " '29': 82,\n",
       " 'additional': 256,\n",
       " 'section': 2276,\n",
       " 'five': 1119,\n",
       " 'corresponding': 747,\n",
       " 'choice': 573,\n",
       " 'statements': 2417,\n",
       " 'describing': 833,\n",
       " 'potential': 1951,\n",
       " 'results': 2199,\n",
       " 'analyzed': 321,\n",
       " 'regression': 2133,\n",
       " 'anova': 328,\n",
       " 'majority': 1599,\n",
       " 'several': 2310,\n",
       " 'high': 1281,\n",
       " 'quality': 2058,\n",
       " 'casebooks': 527,\n",
       " 'focused': 1131,\n",
       " '30': 83,\n",
       " '31': 86,\n",
       " 'felt': 1097,\n",
       " 'participant': 1861,\n",
       " 'enhanced': 970,\n",
       " 'if': 1334,\n",
       " 'everywhere': 1008,\n",
       " 'presented': 1983,\n",
       " 'groups': 1239,\n",
       " 'reflected': 2124,\n",
       " 'lived': 1569,\n",
       " 'experience': 1044,\n",
       " 'find': 1115,\n",
       " 'dilemmas': 870,\n",
       " 'embedded': 948,\n",
       " 'relate': 2138,\n",
       " 'seem': 2278,\n",
       " 'unlikely': 2638,\n",
       " 'occur': 1780,\n",
       " 'particular': 1866,\n",
       " 'themselves': 2529,\n",
       " 'introduce': 1462,\n",
       " 'dilemma': 869,\n",
       " 'seen': 2280,\n",
       " 'unfamiliar': 2625,\n",
       " 'culture': 766,\n",
       " 'described': 832,\n",
       " 'therapy': 2533,\n",
       " 'elicit': 941,\n",
       " 'conversation': 737,\n",
       " 'between': 447,\n",
       " 'mothers': 1707,\n",
       " 'daughters': 782,\n",
       " 'adolescent': 265,\n",
       " 'sexual': 2314,\n",
       " 'behaviors': 438,\n",
       " 'any': 331,\n",
       " 'setting': 2305,\n",
       " 'intended': 1435,\n",
       " 'stimulate': 2426,\n",
       " 'confidentiality': 686,\n",
       " 'protocol': 2034,\n",
       " 'raised': 2069,\n",
       " 'practical': 1954,\n",
       " 'confidences': 685,\n",
       " 'traditionally': 2575,\n",
       " 'take': 2499,\n",
       " 'mother': 1706,\n",
       " 'daughter': 781,\n",
       " 'young': 2770,\n",
       " 'woman': 2746,\n",
       " 'another': 327,\n",
       " 'older': 1791,\n",
       " 'female': 1099,\n",
       " 'relative': 2142,\n",
       " 'examples': 1020,\n",
       " 'enhance': 969,\n",
       " 'learners': 1530,\n",
       " 'abilities': 203,\n",
       " 'effectively': 926,\n",
       " 'problems': 2003,\n",
       " 'steps': 2423,\n",
       " 'resolution': 2182,\n",
       " 'relevant': 2146,\n",
       " 'attended': 395,\n",
       " 'series': 2295,\n",
       " 'anonymous': 326,\n",
       " 'survey': 2482,\n",
       " 'evaluating': 1002,\n",
       " 'certificates': 547,\n",
       " 'issued': 1483,\n",
       " 'sat': 2241,\n",
       " 'received': 2096,\n",
       " 'tasks': 2504,\n",
       " 'least': 1533,\n",
       " 'four': 1156,\n",
       " 'seventy': 2309,\n",
       " 'registered': 2131,\n",
       " 'september': 2293,\n",
       " 'signed': 2338,\n",
       " 'following': 1138,\n",
       " 'presentation': 1982,\n",
       " 'subject': 2450,\n",
       " 'withdrew': 2743,\n",
       " 'remaining': 2155,\n",
       " '71': 169,\n",
       " 'prospective': 2030,\n",
       " 'employed': 953,\n",
       " '12': 25,\n",
       " 'distribution': 887,\n",
       " 'stages': 2410,\n",
       " '43': 109,\n",
       " '60': 146,\n",
       " '41': 107,\n",
       " '57': 140,\n",
       " 'second': 2272,\n",
       " 'session': 2302,\n",
       " 'december': 791,\n",
       " 'motivated': 1708,\n",
       " '35': 93,\n",
       " '85': 184,\n",
       " 'completing': 646,\n",
       " '54': 136,\n",
       " 'fully': 1166,\n",
       " 'eight': 935,\n",
       " '22': 69,\n",
       " 'partially': 1860,\n",
       " 'twenty': 2606,\n",
       " '62': 154,\n",
       " '40': 105,\n",
       " 'item': 1486,\n",
       " 'hypothesized': 1324,\n",
       " 'improved': 1362,\n",
       " 'when': 2724,\n",
       " 'measured': 1640,\n",
       " 'correct': 743,\n",
       " 'answers': 330,\n",
       " 'battery': 426,\n",
       " 'questions': 2062,\n",
       " 'scores': 2266,\n",
       " 'out': 1834,\n",
       " 'answered': 329,\n",
       " 'mean': 1639,\n",
       " 'range': 2073,\n",
       " '38': 100,\n",
       " 'sd': 2268,\n",
       " '39': 103,\n",
       " 'score': 2265,\n",
       " 'strongly': 2439,\n",
       " 'correlated': 746,\n",
       " '65': 160,\n",
       " '0001': 0,\n",
       " 'assignment': 377,\n",
       " 'did': 860,\n",
       " 'main': 1594,\n",
       " 'effect': 924,\n",
       " 'appeared': 334,\n",
       " 'only': 1803,\n",
       " 'weak': 2712,\n",
       " 'provided': 2039,\n",
       " 'substantially': 2456,\n",
       " 'better': 446,\n",
       " '01': 4,\n",
       " 'average': 414,\n",
       " '003': 2,\n",
       " 'whether': 2727,\n",
       " 'our': 1832,\n",
       " 'issue': 1482,\n",
       " 'identification': 1328,\n",
       " 'examined': 1018,\n",
       " 'sensitivity': 2290,\n",
       " 'detect': 844,\n",
       " 'real': 2090,\n",
       " 'specificity': 2394,\n",
       " 'erroneous': 986,\n",
       " 'nonissues': 1749,\n",
       " 'false': 1077,\n",
       " 'positive': 1944,\n",
       " 'rate': 2080,\n",
       " 'there': 2534,\n",
       " 'total': 2568,\n",
       " 'incorrect': 1381,\n",
       " 'identified': 1329,\n",
       " '001': 1,\n",
       " '91': 194,\n",
       " 'higher': 1282,\n",
       " 'no': 1746,\n",
       " 'minimum': 1673,\n",
       " 'subgroup': 2448,\n",
       " 'just': 1497,\n",
       " 'fewer': 1103,\n",
       " 'chosen': 576,\n",
       " '051': 10,\n",
       " '92': 195,\n",
       " 'proved': 2036,\n",
       " 'popular': 1938,\n",
       " 'conference': 683,\n",
       " 'facilities': 1070,\n",
       " 'away': 417,\n",
       " 'seemed': 2279,\n",
       " 'encourage': 959,\n",
       " 'remain': 2154,\n",
       " 'full': 1165,\n",
       " 'off': 1784,\n",
       " 'contributed': 728,\n",
       " 'dropout': 905,\n",
       " 'attendees': 396,\n",
       " 'trainee': 2577,\n",
       " 'attend': 393,\n",
       " 'sizeable': 2362,\n",
       " 'registration': 2132,\n",
       " 'turnout': 2605,\n",
       " 'due': 908,\n",
       " 'competing': 639,\n",
       " 'demands': 811,\n",
       " 'campus': 502,\n",
       " 'strike': 2438,\n",
       " 'last': 1521,\n",
       " 'minute': 1675,\n",
       " 'changes': 555,\n",
       " 'grading': 1227,\n",
       " 'effort': 931,\n",
       " 'precluded': 1959,\n",
       " 'mandatory': 1607,\n",
       " 'consideration': 696,\n",
       " 'should': 2325,\n",
       " 'future': 1177,\n",
       " 'merits': 1653,\n",
       " 'certification': 548,\n",
       " 'credits': 760,\n",
       " 'achieve': 235,\n",
       " 'maintain': 1596,\n",
       " 'target': 2501,\n",
       " 'concern': 669,\n",
       " 'outset': 1837,\n",
       " 'discourage': 874,\n",
       " 'task': 2503,\n",
       " 'option': 1815,\n",
       " 'attaching': 390,\n",
       " 'times': 2555,\n",
       " 'predicted': 1965,\n",
       " 'complain': 641,\n",
       " 'recurrent': 2112,\n",
       " 'difficulty': 868,\n",
       " 'accessing': 221,\n",
       " 'third': 2541,\n",
       " 'without': 2745,\n",
       " 'home': 1295,\n",
       " 'nine': 1744,\n",
       " 'characterized': 557,\n",
       " 'sketchy': 2364,\n",
       " 'developers': 850,\n",
       " 'determine': 846,\n",
       " 'opting': 1814,\n",
       " 'formats': 1147,\n",
       " 'blogs': 470,\n",
       " 'chat': 560,\n",
       " 'rooms': 2221,\n",
       " 'assignments': 378,\n",
       " 'module': 1691,\n",
       " 'excellent': 1022,\n",
       " 'most': 1704,\n",
       " 'interesting': 1447,\n",
       " 'aspect': 366,\n",
       " 'providing': 2041,\n",
       " ...}"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vectorizer2.vocabulary_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "418"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vectorizer2.vocabulary_.get(\"background\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'analyzer': 'word',\n",
       " 'binary': False,\n",
       " 'decode_error': 'strict',\n",
       " 'dtype': numpy.float64,\n",
       " 'encoding': 'utf-8',\n",
       " 'input': 'content',\n",
       " 'lowercase': True,\n",
       " 'max_df': 1.0,\n",
       " 'max_features': None,\n",
       " 'min_df': 1,\n",
       " 'ngram_range': (1, 1),\n",
       " 'norm': 'l2',\n",
       " 'preprocessor': None,\n",
       " 'smooth_idf': True,\n",
       " 'stop_words': None,\n",
       " 'strip_accents': None,\n",
       " 'sublinear_tf': False,\n",
       " 'token_pattern': '(?u)\\\\b\\\\w\\\\w+\\\\b',\n",
       " 'tokenizer': None,\n",
       " 'use_idf': True,\n",
       " 'vocabulary': None}"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vectorizer2.get_params()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "X = vectorizer2.inverse_transform(Y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "x = X[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "y = Y[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>background</td>\n",
       "      <td>(0, 418)\\t0.006488649324705781\\n  (0, 1611)\\...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            0                                                  1\n",
       "0  background    (0, 418)\\t0.006488649324705781\\n  (0, 1611)\\..."
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(zip(x, y))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Code for viewing documents"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['field work',\n",
       " 'fieldwork',\n",
       " 'field study',\n",
       " 'field site',\n",
       " 'field area',\n",
       " 'study site',\n",
       " 'study location',\n",
       " 'study area',\n",
       " 'research site',\n",
       " 'research location',\n",
       " 'sampling site',\n",
       " 'sampling location',\n",
       " 'sampling area']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "def print_article(article):\n",
    "    print(\"\"\"\n",
    "PMC ID\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TITLE\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "METADATA\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TEXT MATCHES\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "EXTRACTED TEXT\n",
    "\n",
    "{}\"\"\".format(article.get(\"_id\"),\n",
    "             article.get(\"article_title\"),\n",
    "             article.get(\"article_meta\"),\n",
    "             article.get(\"text_matches\"),\n",
    "             article.get(\"extracted_text\").strip()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def view_one_article(query={ \"$match\": { \"text_matches\": { \"$in\": terms } } }):\n",
    "    query = [\n",
    "        query,\n",
    "        { \"$sample\": { \"size\": 1 } }\n",
    "    ]\n",
    "    article = articles.aggregate(query).next()\n",
    "    print(\"\"\"\n",
    "PMC ID\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TITLE\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "METADATA\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TEXT MATCHES\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "EXTRACTED TEXT\n",
    "\n",
    "{}\"\"\".format(article.get(\"_id\"),\n",
    "             article.get(\"article_title\"),\n",
    "             article.get(\"article_meta\"),\n",
    "             article.get(\"text_matches\"),\n",
    "             article.get(\"extracted_text\").strip()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['field work',\n",
       " 'fieldwork',\n",
       " 'field study',\n",
       " 'field site',\n",
       " 'field area',\n",
       " 'study site',\n",
       " 'study location',\n",
       " 'study area',\n",
       " 'research site',\n",
       " 'research location',\n",
       " 'sampling site',\n",
       " 'sampling location',\n",
       " 'sampling area']"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "6210248\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "The Effect of Different Habitat Types and Ontogenetic Stages on the Diet Shift of a Critically Endangered Fish Species, Coreius guichenoti (Sauvage and Dabry de Thiersant, 1874)\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "{'has_body': True, 'article_type': 'research-article'}\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['study area', 'sampling site']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "1. Introduction\n",
      "\n",
      "The identification and protection of fish critical habitats are central to the active management of species at risk [1,2]. Generally, the spawning grounds and nursery, migration, and rearing areas on which fish species depend directly to complete their life history cycles, are identified as the critical habitats for fish species [1]. However, many fish species use different critical habitats within different life history stages [3,4]. The distance between critical habitats may extend to several hundred or thousands of kilometers. Due to environmental heterogeneity in different habitats, the fish species may exhibit plastic dietary patterns and diet shifts due to variation in the accessibility and availability of potential food sources [3,5,6]. Acquiring information on the diet shift among different habitats and identifying the habitat factors causing the diet shift helps guide appropriate conservation measures to protect fish resources [3].\n",
      "\n",
      "For many fish species, diet shifts are also closely related to behavioral and morphological changes resulting from their ontogeny [7,8,9]. Generally, ontogenetic shifts in diet between size classes or life history stages are common among fish populations [7], serving as an adaptive survival strategy for resisting environmental variability. In particular, many studies have reported that the diet shift in the development stages between juvenile and adult or during early life stages, is clearly observed or estimated [8,9]. Some studies have also found that the diet shift between young individuals or young adults and adults, occurs in certain fish species [10]. However, for certain fish species, only fully mature or spawning individuals can be found in the limited spawning grounds, the diet shifts or differences between immature and fully mature stages in the spawning grounds are poorly understood, and the diet shifts of individuals between the spawning grounds and other functional habitats are also not clear. Identifying the diet shift or differences between these immature and fully mature ontogenetic stages in the spawning grounds is very important for the full understanding of the reproductive ecology of these fish species, because it provides information about the reproductive adaptation strategy to specific habitat variability [11]. Additionally, intraspecific diet competition is common for many fish species [10,12], and knowing that the intensity of intraspecific food overlaps between immature and fully mature individuals in the spawning grounds when a large number of spawning fishes aggregate in the spawning grounds is also very important for the management and conservation of fish resources. \n",
      "\n",
      "For characterizing diet differences among different habitats and/or ontogenetic stages, stable isotope analysis (SIA) [13,14] has been widely used in recent years. SIA has been broadly employed to track food sources over long time scales. The stable isotope ratios of carbon (δ13C) and nitrogen (δ15N) have been used most frequently. While δ13C has been used to trace the potential food sources of consumers, δ15N has been used to identify the trophic positioning of consumers, because it shows an obvious stepwise enrichment between trophic levels [14,15,16]. Based on the results of SIA, mixing models are built for estimating the proportions of source (prey) contributions to a mixture (consumer) based on biotracer data to determine diet composition, niche variation, population structure, animal movement and so on [14,17]. In recent years, Bayesian mixing models have been used most frequently, because they can improve upon simpler linear mixing models by explicitly taking into account uncertainty in source values, categorical and continuous covariates, and prior information. Depending on the R software, certain R packages (e.g., SIAR and MixSIAR), including the Bayesian mixing model, have been developed for analyzing biotracer data [17,18].\n",
      "\n",
      "Largemouth bronze gudgeon (LBG), Coreius guichenoti (Sauvage and Dabry de Thiersant, 1874), is an endemic and commercial fish species in the upper Yangtze Basin [19] that plays an important role in the transfer of energy from primary producers and consumers to top predators [20]. Due to the loss of spawning grounds and hydrologic obstructions resulting from the construction of cascaded hydropower plants on the Jinsha River (the upper segments of the upper Yangtze) beginning in 2008, this species has been listed as critically endangered on the Red List of China’s Vertebrates since 2016 [21]. The fish species spawn with drifting eggs, and their spawning grounds are only distributed in the mainstream of the middle and lower reaches of the Jinsha River as well as in the lower mainstream of the Yalong River (one of the main tributaries of the Jinsha River). Before the impoundment of the Three Gorges Reservoir (TGR) in 2003, the fish species made use of two critical natural riverine areas (the spawning area and the feeding, nursery area) during their entire life history cycles. The spawning area in the Jinsha River specifically includes the river segment from Xinshi Town to Tiger Leaping Gorges Town, while the feeding and nursery area generally consist of the river segments downstream of Xinshi Town, which can extend to the middle reaches of the Yangtze River (Figure 1). Compared to the feeding and nursery areas, the spawning areas generally display harsher natural habitat characteristics, for instance, higher mean flow velocity and sediment content, low primary productivity and fewer microhabitat types [22]. Since the impoundment of the TGR in 2003, a newly formed reservoir habitat has been added to the habitats of this fish species, possibly exerting important impacts on the population survival of the fish species [23]. However, little is known about the influences of TGR operation on the feeding ecology of this fish species.\n",
      "\n",
      "In addition, the feeding ecology (e.g., food preference in different development stages) of LBG remains unclear, although the diet compositions of this species in certain non-spawning river sections have been documented [20,24,25]. In particular, the feeding habits of this species in the spawning grounds are poorly understood. Moreover, food preference, diet plasticity, and intraspecific competition in the spawning grounds remain unclear. In particular, the fully mature or spawning individuals of this species can only be found in the spawning grounds [19,26]; hence, whether the intraspecific competition in the spawning grounds occurs or is strengthened during the spawning season is still unknown.\n",
      "\n",
      "In the present study, we first employ SIA to attain the values of δ13C and δ15N of LBG muscle tissues collected from three habitat types (spawning ground, natural riverine feeding and nursery area and TGR area) and then examine the differences in δ13C and δ15N among these three habitat types by use of one-way analysis of variance (ANOVA). Finally, we estimate the food preference and diet plasticity by employing MixSIAR within the R software and the intraspecific competition between immature and fully mature individuals according to the SIBE (Stable Isotope Bayesian Ellipses in R) procedure within the R package SIBR by combining the data with the collections of potential basal food sources in the three habitats. Based on the above analysis, we aim to answer the following questions: (1) Do diet shifts of LBG among three habitat types occur? (2) Does a diet shift of LBG between immature and fully mature individuals occur, and is the intraspecific competition between immature and fully mature individuals obvious in the spawning ground during the spawning season? (3) What are the main food sources of LBG in different habitats, and does the species display evident diet plasticity? (4) What effect has the formation of the TGR had on this species? \n",
      "\n",
      "\n",
      "2. Materials and Methods\n",
      "\n",
      "\n",
      "2.1. Study Area\n",
      "\n",
      "The study was carried out at three sites in the Yangtze River basin representing the three habitat types of LBG (Figure 1). The spawning ground sampling site (SG) located in Jiaopingdu (Kunming, Yunnan Province) is a natural riverine segment, where the substrates are mainly composed of boulders and large gravels [27]. Due to the rapid velocity (generally >2 m/s) and high sediment concentration in the water columns year-round, this river segment generally exhibits low biomasses of plankton and zoobenthos. The natural riverine feeding and nursery area sampling site (NRFA) located in the Jiangjin site (southwestern Chongqing Municipality) is also a natural riverine segment with a wide river surface (generally >200 m in river width), exhibiting the complex flow regime and the diversified substrate combinations that are beneficial as foraging and spawning grounds for many fish species [28]. The TGR area sampling site (TGRA) located in the Fuling District (central Chongqing Municipality) is the newly formed reservoir section where the bed substrates are mainly dominated by small-diameter sands and muds and display low water flow velocity and enriched nutrient loading [23]. Before the sampling period, two vast hydropower reservoirs (the Xiangjiaba and Xiluodu Reservoirs) were filled in October 2012 and May 2013, respectively. During the filling periods of the above two reservoirs, important potamodromous fish species (including the species LBG) were captured by gillnets and hooks from the sections downstream of the Xiangjiaba Dam and then transported to the upstream of the Xiluodu Reservoir. However, no fish passage facilities were constructed on these two dams, although a feasibility analysis of fish passage construction involving these two dams had been extensively discussed. \n",
      "\n",
      "\n",
      "2.2. Sample Collection and Processing\n",
      "\n",
      "Fish sample collections in the SG, NRFA and TGRA were taken from 5 June to 16 June 2013. Care of animals used in scientific procedures was under supervision of the Ethics Committee of the Institute of Hydroecology, Ministry of Water Resources and Chinese Academy of Sciences. The survey period was during the main spawning season of LBG that produced obvious peaks in egg and larvae numbers [19]. Fish specimens were collected using gillnets with mesh sizes of 1.5, 3, 5, 5.5, 8 and 10 cm. The gillnets were usually placed in the water column from dusk until dawn depending on the fishermen’s experience. Generally, a total of 6–12 gillnets were used on each sampling day at each sampling site. For the SG site, due to the construction of cascade dams (three dams have been built downstream of the spawning grounds of LBG) in the lower reaches of the Yalong River since 2000, the LBG individuals upstream of the Jinping-II Hydropower Station (Figure 1) had difficultly passing through the three dams to the SG site [26]; hence, the fish specimens in this study area were mainly collected from the Jinsha River population. A total of 175 specimens were collected, including 68 individuals from SG, 55 from NRFA, and 52 from TGRA. All individuals were collected and measured (total length, TL; standard length, SL, measured to the nearest 1 cm) and weighed (body weight, BW, measured to the nearest 0.1 g). Simultaneously, through dissection, the gonad development of LBG was classified as six stages: (I) immature fish stage; (II) quiescent stage; (III) ripening stage; (IV) ripeness stage; (V) reproduction stage; (VI) spent stage by visual observation [29]. In this study, we defined the individuals in the IV–VI stages as fully mature individuals and the other stages as immature individuals [29]. In addition, scales from all collected fish were extracted from the area below the dorsal fin and above the lateral line as the age-determining materials [30]. The scales were first cleaned with water and NaOH, placed between two slides and then observed in the field using a SZ61TR stereo microscope (Olympus, Japan) to determine the age of each fish [30]. After removing the skins and scales, muscle tissues were taken from below the dorsal fin. These tissues were frozen and freeze-dried for 72 h and stored in glass vials at −20 °C until further analysis [31].\n",
      "\n",
      "At the same time as the fish sampling, the water temperature (WT, °C), dissolved oxygen (DO, mg/L), pH, and velocity of water flow (m/s) were measured in situ using YSI 6600 (YSI Inc., Yellow Springs, OH, USA) and LGY II (Nanjing Shengrong Instrument and Equipment Inc., Nanjing, China) instruments. Samples were simultaneously used to determine chemical parameters, including total nitrogen (TN, mg/L), total phosphorus (TP, mg/L), biochemical oxygen demand (BOD5, mg/L), and chemical oxygen demand (CODMn, mg/L), according to standard methods (Environmental Quality Standards for Surface Water, GB3838-2002) (SEPA and AQSIQ, 2002). Based on the modified criteria of Jiang et al. [32], the predominant substrate was assigned to one of three types: (1) cobble plus boulder (CPB), (2) pebble plus gravel (PPG), or (3) sand plus silt (SPS). \n",
      "\n",
      "According to previous gut-content data [24,25], the four primary food source types (mollusks, macrocrustaceans, aquatic insect larvae and particulate organic matter (POM)) for LGB were collected at the three sampling sites during the fish sample collection. POM was collected with a phytoplankton net, and then samples were filtered using a filtration apparatus coupled to a vacuum pump and previously calcined filters (Whatman QM-A quartz-fiber filters) [33]. Macrocrustacean samples (crabs and shrimps, including Macrobrachium nipponense, Procambarus clarki, Exopalaemon modestus, Caridina and Sinopotamon) were collected using 1.5 cm mesh nylon trap nets. Mollusks (snails and Limnoperna fortunei) were collected using Peterson samplers in areas of slack water or directly by hand from submerged rocks in the sampling section [20]. Mollusk specimens were placed in aerated distilled water overnight to eliminate intestinal contents, and their shells and soft tissue were then collected [20]. Aquatic insect larvae, mainly composed of species of Trichoptera, Odonata and Ephemerida, were captured from shoal areas using hand nets [20]. A total of 7 mollusks, 11 macrocrustaceans, 10 aquatic insect larvae and 6 POM samples were collected in SG. A total of 10 mollusks, 9 crustaceans, 11 aquatic insect larvae and 6 POM samples were collected in NRFA. In total, 8 mollusks, 12 crustaceans, 12 aquatic insect larvae and 6 POM samples were collected in TGRA. All collected samples of potential food sources were frozen at −20 °C in the field, sealed in Styrofoam containers and transported to the laboratory for further processing.\n",
      "\n",
      "All samples were dried at 60 °C for 24 h in an air-circulated oven, after which lipids were removed by extracting the samples in chloroform-methanol 2:1 by volume overnight at room temperature [34]. These samples were then independently homogenized using a mortar and pestle and then weighed to the appropriate mass for SIA [31]. Approximately 2 mg powder was weighed and encapsulated for animal samples, whereas 15–30 mg was taken for POM. The δ13C and δ15N were measured using an IsoPrime100 stable isotope ratio mass spectrometer equipped with a vario PYRO cube elemental analyzer (Environmental Stable Isotope Lab., Chinese Academy of Agricultural Sciences, Beijing) and expressed as parts per thousand (‰) relative to standards. The Vienna Pee Dee Belemnite and atmospheric N2 were used as the standards for carbon and nitrogen, respectively [33].\n",
      "\n",
      "\n",
      "2.3. Data Analysis\n",
      "\n",
      "One-way analysis of variance (ANOVA) was first used to determine whether δ13C and δ15N values varied significantly among the three habitat types [35]. Prior to the ANOVA, the Kolmogorov–Smirnov test and the Bartlett test were conducted. To meet these two test requirements, the parameters (δ13C and δ15N values) were transformed by dividing by the respective minimum value. If significance was detected, the Tukey multiple comparison test was conducted, or a nonparametric Kruskal–Wallis test was conducted. Following the gonadal examination, the samples in the SG were classified as two ontogenetic stages, including the fully matured stage (FMS) and the immature stage (IS). The independent-samples t test was chosen to identify the differences in δ13C and δ15N values between FMS and IS, revealing the diet differences between immature individuals and fully mature individuals in the SG. \n",
      "\n",
      "The SIBER procedure (Stable isotope Bayesian Ellipses in R) [36] within the R package SIAR [37] was applied for examining variation in the trophic niche among the three habitat types (SG, NRFA and TGRA) and between the two ontogenetic stages (FMS and IS). The SIBER procedure is based on the concept that multiple stable isotope ratios measured from consumers represent niche dimensions, and broadly varying stable isotope ratios are indicative of a wider consumer isotopic niche. Here, we used Bayesian Standard Area Ellipses (SEAB, also called isotopic niches area) for comparing individual groups with each other and the Monte Carlo Markov-Chain simulation for constructing ellipses characterizing isotopic variation that provide a robust indicator of isotope niche width [14]. Simultaneously, by calculating the probabilities from posterior distributions of the parameters of model given the prior data, the differences in SEAB between different groups were estimated. To conduct these comparisons, the maximum-likelihood-based probabilities were calculated [14].\n",
      "\n",
      "Finally, the Bayesian mixing model MixSIAR [38] was used to determine the likely contribution of potential food sources to the diets of LBG in three habitat types (SG, NRFA and TGRA) and in two ontogenetic stages (FMS and IS). The MixSIAR model is written in the open-source language R. In this model, the values of two biotracers (δ13C and δ15N) of the 175 collected LBG individuals were used as the mixture data, the mean values and standard deviations of δ13C and δ15N in the four potential foods as the food source data, and the discrimination data were acquired by referring to published literature that involved the study areas of this paper [20,39]. The habitat type or the ontogenetic stage in the spawning ground in this model was considered as the fixed effect, because the diet of each group was separately estimated, rather than the overall diet [38]. Prior to model analysis, we first plotted consumer and source isotope data to ensure that the source data bound the consumer data and then used the diagnostic tests (Gelmin-Rubin and Geweke) and trace plots to examine the model convergence [38]. Finally, the proportional contributions of each potential food estimated from the mixing models were reported as range estimates rather than limited to the mean to avoid misrepresenting the uniqueness of the results [40]. \n",
      "\n",
      "All statistical analyses were performed using R software [41], and significance levels for all analyses were set to p < 0.05.\n",
      "\n",
      "\n",
      "3. Results\n",
      "\n",
      "\n",
      "3.1. Characteristics of Specimens and Sampling Sites\n",
      "\n",
      "A total of 175 LBG samples were collected from the three sampling sites (SG, NRFA and TGRA). The mean values and distribution ranges of total length, standard length, body weight and age for immature and fully mature individuals in the three sampling sites are shown in Table 1. For the immature individuals, the mean values of total length, standard length, body weight and age in NRTA were greater than those in SG and in TGRA.\n",
      "\n",
      "For three sampling sites, eight physical and chemical parameters showed significant differences among the three habitat types (SG, NRFA and TGRA, Table 2). The velocity of water flow and CPB generally decreased from the SG to the TGRA. Conversely, the physiochemical parameters, including pH and SPS, showed significantly increasing patterns from SG or NRFA to TGRA (Table 2).\n",
      "\n",
      "\n",
      "3.2. Comparisons of Carbon and Nitrogen Stable Isotope Ratios among the Three Habitat Types (SG, NRFA and TGRA) and between Two Ontogenetic Stages (FMS and IS)\n",
      "\n",
      "Comparisons of δ13C and δ15N values across this study period (5 June to 16 June 2013) using one-way ANOVA showed strong evidence of isotopic differences among the three habitat types (Figure 2; δ13C: F = 50.88, p < 0.001; δ15N: F = 21.80, p < 0.001). The mean δ13C values at SG were significantly greater than those at NRFA and TGRA (p = 0.007 and p < 0.001), while the mean δ15N values at NRFA were lower than those at SG and TGRA (both ps < 0.001) (Figure 2).\n",
      "\n",
      "No significant difference was detected for δ13C values between the two ontogenetic stages, while strong evidence of δ15N difference between the two ontogenetic stages was detected (Figure 3; t test, δ13C: t = 1.50, p = 0.14; δ15N: t = −2.52, p < 0.001).\n",
      "\n",
      "\n",
      "3.3. Variations in Isotopic Niche Widths among Three Habitat Types (SG, NRFA and TGRA) and between Two Ontogenetic Stages (FMS and IS)\n",
      "\n",
      "The standard area ellipses (SEAB) among habitat types and between ontogenetic stages are shown in Figure 4. Compared to the individuals from SG or NRFA, the individuals from TGRA had the largest mean (95% credibility limits) isotopic niche width of 4.41 (3.30–5.73)‰2 (Figure 4A). However, the maximum-likelihood pairwise comparison showed that there were obvious differences between the individuals from SG and from NRFA (probability (p) = 99.27%) and between the individuals from SG and from TGRA (p = 99.95%), whereas there were no significant differences (p = 80.10%) between the individuals from NRFA and from TGRA. The mean SEAB in FMS individuals (2.12‰2) was lower than that in IS individuals (2.58‰2) (Figure 4B), but there was only an 80.20% probability of differences in isotopic niche width between these two ontogenetic stages. \n",
      "\n",
      "\n",
      "3.4. Differences in Potential Food Sources among the Three Habitat Types (SG, NRFA and TGRA) and between Two Ontogenetic Stages (FMS and IS)\n",
      "\n",
      "For all four potential food sources, significant differences in δ13C and δ15N were observed among the three sampling sites (all Ps < 0.05, Table 3). From SG to TGRA, an increasing δ13C gradient in aquatic insect larvae and POM was observed, while δ15N values for these two prey items showed declining trends (Table 3). For mollusks, the δ13C and δ15N values displayed a decreasing trend from SG to NRFA or TGRA, but for macrocrustaceans, the δ13C values showed an increasing (from SG to NRFA) and decreasing (from NRFA to TGRA) trend, while the δ15N values displayed a decreasing gradient from SG to TGRA (Table 3).\n",
      "\n",
      "The percentages of potential food contributions for LBG among the different habitats were determined using MixSIAR and are displayed in Table 4. The contributions of potential food sources to LBG showed higher proportions of POM and lower proportions of aquatic insect larvae, macrocrustaceans and mollusks in SG than in NRFA and TGRA. POM was identified as the main food sources for LBG in SG, and its mean contribution decreased from 91.0% in SG to 34.4% in NRFA and then to 1.4% in TGRA. In NRFA, the POM and macrocrustaceans became the main food sources for LBG (34.4%; 8.9–73.9% credible interval (CI) and 56.4%; 16.8–82.8% CI, respectively). In TGRA, the main food sources for LBG were mollusks and macrocrustaceans (47.4%; 31.9–61.4% CI and 49.4%; 35.6–65.0% CI, respectively).\n",
      "\n",
      "For the FMS and IS individuals, POM contributed the highest proportion for LBG (85.1%; 74.4–93.7% CI and 94.2%; 84.3–98.9%, respectively), and the other three potential food sources displayed little contribution (Table 5). However, the contributions of aquatic insect larvae, macrocrustaceans and mollusks to FMS were greater than the contributions of these three food sources to IS (Table 5).\n",
      "\n",
      "\n",
      "4. Discussion\n",
      "\n",
      "Previous studies found that differences in diet were strongly related to variations in physical and chemical parameters at different ecological scales [3,33]. In the present study, significant variation existed in the isotope signatures of LBG muscle tissues among three highly heterogeneous habitats, indicating an obvious diet shift among three habitats (SG, NRFA and TGRA). This finding indicated that habitat attributes played an important role in the diet composition of LBG. In particular, the diet shift between NRFA and TGRA showed that the newly-formed reservoir can obviously influence the diet composition of LBG. In addition, the diet shifts among the three habitat types may highlight the importance of maintaining habitat connectivity between different habitat types when considering the completeness of the life cycle. Due to the high velocity of water flow (generally >2 m/s, Table 2), the SG displayed low primary productivity and only provided limited foods for LBG, while the NRFA and TGRA showed relatively low water flow velocity and enriched nutrient levels [23], which can support the survival and maturation of more individuals. The movement between habitats may be a life history strategy for long-term adaptation to environmental variation. \n",
      "\n",
      "Many studies have also documented that diet shift for a given fish was closely associated with maturation and spawning [42,43]. Fuiman and Faulk [44] even found that the diet shift in a marine batch-spawning fish could impact the transfer of arachidonic acid from diet to egg. However, in the present study, the results provided no statistical support for the difference in δ13C values between immature and fully mature stages in SG, but a significant difference in δ15N values between these two stages was detected. These results might imply that for LBG, the diet shift or the difference between immature and fully mature stages was not obvious even though the food intake of these two developmental stages was different. In addition, the individuals investigated in this study only included the young, subadult and adult individuals but did not include larvae or juvenile stages because of the lack of distribution of these life stages in spawning grounds. Gao et al. [45] reported that a large number of larvae and juveniles of LBG could be collected only downstream of Xinshi Town (Figure 1). \n",
      "\n",
      "Previous studies have used muscle tissue to study the isotopic shift between ontogenetic stages (particularly during early ontogeny) [7,8]. In the present study, we used muscle tissue to compare the isotopic difference between immature and fully mature stages based upon a complete investigation of the life history characters of LBG. LBG is a long-lived fish species, and its growth inflexion age is 8.13 years [46], indicating that fish individuals collected in this study (Table 1, age range 3–6 years) still exhibited a rapid growth speed. Thus, sufficient nutrients were available to these individuals to meet growth needs. Additionally, LBG produces eggs and spawns many times in its life [26]; therefore, this species must allocate enough nutrients to muscle tissue in order to adapt to the harsh conditions (e.g., high stream velocity and low diet biomass) of the spawning grounds. \n",
      "\n",
      "Many migratory fishes typically move or migrate to the spawning grounds with limited areas for spawning [47,48], resulting in an obvious increase in population density in spawning grounds during the spawning seasons. This behavior likely intensifies the competition among individuals for foods. In particular, for certain fish that spawn in the upper segments of one large river (with limited food availability), their spawning aggregations may lead to the more serious food shortages to the presence of local populations of the same fish species [49]. Our study showed no significant differences in isotopic niche width between immature and fully mature individuals in SG, indicating the same prey sources for individuals of these two ontogenetic stages. Obviously, marked intraspecific food competition and similar feeding ecology existed between these two ontogenetic stages. Additionally, this finding suggested relatively limited aggregation times of LBG reproductive cohorts in the spawning ground, because of the very similar ecological niche between immature and fully mature individuals and limited food availability in the spawning ground. Fish aggregate when spawning and leave after spawning. Tao et al. [50] reported that LBG aggregated into the spawning grounds for spawning during the flood and departed after spawning. In this context, maintaining the undisturbed state of LBG spawning grounds during the spawning season is vital for the protection of this fish species. \n",
      "\n",
      "There is a longstanding argument about the diet preference of LBG. Huang and Deng [24] and Liu et al. [25] found that LBG was an omnivorous fish with carnivorous preference by using stomach content analysis (SCA) based on the samples from two feeding grounds of LBG, while Li et al. [20] used stable isotope analysis (SIA) and found that LBG was an omnivorous fish with phytophagous preference based upon the samples from another feeding ground of LBG. Our results confirmed that LBG was an omnivorous fish with a carnivorous preference based upon the samples collected across all critical habitat types. Our results were consistent with the SCA results [24,25], but were obviously different from the SIA results [20]. This difference in SIA results possibly resulted from the differences in sample processing and statistical methods. Li et al. [20] did not remove lipids in sample processing and used the IsoSource software for determining the likely contribution of potential food sources to the diets of LBG, both of which likely produced obvious deviations from the true values. Post et al. [51] noted that the lipids in samples should be removed when the C/N ratio of samples is >4 because a high lipid content possibly affects the δ13C value. Moore and Semmens [17] determined that although the IsoSource model can address the uncertainty associated with more sources than an analytical solution, it does not formally incorporate the variation in isotope signatures or fractionation. When used to analyze the same data, IsoSource generally obtained different results from the MixSIAR model [17]. Many studies have suggested that a combination of SCA and SIA is more suitable than using SCA alone to determine temporal and regional variations in the diet compositions of certain fish species [52,53]. However, in view of the uneven digestion and assimilation rates of different potential foods [53], we chose the SIA for determining the potential food sources of LBG. In addition, our findings demonstrated that the determination of diet for a fish species, especially the food preference of an omnivorous species, must include analysis across all critical habitats.\n",
      "\n",
      "Our MixSIAR output indicated that POM was the dominant food source for LBG in the spawning ground (SG), while macrocrustaceans and POM were the major diet components for LBG in the feeding and nursery ground (NRFA). In particular, macrocrustaceans constituted the highest proportion of the LBG diet in NRFA. This finding implied that diet plasticity occurred from the spawning ground to the feeding and nursery ground. This fish species can change its feeding behavior by adapting to the food species composition in the surrounding environment. The main food sources of LBG became macrocrustaceans and mollusks when LBG entered the TGR, further confirming the dietary plasticity of LBG. Furthermore, the isotopic niche widths became wider from the spawning ground to the river section of the reservoir area. These results indicate that downstream feeding grounds, especially the reservoir area, can provide more diversified and abundant food sources. This variation pattern seems common in many fish populations of dietary specialists [54,55], which is possibly an adaptation mechanism for decreasing mortality at the early life history stage. LBG spawn pelagic eggs in resource-limited areas, but its larvae and juveniles grow up in areas with abundant food [19]. This condition is likely a typical case for obtaining the optimal habitat suitability by diet plasticity.\n",
      "\n",
      "Compared to immature individuals, the fully mature individuals feed on higher proportions of aquatic insect larvae, macrocrustaceans and mollusks, although the difference between these two groups of individuals was not significant. This possibly resulted from the size-related difference [7]. In this study, the body sizes of fully mature individuals were obviously larger than those of immature individuals, implying that fully mature individuals likely exhibit a greater ability to feed on larger sizes of macrocrustaceans and mollusks. However, the POM still played a very important role in the spawning ground, suggesting the importance of the inputs of terrestrial organic detritus into the upper segments of the river [56] because these river segments generally displayed low concentrations of autochthonous algae and endogenous organic detritus [57].\n",
      "\n",
      "In addition, the isotope niche width of LBG individuals in the reservoir area was broader than that of both other sampling sites. The results showed that the individuals in the reservoir area fed more heavily on higher-order consumers, suggesting an important function of the newly formed reservoir area as the feeding grounds of LBG. However, there are seventeen very large dams in operation, under construction, or planned, on the mainstem of the upper Yangtze River [57]. This activity will inevitably block the migratory routes of LBG from the feeding and nursery grounds to the spawning grounds, inducing a partial or even total loss of the spawning grounds upstream of Xiludu Dam [57]. Although certain sections of TGR (in particular of the tail of the reservoir) can serve as the feeding grounds of LBG, they cannot function as the spawning area [19]. Therefore, maintaining habitat connectivity between the TGR and the spawning grounds in the Jinsha River is very important for the survival and long-term persistence of LBG. Hence, fish passage facilities, especially upstream facilities, should be constructed in the dams downstream of LBG spawning grounds. \n",
      "\n",
      "\n",
      "5. Conclusions\n",
      "\n",
      "This study of the diet shifts of a critically endangered fish species among three habitat types (SG, NRFA and TGRA) and between two ontogenetic stages (FMS and IS) showed that an obvious diet shift existed among three habitat types, but not between ontogenetic stages. Our results determined the food preference of LBG and confirmed the serious intraspecific competition between immature and fully matured individuals in a resource-limited area, both of which can guide future work conducted in captive breeding (aiming to restore the wild population of LBG via artificial release) of LBG. The results also showed that the formation of the Three Gorges Reservoir (TGR) in the upper Yangtze River altered the trophic structure, likely providing more high-quality food organisms. However, due to the cascade hydropower construction of the Jinsha River, it is necessary to maintain the habitat connectivity between the TGR and the spawning grounds in the Jinsha River for LBG survival and persistence over the long term [57]. In this context, the construction of fish passage facilities in the downstream LBG spawning grounds in the lower reaches of the Jinsha River should be implemented in the near future. In addition, due to significant differences in diet compositions among different critical habitats, a sufficiently broad view of trophic ecology, especially in regard to the food preference of a single fish species, made it clear that the effects of habitat types must be fully investigated. \n",
      "\n",
      "This study was in compliance with the International Union for Conservation of Nature (IUCN) Policy Statement on Research Involving Species at Risk of Extinction because the species C. guichenoti was not be listed as an endangered species during the sampling period (5 June to 16 June 2013), and this species was listed as critically endangered on the Red List of China’s Vertebrates in 2016.\n"
     ]
    }
   ],
   "source": [
    "view_one_article()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "problematic_terms = terms[-3:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "6002991\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "Decadal stability in genetic variation and structure in the intertidal seaweed Fucus serratus (Heterokontophyta: Fucaceae)\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "{'has_body': True, 'article_type': 'research-article'}\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['sampling site', 'sampling location']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Background\n",
      "\n",
      "Understanding temporal stability of genetic structure and diversity is crucial for the utility of temporal snapshots in conservation management and to infer how climate-induced range shifts might affect the future distribution and adaptive potential of species. In trailing edge populations, effective population size and genetic diversity are considered major keys to adaptive potential and subsequent persistence under climate change [1, 2]. In contrast, the evolutionary potential and survival of low-diversity leading edge populations [3] may be either enhanced or impaired by the ‘surfing’ of new mutations that can rapidly increase in frequency over iterated founder events, depending on whether the new mutations are primarily favorable or deleterious [4–9].\n",
      "\n",
      "Studies that assess temporal genetic stability are rare in marine species, and mostly confined to fisheries management to ensure sustainable exploitation of economically important species [10–14]. While high gene flow explained 5 to 24-year long stability in genetic variability and structure in Chinook salmon and Atlantic herring [10, 11], large fluctuations in allele frequencies were recorded over a few months in small and closed populations of the intertidal isopod Jaera albifrons [15]. However, high gene flow does not always warrant temporal genetic stability, as several marine species with long-lived planktonic larvae showed stronger temporal than spatial differentiation over 3 to 9 years [16–18]. On the other hand, low gene flow does not necessarily result in genetic instability over time, although genetic drift in small and closed populations can be expected to be high. For example, genetic diversity and population structure remained stable over 5–12 years in relatively closed populations of the seagrass Zostera marina [19] and over 2 years in nine out of 10 locally differentiated populations of the isopod Excirolana braziliensis [20]. These contrasting results demonstrate that a species’ life history alone does not necessarily predict its genetic stability over time.\n",
      "\n",
      "Due to their high sensitivity to rising temperatures, responses of marine intertidal species are considered as early warning signals for the impact of climate change [21–25]. Among global climate change factors, ocean warming is considered the most severe threat for marine macrophytes [26–28]. Over the next century, ecological niche models predict the disappearance of intertidal fucoid brown algae along their southern trailing edges and a poleward extension of their northern leading edges [26, 29]. Fucoid brown algae (Heterokontophyta; Fucaceae) are habitat-forming ecosystem engineers supporting species-rich intertidal communities along temperate rocky shores [30–33]. Thus, range shifts of fucoids will undoubtedly trigger major ecological changes along temperate rocky shores of the North Atlantic.\n",
      "\n",
      "Ecological niche models, however, do not consider the species’ plastic and adaptive potential that could mitigate the predicted northward shifts. Adaptive potential depends largely on a population’s genetically effective size, Ne [34], or the size of an ideal population that undergoes the same rate of genetic change as the real population [35]. At low Ne, and low gene flow between populations, genetic drift generally plays the predominant role, effectively neutralizing selection, and eroding genetic diversity through stochastic fixation or loss of allelic variations [36–38]. Although Ne and temporal stability of genetic diversity patterns are particularly important for restoration and conservation efforts of fucoid seaweeds, only a single Norwegian population of F. serratus has so far been assessed [39].\n",
      "\n",
      "The canopy-forming seaweed F. serratus is an excellent model for the study of temporal evolution and stability of genetic structure and diversity across a range of contrasting temperature regimes. It is one of the dominant intertidal seaweeds along the Northeast-Atlantic rocky shore from northern Portugal to northern Norway [40]. Arctic regions are predicted to become thermally suitable through 2100 under CO2 emission scenario projections [26]. In contrast, regions south of the Brittany coast of France are predicted to become unsuitable [26], as temperatures will rise beyond the species’ potential for thermal acclimatization [41]. The susceptibility of F. serratus to climate change is expected to vary regionally, given the species’ regional patterns of genetic diversity [42], in combination with low gene flow between local populations [43].\n",
      "\n",
      "Genetic diversity of F. serratus is highest in the two former, large glacial refugia (20–18 thousand years ago (kya)) in Southwest Ireland, and Brittany [42, 43]. The third refugium in the Northwest Iberian peninsula is characterized by a high proportion of private alleles, and currently represents the species’ isolated trailing edge, where recurrent cycles of thermally induced extinction and recolonization have eroded genetic diversity [42, 43]. Currently, sea surface temperatures reach 22 °C, and although below the lethal limit of F. serratus (25 °C) [40, 44], inhibit growth, physiological performance and reproductive capacity [45–48].\n",
      "\n",
      "Genetic diversity of F. serratus decreases from its mid-range of distribution towards higher latitudes and is lowest in leading edge populations in northern Norway [42, 43]. Low genetic diversity in leading-edge populations is explained by the populations’ relatively young age and their derivation from small founder populations that carried only a subset of the genetic variation from glacial refugia to the north after the ice retreated, ca. 15–10 kya.\n",
      "\n",
      "While Ne is a good indicator for temporal genetic stability, its estimation relies on temporally-spaced genetic data [49]. Due to this complication in sampling design, Ne of F. serratus has been estimated in only a single population close to Bergen (Norway) over eight years [39]. The estimated Ne between 73 and 386 was regarded insufficient for long-term survival under environmental change [39]. However, a thorough appraisal of the spatial distribution and temporal stability of Ne and genetic diversity throughout the species’ latitudinal range of distribution cannot be inferred from a single location.\n",
      "\n",
      "Estimating climate change susceptibility in a species with low gamete/zygote dispersal requires to assess temporal genetic stability across its latitudinal and thermal range of distribution. In this study, we estimated Ne of F. serratus across latitude and temperature at seven locations with decadal sampling (2000 and 2010), a period in which Europe experienced three heat waves in 2003, 2006, and 2010 [50–52]. Here we evaluate whether range shifts in the north or strong selection pressures in the south have resulted in measurable changes in genetic diversity and population structure. In populations that are dominated by genetic drift and with small adaptive potential, we expected to find a decline in genetic diversity over the past decade. Finally, we discuss whether genetic diversity may be sufficiently stable to buffer environmental change and mitigate the current range shift predictions.\n",
      "\n",
      "\n",
      "Methods\n",
      "\n",
      "\n",
      "Sampling\n",
      "\n",
      "Individuals were sampled ca. ten years apart from the same seven populations spanning the latitudinal distribution of F. serratus (Fig. 1). Ethical approval is not required for research work with the seaweed/macroalga F. serratus. Field collections did not require specific permits and the species is neither endangered nor protected. Sampling involved removing a thumbnail-sized piece of tissue from ca. 50 to 100 individuals at each sampling site and did not threaten either the individual or the local population. Live samples were never transferred to other countries or locations within any of the countries. In all cases the specimens were collected within the context of various grants (see funding information) that involved at least one of the co-authors and one or more colleagues from the country where the collection was made.\n",
      "\n",
      "Fig. 1 Sampling sites. Coordinates, years of collection, sampling sizes (n), and daily average sea surface temperatures (SST) at each of the seven sampling sites. SSTs were identical between the two Norwegian sampling sites as well as between the two French and the two Spanish sampling sites. Summer temperatures were exceptionally high at the Danish and Spanish sampling sites during the first two of three heat waves that Europe experienced in years 2003, 2006, and 2010\n",
      "\n",
      "Variability in daily average sea surface temperatures and surface air temperatures at the sampling locations (Fig. 1, Additional file 1), recorded from 1999 to 2011, were extracted from the NOAA/OI/SST/V2 dataset (0.25° resolution, described in [53]) and the CPC Global Temperature dataset (0.5° resolution) provided by NOAA/OAR/ESRL/PSD, Boulder, Colorado, USA, [54]). Thermal variability was replicated in the two Norwegian, the two French, and the two Spanish samples, respectively. In Denmark, only a single population was sampled at two time points. Individual tissues were blotted dry and stored in silica prior to transport for subsequent DNA extraction.\n",
      "\n",
      "\n",
      "Microsatellite genotyping\n",
      "\n",
      "DNA was extracted from 2 mg silica dried tissue according to [55] with the modifications described in [56], followed by a purification step with the OneStep-96 PCR Inhibitor Removal Kit (Zymo Research, Irvine, USA) and a 1:3 dilution of the purified product. The samples were genotyped for a total of 31 microsatellite markers: 11 anonymous loci (L20, L38, L58, and L94 described in [57]; B113, B128, E6, E9, D39, A198, and F4 described in [58]) and 20 loci linked to expressed sequence tags (ESTs: F12, F22, F34, F36, F60, F45, F50, F17, F72, F49, F14, F21, F58, F19, F37, F65, F59, F69, F9, and described in [56]) (Additional file 2).\n",
      "\n",
      "Polymerase chain reactions (PCRs) with 5 μl total volume contained 1 μl purified DNA template, 1.34 μl nuclease-free Water (Ambion, Thermo Fisher Scientific), 2.5 μl of AmpliTaq Gold 360 MM (Applied Biosystems, Life Technologies) and 0.08 μl of each forward and reverse primer (each primer at 20 μM; forward primer labeled with 6FAM, NED, PET or VIC; Applied Biosystems, Life Technologies). PCR was performed in a Veriti 96-Well Thermal Cycler (Applied Biosystems, Life Technologies). The conditions are depicted in Additional file 3 and specified for each marker in Additional file 2.\n",
      "\n",
      "The fragment lengths were determined on an ABI 3500xl Genetic Analyzer from 1 μl of diluted PCR products (specified for each marker in Additional file 2) mixed with 8.9 μl of HiDi Formamide (Life Technologies) and 0.1 μl of Gene Scan 500 LIZ Size Standard (Life Technologies) after 5 min denaturation at 95 °C. Allele calling was performed with the GeneMapper v 4.1 Software (Applied Biosystems, Thermo Fisher Scientific).\n",
      "\n",
      "\n",
      "Data analysis\n",
      "\n",
      "The microsatellite raw dataset (Additional file 4) was corrected for allelic dropout with a Maximum Likelihood approach [59] using the program MicroDrop [60]. From the corrected data (Additional file 5), nine markers E9, F14, F17, F36, F37, F59, F60, F65, and L20 were removed from the full set of 31 markers before further analyses with the remaining 22 markers because the proportion of missing data for the excluded markers exceeded 12% in at least one of the populations.\n",
      "\n",
      "\n",
      "Diversity estimates\n",
      "\n",
      "Average locus heterozygosity Hexp (bias-corrected [61]), allelic richness α (the average number of alleles per locus) and multi-locus heterozygosity (MLH), the number of heterozygous loci per individual divided by the number of loci, were calculated for each sampling location. Regional estimates were obtained after pooling the two spatial samples from each of the Norwegian, Spanish and French regions. Regional estimates were not possible for the Danish region because only one population was sampled. Hexp was calculated with the R package ‘DEMEtics’ [62], and α was estimated with the R package ‘PopGenReport’. For local estimates, α was normalized to a sample size of 24, the smallest number samples in a population. For regional estimates α was normalized to a sample size of 24, and additionally, to a sample size of 50. MLH was estimated with the R package ‘InbreedR’. Inbreeding coefficients FIS [63] were estimated with the R package ‘Demerelate’ and tests for significant deviation from 0 were based on 1000 iterations. We tested for significant temporal changes of Hexp, α, FIS, and MLH at each sampling location with Wilcoxon rank sum tests in R [64]. To assess temporal evolution of diversity estimates, we tested for correlation between current and historical local measures with a Spearman’s rank correlation in R [64]. Additionally, we tested for significant differences between average present-day and historical values using Wilcoxon Rank Sum tests in R [64].\n",
      "\n",
      "Effective population sizes (Ne) were estimated with an assumed generation time of 2 years [65] with the R package ‘NB’ after removing loci with only one allele: Locus F9 for the Kirkenes population, locus F72 for the Ribadeo1 population and loci F21 and F72 for the Ribadeo2 population.\n",
      "\n",
      "\n",
      "Genetic differentiation\n",
      "\n",
      "Population structure was determined with Bayesian clustering methods implemented in the software Structure v2.3.4 [66]. Acceptance of six clusters (K) was determined with the δ K Method [67] in the R package ‘pophelper’ [68].\n",
      "\n",
      "Temporal genetic changes at each sampling location and geographic genetic differentiation within and between all historical and recent samples were estimated by the fixation index FST [69] using GENETIX 4.05 [70] and the differentiation index Dest [71]) using the R package ‘DEMEtics’ 0.8–7 [62]. Dest more correctly measures the true genetic differentiation compared with FST for multi-allelic markers such as microsatellites [62, 71]. Statistical significance of the pairwise comparisons was based on 10,000 permutations for FST and on 1000 Bootstrap repeats for Dest. To assess temporal stability of geographic differentiation, we tested for correlation between recent and historical FST and Dest values with Spearmans’s rank correlation in R [64]. Additionally, we tested for significant differences between average present-day and historical values using Wilcoxon Rank Sum tests in R [64]. Finally, we tested for correlation between temporal genetic differentiation (FST, Dest) and Ne with Pearson’s product moment correlation in R [64].\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "\n",
      "Genetic structure\n",
      "\n",
      "Bayesian clustering with the program Structure revealed clear differences between regions but not with time (Fig. 2). Historical and present-day FST values were strongly positively correlated (r = 0.93, p < 0.00001), and the overall historical FST value (0.21) did not differ significantly (p = 0.567) from the present-day value (0.22), indicating that spatial genetic differentiation between populations was globally consistent over time (Fig. 3a). Historical and present-day Dest values supported these findings as the overall values (0.40 and 0.42, respectively) did not differ significantly (p = 0.636) and were positively correlated (r = 0.97, p < 0.00001, Fig. 3b). Isolation by distance was indicated by stronger differentiation among than within countries (Additional file 6).\n",
      "\n",
      "Fig. 2 Clustering of samples. Sample assignment to six clusters (colors) with the program Structure shows consistent geographic differentiation between sampling times. Here, new and old refers to the two sampling years specified in Fig. 1\n",
      "\n",
      "Fig. 3 Present and historical genetic differentiation. Population differentiation estimated by FST (a) and Dest (b) with a 1:1 reference line\n",
      "\n",
      "Temporal changes, however, were noted on a local level. Local differentiation between the Norwegian populations decreased from 2004 to 2010 (Additional file 6). All but the French population ‘Ile de Siec’ changed significantly in genetic variation over time, as indicated by significant changes in FST and Dest (Additional file 7). The Spanish population ‘Ribadeo2’ showed significant temporal change in FST but not in Dest.\n",
      "\n",
      "\n",
      "Genetic variation/diversity\n",
      "\n",
      "Stable population diversities through time were indicated by significant correlations of historical and present-day intra-population diversity indices (Fig. 4; Hexp: r = 0.86, p = 0.02, MLH: r = 1, p = 0.0004; α: r = 0.96, p = 0.003; FIS: r = 0.82, p = 0.03). Moreover, average present-day values did not differ significantly (p > 0.05) from average historical values (Hexp: present = 0.56, historical = 0.56; MLH: present = 0.61, historical = 0.62; α: present = 6.22, historical = 6.36; FIS: present = − 0.10, historical = − 0.10).\n",
      "\n",
      "Fig. 4 Genetic diversity across latitudes. Present and historical diversity estimates of a) multi-locus heterozygosity (MLH), b) allelic richness (α), c) expected heterozygosity (Hexp), and d) inbreeding (FIS), with 1:1 reference lines representing unchanged temporal evolution\n",
      "\n",
      "Local and regional diversity estimates (Additional file 8) were highest in France and lower at the northern and southern distribution edges (Fig. 4). Regional α estimates (standardized to 50 samples) exceeded local estimates (standardized to 24 samples) in all regions (Additional file 8).\n",
      "\n",
      "Effective population size (Ne) was highest in the French population ‘Ile de Siec’ (Ne = 10,000,000) and lowest in the Norwegian population ‘Grense Jakobselv’ (Ne = 62) (Fig. 5, Additional file 7). Ne for the other populations ranged from 700 to 200 in the order: Gjerild Klint > Green Top > Ribadeo2 > Ribadeo1 > Kirkenes. At both sampling time points, none of the diversity estimates were significantly correlated with effective population size (all p > 0.09). The temporal decrease in MLH in Ribadeo2 was strong but not significant (p = 0.051, Additional file 7). The FIS in ‘Kirkenes’ was significantly negative (p = 0.043, Additional file 8).\n",
      "\n",
      "Fig. 5 Effective population sizes across latitudes. Effective population size (Ne) at each sampling location with 95% confidence intervals\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "The spatial distribution of genetic diversity has important implications for conservation and management as it reveals a species’ strong and weak points with regards to stability and evolutionary capacity [72–74]. Given the key ecological role of F. serratus [30–32, 75–77], the positive effect of genetic diversity may reach beyond the species level to affect community structure and increase species richness and stability of the entire associated ecosystem [78–81]. We are not aware of seaweed studies that have investigated positive ecosystem effects of genetic diversity, but genetic diversity enhanced heat-stress survival in germlings of Fucus vesiculosus [27]. Furthermore, in the habitat forming seagrass Z. marina, genotypic diversity not only enhanced biomass production, but also abundance of the associated fauna under near-lethal water temperatures [82] and community resistance to grazing [83]. Thus, maintaining genetic diversity in F. serratus is also expected to be important for conservation and management of the entire intertidal ecosystem along temperate rocky shores. Across the latitudinal range of F. serratus, genetic diversity and differentiation remained stable for 5–10 generations at regional scales, and in all but the Norwegian region at local spatial scales (Figs. 2, 3, 4). This suggests that, despite low gene flow between populations, effective population sizes have remained large enough to maintain genetic variation at least on the short term. Temporal genetic differentiation was systematically lower than local differentiation, and 1–2 orders of magnitude lower than regional differentiation (Additional file 9). This implies that temporal snapshots provide valuable information for conservation management of fucoid seaweeds, as they reliably reflect diversity and differentiation patterns for at least a decade.\n",
      "\n",
      "\n",
      "Ne comparisons\n",
      "\n",
      "In all but the Norwegian populations, Ne was estimated as > 260, a size reported as the median estimate for stable populations in over 83 studies spanning a diverse range of taxa [36]. This suggests low sensitivity to genetic stochasticity [36] in all but the northern edge populations of F. serratus. As in most studies, the precision of Ne decreased as Ne increased (Fig. 5) [36, 84, 85]. Local differentiation in F. serratus is one of the most important assumptions of the employed ‘temporal’ method to estimate Ne, in which neutral genetic change over time is expected to be inversely proportional to Ne. Discrete generations are another important assumption of the ‘temporal’ method. Overlapping generations are unlikely to cause a significant downward bias of Ne when more than 4 generations lie between the temporal samples [49]. This can be expected for most of our temporal samples, assuming a generation time of 1–2 years [65, 86] and a time span of 6–11 years between sampling. Thus, our sample-size-corrected estimates can be regarded as unbiased and indicative of a ‘real’ decline in Ne from the species’ mid-range of distribution to its range-edges.\n",
      "\n",
      "An Ne > 1000, as in the French ‘Ile de Siec’ population, is large enough to ensure evolutionary potential in perpetuity [87], and is likely to provide the best source for adaptive genetic rescue of threatened and declining populations [38, 88]. However, large Ne estimates are commonly associated with a high uncertainty [36, 85]. Accordingly, the point estimate of Ne in the ‘Ile de Siec’ population (ca. 10 Million) has a wide confidence interval as compared with the other populations (Fig. 5, Additional file 7). Consequently, the point estimate is unlikely to be the true value in this population, but is certainly > 1000, and higher than in any other measured populations. The reason for this outlier value is not due to high diversity, since this is comparable to the other French population (Fig. 4a-d), but the high stability in allele frequencies over time. Indeed, the ‘Ile de Siec’ population was the only population for which temporal genetic differentiation was non-significant (Additional file 7).\n",
      "\n",
      "\n",
      "Ne in the other mid-range populations, > 500, may be sufficient in the mid-term [36, 87, 89] to mitigate the predicted extinction by the end of the twenty-first century [26]. However, given that summer temperatures are predicted to rise above the thermal tolerance limit of F. serratus in Brittany within the next 200 years [26, 41], it is important to track its fitness in this region in order to implement early conservation measures in case it loses its current stability.\n",
      "\n",
      "An Ne of 50–100 was regarded necessary for a population to minimize inbreeding depression and associated problems such as accumulation of deleterious mutations and loss of variation [36, 87]. However, despite Ne < 60 in the Norwegian ‘Grense Jakobselv’ population, genetic diversity remained stable for both Norwegian populations over six years and neither population was inbred. In contrast, a previous study on a southern Norwegian population reported significant loss of Ne from 2000 to 2008 and concluded that an Ne between 73 and 386 was insufficient for long-term survival under environmental change [39]. Stable diversity despite small Ne in our two northern Norwegian populations may be ascribed to regional gene flow, suggested by a reduction in genetic differentiation between the two Norwegian populations from 2004 to 2010 and significant outbreeding (negative FIS) in the ‘Kirkenes’ population in 2010. Thus, regional gene flow may uncouple Ne from genetic stochasticity effects at the species’ Northern edge of distribution.\n",
      "\n",
      "\n",
      "Diversity comparisons\n",
      "\n",
      "As expected for neutral loci, genetic diversity was positively related with Ne [36, 38]. Both regional and local diversities are highest in Brittany and make the range-center of F. serratus less sensitive to genetic drift [36, 37]. A decline in genetic diversity towards the northern and southern range-edges is in accordance with the species’ biogeographic history [42].\n",
      "\n",
      "Low genetic diversity does not necessarily lower the evolutionary potential of F. serratus to adapt to Arctic shores [26]. The evolutionary potential, survival, and expansion rate of low-diversity leading edge populations [3] may decrease when deleterious mutations accumulate at expansion range fronts and create a so-called ‘expansion load’ [7, 9]. On the other hand, survival may well be enhanced by the ‘surfing’ of favorable mutations that can rapidly increase in frequency over iterated founder events [4, 5, 90]. An additional consideration is that source populations of Arctic colonists may not be located at the species’ northern edge, but within European harbors with frequent shipping, fishing, and cruise boat traffic to and from the northern polar regions.\n",
      "\n",
      "Our results have clear implications for the conservation of F. serratus at its southern edge. Reductions in MLH from 1999 to 2010 were close to significant (p = 0.0051/0.134 for Ribadeo2/Ribadeo1, respectively), although, Hexp, α, and FIS remained temporally stable. This agrees with stability of Hexp and α over 7–9 years in fragmented southern edge populations of the kelp species Laminaria digitata [91], and is likely due to the measures’ insensitivity to the effects of population bottlenecks [92]. In other words, while the polymorphic state of loci and the diversity of alleles did not decline, alleles occurred more frequently in a homozygous state in the recent samples. In theory, the decline in MLH might be explained by increased selection pressure for heat-tolerance, although there is only indirect experimental evidence for this. Acclimation potential to further thermal stress is likely impeded in this population by chronically high expression of heat shock protein genes [22, 23, 41, 93]. Between 2000 and 2010, the Ribadeo1 population experienced a 90% decline in abundance [26]. Although stable local differentiation favors ecotypic differentiation in thermal stress tolerance [41], heat-stress is becoming too extreme at the southern edge.\n",
      "\n",
      "The value of conserving the southern edge of F. serratus may be high [94]. Because of its separation from Brittany by the uninhabitable sandy warm shores of the Bay of Biscay, the Northwest Iberian glacial refugium did not contribute to postglacial recolonizations of ice-free northern shores, and, thus, preserves unique genetic variation [42]. The conservation value of the species’ southern edge becomes even more apparent when considering that small-scale population structure increases the species’ regional diversity above local diversity within single populations (Additional file 8). High regional diversity, despite low within-population diversity, was previously reported for the southern distribution edge of the seagrass Zostera marina [94, 95]. We are not aware of studies that explicitly addressed this effect in macroalgae, although increased local differentiation at the southern edge of the kelp Laminaria digitata [91] can be expected to increase regional variation as well. Thus, with the loss of its southern edge, the species’ can be expected to lose its most heat-adapted populations sustaining unique genetic variation.\n",
      "\n",
      "\n",
      "Conclusions\n",
      "\n",
      "Temporal snapshots of genetic diversity and structure in F. serratus populations spanning its latitudinal range reliably reflect patterns across local and regional spatial scales and across various thermal backgrounds for at least one decade. Stable small-scale structure enhances regional genetic diversity throughout the species’ range of distribution and is a potential driver for local adaptation [36] that may explain species radiation and diversity in the genus Fucus [96–98].\n",
      "\n",
      "\n",
      "MLH appears to be the most stress-sensitive measure of diversity, displaying a strong decline at the species’ southern edge of distribution. As sandy warm shores separate the Iberian southern edge from the genetically diverse Brittany region, genetic rescue by the influx of genetic variation [38, 88] might only be possible if initiated by conservation efforts.\n",
      "\n",
      "Increasing temperatures are likely the major cause for the decline not only of F. serratus, but also other intertidal and subtidal macroalgae in Northwest Iberia [28, 99–101], as well as temperate seaweeds worldwide [102]. Kelp species may maintain genetic diversity to a certain degree in southern edges by escaping to deep-water refugia to avoid rising temperatures in shallow waters [103]. Accordingly, in Northern Portugal, increasing air-temperature stress depresses the upper boundary limit of F. serratus [104]. However, intertidal seaweeds are less adapted to low light conditions and, thus, have low potential to escape into deeper waters. Another factor that impedes survival of southern edge populations in fucoid seaweeds is their reproductive strategy with fewer gametes and lower dispersal (< 12 m from parental sites [45, 46]) as compared with kelps that release billions of spores dispersing several kilometers [105, 106]. We suspect that without the influx of genetic variation from Brittany, intertidal habitat-forming macroalgae, such as F. serratus, may largely disappear from southern edges but retain potential to persist in small subtidal bottleneck populations in cool upwelling regions [107].\n",
      "\n",
      "\n",
      "Additional files\n",
      "\n",
      "\n",
      "Additional file 1:\n",
      "\n",
      "Surface air temperatures. Daily average surface air temperatures (SAT) at each of the seven sampling sites from 1999 to 2011 with gaps in year 2006 for the French and Spanish sampling sites. SATs were identical between the two Norwegian sampling sites as well as between the two French and the two Spanish sampling sites. (PDF 513 kb)Additional file 2:\n",
      "\n",
      "Microsatellite markers. Characteristics of each microsatellite marker, including cycling conditions and multiplexing. (XLSX 8 kb)Additional file 3:\n",
      "\n",
      "PCR cycling protocols. Time-release (a) and no-time-release (b) PCR cycling protocols. In the time-release protocol, the heat-activated DNA-polymerase was progressively released during the thermal cycling process. Annealing temperatures and number of cycles indicated with an X are specified for each marker in Additional file 2. (PDF 34 kb)Additional file 4:\n",
      "\n",
      "Microsatellite raw data. Microsatellite genotypes in Structure format. The first row contains the names of all 31 markers. The following rows contain the individual genotype data. Each individual is represented in 2 consecutive rows. The first column contains the name of the individual, the second row contains the population number that individual belong to. The following 31 columns show the alleles of each marker as microsatellites base pair lengths. The population numbers (1–14) refer to the following sampling locations and times: 1) Gjerild Klint, present-day; 2) Gjerild Klint, historical; 3) Green Top, present-day; 4) Green Top, historical; 5) Ile de Siec, present-day; 6) Ile de Siec, historical; 7) Grense Jakobselv, present-day; 8) Grense Jakobselv, historical; 9) Kirkenes, present-day; 10) Kirkenes, historical; 11) Ribadeo 1, present-day; 12) Ribadeo 1, historical; 13) Ribadeo 2, present-day; 14) Ribadeo 2, historical. (TXT 195 kb)Additional file 5\n",
      "\n",
      "Corrected microsatellite data. Microsatellite data corrected for allelic dropout in Structure format. The first row contains the names of all 31 markers. The following rows contain the individual genotype data. Each individual is represented in 2 consecutive rows. The first column contains the name of the individual, the second row contains the population number that individual belong to. The following 31 columns show the alleles of each marker as microsatellites base pair lengths. The population numbers (1–14) refer to the same sampling locations and times as in Additional file 4. (TXT 198 kb)Additional file 6:\n",
      "\n",
      "Spatial differentiation. Regional and local genetic differentiation between sampling sites in historical and present samples estimated by FST and Dest with p values. (XLSX 10 kb)Additional file 7:\n",
      "\n",
      "Temporal changes. Estimates of effective population size (Ne), temporal genetic differentiation (FST and Dest), and p values for temporal changes in diversity measures at each sampling site. (XLSX 6 kb)Additional file 8:\n",
      "\n",
      "Diversity estimates. Diversity estimates, including heterozygosity (Hexp), allelic richness (α), and multi-locus heterozygosity (MLH), for each location and region with standard errors, and inbreeding coefficients FIS with p values for each population. (XLSX 8 kb)Additional file 9:\n",
      "\n",
      "Temporal versus local genetic differentiation. Temporal genetic change in comparison to local and regional genetic differentiation(FST and Dest) for each population and sampling time point (historical and present). (XLSX 6 kb)\n"
     ]
    }
   ],
   "source": [
    "view_one_article({ \"$match\": { \"text_matches\": { \"$in\": problematic_terms } } })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "phrase = \"\"\"\n",
    "\"Following optic nerve injury, few axons grow beyond the lesion, but we find these axons branch and form loops proximal to the lesion\"\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "cursor = articles.find({'$text': {'$search': phrase}})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "5575138\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "3D Visualization of Individual Regenerating Retinal Ganglion Cell Axons Reveals Surprisingly Complex Growth Paths\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "{'has_body': True, 'article_type': 'research-article'}\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['field area']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Significance Statement\n",
      "\n",
      "Retinal ganglion cells (RGCs) are viewed as being incapable of mounting lengthy axon regeneration. Using whole tissue immunolabeling, we establish a technique to visualize and trace the entire paths of small populations of genetically labeled RGC axons as they regenerate. Following optic nerve injury, few axons grow beyond the lesion, but we find these axons branch and form loops proximal to the lesion. A regeneration inducing treatment further exacerbates branching and tortuous growth, while only modestly increasing the number of RGC axons that successfully grow beyond the lesion. Our study demonstrates extensive and circuitous RGC axon elongation both in pre- and post-lesion regions, highlighting the need to better understand the factors that inhibit direct axon growth in the optic nerve.\n",
      "\n",
      "\n",
      "Introduction\n",
      "\n",
      "Lack of axon regeneration is a major obstacle preventing functional recovery after axon injury. Like other neurons in the CNS, it is thought that retinal ganglion cells (RGCs) have a limited ability to regenerate spontaneously. Nonetheless, growth factors or modification of genes promote RGC axonal regeneration, to some extent. For example, supplying RGCs with cytokines, or genetic modification of Pten, Pcaf, Stat3, Socs3, c-Myc, dcxl, or Klf4 allows some RGC axons to regenerate with few axons reaching the brain targets (Park et al., 2008; Moore et al., 2009; Smith et al., 2009; Puttagunta et al., 2014; Belin et al., 2015; Nawabi et al., 2015; Leibinger et al., 2016; Mehta et al., 2016).\n",
      "\n",
      "Multiple reports have demonstrated that some regenerating RGC axons travel circuitously within the optic nerve after intraorbital crush injury (Luo et al., 2013; Pernet et al., 2013). However, these studies have used anterograde tracers that label all RGC axons, making it difficult to identify individual fibers and visualize how axons of different RGC types behave as they regenerate. Here we sought to combine sparse neuronal labeling with the immunolabeling-enabled 3D imaging of solvent-cleared organs (iDISCO) technique (Renier et al., 2016) and trace the entire path of individual axons as they regenerate.\n",
      "\n",
      "RGCs are a heterogeneous population of neurons. They are divided into several subclasses based on their morphologic, physiologic, and molecular properties. Previous studies have shown that different RGC types differ in their ability to regenerate axons. For instance, studies in cats have shown that αRGCs regenerate axons better than other RGC types (Watanabe and Fukuda, 2002). Similarly, in mice it was demonstrated that αRGCs have greater propensity to regenerate axons after crush injury (Duan et al., 2015). RGCs in the Thy1-H-YFP mouse line have a “Golgi-like” labeling, allowing for the characterization of their dendrites and axons. Importantly, this line labels few RGCs, most of which are immune-positive for SMI-32, a marker of αRGCs (Feng et al., 2000; Lin et al., 2004; Coombs et al., 2006).\n",
      "\n",
      "In this study, we first validate that the Thy1-H-YFP mouse sparsely labels αRGCs. We used a combination of confocal imaging and iDISCO to analyze the dendrites and axons of the labeled neurons. We find that, after optic nerve crush, (1) αRGC dendrites decrease in complexity, and the dendritic arbors are even less complex in animals treated with an axon growth promoting-stimulator (i.e., CNTF); (2) innately, axons remodel and regrow extensively proximal to the crush site; and (3) following CNTF, axon elongation is extremely circuitous, and these axons never reach the brain, despite growing distances that would allow them to reach the brain, had they grown straight. Overall, our results show that RGCs can re-extend axons very well, even in the absence of a growth stimulator, but their inability to traverse the lesion area along with meandering axon growth limit “meaningful” regeneration, dramatically reducing the number of RGCs successfully regenerating axons into the brain. Our results shift the focus from promoting axon elongation, to understanding factors that prevent direct growth of CNS axons through the injured nerve.\n",
      "\n",
      "\n",
      "Materials and Methods\n",
      "\n",
      "All experimental procedures were performed in compliance with protocols approved by the Institutional Animal Care and Use Committee at the University of Miami. The Thy1-H-YFP mouse strain was used for all experiments (The Jackson Laboratory stock number 003782). All animals were housed in a viral antigen-free facility and kept under standard 12/12 h light/dark cycle conditions. For all surgical procedures, mice where anaesthetized with ketamine and xylazine. For analgesia, buprenorphine (0.05 mg/kg) was administered postoperatively. The exact number of animals used for each group is in the main text and figure legends.\n",
      "\n",
      "\n",
      "Intravitreal injection\n",
      "\n",
      "Female and male mice six to eight weeks old underwent unilateral intravitreal adeno-associated virus serotype 2 (AAV) injection. AAVs carried expression constructs for ciliary neurotrophic factor (CNTF; AAV-CNTF; Yungher et al., 2015) or placental alkaline phosphatase (PLAP) as a control transgene (AAV-PLAP). AAversus were made by the University of Miami's Viral Vector Core. Typical titers were 5 × 1012 GC/ml. A fine glass micropipette was inserted into the posterior chamber taking care to avoid damaging the lens. Using a Hamilton syringe (Hamilton 80900) 2 µl of virus was slowly injected. Cholera toxin β (CTB) conjugated to Alexa Fluor 594 (ThermoFisher C34777, 2 µg/µl in PBS) was injected as described above. CTB was injected 1 h postcrush in the 3 d postcrush (3dpc) group.\n",
      "\n",
      "\n",
      "Optic nerve crush\n",
      "\n",
      "Animals received unilateral optic nerve crush. Time points postcrush included in this study: 3 dpc, 3 weeks postcrush (3wpc), and 6wpc. AAV was injected 7 d before crush in the 3dpc group, or 3 d before crush in the 3wpc and 6wpc groups. For the optic nerve crush procedure, the optic nerve was exposed intra-orbitally by blunt dissection. The optic nerve was crushed with forceps (#5 Dumont, Fine Science Tools) for 10 s ∼1 mm distal to the emergence from the globe.\n",
      "\n",
      "\n",
      "Immunohistochemistry\n",
      "\n",
      "Mice were killed by transcardial perfusion with ice cold PBS and 4% paraformaldehyde. The optic nerve was cut proximal to the optic chiasm. The globe with attached optic nerve was postfixed in 4% paraformaldehyde overnight at 4°C. For retinal whole mount staining, the retina was carefully dissected out of the globe and derestricting cuts were placed in each quadrant. Tissue was washed with PBS and blocked in 5% normal donkey serum (Sigma D9663) in PBS + 0.3% Triton X-100 (PBST). Tissue was then incubated in blocking buffer containing primary antibodies, overnight at 4°C. Primary antibodies: goat-anti-osteopontin (OSPN) 1:500 (R&D Systems, AF808), rabbit-anti-melanopsin (OPN4; UF006) 1:2500 (Advanced Targeting Systems, AB-N38), goat-anti-GFP 1:1000 (Abcam, ab6673), rabbit-anti-GFP 1:1000 (Millipore, ab3080), and rabbit-anti-cocaine- and amphetamine-regulated transcript (CART; 55–102) 1:1000 (Phoenix Pharmaceuticals, H-003-62). Following incubation with primary antibody, tissue was washed extensively in PBST. Tissue was then incubated in blocking buffer containing secondary antibodies 1:500, overnight at 4°C. Secondary antibodies: donkey-anti-rabbit Alexa Fluor 488 (ThermoFisher, A-21206), donkey-anti-rabbit Alexa Fluor 647 (ThermoFisher, A-31573), donkey-anti-goat Alexa Fluor 488 (ThermoFisher, A-11055), and donkey-anti-goat Alexa Fluor 647 (ThermoFisher, A-21447). Following secondary incubation retinas were extensive washed with PBST, mounted with Slowfade (ThermoFisher, S36973) and coverslipped. Imaging was performed on an Olympus FV1000 confocal microscope, objectives: UPlanSApo 10 × 0.40 N.A. and UPlanFLN 40 × 1.3 N.A., and Olympus FV10-ASW Ver 0.200C software. Images were analyzed using Imaris software (Bitplane). Figures were composed using Photoshop CS6 (Adobe) and Illustrator CS5 (Adobe).\n",
      "\n",
      "\n",
      "RGC quantification\n",
      "\n",
      "The number of YFP+ RGCs per retina was quantified from low-magnification z-stack images that tiled the entire retina using an Olympus Fluoview 1000 confocal microscope. RGCs were defined by: soma location in the ganglion cell layer or inner nuclear layer (displaced RGC), and the presence of an axon. To determine the subtype of YFP+ RGCs, every YFP+ RGC in a retina was assessed for immunoreactivity with a RGC subtype marker: OSPN, Opn4, or CART (see Results for n number of retinas assessed for each marker). Survival of YFP+ RGCs was determined as the number of YFP+ RGCs remaining or the percentage of YFP+ RGCs remaining compared to the uninjured contralateral retina. To analyze RGC dendrites, high-magnification z-stack images were taken of individual RGCs from the retinal nerve fiber layer to the inner nuclear layer. Images were tiled and reconstructed if a RGC’s dendrites extended out of field. Dendrites were traced using ImageJ/FIJI and the Simple Neurite Tracer plugin (Longair et al., 2011; Schindelin et al., 2012). Linear Sholl analysis was completed using the Sholl analysis plugin for ImageJ/FIJI (Ferreira et al., 2014). Sholl analysis data were analyzed as the average number of intersections observed in radii bins of 30 µm. Data were preprocessed using R 3.3.1 (https://cran.r-project.org/).\n",
      "\n",
      "\n",
      "iDISCO\n",
      "\n",
      "For whole mount staining and clearing we use the enhanced version of iDISCO. For a full description of the protocol see, Renier et al. (2014), Renier et al. (2016), and website (http://lab.rockefeller.edu/tessier-lavigne/assets/file/whole-mount-staining-bench-protocol-january-2015.pdf). Dissected optic nerves were dehydrated with a methanol/PBS series, 20%, 40%, 60%, 80%, and 100%, bleached overnight with 5% H2O2 in 100% methanol at 4°C. Rehydrated with a methanol series in PBS and 0.2% Triton X-100, 80%, 60%, 40%, 20%, and 0%. Incubated with 1× PBS, 0.2% Triton X-100, 20% DMSO, and 0.3 M glycine, 37°C for 2 d. Block in 1× PBS, 0.2% Triton X-100, 10% DMSO, and 6% donkey serum, 37°C, for 2 d. Wash in 1× PBS, 0.2% Tween 20 with 10 μg/ml heparin (PTwH), room temperature for 1 h, twice. Incubate with a chicken IgY recognizing GFP epitopes (Aves, #GFP-1020, 1:200) in PTwH, 5% DMSO, and 3% donkey serum, 37°C, for 2 d. Wash in PTwH for 10 min, 15 min, 30 min, and 1 h then 2 h or longer to the next day. Incubate with a goat-anti-chicken Alexa Fluor 488 1:300 (ThermoFisher, A-11039) in PTwH and 3% donkey serum, 37°, 2 d. Wash in PTwH for 10, 15, 30, and 60 min each and then 2 h or longer for 2 d. After the final wash, the samples were cleared.\n",
      "\n",
      "\n",
      "Clearing\n",
      "\n",
      "Washed samples were incubated at room temperature with shaking. First, for 1 h for each step with 20%, 40%, 60%, and 80% methanol in water followed by 30 min in 100% methanol twice. Next, they were incubated for 3 h in 66% dichloromethane (DCM) and 33% methanol then 20 min in 100% DCM, twice. Final clearing solution was in dibenzylether (DBE) with no shaking. Cleared optic nerves were mounted onto a cover glass with DBE and imaged on an Olympus confocal microscope (Fluoview 1000) using a 20× UPlanSApo objective (N.A. = 0.75). We used an optical zoom of 1.4× and each optic section was 1–1.2 μm. Individual stacks of images were stitched using the program XuvStitch 1.8.099x64 (http://www.xuvtools.org/doku.php).\n",
      "\n",
      "\n",
      "Axon analysis\n",
      "\n",
      "Reconstructed confocal images were analyzed using the FilamentTracer function in Imaris 8.4.1 (Bitplane). To be included in quantitative analysis an axon had to be traced from the proximal optic nerve head to its termination. An axon also had to be resolvable from surrounding axons. FilamentTracer statistics were exported and preprocessed using R 3.3.1 (https://cran.r-project.org/).\n",
      "\n",
      "\n",
      "Statistical analysis\n",
      "\n",
      "Data preprocessing was conducted using R 3.3.1 (https://cran.r-project.org/). Statistical analysis and graph creation was performed with Prism 6 (GraphPad Software). See Table 1 for the list of statistical tests used.\n",
      "\n",
      "Table 1.\n",
      "Data structure\tType of test\tObserved power (α = 0.05)\n",
      "Results text\tDependent continuous\tPearson correlation\t0.0479\n",
      "Figure 2\t3 groups, normal distribution\tANOVA\n",
      "Uninjured vs 6wpc\tNormal distribution\tPost hoc\t0.7877\n",
      "Uninjured vs CNTF + 6wpc\tNormal distribution\tPost hoc\t0.0032\n",
      "6wpc vs CNTF + 6wpc\tNormal distribution\tPost hoc\t0.0214\n",
      "Figure 2\t3 groups, normal distribution\tANOVA\t0.5523\n",
      "Figure 2\t3 groups, repeated measures, normal distribution\tTwo-way ANOVA\n",
      "Uninjured vs 6wpc\tNormal distribution\tPost hoc\t0.0188\n",
      "Uninjured vs CNTF + 6wpc\tNormal distribution\tPost hoc\t<0.0001\n",
      "6wpc vs CNTF + 6wpc\tNormal distribution\tPost hoc\t0.0353\n",
      "Figure 2\n",
      "Distance: 30 μm\t3 groups, normal distribution\tANOVA\n",
      "Uninjured vs 6wpc\tNormal distribution\tPost hoc\t0.0055\n",
      "Uninjured vs CNTF + 6wpc\tNormal distribution\tPost hoc\t<0.0001\n",
      "6wpc vs CNTF + 6wpc\tNormal distribution\tPost hoc\t0.0024\n",
      "Distance: 60 μm\t3 groups, normal distribution\tANOVA\n",
      "Uninjured vs 6wpc\tNormal distribution\tPost hoc\t0.0001\n",
      "Uninjured vs CNTF + 6wpc\tNormal distribution\tPost hoc\t<0.0001\n",
      "6wpc vs CNTF + 6wpc\tNormal distribution\tPost hoc\t0.0019\n",
      "Distance: 90 μm\t3 groups, normal distribution\tANOVA\n",
      "Uninjured vs 6wpc\tNormal distribution\tPost hoc\t0.0002\n",
      "Uninjured vs CNTF + 6wpc\tNormal distribution\tPost hoc\t<0.0001\n",
      "6wpc vs CNTF + 6wpc\tNormal distribution\tPost hoc\t0.0455\n",
      "Distance: 120 μm\t3 groups, normal distribution\tANOVA\n",
      "Uninjured vs 6wpc\tNormal distribution\tPost hoc\t0.1008\n",
      "Uninjured vs CNTF + 6wpc\tNormal distribution\tPost hoc\t0.0014\n",
      "6wpc vs CNTF + 6wpc\tNormal distribution\tPost hoc\t0.2399\n",
      "Distance: 150 μm\t3 groups, normal distribution\tANOVA\n",
      "Uninjured vs 6wpc\tNormal distribution\tPost hoc\t0.6984\n",
      "Uninjured vs CNTF + 6wpc\tNormal distribution\tPost hoc\t0.1204\n",
      "6wpc vs CNTF + 6wpc\tNormal distribution\tPost hoc\t0.4675\n",
      "Figure 3\tDependent continuous\tPearson correlation\t<0.0001\n",
      "Results text\t2 groups, 2 outcomes\tFisher's exact text\t0.0135\n",
      "Figure 5\tNon-normal distribution\tMann Whitney test\t0.0001\n",
      "Figure 5\tNormal distribution, unequal variance\tWelch's test\t<0.0001\n",
      "Figure 5\tNormal distribution, unequal variance\tWelch's test\t0.1152\n",
      "Figure 5\tNormal distribution, unequal variance\tWelch's test\t<0.0001\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "\n",
      "Thy1-H mouse sparsely labels subpopulation of αRGCs\n",
      "\n",
      "To visualize and track the growth of individual axons in the unsectioned mouse optic nerve we used the Thy1-H-YFP transgenic mouse line, which sparsely labels RGCs, including the regeneration competent αRGCs (Feng et al., 2000; Coombs et al., 2006; Duan et al., 2015). We observed that there are ∼70 YFP+ RGCs in each flat-mounted retina of an adult Thy1-H-YFP mouse (n = 9 retina). While the number of YFP+ RGCs per retina varies between animals (σ = 21), there is a strong correlation (R2 = 0.9943, p < 0.05, n = 3 pairs of retina) between the left and right retina of an animal. To determine what portion of these YFP+ RGCs are αRGCs, we immunostained the retinas with an antibody against OSPN, a molecular marker of αRGCs (Duan et al., 2015; Sanes and Masland, 2015). We found that ∼70% of YFP+ RGCs are immunoreactive for OSPN (Fig. 1). To further define the molecular identity of YFP+ RGCs, we stained the retinas with antibodies against OPN4 (a marker of intrinsically photosensitive RGCs) and CART (a marker of direction selective RGCs). Very few YFP+ RGCs were immunoreactive for OPN4 (i.e., <2% of total YFP+ RGCs) or CART (i.e., <10% of total YFP+ RGCs; Fig. 1A–G). Taken together, these results demonstrate that the Thy1-H-YFP mouse line sparsely labels RGCs, which are primarily OSPN+.\n",
      "\n",
      "Figure 1. \n",
      "Identification of RGC subtypes labeled by the Thy1-H-YFP mouse line. Retinal whole mount preparations were immunostained for YFP (green) and markers of three RGC subtypes (magenta): OSPN (A, D, D’), OPN4 (B, E, E’), and CART (C, F, F’). A–C, Low-magnification images of flat mount retina specimens that demonstrate the number of RGCs labeled by the Thy1-H-YFP mouse line as well as by each RGC subtype marker. Scale bar, 500 µm. D–F, High-magnification images of RGCs labeled by each marker. D’–F’, YFP+ RGCs (green) shown in the presence of different RGC subtypes (magenta). Arrowheads mark YFP+ RGCs immunoreactive for subtype marker. Scale bar, 50 µm. G, The percentage of YFP+ RGCs that are immunoreactive for each marker. Bar graphs of mean ± SEM (retinas: OSPN n = 9, OPN4 n = 3, and CART n = 3).\n",
      "\n",
      "To examine the fate of YFP+ RGCs in response to axon injury, we performed intraorbital optic nerve crush and evaluated their survival in the presence or absence of a growth promoting factor (Fig. 2A–H,J). Six weeks postcrush, ∼13 YFP+ RGCs survived (Fig. 2J, 6wpc). Of these, ∼10 were OSPN+ RGCs, which represents ∼76% of total remaining YFP+ RGCs (Fig. 2J). Several studies have shown that virally transferred CNTF promotes RGC survival and axon regeneration. AAV2 expressing a secreted form of CNTF did not alter the survival of YFP+ and YFP+/OSPN+ RGCs (Fig. 2J, CNTF + 6wpc). These data show that the majority of surviving YFP+ RGCs in the Thy1-H-YFP mouse are αRGCs. These findings underscore the utility of the Thy1-H-YFP mouse line for studying axon regeneration in a small number of regeneration competent RGCs, and establishes the feasibility of examining how a specific RGC type will behave after injury and CNTF.\n",
      "\n",
      "Figure 2. Response of YFP+ RGCs to axonal injury and AAV-CNTF injection. Six weeks following optic nerve crush (6wpc), retinal whole mounts from AAV-PLAP (6wpc; A, C–E)- and AAV-CNTF (CNTF + 6wpc)-injected (B, F–H) animals were immunostained for YFP (green) and OSPN (magenta). A, B, Low-magnification images of retinal whole mount preparations show few YFP+ RGCs surviving six weeks following crush injury. Scale bar, 500 µm. RGCs were defined by: soma location in the ganglion cell layer or inner nuclear layer (displaced RGC), and the presence of an axon. C–H, High-magnification images of YFP+/OSPN+ RGCs from 6wpc (C–E) and CNTF + 6wpc (F–H) animals. Three example RGCs are shown for each animal group. Scale bar, 50 µm. I, Example traces of YFP+/OSPN+ RGC dendrites from each condition. Scale bar, 50 µm. J, Quantification of YFP+ RGCs and YFP+/OSPN+ RGCs in each condition, each dot represents 1 retina (6wpc n = 4, CNTF + 6wpc n = 5). K, Number of primary dendrites observed for each RGC (p < 0.05 6wpc vs CNTF + 6wpc, ANOVA with Tukey’s post hoc). L, Dendritic field size for each RGC (mm2). J–L, Bars, median and interquartile range. M, Sholl analysis of RGC dendrites; bars, mean ± SEM (*p < 0.05 uninjured versus 6wpc, §p < 0.05 uninjured versus CNTF + 6wpc, #p < 0.05 6wpc versus CNTF + 6wpc, ANOVA with Tukey’s post hoc at each distance). K–M, Uninjured n = 16 RGCs from five animals, 6wpc n = 15 RGCs from four animals, CNTF + 6wpc n = 15 RGCs from five animals).\n",
      "\n",
      "\n",
      "Changes in αRGC dendrite morphology following axon injury\n",
      "\n",
      "In addition to investigating the survival of YFP+ RGCs, we examined the changes in dendrite morphology following intraorbitral optic nerve crush. Again, we focused on YFP+ αRGCs (i.e., YFP+/OSPN+ RGCs). Representative images of YFP+/OSPN+ RGCs and their dendrites in uninjured and injured (6wpc) animals are shown in Figures 1D’–F’, 2C–E,I, respectively. At six weeks following injury, we did not observe a significant change in dendritic field area or the number of primary dendrites compared to uninjured animals (Fig. 2K,L). Sholl analysis, which is commonly used to evaluate dendritic field arrangement and density (Sholl, 1953), shows that there is a significant reduction in dendrite complexity following intraorbital crush (Fig. 2M, uninjured versus 6wpc, *p < 0.05). AAV-CNTF and crush injury (CNTF + 6wpc) resulted in RGCs that have fewer primary dendrites and less complex dendritic arbors than injury alone (Fig. 2K–M, #p < 0.05 6wpc vs CNTF + 6wpc; statistically significant difference detected at 30- to 90-µm radii but no other radii). Thus, these results indicate that acute optic nerve injury leads to reduction in αRGCs’ dendritic complexity, and CNTF plus injury causes an even further reduction of dendritic complexity in these neurons.\n",
      "\n",
      "\n",
      "Immunohistochemical staining of unsectioned whole optic nerve using iDISCO\n",
      "\n",
      "To visualize and follow single axons throughout the optic nerve, we subjected adult Thy-H-YFP mouse optic nerves to a tissue clearing procedure which renders tissues transparent and facilitates whole tissue 3D imaging. Since tissue clearing procedures generally reduce endogenous YFP signal, we also immunostained the whole nerves using an antibody against YFP before tissue clearing using the iDISCO technique Representative images of uninjured optic nerve subjected to iDISCO and whole tissue imaging are shown in Figure 3A–E. Sparsely labeled individual YFP+ RGC axons are clearly visible. These axons project linearly from the optic disk to the distal optic nerve. In rare occasions, we also noticed that some axons have a short, rapidly terminating branch (Fig. 3F,G). We also observed occasional YFP+ cells and their processes in the optic nerve. These are likely astrocytes based on their morphology (Figs. 4D, 5A; AC, yellow). We have established the use of iDISCO to visualize the entire course of Thy1-H-YFP RGC axons through the optic nerve.\n",
      "\n",
      "Figure 3. The iDISCO technique was used to immunostain unsectioned optic nerves from Thy1-H-YFP mice. A, Maximum intensity projection (MIP) image of a full thickness optic nerve showing YFP+ axons (white). Orientation: optic nerve head on left, distal toward the optic chiasm to the right. B, Traces superimposed on MIP image. Scale bar, 500 µm. C, Example traces of single YFP+ axons, each color represents one continuous axon. Color assignment was arbitrary. D, E, High-magnification view of boxed area in A. Scale bar, 30 µm. F, Example of an uninjured axon branch. G, Trace overlay of F. Scale bar, 10 µm; arrowhead, branch point. H, Scatter plot of the number of RGCs counted in the retina (abscissa) versus the number of axons counted in each optic nerve (ordinate) per animal, n = 5; dashed line, linear fit.\n",
      "\n",
      "Figure 4. iDISCO based 3D analysis of Thy1H-YFP axons following optic nerve crush. A, Diagram of the eye and optic nerve, indicating the location of intravitreal injection and the site of optic nerve crush. Two RGCs shown in green. Approximate distance of crush site to optic disk, and to optic chiasm are shown. Observed distance in vivo may vary ± 0.2 mm. B–D, Maximum intensity projection (MIP) images of full thickness optic nerves showing YFP+ axons (white). Traces for YFP+ axons are shown with MIP image. Orientation, optic nerve head on left, distal toward the optic chiasm to the right. B, Intravitreal AAV-CNTF-injected optic nerve 3dpc (CNTF + 3dpc), n = 4. C, Optic nerve 3dpc (i.e., no AAV-CNTF), n = 4. B, C, One hour postcrush mice received intravitreal injection of CTB-Alexa Fluor 594 (red). Arrows in B, C indicate a disconnected stump of distal degenerating axon. D, Intravitreal AAV-CNTF-injected optic nerve 3wpc (CNTF + 3wpc), n = 4. AC (yellow) marks an example of a presumed astrocyte. B–D, Lesion site indicated by red *. Each color represents an individual axon. Color assignment was arbitrary. Scale bar, 100 µm. Single axon traces are presented for select axons that displayed growth.\n",
      "\n",
      "Figure 5. iDISCO based 3D analysis of YFP+ axons six weeks following optic nerve crush (6wpc). A, Maximum intensity projection (MIP) images of full thickness optic nerves showing YFP+ axons (white). Lesion site indicated by red *. Orientation, optic nerve head on left, distal toward the optic chiasm to the right. Four optic nerves were members of this group, 6wpc 1–4. Traces for YFP+ axons are shown with MIP image. Each color represents an individual axon. Color assignment was arbitrary. Scale bar, 100 µm. AC (yellow) in 6wpc 2 and 6wpc 3 indicate examples of presumed astrocytes. White # in 6wpc 3: extraocular muscle that was not completely removed during dissection. B, Three axons from 6wpc 1 and 6wpc 3 are shown. Each axon grows but fails to cross lesion site. Branch points marked by arrowheads. Lesion site indicated by red *.\n",
      "\n",
      "Next, we used iDISCO to evaluate how individual axons within a defined population of RGCs regenerate. To determine the validity of our method, we compared the number of RGCs to the number of axons in the optic nerve for five animals. Consistent with the number of YFP+ RGCs in the retina, there were ∼70 axons in each nerve. Furthermore, we find a strong correlation between the number of RGCs and axons (R2 = 0.9975, p < 0.0001, n = 5 retina, optic nerve pairs; Fig. 3H), and on average we identified 98.5% of the axons predicted by the number of YFP+ RGCs. These data indicate that we can identify all YFP+ axons within the optic nerve.\n",
      "\n",
      "\n",
      "Analysis of single YFP+ RGC axons following optic nerve injury\n",
      "\n",
      "To examine the morphology and growth pattern of YFP+ RGC axons following crush injury, first we collected Thy1-H-YFP mouse optic nerves 3dpc and performed iDISCO and 3D confocal imaging (Fig. 4B,C). To determine how an individual axon regenerates, we traced entire axons of some RGCs (Fig. 4B,C). Additionally, to visualize all RGC axons, CTB conjugated to Alexa Fluor 594 was injected 1 h after crush to label axons anterogradely. Even at this early stage (i.e., 3dpc; Fig. 4C, n = 4), some axons regrew within a small area near the cut end. However, there were no axons beyond the lesion site. Axons in the animals subjected to AAV-CNTF and crush injury (CNTF + 3dpc, n = 4) appear similar at this time point (Fig. 4B). Disconnected YFP+ axons which have not yet undergone Wallerian degeneration are visible distal to the lesion site (Fig. 4B,C, arrows indicate disconnected bulbs). All YFP+ axons are clearly disconnected, and no CTB labeled axons are found far distal to the lesion site (i.e., 1 mm distal to lesion site), strongly indicating that the injury is complete and no axons are spared from axotomy. These results show that some YFP+ axons begin growing soon after injury (3dpc). Notably, this initial growth appears independent to the presence of CNTF.\n",
      "\n",
      "Second, we observed the growth pattern of YFP+ axons at an intermediate time point. At 3wpc, YFP+ axons in AAV-CNTF-treated animals (CNTF + 3wpc, n = 4 animals) regrew within regions proximal to the lesion site, forming complex branched and looped structures. Some axons also grew past the lesion site (Fig. 4D).\n",
      "\n",
      "To further investigate this complex growth pattern, we examined axon regeneration at six weeks after injury. Optic nerves from four individual animals (6wpc 1–4) are shown in Figure 5. At six weeks postinjury (6wpc), nearly all surviving YFP+ RGC axons were limited to the region proximal to crush site, consistent with the limited ability of CNS axons to regenerate beyond the injury site. We note that the proximal edge of the lesion site occurs at ∼1 mm (±0.2 mm) away from the optic disk. In one animal, we observed one YPF+ axon past the lesion site, elongating to ∼1 mm away from the lesion site (Fig. 5A, 6wpc 4). However, we did not observe regenerating YFP+ RGC axons far beyond the lesion site (i.e., 3 mm from the lesion), indicating that the crush injury was complete and did not leave axons spared.\n",
      "\n",
      "Figure 6. iDISCO based 3D analysis of YFP+ axons six weeks following AAV-CNTF injection and optic nerve crush (CNTF + 6wpc). A, Maximum intensity projection (MIP) images of full thickness optic nerves showing YFP+ axons (white). Lesion site indicated by red *. Orientation, optic nerve head on left, distal toward the optic chiasm to the right. Four optic nerves were members of this group, CNTF + 6wpc 1–4. Only animals CNTF + 6wpc 1–3 had axons that could be fully traced and were included in the quantification. Traces for YFP+ axons are shown with MIP image. Each color represents an individual axon. Color assignment was arbitrary. Scale bars, 100 µm. B, Dot plot of the number of branches occurring per axon in each group, each dot represents one axon. C, Dot plot of axon total length for each axon. D, Dot plot of the maximum distance each axon was found from the retina/optic nerve head boundary. Dashed horizontal reference line indicating lesion site at 1000 µm. E, Dot plot of aberrant growth for each axon. Bars, median and interquartile range. Uninjured n = 34 axons from three animals, 6wpc n = 42 axons from four animals, CNTF + 6wpc n = 22 axons from three animals (CNTF + 6wpc 1–3).\n",
      "\n",
      "Finally, we examined the response of YFP+ RGC axons six weeks following optic nerve crush when the animals received AAV-CNTF injection. Previous studies have demonstrated that CNTF allows some RGC axons to regenerate long distances in the optic nerve with some axons reaching the brain (Pernet et al., 2013; Yungher et al., 2015). We injected AAV-CNTF 3 d before unilateral crush, and we collected Thy1-H-YFP mouse optic nerves at six weeks after injury. Injured optic nerves from four individual animals are shown in Figures 6A, 7. Three of the four animals (CNTF + 6wpc 1–3) had axons that could be traced in their entirety. The fourth had extensive aberrant growth that prevented the tracing of each axon with certainty (Fig. 7B, CNTF + 6wpc 4); however, many axon segments could be traced effectively.\n",
      "\n",
      "Figure 7. iDISCO based 3D visualization of aberrant axon growth. A, Axon traces from animal CNTF + 6wpc 3 are shown from three perspectives. CNTF + 6wpc 3 and CNTF + 6wpc 3’ show longitudinal views of the same optic nerve. CNTF + 6wpc 3’’ and CNTF + 6wpc 4’’’ show coronal views of the optic nerve. B, Animal CNTF + 6wpc 4. Continuous axon segments that could be traced with a high degree of certainty are shown. Each color represents a continuous segment. Lesion site indicated by red *. Arrowheads mark the three segments that grew beyond the lesion site. Scale bars, 100 µm.\n",
      "\n",
      "Tracing each axon in its entirety allows comprehensive axon pattern analysis. We observed that axon branching occurs frequently following crush. Approximately half of the axons have 1 or more branches (Fig. 6B, 6wpc). AAV-CNTF significantly increased the number of axons with branches (p < 0.05) and the number of branches per axon (p = 0.0001; Fig. 6B, 6wpc vs CNTF + 6wpc). Next, we sought to characterize for each axon (1) total axon length, (2) how far centrally an axon reached (maximum distance achieved from the optic disk), and (3) how much aberrant growth occurred (aberrant growth = axon total length – maximum distance achieved; Fig. 6C–E). Given that the proximal edge of the crush site is ∼1 mm from the optic disk, we see that most axons in the injury control animals remain within or near the lesion site (i.e., within 0.8–1.2 mm from the optic disk). Surprisingly, even in the absence of growth promoting factors many of these axons grew (Figs. 5B, 6C\n",
      "). This growth was mostly within the region proximal to the crush. Some axons grew between 1 and 2 mm in length. This growth consists of branches and loops, growth that does not extend the axon distally along the optic nerve (Figs. 5B, 6E). Together, these results show that YFP+ RGCs are innately capable of regrowing long axons; however, these axons are unable to successfully traverse the lesion.\n",
      "\n",
      "Consistent with the regeneration promoting effects of CNTF, axons grew significantly longer than controls (Fig. 6C, p < 0.001). Like the injured control animals, many axons failed to grow through the lesion site, and most regrowth was aberrant (Fig. 6E). In CNTF animals, the axon length measurement shows that despite growing >4–5 mm, some axons are restricted to the lesion area. Independent of treatment, the maximum distance along the optic nerve most growing axons achieve is associated with the lesion site, around 1000 µm from the optic disk (Fig. 6D, distance of lesion site from optic disk indicated as horizontal reference line at 1000 µm). In fact, the median maximum distance achieved by axons in CNTF animals was only 25% longer than that of the control injured animals despite the fact that the CNTF axons grew 3- to 4-fold longer in total length (Fig. 6C–E). Figure 7A shows reconstructed axons of the CNTF + 6wpc 3 optic nerve from different viewing angles. This pattern of regrowth was also evident in the fourth animal (Fig. 7B, CNTF + 6wpc 4). While we were unable to trace each axon from the optic nerve head to termination, we determined that no more than three axons grew beyond the lesion site (Fig. 7B, arrowheads). As shown in the cross view of the optic nerve (Fig. 7, images in the right panels), some axons wrap around the nerve several rounds, further illustrating the extensiveness of tortuous growth.\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "RGCs of different subtypes are connected to distinct presynaptic partners and exhibit an array of responses to visual stimuli. RGCs of different subtypes also project their axons to different brain targets and contribute to image-forming functions as well as non-imaging forming functions (Sanes and Masland, 2015) Anatomically, how do the dendrites and axons of specific RGC types respond to axotomy, as well as after treatments with factors that stimulate axon growth? In the retina, several studies have characterized RGC type-specific changes in dendrite morphology after traumatic axotomy or in glaucoma models. As such, changes in dendrites after insult have been described to some extent. However, the abilities of specific RGC types to regenerate axons and correctly find their targets in adult mammals are just beginning to be determined. Since some studies have shown that many regenerating RGC axons grow circuitously near the lesion and fail to regenerate far, we sought to combine sparse neuronal labeling with iDISCO and follow individual axons derived primarily from one RGC type; αRGC. The major observations in this study are (1) αRGC dendrites decrease their complexity following axotomy, and this response to axotomy is exacerbated by CNTF treatment; (2) YPF+ RGC axons grow over unexpectedly long distances before the lesion site with only a few axons being able to successfully traverse the lesion; and (3) YFP+ axons that do regenerate beyond the lesion site elongate aberrantly, form many collateral axons in the optic nerve and fail to reach the brain.\n",
      "\n",
      "\n",
      "Considerations for cell types of origin of the YFP+ processes\n",
      "\n",
      "Our results show that the majority of surviving YFP+ RGCs after injury are immunoreactive for OSPN, indicating that most of these YFP+ RGCs belong to αRGCs (Duan et al., 2015; Sanes and Masland, 2015). Together with the observation that αRGCs have higher capacity to regenerate axons compared to other RGCs in general (Watanabe and Fukuda, 2002; Duan et al., 2015), we reason that the majority of YFP+ RGC axons that we analyzed are likely to be of αRGCs. However, because 24% of surviving RGCs are OSPN-negative (i.e., of the 13 surviving YFP+ RGCs per retina, 10 are YFP+/OSPN+ and 3 are YFP+/OSPN−), we are unable to conclusively determine if the axons that grew beyond the lesion site are of αRGC origin.\n",
      "\n",
      "\n",
      "Changes in αRGC dendrites in responses to axotomy and CNTF\n",
      "\n",
      "Using transgenic mice that label defined RGC types, previous studies have examined the morphologic changes that occur in RGC dendrites. In a mouse glaucoma model, OFF transient RGCs showed decreased dendritic arborization (Della Santina et al., 2013). Similarly, the dendritic complexity of transient OFF αRGCs from glaucomatous mouse eyes is reduced (El-Danaf and Huberman, 2015). Thus, the results in our study showing loss of dendritic complexity in αRGCs are in line with these previous reports. In contrast, in adult rats, peripheral nerve graft and AAV-CNTF cause an increase in soma size of RGCs without affecting the dendritic complexity or field size (Rodger et al., 2012). However, when examined specifically in the RGC 1 subtype (i.e., RGCs with a large soma), AAV-CNTF caused a significant reduction in the complexity of the dendritic arbors, again without reducing the field size (Rodger et al., 2012). In line with the previous study, our results show that AAV-CNTF leads to a significant decrease in arbor complexity without altering the dendritic field size in most cells. Thus, our results suggest that while αRGC axons are highly regenerative and quite likely generate complex arbors (i.e., in response to CNTF), their dendrites become less complex. In this regard, one could ask why and how does CNTF cause even greater reduction in dendrite complexity in these RGCs? The molecular and cellular mechanisms underlying such different behaviors by the axons and dendrites are unknown. It is also unclear whether the reduction in dendrite arbor complexity will be potentially disruptive for the function of these RGCs (i.e., receive less input from their presynaptic partners). These are interesting questions that may deserve further investigation.\n",
      "\n",
      "We observed that the majority of YFP-labeled αRGCs (∼80%) in this experimental example died at six weeks after injury. Currently, it is unclear whether these YFP+ αRGCs die long after injury because of lack of intrinsic survival signals or because they are disconnected for a long time and lack trophic support from the target. It is also possible that they die because they remain unmyelinated (i.e., not remyelinated) for a long period time, lacking the survival signal(s) and other type(s) of support from the oligodendrocytes (or even astrocytes). The question why most RGCs die remains nebulous and justifies further investigation.\n",
      "\n",
      "\n",
      "Highly regenerative yet unable to go far\n",
      "\n",
      "Several prior studies have adopted tissue clearing strategies and examined axon regeneration in unsectioned, whole CNS tissues (Ertürk et al., 2011; Laskowski and Bradke, 2013; Luo et al., 2013; Pernet et al., 2013; Soderblom et al., 2015). In this study, we sought to expand the 3D analysis to track single axons in given neuronal types. To our knowledge, our study is the first to report the adaptation of sparse labeling and whole tissue staining to attain the entire projection profiles of regenerating axons. Perhaps the most striking observation in our study was the extensive and circuitous regeneration of RGC axons occurring proximal to the lesion site. Axons start to penetrate through the lesion, sometime more than once, but each time they turn back toward the retina and thus they elongate within the proximal optic nerve region (i.e., near the optic nerve head). Additionally, axons that surpass the lesion continue to branch and misroute, with these events lacking an obvious spatial pattern (e.g., occurring near the lesion site). We also note that none of these axons grew past the optic chiasm. As can be seen in Figures 4–7, regenerating axons stop at different distances away from the chiasm. Previous studies have suggested that optic chiasm may inhibit or halt some axons to grow further (Luo et al., 2013; Crair and Mason, 2016). At least for these neurons however, the reason that they fail to grow past the chiasm does not seem to be due to chiasmatic barrier as these axons terminate or turn toward the eye even before they get close to the optic chiasm.\n",
      "\n",
      "How does an axon successfully traverse the lesion? In the CNS lesion, various growth inhibitory molecules are present. Molecular barriers within the lesion area include chondroitin sulfate proteoglycans and myelin-associated inhibitors (Yiu and He, 2006). Astrocytes and fibroblasts interact to establish a scar, surrounding the lesion. Therefore, to successfully traverse the lesion, a growing axon will need to modify the extracellular matrix (ECM) and overcome inhibitory/repulsive cues. Differential expression of ECM modifying enzymes and cell surface receptors may explain why some axons never cross the lesion while others do. This simple explanation is challenged by our finding that axons proximal to the lesion form extensive branches and loops and then eventually grow through the lesion (i.e., Fig. 6A, CNTF + 6wpc 1). For this to occur, axons would need to be dynamically responsive to their environment and alter gene expression accordingly (i.e., after failing to traverse the lesion, they change expression of certain molecules to better suit the lesion environment), or crossing the lesion site is a stochastic event. Future investigation will be needed to identify if specific RGCs are can cross the lesion barrier, and what interventions can help axons through this environment.\n",
      "\n",
      "The tracing of individual axons allowed us to measure the total length of individual axons as well as the maximum distances from the eye. Unequivocally, we find that while many axons travel long distances, their circuitous growth results in them remaining within close vicinity to the lesion site. The observation that the regenerating axons grow aberrantly within the optic nerve is mostly in agreement with previous studies (Luo et al., 2013; Pernet et al., 2013). Nonetheless, our current study provides a complete picture of individual axon trajectory, from their entry into the optic nerve to the end of their route. This is particularly the case for the portion of axons located in the proximal area to the injury site which was not possible in the previous studies that have used CTB as a tracer for all RGC axons. Overall, our results build on prior studies and further show that the growth of YFP+ RGCs axons are tortuous in their paths, and that these axons are unlikely to reinnervate their brain targets after CNTF treatment.\n",
      "\n",
      "\n",
      "Question of treatment and cell type-specific axon behaviors\n",
      "\n",
      "CNTF is only one of many factors that can induce axon regeneration. Do different regeneration stimulating factors cause a similar degree of aberrant growth and misrouting? Do the axons of all RGC types have similar propensity to misroute? Studies by Lim et al. (2016), and de Lima et al. (2012), showed that at least under certain conditions, some RGC axons are able to regenerate with minimal misrouting and back to their correct targets. In some instances, these RGC axons seem to travel linearly toward the brain (Lim et al., 2016). It appears that under some conditions, RGC axons may be able to navigate through the injured optic nerve. It is unknown what molecular and cellular factors minimize misrouting and produce directed growth within the optic nerve and beyond. It may be interesting to apply 3D analysis and examine the behavior of αRGCs and other RGC types under additional growth-promoting conditions. This could help determine if the aberrant growth seen in the present study is specific to some RGC types or to CNTF treatment per se.\n",
      "\n",
      "It is known that certain types of neurons in different CNS regions including the supraspinal serotonergic neurons are able to spontaneously remodel and reextend axons after axotomy (Hawthorne et al., 2011). On the other hand, other neurons including the corticospinal tract axons are strongly refractory to regeneration where the cut dystrophic axons regress with virtually no signs of regrowth (Thallmair et al., 1998; Liu et al., 2010). In the case of RGCs, we know from studies using CTB tracing that a few of these neurons can spontaneously regrow axons, at least to some short distance into the lesion. The lengthy axons seen even without CNTF in our study indicate that RGCs (and perhaps other CNS neurons) may have much higher growth capacity than generally thought. In the case of peripheral neurons, the dorsal root ganglion (DRG) axons send collaterals and misroute following injury to the central branch (Kerschensteiner et al., 2005). It would be interesting to determine if this aberrant growth of DRG axons is amplified following preconditioning injury, a method to promote regeneration of a DRG’s central branch.\n",
      "\n",
      "These findings highlight the need for prudence when evaluating a growth factor to promote axon regeneration. CNTF and other growth factors are frequently used in regeneration studies to stimulate axon growth. In our study, CNTF produced abundant axon growth, but this growth was highly aberrant. Thus, this growth factor caused “too much of a good thing.” Therefore, future treatments must be adjusted to maximize axon growth into brain targets, and minimize axon branching and tortuous growth.\n",
      "\n",
      "Overall, our study documents the morphologic changes that occur in αRGCs after optic nerve injury. Tracking the entire paths of individual axons reveal that these RGCs can naturally re-extend axons extremely well, but both their inability to traverse the lesion area and their circuitous axon growth limit reconnection with the brain. Our results counter the general view that RGC axons are incapable of lengthy regeneration, and shift the focus from promoting axon elongation, to understanding factors that prevent direct growth of axons through the injured nerve.\n"
     ]
    }
   ],
   "source": [
    "print_article(cursor.next())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "5063180\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "Linking quality of care and training costs: cost‐effectiveness in health professions education\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "{'has_body': True, 'article_type': 'research-article'}\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "None\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Introduction\n",
      "\n",
      "Health professions education involves training and certifying care‐provider groups in specific procedures. However, there are considerable associated costs that have been estimated globally to amount to more than 80 billion Euros per year.1 Because some training is usually more effective than no training2 but often associated with considerable monetary and time costs, identifying the most cost‐effective strategy can be challenging.3 Nonetheless, many institutions have to balance the need for training new health care providers in performing certain types of procedures against the costs associated with training.4 Cost‐effectiveness analyses are suitable for these types of decisions; however, only a few studies have attempted to link training costs to quality of care.5, 6 Experimental research in medical education provides evidence of effectiveness of different training interventions or programmes.7, 8 However, quite often there are substantial differences in the costs of different training programmes. For example, simulation‐based training may improve learners' anatomical knowledge9 but so does reading a textbook at only a fraction of the cost. Administrators and educators therefore have to question not only if an educational intervention works but also whether the associated change in the outcome of interest is significant enough to justify a difference in training costs.10 Consequently, the ability to inform and guide educators and administrators on future practice is limited unless the associated costs of training are reported and compared in experimental trials. If not, educators and administrators may become reluctant to invest in effective training programmes because the return on investment is not readily apparent.11 Even worse, costly training programmes may be adopted over cheaper but equally effective alternatives as a consequence of lacking cost‐effectiveness data. The discrepancy between how education scientists evaluate training programmes based on their effectiveness and the way administrators determine which programmes to adopt based on cost may eventually lead to a gap between education evidence and actual practice.11, 12\n",
      "\n",
      "Previous studies have attempted to establish the cost‐effectiveness of simulation programmes13, 14 and of different simulator fidelity levels.3 Despite these efforts, there is still no consensus on how to assess cost‐effectiveness in medical education.15 This may in part be attributed to difficulties linking training interventions and patient outcomes.16 Accurate estimation of the total effects of training may also prove challenging due to decay in skills over time, study participant attrition, organisational changes, and residual training effects that remain after the trial period has ended. Moreover, the effects of training are not always immediately apparent but may instead result in better preparation for future learning.17, 18 Likewise, cost estimates are prone to differences between countries, institutions and populations, thus potentially limiting the generalisability of cost‐effectiveness studies in medical education.19\n",
      "\n",
      "The underlying principle of cost‐effectiveness analysis is to compare the difference in costs between training programmes with the difference in effectiveness between the same two alternatives.20 Administrators and educators may only be prepared to pay a certain amount for each unit of the outcome variable. This is also known as willingness to pay (WTP) and is the maximum amount that an administrator is prepared to pay to achieve or avoid a certain outcome. A new programme, intervention or training concept may be considered cost‐effective if the cost of the outcome of interest is less than what administrators are willing to pay.19 Uncertainty is represented in each of the steps described above and probabilistic models have therefore been developed for this purpose in health economics theory21, 22, 23 Using these models, the likelihood that a given intervention is cost‐effective can be calculated and reworded into concrete recommendations to administrators. However, the majority of existing economic evaluations in medical education research3 have failed to include probabilistic analyses, thereby weakening their recommendations.\n",
      "\n",
      "As indicated above, there are multiple threats to the internal and external validity of cost‐effectiveness research in health professions education. This underlines the need for models that can guide educators, administrators and decision makers in choosing and prioritising training strategies. Based on current health economics theory19, 20, 21, 22, 23 and existing cost‐effectiveness studies in medical education,3, 4, 5, 6, 10, 11, 13, 14, 15 we propose a general model for conducting cost‐effectiveness studies in health professions education, the Programme Effectiveness and Cost Generalization (PRECOG) model. The PRECOG model includes four steps that represent increasing levels of recommendation strength, from assessment of programme effectiveness and cost to generalisation of cost‐effectiveness evidence. Each step of the PRECOG model is described in further detail in the following sections. In addition, we provide an example of a cost‐effectiveness trial of simulation‐based training in an obstetric emergency unit.\n",
      "\n",
      "\n",
      "Methods\n",
      "\n",
      "The four steps of the PRECOG model of collecting cost‐effectiveness evidence include: (i) gathering data on training outcomes, (ii) assessing total costs, (iii) calculating incremental cost‐effectiveness ratios, and (iv) estimating cost‐effectiveness probability (Fig. 1). Each step of the model was developed through a synthesis of current health economics theory19, 20, 21, 22, 23 and existing cost‐effectiveness studies in medical education.3, 4, 5, 6, 10, 11, 13, 14, 15, 24\n",
      "\n",
      "Figure 1 The Programme Effectiveness and Cost Generalization (PRECOG) model for cost‐effectiveness studies in medical education\n",
      "\n",
      "\n",
      "Step 1. Gathering data on training outcomes\n",
      "\n",
      "The first step includes gathering relevant data on training outcomes of interest. The purpose of this step is to estimate the effect of different training programmes and it follows standard research methods in health professions education. However, special consideration must be given to selection of the primary outcome of interest because it is placed in the denominator of a cost‐effectiveness ratio and any recommendation will be based on potential changes in this outcome. A particular challenge to gathering data on training outcomes is that effects are often measured immediately upon completion of training but may persist long after the study period has ended.25 The true effects of training are therefore underestimated if residual effects are omitted when estimating total training effects.\n",
      "\n",
      "\n",
      "Step 2. Assessing total costs\n",
      "\n",
      "The viewpoint for the cost analysis is rarely reported but is nonetheless important as the cost of training varies depending on who bears it (e.g. patients, hospitals, the national health service, trainees, etc.).19, 26 Existing health professions education literature has reported direct costs in terms of implementation cost,27 equipment cost,3, 13 personnel cost11, 28 and operating costs (e.g. operating theatre costs).28, 29, 30 Unrelated costs that may arise secondary to training are usually not reported. Such costs may involve consequences of ineffective training programmes for future practice (e.g. diagnostic errors or surgical complications). At some point, however, an artificial boundary must be drawn around the cost analyses and the role of unrelated costs is therefore debated in the health economics literature.19\n",
      "\n",
      "\n",
      "Step 3. Determining the incremental cost‐effectiveness ratio (ICER)\n",
      "\n",
      "When the effects and costs of different interventions have been established, the costs of one additional outcome of interest unit can be calculated as the incremental cost‐effectiveness ratio (ICER). This ratio can be determined as the ratio of the difference in costs between training programmes to the difference in effectiveness between the same two alternatives.20 An ICER (ICER = ΔC/ΔE) is calculated as the difference in training costs between different training programmes (ΔC) divided by the difference in their effectiveness (ΔE). The resulting ratio is, however, a relative term with no indication of uncertainty and the generalisability of raw cost‐effectiveness ratios may therefore be limited.10\n",
      "\n",
      "\n",
      "Step 4. Estimating cost‐effectiveness probability\n",
      "\n",
      "A new programme, intervention or training concept may be considered cost‐effective if the cost of one additional unit of the outcome of interest is less than the administrator's willingness to pay (WTP) value.21 Willingness to pay is the maximum amount that an administrator is prepared to pay to achieve or avoid a certain outcome. Uncertainty is represented in each of the steps described above and probabilistic models have therefore been developed for this purpose in health economics theory.22, 23 Using an estimate of this uncertainty, the likelihood that a given intervention is cost‐effective can be calculated for different WTP values, which may then be used to provide concrete recommendations to administrators and educators. Uncertainty can be estimated using net benefit regression (NBR) models that take different WTP values, effects and costs into consideration. The basic regression formula for the NBR model can be written as: Net Benefit = WTP × E – C, where E and C are the observed effects and costs, respectively. Based on the variance in distribution of effects and costs, a cost‐effectiveness acceptability curve (CEAC) can then be generated from the NBR model to provide a graphical illustration of the probability that a training strategy is cost‐effective at different WTP values.19, 22, 23\n",
      "\n",
      "\n",
      "Example of Step 1: gathering data on training outcomes\n",
      "\n",
      "\n",
      "Purpose\n",
      "\n",
      "The purpose of the example study was to determine the cost‐effectiveness of training midwives to perform cervical scans compared with obstetrician‐performed cervical scans with regard to time from patient arrival to admission or discharge.\n",
      "\n",
      "\n",
      "Design, participants and randomisation\n",
      "\n",
      "A randomised study was performed to explore the effects of training a group of midwives to assess cervical length with regard to time from arrival to discharge or admission for women with symptoms of preterm onset of labour. The study was approved by the Danish Data Protection Agency (Protocol No. 2007‐58‐0015), registered at clinicaltrials.gov (Identifier NCT02001467), and was exempt from ethical approval by the Regional Ethical Committee for the Capital Region (Protocol No. H‐4‐2013‐FSP). Participants included 12 certified midwives who volunteered for the study and provided informed consent. The eligibility criteria were employment at the Obstetric Emergency Unit, Nordsjaelland's University Hospital, Hillerød, and no prior transvaginal ultrasound experience. Participants were randomised to ultrasound training or no training in a 1 : 1 ratio.\n",
      "\n",
      "\n",
      "Intervention and control\n",
      "\n",
      "Intervention‐group participants completed two types of simulation‐based ultrasound training. They were trained using a virtual reality (VR) simulator (Scantrainer™, Medaphor, Cardiff, UK) and a physical manikin (BluePhantom™, Sarasota, USA) until predefined performance levels were attained. The validity of scores obtained on the simulator tests and the performance levels used had been collected in previous studies and included content evidence, internal structure, relations to other variables and consequences of testing.31, 32, 33, 34 A simulator instructor provided feedback during all simulation‐based training. After completing the simulator training, the participants were trained and assessed by a foetal medicine consultant (LNN). Training was completed when three consecutive scans were rated above a predefined performance level.33 Control‐group participants did not receive any ultrasound training.\n",
      "\n",
      "\n",
      "Outcomes\n",
      "\n",
      "Once training was completed, intervention‐group participants independently performed CLMs on women who presented with symptoms of preterm onset of labour. If participants had a long cervical length (above 25 mm), they could be discharged without consulting an obstetrician. An obstetrician was consulted for further diagnostics or follow‐up if the CLM was shortened (25 mm or less). The control‐group participants provided standard patient care and consulted an obstetrician if a CLM was needed. Over the next 6 months, all study participants recorded each woman's time of arrival and discharge or admission (waiting time; primary outcome), as well as who performed the CLM (midwife, obstetrician, or midwife followed by obstetrician), to determine the number of changes in the responsible health care provider (gaps in continuity of care; secondary outcome). The primary outcome was chosen because managing women with preterm onset of labour is time‐sensitive with regard to initiation of treatment and because we expected considerable psychological stress to be associated with uncertainty for patients during the waiting time.\n",
      "\n",
      "The total effects of training were not limited to the first 6 months because the participants continued to perform CLMs after the study period was completed. To account for this residual effect, the estimated number of patients treated by the intervention‐group participants was determined over a 60‐month period via the average number of patients managed per participant per month and the observed participant attrition rate per month. The total number of patients was multiplied by the average reduced waiting time to determine the total number of minutes saved. Differences between groups in mean waiting time and changes in the responsible health care provider were calculated using Student's t‐test and the chi‐squared test, respectively.\n",
      "\n",
      "\n",
      "Example of Step 2: assessing total costs\n",
      "\n",
      "The viewpoint for the cost‐effectiveness analysis was that the hospital carried the costs of training. To estimate the implementation costs, participants' time expenditure was assessed. The total number of simulator instructor and clinician hours per participant was estimated and time costs were converted into monetary costs by multiplying the number of hours used with the costs per hour for the midwives, simulator instructor and clinician teacher. The total costs were estimated by adding implementation costs and equipment costs (i.e. simulator costs). Equipment costs were depreciated over 5 years and the costs per month were calculated.\n",
      "\n",
      "\n",
      "Example of Step 3: determining the incremental cost‐effectiveness ratio (ICER)\n",
      "\n",
      "An ICER was calculated for the primary and secondary outcomes. The costs of the training programme minus those of the control group (no cost) were divided by the difference in their outcomes (mean difference in waiting times between groups and difference in frequency of gaps in continuity of care).\n",
      "\n",
      "\n",
      "Example of Step 4: estimating cost‐effectiveness probability for different WTP values\n",
      "\n",
      "A template available online for Excel 2011 was used to generate the NBR estimates.35 To determine the distribution of NBR estimates for different WTP values, Monte Carlo simulations were performed using the Cholesky decomposition technique based on observed means and standard deviations for costs and effects for the intervention and control groups. A CEAC was generated based on the Monte Carlo simulations and the NBR model.35 One‐sided confidence intervals were used to assess which strategy was effective for different WTP values.\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "Intervention‐group participants needed a mean 268.2 (95% confidence interval [CI], 140.2‒392.2) minutes of simulator training and a mean 7.3 (95% CI, 4.4‒10.3) supervised scans to attain proficiency. Table 1 lists implementation, equipment and total training costs. Patients who were scanned by the intervention group had significantly shorter waiting times (n = 50, mean 80.7 [95% CI 67.4‒93.9] minutes) compared with those managed by the control group (n = 65; mean 117.2 [95% CI 90.5‒144.0] minutes; mean difference between groups, 36.6 [95% CI 7.3‒65.8] minutes; p = 0.008). Gaps in continuity of care were reduced by 84% for patients managed by the intervention group as compared with the control group (16% versus 100%, p < 0.001). The estimated total number of women managed by the intervention group over a 60‐month period was 164, corresponding to a total of 5990 saved minutes (99 hours 50 minutes). The ICER for time saved was EUR 0.45 minute−1 and the ICER for gaps in continuity of care was EUR 19.51/shift. Figure 2 shows the CEAC. For WTP values less than EUR 0.26 minute−1, there was a 95% probability that obstetrician‐performed CLM was the most cost‐effective strategy. For WTP values above EUR 0.73 minute−1, there was a 95% probability that midwife‐performed CLM was the most cost‐effective strategy.\n",
      "\n",
      "Table 1\n",
      "Implementation costs\n",
      "Midwives\n",
      "Training time for all midwives (\t34 hours 41 minutes \n",
      "Cost per hour\tEUR 17.4\n",
      "Total cost of all midwives (\tEUR 603.2\n",
      "Simulator instructor\n",
      "Total number of hours of instruction\t26 hours 37 minutes\n",
      "Cost per hour\tEUR 29.1\n",
      "Total cost of instructor\tEUR 774.6\n",
      "Clinician teacher\n",
      "Number of hours of supervision\t7 hours 20 minutes\n",
      "Cost per hour\tEUR 38.2\n",
      "Total cost of clinician teacher\tEUR 280.1\n",
      "Total implementation costs\tEUR 1657.9\n",
      "Equipment costs\n",
      "\n",
      "Virtual reality simulator (Scantrainer, Medaphor)\tEUR 56 979.0/EUR 949.7\n",
      "\n",
      "Physical manikin (BluePhantom, CAE)\tEUR 4841.8/EUR 80.7\n",
      "Equipment costs during study period (1 month)\tEUR 1030.4\n",
      "Total costs (implementation cost + equipment costs during study period)\tEUR 2688.3\n",
      "\n",
      "Figure 2 A cost‐effectiveness acceptability curve (CAEC) showing the probability that the training strategies were cost‐effective for different willingness‐to‐pay values. The dashed line denotes the 95% confidence limit that the training strategy is cost‐effective.\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "We provided a methodological basis for conducting future cost‐effectiveness studies in order to meet the calls for economic evaluations in health professions education.5, 6 We demonstrated a 4‐step model to determine cost effectiveness of training interventions (the PRECOG model). The model included gathering training outcomes, assessing costs, calculating cost‐effectiveness ratios, and estimating the probability that the training strategies were cost‐effective for different willingness to pay values.\n",
      "\n",
      "A substantial challenge for medical education cost‐effectiveness studies is the lack of generalisability of results due to differences in training costs, study context and curriculum design across institutions and countries.3, 19 Systematic reviews and meta‐analyses may overcome some of these limitations, but experimental research in medical education often fail to report costs of training interventions, thereby making economic evaluations across studies difficult. Furthermore, comparisons of effects may be complicated by different evaluation methods, outcome measures6 and follow‐up durations. We proposed a model where the total effects of training were estimated by extrapolating from immediate effects while adjusting for participant attrition rate and skills decay, when applicable. If only immediate effects are taken into account, the costs per unit of the outcome variable may be overestimated. To account for decision uncertainty, we used a probabilistic method for estimating cost‐effectiveness. The importance of conducting this final step was highlighted by both training strategies being recommended as the most cost‐effective in the example provided, depending on administrators' willingness to pay. If administrators were willing to pay more than EUR 0.73 per minute of reduced waiting time, there was a high probability that training a group of midwives in performing CLMs would be cost‐effective. On the other hand, if administrators were willing to pay less than EUR 0.26 per minute, there was a high probability that maintaining the obstetrician‐performed CLMs would be cost‐effective.\n",
      "\n",
      "The example used in this study involved training a group of midwives to perform CLMs to reduce patient waiting time before admission or discharge and to reduce gaps in continuity of care. Time is an important parameter when administrating corticosteroids to women with preterm onset of labour36 and gaps in continuity of care are associated with errors during patient handover.37 Both parameters were therefore proxy indicators of care quality. However, we did not attempt to explore their effects on patient outcomes such as neonatal morbidity or number of errors committed during patient handover.\n",
      "\n",
      "Strengths of this study include the use of patient‐level data, a randomised design, standardised protocols for training, assessment and certification, and robust statistical models. There are also some limitations. First, the small sample size and single‐centre design limit generalisability of the study results. Second, only direct effects of the training intervention were assessed and not indirect effects such as reduced and increased workload imposed on obstetricians and midwives, respectively. Third, the equipment costs may have been slightly overestimated in this study as we anticipated that the simulators could not be used for other purposes during the study period. Finally, the model does not provide a one‐size‐fits‐all approach to economic evaluations in medical education, but may be used to improve the methodological rigour of future cost‐effectiveness studies. We chose to select the current model over competing models because it enabled us to link training outcomes to costs and to provide a generalisable estimate of the probability that the intervention was cost‐effective. However, the general principles described in the present model may just as well apply to different methods for estimation of effects, cost, uncertainty and generalisation.\n",
      "\n",
      "Providing evidence of cost‐effectiveness in future experimental medical education trials may bridge the gap between best practice and actual practice.12 Administrators are thereby enabled to prioritise training interventions based on cost‐effectiveness evidence rather than effectiveness evidence alone. However, this requires cost estimates to be reported in future effectiveness studies.\n",
      "\n",
      "\n",
      "Conclusion\n",
      "\n",
      "Cost‐effectiveness models can be used to link quality of care to training costs. We demonstrated a 4‐step model to determine cost‐effectiveness of training interventions. In the example used in the present study, training midwives to perform cervical length measurements was cost‐effective for WTP values above EUR 0.73 per minute reduced waiting time. Maintaining obstetrician‐performed cervical length measurements was cost‐effective for WTP values below EUR 0.26 per minute waiting time.\n",
      "\n",
      "\n",
      "Contributors\n",
      "\n",
      "MGT, LNN and AT contributed to the conception of the study. MGT, LN, AT and MEM contributed to the study design. MGT, MEM, CW and LD contributed to the data acquisition. CR contributed to data interpretation. MGT prepared the first draft of the article. CR and AT contributed to the first draft and CR, AT, LN, MEM, CW and LD critically revised the subsequent drafts. All authors approved the final manuscript and agree to be accountable for all aspects of the work.\n",
      "\n",
      "\n",
      "Funding\n",
      "\n",
      "this study was funded by the Tryg Foundation.\n",
      "\n",
      "\n",
      "Conflicts of interest\n",
      "\n",
      "none.\n",
      "\n",
      "\n",
      "Ethical approval\n",
      "\n",
      "the study was approved by the Danish Data Protection Agency (Protocol No. 2007‐58‐0015) and was exempt from ethical approval by the Regional Ethical Committee for the Capital Region (Protocol No. H‐4‐2013‐FSP).\n"
     ]
    }
   ],
   "source": [
    "view_one_article({ \"$match\": {'$text': {'$search': phrase}}})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "3543354\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "None\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "None\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['study area']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Introduction\n",
      "\n",
      "The transfer of energy and recycling of nutrients via the decomposition of organic matter is a central unifying process that links all organisms to the functioning of ecosystems [1]. However, dead organic matter varies enormously in its spatial and temporal distribution [2], and this determines the magnitude of its contribution to nutrient cycling [3] and the diversity and dynamics of its consumers [4]. Animal carrion is the most nutrient-rich form of dead organic matter [5], and recent reviews have highlighted the overlooked role of carrion in food webs [6], [7], [8], and driving variation in biodiversity and ecological processes in landscapes [5], [9]. Widespread changes to the population dynamics of large vertebrates, through loss of top predators [10] or hunting and harvesting of wild herbivores [11], are affecting the distribution and input of carrion resources in some terrestrial ecosystems [6], [12]. This creates a strong imperative to establish a deeper understanding of the links between carrion, biodiversity and ecosystem functioning.\n",
      "\n",
      "Ephemeral resource patches, such as rotting logs and dung pads, occur in ecosystems worldwide and are focal points of nutrient cycling and aggregation of invertebrates [13], [14]. Carrion is distinct from many other ephemeral resources due to its faster decomposition rate, and its patchy spatial occurrence [9]. This means that carrion has very different spatial and temporal dynamic to other ephemeral resources, and suggests that carrion may have a disproportionate effect on biodiversity and nutrient cycling in ecosystems.\n",
      "\n",
      "The decomposition of carrion affects its surroundings through pathways to vertebrate and invertebrate scavengers and detritivores, and their predators, as well as through microbial and fungal decomposers and plants. This generates a temporally dynamic patch with high nutrient concentration and biological activity that contrasts strongly with the nearby environment [5], [15]. To date, studies on the effects of carrion on variation in biodiversity have been restricted to single groups of organisms, such as the detailed examination of successional patterns of arthropods [16], [17], and its localised effects on soil nitrogen or plant growth and composition [15], [18]. This has resulted in a detailed understanding of the specialised role that many insect groups have in carrion decomposition, including the critical role of fly larvae [19], [20], [21], and the complex food webs of associated predators and detritivores [16], [22], [23], [24]. Studies on soil and plant responses have demonstrated that the significant spike in soil nitrogen at a carcass can be detected in nearby plant tissue [15], [25], leading to changes in plant growth and composition [18], [26], and is likely moderated by soil microbial activity [5]. Key studies have also shown the importance of carrion for soil nitrogen and plant dynamics across whole ecosystems [27], [28]. Nevertheless, there are only a handful of comprehensive studies that have integrated multiple groups of organisms into the one study design [3], and there are fewer still that have specifically examined the functional traits of different species that respond to carrion. This leaves much to learn about what traits characterise carrion opportunists, and therefore what traits of species can help us understand the links between ephemeral carrion patches and its effect on variation in biodiversity within landscapes [9].\n",
      "\n",
      "We present data from an experiment designed to answer a number of questions about how ephemeral carrion resource patches generates variation in ecological communities. We did this by quantifying the responses of different aspects of the carrion food web, including components of the insect community, soil nutrients, and the plant community. We sought to identify the traits associated with insect responses to the patchiness of carrion, and the responses of plants to subsequent changes in soil nutrients. To do this, we first examined changes in assemblage diversity and composition in response to the experimental addition of animal carcasses. Second, we examined whether assemblage changes corresponded with particular combinations of traits, for both insects and plants. We predicted that insect responses would be associated with their dispersal ability, with specialist species having wing morphologies that enable the rapid location of new carrion resources. We also predicted that plant responses would be associated with their growth potential and ability to exploit elevated soil nutrient levels. Our experiment compared soil, plant and insect responses between paired carrion and control sites with spatial replication. The results of our analysis therefore provide new insight into the magnitude of the spatial contrast between carrion and its immediate surroundings. This allowed us to examine how the spatial contrast changed over time, and how carrion contributes to both spatial and temporal variation in biodiversity and ecological processes within a landscape.\n",
      "\n",
      "\n",
      "Methods\n",
      "\n",
      "\n",
      "Ethics Statement\n",
      "\n",
      "Insects were sampled under an ACT government ‘Licence to take’ (LT2010417).\n",
      "\n",
      "\n",
      "Study Area and Experimental Design\n",
      "\n",
      "In November 2010 (southern hemisphere spring), we added carcasses of kangaroos to 18 sites in temperate Eucalyptus woodland near Canberra, Australia (149°10′ East, 35°10′ South). This study area is part of the Mulligans Flat-Goorooyarroo Woodland Experiment, a long-term ecological restoration project [29]. Specific details on temperature, rainfall, floristic composition and soil types from this study area have been described by McIntyre et al [30]. We sourced carcasses of eighteen eastern grey kangaroos (Macropus giganteus) over a 10 day period from road kill across the city of Canberra, and positioned them at sites within 12 hours of death. Note that no animals were killed for the purposes of this study. Carcasses were only used when there was not significant blood loss. Average mass of the kangaroos was 31 kg (range 17–80 kg). Each carcass was paired with a control site 10 metres away to allow spatial comparisons between carcass and control treatments through time. We positioned each carcass (and control site) at the base a young Eucalyptus blakelyi tree (<2 m tall at beginning of experiment). Our study area has a history of livestock grazing, resulting in some sites being dominated by exotic grass species [30], [31], [32]. We placed half the carcasses in native-dominated grassland (Themeda sp. and Austrodanthonia sp.) and half in exotic-dominated grassland (Phalaris sp. and Holcus sp.).\n",
      "\n",
      "\n",
      "Soil and Plant Data Collection\n",
      "\n",
      "We took a 4 cm deep soil sample (approximately 4 cm diameter) using a trowel at each carcass and control site at one week (week 0) before carcass addition, and then 12 and 26 weeks after carcass addition. Samples were taken from near the centre of each carcass. Total N content of soil and leaf samples (see below) was quantified with high temperature combustion using a LECA2000 CNS analyser [33]. We performed two analyses of total N, and used the average in our calculations. The potential mineralisable soil nitrogen (organic nitrogen) was quantified using the Illinois soil nitrogen test for amino sugar [34].\n",
      "\n",
      "We took samples of between five and ten leaves from the tips of branches (i.e. active growth zones) in each E. blakelyi tree at carcass and control sites at weeks 12 and 26. No leaves were taken at week 0. Leaf nitrogen content was determined as per soil nitrogen (see above). A higher nitrogen content in leaves indicates potentially greater photosynthetic capacity, and therefore growth potential [35]. Each carcass was positioned on top of existing grasses and forbs, but these were killed off during the early stages of decay. We examined plant re-colonization of carcass patches after 52 weeks by recording the presence of grass and forb species inside a 40×40 cm quadrat at (i) the centre and (ii) edge of each carcass patch, and at (iii) the paired control site. The edge of each carcass was able to be delineated by estimation of the size of the original carcass, and the presence of a ‘halo’ of plant growth. In addition to these standardised quadrats, we also recorded the presence of all other plant species present in the remainder of a 40 cm deep perimeter around each carcass (the ‘surrounding’ vegetation) and the entire carcass patch to compile a complete inventory of grass and forb species. We collected data on plant life forms and specific leaf area (SLA) from the literature [36], [37], [38], and these are summarised in Appendix S2. SLA is the one-sided area of a fresh leaf divided by its dry mass (mm2.mg−1), and is indicative of potential relative growth rate [36]. SLA has also been shown not to vary significantly within plant species [37], thus justifying our use of data from the literature. We used the mean SLA value of each plant species presented in each published account, and excluded rare species for which data were unavailable.\n",
      "\n",
      "\n",
      "Insect Data Collection\n",
      "\n",
      "We sampled insects using a pair of pitfall traps at each carcass and control site. We chose ants and beetles because they are an abundant and diverse components of carrion arthropod communities [3], [16], [39], and yet differ strongly in their dispersal ability and resource use. We recognise that flies are an important part of carrion arthropod communities [16], but we wanted to focus on two diverse groups with contrasting dispersal ability. Many beetles are flight-capable and have high resource specialisation [40], [41], with some specialist species requiring carrion to lay eggs and complete their life cycle. In contrast, ants are generally restricted to the ground (after post-mating dispersal), and may opportunistically scavenge from carcasses or predate on the other insects present. For these reasons, we expected beetles and ants to show very different responses to carrion. Pitfall traps were 9 cm in diameter, and were half filled with glycol solution. The traps were opened for seven days during weeks 1, 6, 12 and 26 of the experiment, which represented distinct stages of decomposition of the carcasses (see Appendix S1). We pooled pitfall trap pairs, giving n = 36 at each sampling time. All insects were extracted from the pitfall traps, stored in 70% ethanol and sorted to order and family. We further sorted beetles and ants to genus and morphospecies using relevant keys [42], [43].\n",
      "\n",
      "\n",
      "Insect Morphological Measurements\n",
      "\n",
      "We measured body length and wing dimensions of each beetle species for comparison of mean trait values of species present at each carcass at control site. We used body length to estimate the body mass of each species of beetle using family-specific length versus weight power relationships for species of Scarabaeidae, Staphylinidae and Carabidae, and a general Coleoptera relationship for all remaining species [44]. We calculated wing loading [45] by dividing body mass by wing area (length x width), expressed as (mg/mm2). We calculated wing aspect ratio by dividing wing length by width, with higher ratios indicating longer and narrower wings [45]. Ant body size is correlated with microhabitat use and resource discovery rates [46], and we regarded this as a useful measure of ant dispersal ability. We measured ant head width and rear femur length, and multiplied these together to calculate a body size index for each species [47]. All morphological measurements were made on up to three individuals of each species (then averaged) using an ocular micrometer with magnification between 10× and 100×, and resolution between 0.1 and 0.01 mm, depending on the measurement and size of the insect. This approach was taken as we were interested in variation in traits between species rather than within species [48].\n",
      "\n",
      "\n",
      "Data Analysis\n",
      "\n",
      "We were first interested in quantifying the magnitude of the contrast between carcass and control sites. We used repeated measures linear mixed models in GenStat 14 [49] to test for interactive effects of carcass addition and sampling time on total insect abundance, species richness and evenness, as well as soil and plant nitrogen content. Carcass addition and time were treated as fixed factors, and site as a random factor. We used ANOVA, also in GenStat 14, to test for differences in insect and plant species richness and evenness or traits between carcass and control treatments at a particular sampling time, with blocking by carcass-control pair. Plant SLA was log10 transformed, and insect abundance data were log10(x+1) transformed to meet assumptions of normality and homogeneity of variances. Percentage similarity in plant species composition between the edge and centre of carcasses was expressed as the number of shared species divided by the total number (i.e. Jaccard similarity index). We compared similarity between the centre and edge of carcasses because plant recolonisation was more likely to occur from the edge rather than from more distant control sites. We tested for effects of quadrat position (carcass centre, edge or control site) and grassland type (native or exotic) on average SLA across plant species. For this analysis, we excluded rare species that occurred only in one quadrat. We wanted to test whether insect assemblages were more homogenous in composition at carcasses than at control sites due to the widespread dominance of carrion specialists. We used Permutational Analysis of Multivariate Dispersion [50] to test for differences in among-sample heterogeneity of insect species composition between carcass and control treatments, with statistical significance determined from 10 000 permutations of the data. This test calculates the average distance to the centroid of a group of samples projected in multivariate space, with a greater distance to centroid indicating greater among-sample heterogeneity [51]. Multivariate insect abundance data were square-root transformed prior to analysis to reduce the influence of highly abundant species on sample dissimilarity. Shannon evenness was defined as E = H′/lnS, where S is the number of species in a sample, and H′ = −∑ pi ln pi, where pi is the proportion of a species in a sample which is multiplied by the natural logarithm of itself [52].\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "\n",
      "Soil and Leaf Nitrogen\n",
      "\n",
      "The addition of carrion had significant effects on soil nitrogen (Table 1). The soils in our study environment were originally low in phosphorus and nitrogen compared with grasslands improved for grazing [31], and thus have low productivity potential. The highly concentrated nutrients in each carcass (a 30 kg vertebrate carcass contains approximately one kilogram of nitrogen [5]) produced a dynamic change in soil nitrogen. We found that total soil nitrogen beneath carcasses was 200% higher than control sites after 12 weeks, and 70% higher than control sites after 26 weeks of carrion being added (Fig. 1A). In addition, the concentration of the organic fraction of soil nitrogen was 460% higher than control sites after 12 weeks, and 290% higher after 26 weeks (Fig. 1B). This nutrient pulse also was detected in plant leaf nitrogen levels from nearby tree saplings, which were 18% higher in plants near to carcasses than control sites after 12 weeks (Fig. 1C). This demonstrates the cycling of nitrogen from carcasses into soil, and the uptake and assimilation of soil nitrogen into leaf tissue during the first few weeks of decomposition.\n",
      "\n",
      "10.1371/journal.pone.0053961.g001Figure 1 Effect of carrion on soil and foliar nitrogen.Carrion increased (a) soil total nitrogen, (b) soil organic nitrogen, and (c) tree foliar total nitrogen relative to control sites. Error bars show mean ± SE.\n",
      "\n",
      "Table 1\n",
      "Effects\tF\td.f.\tP\n",
      "Soil total nitrogen (%)\tTime\t55.85\t2\t<0.001\n",
      "Treatment\t29.36\t1\t<0.001\n",
      "Time×Treatment\t37.59\t2\t<0.001\n",
      "Soil organic nitrogen (µg/kg)\tTime\t136.31\t2\t<0.001\n",
      "Treatment\t26.68\t1\t<0.001\n",
      "Time×Treatment\t107.28\t2\t<0.001\n",
      "Tree foliar nitrogen (%)\tTime\t29.73\t1\t<0.001\n",
      "Treatment\t12.71\t1\t<0.001\n",
      "Time×Treatment\t0.26\t1\t0.609\n",
      "\n",
      "\n",
      "Plant Assemblage Richness and Composition\n",
      "\n",
      "Plant recolonisation of carcasses was faster in the exotic-dominated grassland where, after 52 weeks, the composition of plants in the centre of each carcass patch more closely matched the composition of species at the edge of each carcass patch in the exotic-dominated grassland (56% shared species) than in native-dominated grassland (37% shared species). We found a significant interaction between carcass treatment and grassland type (F2 = 11.48, P<0.001) and their effect on plant species richness at carcass patches after 52 weeks. Plant species richness was similar across carcass and control sites in the exotic grassland, but the number of species in the centre of carcass patches in native grassland was only a third that of species at the edge of carcass patches or at control sites (Fig. 2A). This result is further highlighted by our observation that four of the nine native-dominated carcass patches had no re-colonising plants at all, whereas all nine of the exotic-dominated carcass patches had re-colonising plants 52 week after carrion was added (compare Fig. 3A with 3B). The rapid response of plants in the exotic grassland was associated with a significantly higher specific leaf area among the constituent plant species (Fig. 2B, F1 = 18.03, P<0.001). Regardless of grassland type, plants recolonising the centre of carcass patches tended to be annuals rather perennials, and graminoids rather than forbs (Fig. 2C, 2D).\n",
      "\n",
      "10.1371/journal.pone.0053961.g002Figure 2 Differences in plant species richness and traits between carcass centre, carcass edge, and control sites.(a) Species richness of plants species in exotic grassland compared with native grassland. (b) Specific leaf area of plants in exotic grassland compared with native grassland. (c) Percentage of annuals versus perennials. (d) Percentage of graminoids versus forbs. Error bars show mean ± SE.\n",
      "\n",
      "10.1371/journal.pone.0053961.g003Figure 3 Recolonisation of plants 52 weeks after carrion addition.Very little recolonisation had occurred in the native-dominated grassland sites (a), whereas re-colonisation had progressed further in all exotic-dominated grassland sites (b), reflecting greater tolerance of some exotic species to extremely high nutrients. These examples indicate a different succession trajectory for native versus exotic grassland plant assemblages, with recolonisation occurring over different time scales.\n",
      "\n",
      "\n",
      "Insect Assemblage Diversity and Composition\n",
      "\n",
      "We sampled a total of 33 ant species (3330 individuals) and 119 beetle species (3614 individuals) from all carcass and control sites over the four sampling times (Appendix S3 and S4). Beetle and ant assemblages displayed contrasting dynamic responses to carrion. We found that beetle assemblages had a significantly higher species richness and lower evenness during early stages of decay, with a gradual return to levels similar to control sites by week 26 (Fig. 4A, Table 2). This contrasted strongly with ant assemblages, which initially had lower species richness at carcass sites, and did not show the same decrease in evenness as the beetle assemblage (Fig. 4B). We found an overall significant difference in among-sample heterogeneity from carcass and control sites at each sample times for both beetles (F = 21.11, P<0.001) and ants (F = 11.81, P<0.001). However, comparisons between carcass and control sites for each sampling time showed distinctive and contrasting patterns for the beetles and ants (Fig. 5). In the early weeks of decomposition, the composition of beetle assemblages at carrion sites was less variable than at control sites, indicating spatial monopolisation of carcass patches by a few abundant species (Fig. 5A). This changed as decay progressed, with the beetle assemblage becoming more heterogeneous at carrion sites relative to control sites. This was due to widespread species at carcasses becoming less dominant, while assemblages at control sites showed a seasonal decrease in heterogeneity (Fig. 5A). In contrast, there was never a difference in ant assemblage heterogeneity between carcass and control sites (Fig. 5B), with the overall significant difference among groups due a decrease in heterogeneity in week 26 in late autumn. This indicated there was no spatial monopolisation of carcasses by a few ant species, in contrast to that which we observed for beetles.\n",
      "\n",
      "10.1371/journal.pone.0053961.g004Figure 4 Effect of carrion on beetle and ant diversity.(a) Species richness and evenness of beetles at carcasses showed a trajectory through time clearly different from control sites at weeks 1, 6 and 12, with a return to similar levels at week 26 (blue arrows). (b) Species richness of ants at carcasses was half that at control sites during week 1 (red arrows), but was then similar to control sites at week 6, 12 and 26 of the experiment. Error bars show mean ± SE.\n",
      "\n",
      "10.1371/journal.pone.0053961.g005Figure 5 Effect of carrion on insect assemblage heterogeneity.Average distance to group centroid, reflecting heterogeneity in composition among samples of (a) beetles and (b) ants from carrion and control treatments. Error bars show mean ± SE.\n",
      "\n",
      "Table 2\n",
      "Effects\tF\td.f.\tP\n",
      "(A)\tAbundance\tTime\t48.81\t3\t<0.001\n",
      "Treatment\t3.25\t1\t0.080\n",
      "Time×Treatment\t4.55\t3\t0.009\n",
      "Species richness\tTime\t26.05\t3\t<0.001\n",
      "Treatment\t3.35\t1\t0.076\n",
      "Time×Treatment\t4.51\t3\t0.009\n",
      "Evenness\tTime\t44.38\t3\t<0.001\n",
      "Treatment\t1.08\t1\t0.322\n",
      "Time×Treatment\t1.41\t3\t0.264\n",
      "(B)\tAbundance\tTime\t40.09\t3\t<0.001\n",
      "Treatment\t230.92\t1\t<0.001\n",
      "Time×Treatment\t19.05\t3\t<0.001\n",
      "Species richness\tTime\t21.46\t3\t<0.001\n",
      "Treatment\t70.07\t1\t<0.001\n",
      "Time×Treatment\t11.51\t3\t<0.001\n",
      "Evenness\tTime\t50.49\t3\t<0.001\n",
      "Treatment\t28.86\t1\t<0.001\n",
      "Time×Treatment\t12.83\t3\t<0.001\n",
      "\n",
      "\n",
      "Insect Dispersal Traits\n",
      "\n",
      "The contrasting response of beetle and ant assemblages was associated with their different dispersal abilities, with significant differences in morphological traits across beetle species but not ant species (Table 3). Beetle species at carrion were larger than species at control sites one week after carrion was added (Fig. 6A), whereas ants showed no significant difference in body size at any of the four sampling times (Fig. 6B). Beetles at carcasses also had significantly higher wing loading, but only at one week after carrion was added (Fig. 6C). We also found that a higher percentage of beetles at carcasses had functional wings compared with beetles at control sites (Fig. 6D).\n",
      "\n",
      "10.1371/journal.pone.0053961.g006Figure 6 Differences in insect morphological traits between carcass and control sites.(a) Beetle species at carrion during week 1 were larger on average than those at control sites. (b) Beetle species at carrion during week 1 had higher wing loadings on average than beetle species at control sites. (c) No differences in ant body size were observed at different stages of carrion decomposition. (d) Macropterous beetles were always more prevalent at carcass sites that control sites. Error bars show mean ± SE.\n",
      "\n",
      "Table 3\n",
      "Effect\tF\td.f.\tP\n",
      "(A) Ant body size\tTime\t0.99\t3\t0.395\n",
      "Treatment\t0.09\t1\t0.769\n",
      "Time×Treatment\t0.62\t3\t0.600\n",
      "(B) Beetle body length\tTime\t12.42\t3\t<0.001\n",
      "Treatment\t39.37\t1\t<0.001\n",
      "Time×Treatment\t8.82\t3\t<0.001\n",
      "(C) Beetle wing loading\tTime\t9.25\t3\t<0.001\n",
      "Treatment\t1.81\t1\t0.179\n",
      "Time×Treatment\t1.64\t3\t0.180\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "Our results show that carrion in ecosystems represents a dynamic driver of species richness and nutrient cycling. This is supported by our observations, which showed elevated nitrogen in soil and plant leaf tissue at carrion patches, different rates of recolonization by plants, and a diverse and functionally distinct set of insects occurring at carcasses compared with control sites. Importantly, we have shown that the pattern of response by insects and plants was associated with particular functional traits that allow them to respond to and then exploit carrion. This builds on previous work that has demonstrated separately the effects of carrion on soil nutrient heterogeneity [53], scavenger community structure [16], [54], and plant composition [18], and develops further our understanding of how carrion generates spatial and temporal variation in ecological communities.\n",
      "\n",
      "We found that the nitrogen spike in soils translated into an increase in leaf nitrogen in the nearby E. blakelyi trees, indicating the movement of nutrients from the decomposing carcasses into live plant tissue via the soil subsystem [15], [25]. Trees in the genus Eucalyptus are not nitrogen-fixing and must use available nitrogen in the soil. This suggests that the increase in soil nitrogen due to carcass decomposition was actively taken up by the trees we sampled. The differences we detected in foliage nitrogen between carcass and control sites was smaller than that reported from analogous experimental studies in North America [15], [28], but this may be due to the very different plant species involved, as well as underlying microbial and soil properties. The dramatic change in soil nitrogen also resulted in a greater dominance of grass species with high average specific leaf area. It is well established that these functional plant types typically respond to elevated nitrogen [31]. This reflects observations in a tallgrass prairie ecosystem of North America, where forbs decreased in cover at the centre of ungulate carcasses after 1 year, but grasses increased in cover [18]. This is evidence of in situ organisms displaying trait-dependent responses to new nutrient-rich resource patches. The effects of carcasses on the diversity and composition of ground-layer plant assemblages depended on whether the carcasses were placed in native- or exotic-dominated grassland. Plants at the carcass treatment sites were initially killed by the carcasses due to prolonged shading by carcass remains and rapid soil chemistry changes, but later recolonised from the edges. The quick recolonization of carrion patches by plants in the exotic grassland was largely due to seeding, except for one species that used stolons (Cynodon dactylon). This underlines a context-dependent effect of carrion not previously described at local scales. This parallels a pattern observed for the effects of migrating salmon (Oncorhynchus spp.) carcasses on plants in North America, with effects reduced in the more highly productive catchments [27]. Together the results suggest a general principle of context-dependency resulting in temporally attenuated effects of carrion in high productivity ecosystems or in habitats with a high proportion of nutrient-responsive plant species.\n",
      "\n",
      "We showed that dispersal traits of insects were associated with their responses to ephemeral resource opportunities. The distinct dispersal traits of beetles compared with ants was likely to explain, in part, their contrasting dynamics in assemblage composition, and led to their changes in spatial heterogeneity during different stages of carcass decomposition. Dispersal plays a key role in insect spatial diversity patterns [55], and our study provides a further example relevant to carrion. Although carrion affected the relative abundance of ants present in the landscape, it did not affect species richness or spatial heterogeneity. The lack of flight to facilitate rapid foraging means that ants are generally less able to move as rapidly as flight-capable beetles, regardless of any resource specialisation they might have. Although we found that some ant species dominated individual carcasses, such as the meat ant Iridomyrmex purpureus, there were no species that monopolised all the carcasses in the same way some beetle species did. This result is due, in part, to the lack of strong specialisation of ants on carrion resources, with scavenging more likely to be opportunistic. For beetles, capacity for flight and higher wing-aspect ratios were associated with beetles at carrion generally. A larger body size and higher wing loading were important functional traits that were associated with the early colonisation of fresh carrion and spatial monopolisation of this resource. Beetles visiting fresh carcasses were typically equipped with wings that enable manoeuvrable flight [45], suggesting that they can actively seek carrion, and with their large body size have the metabolic capacity to do so over greater distances than smaller beetles. These traits are exemplified by the carrion specialists Ptomaphila lacrysoma and Creophilus erythrocephalus, which are well known to frequent carrion in south-eastern Australia [56], [57]. These species, as well as several others, were not recorded at control sites (Appendix S4), and have not been recorded as part of previous research in the study area [48]. This highlights the unique composition of carrion insect assemblages. Ants at carrion displayed no characteristic morphologies during different stages of carrion decay, and this matched their lack of difference in spatial heterogeneity between carcasses or control sites. Although we focused on morphological traits related to dispersal ability, it is important to point out that the feeding traits of insects are equally important in determining their responses to carrion [16]. For example, the carrion specialists mentioned above also feed on the flesh of carcasses or prey on the insects that do. Their early arrival at a carcass is therefore critical if they are to exploit a resource that will quickly be exhausted. Further research is needed to test if dispersal limitation or small body size actually prevents some species from exploiting carrion, as this would provide a mechanistic link between morphology and assemblage responses to carrion resources.\n",
      "\n",
      "Together, our findings provide an integrated example of how traits of opportunist species in two different taxonomic groups are associated with their ability to exploit patchy and dynamic resources. This adds to the literature on carrion resource specialisation traits and temporal assemblage dynamics [16], [23], to give a fuller picture of the processes that drive spatial changes in species composition. It also supports previous claims that carrion might need to be managed in some ecosystems to maintenance of biodiversity and ecological processes [9], [58]. This includes, for example, the role of carrion in the persistence of dependent insects [59] or the nutrient opportunities it can provide to plants [15]. Our study indicates that appropriate consideration of spatial and temporal continuity in carrion will enable insects of variable dispersal ability to exploit this resource, and will promote spatial heterogeneity in nutrient cycling and plant communities.\n",
      "\n",
      "\n",
      "Supporting InformationAppendix S1\n",
      "\n",
      "Stages of decomposition when insects were sampled.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.Appendix S2\n",
      "\n",
      "Summary of plant species and their traits.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.Appendix S3\n",
      "\n",
      "Summary of ant species.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.Appendix S4\n",
      "\n",
      "Summary of beetle species.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.\n"
     ]
    }
   ],
   "source": [
    "# To sample an article with text matches from our terms:\n",
    "art = articles.aggregate([\n",
    "    { \"$match\": { \"text_matches\": { \"$in\": terms } } },\n",
    "    { \"$sample\": { \"size\": 1 } }\n",
    "]).next()\n",
    "\n",
    "# Print the ID, metadata, matches, and extracted text of your article\n",
    "print(\"\"\"\n",
    "PMC ID\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TITLE\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "METADATA\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TEXT MATCHES\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "EXTRACTED TEXT\n",
    "\n",
    "{}\"\"\".format(art.get(\"_id\"),\n",
    "             art.get(\"article_title\"),\n",
    "             art.get(\"article_meta\"),\n",
    "             art.get(\"text_matches\"),\n",
    "             art.get(\"extracted_text\").strip()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "soup = BeautifulSoup(art[\"xml\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3573865'"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "soup.front.find(\"article-id\", attrs={\"pub-id-type\": \"pmc\"}).get_text()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'research-article'"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.article_type()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'PLoS ONE'"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.soup.find(\"front\").find(\"journal-meta\").find(\"journal-title\").get_text()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Methods in Ecology and Evolution'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.journal_title()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Advances in multiplex PCR: balancing primer efficiencies and improving detection success'"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.article_title()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DataFrame scratch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "foo = pd.DataFrame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>one</th>\n",
       "      <th>two</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>dog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   one  two\n",
       "0   10  dog"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame({\"one\": [10], \"two\": [\"dog\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>one</th>\n",
       "      <th>two</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>dog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   one  two\n",
       "0   10  dog"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame.from_records([{\"one\": 10, \"two\": \"dog\"}])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   A\n",
       "0  0\n",
       "1  1\n",
       "2  2\n",
       "3  3\n",
       "4  4"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.concat([pd.DataFrame([i], columns=['A']) for i in range(5)],ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.concat([foo])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Count article types and other metadata"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO: Check if an article's `_id` is in `result_set` and add those to a different default_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "211"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.count_documents({})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-15-ff577f9bb7c5>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcursor\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marticles\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maggregate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m{\u001b[0m \u001b[0;34m\"$sample\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m{\u001b[0m \u001b[0;34m\"size\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0marticles\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcount_documents\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m}\u001b[0m \u001b[0;34m}\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/collection.py\u001b[0m in \u001b[0;36mcount_documents\u001b[0;34m(self, filter, session, **kwargs)\u001b[0m\n\u001b[1;32m   1683\u001b[0m         \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_socket_for_reads\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msession\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0msock_info\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mslave_ok\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1684\u001b[0m             result = self._aggregate_one_result(\n\u001b[0;32m-> 1685\u001b[0;31m                 sock_info, slave_ok, cmd, collation, session)\n\u001b[0m\u001b[1;32m   1686\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1687\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/collection.py\u001b[0m in \u001b[0;36m_aggregate_one_result\u001b[0;34m(self, sock_info, slave_ok, cmd, collation, session)\u001b[0m\n\u001b[1;32m   1585\u001b[0m             \u001b[0mread_concern\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_concern\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1586\u001b[0m             \u001b[0mcollation\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcollation\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1587\u001b[0;31m             session=session)\n\u001b[0m\u001b[1;32m   1588\u001b[0m         \u001b[0mbatch\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'cursor'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'firstBatch'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1589\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mbatch\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/collection.py\u001b[0m in \u001b[0;36m_command\u001b[0;34m(self, sock_info, command, slave_ok, read_preference, codec_options, check, allowable_errors, read_concern, write_concern, collation, session, retryable_write)\u001b[0m\n\u001b[1;32m    242\u001b[0m                 \u001b[0msession\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    243\u001b[0m                 \u001b[0mclient\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__database\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclient\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 244\u001b[0;31m                 retryable_write=retryable_write)\n\u001b[0m\u001b[1;32m    245\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    246\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__create\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moptions\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcollation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/pool.py\u001b[0m in \u001b[0;36mcommand\u001b[0;34m(self, dbname, spec, slave_ok, read_preference, codec_options, check, allowable_errors, check_keys, read_concern, write_concern, parse_write_concern_error, collation, session, client, retryable_write, publish_events)\u001b[0m\n\u001b[1;32m    582\u001b[0m         \u001b[0;31m# Catch socket.error, KeyboardInterrupt, etc. and close ourselves.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    583\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mBaseException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 584\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_raise_connection_failure\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merror\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    585\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    586\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0msend_message\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmessage\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmax_doc_size\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/pool.py\u001b[0m in \u001b[0;36m_raise_connection_failure\u001b[0;34m(self, error)\u001b[0m\n\u001b[1;32m    743\u001b[0m             \u001b[0m_raise_connection_failure\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maddress\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    744\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 745\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    746\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    747\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__eq__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/pool.py\u001b[0m in \u001b[0;36mcommand\u001b[0;34m(self, dbname, spec, slave_ok, read_preference, codec_options, check, allowable_errors, check_keys, read_concern, write_concern, parse_write_concern_error, collation, session, client, retryable_write, publish_events)\u001b[0m\n\u001b[1;32m    577\u001b[0m                            \u001b[0mcompression_ctx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompression_context\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    578\u001b[0m                            \u001b[0muse_op_msg\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mop_msg_enabled\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 579\u001b[0;31m                            unacknowledged=unacknowledged)\n\u001b[0m\u001b[1;32m    580\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mOperationFailure\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    581\u001b[0m             \u001b[0;32mraise\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/network.py\u001b[0m in \u001b[0;36mcommand\u001b[0;34m(sock, dbname, spec, slave_ok, is_mongos, read_preference, codec_options, session, client, check, allowable_errors, address, check_keys, listeners, max_bson_size, read_concern, parse_write_concern_error, collation, compression_ctx, use_op_msg, unacknowledged)\u001b[0m\n\u001b[1;32m    139\u001b[0m             \u001b[0mresponse_doc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m\"ok\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    140\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 141\u001b[0;31m             \u001b[0mreply\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mreceive_message\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msock\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    142\u001b[0m             \u001b[0munpacked_docs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mreply\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munpack_response\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcodec_options\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcodec_options\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/network.py\u001b[0m in \u001b[0;36mreceive_message\u001b[0;34m(sock, request_id, max_message_size)\u001b[0m\n\u001b[1;32m    171\u001b[0m     \u001b[0;31m# Ignore the response's request id.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    172\u001b[0m     length, _, response_to, op_code = _UNPACK_HEADER(\n\u001b[0;32m--> 173\u001b[0;31m         _receive_data_on_socket(sock, 16))\n\u001b[0m\u001b[1;32m    174\u001b[0m     \u001b[0;31m# No request_id for exhaust cursor \"getMore\".\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    175\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mrequest_id\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/network.py\u001b[0m in \u001b[0;36m_receive_data_on_socket\u001b[0;34m(sock, length)\u001b[0m\n\u001b[1;32m    230\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0mbytes_read\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0mlength\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    231\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 232\u001b[0;31m                 \u001b[0mchunk_length\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmv\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mbytes_read\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    233\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mIOError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mOSError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mexc\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    234\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0m_errno_from_exception\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexc\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0merrno\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mEINTR\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "cursor = articles.aggregate([{ \"$sample\": { \"size\": articles.count_documents({}) } }])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "count = articles.count_documents({})\n",
    "cursor = articles.find({})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[F\u001b[KProcessed 175 articles (82.9%) in 0m8s; about 0m2s left.\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-46-1f74dba360bf>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0mrow\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m     \u001b[0marticle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mArticle\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"xml\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     10\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m     \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"id\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"_id\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Dropbox (EHA)/repositories/pubcrawler/pubcrawler/article.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, xml)\u001b[0m\n\u001b[1;32m     13\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mxml\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mxml\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mxml\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 15\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msoup\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mBeautifulSoup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mxml\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'lxml-xml'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     16\u001b[0m         \u001b[0;31m# self._id = article_dict['_id']\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     17\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, markup, features, builder, parse_only, from_encoding, exclude_encodings, **kwargs)\u001b[0m\n\u001b[1;32m    279\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    280\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 281\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_feed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    282\u001b[0m                 \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    283\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mParserRejectedMarkup\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36m_feed\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    340\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuilder\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    341\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 342\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuilder\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmarkup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    343\u001b[0m         \u001b[0;31m# Close out any unfinished strings and close all the open tags.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    344\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mendData\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/builder/_lxml.py\u001b[0m in \u001b[0;36mfeed\u001b[0;34m(self, markup)\u001b[0m\n\u001b[1;32m    166\u001b[0m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmarkup\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mCHUNK_SIZE\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    167\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 168\u001b[0;31m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    169\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    170\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mUnicodeDecodeError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mLookupError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0metree\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mParserError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parser.pxi\u001b[0m in \u001b[0;36mlxml.etree._FeedParser.feed\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parser.pxi\u001b[0m in \u001b[0;36mlxml.etree._FeedParser.feed\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._TargetParserContext._handleParseResult\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._TargetParserContext._handleParseResult\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/etree.pyx\u001b[0m in \u001b[0;36mlxml.etree._ExceptionContext._raise_if_stored\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/saxparser.pxi\u001b[0m in \u001b[0;36mlxml.etree._handleSaxTargetStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/saxparser.pxi\u001b[0m in \u001b[0;36mlxml.etree._callTargetSaxStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._PythonSaxParserTarget._handleSaxStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/builder/_lxml.py\u001b[0m in \u001b[0;36mstart\u001b[0;34m(self, name, attrs, nsmap)\u001b[0m\n\u001b[1;32m    218\u001b[0m         \u001b[0mnamespace\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getNsTag\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    219\u001b[0m         \u001b[0mnsprefix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_prefix_for_namespace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnamespace\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 220\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msoup\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhandle_starttag\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnamespace\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnsprefix\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mattrs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    221\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    222\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_prefix_for_namespace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnamespace\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36mhandle_starttag\u001b[0;34m(self, name, namespace, nsprefix, attrs)\u001b[0m\n\u001b[1;32m    528\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    529\u001b[0m         tag = Tag(self, self.builder, name, namespace, nsprefix, attrs,\n\u001b[0;32m--> 530\u001b[0;31m                   self.currentTag, self._most_recent_element)\n\u001b[0m\u001b[1;32m    531\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mtag\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    532\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mtag\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/element.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, parser, builder, name, namespace, prefix, attrs, parent, previous, is_xml)\u001b[0m\n\u001b[1;32m    859\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mattrs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mattrs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    860\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcontents\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 861\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mparent\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprevious\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    862\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhidden\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    863\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/element.py\u001b[0m in \u001b[0;36msetup\u001b[0;34m(self, parent, previous_element, next_element, previous_sibling, next_sibling)\u001b[0m\n\u001b[1;32m    260\u001b[0m         \"\"\"Sets up the initial relations between this element and\n\u001b[1;32m    261\u001b[0m         other elements.\"\"\"\n\u001b[0;32m--> 262\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparent\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparent\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    263\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    264\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprevious_element\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprevious_element\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "reporter = Reporter(25, len(matches))\n",
    "\n",
    "rows = []\n",
    "\n",
    "for idx, record in enumerate(matches):\n",
    "    reporter.report(idx)\n",
    "\n",
    "    row = {}\n",
    "    article = Article(record[\"xml\"])\n",
    "\n",
    "    row[\"id\"] = record[\"_id\"]\n",
    "    row[\"keywords\"] = article.keywords()\n",
    "    row[\"id_types\"] = list(article.pub_ids().keys())\n",
    "    row[\"article_type\"] = article.article_type()\n",
    "    row[\"has_body\"] = True if article.soup.body else False\n",
    "    row[\"text_matches\"] = record.get(\"text_matches\")\n",
    "    row[\"article_title\"] = article.article_title()\n",
    "    row[\"journal_title\"] = article.journal_title()\n",
    "    row[\"text_length\"] = len(record[\"extracted_text\"])\n",
    "\n",
    "    rows.append(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "article_df = pd.DataFrame.from_records(rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "any_matches\n",
       "False    19519\n",
       "True       481\n",
       "dtype: int64"
      ]
     },
     "execution_count": 282,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article_df[\"any_matches\"] = [False if row is None else True for row in article_df[\"text_matches\"]]\n",
    "article_df.groupby([\"any_matches\"]).size()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "metadata": {},
   "outputs": [],
   "source": [
    "article_df.to_csv(\"article_sample.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>any_matches</th>\n",
       "      <th>article_type</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>True</td>\n",
       "      <td>research-article</td>\n",
       "      <td>446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>True</td>\n",
       "      <td>review-article</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>True</td>\n",
       "      <td>brief-report</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>True</td>\n",
       "      <td>protocol</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>True</td>\n",
       "      <td>case-report</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>True</td>\n",
       "      <td>other</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>True</td>\n",
       "      <td>data-paper</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>True</td>\n",
       "      <td>editorial</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>True</td>\n",
       "      <td>letter</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>False</td>\n",
       "      <td>research-article</td>\n",
       "      <td>14138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>False</td>\n",
       "      <td>review-article</td>\n",
       "      <td>1279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>case-report</td>\n",
       "      <td>896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>abstract</td>\n",
       "      <td>507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>False</td>\n",
       "      <td>editorial</td>\n",
       "      <td>314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>brief-report</td>\n",
       "      <td>311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>book-review</td>\n",
       "      <td>306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>False</td>\n",
       "      <td>letter</td>\n",
       "      <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>False</td>\n",
       "      <td>correction</td>\n",
       "      <td>245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>article-commentary</td>\n",
       "      <td>92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>False</td>\n",
       "      <td>news</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>False</td>\n",
       "      <td>meeting-report</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>False</td>\n",
       "      <td>discussion</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>False</td>\n",
       "      <td>product-review</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>False</td>\n",
       "      <td>obituary</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>False</td>\n",
       "      <td>protocol</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>False</td>\n",
       "      <td>in-brief</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>False</td>\n",
       "      <td>rapid-communication</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>False</td>\n",
       "      <td>methods-article</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>False</td>\n",
       "      <td>retraction</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>False</td>\n",
       "      <td>systematic-review</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>False</td>\n",
       "      <td>introduction</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>False</td>\n",
       "      <td>reply</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>False</td>\n",
       "      <td>data-paper</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>addendum</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>books-received</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>announcement</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>False</td>\n",
       "      <td>oration</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>False</td>\n",
       "      <td>report</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>case-report</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>calendar</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    any_matches         article_type  count\n",
       "39         True     research-article    446\n",
       "40         True       review-article     15\n",
       "32         True         brief-report      7\n",
       "38         True             protocol      4\n",
       "33         True          case-report      3\n",
       "37         True                other      3\n",
       "34         True           data-paper      1\n",
       "35         True            editorial      1\n",
       "36         True               letter      1\n",
       "28        False     research-article  14138\n",
       "30        False       review-article   1279\n",
       "9         False          case-report    896\n",
       "22        False                other    745\n",
       "1         False             abstract    507\n",
       "13        False            editorial    314\n",
       "7         False         brief-report    311\n",
       "5         False          book-review    306\n",
       "16        False               letter    301\n",
       "10        False           correction    245\n",
       "4         False   article-commentary     92\n",
       "19        False                 news     82\n",
       "17        False       meeting-report     67\n",
       "12        False           discussion     42\n",
       "23        False       product-review     31\n",
       "20        False             obituary     29\n",
       "24        False             protocol     27\n",
       "14        False             in-brief     17\n",
       "25        False  rapid-communication     15\n",
       "18        False      methods-article     13\n",
       "29        False           retraction     12\n",
       "31        False    systematic-review     11\n",
       "15        False         introduction      9\n",
       "26        False                reply      9\n",
       "11        False           data-paper      7\n",
       "2         False             addendum      3\n",
       "6         False       books-received      3\n",
       "3         False         announcement      2\n",
       "21        False              oration      2\n",
       "27        False               report      2\n",
       "0         False          case-report      1\n",
       "8         False             calendar      1"
      ]
     },
     "execution_count": 285,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article_df.groupby([\"any_matches\", \"article_type\"]).size()\n",
    "art_types = article_df.groupby([\"any_matches\", \"article_type\"]).size().reset_index()\n",
    "art_types.columns = [\"any_matches\", \"article_type\", \"count\"]\n",
    "art_types.sort_values(by = [\"any_matches\", \"count\"], ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "metadata": {},
   "outputs": [],
   "source": [
    "keep_vars = [\"id\", \"any_matches\"]\n",
    "unnest_var = \"keywords\"\n",
    "\n",
    "def unnest(data, unnest_var, keep_vars):\n",
    "    all_vars = keep_vars + [unnest_var]\n",
    "    nested = article_df.loc[:, keep_vars + [unnest_var]]\n",
    "    lens = [len(item) if item is not None else 1 for item in nested[unnest_var]]\n",
    "    unnested_dict = {var: np.repeat([nested[var].values], lens) for var in keep_vars}\n",
    "    unnested_dict[unnest_var] = np.hstack(nested[unnest_var])\n",
    "    unnested = pd.DataFrame(unnested_dict)\n",
    "    return(unnested)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "metadata": {},
   "outputs": [],
   "source": [
    "keywords = unnest(article_df, \"keywords\", [\"id\", \"any_matches\"])\n",
    "\n",
    "keywords.to_csv(\"keywords.csv\")\n",
    "\n",
    "id_types = unnest(article_df, \"id_types\", [\"id\", \"any_matches\"])\n",
    "\n",
    "id_types.to_csv(\"id_types.csv\")\n",
    "\n",
    "text_matches = unnest(article_df, \"text_matches\", [\"id\"])\n",
    "\n",
    "text_matches.to_csv(\"text_matches.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Old version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processed 240 articles (96.0%) in 0m12s; about 0m1s left.\n"
     ]
    }
   ],
   "source": [
    "id_types = defaultdict(int)\n",
    "article_types = defaultdict(int)\n",
    "keywords = defaultdict(int)\n",
    "\n",
    "reporter = Reporter(10, len(matches))\n",
    "\n",
    "for idx, doc in enumerate(matches):\n",
    "    if idx is not 0 and idx % reporter.interval is 0:\n",
    "        reporter.report(idx)\n",
    "    article = Article(doc[\"xml\"])\n",
    "    \n",
    "    kwds = article.keywords()\n",
    "    if kwds:\n",
    "        for kwd in kwds:\n",
    "            keywords[kwd] += 1\n",
    "    \n",
    "    ids = article.pub_ids().keys()\n",
    "    if len(ids) is 0:\n",
    "        id_types[\"none\"] += 1\n",
    "    else:\n",
    "        for item in ids:\n",
    "            id_types[item] += 1\n",
    "    \n",
    "    article_type = article.article_type()\n",
    "    if article_type:\n",
    "        article_types[article_type] += 1\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "keywords = pd.DataFrame(list(keywords.items()), columns=[\"keyword\", \"count\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>keyword</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>inflammation</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>oxidative stress</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>epidemiology</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>Obesity</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>Prostate cancer</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220</th>\n",
       "      <td>breast cancer</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>SNP</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1833</th>\n",
       "      <td>cytotoxicity</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>mitochondria</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>depression</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>801</th>\n",
       "      <td>Parkinson’s disease</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>388</th>\n",
       "      <td>Breast cancer</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1566</th>\n",
       "      <td>Prognosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2259</th>\n",
       "      <td>autophagy</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1274</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>obesity</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>819</th>\n",
       "      <td>Saccharomyces cerevisiae</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>459</th>\n",
       "      <td>Aging</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>Proliferation</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464</th>\n",
       "      <td>drug resistance</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>466</th>\n",
       "      <td>tuberculosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1320</th>\n",
       "      <td>cancer</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1523</th>\n",
       "      <td>diagnosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1731</th>\n",
       "      <td>apoptosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>PD-1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>694</th>\n",
       "      <td>Depression</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1223</th>\n",
       "      <td>clinical trial</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1762</th>\n",
       "      <td>Apoptosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>662</th>\n",
       "      <td>rehabilitation</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1003</th>\n",
       "      <td>Personality disorders</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1004</th>\n",
       "      <td>Carbon footprint</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1005</th>\n",
       "      <td>Economic evaluation</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1006</th>\n",
       "      <td>Online education</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1008</th>\n",
       "      <td>evaluation</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1009</th>\n",
       "      <td>star trackers</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1010</th>\n",
       "      <td>slew tolerance</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1011</th>\n",
       "      <td>attitude availability</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1012</th>\n",
       "      <td>star detection</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>celiac disease</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>Helicobacter pylori</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>immunoglobulin A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>981</th>\n",
       "      <td>data association</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>974</th>\n",
       "      <td>[18F]FP-(+)-DTBZ</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>975</th>\n",
       "      <td>PET</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>Test-retest</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>977</th>\n",
       "      <td>Cryptococcosis</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>978</th>\n",
       "      <td>Pulmonary Tuberculosis</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>Coinfection</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>980</th>\n",
       "      <td>multi-object tracking</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>982</th>\n",
       "      <td>deep neural network</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>lysosomal enzymes</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983</th>\n",
       "      <td>long short-term memory network</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>nanoparticles</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>turkey</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>aminopeptidases</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>poultry meat</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>988</th>\n",
       "      <td>nutrition</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989</th>\n",
       "      <td>supplementation</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2796</th>\n",
       "      <td>argonaute</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2797 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             keyword  count\n",
       "43                      inflammation      6\n",
       "587                 oxidative stress      5\n",
       "159                     epidemiology      5\n",
       "183                          Obesity      4\n",
       "279                  Prostate cancer      4\n",
       "1220                   breast cancer      4\n",
       "647                              SNP      4\n",
       "1833                    cytotoxicity      4\n",
       "45                      mitochondria      4\n",
       "294                       depression      4\n",
       "801              Parkinson’s disease      4\n",
       "388                    Breast cancer      4\n",
       "1566                       Prognosis      3\n",
       "2259                       autophagy      3\n",
       "1274                          Cancer      3\n",
       "42                           obesity      3\n",
       "819         Saccharomyces cerevisiae      3\n",
       "459                            Aging      3\n",
       "543                    Proliferation      3\n",
       "464                  drug resistance      3\n",
       "466                     tuberculosis      3\n",
       "1320                          cancer      3\n",
       "1523                       diagnosis      3\n",
       "1731                       apoptosis      3\n",
       "361                             PD-1      3\n",
       "694                       Depression      3\n",
       "1223                  clinical trial      3\n",
       "1762                       Apoptosis      3\n",
       "662                   rehabilitation      3\n",
       "396                colorectal cancer      3\n",
       "...                              ...    ...\n",
       "1003           Personality disorders      1\n",
       "1004                Carbon footprint      1\n",
       "1005             Economic evaluation      1\n",
       "1006                Online education      1\n",
       "1008                      evaluation      1\n",
       "1009                   star trackers      1\n",
       "1010                  slew tolerance      1\n",
       "1011           attitude availability      1\n",
       "1012                  star detection      1\n",
       "993                   celiac disease      1\n",
       "992              Helicobacter pylori      1\n",
       "991                 immunoglobulin A      1\n",
       "981                 data association      1\n",
       "974                 [18F]FP-(+)-DTBZ      1\n",
       "975                              PET      1\n",
       "976                      Test-retest      1\n",
       "977                   Cryptococcosis      1\n",
       "978           Pulmonary Tuberculosis      1\n",
       "979                      Coinfection      1\n",
       "980            multi-object tracking      1\n",
       "982              deep neural network      1\n",
       "990                lysosomal enzymes      1\n",
       "983   long short-term memory network      1\n",
       "984                    nanoparticles      1\n",
       "985                           turkey      1\n",
       "986                  aminopeptidases      1\n",
       "987                     poultry meat      1\n",
       "988                        nutrition      1\n",
       "989                  supplementation      1\n",
       "2796                       argonaute      1\n",
       "\n",
       "[2797 rows x 2 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "keywords.sort_values(\"count\", ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(int,\n",
       "            {'pmid': 18771,\n",
       "             'pmc': 20000,\n",
       "             'publisher-id': 13829,\n",
       "             'doi': 17408,\n",
       "             'pii': 1117,\n",
       "             'pmc-scan': 1198,\n",
       "             'manuscript': 206,\n",
       "             'coden': 219,\n",
       "             'art-access-id': 208,\n",
       "             'other': 134,\n",
       "             'publisher-manuscript': 22,\n",
       "             'sici': 27,\n",
       "             'medline': 4})"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(int,\n",
       "            {'research-article': 14584,\n",
       "             'abstract': 507,\n",
       "             'brief-report': 318,\n",
       "             'retraction': 12,\n",
       "             'other': 748,\n",
       "             'review-article': 1294,\n",
       "             'editorial': 315,\n",
       "             'case-report': 899,\n",
       "             'correction': 245,\n",
       "             'news': 82,\n",
       "             'discussion': 42,\n",
       "             'letter': 302,\n",
       "             'meeting-report': 67,\n",
       "             'product-review': 31,\n",
       "             'reply': 9,\n",
       "             'article-commentary': 92,\n",
       "             'addendum': 3,\n",
       "             'book-review': 306,\n",
       "             'protocol': 31,\n",
       "             'obituary': 29,\n",
       "             'rapid-communication': 15,\n",
       "             'methods-article': 13,\n",
       "             'in-brief': 17,\n",
       "             'systematic-review': 11,\n",
       "             'introduction': 9,\n",
       "             'oration': 2,\n",
       "             'books-received': 3,\n",
       "             ' case-report': 1,\n",
       "             'data-paper': 8,\n",
       "             'report': 2,\n",
       "             'announcement': 2,\n",
       "             'calendar': 1})"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article_types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Look at the text of a single document matching a search result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pymongo.results.UpdateResult at 0x10d353408>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.update_many(\n",
    "    filter={},\n",
    "    update={ \"$unset\": { \"text_matches\": \"\" } }\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
