{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pymongo\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.display import clear_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pubcrawler.article import Article\n",
    "from bs4 import BeautifulSoup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "from reporter import Reporter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "client = pymongo.MongoClient(\"localhost\", 27017)\n",
    "articles = client.pmc.articles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"terms\") as f:\n",
    "    terms = [line.strip() for line in f.readlines()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "art = articles.find_one()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To sample a random article:\n",
    "art = articles.aggregate([\n",
    "    { \"$sample\": { \"size\": 1 } }\n",
    "]).next()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['field work',\n",
       " 'fieldwork',\n",
       " 'field study',\n",
       " 'field site',\n",
       " 'field area',\n",
       " 'study site',\n",
       " 'study location',\n",
       " 'study area',\n",
       " 'research site',\n",
       " 'research location',\n",
       " 'sampling site',\n",
       " 'sampling location',\n",
       " 'sampling area']"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "def view_one_article(query={ \"$match\": { \"text_matches\": { \"$in\": terms } } }):\n",
    "    query = [\n",
    "        query,\n",
    "        { \"$sample\": { \"size\": 1 } }\n",
    "    ]\n",
    "    article = articles.aggregate(query).next()\n",
    "    print(\"\"\"\n",
    "PMC ID\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TITLE\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "METADATA\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TEXT MATCHES\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "EXTRACTED TEXT\n",
    "\n",
    "{}\"\"\".format(article.get(\"_id\"),\n",
    "             article.get(\"article_title\"),\n",
    "             article.get(\"article_meta\"),\n",
    "             article.get(\"text_matches\"),\n",
    "             article.get(\"extracted_text\").strip()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "problematic_terms = terms[-3:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "3538743\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "None\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "None\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['study site', 'study area', 'sampling location', 'sampling area']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Introduction\n",
      "\n",
      "Comparisons of persisting versus shorter-term effects of a given long-term disturbance are less common than might be expected; grazing management has provided a good laboratory for such studies, because of detailed, long-term stock use records, the presence of de facto long-term exclosures, and an understanding among managers that long- and short-term grazing effects may differ [1]–[4]. These studies have shown both similarities between long- and short-term effects as well as divergent effects [5], [6]. A significant period of time may be required for indirect effects or other subtle, slow, or complex processes to operate in ecosystems [2], [3], [7] resulting in, for example, short-term increases in nutrients due to grazing eventually transitioning to long-term nutrient decreases [2], [8], which in turn may influence arthropods in complex ways [9], [10]. Some short-term investigations predict long-term species-level effects well but predict assemblage or ecosystem effects poorly (discussion in [2]). In our current work, we contrast effects of pack stock grazing manifested during early and mid-growing season in vulnerable subalpine wetlands. We make use of a landscape-scale, long-term manipulation (see also [11]) of wetland grazing created by the management of recreational pack stock in Sequoia National Park (Sierra Nevada mountains of California, USA) [12]. Effects of mules and other pack stock have been the focus of comparatively few investigations [13]. Our study uses an unusually broad suite of response variables to capture ecosystem response across multiple trophic levels rather than relying on primary producers in isolation [14], [15]. We evaluate responses across the full breadth of Arthropoda in addition to vegetation structural characteristics (see also [16], [17], [18]), allowing detection of both direct and indirect grazing effects. Such indirect effects are under-investigated in assessments of recreational impacts, particularly on larger spatial and temporal scales [19], [20].\n",
      "\n",
      "Absence of apparent long-term disturbance effects does not render shorter-term effects trivial. Invertebrates may be particularly susceptible to such additional short-term effects, but these impacts may not be easily ascertained, because a) invertebrates have been under-investigated in ecosystem studies in general, and b) short-term effects on invertebrates may not be detected by long-term sampling as a result of masking by dispersal and/or recolonization [21]–[23]. Some studies report mid-season effects on arthropods from grazing or other forms of canopy removal [24]–[26]; canopy removal experiments suggest that univoltine arthropods may be particularly vulnerable to mid-season stock effects [27]. Flowering in the subalpine growing season occurs while stock are present [12], and flower removal can impact butterflies [28] and other nectivores [22], [29], which could result in a feedback loop that reduces populations of both flowering plants and pollinators. Although habitats may recover during the winter, resulting in few apparent long-term effects, it is possible that there are impacts on arthropod assemblages during each growing season that could cascade into vertebrate or upland assemblages [29].\n",
      "\n",
      "High elevation and high latitude wetlands are valued ecosystem components (e.g., [30]) that have short growing seasons and tend to be vulnerable to anthropogenic disturbance [9], [17], [31], in part because these habitats retain high levels of soil moisture through the growing season and because of slow vegetation growth/regrowth [32]. Conclusions based solely on infrequent, long-term sampling may be particularly likely to be misleading in snow-dominated ecosystems, because the long winter may provide annual recovery periods of nine months or longer that may erase severe, short-term impacts occurring at the height of the abbreviated growing season. Pack stock are an example of a potential disturbance source that coincides with the apex of the growing season in the mountain environment [12], [32], [33]; such timing of disturbance is relatively common in mountain ecosystems [34]. Pack stock graze subalpine wet meadows in short two-three month pulses, followed by a long winter recovery period [12]. Stock are primarily mules and horses, but there are occasional burros or llamas [17], [18], [35]; these animals are used to transport recreationists and materials deep into wilderness areas.\n",
      "\n",
      "In an initial, one-year study, we sought to determine if these grazing patterns caused lasting effects on terrestrial, epigeal arthropods and associated wetland vegetation, or if the long winters without stock allowed an annual recovery of assemblages from any impacts that occur during summer usage [12]. We sampled in early season only, and stock grazing had moderate long-term effects on vegetation but little apparent effect on arthropod fauna in these wetlands during early season [12]. The primary objective of the current two-year study was to compare a) long-term grazing effects that are manifested in early season before annual usage, with b) the full suite of effects that might become apparent at mid-season, from either the numerous, known components of short-term, annual disturbance (e.g., cropping, hoof punching, feces and urine deposition) or from unknown but possible lasting phenomena that might be detectable later in the stock/growing season but not in early season (e.g., delayed changes in productivity/canopy structure). Our secondary objective was comparison of arthropod assemblage structure, irrespective of grazing, at the beginning and end of the growing season. Higher diversity and abundance might be expected in early season because of high soil moisture or the initial arrival of warmer temperatures, or alternatively in mid-season as a response to better-developed vegetation structure.\n",
      "\n",
      "\n",
      "Materials and Methods\n",
      "\n",
      "\n",
      "Study Area, Sites, and Design\n",
      "\n",
      "Wet meadows are saturated with water during much of the year [36], [37], and terrestrial arthropods are abundant and diverse in these habitats [23]. Stock access to subalpine wetlands is generally prohibited by Sequoia National Park until about one month after snowmelt and sometimes substantially longer, depending upon the length of the preceding winter at this elevation (2587–3242 masl). Stock are thus typically present from July-September. We used the same reed grass-dominated (Calamagrostis muiriana B.L. Wilson and S. Gray; see [32] for a good image) sites and the same site-scale methodology as in our early season-only study [12]. Grazed wetlands used in the study received a mean of 18.5 (SE  = 4.2) stock nights/ha/year over the last two decades.\n",
      "\n",
      "Pack stock grazing and associated management practices in Sequoia National Park present an ideal scenario for examination of long-term grazing effects. This work was facilitated by a) the presence of many wet meadows that had been closed to stock for decades that could be paired for contrast with grazed wet meadows with known usage patterns, and b) a controlled opening date for grazing on each wet meadow, so we could sample immediately after greenup, i.e., after there was high quality arthropod habitat, but just before stock grazing. The grazing patterns and management regime enabled us to design what was in essence a subsequent long-term and large-scale experiment (see also [11]). We investigated effects on arthropods and vegetation in subalpine wetlands during early season, and we also sampled after these wetlands had been grazed by stock for the entirety of the growing season.\n",
      "\n",
      "The study was cast as a 2×2×2 blocked design (Treatment: Control, Grazed; Season: Early, Mid; Year: 2010, 2011) using ten pairs of control and grazed subalpine wet meadows. Each study site had two randomly-selected subsample locations, with two additional randomly-selected subsamples nested within each of the first pair of subsamples. Vegetation in these wetlands typically begins to senesce in mid- to late September [33], and arthropod diversity and abundance decline at this time and remain low through the late, senescent season (Holmquist and Schmidt-Gengenbach, unpublished report). We did mid-season sampling at the end of the growing season, i.e., just before vegetation senescence. Early season comparison of grazed and control sites should reveal long-term effects in isolation from annual, growing season effects. Mid-season comparison of grazed and control sites, in concert with comparison of the same wetlands in early season, should detect any additional impacts manifested during the current growing season. Treatment x Season interactions were thus of particular interest, especially any results that might indicate lack of grazed-control differences in early season but presence of such effects at mid-season. We used a paired design in order to minimize the influence of differences among watersheds, vegetation types, and spatial effects. Site selections were driven by a limited number of sites that had a long history of grazing and that could be paired with control sites, having similar vegetation, that had been closed to grazing for many years. Individual wet meadow pairs were separated by a mean of only 960 m (SE  = 160) and were in the same watersheds, allowing us to sample both meadows of a pair rapidly, before there could be significant meteorological changes. Although individual wet meadow pairs were tightly co-located, meadow blocks were separated by up to 40 km, requiring up to five days of foot travel to sample some groups of sites. The design thus included good replicate dispersion ([38], see also [3]) across a large, wilderness landscape. Distances between grazed and control sites were normally distributed (Lilliefors and Shapiro-Wilks tests, p = 0.24 and 0.94, respectively). Voronoi tessellations and spanning trees were similar for both site categories, and there was no pattern and little roughness apparent for spatial surface models of either control or grazed sites across representative response variables.\n",
      "\n",
      "We sampled all sites four days or less before grazed sites were opened to pack stock in early season and similarly just before vegetation senescence toward the end of mid-season conditions in both 2010 and 2011. The two years captured varying antecedent conditions, because the winter preceding 2011 sampling produced greater snow water equivalent (SWE  = 131 cm at our Hockett Meadow block; ≈590 cm snow depth) than the winter preceding 2010 sampling (SWE  = 89 cm). Stock opening date at individual meadows is determined by the Park based on soil saturation and vegetation characteristics, so we sampled under similar phenological conditions in each year. Sampling in 2010 began in early July and concluded in early September, whereas 2011 sampling ran from early August through mid-September because of late snowmelt. See [12] for additional details on Park grazing management, vegetation assemblages, site locations, and stock use patterns in individual wet meadows.\n",
      "\n",
      "\n",
      "Ethics Statement\n",
      "\n",
      "A Scientific Research and Collecting permit was obtained from the US National Park Service for work in Sequoia National Park for each year of the study. No protected species were sampled.\n",
      "\n",
      "\n",
      "Faunal Methodology\n",
      "\n",
      "We sampled the wetland canopy assemblage with sweep nets and secondarily targeted ground-dwelling fauna, especially ants, by baiting. Sweep nets are likely the most frequently used device for sampling epigaeic arthropods, can detect sparsely distributed taxa [39], integrate relatively large areas, are easy to transport while backpacking, and do not negatively affect wilderness qualities (see also [12]). Limitations include the potential for efficiency to be influenced by habitat and arthropod assemblage structure, weather, and diel periodicity, vertical distribution, and activity of fauna, [39]. Similarly, baits are the most common tool used to sample ant assemblages [40] and offer many of the same advantages as sweep nets, but there is a higher probability of capture for omnivorous, wide-ranging, or dominant taxa [40], [41]. Differential capture probability can be mitigated by use of baits of differing types [40].\n",
      "\n",
      "Each sweep sample for a site consisted of 50 standard sweep net sweeps [39], [42] divided between the two subsampling locations at each site and covering a total of 400 m2. The collapsible net had a 30.5 cm aperture and mesh size of 0.5×0.75 mm (BioQuip #7112CP). We collected sweep samples prior to the disruption associated with other data collection at the sites, and samples were killed with 99% ethyl acetate [43]. We placed a honey bait at one subsample location and a tuna bait at the other subsample location immediately after sweep netting. The baits were ∼1 cm2 portions of honey or tuna that were placed on green construction paper cards and weighted with rocks. After 30 minutes, ants, mites, and other arthropods were removed with forceps and placed in a vial containing 70% ethanol (additional sampling details in [12]).\n",
      "\n",
      "Sweep samples were sorted in the lab and identified to family (see also [44], [45]) and morphospecies ([46]–[49]; see also [12], [23], [50]). Ants from the bait samples were identified to species. If taxonomic ambiguity (sensu [51]) existed, we used the “distribute parents among children” approach [51] on a per sample basis. Vouchers were stored with the University of California. This study was an investigation targeting specific ecological questions by examining responses across the full breadth of Arthropoda [44], [45]; higher taxonomic resolution was often impractical due to the large collections, an abundance of immature specimens and undescribed species, and because a number of groups await revision.\n",
      "\n",
      "\n",
      "Vegetation and Physical Data\n",
      "\n",
      "We estimated percent bare ground, percent green, standing brown (senescent), and litter cover, as well as canopy height and litter depth on each of the subsample locations. Such coarse vegetation parameters are effective in detecting pack stock impacts on vegetation assemblages [17], [52], [53]. Cover categories were estimated with a point-intercept transect (20 points) centered in each subsample location and randomly oriented. We measured canopy height and litter depth at two random locations within each subsample. Vegetation variables were thus means of two or four measurements at each site.\n",
      "\n",
      "Air temperature and average wind speed were recorded midway between the two subsample locations with a Kestrel 3000 digital meter in order to verify that similar meteorological conditions obtained between paired grazed and control wetlands. We used a pocket penetrometer (Ben Meadows) to estimate soil strength at each of the locations used for canopy height and litter depth measurement, and the average of these four estimates was the site mean.\n",
      "\n",
      "\n",
      "Analysis\n",
      "\n",
      "We examined the influences of grazing, season, and year on invertebrate assemblages and vegetation structure with both uni- and multivariate approaches. Univariate analyses were 2×2×2 blocked ANCOVAs (df  = 49) using a general linear model in SYSTAT 12. Analysis of site elevation as a covariate was necessary because elevation differed by treatment but could not be affected by the treatment [21], [54] (although mean elevation differences between grazed and control wet meadows were <60 m [12]). Faunal metrics included order and family abundances, family and morphospecies richness and dominance, evenness (probability of interspecific encounter, PIE, [55]), expected number of species scaled to the number of individuals in the sample with the fewest individuals (E(S18); [55], [56]), and percentages of predators, herbivores, and more- and less-motile taxa. Use of E(S) is valuable, because samples with larger numbers of individuals will tend to have more species, even if all samples represent equal effort and are collected from the same assemblage. We calculated E(S18), and PIE using the application Diversity. Metrics that showed departures from normality (Lilliefors tests; [57]) or heteroscedasticity (Fmax and Cochran's tests; [58]), were modified with square-root transformations ((y)0.5+(y+1)0.5) of proportional data and log transformations (log (y+1)) of all other data such that parametric assumptions were met. Substitutions were not made for cells with missing values. We used G*Power [59], our known sampling design and sample size, and the standard a priori estimate for effect size of 0.5 [60] to estimate power a priori. Estimated power was high (0.98). We used the sequential Bonferroni adjustment [61] and the application MacBonferroni to calculate alternative, conservative probability values for an overall family-wise error rate of 0.05 across suites of ANCOVAs.\n",
      "\n",
      "We also used multivariate analyses in order to detect patterns as a function of study factors, across both family and morphospecies matrices, that might not emerge via univariate tests of individual taxa or assemblage metrics (only univariate methods were used in the earlier study [12]). We used hierarchical, polythetic, agglomerative cluster analyses to group samples based upon similarities in response patterns [62], [63]. Differences among groups were then assessed with MRPP using both a priori and a posteriori groupings [62], [63]. We also performed permutational analyses of dispersion, because dispersion patterns of observations have the potential to influence MRPP results. Aphid morphospecies were excluded from multivariate analyses, because of difficulty in distinguishing individual aphid morphospecies across samples. We performed cluster and MRPP analyses using PC-ORD 6 [62], [63] and analyses of dispersion using PERMDISP2 software developed by MJ Anderson (see also [64], [65]). Response and explanatory matrices contained all sites. The response matrices of families and morphospecies included taxa that we sampled at three or more sites (62 families and 120 morphospecies; [62], [63] but see [66]). These matrices were relativized by maximum abundance for each family or morphospecies. The final family primary matrix had a coefficient of variation of 56%, and 71% of the cells contained zeros, and the primary morphospecies matrix had a CV of 45% and 81% zeros. The Sørensen (Bray-Curtis) distance measure [63] was used because this measure is superior for sparse data (with many zeros, as often occurs in taxonomic matrices), retaining sensitivity to heterogeneity without being overly sensitive to outliers [63]. We used group average linkage for the cluster analysis. Treatment and Season coding variables from an explanatory matrix were used for the initial cluster analyses, and an additional group membership variable based on the four highest level groups from the resulting dendrograms, versus initially coded factors, was added to the explanatory matrix for both families and morphospecies. We used MRPP, including pairwise comparisons, with both the initial coding variables and the additional group membership variables. The distance matrices were rank-transformed prior to MRPP. The permutational dispersion analyses were based on 9,999 permutations and used the same datasets and distance measure used for MRPP without additional transformations or other data modifications.\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "\n",
      "Vegetation and Physical\n",
      "\n",
      "A number of vegetation and physical parameters had significant responses to grazing, season, and year, and there were also significant block effects and interaction terms (Table 1, Figs. 1, 2). Litter depth, litter cover, bare ground, and soil strength indicated impacts from the grazing treatment. There was also an overall significant trend of poorer habitat conditions under grazing exposure across metrics, seasons, and years (p = 0.0015; two-tailed sign test; Table 1). Three variables showed seasonal differences: litter cover was lower later in the growing season, whereas brown cover and soil strength (compaction) increased, but most variables had a significant Season x Year interaction term. There was only a single significant Treatment x Season interaction; grazed sites had less brown, senescent cover in early season than control sites, but more brown cover in mid-season (Table 1, Fig. 1). Block effects were present for all variables except bare ground (Table 1).\n",
      "\n",
      "10.1371/journal.pone.0054109.g001Figure 1 Means (SEs) for percent cover components as a function of grazing treatment and season.See Table 1 for test results.\n",
      "\n",
      "10.1371/journal.pone.0054109.g002Figure 2 Means (SEs) for canopy height and litter depth by grazing treatment and season.See Table 1 for test results.\n",
      "\n",
      "Table 1\n",
      "Early\tMid\tANCOVA\n",
      "Control\tGrazed\tControl\tGrazed\tT\tS\tY\tTxS\tTxY\tSxY\tB\n",
      "Canopy height (cm)\t'10\t7.10 (1.0)\t7.98 (1.2)\t14.9 (1.6)\t11.1 (1.5)\t**\t*\n",
      "'11\t9.33 (1.2)\t9.55 (1.4)\t7.30 (0.73)\t6.54 (0.73)\n",
      "Litter depth (cm)\t'10\t2.08 (0.17)\t0.469 (0.23)\t1.53 (0.18)\t0.393 (0.13)\t**\t*\t**\n",
      "'11\t1.04 (0.17)\t1.00 (0.21)\t1.25 (0.18)\t1.13 (0.19)\n",
      "Litter cover (%)\t'10\t22.8 (4.0)\t10.9 (3.2)\t12.5 (3.4)\t0.714 (0.46)\t**\t*\t**\n",
      "'11\t10.8 (2.4)\t10.4 (3.3)\t8.25 (2.0)\t6.50 (2.3)\n",
      "Bare ground (%)\t'10\t4.50 (1.1)\t22.5 (3.1)\t1.25 (0.56)\t17.1 (5.4)\t*\t**\t*\n",
      "'11\t9.00 (1.9)\t9.17 (2.0)\t9.25 (1.5)\t12.0 (5.2)\n",
      "Brown cover (%)\t'10\t14.5 (2.8)\t5.94 (3.2)\t10.8 (2.1)\t13.2 (3.7)\t**\t*\n",
      "'11\t12.8 (1.8)\t6.67 (1.9)\t18.0 (3.0)\t27.5 (7.5)\n",
      "Green cover (%)\t'10\t58.3 (6.3)\t60.6 (7.5)\t75.5 (3.6)\t68.9 (6.6)\t**\t**\n",
      "'11\t67.5 (5.0)\t73.8 (5.7)\t64.5 (4.2)\t54.0 (7.9)\n",
      "Soil strength (kg/cm\t'10\t0.967 (0.10)\t1.41 (0.12)\t1.52 (0.15)\t1.96 (0.13)\t**\t**\n",
      "'11\t1.71 (0.18)\t2.19 (0.20)\t2.01 (0.084)\t2.73 (0.42)\n",
      "Wind speed (km/hr)\t'10\t7.39 (1.4)\t4.66 (1.4)\t5.50 (0.69)\t3.71 (1.1)\t*\n",
      "'11\t6.35 (0.95)\t4.10 (0.91)\t5.90 (1.0)\t5.12 (1.1)\n",
      "Air temperature (°C)\t'10\t17.4 (0.77)\t18.3 (0.63)\t19.6 (1.2)\t18.7 (1.8)\t*\n",
      "'11\t19.8 (0.41)\t21.3 (0.85)\t17.7 (0.77)\t18.8 (0.63)\n",
      "\n",
      "a\n",
      "\n",
      "b\n",
      "\n",
      "c\n",
      "\n",
      "d\n",
      "\n",
      "e\n",
      "\n",
      "f\n",
      "\n",
      "*\n",
      "\n",
      "**\n",
      "\n",
      "\n",
      "Fauna\n",
      "\n",
      "We identified 9,633 arthropods representing 99 families (Table S1) and 239 morphospecies; 63% of the families occurred in three or more samples. Diptera (flies) and Hemiptera (true bugs and leafhoppers) were dominant across all site categories; the most abundant families included ephydrid ( = 39.2/50 sweeps, SE  = 12.5), muscid ( = 22.2, SE  = 3.0), and anthomyiid ( = 13.0, SE  = 1.9) flies and cicadellid ( = 28.6, SE  = 4.9) and delphacid ( = 11.4, SE  = 3.0) leafhoppers. Family and morphospecies richness were highest for Diptera (37 and 92, respectively), followed by Hymenoptera (wasps, 17, 64), Hemiptera (14, 37), and Coleoptera (beetles, 12, 18; Table S1). Cicadellids and braconid and pteromalid wasps were the families with the largest number of morphospecies (21, 17, and 17, respectively). Relatively motile taxa represented 88% of the assemblage, and there were about twice as many herbivores as predators (Table 2).\n",
      "\n",
      "Table 2\n",
      "Early\tMid\tANCOVA\n",
      "Control\tGrazed\tControl\tGrazed\tT\tS\tY\tTxS\tTxY\tSxY\tB\n",
      "Total individuals\t'10\t120 (32)\t138 (47)\t121 (23)\t215 (114)\n",
      "'11\t146 (15)\t190 (58)\t148 (32)\t106 (36)\n",
      "Family richness\t'10\t15.4 (2.3)\t15.6 (1.9)\t15.8 (2.1)\t15.1 (2.1)\t**\n",
      "'11\t23.1 (0.89)\t21.8 (2.2)\t21.4 (2.1)\t19.2 (2.7)\n",
      "Morphospecies richness\t'10\t21.6 (3.3)\t21.3 (3.3)\t22.1 (3.6)\t21.1 (3.7)\t**\n",
      "'11\t31.1 (1.2)\t30.8 (3.5)\t33.7 (4.2)\t26.8 (5.3)\n",
      "Expected number of morphospecies\t'10\t5.27 (0.29)\t5.01 (0.26)\t4.49 (0.51)\t4.29 (0.82)\t**\t*\n",
      "'11\t5.54 (0.33)\t5.97 (0.12)\t6.15 (0.26)\t6.11 (0.31)\n",
      "Probability of interspecific encounter\t'10\t0.827 (0.026)\t0.805 (0.022)\t0.676 (0.083)\t0.615 (0.12)\t**\t*\n",
      "'11\t0.833 (0.039)\t0.892 (0.0073)\t0.890 (0.20)\t0.885 (0.021)\n",
      "% Family dominance\t'10\t34.0 (3.7)\t37.3 (4.0)\t48.4 (8.1)\t54.9 (11)\t**\n",
      "'11\t34.0 (5.3)\t27.7 (3.1)\t30.6 (3.4)\t26.3 (4.0)\n",
      "% Species dominance\t'10\t32.3 (3.4)\t34.5 (3.5)\t47.8 (8.2)\t53.7 (11)\t**\t*\n",
      "'11\t31.7 (5.5)\t23.5 (2.5)\t23.1 (4.2)\t24.6 (4.3)\n",
      "% Predators\t'10\t7.26 (1.3)\t6.43 (1.2)\t10.3 (2.2)\t9.23 (3.5)\t**\n",
      "'11\t9.54 (1.5)\t10.9 (2.2)\t12.2 (1.5)\t15.8 (3.3)\n",
      "% Herbivores\t'10\t29.6 (6.0)\t30.0 (8.8)\t10.5 (3.2)\t8.56 (2.7)\t**\n",
      "'11\t20.2 (4.4)\t26.4 (6.4)\t33.7 (4.4)\t47.2 (7.5)\n",
      "% More motile taxa\t'10\t87.8 (4.4)\t91.2 (3.2)\t94.3 (1.5)\t92.0 (3.4)\n",
      "'11\t87.4 (2.6)\t88.4 (4.1)\t79.9 (3.7)\t87.4 (3.8)\n",
      "% Less motile taxa\t'10\t12.1 (4.3)\t8.76 (3.2)\t5.67 (1.5)\t8.00 (3.4)\t*\n",
      "'11\t12.5 (2.6)\t11.6 (4.1)\t20.1 (3.8)\t12.6 (3.8)\n",
      "Baits: Formicidae\t'10\t0.800 (0.25)\t0.875 (0.23)\t0.200 (0.13)\t0.714 (0.18)\t**\t**\n",
      "species richness\t'11\t0.600 (0.22)\t1.00 (0.37)\t0.200 (0.13)\t0.600 (0.40)\n",
      "\n",
      "a\n",
      "\n",
      "b\n",
      "\n",
      "c\n",
      "\n",
      "d\n",
      "\n",
      "e\n",
      "\n",
      "f\n",
      "\n",
      "*\n",
      "\n",
      "**\n",
      "\n",
      "In contrast to vegetation and related variables, fauna showed limited response to both grazing and season at the assemblage level (Table 2, Figs. 3, 4). There were no significant results among assemblage variables for Treatment, and there was not a directional trend across metrics (p = 0.41; two-tailed sign test). Further, there were no significant interactions with either Season or Year. When mid-season plots were examined in isolation, there was a weak trend (p = 0.049) of more depauperate assemblage characteristics on grazed plots among canopy fauna (i.e., bait variable excluded; otherwise p = 0.096). Season effects were only present for ant species richness, but there were some Season x Year interactions, particularly for percentage of herbivores (Table 1, Fig. 4). Year effects, however, occurred across almost all parameters: diversity was higher in 2011 than in 2010 (Table 1, Fig. 3). Year effects were most striking for species dominance at mid-season; 2011 dominance was less than half that of 2010. Block effects, again in contrast to vegetation results, were only present for ant species richness.\n",
      "\n",
      "10.1371/journal.pone.0054109.g003Figure 3 Means (SEs) for families, morphospecies, and expected number of morphospecies by grazing treatment and season.See Table 2 for test results.\n",
      "\n",
      "10.1371/journal.pone.0054109.g004Figure 4 Means (SEs) for predators, herbivores, and more- and less-motile taxa by grazing treatment and season.See Table 2 for test results.\n",
      "\n",
      "There were more seasonal effects at the order and family level, but there were again few grazing effects apparent (Table S2). Three significant Treatment effects were present: for cicadellid leafhoppers, muscid flies, and spiders, and all three taxa had fewer representatives on grazed wetlands than on the paired control wetlands. There was no Treatment trend across all collected families for either early season (p = 0.90; two-tailed sign test; Table S1) or mid-season (p = 0.20). Although there was similarly no Treatment trend across morphospecies in early season (p = 0.76), there was a trend of lower abundance across all morphospecies on grazed plots in mid-season (p = 0.0023). There were again no Treatment x Season interactions and only three significant Treatment x Year interactions without consistent directionality (Table S2). Significant Season effects generally indicated higher abundances in early season. There were numerous Season x Year interactions as well, which were strongest for ephydrid shore flies: abundances were fourteen times greater in mid-season than in early season in 2010, but twice as high in early season as in mid-season in 2011. We collected 85 total families in early season versus 75 in mid-season. Hemiptera generally had highest abundances in early season of 2010 and mid-season of 2011, whereas Diptera showed the inverse relationship (Table S2). Almost half of the abundant taxa showed Year effects, and all of these taxa had higher abundances in 2011, with the exception of bait Formicidae (ants) and the most abundant ant species, Myrmica discontinua Weber. Aphids showed the strongest year effects: 28-fold higher abundance in 2011 than in 2010. There were block effects only for leafhoppers and bait Formicidae (Table S2).\n",
      "\n",
      "There was little clustering by different Treatment and Season combinations for family data (Fig. S1). The initial MRPP randomization test using study factors (T = −2.64, p = 0.0083) did indicate that some treatment combinations were distinct compositionally, but the low within-group agreement value, or effect size (A = 0.045), also suggested substantial variation within each combination. The only significant pairwise comparisons were Early Control vs. Mid Control (p = 0.020) and Early Control vs. Mid Grazed (p = 0.0016). The second MRPP using the new group membership variable generated from cluster analysis resulted in a lower p value (T = −9.79, p<0.0001) and higher A ( = 0.24), and all pairwise comparisons were significant (p<0.024) emphasizing the potential importance of other influences in addition to Treatment and Season. The overall dispersion analysis did not yield significant results. Results were consistent, whether interpreted on the basis of deviations from centroids or from spatial medians (in each case from both ANOVA tables and permutation of least squares residuals); p-values ranged from 0.86 to 0.90. None of the pairwise tests for differences in dispersion among groups were significant (0.42<p<0.92). The non-significant dispersion results in combination with the significant MRPP results, albeit with low effect size, suggest that assemblages did differ compositionally at the family level rather than in variability/dispersion.\n",
      "\n",
      "There was similarly little clustering at the morphospecies level (Fig. S2). As with family data, the initial MRPP based on morphospecies was significant (T = −3.72, p = 0.00043), but effect size was again low (A = 0.069). Three pairwise comparisons were significant: Early vs Mid Control (p = 0.00019), Early Control vs. Mid Grazed (p = 0.0060), and Early Grazed vs. Mid Control (p = 0.0010). The strongest MRPP results in the study were obtained for morphospecies data in combination with the new group membership variable written from the cluster analyses (T = −11.9, p<0.0001), although effect size was the same as for families (A = 0.24). All of the ensuing pairwise comparisons were significant (p<0.0017). There were no significant results from the overall dispersion analyses (0.12<p<0.23), regardless of derivation from centroids or spatial medians (or ANOVA tables or permutation of residuals), and there were no significant pairwise results (0.13<p<0.86).\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "There were moderate long-term grazing effects on subalpine wetland vegetation, but few negative indirect effects on fauna, and these relationships changed only subtly at mid-season. Although long-term vs. short-term differences in grazing effects have been documented elsewhere [1], Treatment x Season interactions were in fact the least common of all the examined sources of variance in our study (only one significant interaction). The only negative effects on fauna were: three Treatment differences for individual taxa (of 32 tests) that indicated lower numbers on grazed plots, weakly-significant lower diversity on mid-season grazed sites across canopy assemblage metrics, and a significant trend of lower abundances on mid-season grazed sites across all morphospecies. These effects, though certainly minimal, do differ somewhat from results of the early season-only study that detected no effects of grazing on fauna, the exceptions being a small number of positive effects [12]. Nevertheless, it seems clear that the lack of long-term effects was not obscuring major impacts on either flora or fauna manifesting at mid-season, thus corroborating the overall conclusion of the earlier study: little negative impact of grazing on fauna.\n",
      "\n",
      "The subalpine vegetation metrics negatively influenced by grazing disturbance in our study have often been shown to be affected in other, more intensively grazed, environments [53], [67]–[69]. Effects on bare ground and litter are sufficiently established that these characteristics have been used as proxies for grazing intensity and/or as manipulated variables used in experimental assessments of grazing pressure [52], [53], [70]. Litter removal can cause shifts in soil moisture [70] which could, along with direct pressure from trampling, mediate some of the observed increases in soil strength (in the form of compaction) on our sites. The significant interaction for senescent, brown cover (less common on grazed than on control sites in early season but more common on grazed sites in mid-season) may have been driven by a) breakage of the less flexible standing senescent vegetation that dominates these wet meadows in late season [33] with effects that remained through the ensuing winter and into the next growing season (see also [1], [12]), and b) mid-season impacts, such as urine patches, that may temporarily increase standing dead vegetation [71], [72] in otherwise non-senescent wetlands.\n",
      "\n",
      "Wetland arthropods are often sensitive to disturbance [73], and livestock grazing can negatively affect arthropods [24], [26], [74], so sufficiently strong grazing pressure would have been expected to produce effects on this arthropod assemblage. Grazing can affect arthropods indirectly via effects on litter and structural diversity [69], [75], [76], such as we observed in our grazed wetlands. Because of such effects on vegetation, herbivores [75], [77], particularly leafhoppers, can be sensitive to livestock grazing ([27]; but see [74]), and indeed these abundant animals were the group most affected by grazing in our study. Grazing can alternatively have neutral or positive indirect effects on arthropods [50], [67], [78] despite negative effects on vegetation [12], [68]. The minor negative effects of grazing on fauna in the current study indicate that pack stock disturbance in Sequoia does not fall to either extreme of this spectrum of effects, and the subtle indications of additional mid-season grazing effects most closely align with studies that have found only limited divergence between long- and short-term grazing effects on ecosystems [2]–[4].\n",
      "\n",
      "A variety of factors may have contributed to both the lack of overall effects of grazing disturbance on fauna and the minor additional grazing effect apparent during the growing season. Stock usage of these wetlands is relatively low and is prohibited during the most vulnerable period immediately following snowmelt ([12]; see also [33]). Stock usage was further reduced during the two study years as a result of late wetland openings due to heavy preceding winters. The early-season wetland closures in the Park made the study possible by allowing examination of early-season effects in isolation, but earlier and more intensive stock use in other managed areas would likely produce greater effects. Sierran wet meadows may also be resistant to grazing as a result of decades-long use (see also [16], [33]), possibly reducing, but clearly not eliminating, effects on vegetation structure. Resistance to pack stock may further vary as a function of vegetation assemblage [18], [33]; the reed grass-dominated assemblage, frequently used by packers and emphasized in our study, may tolerate stock better than wetter vegetation assemblages. Faunal differences as a function of grazing may be mitigated by frequent movement of arthropods among habitats ([21]–[23], c.f. [79]) facilitated by permeable micro-landscape boundaries [80]–[82] between grazed and ungrazed habitat, particularly for the motile taxa that dominated the assemblage. A number of taxa have life history stages in other habitats, for instance the subalpine streams typically bisecting these wetlands [83], that may function as refugia from terrestrial disturbance.\n",
      "\n",
      "Sampling grain can also mediate detection of livestock grazing effects [34], [84]–[86]. Our study was cast at the landscape scale, i.e., the scale of management interest, and the experimental units in our study were entire wetlands [3]. Cropping in our study wetlands was patchy, as might be expected with low stock densities [85], and cropping intensity varied across our randomly selected subsampling areas. Although we believe that our work captured grazing effects at the wetland and landscape scale, greater effects may have been present in small, intensively-grazed patches (see also [34], [85]).\n",
      "\n",
      "Season and Year had approximately equal influence on overall vegetation structure in these high elevation wetlands, but inter-annual variability was greater than seasonal variability for fauna, particularly at the assemblage level. There was greater faunal diversity and abundance in the particularly wet 2011 than in 2010, and this result is consistent with correlations between these faunal metrics and snow water equivalent over several years in both the Sierra Nevada and White Mountains of California (Holmquist and Schmidt-Gengenbach, unpublished meeting abstract). Clearly, idiosyncrasies of the year or years selected for study can influence conclusions regarding assemblage structure and response to grazing disturbance (see also [3]), as has been demonstrated in other ecosystems (e.g., [6]). Although the early/mid-season differences were of secondary importance for fauna, diversity and abundance decline by about a factor of four with wetland senescence during late season (Holmquist and Schmidt-Gengenbach, unpublished report). The trend of higher abundances for individual taxa in early season may be a function of early season hatching/emergence, or may suggest that soil moisture or the initial arrival of warmer weather is more important than the more fully developed vegetation structure found in mid-season, but our study was not designed to establish mechanisms for seasonal differences. The numerous Season x Year interactions were also likely related to the divergent snow years, and canopy height, green cover, and a number of faunal assemblage and family metrics showed congruent responses.\n",
      "\n",
      "It appears that current management of pack stock in the Park has produced moderate negative effects on coarse vegetation structure, but only minimal effects on the arthropod assemblage. This study, however, did capture some minor grazing effects at mid-season that were not apparent from early season sampling that targeted persisting effects only [12]. These results, coupled with the moderate disturbance to vegetation, raise the concern that increases in stock usage of these wetlands could cause more significant impact. Conversely, current Park maintenance of low stock densities and attention to wetland soil moisture and plant phenology in determining opening dates likely combine to prevent greater impact, and other land management agencies might benefit from application of these standards. The results of the current study should assist with the resolution of hiker-equestrian conflict associated with pack stock use [32], [35], [87] by demonstrating the complexity of ecosystem response to this disturbance: current stock use appears to be neither completely benign nor vastly destructive. Reliance on either vegetation or faunal metrics exclusively would have yielded different overall conclusions, whereas examining effects though the lenses of both flora and fauna provided a more integrative and nuanced view of ecosystem response.\n",
      "\n",
      "\n",
      "Supporting InformationFigure S1\n",
      "\n",
      "\n",
      "Agglomerative cluster analysis of site family data with overlay by grazing treatment and season.\n",
      "\n",
      "(PPT)\n",
      "\n",
      "Click here for additional data file.Figure S2\n",
      "\n",
      "\n",
      "Agglomerative cluster analysis of site morphospecies data with overlay by grazing treatment and season.\n",
      "\n",
      "(PPT)\n",
      "\n",
      "Click here for additional data file.Table S1\n",
      "\n",
      "\n",
      "Mean relative abundance/50 sweeps (standard error), both years combined, as a function of grazing and season; zeros are omitted for clarity.\n",
      "\n",
      "(DOC)\n",
      "\n",
      "Click here for additional data file.Table S2\n",
      "\n",
      "\n",
      "Abundance means (standard errors) for orders and ten most abundant families as a function of Treatment (Control, Grazed), Season (Early, Mid), and Year (2010, 2011) and results of 2×2×2 blocked ANCOVAs with elevation as a covariate.\n",
      "\n",
      "(DOC)\n",
      "\n",
      "Click here for additional data file.\n"
     ]
    }
   ],
   "source": [
    "view_one_article({ \"$match\": { \"text_matches\": { \"$in\": problematic_terms } } })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "3543354\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "None\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "None\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['study area']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Introduction\n",
      "\n",
      "The transfer of energy and recycling of nutrients via the decomposition of organic matter is a central unifying process that links all organisms to the functioning of ecosystems [1]. However, dead organic matter varies enormously in its spatial and temporal distribution [2], and this determines the magnitude of its contribution to nutrient cycling [3] and the diversity and dynamics of its consumers [4]. Animal carrion is the most nutrient-rich form of dead organic matter [5], and recent reviews have highlighted the overlooked role of carrion in food webs [6], [7], [8], and driving variation in biodiversity and ecological processes in landscapes [5], [9]. Widespread changes to the population dynamics of large vertebrates, through loss of top predators [10] or hunting and harvesting of wild herbivores [11], are affecting the distribution and input of carrion resources in some terrestrial ecosystems [6], [12]. This creates a strong imperative to establish a deeper understanding of the links between carrion, biodiversity and ecosystem functioning.\n",
      "\n",
      "Ephemeral resource patches, such as rotting logs and dung pads, occur in ecosystems worldwide and are focal points of nutrient cycling and aggregation of invertebrates [13], [14]. Carrion is distinct from many other ephemeral resources due to its faster decomposition rate, and its patchy spatial occurrence [9]. This means that carrion has very different spatial and temporal dynamic to other ephemeral resources, and suggests that carrion may have a disproportionate effect on biodiversity and nutrient cycling in ecosystems.\n",
      "\n",
      "The decomposition of carrion affects its surroundings through pathways to vertebrate and invertebrate scavengers and detritivores, and their predators, as well as through microbial and fungal decomposers and plants. This generates a temporally dynamic patch with high nutrient concentration and biological activity that contrasts strongly with the nearby environment [5], [15]. To date, studies on the effects of carrion on variation in biodiversity have been restricted to single groups of organisms, such as the detailed examination of successional patterns of arthropods [16], [17], and its localised effects on soil nitrogen or plant growth and composition [15], [18]. This has resulted in a detailed understanding of the specialised role that many insect groups have in carrion decomposition, including the critical role of fly larvae [19], [20], [21], and the complex food webs of associated predators and detritivores [16], [22], [23], [24]. Studies on soil and plant responses have demonstrated that the significant spike in soil nitrogen at a carcass can be detected in nearby plant tissue [15], [25], leading to changes in plant growth and composition [18], [26], and is likely moderated by soil microbial activity [5]. Key studies have also shown the importance of carrion for soil nitrogen and plant dynamics across whole ecosystems [27], [28]. Nevertheless, there are only a handful of comprehensive studies that have integrated multiple groups of organisms into the one study design [3], and there are fewer still that have specifically examined the functional traits of different species that respond to carrion. This leaves much to learn about what traits characterise carrion opportunists, and therefore what traits of species can help us understand the links between ephemeral carrion patches and its effect on variation in biodiversity within landscapes [9].\n",
      "\n",
      "We present data from an experiment designed to answer a number of questions about how ephemeral carrion resource patches generates variation in ecological communities. We did this by quantifying the responses of different aspects of the carrion food web, including components of the insect community, soil nutrients, and the plant community. We sought to identify the traits associated with insect responses to the patchiness of carrion, and the responses of plants to subsequent changes in soil nutrients. To do this, we first examined changes in assemblage diversity and composition in response to the experimental addition of animal carcasses. Second, we examined whether assemblage changes corresponded with particular combinations of traits, for both insects and plants. We predicted that insect responses would be associated with their dispersal ability, with specialist species having wing morphologies that enable the rapid location of new carrion resources. We also predicted that plant responses would be associated with their growth potential and ability to exploit elevated soil nutrient levels. Our experiment compared soil, plant and insect responses between paired carrion and control sites with spatial replication. The results of our analysis therefore provide new insight into the magnitude of the spatial contrast between carrion and its immediate surroundings. This allowed us to examine how the spatial contrast changed over time, and how carrion contributes to both spatial and temporal variation in biodiversity and ecological processes within a landscape.\n",
      "\n",
      "\n",
      "Methods\n",
      "\n",
      "\n",
      "Ethics Statement\n",
      "\n",
      "Insects were sampled under an ACT government ‘Licence to take’ (LT2010417).\n",
      "\n",
      "\n",
      "Study Area and Experimental Design\n",
      "\n",
      "In November 2010 (southern hemisphere spring), we added carcasses of kangaroos to 18 sites in temperate Eucalyptus woodland near Canberra, Australia (149°10′ East, 35°10′ South). This study area is part of the Mulligans Flat-Goorooyarroo Woodland Experiment, a long-term ecological restoration project [29]. Specific details on temperature, rainfall, floristic composition and soil types from this study area have been described by McIntyre et al [30]. We sourced carcasses of eighteen eastern grey kangaroos (Macropus giganteus) over a 10 day period from road kill across the city of Canberra, and positioned them at sites within 12 hours of death. Note that no animals were killed for the purposes of this study. Carcasses were only used when there was not significant blood loss. Average mass of the kangaroos was 31 kg (range 17–80 kg). Each carcass was paired with a control site 10 metres away to allow spatial comparisons between carcass and control treatments through time. We positioned each carcass (and control site) at the base a young Eucalyptus blakelyi tree (<2 m tall at beginning of experiment). Our study area has a history of livestock grazing, resulting in some sites being dominated by exotic grass species [30], [31], [32]. We placed half the carcasses in native-dominated grassland (Themeda sp. and Austrodanthonia sp.) and half in exotic-dominated grassland (Phalaris sp. and Holcus sp.).\n",
      "\n",
      "\n",
      "Soil and Plant Data Collection\n",
      "\n",
      "We took a 4 cm deep soil sample (approximately 4 cm diameter) using a trowel at each carcass and control site at one week (week 0) before carcass addition, and then 12 and 26 weeks after carcass addition. Samples were taken from near the centre of each carcass. Total N content of soil and leaf samples (see below) was quantified with high temperature combustion using a LECA2000 CNS analyser [33]. We performed two analyses of total N, and used the average in our calculations. The potential mineralisable soil nitrogen (organic nitrogen) was quantified using the Illinois soil nitrogen test for amino sugar [34].\n",
      "\n",
      "We took samples of between five and ten leaves from the tips of branches (i.e. active growth zones) in each E. blakelyi tree at carcass and control sites at weeks 12 and 26. No leaves were taken at week 0. Leaf nitrogen content was determined as per soil nitrogen (see above). A higher nitrogen content in leaves indicates potentially greater photosynthetic capacity, and therefore growth potential [35]. Each carcass was positioned on top of existing grasses and forbs, but these were killed off during the early stages of decay. We examined plant re-colonization of carcass patches after 52 weeks by recording the presence of grass and forb species inside a 40×40 cm quadrat at (i) the centre and (ii) edge of each carcass patch, and at (iii) the paired control site. The edge of each carcass was able to be delineated by estimation of the size of the original carcass, and the presence of a ‘halo’ of plant growth. In addition to these standardised quadrats, we also recorded the presence of all other plant species present in the remainder of a 40 cm deep perimeter around each carcass (the ‘surrounding’ vegetation) and the entire carcass patch to compile a complete inventory of grass and forb species. We collected data on plant life forms and specific leaf area (SLA) from the literature [36], [37], [38], and these are summarised in Appendix S2. SLA is the one-sided area of a fresh leaf divided by its dry mass (mm2.mg−1), and is indicative of potential relative growth rate [36]. SLA has also been shown not to vary significantly within plant species [37], thus justifying our use of data from the literature. We used the mean SLA value of each plant species presented in each published account, and excluded rare species for which data were unavailable.\n",
      "\n",
      "\n",
      "Insect Data Collection\n",
      "\n",
      "We sampled insects using a pair of pitfall traps at each carcass and control site. We chose ants and beetles because they are an abundant and diverse components of carrion arthropod communities [3], [16], [39], and yet differ strongly in their dispersal ability and resource use. We recognise that flies are an important part of carrion arthropod communities [16], but we wanted to focus on two diverse groups with contrasting dispersal ability. Many beetles are flight-capable and have high resource specialisation [40], [41], with some specialist species requiring carrion to lay eggs and complete their life cycle. In contrast, ants are generally restricted to the ground (after post-mating dispersal), and may opportunistically scavenge from carcasses or predate on the other insects present. For these reasons, we expected beetles and ants to show very different responses to carrion. Pitfall traps were 9 cm in diameter, and were half filled with glycol solution. The traps were opened for seven days during weeks 1, 6, 12 and 26 of the experiment, which represented distinct stages of decomposition of the carcasses (see Appendix S1). We pooled pitfall trap pairs, giving n = 36 at each sampling time. All insects were extracted from the pitfall traps, stored in 70% ethanol and sorted to order and family. We further sorted beetles and ants to genus and morphospecies using relevant keys [42], [43].\n",
      "\n",
      "\n",
      "Insect Morphological Measurements\n",
      "\n",
      "We measured body length and wing dimensions of each beetle species for comparison of mean trait values of species present at each carcass at control site. We used body length to estimate the body mass of each species of beetle using family-specific length versus weight power relationships for species of Scarabaeidae, Staphylinidae and Carabidae, and a general Coleoptera relationship for all remaining species [44]. We calculated wing loading [45] by dividing body mass by wing area (length x width), expressed as (mg/mm2). We calculated wing aspect ratio by dividing wing length by width, with higher ratios indicating longer and narrower wings [45]. Ant body size is correlated with microhabitat use and resource discovery rates [46], and we regarded this as a useful measure of ant dispersal ability. We measured ant head width and rear femur length, and multiplied these together to calculate a body size index for each species [47]. All morphological measurements were made on up to three individuals of each species (then averaged) using an ocular micrometer with magnification between 10× and 100×, and resolution between 0.1 and 0.01 mm, depending on the measurement and size of the insect. This approach was taken as we were interested in variation in traits between species rather than within species [48].\n",
      "\n",
      "\n",
      "Data Analysis\n",
      "\n",
      "We were first interested in quantifying the magnitude of the contrast between carcass and control sites. We used repeated measures linear mixed models in GenStat 14 [49] to test for interactive effects of carcass addition and sampling time on total insect abundance, species richness and evenness, as well as soil and plant nitrogen content. Carcass addition and time were treated as fixed factors, and site as a random factor. We used ANOVA, also in GenStat 14, to test for differences in insect and plant species richness and evenness or traits between carcass and control treatments at a particular sampling time, with blocking by carcass-control pair. Plant SLA was log10 transformed, and insect abundance data were log10(x+1) transformed to meet assumptions of normality and homogeneity of variances. Percentage similarity in plant species composition between the edge and centre of carcasses was expressed as the number of shared species divided by the total number (i.e. Jaccard similarity index). We compared similarity between the centre and edge of carcasses because plant recolonisation was more likely to occur from the edge rather than from more distant control sites. We tested for effects of quadrat position (carcass centre, edge or control site) and grassland type (native or exotic) on average SLA across plant species. For this analysis, we excluded rare species that occurred only in one quadrat. We wanted to test whether insect assemblages were more homogenous in composition at carcasses than at control sites due to the widespread dominance of carrion specialists. We used Permutational Analysis of Multivariate Dispersion [50] to test for differences in among-sample heterogeneity of insect species composition between carcass and control treatments, with statistical significance determined from 10 000 permutations of the data. This test calculates the average distance to the centroid of a group of samples projected in multivariate space, with a greater distance to centroid indicating greater among-sample heterogeneity [51]. Multivariate insect abundance data were square-root transformed prior to analysis to reduce the influence of highly abundant species on sample dissimilarity. Shannon evenness was defined as E = H′/lnS, where S is the number of species in a sample, and H′ = −∑ pi ln pi, where pi is the proportion of a species in a sample which is multiplied by the natural logarithm of itself [52].\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "\n",
      "Soil and Leaf Nitrogen\n",
      "\n",
      "The addition of carrion had significant effects on soil nitrogen (Table 1). The soils in our study environment were originally low in phosphorus and nitrogen compared with grasslands improved for grazing [31], and thus have low productivity potential. The highly concentrated nutrients in each carcass (a 30 kg vertebrate carcass contains approximately one kilogram of nitrogen [5]) produced a dynamic change in soil nitrogen. We found that total soil nitrogen beneath carcasses was 200% higher than control sites after 12 weeks, and 70% higher than control sites after 26 weeks of carrion being added (Fig. 1A). In addition, the concentration of the organic fraction of soil nitrogen was 460% higher than control sites after 12 weeks, and 290% higher after 26 weeks (Fig. 1B). This nutrient pulse also was detected in plant leaf nitrogen levels from nearby tree saplings, which were 18% higher in plants near to carcasses than control sites after 12 weeks (Fig. 1C). This demonstrates the cycling of nitrogen from carcasses into soil, and the uptake and assimilation of soil nitrogen into leaf tissue during the first few weeks of decomposition.\n",
      "\n",
      "10.1371/journal.pone.0053961.g001Figure 1 Effect of carrion on soil and foliar nitrogen.Carrion increased (a) soil total nitrogen, (b) soil organic nitrogen, and (c) tree foliar total nitrogen relative to control sites. Error bars show mean ± SE.\n",
      "\n",
      "Table 1\n",
      "Effects\tF\td.f.\tP\n",
      "Soil total nitrogen (%)\tTime\t55.85\t2\t<0.001\n",
      "Treatment\t29.36\t1\t<0.001\n",
      "Time×Treatment\t37.59\t2\t<0.001\n",
      "Soil organic nitrogen (µg/kg)\tTime\t136.31\t2\t<0.001\n",
      "Treatment\t26.68\t1\t<0.001\n",
      "Time×Treatment\t107.28\t2\t<0.001\n",
      "Tree foliar nitrogen (%)\tTime\t29.73\t1\t<0.001\n",
      "Treatment\t12.71\t1\t<0.001\n",
      "Time×Treatment\t0.26\t1\t0.609\n",
      "\n",
      "\n",
      "Plant Assemblage Richness and Composition\n",
      "\n",
      "Plant recolonisation of carcasses was faster in the exotic-dominated grassland where, after 52 weeks, the composition of plants in the centre of each carcass patch more closely matched the composition of species at the edge of each carcass patch in the exotic-dominated grassland (56% shared species) than in native-dominated grassland (37% shared species). We found a significant interaction between carcass treatment and grassland type (F2 = 11.48, P<0.001) and their effect on plant species richness at carcass patches after 52 weeks. Plant species richness was similar across carcass and control sites in the exotic grassland, but the number of species in the centre of carcass patches in native grassland was only a third that of species at the edge of carcass patches or at control sites (Fig. 2A). This result is further highlighted by our observation that four of the nine native-dominated carcass patches had no re-colonising plants at all, whereas all nine of the exotic-dominated carcass patches had re-colonising plants 52 week after carrion was added (compare Fig. 3A with 3B). The rapid response of plants in the exotic grassland was associated with a significantly higher specific leaf area among the constituent plant species (Fig. 2B, F1 = 18.03, P<0.001). Regardless of grassland type, plants recolonising the centre of carcass patches tended to be annuals rather perennials, and graminoids rather than forbs (Fig. 2C, 2D).\n",
      "\n",
      "10.1371/journal.pone.0053961.g002Figure 2 Differences in plant species richness and traits between carcass centre, carcass edge, and control sites.(a) Species richness of plants species in exotic grassland compared with native grassland. (b) Specific leaf area of plants in exotic grassland compared with native grassland. (c) Percentage of annuals versus perennials. (d) Percentage of graminoids versus forbs. Error bars show mean ± SE.\n",
      "\n",
      "10.1371/journal.pone.0053961.g003Figure 3 Recolonisation of plants 52 weeks after carrion addition.Very little recolonisation had occurred in the native-dominated grassland sites (a), whereas re-colonisation had progressed further in all exotic-dominated grassland sites (b), reflecting greater tolerance of some exotic species to extremely high nutrients. These examples indicate a different succession trajectory for native versus exotic grassland plant assemblages, with recolonisation occurring over different time scales.\n",
      "\n",
      "\n",
      "Insect Assemblage Diversity and Composition\n",
      "\n",
      "We sampled a total of 33 ant species (3330 individuals) and 119 beetle species (3614 individuals) from all carcass and control sites over the four sampling times (Appendix S3 and S4). Beetle and ant assemblages displayed contrasting dynamic responses to carrion. We found that beetle assemblages had a significantly higher species richness and lower evenness during early stages of decay, with a gradual return to levels similar to control sites by week 26 (Fig. 4A, Table 2). This contrasted strongly with ant assemblages, which initially had lower species richness at carcass sites, and did not show the same decrease in evenness as the beetle assemblage (Fig. 4B). We found an overall significant difference in among-sample heterogeneity from carcass and control sites at each sample times for both beetles (F = 21.11, P<0.001) and ants (F = 11.81, P<0.001). However, comparisons between carcass and control sites for each sampling time showed distinctive and contrasting patterns for the beetles and ants (Fig. 5). In the early weeks of decomposition, the composition of beetle assemblages at carrion sites was less variable than at control sites, indicating spatial monopolisation of carcass patches by a few abundant species (Fig. 5A). This changed as decay progressed, with the beetle assemblage becoming more heterogeneous at carrion sites relative to control sites. This was due to widespread species at carcasses becoming less dominant, while assemblages at control sites showed a seasonal decrease in heterogeneity (Fig. 5A). In contrast, there was never a difference in ant assemblage heterogeneity between carcass and control sites (Fig. 5B), with the overall significant difference among groups due a decrease in heterogeneity in week 26 in late autumn. This indicated there was no spatial monopolisation of carcasses by a few ant species, in contrast to that which we observed for beetles.\n",
      "\n",
      "10.1371/journal.pone.0053961.g004Figure 4 Effect of carrion on beetle and ant diversity.(a) Species richness and evenness of beetles at carcasses showed a trajectory through time clearly different from control sites at weeks 1, 6 and 12, with a return to similar levels at week 26 (blue arrows). (b) Species richness of ants at carcasses was half that at control sites during week 1 (red arrows), but was then similar to control sites at week 6, 12 and 26 of the experiment. Error bars show mean ± SE.\n",
      "\n",
      "10.1371/journal.pone.0053961.g005Figure 5 Effect of carrion on insect assemblage heterogeneity.Average distance to group centroid, reflecting heterogeneity in composition among samples of (a) beetles and (b) ants from carrion and control treatments. Error bars show mean ± SE.\n",
      "\n",
      "Table 2\n",
      "Effects\tF\td.f.\tP\n",
      "(A)\tAbundance\tTime\t48.81\t3\t<0.001\n",
      "Treatment\t3.25\t1\t0.080\n",
      "Time×Treatment\t4.55\t3\t0.009\n",
      "Species richness\tTime\t26.05\t3\t<0.001\n",
      "Treatment\t3.35\t1\t0.076\n",
      "Time×Treatment\t4.51\t3\t0.009\n",
      "Evenness\tTime\t44.38\t3\t<0.001\n",
      "Treatment\t1.08\t1\t0.322\n",
      "Time×Treatment\t1.41\t3\t0.264\n",
      "(B)\tAbundance\tTime\t40.09\t3\t<0.001\n",
      "Treatment\t230.92\t1\t<0.001\n",
      "Time×Treatment\t19.05\t3\t<0.001\n",
      "Species richness\tTime\t21.46\t3\t<0.001\n",
      "Treatment\t70.07\t1\t<0.001\n",
      "Time×Treatment\t11.51\t3\t<0.001\n",
      "Evenness\tTime\t50.49\t3\t<0.001\n",
      "Treatment\t28.86\t1\t<0.001\n",
      "Time×Treatment\t12.83\t3\t<0.001\n",
      "\n",
      "\n",
      "Insect Dispersal Traits\n",
      "\n",
      "The contrasting response of beetle and ant assemblages was associated with their different dispersal abilities, with significant differences in morphological traits across beetle species but not ant species (Table 3). Beetle species at carrion were larger than species at control sites one week after carrion was added (Fig. 6A), whereas ants showed no significant difference in body size at any of the four sampling times (Fig. 6B). Beetles at carcasses also had significantly higher wing loading, but only at one week after carrion was added (Fig. 6C). We also found that a higher percentage of beetles at carcasses had functional wings compared with beetles at control sites (Fig. 6D).\n",
      "\n",
      "10.1371/journal.pone.0053961.g006Figure 6 Differences in insect morphological traits between carcass and control sites.(a) Beetle species at carrion during week 1 were larger on average than those at control sites. (b) Beetle species at carrion during week 1 had higher wing loadings on average than beetle species at control sites. (c) No differences in ant body size were observed at different stages of carrion decomposition. (d) Macropterous beetles were always more prevalent at carcass sites that control sites. Error bars show mean ± SE.\n",
      "\n",
      "Table 3\n",
      "Effect\tF\td.f.\tP\n",
      "(A) Ant body size\tTime\t0.99\t3\t0.395\n",
      "Treatment\t0.09\t1\t0.769\n",
      "Time×Treatment\t0.62\t3\t0.600\n",
      "(B) Beetle body length\tTime\t12.42\t3\t<0.001\n",
      "Treatment\t39.37\t1\t<0.001\n",
      "Time×Treatment\t8.82\t3\t<0.001\n",
      "(C) Beetle wing loading\tTime\t9.25\t3\t<0.001\n",
      "Treatment\t1.81\t1\t0.179\n",
      "Time×Treatment\t1.64\t3\t0.180\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "Our results show that carrion in ecosystems represents a dynamic driver of species richness and nutrient cycling. This is supported by our observations, which showed elevated nitrogen in soil and plant leaf tissue at carrion patches, different rates of recolonization by plants, and a diverse and functionally distinct set of insects occurring at carcasses compared with control sites. Importantly, we have shown that the pattern of response by insects and plants was associated with particular functional traits that allow them to respond to and then exploit carrion. This builds on previous work that has demonstrated separately the effects of carrion on soil nutrient heterogeneity [53], scavenger community structure [16], [54], and plant composition [18], and develops further our understanding of how carrion generates spatial and temporal variation in ecological communities.\n",
      "\n",
      "We found that the nitrogen spike in soils translated into an increase in leaf nitrogen in the nearby E. blakelyi trees, indicating the movement of nutrients from the decomposing carcasses into live plant tissue via the soil subsystem [15], [25]. Trees in the genus Eucalyptus are not nitrogen-fixing and must use available nitrogen in the soil. This suggests that the increase in soil nitrogen due to carcass decomposition was actively taken up by the trees we sampled. The differences we detected in foliage nitrogen between carcass and control sites was smaller than that reported from analogous experimental studies in North America [15], [28], but this may be due to the very different plant species involved, as well as underlying microbial and soil properties. The dramatic change in soil nitrogen also resulted in a greater dominance of grass species with high average specific leaf area. It is well established that these functional plant types typically respond to elevated nitrogen [31]. This reflects observations in a tallgrass prairie ecosystem of North America, where forbs decreased in cover at the centre of ungulate carcasses after 1 year, but grasses increased in cover [18]. This is evidence of in situ organisms displaying trait-dependent responses to new nutrient-rich resource patches. The effects of carcasses on the diversity and composition of ground-layer plant assemblages depended on whether the carcasses were placed in native- or exotic-dominated grassland. Plants at the carcass treatment sites were initially killed by the carcasses due to prolonged shading by carcass remains and rapid soil chemistry changes, but later recolonised from the edges. The quick recolonization of carrion patches by plants in the exotic grassland was largely due to seeding, except for one species that used stolons (Cynodon dactylon). This underlines a context-dependent effect of carrion not previously described at local scales. This parallels a pattern observed for the effects of migrating salmon (Oncorhynchus spp.) carcasses on plants in North America, with effects reduced in the more highly productive catchments [27]. Together the results suggest a general principle of context-dependency resulting in temporally attenuated effects of carrion in high productivity ecosystems or in habitats with a high proportion of nutrient-responsive plant species.\n",
      "\n",
      "We showed that dispersal traits of insects were associated with their responses to ephemeral resource opportunities. The distinct dispersal traits of beetles compared with ants was likely to explain, in part, their contrasting dynamics in assemblage composition, and led to their changes in spatial heterogeneity during different stages of carcass decomposition. Dispersal plays a key role in insect spatial diversity patterns [55], and our study provides a further example relevant to carrion. Although carrion affected the relative abundance of ants present in the landscape, it did not affect species richness or spatial heterogeneity. The lack of flight to facilitate rapid foraging means that ants are generally less able to move as rapidly as flight-capable beetles, regardless of any resource specialisation they might have. Although we found that some ant species dominated individual carcasses, such as the meat ant Iridomyrmex purpureus, there were no species that monopolised all the carcasses in the same way some beetle species did. This result is due, in part, to the lack of strong specialisation of ants on carrion resources, with scavenging more likely to be opportunistic. For beetles, capacity for flight and higher wing-aspect ratios were associated with beetles at carrion generally. A larger body size and higher wing loading were important functional traits that were associated with the early colonisation of fresh carrion and spatial monopolisation of this resource. Beetles visiting fresh carcasses were typically equipped with wings that enable manoeuvrable flight [45], suggesting that they can actively seek carrion, and with their large body size have the metabolic capacity to do so over greater distances than smaller beetles. These traits are exemplified by the carrion specialists Ptomaphila lacrysoma and Creophilus erythrocephalus, which are well known to frequent carrion in south-eastern Australia [56], [57]. These species, as well as several others, were not recorded at control sites (Appendix S4), and have not been recorded as part of previous research in the study area [48]. This highlights the unique composition of carrion insect assemblages. Ants at carrion displayed no characteristic morphologies during different stages of carrion decay, and this matched their lack of difference in spatial heterogeneity between carcasses or control sites. Although we focused on morphological traits related to dispersal ability, it is important to point out that the feeding traits of insects are equally important in determining their responses to carrion [16]. For example, the carrion specialists mentioned above also feed on the flesh of carcasses or prey on the insects that do. Their early arrival at a carcass is therefore critical if they are to exploit a resource that will quickly be exhausted. Further research is needed to test if dispersal limitation or small body size actually prevents some species from exploiting carrion, as this would provide a mechanistic link between morphology and assemblage responses to carrion resources.\n",
      "\n",
      "Together, our findings provide an integrated example of how traits of opportunist species in two different taxonomic groups are associated with their ability to exploit patchy and dynamic resources. This adds to the literature on carrion resource specialisation traits and temporal assemblage dynamics [16], [23], to give a fuller picture of the processes that drive spatial changes in species composition. It also supports previous claims that carrion might need to be managed in some ecosystems to maintenance of biodiversity and ecological processes [9], [58]. This includes, for example, the role of carrion in the persistence of dependent insects [59] or the nutrient opportunities it can provide to plants [15]. Our study indicates that appropriate consideration of spatial and temporal continuity in carrion will enable insects of variable dispersal ability to exploit this resource, and will promote spatial heterogeneity in nutrient cycling and plant communities.\n",
      "\n",
      "\n",
      "Supporting InformationAppendix S1\n",
      "\n",
      "Stages of decomposition when insects were sampled.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.Appendix S2\n",
      "\n",
      "Summary of plant species and their traits.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.Appendix S3\n",
      "\n",
      "Summary of ant species.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.Appendix S4\n",
      "\n",
      "Summary of beetle species.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.\n"
     ]
    }
   ],
   "source": [
    "# To sample an article with text matches from our terms:\n",
    "art = articles.aggregate([\n",
    "    { \"$match\": { \"text_matches\": { \"$in\": terms } } },\n",
    "    { \"$sample\": { \"size\": 1 } }\n",
    "]).next()\n",
    "\n",
    "# Print the ID, metadata, matches, and extracted text of your article\n",
    "print(\"\"\"\n",
    "PMC ID\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TITLE\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "METADATA\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TEXT MATCHES\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "EXTRACTED TEXT\n",
    "\n",
    "{}\"\"\".format(art.get(\"_id\"),\n",
    "             art.get(\"article_title\"),\n",
    "             art.get(\"article_meta\"),\n",
    "             art.get(\"text_matches\"),\n",
    "             art.get(\"extracted_text\").strip()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "soup = BeautifulSoup(art[\"xml\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3573865'"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "soup.front.find(\"article-id\", attrs={\"pub-id-type\": \"pmc\"}).get_text()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'research-article'"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.article_type()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'PLoS ONE'"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.soup.find(\"front\").find(\"journal-meta\").find(\"journal-title\").get_text()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Methods in Ecology and Evolution'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.journal_title()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Advances in multiplex PCR: balancing primer efficiencies and improving detection success'"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.article_title()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DataFrame scratch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "foo = pd.DataFrame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>one</th>\n",
       "      <th>two</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>dog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   one  two\n",
       "0   10  dog"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame({\"one\": [10], \"two\": [\"dog\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>one</th>\n",
       "      <th>two</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>dog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   one  two\n",
       "0   10  dog"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame.from_records([{\"one\": 10, \"two\": \"dog\"}])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   A\n",
       "0  0\n",
       "1  1\n",
       "2  2\n",
       "3  3\n",
       "4  4"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.concat([pd.DataFrame([i], columns=['A']) for i in range(5)],ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.concat([foo])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Count article types and other metadata"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO: Check if an article's `_id` is in `result_set` and add those to a different default_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "211"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.count_documents({})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-15-ff577f9bb7c5>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcursor\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marticles\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maggregate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m{\u001b[0m \u001b[0;34m\"$sample\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m{\u001b[0m \u001b[0;34m\"size\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0marticles\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcount_documents\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m}\u001b[0m \u001b[0;34m}\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/collection.py\u001b[0m in \u001b[0;36mcount_documents\u001b[0;34m(self, filter, session, **kwargs)\u001b[0m\n\u001b[1;32m   1683\u001b[0m         \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_socket_for_reads\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msession\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0msock_info\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mslave_ok\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1684\u001b[0m             result = self._aggregate_one_result(\n\u001b[0;32m-> 1685\u001b[0;31m                 sock_info, slave_ok, cmd, collation, session)\n\u001b[0m\u001b[1;32m   1686\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1687\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/collection.py\u001b[0m in \u001b[0;36m_aggregate_one_result\u001b[0;34m(self, sock_info, slave_ok, cmd, collation, session)\u001b[0m\n\u001b[1;32m   1585\u001b[0m             \u001b[0mread_concern\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_concern\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1586\u001b[0m             \u001b[0mcollation\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcollation\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1587\u001b[0;31m             session=session)\n\u001b[0m\u001b[1;32m   1588\u001b[0m         \u001b[0mbatch\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'cursor'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'firstBatch'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1589\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mbatch\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/collection.py\u001b[0m in \u001b[0;36m_command\u001b[0;34m(self, sock_info, command, slave_ok, read_preference, codec_options, check, allowable_errors, read_concern, write_concern, collation, session, retryable_write)\u001b[0m\n\u001b[1;32m    242\u001b[0m                 \u001b[0msession\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    243\u001b[0m                 \u001b[0mclient\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__database\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclient\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 244\u001b[0;31m                 retryable_write=retryable_write)\n\u001b[0m\u001b[1;32m    245\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    246\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__create\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moptions\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcollation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/pool.py\u001b[0m in \u001b[0;36mcommand\u001b[0;34m(self, dbname, spec, slave_ok, read_preference, codec_options, check, allowable_errors, check_keys, read_concern, write_concern, parse_write_concern_error, collation, session, client, retryable_write, publish_events)\u001b[0m\n\u001b[1;32m    582\u001b[0m         \u001b[0;31m# Catch socket.error, KeyboardInterrupt, etc. and close ourselves.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    583\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mBaseException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 584\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_raise_connection_failure\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merror\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    585\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    586\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0msend_message\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmessage\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmax_doc_size\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/pool.py\u001b[0m in \u001b[0;36m_raise_connection_failure\u001b[0;34m(self, error)\u001b[0m\n\u001b[1;32m    743\u001b[0m             \u001b[0m_raise_connection_failure\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maddress\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    744\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 745\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    746\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    747\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__eq__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/pool.py\u001b[0m in \u001b[0;36mcommand\u001b[0;34m(self, dbname, spec, slave_ok, read_preference, codec_options, check, allowable_errors, check_keys, read_concern, write_concern, parse_write_concern_error, collation, session, client, retryable_write, publish_events)\u001b[0m\n\u001b[1;32m    577\u001b[0m                            \u001b[0mcompression_ctx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompression_context\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    578\u001b[0m                            \u001b[0muse_op_msg\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mop_msg_enabled\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 579\u001b[0;31m                            unacknowledged=unacknowledged)\n\u001b[0m\u001b[1;32m    580\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mOperationFailure\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    581\u001b[0m             \u001b[0;32mraise\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/network.py\u001b[0m in \u001b[0;36mcommand\u001b[0;34m(sock, dbname, spec, slave_ok, is_mongos, read_preference, codec_options, session, client, check, allowable_errors, address, check_keys, listeners, max_bson_size, read_concern, parse_write_concern_error, collation, compression_ctx, use_op_msg, unacknowledged)\u001b[0m\n\u001b[1;32m    139\u001b[0m             \u001b[0mresponse_doc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m\"ok\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    140\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 141\u001b[0;31m             \u001b[0mreply\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mreceive_message\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msock\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    142\u001b[0m             \u001b[0munpacked_docs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mreply\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munpack_response\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcodec_options\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcodec_options\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/network.py\u001b[0m in \u001b[0;36mreceive_message\u001b[0;34m(sock, request_id, max_message_size)\u001b[0m\n\u001b[1;32m    171\u001b[0m     \u001b[0;31m# Ignore the response's request id.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    172\u001b[0m     length, _, response_to, op_code = _UNPACK_HEADER(\n\u001b[0;32m--> 173\u001b[0;31m         _receive_data_on_socket(sock, 16))\n\u001b[0m\u001b[1;32m    174\u001b[0m     \u001b[0;31m# No request_id for exhaust cursor \"getMore\".\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    175\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mrequest_id\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/network.py\u001b[0m in \u001b[0;36m_receive_data_on_socket\u001b[0;34m(sock, length)\u001b[0m\n\u001b[1;32m    230\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0mbytes_read\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0mlength\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    231\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 232\u001b[0;31m                 \u001b[0mchunk_length\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmv\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mbytes_read\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    233\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mIOError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mOSError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mexc\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    234\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0m_errno_from_exception\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexc\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0merrno\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mEINTR\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "cursor = articles.aggregate([{ \"$sample\": { \"size\": articles.count_documents({}) } }])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "count = articles.count_documents({})\n",
    "cursor = articles.find({})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[F\u001b[KProcessed 175 articles (82.9%) in 0m8s; about 0m2s left.\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-46-1f74dba360bf>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0mrow\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m     \u001b[0marticle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mArticle\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"xml\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     10\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m     \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"id\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"_id\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Dropbox (EHA)/repositories/pubcrawler/pubcrawler/article.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, xml)\u001b[0m\n\u001b[1;32m     13\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mxml\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mxml\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mxml\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 15\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msoup\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mBeautifulSoup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mxml\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'lxml-xml'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     16\u001b[0m         \u001b[0;31m# self._id = article_dict['_id']\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     17\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, markup, features, builder, parse_only, from_encoding, exclude_encodings, **kwargs)\u001b[0m\n\u001b[1;32m    279\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    280\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 281\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_feed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    282\u001b[0m                 \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    283\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mParserRejectedMarkup\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36m_feed\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    340\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuilder\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    341\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 342\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuilder\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmarkup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    343\u001b[0m         \u001b[0;31m# Close out any unfinished strings and close all the open tags.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    344\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mendData\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/builder/_lxml.py\u001b[0m in \u001b[0;36mfeed\u001b[0;34m(self, markup)\u001b[0m\n\u001b[1;32m    166\u001b[0m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmarkup\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mCHUNK_SIZE\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    167\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 168\u001b[0;31m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    169\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    170\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mUnicodeDecodeError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mLookupError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0metree\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mParserError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parser.pxi\u001b[0m in \u001b[0;36mlxml.etree._FeedParser.feed\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parser.pxi\u001b[0m in \u001b[0;36mlxml.etree._FeedParser.feed\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._TargetParserContext._handleParseResult\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._TargetParserContext._handleParseResult\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/etree.pyx\u001b[0m in \u001b[0;36mlxml.etree._ExceptionContext._raise_if_stored\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/saxparser.pxi\u001b[0m in \u001b[0;36mlxml.etree._handleSaxTargetStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/saxparser.pxi\u001b[0m in \u001b[0;36mlxml.etree._callTargetSaxStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._PythonSaxParserTarget._handleSaxStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/builder/_lxml.py\u001b[0m in \u001b[0;36mstart\u001b[0;34m(self, name, attrs, nsmap)\u001b[0m\n\u001b[1;32m    218\u001b[0m         \u001b[0mnamespace\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getNsTag\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    219\u001b[0m         \u001b[0mnsprefix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_prefix_for_namespace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnamespace\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 220\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msoup\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhandle_starttag\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnamespace\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnsprefix\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mattrs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    221\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    222\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_prefix_for_namespace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnamespace\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36mhandle_starttag\u001b[0;34m(self, name, namespace, nsprefix, attrs)\u001b[0m\n\u001b[1;32m    528\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    529\u001b[0m         tag = Tag(self, self.builder, name, namespace, nsprefix, attrs,\n\u001b[0;32m--> 530\u001b[0;31m                   self.currentTag, self._most_recent_element)\n\u001b[0m\u001b[1;32m    531\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mtag\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    532\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mtag\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/element.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, parser, builder, name, namespace, prefix, attrs, parent, previous, is_xml)\u001b[0m\n\u001b[1;32m    859\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mattrs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mattrs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    860\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcontents\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 861\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mparent\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprevious\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    862\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhidden\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    863\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/element.py\u001b[0m in \u001b[0;36msetup\u001b[0;34m(self, parent, previous_element, next_element, previous_sibling, next_sibling)\u001b[0m\n\u001b[1;32m    260\u001b[0m         \"\"\"Sets up the initial relations between this element and\n\u001b[1;32m    261\u001b[0m         other elements.\"\"\"\n\u001b[0;32m--> 262\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparent\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparent\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    263\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    264\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprevious_element\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprevious_element\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "reporter = Reporter(25, len(matches))\n",
    "\n",
    "rows = []\n",
    "\n",
    "for idx, record in enumerate(matches):\n",
    "    reporter.report(idx)\n",
    "\n",
    "    row = {}\n",
    "    article = Article(record[\"xml\"])\n",
    "\n",
    "    row[\"id\"] = record[\"_id\"]\n",
    "    row[\"keywords\"] = article.keywords()\n",
    "    row[\"id_types\"] = list(article.pub_ids().keys())\n",
    "    row[\"article_type\"] = article.article_type()\n",
    "    row[\"has_body\"] = True if article.soup.body else False\n",
    "    row[\"text_matches\"] = record.get(\"text_matches\")\n",
    "    row[\"article_title\"] = article.article_title()\n",
    "    row[\"journal_title\"] = article.journal_title()\n",
    "    row[\"text_length\"] = len(record[\"extracted_text\"])\n",
    "\n",
    "    rows.append(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "article_df = pd.DataFrame.from_records(rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "any_matches\n",
       "False    19519\n",
       "True       481\n",
       "dtype: int64"
      ]
     },
     "execution_count": 282,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article_df[\"any_matches\"] = [False if row is None else True for row in article_df[\"text_matches\"]]\n",
    "article_df.groupby([\"any_matches\"]).size()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "metadata": {},
   "outputs": [],
   "source": [
    "article_df.to_csv(\"article_sample.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>any_matches</th>\n",
       "      <th>article_type</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>True</td>\n",
       "      <td>research-article</td>\n",
       "      <td>446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>True</td>\n",
       "      <td>review-article</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>True</td>\n",
       "      <td>brief-report</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>True</td>\n",
       "      <td>protocol</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>True</td>\n",
       "      <td>case-report</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>True</td>\n",
       "      <td>other</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>True</td>\n",
       "      <td>data-paper</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>True</td>\n",
       "      <td>editorial</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>True</td>\n",
       "      <td>letter</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>False</td>\n",
       "      <td>research-article</td>\n",
       "      <td>14138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>False</td>\n",
       "      <td>review-article</td>\n",
       "      <td>1279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>case-report</td>\n",
       "      <td>896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>abstract</td>\n",
       "      <td>507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>False</td>\n",
       "      <td>editorial</td>\n",
       "      <td>314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>brief-report</td>\n",
       "      <td>311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>book-review</td>\n",
       "      <td>306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>False</td>\n",
       "      <td>letter</td>\n",
       "      <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>False</td>\n",
       "      <td>correction</td>\n",
       "      <td>245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>article-commentary</td>\n",
       "      <td>92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>False</td>\n",
       "      <td>news</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>False</td>\n",
       "      <td>meeting-report</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>False</td>\n",
       "      <td>discussion</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>False</td>\n",
       "      <td>product-review</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>False</td>\n",
       "      <td>obituary</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>False</td>\n",
       "      <td>protocol</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>False</td>\n",
       "      <td>in-brief</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>False</td>\n",
       "      <td>rapid-communication</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>False</td>\n",
       "      <td>methods-article</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>False</td>\n",
       "      <td>retraction</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>False</td>\n",
       "      <td>systematic-review</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>False</td>\n",
       "      <td>introduction</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>False</td>\n",
       "      <td>reply</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>False</td>\n",
       "      <td>data-paper</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>addendum</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>books-received</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>announcement</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>False</td>\n",
       "      <td>oration</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>False</td>\n",
       "      <td>report</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>case-report</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>calendar</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    any_matches         article_type  count\n",
       "39         True     research-article    446\n",
       "40         True       review-article     15\n",
       "32         True         brief-report      7\n",
       "38         True             protocol      4\n",
       "33         True          case-report      3\n",
       "37         True                other      3\n",
       "34         True           data-paper      1\n",
       "35         True            editorial      1\n",
       "36         True               letter      1\n",
       "28        False     research-article  14138\n",
       "30        False       review-article   1279\n",
       "9         False          case-report    896\n",
       "22        False                other    745\n",
       "1         False             abstract    507\n",
       "13        False            editorial    314\n",
       "7         False         brief-report    311\n",
       "5         False          book-review    306\n",
       "16        False               letter    301\n",
       "10        False           correction    245\n",
       "4         False   article-commentary     92\n",
       "19        False                 news     82\n",
       "17        False       meeting-report     67\n",
       "12        False           discussion     42\n",
       "23        False       product-review     31\n",
       "20        False             obituary     29\n",
       "24        False             protocol     27\n",
       "14        False             in-brief     17\n",
       "25        False  rapid-communication     15\n",
       "18        False      methods-article     13\n",
       "29        False           retraction     12\n",
       "31        False    systematic-review     11\n",
       "15        False         introduction      9\n",
       "26        False                reply      9\n",
       "11        False           data-paper      7\n",
       "2         False             addendum      3\n",
       "6         False       books-received      3\n",
       "3         False         announcement      2\n",
       "21        False              oration      2\n",
       "27        False               report      2\n",
       "0         False          case-report      1\n",
       "8         False             calendar      1"
      ]
     },
     "execution_count": 285,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article_df.groupby([\"any_matches\", \"article_type\"]).size()\n",
    "art_types = article_df.groupby([\"any_matches\", \"article_type\"]).size().reset_index()\n",
    "art_types.columns = [\"any_matches\", \"article_type\", \"count\"]\n",
    "art_types.sort_values(by = [\"any_matches\", \"count\"], ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "metadata": {},
   "outputs": [],
   "source": [
    "keep_vars = [\"id\", \"any_matches\"]\n",
    "unnest_var = \"keywords\"\n",
    "\n",
    "def unnest(data, unnest_var, keep_vars):\n",
    "    all_vars = keep_vars + [unnest_var]\n",
    "    nested = article_df.loc[:, keep_vars + [unnest_var]]\n",
    "    lens = [len(item) if item is not None else 1 for item in nested[unnest_var]]\n",
    "    unnested_dict = {var: np.repeat([nested[var].values], lens) for var in keep_vars}\n",
    "    unnested_dict[unnest_var] = np.hstack(nested[unnest_var])\n",
    "    unnested = pd.DataFrame(unnested_dict)\n",
    "    return(unnested)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "metadata": {},
   "outputs": [],
   "source": [
    "keywords = unnest(article_df, \"keywords\", [\"id\", \"any_matches\"])\n",
    "\n",
    "keywords.to_csv(\"keywords.csv\")\n",
    "\n",
    "id_types = unnest(article_df, \"id_types\", [\"id\", \"any_matches\"])\n",
    "\n",
    "id_types.to_csv(\"id_types.csv\")\n",
    "\n",
    "text_matches = unnest(article_df, \"text_matches\", [\"id\"])\n",
    "\n",
    "text_matches.to_csv(\"text_matches.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Old version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processed 240 articles (96.0%) in 0m12s; about 0m1s left.\n"
     ]
    }
   ],
   "source": [
    "id_types = defaultdict(int)\n",
    "article_types = defaultdict(int)\n",
    "keywords = defaultdict(int)\n",
    "\n",
    "reporter = Reporter(10, len(matches))\n",
    "\n",
    "for idx, doc in enumerate(matches):\n",
    "    if idx is not 0 and idx % reporter.interval is 0:\n",
    "        reporter.report(idx)\n",
    "    article = Article(doc[\"xml\"])\n",
    "    \n",
    "    kwds = article.keywords()\n",
    "    if kwds:\n",
    "        for kwd in kwds:\n",
    "            keywords[kwd] += 1\n",
    "    \n",
    "    ids = article.pub_ids().keys()\n",
    "    if len(ids) is 0:\n",
    "        id_types[\"none\"] += 1\n",
    "    else:\n",
    "        for item in ids:\n",
    "            id_types[item] += 1\n",
    "    \n",
    "    article_type = article.article_type()\n",
    "    if article_type:\n",
    "        article_types[article_type] += 1\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "keywords = pd.DataFrame(list(keywords.items()), columns=[\"keyword\", \"count\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>keyword</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>inflammation</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>oxidative stress</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>epidemiology</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>Obesity</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>Prostate cancer</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220</th>\n",
       "      <td>breast cancer</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>SNP</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1833</th>\n",
       "      <td>cytotoxicity</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>mitochondria</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>depression</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>801</th>\n",
       "      <td>Parkinson’s disease</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>388</th>\n",
       "      <td>Breast cancer</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1566</th>\n",
       "      <td>Prognosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2259</th>\n",
       "      <td>autophagy</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1274</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>obesity</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>819</th>\n",
       "      <td>Saccharomyces cerevisiae</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>459</th>\n",
       "      <td>Aging</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>Proliferation</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464</th>\n",
       "      <td>drug resistance</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>466</th>\n",
       "      <td>tuberculosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1320</th>\n",
       "      <td>cancer</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1523</th>\n",
       "      <td>diagnosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1731</th>\n",
       "      <td>apoptosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>PD-1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>694</th>\n",
       "      <td>Depression</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1223</th>\n",
       "      <td>clinical trial</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1762</th>\n",
       "      <td>Apoptosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>662</th>\n",
       "      <td>rehabilitation</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1003</th>\n",
       "      <td>Personality disorders</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1004</th>\n",
       "      <td>Carbon footprint</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1005</th>\n",
       "      <td>Economic evaluation</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1006</th>\n",
       "      <td>Online education</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1008</th>\n",
       "      <td>evaluation</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1009</th>\n",
       "      <td>star trackers</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1010</th>\n",
       "      <td>slew tolerance</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1011</th>\n",
       "      <td>attitude availability</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1012</th>\n",
       "      <td>star detection</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>celiac disease</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>Helicobacter pylori</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>immunoglobulin A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>981</th>\n",
       "      <td>data association</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>974</th>\n",
       "      <td>[18F]FP-(+)-DTBZ</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>975</th>\n",
       "      <td>PET</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>Test-retest</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>977</th>\n",
       "      <td>Cryptococcosis</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>978</th>\n",
       "      <td>Pulmonary Tuberculosis</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>Coinfection</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>980</th>\n",
       "      <td>multi-object tracking</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>982</th>\n",
       "      <td>deep neural network</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>lysosomal enzymes</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983</th>\n",
       "      <td>long short-term memory network</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>nanoparticles</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>turkey</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>aminopeptidases</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>poultry meat</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>988</th>\n",
       "      <td>nutrition</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989</th>\n",
       "      <td>supplementation</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2796</th>\n",
       "      <td>argonaute</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2797 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             keyword  count\n",
       "43                      inflammation      6\n",
       "587                 oxidative stress      5\n",
       "159                     epidemiology      5\n",
       "183                          Obesity      4\n",
       "279                  Prostate cancer      4\n",
       "1220                   breast cancer      4\n",
       "647                              SNP      4\n",
       "1833                    cytotoxicity      4\n",
       "45                      mitochondria      4\n",
       "294                       depression      4\n",
       "801              Parkinson’s disease      4\n",
       "388                    Breast cancer      4\n",
       "1566                       Prognosis      3\n",
       "2259                       autophagy      3\n",
       "1274                          Cancer      3\n",
       "42                           obesity      3\n",
       "819         Saccharomyces cerevisiae      3\n",
       "459                            Aging      3\n",
       "543                    Proliferation      3\n",
       "464                  drug resistance      3\n",
       "466                     tuberculosis      3\n",
       "1320                          cancer      3\n",
       "1523                       diagnosis      3\n",
       "1731                       apoptosis      3\n",
       "361                             PD-1      3\n",
       "694                       Depression      3\n",
       "1223                  clinical trial      3\n",
       "1762                       Apoptosis      3\n",
       "662                   rehabilitation      3\n",
       "396                colorectal cancer      3\n",
       "...                              ...    ...\n",
       "1003           Personality disorders      1\n",
       "1004                Carbon footprint      1\n",
       "1005             Economic evaluation      1\n",
       "1006                Online education      1\n",
       "1008                      evaluation      1\n",
       "1009                   star trackers      1\n",
       "1010                  slew tolerance      1\n",
       "1011           attitude availability      1\n",
       "1012                  star detection      1\n",
       "993                   celiac disease      1\n",
       "992              Helicobacter pylori      1\n",
       "991                 immunoglobulin A      1\n",
       "981                 data association      1\n",
       "974                 [18F]FP-(+)-DTBZ      1\n",
       "975                              PET      1\n",
       "976                      Test-retest      1\n",
       "977                   Cryptococcosis      1\n",
       "978           Pulmonary Tuberculosis      1\n",
       "979                      Coinfection      1\n",
       "980            multi-object tracking      1\n",
       "982              deep neural network      1\n",
       "990                lysosomal enzymes      1\n",
       "983   long short-term memory network      1\n",
       "984                    nanoparticles      1\n",
       "985                           turkey      1\n",
       "986                  aminopeptidases      1\n",
       "987                     poultry meat      1\n",
       "988                        nutrition      1\n",
       "989                  supplementation      1\n",
       "2796                       argonaute      1\n",
       "\n",
       "[2797 rows x 2 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "keywords.sort_values(\"count\", ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(int,\n",
       "            {'pmid': 18771,\n",
       "             'pmc': 20000,\n",
       "             'publisher-id': 13829,\n",
       "             'doi': 17408,\n",
       "             'pii': 1117,\n",
       "             'pmc-scan': 1198,\n",
       "             'manuscript': 206,\n",
       "             'coden': 219,\n",
       "             'art-access-id': 208,\n",
       "             'other': 134,\n",
       "             'publisher-manuscript': 22,\n",
       "             'sici': 27,\n",
       "             'medline': 4})"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(int,\n",
       "            {'research-article': 14584,\n",
       "             'abstract': 507,\n",
       "             'brief-report': 318,\n",
       "             'retraction': 12,\n",
       "             'other': 748,\n",
       "             'review-article': 1294,\n",
       "             'editorial': 315,\n",
       "             'case-report': 899,\n",
       "             'correction': 245,\n",
       "             'news': 82,\n",
       "             'discussion': 42,\n",
       "             'letter': 302,\n",
       "             'meeting-report': 67,\n",
       "             'product-review': 31,\n",
       "             'reply': 9,\n",
       "             'article-commentary': 92,\n",
       "             'addendum': 3,\n",
       "             'book-review': 306,\n",
       "             'protocol': 31,\n",
       "             'obituary': 29,\n",
       "             'rapid-communication': 15,\n",
       "             'methods-article': 13,\n",
       "             'in-brief': 17,\n",
       "             'systematic-review': 11,\n",
       "             'introduction': 9,\n",
       "             'oration': 2,\n",
       "             'books-received': 3,\n",
       "             ' case-report': 1,\n",
       "             'data-paper': 8,\n",
       "             'report': 2,\n",
       "             'announcement': 2,\n",
       "             'calendar': 1})"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article_types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Look at the text of a single document matching a search result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pymongo.results.UpdateResult at 0x10d353408>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.update_many(\n",
    "    filter={},\n",
    "    update={ \"$unset\": { \"text_matches\": \"\" } }\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
