{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pymongo\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.display import clear_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pubcrawler.article import Article\n",
    "from bs4 import BeautifulSoup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "from reporter import Reporter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "client = pymongo.MongoClient(\"localhost\", 27017)\n",
    "articles = client.pmc.articles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"terms\") as f:\n",
    "    terms = [line.strip() for line in f.readlines()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "art = articles.find_one()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# To sample a random article:\n",
    "art = articles.aggregate([\n",
    "    { \"$sample\": { \"size\": 1 } }\n",
    "]).next()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['field work',\n",
       " 'fieldwork',\n",
       " 'field study',\n",
       " 'field site',\n",
       " 'field area',\n",
       " 'study site',\n",
       " 'study location',\n",
       " 'study area',\n",
       " 'research site',\n",
       " 'research location',\n",
       " 'sampling site',\n",
       " 'sampling location',\n",
       " 'sampling area']"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "def view_one_article(query={ \"$match\": { \"text_matches\": { \"$in\": terms } } }):\n",
    "    query = [\n",
    "        query,\n",
    "        { \"$sample\": { \"size\": 1 } }\n",
    "    ]\n",
    "    article = articles.aggregate(query).next()\n",
    "    print(\"\"\"\n",
    "PMC ID\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TITLE\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "METADATA\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TEXT MATCHES\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "EXTRACTED TEXT\n",
    "\n",
    "{}\"\"\".format(article.get(\"_id\"),\n",
    "             article.get(\"article_title\"),\n",
    "             article.get(\"article_meta\"),\n",
    "             article.get(\"text_matches\"),\n",
    "             article.get(\"extracted_text\").strip()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "6221015\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "Separation of Scales in Transpiration Effects on Low Flows: A Spatial Analysis in the Hydrological Open Air Laboratory\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "{'has_body': True, 'article_type': 'research-article'}\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['study site', 'study area']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "1\n",
      "\n",
      "\n",
      "Introduction\n",
      "\n",
      "Evaporation and transpiration in midlatitude humid catchments affect streamflow at two main time scales. At the seasonal time scale, the energy input is at a maximum in summer; therefore, evapotranspiration is also at a maximum. This depletes soil moisture in summer below the annual mean, which affects runoff generation during storms. Soil moisture depletion also tends to reduce groundwater recharge and hence discharge to the streams. Streamflow during recession periods is the net result of the interplay of the hydraulic aquifer characteristics and evapotranspiration within the catchment. Several studies observed faster streamflow recessions in summer than during the rest of the year due to summer evapotranspiration (Federer, 1973; Shaw & Riha, 2012; Szilágyi et al., 2007). At the daily time scale, there are similar fluctuations in the energy input between day and night, leading to diurnal fluctuations in evaporation and transpiration, which, again, affect soil moisture and subsurface flow. In small headwater catchments these diurnal fluctuations of transpiration usually imprint a diurnal signal on the streamflow during low flow periods (Gribovszki et al., 2010).\n",
      "\n",
      "Even though there are two distinct time scales of variability in evaporation and transpiration, diurnal, and seasonal, a relationship between the two time scales would be expected. Only a few studies examined the seasonality in the diurnal fluctuations and how the diurnal transpiration was related to the seasonal transpiration that determines the catchment water balance. Bond et al. (2002) and Wondzell et al. (2007, 2010) showed that the area contributing to streamflow fluctuations decreased as the catchment gradually dried out. This dynamic was explained by the weakening of the coupling between the vegetation and stream during summer as the groundwater levels dropped. The time lags between transpiration and streamflow also varies seasonally due to changes in the flow paths (Barnard et al., 2010; Cadol et al., 2012; Deutscher et al., 2016; Fonley et al., 2016; Graham et al., 2013; Gribovszki et al., 2008; Kirchner, 2009; Szeftel, 2010; Szilágyi et al., 2008; Wondzell et al., 2007, 2010; Yue et al., 2016).\n",
      "\n",
      "One of the main questions to investigate diurnal streamflow fluctuations during low flows in the past was to determine where the streamflow fluctuations originate. Numerous studies have confirmed that these fluctuations were induced by transpiration from the riparian and near river vegetation. Experiments where the riparian forest was removed showed that the streamflow fluctuations stopped (e.g., Dunford & Fletcher, 1947; Lawrence, 1990; O'Loughlin et al., 1982); however, when the hillslope vegetation was removed instead, the fluctuations persisted but in a modified way (Bren, 1997). Another question was related to the main flow paths in the subsurface between the vegetation and the stream. The integrated effect of site conditions and the species assemblage may determine the sources of root water uptake (Snyder & Williams, 2000). Certain trees may extract water directly from the groundwater (Barbeta & Peñuelas, 2017; Gribovszki et al., 2010; Snyder & Williams, 2000), but the amplitude of the water table fluctuations depends on the distance from the stream and the vegetation type. For example, Reigner (1966) showed that the groundwater fluctuations were significant only up to a 2‐m distance from the stream. Irrigation experiments of Barnard et al. (2010) in Oregon suggested, however, that during higher soil moisture conditions after irrigation, hillslope vegetation located further from the stream could contribute to the diurnal streamflow fluctuations. Yue et al. (2016) observed diurnal groundwater table fluctuations along a part of a river that was covered by woody species and wet slough; however, the middle section of the river with shallower‐rooted grasses did not exhibit water table fluctuations. When the depth to the water table exceeded a certain threshold during a long recession period, the amplitudes of the diurnal water table fluctuations observed in an area covered by wet slough decreased.\n",
      "\n",
      "There are no robust methods for measuring the evapotranspiration rates in riparian forests with mixed vegetation types (Drexler et al., 2004; Goodrich et al., 2000; Landon et al., 2009; Leenhouts et al., 2006; Loheide et al., 2005). For many of the narrow riparian corridors, the fetch requirement of the eddy covariance method often exceeds the width of the riparian forest (Goodrich et al., 2000). Upscaling sap flux measurements from tree to stand level can also be problematic (Cermák et al., 2004; Oishi et al., 2008; Schaeffer et al., 2000). The empirical approaches based on crop coefficients and vegetation index‐based crop coefficients (Glenn et al., 2011; Nagler et al., 2005) are not widely used for riparian ecosystems due to the heterogeneous species composition. Therefore, estimating the evapotranspiration rates for the riparian zone based on measurements of diurnal water level fluctuations can be a valuable alternative (e.g., Cernohous & Šach, 2008; Dvorakova et al., 2014; Gribovszki et al., 2008; Loheide et al., 2005; White, 1932; Wondzell et al., 2007, 2010). However, estimating riparian zone evapotranspiration in terms of volumes might be uncertain at some study sites where estimated evapotranspiration rates are only representative of a fraction of the riparian zone (Butler et al., 2007; Loheide et al., 2005; White, 1932).\n",
      "\n",
      "Previous studies examined diurnal streamflow fluctuations and their temporal changes in single or nested catchments with spatially rather uniform runoff generation mechanisms (e.g., Bond et al., 2002; Szeftel, 2010). However, it is not clear how spatial differences in the runoff generation mechanisms affect the flow paths and total evapotranspiration. The main objective of our study therefore was to evaluate spatial and temporal patterns of the diurnal streamflow fluctuations in relation to the seasonal cycle of evapotranspiration in a small experimental catchment with different runoff generation mechanisms and mixed land cover types. The study was performed in the Hydrological Open Air Laboratory (HOAL), a 66‐ha Austrian experimental catchment, where both the main stream and the tributaries draining nearly all the surface contributing area to the stream are gauged, allowing the main stream and tributary contributions to be separated. The goal of our study was to determine where the diurnal low flow fluctuations originate, that is, how much the streamflow fluctuations observed at the tributaries with different characteristics and runoff generation mechanisms (such as wetlands, springs, and tile drainage systems) contribute to the streamflow fluctuations observed at the catchment outlet. Furthermore, we aimed to understand which part of the catchment induces the streamflow fluctuations on the diurnal time scale and which part influences the evapotranspiration on the seasonal time scale. In order to explore the causal link between the process drivers and the streamflow signal, we used a solar radiation‐driven model. The estimated model parameters assisted in generalizing the response time scales associated with the diurnal fluctuations and the seasonal recession of streamflow.\n",
      "\n",
      "2\n",
      "\n",
      "\n",
      "Study Area and Data2.1\n",
      "\n",
      "\n",
      "Study Area\n",
      "\n",
      "The study was conducted in a small experimental catchment, the HOAL in Petzenkirchen, located in the western part of Lower Austria (Figure 1a), approximately 100 km west of Vienna (48°9′N, 15°9′E). The drainage area of the catchment is 66 ha at the catchment outlet (MW). The natural surface water outlet of the catchment is the Seitengraben stream. The elevation of the catchment ranges from 268 to 323 m above sea level, with a mean slope of 8%. The stream is approximately 620 m long and has a medium slope of 2.4% (Blöschl et al., 2016; Eder et al., 2014, 2010).\n",
      "\n",
      "Figure 1 Hydrological Open Air Laboratory in Petzenkirchen, Lower Austria (panel a). Streamflow is monitored at the main outlet (MW) and at 11 tributaries (detail, panel b) that have different dominant runoff generation mechanisms.\n",
      "\n",
      "The climate is humid. Mean annual (2002–2015) air temperature, precipitation, runoff, and evapotranspiration are 9.6 °C, 784 mm/year, 178 mm/year, and 606 mm/year, respectively. Seasonal maxima of air temperature, rainfall amount, and intensity occur in the summer (Figure 3 in Blöschl et al., 2016). Mean monthly runoff tends to peak in winter or early spring. The seasonal variability in evapotranspiration plus storage change estimated from the water balance is presented in Figure 2.\n",
      "\n",
      "Figure 2 Monthly evapotranspiration (ET) and storage change (dS) estimated from the water balance in the Hydrological Open Air Laboratory for the period 2002–2015.\n",
      "\n",
      "The geology of the HOAL consists of Tertiary fine sediments and fractured siltstone of the Molasse zone. The dominant soil types are Cambisols (57%), Kolluvisol (16%), and Planosols (21%) with moderate to low permeability. Gleysols (6%) occur close to the stream (Blöschl et al., 2016; FAO et al., 1998).\n",
      "\n",
      "The catchment is dominated by agricultural land use. At the time of the study 87% of the catchment area was arable land, the rest was forested, paved, or used as pasture. The crops were mainly winter wheat, maize, winter barley, and winter oilseed rape. Winter wheat and winter barley resume growing in March, reaching their full height in June and are harvested typically in July. Maize is planted middle/end of April and harvested in late September. Oil seed rape develops from March to June and is harvested in July. Most of the forested area is located close to the stream (Figure 1). The most common trees in the riparian zone are different species of willow (Salix), poplar (Populus), ash (Fraxinus), field maple (Acer), and black alder (Alnus glutinosa). The main shrub species in the riparian zone are cornel (Cornus), elder (Sambucus), hazelnut (Corylus avellana), honeysuckle (Lonicera), and guelder rose (Viburnum).\n",
      "\n",
      "One of the main features of the HOAL is the wide range of observed runoff generation mechanisms, such as infiltration excess overland flow, reinfiltration of overland flow, saturation excess runoff from wetlands, tile drainage flow, shallow aquifer seepage flow, and groundwater discharge from springs (Blöschl et al., 2016). Due to the thin soil layers with low permeability and the agricultural land use, subsurface tile drainage systems were installed in the 1950s to eliminate waterlogging (Figure 1a). These systems contribute to the stream at five points: Sys1 and Sys2 flow throughout the whole year, while Frau1, Frau2, and Sys3 are ephemeral. Sys3 behaves as a combination of a tile drain and a wetland during low flow conditions as it also collects the water from the neighboring highly saturated parts of the catchment. Exner‐Kittridge et al. (2016) found that, chemically and dynamically, the water at Sys1 tile drain originates from the deep aquifer and not the shallow aquifer; therefore, Sys1 behaves like a spring rather than a drainage system. The upper 25% of the stream length was piped in the 1950s to expand the agricultural area. The concrete pipe enters the main stream at inlet Sys4. The dynamics of Sys4 are similar to the perennial drains. A perennial spring, Q1, originates from a fractured siltstone aquifer and directly enters the main stream. Two wetlands (A1 and A2) are fed by springs that seep into the stream through rivulets in the southeastern part of the catchment. Due to their high saturation, the runoff response of the wetlands is fast. During major storm events, saturation overland flow laterally enters the stream at E1 and E2 erosion gullies and, potentially, a few other points, depending on the event magnitude, soil moisture state, and the direction of the plowing. As both the main stream and the tributaries draining nearly all the surface contributing area to the stream are gauged (12 gauging stations in total), the main stream and tributary contributions can be separated.\n",
      "\n",
      "The estimated drainage area, the percentage of the total drainage area occupied by the riparian forest, the forest cover, the mean streamflow, and the mean low flow episode streamflow (according to sections 3.1 and 4.1) are summarized in Table 1 for each gauge in the HOAL. Comparing the mean streamflow at MW catchment outlet with the sum of the tributaries for the time period 2013–2015, the tributaries contribute approximately 56% to the total runoff at MW outlet. The contribution of the tributaries to the main stream is higher in the low flow periods (63%). This means that about 40% of the streamflow observed at MW enters the stream laterally in a diffuse way, mostly through the subsurface.\n",
      "\n",
      "Table 1\n",
      "Gauge\tRunoff generation mechanism\tEstimated drainage area (ha)\tRiparian zone (%)\tForest cover (ha)\tMean streamflow (L/s)\tMean low flow episode streamflow (L/s)\n",
      "MW\tOutlet\t65.8\t3.5\t6.32\t4.23 (3.94)\t3.12 (2.82)\n",
      "Sys4\tInlet pipe\t37.4\t0.2\t1.73\t0.74\t0.62\n",
      "Frau1\tTile drain\t3.1\t0.1\t0.00\t0.03\t0.00\n",
      "Frau2\tTile drain\t4.8\t0.3\t0.01\t0.15\t0.09\n",
      "Sys1\tTile drain (deep aquifer)\t6.5\t5.0\t0.77\t0.43\t0.41\n",
      "Sys2\tTile drain\t2.4\t0.7\t0.45\t0.18\t0.16\n",
      "Sys3\tTile drain/Wetland\t4.3\t0.1\t0.61\t0.09\t0.06\n",
      "A1\tWetland\t2.1\t1.6\t0.25\t0.09\n",
      "A2\tWetland\t1.1\t7.0\t0.17\t0.09\t0.06\n",
      "Q1\tDeep aquifer\t2.0\t0.8\t0.02\t0.46\n",
      "E1\tErosion gully\t0.8\t0.9\t0.01\t0.00\n",
      "E2\tErosion gully\t1.0\t0.2\t0.00\t0.09\t0.002.2\n",
      "\n",
      "\n",
      "Data\n",
      "\n",
      "Streamflow has been monitored at MW outlet of the catchment by a calibrated H‐flume with a pressure transducer since 2001 with 1‐min temporal resolution (Figure 1). Streamflow measurements at the tributaries started in 2011. At Sys4 inlet, at Frau1, Frau2, and Sys2 tile drainage systems, Sys1 tile drain (deep aquifer), Sys3 tile drain/wetland, at the wetlands (A1 and A2) and at the erosion gullies (E1 and E2) H‐flumes and pressure transducers are used to monitor the flow; at Q1 spring a V‐notch weir and a pressure transducer are used to measure the flow. Details on the sensors are given in Blöschl et al. (2016).\n",
      "\n",
      "Two time periods were selected for the analysis because of the differences in the record lengths between the main outlet and the tributaries. While the streamflow fluctuations at the main outlet were analyzed for the period 2002–2015, the fluctuations at the tributaries were analyzed for the period 2013–2015. Typical examples of streamflow fluctuations at the main outlet (MW) during a low flow period in summer and in autumn 2006 are presented in Figure 3. The relatively long rainless period during June 2006 resulted in a recession of the streamflow, gradually decreasing from 5 to 3 L/s over 21 days. Due to evapotranspiration, diurnal streamflow fluctuations were superimposed on the recession curve (Figure 3a). In the middle of October 2006, the amplitude of the diurnal streamflow fluctuations was around 0.4 L/s. Following a storm event (28–31 October), the air temperature dropped below freezing point and the diurnal streamflow fluctuations stopped. Even though later the air temperature increased above freezing point, the diurnal streamflow fluctuations did not resume during the rest of the year (Figure 3b).\n",
      "\n",
      "Figure 3 Example of air temperature (Tair) interpolated between 7, 14, and 19 hr, rainfall (P), and streamflow (Qm) fluctuations at the main outlet (MW) of the Hydrological Open Air Laboratory in (panel a) June 2006 and (panel b) October–November 2006.\n",
      "\n",
      "Nineteen piezometers located in the riparian forest close to the stream have monitored the groundwater level at a resolution of 5 min since 2013. Consistent with the streamflow records, groundwater level fluctuations were analyzed for the period 2013–2015. Five piezometers at different distances from the stream were selected for the analyses. The average depth to the groundwater table between 2013 and 2015 was larger further away from the stream: 0.3 m at BP02 (0.3 m from the stream), 0.4 m at BP07 and H04 (1.6 and 1.4 m from the stream respectively), 2.7 m at H02 (7.4 m from the stream), and 4.3 m at H01 (14.7 m from the stream).\n",
      "\n",
      "Rainfall has been measured with high temporal resolution (1 min) since 2002 by a weighing rain gauge situated 200 m from the catchment outlet. In 2012 four additional weighing rain gauges were installed in the HOAL (Figure 1).\n",
      "\n",
      "Since 1986 air temperature has been recorded at 7, 14, and 19 hr by a thermometer and cumulative daily solar radiation has been measured with a pyranometer located about 500 m from the catchment outlet. Since October 2012 air temperature, incoming and outgoing solar and long wave radiation have been measured at the HOAL weather station at 1‐min temporal resolution. Three eddy covariance stations have measured crop evapotranspiration in the HOAL since August 2012.\n",
      "\n",
      "Streamflow, groundwater level, rainfall, and solar radiation data were aggregated to hourly values for the analysis. Test simulations indicated that the difference between hourly and shorter time steps was negligible during low flows. For the time period 2002–2012, hourly solar radiation on a horizontal surface was estimated from measured cumulative daily solar radiation as a function of the day of the year, the solar time and the latitude (Duffie & Beckman, 2013).3\n",
      "\n",
      "\n",
      "Methodology3.1\n",
      "\n",
      "\n",
      "Identification of Periods With Streamflow Fluctuations\n",
      "\n",
      "In order to analyze the diurnal streamflow fluctuations, periods during streamflow recession with diel streamflow and groundwater level signals were identified for the main outlet, for all tributaries and for five piezometers based on four criteria:\n",
      "\n",
      "\n",
      "the day and the night preceding the day had no precipitation,\n",
      "\n",
      "the difference between the daily minimum and maximum streamflow or groundwater level exceeded a threshold (see supporting information Table S1),\n",
      "\n",
      "the daily minimum streamflow or groundwater level was observed between 8 a.m. and 7 p.m., and\n",
      "\n",
      "the daily maximum streamflow or groundwater level was observed either before 3 p.m. or after 9 p.m.\n",
      "\n",
      "For the selected episodes, the amplitude was estimated as the difference between the daily minimum and maximum streamflow or groundwater level (see supporting information Text S1). If an episode was identified at the main outlet, the streamflow records of the tributaries and the groundwater levels of the piezometers were checked with regard to the presence of fluctuations. If one or more of the criteria above were not matched at the tributaries or at the piezometers, the episode was considered to have zero amplitude; otherwise, the amplitudes were evaluated in a similar way as for the main outlet. The mean streamflow of the episodes was calculated for the same time periods for all streamflow gauges and became 0 only if a tributary dried out.\n",
      "\n",
      "3.2\n",
      "\n",
      "\n",
      "Modeling of Streamflow Fluctuations\n",
      "\n",
      "The study proceeded along two approaches of analyzing streamflow fluctuations, that is, direct analyses and model simulations. The spatiotemporal patterns and the seasonal variability of the low flow fluctuations were described through direct analyses of the observations. The model simulations aimed at estimating the time lags of the streamflow response relative to its forcing. A new modeling approach based on a simple impulse response model was used assuming that\n",
      "\n",
      "\n",
      "the amplitude of the diurnal streamflow fluctuations is proportional to incoming shortwave solar radiation; that is, solar radiation can be used as a proxy for transpiration (e.g., Renner et al., 2016),\n",
      "\n",
      "the temporal pattern of the diurnal streamflow fluctuations can be modeled by an exponential response function to solar radiation, and\n",
      "\n",
      "the main recession trend during the low flow period is exponential.\n",
      "\n",
      "The model has three free parameters (f, λ, and α). The parameter f expresses the proportion of the maximum available energy in the entire catchment or subcatchment, which influences the amplitude of the diurnal streamflow fluctuations, assuming that the energy consumed by evapotranspiration is equal to incoming shortwave solar radiation. If this assumption is true, f is equal to 1, if the entire maximum catchment energy contributes to diurnal streamflow variations. The parameter λ represents the time lag between incoming shortwave solar radiation and the diurnal streamflow fluctuations. The parameter α is the recession time scale. If α is significantly larger than λ, that is, there is a magnitude difference between the two time scales, then a separation of scales exists in the time domain.\n",
      "\n",
      "Based on assumption (i) evapotranspiration ET is estimated from the incoming shortwave solar radiation according to (1)(1)ET=f·S·GρΔHvapwhere ET is evapotranspiration in (L3/T), f is the dimensionless amplitude factor, S (L2) is the drainage area of the entire catchment or subcatchment, G (M/T3) is incoming shortwave solar radiation, ρ (M/L3) is the density of water, and ΔHvap (L2/T2) is the latent heat of vaporization of water. This approach assumes no plant regulation of evapotranspiration. Based on assumption (ii), the evapotranspiration pattern is convoluted with an exponential response function to obtain the evapotranspiration signal in the hydrograph Qd (L3/T) according to (2):\n",
      "(2)Qdt=∫0tETτut−τdτwhere τ (T) is the integration variable and ut (1/T1) is the response function according to (3):\n",
      "(3)ut=1λe−tλwhere λ (T) is the time lag and t (T) is time. Based on assumption (iii), the recession curve Qr (L3/T) is expressed as an exponential function (4):\n",
      "(4)Qr=Q0e−tαwhere α (T) is the recession time scale and Q0 (L3/T) is set to the maximum measured discharge during the first day of the time period analyzed. Streamflow Q (L3/T) is calculated according to (5), subtracting the evapotranspiration signal from the recession curve:\n",
      "(5)Q=Qr−Qd\n",
      "\n",
      "It is important to note that the model is not a predictive model. It was fitted to the streamflow data using a multiple objective calibration approach (combining the root‐mean‐square error, the amplitude error, and the error of timing, see supporting information Text S2, Chu et al. (2011)) for the purpose of interpreting the streamflow fluctuations and low flow recessions at different times of the year. As the model was fitted to each recession period independently, the result was one parameter set (f, λ, and α) for each episode.\n",
      "\n",
      "The modeled evapotranspiration volumes were compared with upscaled, literature‐based riparian evapotranspiration volumes (see supporting information Text S3 and Table S2), (Beeson, 2011; Hinckley et al., 1994; Köcher et al., 2008). The groundwater evapotranspiration, that is, the evapotranspiration calculated from the diurnal fluctuations of the shallow groundwater levels, was estimated by the White method (White, 1932) and the empirical method of Gribovszki et al. (2008; see supporting information Text S4).4\n",
      "\n",
      "\n",
      "Results4.1\n",
      "\n",
      "\n",
      "Amplitude of the Observed Diurnal Signals\n",
      "\n",
      "The number of episodes with diurnal variations at the main outlet was 549 and 138 for the two time periods (Table 2). Depending on the runoff generation mechanism, the streamflow data of some tributaries featured diel signals simultaneously with the catchment outlet (approximately 60% of the episodes for the wetlands), while other tributaries were dry during long rainless periods (for instance the erosion gullies E1 and E2). While all periods were used for the direct analyses, periods shorter than 3 days were excluded from the model simulations (Table 2). The mean lengths of the episodes for the direct analyses and model simulations were about 60 and 110 hr, respectively.\n",
      "\n",
      "Table 2\n",
      "Time period\tGauges\tDirect analyses\tModel simulations\n",
      "Number of episodes\tTotal number of days in episodes\tMean episode length (hr)\tNumber of episodes\tTotal number of days in episodes\tMean episode length (hr)\n",
      "2002–2015\tMW\t549\t1364\t57\t185\t832\t108\n",
      "2013–2015\tMW and tributaries\t138\t344\t56\t42\t197\t113\n",
      "\n",
      "The monthly average amplitudes of the streamflow diel signals at the MW catchment outlet and diurnal fluctuations of the groundwater levels at two piezometers (1.6 m from the stream at BP07 and 1.4 m away from the stream at H04) show a clear seasonal pattern (Figures 4a and 4c). The amplitudes of the diurnal streamflow and groundwater level fluctuations increased from spring to summer. During the summer season (May to September) the amplitude of the streamflow fluctuations was usually larger than 0.4 L/s at MW outlet, while during the winter season (November to March) the amplitudes approached 0. The amplitude is controlled by both the energy input, which peaks in the summer, and the efficiency of the plants as they allocate the available energy to transpiration. The very small values of the amplitudes in the late autumn and winter months suggest that the efficiency is the lowest in the winter season. This is consistent with observations (e.g., Figure 3b) that the first late autumn frost terminates the diurnal streamflow fluctuations.\n",
      "\n",
      "Figure 4 Seasonal variability of the monthly average measured amplitudes Am of the streamflow diel signals (panel a) and the time of the day when the minimum measured discharge occurs tm,min (panel b) at MW catchment outlet based on observations during 2002–2015; seasonal variability of the monthly average measured amplitudes Am,gwl of the groundwater level diel signals (panel c), and the time of the day when the minimum measured groundwater level occurs tm,min,gwl (panel d) at BP07 and H04 piezometers based on observations during 2013–2015. For the piezometers the results have been lumped into bimonthly bins because of the shorter observation period. Direct analyses, see supporting information Text S1.\n",
      "\n",
      "The magnitudes of the streamflow fluctuations varied between the locations (Figures 5 and 6). During a 5‐day dry period in August 2013 diel signals were observed for most of the tributaries and piezometers in the HOAL, although with very different magnitudes (Figure 5). The differences between the tributaries are highlighted in Figure 6, which shows the relative amplitudes, that is, the mean measured amplitudes as a function of the mean streamflow of the episodes at the nine gauges of the catchment that usually have nonzero runoff during rainless periods. The other three gauges, the ephemeral Frau1 tile drain and the erosion gullies (E1 and E2), are not included in Figure 6 because they are always dry during rainless periods.\n",
      "\n",
      "Figure 5 A 5‐day episode in the Hydrological Open Air Laboratory, 14–18 August 2013. (panel a) Measured runoff is shown as solid lines, two virtual gauges (i.e., combination of gauges, LF with large amplitudes and SF with small amplitudes) are shown as dotted lines. Incoming shortwave radiation G and latent heat of vaporization LE at the weather station are shown at the top. (panel b) Normalized (between 0 and 1) groundwater levels of five piezometers (BP02, H04, BP07, H02, and H01, Figure 1).\n",
      "\n",
      "Figure 6 Mean measured streamflow amplitudes (Am,mean) as a function of the mean measured streamflow (Qm,mean) of the episodes at nine stream gauges. Solid line indicates the relative low flow amplitudes of the MW outlet; dashed line indicates the trend of the relative average low flow amplitudes of the tributaries.\n",
      "\n",
      "In order to understand from which tributaries the fluctuations detected at MW catchment outlet originate, vegetation cover, dominant soil types, depths to groundwater level, and runoff generation mechanisms were explored for the respective subcatchments. Sys3 tile drain/wetland is located on the left side of the stream, where the depth to groundwater level is small. The drainage area of Sys3 tile drain/wetland is dominated by Kolluvisol around the outlet; the soil texture is characterized by a large percentage of silty loam causing low permeability and generally wet soil conditions. The drainage area around the outlet is covered by forest; the dominant tree types are ash, poplar, and black alder. The root system of these tree species in moisture retaining and organic rich soil types can reach depths of 2 m (Crow, 2005); therefore, they are apparently exceptionally well connected to the shallow groundwater table. Similarly to Sys3 tile drain/wetland, A1 and A2 wetlands were characterized by diel fluctuations with large magnitudes (Figures 5a and 6). The wetlands are also located on the left side of the stream, and due to their high wetness conditions and large riparian forest cover, the trees can be expected to be well connected to the shallow groundwater table as well.\n",
      "\n",
      "The relative amplitude was 1 magnitude smaller at Sys4 inlet and Sys1, Sys2, Frau2 tile drains compared to the wetlands (Figures 5a and 6). One of the reasons for the smaller relative amplitudes is the different vegetation cover (Table 1). The catchment area of Sys4 inlet and Frau2 tile drains is covered mainly by crop; the forest cover is minimal (Table 1). The fluctuations at these tributaries are possibly caused by the narrow riparian forest zone close to the outlets. Furthermore, Frau2 tile drain enters the main stream in an oblique way from the crop fields so is more exposed to riparian vegetation, unlike Frau1 tile drain that enters perpendicularly and is dry in rainless periods. Even though Sys1 tile drain behaves as a spring according to chemical analyses and it has basically constant contribution to the outflow at the catchment outlet even in the driest months, the root zone of the riparian forest may reach the aquifer, causing diurnal fluctuations of the streamflow. At Sys1 tile drain (deep aquifer) the relative amplitudes of the fluctuations induced by the riparian vegetation were smaller than the relative amplitudes at the wetlands and they were superimposed on the high baseflow rates. The deep aquifer spring Q1, which produces runoff throughout the year from a fractured siltstone layer, is not exposed to the local effect of riparian vegetation; hence, diurnal variations of the outflow were never observed (Figures 5a and 6).\n",
      "\n",
      "Based on the magnitudes of the observed relative amplitudes and the process reasoning above, the tributaries in the HOAL were separated into two distinct groups, which were lumped into two virtual gauges by taking the sum of the measured streamflow rates. Virtual gauge LF consists of three gauges (A1 and A2 wetlands, and Sys3 tile drain/wetland) with large amplitudes relative to the streamflow rates. Virtual gauge SF consists of four gauges (Sys4 inlet pipe, Frau2, Sys1, and Sys2 tile drainage systems) with small or no diurnal fluctuations in streamflow (Figure 5a).\n",
      "\n",
      "MW catchment outlet integrates the characteristics of the different runoff mechanisms, and therefore, it can be regarded as a mixture of the different systems. However, Figure 6 suggests that the relative fluctuations at MW (about one fifth of the average low flow, solid line) were much larger than the average of all the tributaries (dashed line). This means that there must be significant additional mechanisms that give rise to the observed fluctuations at MW. About 40% of the streamflow observed at MW enters the stream laterally in a diffuse way (Table 1), mostly through the subsurface, and is not captured by the tributary gauges, which suggests that the remaining part of the fluctuations is related to these diffusive inflows. These are the areas of significant riparian vegetation.\n",
      "\n",
      "For comparison, Figure 5b shows the groundwater levels normalized to the minimum and maximum levels during the episode for five piezometers (supporting information Table S3), one of them located on the left bank, 1.6 m from the stream (BP07), and four of them on the right bank (BP02, H04, H02, and H01, which are installed 0.3, 1.4, 7.4, and 14.7 m from the stream, respectively). The mean amplitude of the measured groundwater level fluctuations during the 5‐day recession period in August 2013 (Figure 5) was 16.9 cm at BP07 piezometer, while no diurnal fluctuations were observed further away from the stream at H01 piezometer. The amplitude of the groundwater table fluctuations was 13 times larger at BP02 piezometer (0.3 m from the stream) than the water level fluctuations at MW Outlet.\n",
      "\n",
      "The impact of distance from the stream on the magnitude of the diurnal fluctuations was further investigated (Figure 7). Similarly to Figure 5b, the mean amplitude of the measured groundwater level fluctuations through the episodes decreased with larger distance from the stream. While the mean amplitude was 6.0 cm at BP02 piezometer 0.3 m from the stream, at H01 piezometer (14.7 m from the stream) it was 0.1 cm. These results indicate that the signal that is propagated into the streamflow decreased with larger distances from the stream.\n",
      "\n",
      "Figure 7 Mean amplitude of the measured groundwater level fluctuations (Am,gwl,mean) as a function of logarithmic distance (log L) from the stream.4.2\n",
      "\n",
      "\n",
      "Timing of the Observed Diurnal Signals\n",
      "\n",
      "The timing of the diurnal fluctuations, that is, the time of the day when the minimum streamflow or groundwater level occurs, showed a clear seasonal pattern (Figures 4b and 4d). In February and March the daily minimum discharge and groundwater level occurred in the early afternoon. As the season progressed, the daily minimum discharge and groundwater level occurred later in the afternoon, at around 16:00 at MW outlet and around 18:00 at the piezometers. These changes may be related to the response times of the system to the energy input. This phenomenon was analyzed in detail later by the model (section 4.5).\n",
      "\n",
      "The timing of streamflow fluctuations was compared with the timing of latent heat of vaporization measured by the eddy covariance technique at the weather station (Figure 5a at the top) and groundwater level fluctuations (Figure 5b). The latent heat of vaporization representative of the crop fields and grassland surrounding the weather station lagged behind the incoming solar radiation by 0.5–2 hr (Figure 5a at the top). The phase shift between streamflow and groundwater table fluctuations varied between locations. Piezometers were either in phase with MW, such as closest to the stream at BP02 piezometer (0.3 m away), while further away they changed earlier (BP07) or later (H02 and H04) than MW.\n",
      "\n",
      "4.3\n",
      "\n",
      "\n",
      "Model Performance\n",
      "\n",
      "For the 185 recession periods between 2002 and 2015 (Table 2), the median of the model efficiency, that is, Nash‐Sutcliffe coefficient, for MW was 0.89; the 25th and 75th percentiles were 0.68 and 0.95, respectively (see supporting information Table S4). For autumn, the model efficiency was lower due to the smaller amplitudes and signal‐to‐noise ratios. Figures 8 and 9 show examples of the model fit. A 17‐day recession period in June 2009 had several overcast days (Figure 8). The significant differences in the amplitudes between the days were captured very well by the model. For example, 11 June was an overcast day with a diurnal amplitude of streamflow at MW of only 0.3 L/s compared to the other days with amplitudes of about 1.1 L/s. During a 5‐day period in August 2013 (Figure 9), diel signals were observed at most of the tributaries. Virtual gauge LF (A1 and A2 wetlands and Sys3 tile drain/wetland) was characterized by large amplitudes, while amplitudes at virtual gauge SF (Sys4 inlet, Frau2, Sys1, and Sys2 tile drainage systems) were small.\n",
      "\n",
      "Figure 8 Observed (blue line) and simulated (red line) streamflow fluctuations Q at the MW catchment outlet in the period 2–18 June 2009. Incoming shortwave radiation G is shown at the top.\n",
      "\n",
      "Figure 9 Observed (blue line) and simulated (red line) streamflow Q for 14–18 August 2013. The tributaries are lumped into virtual gauge LF (A1 and A2 wetlands, Sys3 tile drain/wetland) and virtual gauge SF (Sys4 inlet, Frau2, Sys1, and Sys2 tile drainage systems). Incoming shortwave radiation G is shown at the top.4.4\n",
      "\n",
      "\n",
      "Process Controls on Amplitude of Diurnal Streamflow Variation\n",
      "\n",
      "In order to describe the dynamics of the diurnal low flow fluctuations, we analyzed the seasonal evolution of the calibrated model parameters. Figures 10a and 10c show the seasonal variability of the amplitude factor f at MW catchment outlet (a) and at Virtual gauge LF (c). Amplitude factor f expresses the proportion of maximum available energy in the catchment which affects the diurnal streamflow variations, if the energy consumed by evapotranspiration is equal to incoming shortwave solar radiation. The amplitude factor increased from early spring until the beginning of summer. Assuming that all the entire incoming shortwave radiation is allocated to transpiration, f = 0.004 in April and 0.008 in the summer would imply that only a small portion of the available energy in the catchment (0.004 and 0.008, respectively) contributes to the component of transpiration that causes the diurnal streamflow fluctuations.\n",
      "\n",
      "Figure 10 Seasonal evolution of the calibrated amplitude factor f (panels a, c) and f as a function of the maximum measured discharge Q0 on the first day of each recession period (panels b, d) for MW catchment outlet (panels a, b) and for Virtual gauge LF comprising the three gauges (A1 and A2 wetlands, Sys3 tile drain/wetland) with large amplitudes (panels c, d). For LF the results have been lumped into bimonthly bins because of the small sample size.\n",
      "\n",
      "The seasonal variability of the amplitude factor f at virtual gauge LF (A1 and A2 wetlands and Sys3 tile drain/wetland) was similar to the value for the MW catchment outlet (Figure 10c). It increased from early spring (median of 0.003) until the summer (median of 0.009) and decreased in the autumn (median of 0.002). At virtual gauge LF the mean of the amplitude factor f was 0.0081, 5 times larger than at virtual gauge SF (0.0018), where the relative amplitude of the fluctuations was 1 magnitude smaller.\n",
      "\n",
      "Figures 10b and 10d show the calibrated amplitude factor f as a function of the maximum measured discharge Q0 on the first day of each recession period. Amplitude factor f increased with discharge both for MW catchment outlet (Figure 10b) and even more clearly for virtual gauge LF (Figure 10d).\n",
      "\n",
      "The amplitude factor of the model comprises several factors influencing the diurnal streamflow fluctuations. One of these factors is the efficiency of the vegetation in root water uptake, which we estimated using eddy covariance measurements. Although the flux footprints of the eddy covariance stations located in the crop fields do not cover the riparian zone close to the stream, we estimated the proportion of the incoming solar radiation, which is allocated for transpiration in the crop fields under the climatic conditions in the HOAL. The seasonal variability of the ratio of daily evapotranspiration (expressed as latent heat) and incoming shortwave radiation is shown in Figure 11a. Analyzing only the clear‐sky days between June and August (2012–2014), the average of the incoming shortwave radiation was approximately 25.3 MJ/m2/day (or 293 W/m2, 10 mm/day), and the latent heat measured in the crop fields was around 7.4 MJ/m2/d (or 86 W/m2, 3 mm/day). The LE/G ratio, that is, the plants' efficiency of allocating energy to transpiration, increased from 0.1 in January to about 0.3 in summer (Figure 11a). This means that in summer only about 30% of the incoming shortwave radiation was balanced by latent heat in the crop fields. Assuming similarity between tree and crop transpiration, the ratio of the calibrated model parameter f and the median of the LE/G ratios represents a measure of how well the trees are connected to the stream (Figure 11b). Amplitude factor f is essentially the ratio of evapotranspiration seen by the stream and G, so f/(LE/G) is the ratio of evapotranspiration seen by the stream and the evapotranspiration measured by the eddy covariance method and hence a measure of the proportion of the available energy in the entire catchment influencing the streamflow fluctuations at the diurnal time scale. Figure 11b suggests that this proportion slightly increased from spring to summer, but the change was much smaller than that of the transpiration efficiency.\n",
      "\n",
      "Figure 11 Seasonal variability of the ratio of daily latent heat LE measured by eddy covariance and daily incoming shortwave radiation G (panel a) and the ratio of the calibrated model parameter f and the median of the LE/G ratios (panel b).4.5\n",
      "\n",
      "\n",
      "Process Controls on Lag Times of Diurnal Streamflow Variation\n",
      "\n",
      "The calibrated values of time lag λ at MW catchment outlet and virtual gauge LF including only those periods when the model accurately reproduced the observed time series (Nash‐Sutcliffe coefficient > 0.2; 90% and 69% of the modeled episodes were considered for MW outlet and virtual gauge LF, respectively) gradually increased through the year (Figure 12). In March the median of the time lag λ was around 3 hr; it increased to 8 hr in May and 11 hr in October. For virtual gauge LF (A1 and A2 wetlands and Sys3 tile drain/wetland) there was a similar increase (Figure 12c). The mean value of the time lag λ was higher at Virtual gauge SF (11.1 hr) than at virtual gauge LF (10.3 hr; Table 3).\n",
      "\n",
      "Figure 12 Seasonal evolution of the calibrated model parameter time lag λ (panels a, c) and α (panels b, d) for MW catchment outlet (panels a, b) and for Virtual gauge LF comprising the three gauges (A1 and A2 wetlands, Sys3 tile drain/wetland) with large amplitudes (panels c, d). For LF the results have been lumped into bimonthly bins because of the small sample size.\n",
      "\n",
      "Table 3\n",
      "Gauge\tTime period\tf\tλ\tα\n",
      "MW Catchment Outlet\t2013–2015\t0.0096\t11.3\t69.3\n",
      "MW Catchment Outlet\t2002–2015\t0.0076\t8.9\t58.4\n",
      "Virtual LF (A1 and A2 Wetland, Sys3 Tile drain/Wetland)\t2013–2015\t0.0081\t10.3\t19.8\n",
      "Virtual SF (Sys4 Inlet, Frau2, Sys1, Sys2 Tile drain)\t2013–2015\t0.0018\t11.1\t77.6\n",
      "\n",
      "The second lag time of the model is the recession time scale α. It operates at longer time scales than λ and is related to the catchment drainage and evapotranspiration as the catchment dries out during the recession periods. Figures 12b and 12d show the seasonal evolution of the calibrated values of recession time scale α at MW catchment outlet (b) and virtual gauge LF (d) for only those periods when the Nash‐Sutcliffe coefficient was larger than 0.2 and there was a clear recession in the hydrographs at MW catchment outlet (66% and 38% of the modeled episodes were considered for MW outlet and virtual gauge LF, respectively). A few episodes had increasing baseflow, which would imply a rainfall input not considered. Therefore, these episodes were not considered in estimating α. There was a tendency for time scale α to gradually increase from spring to autumn, from a median value of 21 days in March to 54 days in October. According to Table 3, the mean recession time scale α was 4 times larger at virtual gauge SF than at virtual gauge LF, where the systems are fed by deeper subsurface flow.5\n",
      "\n",
      "\n",
      "Discussion5.1\n",
      "\n",
      "\n",
      "Spatiotemporal Patterns of Streamflow Fluctuations\n",
      "\n",
      "Several studies analyzed the spatial and temporal differences of diurnal streamflow fluctuations between catchments and in single or nested catchments but focused only on spatially uniform runoff generation mechanisms. For instance, Lundquist and Cayan (2002) analyzed diurnal variations in 100 rivers in the western United States and found a weak correlation between the amplitude of the diurnal cycle and mean monthly temperature, discharge, basin area, and mean basin elevation. They concluded that each catchment was unique and that the spatiotemporal patterns of diel streamflow signals were determined by local physiographic and hydrologic characteristics.\n",
      "\n",
      "The observation and modeling results presented in this study show that the spatial differences in runoff generation greatly influence the magnitude of the diurnal streamflow fluctuations. We found that the amplitude of the diel streamflow signal relative to the streamflow at the wetlands was 1 magnitude larger than those of the tile drains and the deep aquifers (Figures 5a and 6). The wetlands are the areas in the catchment with high wetness conditions, shallow groundwater table, and large riparian forest cover (Table 1). In contrast, the inlet pipe, the tile drainage systems, and deep aquifers are fed by deeper water sources, covered mainly by crop fields and the depth to the groundwater level is larger than in the wetlands.\n",
      "\n",
      "The modeling results also reflected the differences in the relative fluctuations between the tributaries. The amplitude factor f of the solar radiation‐driven model was larger in the wetlands; that is, a larger proportion of the maximum catchment energy was allocated for transpiration, which influenced the diurnal streamflow fluctuations. The recession time scale α of the model was longer for systems fed by deeper subsurface flow (tile drainage systems and deep aquifers) than for the shallower systems as would be expected.\n",
      "\n",
      "5.2\n",
      "\n",
      "\n",
      "Simplified Process Representation\n",
      "\n",
      "In this study incoming shortwave radiation proved to be a useful proxy for representing streamflow fluctuations due to riparian transpiration (see section 4.2). Dominant controls on transpiration, that is, stomatal (vapor pressure deficit and conductances) versus boundary layer (radiation) control, depend on the vegetation type, scale, and meteorological conditions (Jarvis & McNaughton, 1986; Martin et al., 2001). Due to the energy that is advected in the form of vapor pressure deficit, daily latent heat flux can exceed the daily sum of net radiation (e.g., Hall et al., 1998, observed such a phenomenon on a few, dry summer days). Therefore, using net radiation as a driver of the diurnal streamflow fluctuations might introduce a bias on longer time scales. Some studies suggested the atmospheric moisture deficit as the main driver of transpiration (e.g., Granier et al., 2000; Szeftel, 2010) or a combination of both energy and vapor pressure deficit (e.g., different versions of the Penman‐Monteith equation). However, other studies showed that transpiration of various tree species was closely related to solar radiation (e.g., Dragoni et al., 2005; Granier et al., 2000; Kume et al., 2008; Oguntunde & Oguntuaseb, 2007). For example, Pieruschka et al. (2010) suggested that the absorbed solar energy by the leaves influenced the stomatal control of transpiration; therefore, solar radiation was an important control on transpiration. Using 110 FLUXNET eddy covariance sites, Boese et al. (2017) observed a substantial transpiration component, also termed as equilibrium transpiration (Jarvis & McNaughton, 1986), which was independent of stomatal conductance and driven by incoming solar radiation. Phillips et al. (1999) showed that diurnal sap flow in a Panamanian humid forest was more correlated with radiation than with atmospheric moisture deficit and Williams et al. (2004) found that transpiration was not correlated with atmospheric vapor pressure deficit in an olive orchard in Morocco. Renner et al. (2016) argued that vapor pressure deficit and wind speed, two variables widely used in evapotranspiration estimations, only slightly increased the predictability of atmospheric demand in a beech forest in Luxembourg.\n",
      "\n",
      "In this study the evapotranspiration pattern was convoluted with an exponential response function that resulted in a hydrograph shape with convex rising limbs and concave recessions. Kovar and Bacinova (2015) used similar methods when they simulated the diurnal streamflow fluctuations with the Fourier series model. They applied both a linear and an exponential regression to simulate the depletion process and found a very small difference between the two approaches. Dvorakova et al. (2012, 2014) developed and calibrated a linear storage model to describe the recession process, where the actual evapotranspiration was reproduced by simplified Fourier series or sine curves (Dvorakova et al., 2012, 2014). Similar to our results, these studies also showed that the depletion of the catchment storage during low flow conditions could be captured by an exponential function.\n",
      "\n",
      "5.3\n",
      "\n",
      "\n",
      "Estimated Evapotranspiration Volumes and Rates\n",
      "\n",
      "The daily transpiration rates depend on the type of vegetation, structure, age, and leaf area index (Farid et al., 2008; Schaeffer et al., 2000). Numerous studies compared evapotranspiration or transpiration rates of different vegetation cover using eddy covariance or sap flow measurements. For example, Granier et al. (2000) found summer transpiration rates of about 4.5 mm/day in a beech forest in France. Water abstraction of willows next to the stream can reach 5.6 mm/day (Marttila et al., 2017). Crop evapotranspiration, typically, is also on the order of 4 mm/day, depending on crop type (e.g., Delzon & Loustau, 2005).\n",
      "\n",
      "The volumes of the diurnal streamflow fluctuations may be interpreted in two ways. The first and most common interpretation, as presented, for example, in Gribovszki et al. (2010), is to attribute the entire missing volume in the hydrograph to evapotranspiration. This is the assumption underlying our estimations. In terms of volumes, independently from the LE/G ratio, by integrating equation (1) over the day and taking the average over the episode, the daily average summer (between June and August) evapotranspiration at MW, LF, and LS was 69, 8, and 8 m3/day, respectively (Table 4). The tributary influences are expected to propagate synchronously along the main stream due to the short, approximately 50‐ and 80‐min‐long lag times of celerity and velocity, respectively, between the upstream tributaries and MW outlet (Eder et al., 2014). Therefore, this result means that 53 m3/day are not accounted for by the tributaries (LF + LS) so needs to be due to the diffusive subsurface inflow to the stream. In other words about 77% (53/69) of the volumes associated with the streamflow fluctuations are related to (hyporheic) exchange along the riparian zone of the main stream.\n",
      "\n",
      "Table 4\n",
      "Method\tReference gauge/piezometer\tTime period\tET\n",
      "Incoming shortwave radiation‐driven model\tMW Outlet\t2002–2015\t55\n",
      "Incoming shortwave radiation‐driven model\tMW Outlet\t2013–2015\t69\n",
      "Incoming shortwave radiation‐driven model\tVirtual gauge LF (A1 and A2 Wetland, Sys3 Tile drain/Wetland)\t2013–2015\t8\n",
      "Incoming shortwave radiation‐driven model\tVirtual gauge SF (Sys4 Inlet, Frau2, Sys1, Sys2 Tile drain)\t2013–2015\t8\n",
      "Upscaling literature‐based evapotranspiration values for the entire riparian zone\t—\t81\n",
      "\n",
      "The second interpretation of the diurnal streamflow fluctuations is that subsurface flow is controlled by slight changes in the potential gradients that, in turn, are controlled by the diurnal cycle of evapotranspiration as the suction of the roots changes during the day. If this is the case, the summer evapotranspiration rates causing the diurnal streamflow fluctuations are smaller than the estimated ones. Szilágyi et al. (2008) distinguished between a local and an overall hydraulic gradient driving the water transport in the vadose and saturated zones during recession flow periods using a 2‐D finite element numerical model. Voltz et al. (2013) observed an overall relatively small response of the hydraulic gradients during a summer recession period in a steep headwater catchment in Oregon, where the ratio of the cross‐to down‐valley hydraulic gradient showed the largest diurnal fluctuations in wells closest to the stream. Given the topography, where the roots of riparian trees can easily reach into the groundwater, and groundwater levels in the riparian zones themselves fluctuate, this mechanism is not likely important in the HOAL.\n",
      "\n",
      "In order to compare the mean daily evapotranspiration rates from the main outlet with evapotranspiration rates estimated from the shallow groundwater level fluctuations (ETG), the estimated evapotranspiration volumes (equation (1)) were divided by the product of the catchment area and the calibrated amplitude factor f. The comparison indicates that the daily rates from diurnal streamflow fluctuations from the groundwater levels were 7 and 5 mm/day based on the empirical method of Gribovszki et al. (2008) and the White (1932) method, respectively, which was slightly lower than the simulated rates for MW catchment outlet (9.5 mm/day). These results are consistent with the range (8–11 mm/day) found by Gribovszki et al. (2008) in an alder forest in Hungary. As the depth to the groundwater did not exceed 1 m (2013–2015) in the riparian zone within 0.5‐m distance from the stream in the driest summer months, the root system of the trees (especially the 40–70 year olds and >10‐m species) is likely in direct contact with the saturated zone and evapotranspiration estimated from the diurnal groundwater level fluctuations could be close to groundwater evapotranspiration (Dawson, 1996; Shah et al., 2007; Williams et al., 2006).\n",
      "\n",
      "Because the estimated evapotranspiration volumes could not be validated against measurements of sap flow and stomatal conductances, we estimated the transpired volumes for the entire riparian forest using aerial photographs, a tree survey and literature‐based transpiration values of different tree species (Table 4). The estimated summer transpiration rates (81 m3/day) show a good agreement with the modeling results (e.g., incoming shortwave radiation‐driven model set up for MW Outlet, for the time period 2013–15: 69 m3/day). The slightly larger values are expected as the literature‐based estimate was calculated for the entire riparian zone (Table 4), which may not fully contribute to the streamflow fluctuations.\n",
      "\n",
      "5.4\n",
      "\n",
      "\n",
      "Separation of Scales in Time Implies a Separation of Scales in Space\n",
      "\n",
      "The analysis of the streamflow fluctuations during low flow conditions at MW catchment outlet shows that the time lag λ between radiation and the diurnal low flow fluctuations gradually increased from 3 to 11 hr as the season progressed. The time lag represents the total response consisting of a cascade of responses, which includes the time lags between radiation and evaporation from the stomata, sap flow in the branches and the stem, root water movement, groundwater movement, and groundwater‐stream interactions (Figure 13). Each component has its own time lag. A number of studies found that the time lag between the diurnal fluctuations of radiation and sap flow in the tree was approximately 30 min for species such as apple trees (Dragoni et al., 2005), beech (Granier et al., 2000), and Japanese cedar (Kumagai et al., 2009). Gartner et al. (2009) found that the sap flow of birch and spruce in the southeastern part of Austria lagged solar radiation by 1 hr during early August. When the soils dried out during a significant drought, the time lags increased to approximately 2.5 hr. Hence, it is likely that the time lag of the mixed vegetation in the riparian zone of the HOAL catchment is also on the order of 0.5 to 2.5 hr. In some of the studies above, the time lag of sap flow from stem to branch was included in the estimates; therefore, this value is also small. Similarly, it is likely that the time lags for root water uptake are small in the riparian zone, where the groundwater table is high, which is typical in the HOAL, especially on the left side of the stream. The remaining time lag components are associated with subsurface processes (groundwater movement and groundwater‐stream interactions). Assuming that the lag components are additive, one would estimate lags of the subsurface processes of about 1.5 hr in early spring to about 9 hr in autumn for MW catchment outlet. The time lags are shorter on those tributaries that are located on the left side of the stream with a western aspect where the groundwater levels are shallow and the riparian forest cover is more dense (virtual gauge LF).\n",
      "\n",
      "Figure 13 A possible conceptualization of the catchment during low flow periods: schematic of the decoupling of riparian zone fluxes and catchment zone fluxes for the catchment outlet. Width of the arrows indicates the time scale. The thick arrows represent time scales of hours (i.e., well connected at the diurnal scale); the thin arrows represent time scales of weeks or months (i.e., decoupled at the diurnal scale).\n",
      "\n",
      "The reason for the longer subsurface lag times in the summer months are presumably a consequence of the soil moisture status of the riparian zone. Along the stream in the HOAL there is probably a mixture of infiltration and exfiltration due to the heterogeneity of the topography in the near‐stream zone. Previous studies (e.g., Caldwell et al., 1998; Voltz et al., 2013) found a reversal of groundwater gradients in the riparian zone due to nighttime infiltration of groundwater into the stream and reversed flow during the day as the trees take up groundwater and part of the stream water to satisfy their transpiration needs (Figure 13). Although we did not observe the complete reversal of the groundwater gradient in the riparian zone with the current measurement setup, a diurnal change in the magnitude of the gradient was observed. The fact that the main part of the time lag is related to the movement of water in the subsurface suggests that the increase in lag times during the summer months is a consequence of the amount of water stored in the near‐stream zone. The subsurface water storage, both soil moisture and groundwater, is largest in spring, which is also indicated by the seasonal maximum in the discharge. Therefore, it is likely that the short lag times in spring are a consequence of the fact that the roots are well connected to the water sources and the celerities are higher. As the season progresses, the catchment gradually dries out and the roots become less connected to the subsurface water storage. When the soils get drier, the unsaturated zone becomes thicker and flow paths get longer and the overall hydraulic conductivity gets smaller. These mechanisms cause an increase in the lag times from 3 to 11 hr for MW catchment outlet. This interpretation is consistent with the findings of Bond et al. (2002), Fonley et al. (2016), Moore et al. (2011), and Wondzell et al. (2007, 2010).\n",
      "\n",
      "Similarly to time lag λ, the recession time scale α could be also described by a cascade of responses. This cascade includes the time lags between transpiration of the crop fields in the catchment and root water uptake, water movement in the unsaturated zone, groundwater movement, and groundwater‐stream interactions. While it is difficult to separate the individual components, it is clear that the dynamics of root water uptake and transpiration of crops are faster than the water movement in the unsaturated zone and the groundwater movement. The former operates at a daily scale with significant diurnal variations, while the latter operates at time scales of weeks and months. The differences in the magnitudes of the time scales are indicated in Figure 13 by the thickness of the arrows. Thick arrows represent time scales of hours; thin arrows represent time scales of weeks or months. The recession time scale shows a gradual seasonal increase from 21 to 54 days for MW catchment outlet. This increase is presumably related to the amount of water stored in the subsurface at the catchment scale. As the catchment dries out, groundwater flow seems to follow deeper flow paths, which are associated with longer response times consistent with the findings of Bond et al. (2002). A similar, increasing trend of the recession time scale α was found for the subcatchments with shallow groundwater levels (virtual gauge LF) although, overall, the time scales are smaller, as would be expected because of the shorter distances. At subcatchments fed by deeper subsurface flow (virtual gauge SF) the recession time scale α was longer indicating a reduced dependence on riparian processes.\n",
      "\n",
      "Observations and model simulations showed that the diurnal signal observed in streamflow at different outlet points of the HOAL mainly originates from diurnal fluctuations in the riparian evapotranspiration. There is a clear spring onset, a late autumn offset (Figure 3b) of the diurnal diel signal in streamflow and a clear seasonal pattern in the amplitudes and timings (Figure 4). While the first harvest of the crop fields in July does not influence the amplitude of the diurnal signal in the stream (Figure 4), the amplitudes start to decrease only in the autumn months and the first frost when the trees in the riparian zone next to the stream drop their leaves terminates the diurnal streamflow fluctuations. Streamflow from tributaries with large riparian forest cover fluctuated more than streamflow from tributaries which were covered mainly by crop fields and fed by deep aquifers. These observations imply that the riparian zone is the main driver of the diurnal streamflow fluctuations. The results of a solar radiation‐driven model showed that only a small proportion of the maximum available catchment energy induced the diurnal streamflow fluctuations (Figure 10). This proportion was larger during higher baseflow conditions, and it increased from spring to summer and decreased in autumn. The daily minimum streamflow occurred later as the season progressed, which was also reflected in the time lags of the solar radiation‐driven model. The change in the timing of the daily minimum streamflow and the increase in lag time are likely related to the drying of the catchment during the summer and autumn, which may lead to a partial disconnect of the riparian zone and the stream. This is consistent with the findings of previous studies, such as Bond et al. (2002) and Wondzell et al. (2007, 2010) who showed that the area contributing to streamflow fluctuations decreased as the catchment gradually dried out.\n",
      "\n",
      "A clear separation of scales (Blöschl & Sivapalan, 1995) exists in the time domain, which is apparent in the streamflow signal. Conceptually, time scales and space scales of variability are linked through their characteristic velocities; therefore, a separation of scales in the space domain would be expected, if a separation of scales in the time domain exists (Skøien et al., 2003). There is also interaction across the diurnal and seasonal time scales. The diurnal low flow fluctuations due to riparian and near‐riparian transpiration are modulated by the soil moisture state at the seasonal time scales. Conversely, daily transpiration contributes to the seasonal totals. This interaction across time scales is reminiscent of the effects of climate variability on floods—in a direct way through the seasonal variability of storm characteristics and indirectly through the seasonality of rainfall and evapotranspiration that affect the antecedent catchment conditions for individual storm events (Sivapalan et al., 2005).\n",
      "\n",
      "The streamflow and piezometer observations imply that at the daily time scale, riparian evapotranspiration induces the diurnal streamflow fluctuations, while most of the catchment evapotranspiration, such as evapotranspiration from the crop fields further away from the stream does not contribute to these fluctuations. This implication was confirmed by piezometer data. We found that the amplitudes of the fluctuations in the riparian zone within 2‐m distance from the stream were significant and did not find significant fluctuations 15 m from the stream. This is consistent with the findings of Reigner (1966), who also found that the amplitudes were significant within 2‐m distance from the stream. This behavior is also apparent in the time domain. As the first frost occurs, the low flow fluctuations at MW stop within days (Figure 3b). A similar behavior has been observed by Goodrich et al. (2000) and for groundwater fluctuations by Lautz (2008). Deutscher et al. (2016) distinguished between two distinct parts of the catchment with different connectivity, and connectivity may also exist in terms of soil moisture patterns (Western et al., 1998). Conversely, there is a quick onset of the fluctuations in spring at the beginning of the growing season, when the vegetation is growing. Preferred states and switching behavior associated with thresholds seem to be more common characteristics than what is usually assumed (Blöschl & Zehe, 2005; Zehe & Sivapalan, 2009). Spatial patterns of hydrologic dynamics may help identify preferred states (Blöschl, 2006; Grayson et al., 1997). Analyzing spatial patterns may also help to address the difficulties in predicting the whole‐catchment water balance from observations at the local scale (Thompson et al., 2011).6\n",
      "\n",
      "\n",
      "Conclusions\n",
      "\n",
      "This study investigated the spatial and temporal patterns of diurnal low flow fluctuations for different runoff generation mechanisms in a 66 ha Austrian experimental catchment, the HOAL. Our results showed that:\n",
      "\n",
      "‐The ratio of streamflow fluctuations and mean streamflow was around 0.3 for wetlands, where the riparian forest cover is the largest and the depth to the groundwater table does not exceed 1 m. The amplitudes were much smaller for tile drainage systems and springs that are fed by deeper subsurface flow, where the dominant land cover is crop and the ratio is around 0.04.\n",
      "\n",
      "‐The separation of scales in the time domain could be reproduced by a solar radiation‐driven model. Lag times between radiative forcing and evapotranspiration increased from 3 to 11 hr from spring to autumn as the catchment became more disconnected from the stream. The recession time scales increased from 25 days in spring to 60 days in autumn, which was likely a consequence of the decreasing storage of subsurface water at the catchment scale.\n",
      "\n",
      "‐A separation of scales in the time domain is apparent in the streamflow signal, that is, diurnal and seasonal fluctuations induced by transpiration, implies a separation of scales in the space domain: the diurnal streamflow fluctuation are driven by the riparian zone along the main stream, while most of the catchment (the crop fields located further away from the stream) did not affect the diel signals. This interpretation is supported by the groundwater level data.\n",
      "\n",
      "\n",
      "Supporting information\n",
      "\n",
      "Text S1\n",
      "\n",
      "Click here for additional data file.\n",
      "\n",
      "Text S2\n",
      "\n",
      "Click here for additional data file.\n",
      "\n",
      "Text S3\n",
      "\n",
      "Click here for additional data file.\n",
      "\n",
      "Text S4\n",
      "\n",
      "Click here for additional data file.\n",
      "\n",
      "Table S1\n",
      "\n",
      "Click here for additional data file.\n",
      "\n",
      "Table S2\n",
      "\n",
      "Click here for additional data file.\n",
      "\n",
      "Table S3\n",
      "\n",
      "Click here for additional data file.\n",
      "\n",
      "Table S4\n",
      "\n",
      "Click here for additional data file.\n"
     ]
    }
   ],
   "source": [
    "view_one_article()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "problematic_terms = terms[-3:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "6002991\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "Decadal stability in genetic variation and structure in the intertidal seaweed Fucus serratus (Heterokontophyta: Fucaceae)\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "{'has_body': True, 'article_type': 'research-article'}\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['sampling site', 'sampling location']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Background\n",
      "\n",
      "Understanding temporal stability of genetic structure and diversity is crucial for the utility of temporal snapshots in conservation management and to infer how climate-induced range shifts might affect the future distribution and adaptive potential of species. In trailing edge populations, effective population size and genetic diversity are considered major keys to adaptive potential and subsequent persistence under climate change [1, 2]. In contrast, the evolutionary potential and survival of low-diversity leading edge populations [3] may be either enhanced or impaired by the ‘surfing’ of new mutations that can rapidly increase in frequency over iterated founder events, depending on whether the new mutations are primarily favorable or deleterious [4–9].\n",
      "\n",
      "Studies that assess temporal genetic stability are rare in marine species, and mostly confined to fisheries management to ensure sustainable exploitation of economically important species [10–14]. While high gene flow explained 5 to 24-year long stability in genetic variability and structure in Chinook salmon and Atlantic herring [10, 11], large fluctuations in allele frequencies were recorded over a few months in small and closed populations of the intertidal isopod Jaera albifrons [15]. However, high gene flow does not always warrant temporal genetic stability, as several marine species with long-lived planktonic larvae showed stronger temporal than spatial differentiation over 3 to 9 years [16–18]. On the other hand, low gene flow does not necessarily result in genetic instability over time, although genetic drift in small and closed populations can be expected to be high. For example, genetic diversity and population structure remained stable over 5–12 years in relatively closed populations of the seagrass Zostera marina [19] and over 2 years in nine out of 10 locally differentiated populations of the isopod Excirolana braziliensis [20]. These contrasting results demonstrate that a species’ life history alone does not necessarily predict its genetic stability over time.\n",
      "\n",
      "Due to their high sensitivity to rising temperatures, responses of marine intertidal species are considered as early warning signals for the impact of climate change [21–25]. Among global climate change factors, ocean warming is considered the most severe threat for marine macrophytes [26–28]. Over the next century, ecological niche models predict the disappearance of intertidal fucoid brown algae along their southern trailing edges and a poleward extension of their northern leading edges [26, 29]. Fucoid brown algae (Heterokontophyta; Fucaceae) are habitat-forming ecosystem engineers supporting species-rich intertidal communities along temperate rocky shores [30–33]. Thus, range shifts of fucoids will undoubtedly trigger major ecological changes along temperate rocky shores of the North Atlantic.\n",
      "\n",
      "Ecological niche models, however, do not consider the species’ plastic and adaptive potential that could mitigate the predicted northward shifts. Adaptive potential depends largely on a population’s genetically effective size, Ne [34], or the size of an ideal population that undergoes the same rate of genetic change as the real population [35]. At low Ne, and low gene flow between populations, genetic drift generally plays the predominant role, effectively neutralizing selection, and eroding genetic diversity through stochastic fixation or loss of allelic variations [36–38]. Although Ne and temporal stability of genetic diversity patterns are particularly important for restoration and conservation efforts of fucoid seaweeds, only a single Norwegian population of F. serratus has so far been assessed [39].\n",
      "\n",
      "The canopy-forming seaweed F. serratus is an excellent model for the study of temporal evolution and stability of genetic structure and diversity across a range of contrasting temperature regimes. It is one of the dominant intertidal seaweeds along the Northeast-Atlantic rocky shore from northern Portugal to northern Norway [40]. Arctic regions are predicted to become thermally suitable through 2100 under CO2 emission scenario projections [26]. In contrast, regions south of the Brittany coast of France are predicted to become unsuitable [26], as temperatures will rise beyond the species’ potential for thermal acclimatization [41]. The susceptibility of F. serratus to climate change is expected to vary regionally, given the species’ regional patterns of genetic diversity [42], in combination with low gene flow between local populations [43].\n",
      "\n",
      "Genetic diversity of F. serratus is highest in the two former, large glacial refugia (20–18 thousand years ago (kya)) in Southwest Ireland, and Brittany [42, 43]. The third refugium in the Northwest Iberian peninsula is characterized by a high proportion of private alleles, and currently represents the species’ isolated trailing edge, where recurrent cycles of thermally induced extinction and recolonization have eroded genetic diversity [42, 43]. Currently, sea surface temperatures reach 22 °C, and although below the lethal limit of F. serratus (25 °C) [40, 44], inhibit growth, physiological performance and reproductive capacity [45–48].\n",
      "\n",
      "Genetic diversity of F. serratus decreases from its mid-range of distribution towards higher latitudes and is lowest in leading edge populations in northern Norway [42, 43]. Low genetic diversity in leading-edge populations is explained by the populations’ relatively young age and their derivation from small founder populations that carried only a subset of the genetic variation from glacial refugia to the north after the ice retreated, ca. 15–10 kya.\n",
      "\n",
      "While Ne is a good indicator for temporal genetic stability, its estimation relies on temporally-spaced genetic data [49]. Due to this complication in sampling design, Ne of F. serratus has been estimated in only a single population close to Bergen (Norway) over eight years [39]. The estimated Ne between 73 and 386 was regarded insufficient for long-term survival under environmental change [39]. However, a thorough appraisal of the spatial distribution and temporal stability of Ne and genetic diversity throughout the species’ latitudinal range of distribution cannot be inferred from a single location.\n",
      "\n",
      "Estimating climate change susceptibility in a species with low gamete/zygote dispersal requires to assess temporal genetic stability across its latitudinal and thermal range of distribution. In this study, we estimated Ne of F. serratus across latitude and temperature at seven locations with decadal sampling (2000 and 2010), a period in which Europe experienced three heat waves in 2003, 2006, and 2010 [50–52]. Here we evaluate whether range shifts in the north or strong selection pressures in the south have resulted in measurable changes in genetic diversity and population structure. In populations that are dominated by genetic drift and with small adaptive potential, we expected to find a decline in genetic diversity over the past decade. Finally, we discuss whether genetic diversity may be sufficiently stable to buffer environmental change and mitigate the current range shift predictions.\n",
      "\n",
      "\n",
      "Methods\n",
      "\n",
      "\n",
      "Sampling\n",
      "\n",
      "Individuals were sampled ca. ten years apart from the same seven populations spanning the latitudinal distribution of F. serratus (Fig. 1). Ethical approval is not required for research work with the seaweed/macroalga F. serratus. Field collections did not require specific permits and the species is neither endangered nor protected. Sampling involved removing a thumbnail-sized piece of tissue from ca. 50 to 100 individuals at each sampling site and did not threaten either the individual or the local population. Live samples were never transferred to other countries or locations within any of the countries. In all cases the specimens were collected within the context of various grants (see funding information) that involved at least one of the co-authors and one or more colleagues from the country where the collection was made.\n",
      "\n",
      "Fig. 1 Sampling sites. Coordinates, years of collection, sampling sizes (n), and daily average sea surface temperatures (SST) at each of the seven sampling sites. SSTs were identical between the two Norwegian sampling sites as well as between the two French and the two Spanish sampling sites. Summer temperatures were exceptionally high at the Danish and Spanish sampling sites during the first two of three heat waves that Europe experienced in years 2003, 2006, and 2010\n",
      "\n",
      "Variability in daily average sea surface temperatures and surface air temperatures at the sampling locations (Fig. 1, Additional file 1), recorded from 1999 to 2011, were extracted from the NOAA/OI/SST/V2 dataset (0.25° resolution, described in [53]) and the CPC Global Temperature dataset (0.5° resolution) provided by NOAA/OAR/ESRL/PSD, Boulder, Colorado, USA, [54]). Thermal variability was replicated in the two Norwegian, the two French, and the two Spanish samples, respectively. In Denmark, only a single population was sampled at two time points. Individual tissues were blotted dry and stored in silica prior to transport for subsequent DNA extraction.\n",
      "\n",
      "\n",
      "Microsatellite genotyping\n",
      "\n",
      "DNA was extracted from 2 mg silica dried tissue according to [55] with the modifications described in [56], followed by a purification step with the OneStep-96 PCR Inhibitor Removal Kit (Zymo Research, Irvine, USA) and a 1:3 dilution of the purified product. The samples were genotyped for a total of 31 microsatellite markers: 11 anonymous loci (L20, L38, L58, and L94 described in [57]; B113, B128, E6, E9, D39, A198, and F4 described in [58]) and 20 loci linked to expressed sequence tags (ESTs: F12, F22, F34, F36, F60, F45, F50, F17, F72, F49, F14, F21, F58, F19, F37, F65, F59, F69, F9, and described in [56]) (Additional file 2).\n",
      "\n",
      "Polymerase chain reactions (PCRs) with 5 μl total volume contained 1 μl purified DNA template, 1.34 μl nuclease-free Water (Ambion, Thermo Fisher Scientific), 2.5 μl of AmpliTaq Gold 360 MM (Applied Biosystems, Life Technologies) and 0.08 μl of each forward and reverse primer (each primer at 20 μM; forward primer labeled with 6FAM, NED, PET or VIC; Applied Biosystems, Life Technologies). PCR was performed in a Veriti 96-Well Thermal Cycler (Applied Biosystems, Life Technologies). The conditions are depicted in Additional file 3 and specified for each marker in Additional file 2.\n",
      "\n",
      "The fragment lengths were determined on an ABI 3500xl Genetic Analyzer from 1 μl of diluted PCR products (specified for each marker in Additional file 2) mixed with 8.9 μl of HiDi Formamide (Life Technologies) and 0.1 μl of Gene Scan 500 LIZ Size Standard (Life Technologies) after 5 min denaturation at 95 °C. Allele calling was performed with the GeneMapper v 4.1 Software (Applied Biosystems, Thermo Fisher Scientific).\n",
      "\n",
      "\n",
      "Data analysis\n",
      "\n",
      "The microsatellite raw dataset (Additional file 4) was corrected for allelic dropout with a Maximum Likelihood approach [59] using the program MicroDrop [60]. From the corrected data (Additional file 5), nine markers E9, F14, F17, F36, F37, F59, F60, F65, and L20 were removed from the full set of 31 markers before further analyses with the remaining 22 markers because the proportion of missing data for the excluded markers exceeded 12% in at least one of the populations.\n",
      "\n",
      "\n",
      "Diversity estimates\n",
      "\n",
      "Average locus heterozygosity Hexp (bias-corrected [61]), allelic richness α (the average number of alleles per locus) and multi-locus heterozygosity (MLH), the number of heterozygous loci per individual divided by the number of loci, were calculated for each sampling location. Regional estimates were obtained after pooling the two spatial samples from each of the Norwegian, Spanish and French regions. Regional estimates were not possible for the Danish region because only one population was sampled. Hexp was calculated with the R package ‘DEMEtics’ [62], and α was estimated with the R package ‘PopGenReport’. For local estimates, α was normalized to a sample size of 24, the smallest number samples in a population. For regional estimates α was normalized to a sample size of 24, and additionally, to a sample size of 50. MLH was estimated with the R package ‘InbreedR’. Inbreeding coefficients FIS [63] were estimated with the R package ‘Demerelate’ and tests for significant deviation from 0 were based on 1000 iterations. We tested for significant temporal changes of Hexp, α, FIS, and MLH at each sampling location with Wilcoxon rank sum tests in R [64]. To assess temporal evolution of diversity estimates, we tested for correlation between current and historical local measures with a Spearman’s rank correlation in R [64]. Additionally, we tested for significant differences between average present-day and historical values using Wilcoxon Rank Sum tests in R [64].\n",
      "\n",
      "Effective population sizes (Ne) were estimated with an assumed generation time of 2 years [65] with the R package ‘NB’ after removing loci with only one allele: Locus F9 for the Kirkenes population, locus F72 for the Ribadeo1 population and loci F21 and F72 for the Ribadeo2 population.\n",
      "\n",
      "\n",
      "Genetic differentiation\n",
      "\n",
      "Population structure was determined with Bayesian clustering methods implemented in the software Structure v2.3.4 [66]. Acceptance of six clusters (K) was determined with the δ K Method [67] in the R package ‘pophelper’ [68].\n",
      "\n",
      "Temporal genetic changes at each sampling location and geographic genetic differentiation within and between all historical and recent samples were estimated by the fixation index FST [69] using GENETIX 4.05 [70] and the differentiation index Dest [71]) using the R package ‘DEMEtics’ 0.8–7 [62]. Dest more correctly measures the true genetic differentiation compared with FST for multi-allelic markers such as microsatellites [62, 71]. Statistical significance of the pairwise comparisons was based on 10,000 permutations for FST and on 1000 Bootstrap repeats for Dest. To assess temporal stability of geographic differentiation, we tested for correlation between recent and historical FST and Dest values with Spearmans’s rank correlation in R [64]. Additionally, we tested for significant differences between average present-day and historical values using Wilcoxon Rank Sum tests in R [64]. Finally, we tested for correlation between temporal genetic differentiation (FST, Dest) and Ne with Pearson’s product moment correlation in R [64].\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "\n",
      "Genetic structure\n",
      "\n",
      "Bayesian clustering with the program Structure revealed clear differences between regions but not with time (Fig. 2). Historical and present-day FST values were strongly positively correlated (r = 0.93, p < 0.00001), and the overall historical FST value (0.21) did not differ significantly (p = 0.567) from the present-day value (0.22), indicating that spatial genetic differentiation between populations was globally consistent over time (Fig. 3a). Historical and present-day Dest values supported these findings as the overall values (0.40 and 0.42, respectively) did not differ significantly (p = 0.636) and were positively correlated (r = 0.97, p < 0.00001, Fig. 3b). Isolation by distance was indicated by stronger differentiation among than within countries (Additional file 6).\n",
      "\n",
      "Fig. 2 Clustering of samples. Sample assignment to six clusters (colors) with the program Structure shows consistent geographic differentiation between sampling times. Here, new and old refers to the two sampling years specified in Fig. 1\n",
      "\n",
      "Fig. 3 Present and historical genetic differentiation. Population differentiation estimated by FST (a) and Dest (b) with a 1:1 reference line\n",
      "\n",
      "Temporal changes, however, were noted on a local level. Local differentiation between the Norwegian populations decreased from 2004 to 2010 (Additional file 6). All but the French population ‘Ile de Siec’ changed significantly in genetic variation over time, as indicated by significant changes in FST and Dest (Additional file 7). The Spanish population ‘Ribadeo2’ showed significant temporal change in FST but not in Dest.\n",
      "\n",
      "\n",
      "Genetic variation/diversity\n",
      "\n",
      "Stable population diversities through time were indicated by significant correlations of historical and present-day intra-population diversity indices (Fig. 4; Hexp: r = 0.86, p = 0.02, MLH: r = 1, p = 0.0004; α: r = 0.96, p = 0.003; FIS: r = 0.82, p = 0.03). Moreover, average present-day values did not differ significantly (p > 0.05) from average historical values (Hexp: present = 0.56, historical = 0.56; MLH: present = 0.61, historical = 0.62; α: present = 6.22, historical = 6.36; FIS: present = − 0.10, historical = − 0.10).\n",
      "\n",
      "Fig. 4 Genetic diversity across latitudes. Present and historical diversity estimates of a) multi-locus heterozygosity (MLH), b) allelic richness (α), c) expected heterozygosity (Hexp), and d) inbreeding (FIS), with 1:1 reference lines representing unchanged temporal evolution\n",
      "\n",
      "Local and regional diversity estimates (Additional file 8) were highest in France and lower at the northern and southern distribution edges (Fig. 4). Regional α estimates (standardized to 50 samples) exceeded local estimates (standardized to 24 samples) in all regions (Additional file 8).\n",
      "\n",
      "Effective population size (Ne) was highest in the French population ‘Ile de Siec’ (Ne = 10,000,000) and lowest in the Norwegian population ‘Grense Jakobselv’ (Ne = 62) (Fig. 5, Additional file 7). Ne for the other populations ranged from 700 to 200 in the order: Gjerild Klint > Green Top > Ribadeo2 > Ribadeo1 > Kirkenes. At both sampling time points, none of the diversity estimates were significantly correlated with effective population size (all p > 0.09). The temporal decrease in MLH in Ribadeo2 was strong but not significant (p = 0.051, Additional file 7). The FIS in ‘Kirkenes’ was significantly negative (p = 0.043, Additional file 8).\n",
      "\n",
      "Fig. 5 Effective population sizes across latitudes. Effective population size (Ne) at each sampling location with 95% confidence intervals\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "The spatial distribution of genetic diversity has important implications for conservation and management as it reveals a species’ strong and weak points with regards to stability and evolutionary capacity [72–74]. Given the key ecological role of F. serratus [30–32, 75–77], the positive effect of genetic diversity may reach beyond the species level to affect community structure and increase species richness and stability of the entire associated ecosystem [78–81]. We are not aware of seaweed studies that have investigated positive ecosystem effects of genetic diversity, but genetic diversity enhanced heat-stress survival in germlings of Fucus vesiculosus [27]. Furthermore, in the habitat forming seagrass Z. marina, genotypic diversity not only enhanced biomass production, but also abundance of the associated fauna under near-lethal water temperatures [82] and community resistance to grazing [83]. Thus, maintaining genetic diversity in F. serratus is also expected to be important for conservation and management of the entire intertidal ecosystem along temperate rocky shores. Across the latitudinal range of F. serratus, genetic diversity and differentiation remained stable for 5–10 generations at regional scales, and in all but the Norwegian region at local spatial scales (Figs. 2, 3, 4). This suggests that, despite low gene flow between populations, effective population sizes have remained large enough to maintain genetic variation at least on the short term. Temporal genetic differentiation was systematically lower than local differentiation, and 1–2 orders of magnitude lower than regional differentiation (Additional file 9). This implies that temporal snapshots provide valuable information for conservation management of fucoid seaweeds, as they reliably reflect diversity and differentiation patterns for at least a decade.\n",
      "\n",
      "\n",
      "Ne comparisons\n",
      "\n",
      "In all but the Norwegian populations, Ne was estimated as > 260, a size reported as the median estimate for stable populations in over 83 studies spanning a diverse range of taxa [36]. This suggests low sensitivity to genetic stochasticity [36] in all but the northern edge populations of F. serratus. As in most studies, the precision of Ne decreased as Ne increased (Fig. 5) [36, 84, 85]. Local differentiation in F. serratus is one of the most important assumptions of the employed ‘temporal’ method to estimate Ne, in which neutral genetic change over time is expected to be inversely proportional to Ne. Discrete generations are another important assumption of the ‘temporal’ method. Overlapping generations are unlikely to cause a significant downward bias of Ne when more than 4 generations lie between the temporal samples [49]. This can be expected for most of our temporal samples, assuming a generation time of 1–2 years [65, 86] and a time span of 6–11 years between sampling. Thus, our sample-size-corrected estimates can be regarded as unbiased and indicative of a ‘real’ decline in Ne from the species’ mid-range of distribution to its range-edges.\n",
      "\n",
      "An Ne > 1000, as in the French ‘Ile de Siec’ population, is large enough to ensure evolutionary potential in perpetuity [87], and is likely to provide the best source for adaptive genetic rescue of threatened and declining populations [38, 88]. However, large Ne estimates are commonly associated with a high uncertainty [36, 85]. Accordingly, the point estimate of Ne in the ‘Ile de Siec’ population (ca. 10 Million) has a wide confidence interval as compared with the other populations (Fig. 5, Additional file 7). Consequently, the point estimate is unlikely to be the true value in this population, but is certainly > 1000, and higher than in any other measured populations. The reason for this outlier value is not due to high diversity, since this is comparable to the other French population (Fig. 4a-d), but the high stability in allele frequencies over time. Indeed, the ‘Ile de Siec’ population was the only population for which temporal genetic differentiation was non-significant (Additional file 7).\n",
      "\n",
      "\n",
      "Ne in the other mid-range populations, > 500, may be sufficient in the mid-term [36, 87, 89] to mitigate the predicted extinction by the end of the twenty-first century [26]. However, given that summer temperatures are predicted to rise above the thermal tolerance limit of F. serratus in Brittany within the next 200 years [26, 41], it is important to track its fitness in this region in order to implement early conservation measures in case it loses its current stability.\n",
      "\n",
      "An Ne of 50–100 was regarded necessary for a population to minimize inbreeding depression and associated problems such as accumulation of deleterious mutations and loss of variation [36, 87]. However, despite Ne < 60 in the Norwegian ‘Grense Jakobselv’ population, genetic diversity remained stable for both Norwegian populations over six years and neither population was inbred. In contrast, a previous study on a southern Norwegian population reported significant loss of Ne from 2000 to 2008 and concluded that an Ne between 73 and 386 was insufficient for long-term survival under environmental change [39]. Stable diversity despite small Ne in our two northern Norwegian populations may be ascribed to regional gene flow, suggested by a reduction in genetic differentiation between the two Norwegian populations from 2004 to 2010 and significant outbreeding (negative FIS) in the ‘Kirkenes’ population in 2010. Thus, regional gene flow may uncouple Ne from genetic stochasticity effects at the species’ Northern edge of distribution.\n",
      "\n",
      "\n",
      "Diversity comparisons\n",
      "\n",
      "As expected for neutral loci, genetic diversity was positively related with Ne [36, 38]. Both regional and local diversities are highest in Brittany and make the range-center of F. serratus less sensitive to genetic drift [36, 37]. A decline in genetic diversity towards the northern and southern range-edges is in accordance with the species’ biogeographic history [42].\n",
      "\n",
      "Low genetic diversity does not necessarily lower the evolutionary potential of F. serratus to adapt to Arctic shores [26]. The evolutionary potential, survival, and expansion rate of low-diversity leading edge populations [3] may decrease when deleterious mutations accumulate at expansion range fronts and create a so-called ‘expansion load’ [7, 9]. On the other hand, survival may well be enhanced by the ‘surfing’ of favorable mutations that can rapidly increase in frequency over iterated founder events [4, 5, 90]. An additional consideration is that source populations of Arctic colonists may not be located at the species’ northern edge, but within European harbors with frequent shipping, fishing, and cruise boat traffic to and from the northern polar regions.\n",
      "\n",
      "Our results have clear implications for the conservation of F. serratus at its southern edge. Reductions in MLH from 1999 to 2010 were close to significant (p = 0.0051/0.134 for Ribadeo2/Ribadeo1, respectively), although, Hexp, α, and FIS remained temporally stable. This agrees with stability of Hexp and α over 7–9 years in fragmented southern edge populations of the kelp species Laminaria digitata [91], and is likely due to the measures’ insensitivity to the effects of population bottlenecks [92]. In other words, while the polymorphic state of loci and the diversity of alleles did not decline, alleles occurred more frequently in a homozygous state in the recent samples. In theory, the decline in MLH might be explained by increased selection pressure for heat-tolerance, although there is only indirect experimental evidence for this. Acclimation potential to further thermal stress is likely impeded in this population by chronically high expression of heat shock protein genes [22, 23, 41, 93]. Between 2000 and 2010, the Ribadeo1 population experienced a 90% decline in abundance [26]. Although stable local differentiation favors ecotypic differentiation in thermal stress tolerance [41], heat-stress is becoming too extreme at the southern edge.\n",
      "\n",
      "The value of conserving the southern edge of F. serratus may be high [94]. Because of its separation from Brittany by the uninhabitable sandy warm shores of the Bay of Biscay, the Northwest Iberian glacial refugium did not contribute to postglacial recolonizations of ice-free northern shores, and, thus, preserves unique genetic variation [42]. The conservation value of the species’ southern edge becomes even more apparent when considering that small-scale population structure increases the species’ regional diversity above local diversity within single populations (Additional file 8). High regional diversity, despite low within-population diversity, was previously reported for the southern distribution edge of the seagrass Zostera marina [94, 95]. We are not aware of studies that explicitly addressed this effect in macroalgae, although increased local differentiation at the southern edge of the kelp Laminaria digitata [91] can be expected to increase regional variation as well. Thus, with the loss of its southern edge, the species’ can be expected to lose its most heat-adapted populations sustaining unique genetic variation.\n",
      "\n",
      "\n",
      "Conclusions\n",
      "\n",
      "Temporal snapshots of genetic diversity and structure in F. serratus populations spanning its latitudinal range reliably reflect patterns across local and regional spatial scales and across various thermal backgrounds for at least one decade. Stable small-scale structure enhances regional genetic diversity throughout the species’ range of distribution and is a potential driver for local adaptation [36] that may explain species radiation and diversity in the genus Fucus [96–98].\n",
      "\n",
      "\n",
      "MLH appears to be the most stress-sensitive measure of diversity, displaying a strong decline at the species’ southern edge of distribution. As sandy warm shores separate the Iberian southern edge from the genetically diverse Brittany region, genetic rescue by the influx of genetic variation [38, 88] might only be possible if initiated by conservation efforts.\n",
      "\n",
      "Increasing temperatures are likely the major cause for the decline not only of F. serratus, but also other intertidal and subtidal macroalgae in Northwest Iberia [28, 99–101], as well as temperate seaweeds worldwide [102]. Kelp species may maintain genetic diversity to a certain degree in southern edges by escaping to deep-water refugia to avoid rising temperatures in shallow waters [103]. Accordingly, in Northern Portugal, increasing air-temperature stress depresses the upper boundary limit of F. serratus [104]. However, intertidal seaweeds are less adapted to low light conditions and, thus, have low potential to escape into deeper waters. Another factor that impedes survival of southern edge populations in fucoid seaweeds is their reproductive strategy with fewer gametes and lower dispersal (< 12 m from parental sites [45, 46]) as compared with kelps that release billions of spores dispersing several kilometers [105, 106]. We suspect that without the influx of genetic variation from Brittany, intertidal habitat-forming macroalgae, such as F. serratus, may largely disappear from southern edges but retain potential to persist in small subtidal bottleneck populations in cool upwelling regions [107].\n",
      "\n",
      "\n",
      "Additional files\n",
      "\n",
      "\n",
      "Additional file 1:\n",
      "\n",
      "Surface air temperatures. Daily average surface air temperatures (SAT) at each of the seven sampling sites from 1999 to 2011 with gaps in year 2006 for the French and Spanish sampling sites. SATs were identical between the two Norwegian sampling sites as well as between the two French and the two Spanish sampling sites. (PDF 513 kb)Additional file 2:\n",
      "\n",
      "Microsatellite markers. Characteristics of each microsatellite marker, including cycling conditions and multiplexing. (XLSX 8 kb)Additional file 3:\n",
      "\n",
      "PCR cycling protocols. Time-release (a) and no-time-release (b) PCR cycling protocols. In the time-release protocol, the heat-activated DNA-polymerase was progressively released during the thermal cycling process. Annealing temperatures and number of cycles indicated with an X are specified for each marker in Additional file 2. (PDF 34 kb)Additional file 4:\n",
      "\n",
      "Microsatellite raw data. Microsatellite genotypes in Structure format. The first row contains the names of all 31 markers. The following rows contain the individual genotype data. Each individual is represented in 2 consecutive rows. The first column contains the name of the individual, the second row contains the population number that individual belong to. The following 31 columns show the alleles of each marker as microsatellites base pair lengths. The population numbers (1–14) refer to the following sampling locations and times: 1) Gjerild Klint, present-day; 2) Gjerild Klint, historical; 3) Green Top, present-day; 4) Green Top, historical; 5) Ile de Siec, present-day; 6) Ile de Siec, historical; 7) Grense Jakobselv, present-day; 8) Grense Jakobselv, historical; 9) Kirkenes, present-day; 10) Kirkenes, historical; 11) Ribadeo 1, present-day; 12) Ribadeo 1, historical; 13) Ribadeo 2, present-day; 14) Ribadeo 2, historical. (TXT 195 kb)Additional file 5\n",
      "\n",
      "Corrected microsatellite data. Microsatellite data corrected for allelic dropout in Structure format. The first row contains the names of all 31 markers. The following rows contain the individual genotype data. Each individual is represented in 2 consecutive rows. The first column contains the name of the individual, the second row contains the population number that individual belong to. The following 31 columns show the alleles of each marker as microsatellites base pair lengths. The population numbers (1–14) refer to the same sampling locations and times as in Additional file 4. (TXT 198 kb)Additional file 6:\n",
      "\n",
      "Spatial differentiation. Regional and local genetic differentiation between sampling sites in historical and present samples estimated by FST and Dest with p values. (XLSX 10 kb)Additional file 7:\n",
      "\n",
      "Temporal changes. Estimates of effective population size (Ne), temporal genetic differentiation (FST and Dest), and p values for temporal changes in diversity measures at each sampling site. (XLSX 6 kb)Additional file 8:\n",
      "\n",
      "Diversity estimates. Diversity estimates, including heterozygosity (Hexp), allelic richness (α), and multi-locus heterozygosity (MLH), for each location and region with standard errors, and inbreeding coefficients FIS with p values for each population. (XLSX 8 kb)Additional file 9:\n",
      "\n",
      "Temporal versus local genetic differentiation. Temporal genetic change in comparison to local and regional genetic differentiation(FST and Dest) for each population and sampling time point (historical and present). (XLSX 6 kb)\n"
     ]
    }
   ],
   "source": [
    "view_one_article({ \"$match\": { \"text_matches\": { \"$in\": problematic_terms } } })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "PMC ID\n",
      "\n",
      "3543354\n",
      "\n",
      "\n",
      "TITLE\n",
      "\n",
      "None\n",
      "\n",
      "\n",
      "METADATA\n",
      "\n",
      "None\n",
      "\n",
      "\n",
      "TEXT MATCHES\n",
      "\n",
      "['study area']\n",
      "\n",
      "\n",
      "EXTRACTED TEXT\n",
      "\n",
      "Introduction\n",
      "\n",
      "The transfer of energy and recycling of nutrients via the decomposition of organic matter is a central unifying process that links all organisms to the functioning of ecosystems [1]. However, dead organic matter varies enormously in its spatial and temporal distribution [2], and this determines the magnitude of its contribution to nutrient cycling [3] and the diversity and dynamics of its consumers [4]. Animal carrion is the most nutrient-rich form of dead organic matter [5], and recent reviews have highlighted the overlooked role of carrion in food webs [6], [7], [8], and driving variation in biodiversity and ecological processes in landscapes [5], [9]. Widespread changes to the population dynamics of large vertebrates, through loss of top predators [10] or hunting and harvesting of wild herbivores [11], are affecting the distribution and input of carrion resources in some terrestrial ecosystems [6], [12]. This creates a strong imperative to establish a deeper understanding of the links between carrion, biodiversity and ecosystem functioning.\n",
      "\n",
      "Ephemeral resource patches, such as rotting logs and dung pads, occur in ecosystems worldwide and are focal points of nutrient cycling and aggregation of invertebrates [13], [14]. Carrion is distinct from many other ephemeral resources due to its faster decomposition rate, and its patchy spatial occurrence [9]. This means that carrion has very different spatial and temporal dynamic to other ephemeral resources, and suggests that carrion may have a disproportionate effect on biodiversity and nutrient cycling in ecosystems.\n",
      "\n",
      "The decomposition of carrion affects its surroundings through pathways to vertebrate and invertebrate scavengers and detritivores, and their predators, as well as through microbial and fungal decomposers and plants. This generates a temporally dynamic patch with high nutrient concentration and biological activity that contrasts strongly with the nearby environment [5], [15]. To date, studies on the effects of carrion on variation in biodiversity have been restricted to single groups of organisms, such as the detailed examination of successional patterns of arthropods [16], [17], and its localised effects on soil nitrogen or plant growth and composition [15], [18]. This has resulted in a detailed understanding of the specialised role that many insect groups have in carrion decomposition, including the critical role of fly larvae [19], [20], [21], and the complex food webs of associated predators and detritivores [16], [22], [23], [24]. Studies on soil and plant responses have demonstrated that the significant spike in soil nitrogen at a carcass can be detected in nearby plant tissue [15], [25], leading to changes in plant growth and composition [18], [26], and is likely moderated by soil microbial activity [5]. Key studies have also shown the importance of carrion for soil nitrogen and plant dynamics across whole ecosystems [27], [28]. Nevertheless, there are only a handful of comprehensive studies that have integrated multiple groups of organisms into the one study design [3], and there are fewer still that have specifically examined the functional traits of different species that respond to carrion. This leaves much to learn about what traits characterise carrion opportunists, and therefore what traits of species can help us understand the links between ephemeral carrion patches and its effect on variation in biodiversity within landscapes [9].\n",
      "\n",
      "We present data from an experiment designed to answer a number of questions about how ephemeral carrion resource patches generates variation in ecological communities. We did this by quantifying the responses of different aspects of the carrion food web, including components of the insect community, soil nutrients, and the plant community. We sought to identify the traits associated with insect responses to the patchiness of carrion, and the responses of plants to subsequent changes in soil nutrients. To do this, we first examined changes in assemblage diversity and composition in response to the experimental addition of animal carcasses. Second, we examined whether assemblage changes corresponded with particular combinations of traits, for both insects and plants. We predicted that insect responses would be associated with their dispersal ability, with specialist species having wing morphologies that enable the rapid location of new carrion resources. We also predicted that plant responses would be associated with their growth potential and ability to exploit elevated soil nutrient levels. Our experiment compared soil, plant and insect responses between paired carrion and control sites with spatial replication. The results of our analysis therefore provide new insight into the magnitude of the spatial contrast between carrion and its immediate surroundings. This allowed us to examine how the spatial contrast changed over time, and how carrion contributes to both spatial and temporal variation in biodiversity and ecological processes within a landscape.\n",
      "\n",
      "\n",
      "Methods\n",
      "\n",
      "\n",
      "Ethics Statement\n",
      "\n",
      "Insects were sampled under an ACT government ‘Licence to take’ (LT2010417).\n",
      "\n",
      "\n",
      "Study Area and Experimental Design\n",
      "\n",
      "In November 2010 (southern hemisphere spring), we added carcasses of kangaroos to 18 sites in temperate Eucalyptus woodland near Canberra, Australia (149°10′ East, 35°10′ South). This study area is part of the Mulligans Flat-Goorooyarroo Woodland Experiment, a long-term ecological restoration project [29]. Specific details on temperature, rainfall, floristic composition and soil types from this study area have been described by McIntyre et al [30]. We sourced carcasses of eighteen eastern grey kangaroos (Macropus giganteus) over a 10 day period from road kill across the city of Canberra, and positioned them at sites within 12 hours of death. Note that no animals were killed for the purposes of this study. Carcasses were only used when there was not significant blood loss. Average mass of the kangaroos was 31 kg (range 17–80 kg). Each carcass was paired with a control site 10 metres away to allow spatial comparisons between carcass and control treatments through time. We positioned each carcass (and control site) at the base a young Eucalyptus blakelyi tree (<2 m tall at beginning of experiment). Our study area has a history of livestock grazing, resulting in some sites being dominated by exotic grass species [30], [31], [32]. We placed half the carcasses in native-dominated grassland (Themeda sp. and Austrodanthonia sp.) and half in exotic-dominated grassland (Phalaris sp. and Holcus sp.).\n",
      "\n",
      "\n",
      "Soil and Plant Data Collection\n",
      "\n",
      "We took a 4 cm deep soil sample (approximately 4 cm diameter) using a trowel at each carcass and control site at one week (week 0) before carcass addition, and then 12 and 26 weeks after carcass addition. Samples were taken from near the centre of each carcass. Total N content of soil and leaf samples (see below) was quantified with high temperature combustion using a LECA2000 CNS analyser [33]. We performed two analyses of total N, and used the average in our calculations. The potential mineralisable soil nitrogen (organic nitrogen) was quantified using the Illinois soil nitrogen test for amino sugar [34].\n",
      "\n",
      "We took samples of between five and ten leaves from the tips of branches (i.e. active growth zones) in each E. blakelyi tree at carcass and control sites at weeks 12 and 26. No leaves were taken at week 0. Leaf nitrogen content was determined as per soil nitrogen (see above). A higher nitrogen content in leaves indicates potentially greater photosynthetic capacity, and therefore growth potential [35]. Each carcass was positioned on top of existing grasses and forbs, but these were killed off during the early stages of decay. We examined plant re-colonization of carcass patches after 52 weeks by recording the presence of grass and forb species inside a 40×40 cm quadrat at (i) the centre and (ii) edge of each carcass patch, and at (iii) the paired control site. The edge of each carcass was able to be delineated by estimation of the size of the original carcass, and the presence of a ‘halo’ of plant growth. In addition to these standardised quadrats, we also recorded the presence of all other plant species present in the remainder of a 40 cm deep perimeter around each carcass (the ‘surrounding’ vegetation) and the entire carcass patch to compile a complete inventory of grass and forb species. We collected data on plant life forms and specific leaf area (SLA) from the literature [36], [37], [38], and these are summarised in Appendix S2. SLA is the one-sided area of a fresh leaf divided by its dry mass (mm2.mg−1), and is indicative of potential relative growth rate [36]. SLA has also been shown not to vary significantly within plant species [37], thus justifying our use of data from the literature. We used the mean SLA value of each plant species presented in each published account, and excluded rare species for which data were unavailable.\n",
      "\n",
      "\n",
      "Insect Data Collection\n",
      "\n",
      "We sampled insects using a pair of pitfall traps at each carcass and control site. We chose ants and beetles because they are an abundant and diverse components of carrion arthropod communities [3], [16], [39], and yet differ strongly in their dispersal ability and resource use. We recognise that flies are an important part of carrion arthropod communities [16], but we wanted to focus on two diverse groups with contrasting dispersal ability. Many beetles are flight-capable and have high resource specialisation [40], [41], with some specialist species requiring carrion to lay eggs and complete their life cycle. In contrast, ants are generally restricted to the ground (after post-mating dispersal), and may opportunistically scavenge from carcasses or predate on the other insects present. For these reasons, we expected beetles and ants to show very different responses to carrion. Pitfall traps were 9 cm in diameter, and were half filled with glycol solution. The traps were opened for seven days during weeks 1, 6, 12 and 26 of the experiment, which represented distinct stages of decomposition of the carcasses (see Appendix S1). We pooled pitfall trap pairs, giving n = 36 at each sampling time. All insects were extracted from the pitfall traps, stored in 70% ethanol and sorted to order and family. We further sorted beetles and ants to genus and morphospecies using relevant keys [42], [43].\n",
      "\n",
      "\n",
      "Insect Morphological Measurements\n",
      "\n",
      "We measured body length and wing dimensions of each beetle species for comparison of mean trait values of species present at each carcass at control site. We used body length to estimate the body mass of each species of beetle using family-specific length versus weight power relationships for species of Scarabaeidae, Staphylinidae and Carabidae, and a general Coleoptera relationship for all remaining species [44]. We calculated wing loading [45] by dividing body mass by wing area (length x width), expressed as (mg/mm2). We calculated wing aspect ratio by dividing wing length by width, with higher ratios indicating longer and narrower wings [45]. Ant body size is correlated with microhabitat use and resource discovery rates [46], and we regarded this as a useful measure of ant dispersal ability. We measured ant head width and rear femur length, and multiplied these together to calculate a body size index for each species [47]. All morphological measurements were made on up to three individuals of each species (then averaged) using an ocular micrometer with magnification between 10× and 100×, and resolution between 0.1 and 0.01 mm, depending on the measurement and size of the insect. This approach was taken as we were interested in variation in traits between species rather than within species [48].\n",
      "\n",
      "\n",
      "Data Analysis\n",
      "\n",
      "We were first interested in quantifying the magnitude of the contrast between carcass and control sites. We used repeated measures linear mixed models in GenStat 14 [49] to test for interactive effects of carcass addition and sampling time on total insect abundance, species richness and evenness, as well as soil and plant nitrogen content. Carcass addition and time were treated as fixed factors, and site as a random factor. We used ANOVA, also in GenStat 14, to test for differences in insect and plant species richness and evenness or traits between carcass and control treatments at a particular sampling time, with blocking by carcass-control pair. Plant SLA was log10 transformed, and insect abundance data were log10(x+1) transformed to meet assumptions of normality and homogeneity of variances. Percentage similarity in plant species composition between the edge and centre of carcasses was expressed as the number of shared species divided by the total number (i.e. Jaccard similarity index). We compared similarity between the centre and edge of carcasses because plant recolonisation was more likely to occur from the edge rather than from more distant control sites. We tested for effects of quadrat position (carcass centre, edge or control site) and grassland type (native or exotic) on average SLA across plant species. For this analysis, we excluded rare species that occurred only in one quadrat. We wanted to test whether insect assemblages were more homogenous in composition at carcasses than at control sites due to the widespread dominance of carrion specialists. We used Permutational Analysis of Multivariate Dispersion [50] to test for differences in among-sample heterogeneity of insect species composition between carcass and control treatments, with statistical significance determined from 10 000 permutations of the data. This test calculates the average distance to the centroid of a group of samples projected in multivariate space, with a greater distance to centroid indicating greater among-sample heterogeneity [51]. Multivariate insect abundance data were square-root transformed prior to analysis to reduce the influence of highly abundant species on sample dissimilarity. Shannon evenness was defined as E = H′/lnS, where S is the number of species in a sample, and H′ = −∑ pi ln pi, where pi is the proportion of a species in a sample which is multiplied by the natural logarithm of itself [52].\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "\n",
      "Soil and Leaf Nitrogen\n",
      "\n",
      "The addition of carrion had significant effects on soil nitrogen (Table 1). The soils in our study environment were originally low in phosphorus and nitrogen compared with grasslands improved for grazing [31], and thus have low productivity potential. The highly concentrated nutrients in each carcass (a 30 kg vertebrate carcass contains approximately one kilogram of nitrogen [5]) produced a dynamic change in soil nitrogen. We found that total soil nitrogen beneath carcasses was 200% higher than control sites after 12 weeks, and 70% higher than control sites after 26 weeks of carrion being added (Fig. 1A). In addition, the concentration of the organic fraction of soil nitrogen was 460% higher than control sites after 12 weeks, and 290% higher after 26 weeks (Fig. 1B). This nutrient pulse also was detected in plant leaf nitrogen levels from nearby tree saplings, which were 18% higher in plants near to carcasses than control sites after 12 weeks (Fig. 1C). This demonstrates the cycling of nitrogen from carcasses into soil, and the uptake and assimilation of soil nitrogen into leaf tissue during the first few weeks of decomposition.\n",
      "\n",
      "10.1371/journal.pone.0053961.g001Figure 1 Effect of carrion on soil and foliar nitrogen.Carrion increased (a) soil total nitrogen, (b) soil organic nitrogen, and (c) tree foliar total nitrogen relative to control sites. Error bars show mean ± SE.\n",
      "\n",
      "Table 1\n",
      "Effects\tF\td.f.\tP\n",
      "Soil total nitrogen (%)\tTime\t55.85\t2\t<0.001\n",
      "Treatment\t29.36\t1\t<0.001\n",
      "Time×Treatment\t37.59\t2\t<0.001\n",
      "Soil organic nitrogen (µg/kg)\tTime\t136.31\t2\t<0.001\n",
      "Treatment\t26.68\t1\t<0.001\n",
      "Time×Treatment\t107.28\t2\t<0.001\n",
      "Tree foliar nitrogen (%)\tTime\t29.73\t1\t<0.001\n",
      "Treatment\t12.71\t1\t<0.001\n",
      "Time×Treatment\t0.26\t1\t0.609\n",
      "\n",
      "\n",
      "Plant Assemblage Richness and Composition\n",
      "\n",
      "Plant recolonisation of carcasses was faster in the exotic-dominated grassland where, after 52 weeks, the composition of plants in the centre of each carcass patch more closely matched the composition of species at the edge of each carcass patch in the exotic-dominated grassland (56% shared species) than in native-dominated grassland (37% shared species). We found a significant interaction between carcass treatment and grassland type (F2 = 11.48, P<0.001) and their effect on plant species richness at carcass patches after 52 weeks. Plant species richness was similar across carcass and control sites in the exotic grassland, but the number of species in the centre of carcass patches in native grassland was only a third that of species at the edge of carcass patches or at control sites (Fig. 2A). This result is further highlighted by our observation that four of the nine native-dominated carcass patches had no re-colonising plants at all, whereas all nine of the exotic-dominated carcass patches had re-colonising plants 52 week after carrion was added (compare Fig. 3A with 3B). The rapid response of plants in the exotic grassland was associated with a significantly higher specific leaf area among the constituent plant species (Fig. 2B, F1 = 18.03, P<0.001). Regardless of grassland type, plants recolonising the centre of carcass patches tended to be annuals rather perennials, and graminoids rather than forbs (Fig. 2C, 2D).\n",
      "\n",
      "10.1371/journal.pone.0053961.g002Figure 2 Differences in plant species richness and traits between carcass centre, carcass edge, and control sites.(a) Species richness of plants species in exotic grassland compared with native grassland. (b) Specific leaf area of plants in exotic grassland compared with native grassland. (c) Percentage of annuals versus perennials. (d) Percentage of graminoids versus forbs. Error bars show mean ± SE.\n",
      "\n",
      "10.1371/journal.pone.0053961.g003Figure 3 Recolonisation of plants 52 weeks after carrion addition.Very little recolonisation had occurred in the native-dominated grassland sites (a), whereas re-colonisation had progressed further in all exotic-dominated grassland sites (b), reflecting greater tolerance of some exotic species to extremely high nutrients. These examples indicate a different succession trajectory for native versus exotic grassland plant assemblages, with recolonisation occurring over different time scales.\n",
      "\n",
      "\n",
      "Insect Assemblage Diversity and Composition\n",
      "\n",
      "We sampled a total of 33 ant species (3330 individuals) and 119 beetle species (3614 individuals) from all carcass and control sites over the four sampling times (Appendix S3 and S4). Beetle and ant assemblages displayed contrasting dynamic responses to carrion. We found that beetle assemblages had a significantly higher species richness and lower evenness during early stages of decay, with a gradual return to levels similar to control sites by week 26 (Fig. 4A, Table 2). This contrasted strongly with ant assemblages, which initially had lower species richness at carcass sites, and did not show the same decrease in evenness as the beetle assemblage (Fig. 4B). We found an overall significant difference in among-sample heterogeneity from carcass and control sites at each sample times for both beetles (F = 21.11, P<0.001) and ants (F = 11.81, P<0.001). However, comparisons between carcass and control sites for each sampling time showed distinctive and contrasting patterns for the beetles and ants (Fig. 5). In the early weeks of decomposition, the composition of beetle assemblages at carrion sites was less variable than at control sites, indicating spatial monopolisation of carcass patches by a few abundant species (Fig. 5A). This changed as decay progressed, with the beetle assemblage becoming more heterogeneous at carrion sites relative to control sites. This was due to widespread species at carcasses becoming less dominant, while assemblages at control sites showed a seasonal decrease in heterogeneity (Fig. 5A). In contrast, there was never a difference in ant assemblage heterogeneity between carcass and control sites (Fig. 5B), with the overall significant difference among groups due a decrease in heterogeneity in week 26 in late autumn. This indicated there was no spatial monopolisation of carcasses by a few ant species, in contrast to that which we observed for beetles.\n",
      "\n",
      "10.1371/journal.pone.0053961.g004Figure 4 Effect of carrion on beetle and ant diversity.(a) Species richness and evenness of beetles at carcasses showed a trajectory through time clearly different from control sites at weeks 1, 6 and 12, with a return to similar levels at week 26 (blue arrows). (b) Species richness of ants at carcasses was half that at control sites during week 1 (red arrows), but was then similar to control sites at week 6, 12 and 26 of the experiment. Error bars show mean ± SE.\n",
      "\n",
      "10.1371/journal.pone.0053961.g005Figure 5 Effect of carrion on insect assemblage heterogeneity.Average distance to group centroid, reflecting heterogeneity in composition among samples of (a) beetles and (b) ants from carrion and control treatments. Error bars show mean ± SE.\n",
      "\n",
      "Table 2\n",
      "Effects\tF\td.f.\tP\n",
      "(A)\tAbundance\tTime\t48.81\t3\t<0.001\n",
      "Treatment\t3.25\t1\t0.080\n",
      "Time×Treatment\t4.55\t3\t0.009\n",
      "Species richness\tTime\t26.05\t3\t<0.001\n",
      "Treatment\t3.35\t1\t0.076\n",
      "Time×Treatment\t4.51\t3\t0.009\n",
      "Evenness\tTime\t44.38\t3\t<0.001\n",
      "Treatment\t1.08\t1\t0.322\n",
      "Time×Treatment\t1.41\t3\t0.264\n",
      "(B)\tAbundance\tTime\t40.09\t3\t<0.001\n",
      "Treatment\t230.92\t1\t<0.001\n",
      "Time×Treatment\t19.05\t3\t<0.001\n",
      "Species richness\tTime\t21.46\t3\t<0.001\n",
      "Treatment\t70.07\t1\t<0.001\n",
      "Time×Treatment\t11.51\t3\t<0.001\n",
      "Evenness\tTime\t50.49\t3\t<0.001\n",
      "Treatment\t28.86\t1\t<0.001\n",
      "Time×Treatment\t12.83\t3\t<0.001\n",
      "\n",
      "\n",
      "Insect Dispersal Traits\n",
      "\n",
      "The contrasting response of beetle and ant assemblages was associated with their different dispersal abilities, with significant differences in morphological traits across beetle species but not ant species (Table 3). Beetle species at carrion were larger than species at control sites one week after carrion was added (Fig. 6A), whereas ants showed no significant difference in body size at any of the four sampling times (Fig. 6B). Beetles at carcasses also had significantly higher wing loading, but only at one week after carrion was added (Fig. 6C). We also found that a higher percentage of beetles at carcasses had functional wings compared with beetles at control sites (Fig. 6D).\n",
      "\n",
      "10.1371/journal.pone.0053961.g006Figure 6 Differences in insect morphological traits between carcass and control sites.(a) Beetle species at carrion during week 1 were larger on average than those at control sites. (b) Beetle species at carrion during week 1 had higher wing loadings on average than beetle species at control sites. (c) No differences in ant body size were observed at different stages of carrion decomposition. (d) Macropterous beetles were always more prevalent at carcass sites that control sites. Error bars show mean ± SE.\n",
      "\n",
      "Table 3\n",
      "Effect\tF\td.f.\tP\n",
      "(A) Ant body size\tTime\t0.99\t3\t0.395\n",
      "Treatment\t0.09\t1\t0.769\n",
      "Time×Treatment\t0.62\t3\t0.600\n",
      "(B) Beetle body length\tTime\t12.42\t3\t<0.001\n",
      "Treatment\t39.37\t1\t<0.001\n",
      "Time×Treatment\t8.82\t3\t<0.001\n",
      "(C) Beetle wing loading\tTime\t9.25\t3\t<0.001\n",
      "Treatment\t1.81\t1\t0.179\n",
      "Time×Treatment\t1.64\t3\t0.180\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "Our results show that carrion in ecosystems represents a dynamic driver of species richness and nutrient cycling. This is supported by our observations, which showed elevated nitrogen in soil and plant leaf tissue at carrion patches, different rates of recolonization by plants, and a diverse and functionally distinct set of insects occurring at carcasses compared with control sites. Importantly, we have shown that the pattern of response by insects and plants was associated with particular functional traits that allow them to respond to and then exploit carrion. This builds on previous work that has demonstrated separately the effects of carrion on soil nutrient heterogeneity [53], scavenger community structure [16], [54], and plant composition [18], and develops further our understanding of how carrion generates spatial and temporal variation in ecological communities.\n",
      "\n",
      "We found that the nitrogen spike in soils translated into an increase in leaf nitrogen in the nearby E. blakelyi trees, indicating the movement of nutrients from the decomposing carcasses into live plant tissue via the soil subsystem [15], [25]. Trees in the genus Eucalyptus are not nitrogen-fixing and must use available nitrogen in the soil. This suggests that the increase in soil nitrogen due to carcass decomposition was actively taken up by the trees we sampled. The differences we detected in foliage nitrogen between carcass and control sites was smaller than that reported from analogous experimental studies in North America [15], [28], but this may be due to the very different plant species involved, as well as underlying microbial and soil properties. The dramatic change in soil nitrogen also resulted in a greater dominance of grass species with high average specific leaf area. It is well established that these functional plant types typically respond to elevated nitrogen [31]. This reflects observations in a tallgrass prairie ecosystem of North America, where forbs decreased in cover at the centre of ungulate carcasses after 1 year, but grasses increased in cover [18]. This is evidence of in situ organisms displaying trait-dependent responses to new nutrient-rich resource patches. The effects of carcasses on the diversity and composition of ground-layer plant assemblages depended on whether the carcasses were placed in native- or exotic-dominated grassland. Plants at the carcass treatment sites were initially killed by the carcasses due to prolonged shading by carcass remains and rapid soil chemistry changes, but later recolonised from the edges. The quick recolonization of carrion patches by plants in the exotic grassland was largely due to seeding, except for one species that used stolons (Cynodon dactylon). This underlines a context-dependent effect of carrion not previously described at local scales. This parallels a pattern observed for the effects of migrating salmon (Oncorhynchus spp.) carcasses on plants in North America, with effects reduced in the more highly productive catchments [27]. Together the results suggest a general principle of context-dependency resulting in temporally attenuated effects of carrion in high productivity ecosystems or in habitats with a high proportion of nutrient-responsive plant species.\n",
      "\n",
      "We showed that dispersal traits of insects were associated with their responses to ephemeral resource opportunities. The distinct dispersal traits of beetles compared with ants was likely to explain, in part, their contrasting dynamics in assemblage composition, and led to their changes in spatial heterogeneity during different stages of carcass decomposition. Dispersal plays a key role in insect spatial diversity patterns [55], and our study provides a further example relevant to carrion. Although carrion affected the relative abundance of ants present in the landscape, it did not affect species richness or spatial heterogeneity. The lack of flight to facilitate rapid foraging means that ants are generally less able to move as rapidly as flight-capable beetles, regardless of any resource specialisation they might have. Although we found that some ant species dominated individual carcasses, such as the meat ant Iridomyrmex purpureus, there were no species that monopolised all the carcasses in the same way some beetle species did. This result is due, in part, to the lack of strong specialisation of ants on carrion resources, with scavenging more likely to be opportunistic. For beetles, capacity for flight and higher wing-aspect ratios were associated with beetles at carrion generally. A larger body size and higher wing loading were important functional traits that were associated with the early colonisation of fresh carrion and spatial monopolisation of this resource. Beetles visiting fresh carcasses were typically equipped with wings that enable manoeuvrable flight [45], suggesting that they can actively seek carrion, and with their large body size have the metabolic capacity to do so over greater distances than smaller beetles. These traits are exemplified by the carrion specialists Ptomaphila lacrysoma and Creophilus erythrocephalus, which are well known to frequent carrion in south-eastern Australia [56], [57]. These species, as well as several others, were not recorded at control sites (Appendix S4), and have not been recorded as part of previous research in the study area [48]. This highlights the unique composition of carrion insect assemblages. Ants at carrion displayed no characteristic morphologies during different stages of carrion decay, and this matched their lack of difference in spatial heterogeneity between carcasses or control sites. Although we focused on morphological traits related to dispersal ability, it is important to point out that the feeding traits of insects are equally important in determining their responses to carrion [16]. For example, the carrion specialists mentioned above also feed on the flesh of carcasses or prey on the insects that do. Their early arrival at a carcass is therefore critical if they are to exploit a resource that will quickly be exhausted. Further research is needed to test if dispersal limitation or small body size actually prevents some species from exploiting carrion, as this would provide a mechanistic link between morphology and assemblage responses to carrion resources.\n",
      "\n",
      "Together, our findings provide an integrated example of how traits of opportunist species in two different taxonomic groups are associated with their ability to exploit patchy and dynamic resources. This adds to the literature on carrion resource specialisation traits and temporal assemblage dynamics [16], [23], to give a fuller picture of the processes that drive spatial changes in species composition. It also supports previous claims that carrion might need to be managed in some ecosystems to maintenance of biodiversity and ecological processes [9], [58]. This includes, for example, the role of carrion in the persistence of dependent insects [59] or the nutrient opportunities it can provide to plants [15]. Our study indicates that appropriate consideration of spatial and temporal continuity in carrion will enable insects of variable dispersal ability to exploit this resource, and will promote spatial heterogeneity in nutrient cycling and plant communities.\n",
      "\n",
      "\n",
      "Supporting InformationAppendix S1\n",
      "\n",
      "Stages of decomposition when insects were sampled.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.Appendix S2\n",
      "\n",
      "Summary of plant species and their traits.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.Appendix S3\n",
      "\n",
      "Summary of ant species.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.Appendix S4\n",
      "\n",
      "Summary of beetle species.\n",
      "\n",
      "(DOCX)\n",
      "\n",
      "Click here for additional data file.\n"
     ]
    }
   ],
   "source": [
    "# To sample an article with text matches from our terms:\n",
    "art = articles.aggregate([\n",
    "    { \"$match\": { \"text_matches\": { \"$in\": terms } } },\n",
    "    { \"$sample\": { \"size\": 1 } }\n",
    "]).next()\n",
    "\n",
    "# Print the ID, metadata, matches, and extracted text of your article\n",
    "print(\"\"\"\n",
    "PMC ID\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TITLE\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "METADATA\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "TEXT MATCHES\n",
    "\n",
    "{}\n",
    "\n",
    "\n",
    "EXTRACTED TEXT\n",
    "\n",
    "{}\"\"\".format(art.get(\"_id\"),\n",
    "             art.get(\"article_title\"),\n",
    "             art.get(\"article_meta\"),\n",
    "             art.get(\"text_matches\"),\n",
    "             art.get(\"extracted_text\").strip()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "soup = BeautifulSoup(art[\"xml\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3573865'"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "soup.front.find(\"article-id\", attrs={\"pub-id-type\": \"pmc\"}).get_text()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'research-article'"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.article_type()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'PLoS ONE'"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.soup.find(\"front\").find(\"journal-meta\").find(\"journal-title\").get_text()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Methods in Ecology and Evolution'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.journal_title()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Advances in multiplex PCR: balancing primer efficiencies and improving detection success'"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.article_title()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DataFrame scratch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "foo = pd.DataFrame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>one</th>\n",
       "      <th>two</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>dog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   one  two\n",
       "0   10  dog"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame({\"one\": [10], \"two\": [\"dog\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>one</th>\n",
       "      <th>two</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>dog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   one  two\n",
       "0   10  dog"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame.from_records([{\"one\": 10, \"two\": \"dog\"}])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   A\n",
       "0  0\n",
       "1  1\n",
       "2  2\n",
       "3  3\n",
       "4  4"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.concat([pd.DataFrame([i], columns=['A']) for i in range(5)],ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.concat([foo])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Count article types and other metadata"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO: Check if an article's `_id` is in `result_set` and add those to a different default_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "211"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.count_documents({})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-15-ff577f9bb7c5>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcursor\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marticles\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maggregate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m{\u001b[0m \u001b[0;34m\"$sample\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m{\u001b[0m \u001b[0;34m\"size\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0marticles\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcount_documents\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m}\u001b[0m \u001b[0;34m}\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/collection.py\u001b[0m in \u001b[0;36mcount_documents\u001b[0;34m(self, filter, session, **kwargs)\u001b[0m\n\u001b[1;32m   1683\u001b[0m         \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_socket_for_reads\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msession\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0msock_info\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mslave_ok\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1684\u001b[0m             result = self._aggregate_one_result(\n\u001b[0;32m-> 1685\u001b[0;31m                 sock_info, slave_ok, cmd, collation, session)\n\u001b[0m\u001b[1;32m   1686\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1687\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/collection.py\u001b[0m in \u001b[0;36m_aggregate_one_result\u001b[0;34m(self, sock_info, slave_ok, cmd, collation, session)\u001b[0m\n\u001b[1;32m   1585\u001b[0m             \u001b[0mread_concern\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_concern\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1586\u001b[0m             \u001b[0mcollation\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcollation\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1587\u001b[0;31m             session=session)\n\u001b[0m\u001b[1;32m   1588\u001b[0m         \u001b[0mbatch\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'cursor'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'firstBatch'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1589\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mbatch\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/collection.py\u001b[0m in \u001b[0;36m_command\u001b[0;34m(self, sock_info, command, slave_ok, read_preference, codec_options, check, allowable_errors, read_concern, write_concern, collation, session, retryable_write)\u001b[0m\n\u001b[1;32m    242\u001b[0m                 \u001b[0msession\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    243\u001b[0m                 \u001b[0mclient\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__database\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclient\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 244\u001b[0;31m                 retryable_write=retryable_write)\n\u001b[0m\u001b[1;32m    245\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    246\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__create\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moptions\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcollation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/pool.py\u001b[0m in \u001b[0;36mcommand\u001b[0;34m(self, dbname, spec, slave_ok, read_preference, codec_options, check, allowable_errors, check_keys, read_concern, write_concern, parse_write_concern_error, collation, session, client, retryable_write, publish_events)\u001b[0m\n\u001b[1;32m    582\u001b[0m         \u001b[0;31m# Catch socket.error, KeyboardInterrupt, etc. and close ourselves.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    583\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mBaseException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 584\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_raise_connection_failure\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merror\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    585\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    586\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0msend_message\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmessage\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmax_doc_size\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/pool.py\u001b[0m in \u001b[0;36m_raise_connection_failure\u001b[0;34m(self, error)\u001b[0m\n\u001b[1;32m    743\u001b[0m             \u001b[0m_raise_connection_failure\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maddress\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    744\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 745\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    746\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    747\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__eq__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/pool.py\u001b[0m in \u001b[0;36mcommand\u001b[0;34m(self, dbname, spec, slave_ok, read_preference, codec_options, check, allowable_errors, check_keys, read_concern, write_concern, parse_write_concern_error, collation, session, client, retryable_write, publish_events)\u001b[0m\n\u001b[1;32m    577\u001b[0m                            \u001b[0mcompression_ctx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompression_context\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    578\u001b[0m                            \u001b[0muse_op_msg\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mop_msg_enabled\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 579\u001b[0;31m                            unacknowledged=unacknowledged)\n\u001b[0m\u001b[1;32m    580\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mOperationFailure\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    581\u001b[0m             \u001b[0;32mraise\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/network.py\u001b[0m in \u001b[0;36mcommand\u001b[0;34m(sock, dbname, spec, slave_ok, is_mongos, read_preference, codec_options, session, client, check, allowable_errors, address, check_keys, listeners, max_bson_size, read_concern, parse_write_concern_error, collation, compression_ctx, use_op_msg, unacknowledged)\u001b[0m\n\u001b[1;32m    139\u001b[0m             \u001b[0mresponse_doc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m\"ok\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    140\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 141\u001b[0;31m             \u001b[0mreply\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mreceive_message\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msock\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    142\u001b[0m             \u001b[0munpacked_docs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mreply\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munpack_response\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcodec_options\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcodec_options\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/network.py\u001b[0m in \u001b[0;36mreceive_message\u001b[0;34m(sock, request_id, max_message_size)\u001b[0m\n\u001b[1;32m    171\u001b[0m     \u001b[0;31m# Ignore the response's request id.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    172\u001b[0m     length, _, response_to, op_code = _UNPACK_HEADER(\n\u001b[0;32m--> 173\u001b[0;31m         _receive_data_on_socket(sock, 16))\n\u001b[0m\u001b[1;32m    174\u001b[0m     \u001b[0;31m# No request_id for exhaust cursor \"getMore\".\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    175\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mrequest_id\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/pymongo/network.py\u001b[0m in \u001b[0;36m_receive_data_on_socket\u001b[0;34m(sock, length)\u001b[0m\n\u001b[1;32m    230\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0mbytes_read\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0mlength\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    231\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 232\u001b[0;31m                 \u001b[0mchunk_length\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmv\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mbytes_read\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    233\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mIOError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mOSError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mexc\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    234\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0m_errno_from_exception\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexc\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0merrno\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mEINTR\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "cursor = articles.aggregate([{ \"$sample\": { \"size\": articles.count_documents({}) } }])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "count = articles.count_documents({})\n",
    "cursor = articles.find({})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[F\u001b[KProcessed 175 articles (82.9%) in 0m8s; about 0m2s left.\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-46-1f74dba360bf>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0mrow\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m     \u001b[0marticle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mArticle\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"xml\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     10\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m     \u001b[0mrow\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"id\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"_id\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Dropbox (EHA)/repositories/pubcrawler/pubcrawler/article.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, xml)\u001b[0m\n\u001b[1;32m     13\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mxml\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mxml\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mxml\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 15\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msoup\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mBeautifulSoup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mxml\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'lxml-xml'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     16\u001b[0m         \u001b[0;31m# self._id = article_dict['_id']\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     17\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, markup, features, builder, parse_only, from_encoding, exclude_encodings, **kwargs)\u001b[0m\n\u001b[1;32m    279\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    280\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 281\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_feed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    282\u001b[0m                 \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    283\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mParserRejectedMarkup\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36m_feed\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    340\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuilder\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    341\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 342\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuilder\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmarkup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    343\u001b[0m         \u001b[0;31m# Close out any unfinished strings and close all the open tags.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    344\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mendData\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/builder/_lxml.py\u001b[0m in \u001b[0;36mfeed\u001b[0;34m(self, markup)\u001b[0m\n\u001b[1;32m    166\u001b[0m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmarkup\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mCHUNK_SIZE\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    167\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 168\u001b[0;31m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    169\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    170\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mUnicodeDecodeError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mLookupError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0metree\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mParserError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parser.pxi\u001b[0m in \u001b[0;36mlxml.etree._FeedParser.feed\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parser.pxi\u001b[0m in \u001b[0;36mlxml.etree._FeedParser.feed\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._TargetParserContext._handleParseResult\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._TargetParserContext._handleParseResult\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/etree.pyx\u001b[0m in \u001b[0;36mlxml.etree._ExceptionContext._raise_if_stored\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/saxparser.pxi\u001b[0m in \u001b[0;36mlxml.etree._handleSaxTargetStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/saxparser.pxi\u001b[0m in \u001b[0;36mlxml.etree._callTargetSaxStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._PythonSaxParserTarget._handleSaxStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/builder/_lxml.py\u001b[0m in \u001b[0;36mstart\u001b[0;34m(self, name, attrs, nsmap)\u001b[0m\n\u001b[1;32m    218\u001b[0m         \u001b[0mnamespace\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getNsTag\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    219\u001b[0m         \u001b[0mnsprefix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_prefix_for_namespace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnamespace\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 220\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msoup\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhandle_starttag\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnamespace\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnsprefix\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mattrs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    221\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    222\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_prefix_for_namespace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnamespace\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36mhandle_starttag\u001b[0;34m(self, name, namespace, nsprefix, attrs)\u001b[0m\n\u001b[1;32m    528\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    529\u001b[0m         tag = Tag(self, self.builder, name, namespace, nsprefix, attrs,\n\u001b[0;32m--> 530\u001b[0;31m                   self.currentTag, self._most_recent_element)\n\u001b[0m\u001b[1;32m    531\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mtag\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    532\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mtag\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/element.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, parser, builder, name, namespace, prefix, attrs, parent, previous, is_xml)\u001b[0m\n\u001b[1;32m    859\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mattrs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mattrs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    860\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcontents\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 861\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mparent\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprevious\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    862\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhidden\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    863\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/element.py\u001b[0m in \u001b[0;36msetup\u001b[0;34m(self, parent, previous_element, next_element, previous_sibling, next_sibling)\u001b[0m\n\u001b[1;32m    260\u001b[0m         \"\"\"Sets up the initial relations between this element and\n\u001b[1;32m    261\u001b[0m         other elements.\"\"\"\n\u001b[0;32m--> 262\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparent\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparent\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    263\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    264\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprevious_element\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprevious_element\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "reporter = Reporter(25, len(matches))\n",
    "\n",
    "rows = []\n",
    "\n",
    "for idx, record in enumerate(matches):\n",
    "    reporter.report(idx)\n",
    "\n",
    "    row = {}\n",
    "    article = Article(record[\"xml\"])\n",
    "\n",
    "    row[\"id\"] = record[\"_id\"]\n",
    "    row[\"keywords\"] = article.keywords()\n",
    "    row[\"id_types\"] = list(article.pub_ids().keys())\n",
    "    row[\"article_type\"] = article.article_type()\n",
    "    row[\"has_body\"] = True if article.soup.body else False\n",
    "    row[\"text_matches\"] = record.get(\"text_matches\")\n",
    "    row[\"article_title\"] = article.article_title()\n",
    "    row[\"journal_title\"] = article.journal_title()\n",
    "    row[\"text_length\"] = len(record[\"extracted_text\"])\n",
    "\n",
    "    rows.append(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "article_df = pd.DataFrame.from_records(rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "any_matches\n",
       "False    19519\n",
       "True       481\n",
       "dtype: int64"
      ]
     },
     "execution_count": 282,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article_df[\"any_matches\"] = [False if row is None else True for row in article_df[\"text_matches\"]]\n",
    "article_df.groupby([\"any_matches\"]).size()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "metadata": {},
   "outputs": [],
   "source": [
    "article_df.to_csv(\"article_sample.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>any_matches</th>\n",
       "      <th>article_type</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>True</td>\n",
       "      <td>research-article</td>\n",
       "      <td>446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>True</td>\n",
       "      <td>review-article</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>True</td>\n",
       "      <td>brief-report</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>True</td>\n",
       "      <td>protocol</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>True</td>\n",
       "      <td>case-report</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>True</td>\n",
       "      <td>other</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>True</td>\n",
       "      <td>data-paper</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>True</td>\n",
       "      <td>editorial</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>True</td>\n",
       "      <td>letter</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>False</td>\n",
       "      <td>research-article</td>\n",
       "      <td>14138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>False</td>\n",
       "      <td>review-article</td>\n",
       "      <td>1279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>False</td>\n",
       "      <td>case-report</td>\n",
       "      <td>896</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>False</td>\n",
       "      <td>other</td>\n",
       "      <td>745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>abstract</td>\n",
       "      <td>507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>False</td>\n",
       "      <td>editorial</td>\n",
       "      <td>314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>False</td>\n",
       "      <td>brief-report</td>\n",
       "      <td>311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>False</td>\n",
       "      <td>book-review</td>\n",
       "      <td>306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>False</td>\n",
       "      <td>letter</td>\n",
       "      <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>False</td>\n",
       "      <td>correction</td>\n",
       "      <td>245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>article-commentary</td>\n",
       "      <td>92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>False</td>\n",
       "      <td>news</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>False</td>\n",
       "      <td>meeting-report</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>False</td>\n",
       "      <td>discussion</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>False</td>\n",
       "      <td>product-review</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>False</td>\n",
       "      <td>obituary</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>False</td>\n",
       "      <td>protocol</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>False</td>\n",
       "      <td>in-brief</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>False</td>\n",
       "      <td>rapid-communication</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>False</td>\n",
       "      <td>methods-article</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>False</td>\n",
       "      <td>retraction</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>False</td>\n",
       "      <td>systematic-review</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>False</td>\n",
       "      <td>introduction</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>False</td>\n",
       "      <td>reply</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>False</td>\n",
       "      <td>data-paper</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>addendum</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>False</td>\n",
       "      <td>books-received</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>announcement</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>False</td>\n",
       "      <td>oration</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>False</td>\n",
       "      <td>report</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>case-report</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>False</td>\n",
       "      <td>calendar</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    any_matches         article_type  count\n",
       "39         True     research-article    446\n",
       "40         True       review-article     15\n",
       "32         True         brief-report      7\n",
       "38         True             protocol      4\n",
       "33         True          case-report      3\n",
       "37         True                other      3\n",
       "34         True           data-paper      1\n",
       "35         True            editorial      1\n",
       "36         True               letter      1\n",
       "28        False     research-article  14138\n",
       "30        False       review-article   1279\n",
       "9         False          case-report    896\n",
       "22        False                other    745\n",
       "1         False             abstract    507\n",
       "13        False            editorial    314\n",
       "7         False         brief-report    311\n",
       "5         False          book-review    306\n",
       "16        False               letter    301\n",
       "10        False           correction    245\n",
       "4         False   article-commentary     92\n",
       "19        False                 news     82\n",
       "17        False       meeting-report     67\n",
       "12        False           discussion     42\n",
       "23        False       product-review     31\n",
       "20        False             obituary     29\n",
       "24        False             protocol     27\n",
       "14        False             in-brief     17\n",
       "25        False  rapid-communication     15\n",
       "18        False      methods-article     13\n",
       "29        False           retraction     12\n",
       "31        False    systematic-review     11\n",
       "15        False         introduction      9\n",
       "26        False                reply      9\n",
       "11        False           data-paper      7\n",
       "2         False             addendum      3\n",
       "6         False       books-received      3\n",
       "3         False         announcement      2\n",
       "21        False              oration      2\n",
       "27        False               report      2\n",
       "0         False          case-report      1\n",
       "8         False             calendar      1"
      ]
     },
     "execution_count": 285,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article_df.groupby([\"any_matches\", \"article_type\"]).size()\n",
    "art_types = article_df.groupby([\"any_matches\", \"article_type\"]).size().reset_index()\n",
    "art_types.columns = [\"any_matches\", \"article_type\", \"count\"]\n",
    "art_types.sort_values(by = [\"any_matches\", \"count\"], ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "metadata": {},
   "outputs": [],
   "source": [
    "keep_vars = [\"id\", \"any_matches\"]\n",
    "unnest_var = \"keywords\"\n",
    "\n",
    "def unnest(data, unnest_var, keep_vars):\n",
    "    all_vars = keep_vars + [unnest_var]\n",
    "    nested = article_df.loc[:, keep_vars + [unnest_var]]\n",
    "    lens = [len(item) if item is not None else 1 for item in nested[unnest_var]]\n",
    "    unnested_dict = {var: np.repeat([nested[var].values], lens) for var in keep_vars}\n",
    "    unnested_dict[unnest_var] = np.hstack(nested[unnest_var])\n",
    "    unnested = pd.DataFrame(unnested_dict)\n",
    "    return(unnested)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "metadata": {},
   "outputs": [],
   "source": [
    "keywords = unnest(article_df, \"keywords\", [\"id\", \"any_matches\"])\n",
    "\n",
    "keywords.to_csv(\"keywords.csv\")\n",
    "\n",
    "id_types = unnest(article_df, \"id_types\", [\"id\", \"any_matches\"])\n",
    "\n",
    "id_types.to_csv(\"id_types.csv\")\n",
    "\n",
    "text_matches = unnest(article_df, \"text_matches\", [\"id\"])\n",
    "\n",
    "text_matches.to_csv(\"text_matches.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Old version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processed 240 articles (96.0%) in 0m12s; about 0m1s left.\n"
     ]
    }
   ],
   "source": [
    "id_types = defaultdict(int)\n",
    "article_types = defaultdict(int)\n",
    "keywords = defaultdict(int)\n",
    "\n",
    "reporter = Reporter(10, len(matches))\n",
    "\n",
    "for idx, doc in enumerate(matches):\n",
    "    if idx is not 0 and idx % reporter.interval is 0:\n",
    "        reporter.report(idx)\n",
    "    article = Article(doc[\"xml\"])\n",
    "    \n",
    "    kwds = article.keywords()\n",
    "    if kwds:\n",
    "        for kwd in kwds:\n",
    "            keywords[kwd] += 1\n",
    "    \n",
    "    ids = article.pub_ids().keys()\n",
    "    if len(ids) is 0:\n",
    "        id_types[\"none\"] += 1\n",
    "    else:\n",
    "        for item in ids:\n",
    "            id_types[item] += 1\n",
    "    \n",
    "    article_type = article.article_type()\n",
    "    if article_type:\n",
    "        article_types[article_type] += 1\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "keywords = pd.DataFrame(list(keywords.items()), columns=[\"keyword\", \"count\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>keyword</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>inflammation</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>oxidative stress</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>epidemiology</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>Obesity</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>Prostate cancer</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1220</th>\n",
       "      <td>breast cancer</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>SNP</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1833</th>\n",
       "      <td>cytotoxicity</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>mitochondria</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>depression</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>801</th>\n",
       "      <td>Parkinson’s disease</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>388</th>\n",
       "      <td>Breast cancer</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1566</th>\n",
       "      <td>Prognosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2259</th>\n",
       "      <td>autophagy</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1274</th>\n",
       "      <td>Cancer</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>obesity</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>819</th>\n",
       "      <td>Saccharomyces cerevisiae</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>459</th>\n",
       "      <td>Aging</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>Proliferation</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464</th>\n",
       "      <td>drug resistance</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>466</th>\n",
       "      <td>tuberculosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1320</th>\n",
       "      <td>cancer</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1523</th>\n",
       "      <td>diagnosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1731</th>\n",
       "      <td>apoptosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>PD-1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>694</th>\n",
       "      <td>Depression</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1223</th>\n",
       "      <td>clinical trial</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1762</th>\n",
       "      <td>Apoptosis</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>662</th>\n",
       "      <td>rehabilitation</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1003</th>\n",
       "      <td>Personality disorders</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1004</th>\n",
       "      <td>Carbon footprint</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1005</th>\n",
       "      <td>Economic evaluation</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1006</th>\n",
       "      <td>Online education</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1008</th>\n",
       "      <td>evaluation</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1009</th>\n",
       "      <td>star trackers</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1010</th>\n",
       "      <td>slew tolerance</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1011</th>\n",
       "      <td>attitude availability</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1012</th>\n",
       "      <td>star detection</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>celiac disease</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>Helicobacter pylori</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>immunoglobulin A</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>981</th>\n",
       "      <td>data association</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>974</th>\n",
       "      <td>[18F]FP-(+)-DTBZ</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>975</th>\n",
       "      <td>PET</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>Test-retest</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>977</th>\n",
       "      <td>Cryptococcosis</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>978</th>\n",
       "      <td>Pulmonary Tuberculosis</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>Coinfection</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>980</th>\n",
       "      <td>multi-object tracking</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>982</th>\n",
       "      <td>deep neural network</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>lysosomal enzymes</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983</th>\n",
       "      <td>long short-term memory network</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>nanoparticles</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>turkey</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>aminopeptidases</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>poultry meat</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>988</th>\n",
       "      <td>nutrition</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989</th>\n",
       "      <td>supplementation</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2796</th>\n",
       "      <td>argonaute</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2797 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             keyword  count\n",
       "43                      inflammation      6\n",
       "587                 oxidative stress      5\n",
       "159                     epidemiology      5\n",
       "183                          Obesity      4\n",
       "279                  Prostate cancer      4\n",
       "1220                   breast cancer      4\n",
       "647                              SNP      4\n",
       "1833                    cytotoxicity      4\n",
       "45                      mitochondria      4\n",
       "294                       depression      4\n",
       "801              Parkinson’s disease      4\n",
       "388                    Breast cancer      4\n",
       "1566                       Prognosis      3\n",
       "2259                       autophagy      3\n",
       "1274                          Cancer      3\n",
       "42                           obesity      3\n",
       "819         Saccharomyces cerevisiae      3\n",
       "459                            Aging      3\n",
       "543                    Proliferation      3\n",
       "464                  drug resistance      3\n",
       "466                     tuberculosis      3\n",
       "1320                          cancer      3\n",
       "1523                       diagnosis      3\n",
       "1731                       apoptosis      3\n",
       "361                             PD-1      3\n",
       "694                       Depression      3\n",
       "1223                  clinical trial      3\n",
       "1762                       Apoptosis      3\n",
       "662                   rehabilitation      3\n",
       "396                colorectal cancer      3\n",
       "...                              ...    ...\n",
       "1003           Personality disorders      1\n",
       "1004                Carbon footprint      1\n",
       "1005             Economic evaluation      1\n",
       "1006                Online education      1\n",
       "1008                      evaluation      1\n",
       "1009                   star trackers      1\n",
       "1010                  slew tolerance      1\n",
       "1011           attitude availability      1\n",
       "1012                  star detection      1\n",
       "993                   celiac disease      1\n",
       "992              Helicobacter pylori      1\n",
       "991                 immunoglobulin A      1\n",
       "981                 data association      1\n",
       "974                 [18F]FP-(+)-DTBZ      1\n",
       "975                              PET      1\n",
       "976                      Test-retest      1\n",
       "977                   Cryptococcosis      1\n",
       "978           Pulmonary Tuberculosis      1\n",
       "979                      Coinfection      1\n",
       "980            multi-object tracking      1\n",
       "982              deep neural network      1\n",
       "990                lysosomal enzymes      1\n",
       "983   long short-term memory network      1\n",
       "984                    nanoparticles      1\n",
       "985                           turkey      1\n",
       "986                  aminopeptidases      1\n",
       "987                     poultry meat      1\n",
       "988                        nutrition      1\n",
       "989                  supplementation      1\n",
       "2796                       argonaute      1\n",
       "\n",
       "[2797 rows x 2 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "keywords.sort_values(\"count\", ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(int,\n",
       "            {'pmid': 18771,\n",
       "             'pmc': 20000,\n",
       "             'publisher-id': 13829,\n",
       "             'doi': 17408,\n",
       "             'pii': 1117,\n",
       "             'pmc-scan': 1198,\n",
       "             'manuscript': 206,\n",
       "             'coden': 219,\n",
       "             'art-access-id': 208,\n",
       "             'other': 134,\n",
       "             'publisher-manuscript': 22,\n",
       "             'sici': 27,\n",
       "             'medline': 4})"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(int,\n",
       "            {'research-article': 14584,\n",
       "             'abstract': 507,\n",
       "             'brief-report': 318,\n",
       "             'retraction': 12,\n",
       "             'other': 748,\n",
       "             'review-article': 1294,\n",
       "             'editorial': 315,\n",
       "             'case-report': 899,\n",
       "             'correction': 245,\n",
       "             'news': 82,\n",
       "             'discussion': 42,\n",
       "             'letter': 302,\n",
       "             'meeting-report': 67,\n",
       "             'product-review': 31,\n",
       "             'reply': 9,\n",
       "             'article-commentary': 92,\n",
       "             'addendum': 3,\n",
       "             'book-review': 306,\n",
       "             'protocol': 31,\n",
       "             'obituary': 29,\n",
       "             'rapid-communication': 15,\n",
       "             'methods-article': 13,\n",
       "             'in-brief': 17,\n",
       "             'systematic-review': 11,\n",
       "             'introduction': 9,\n",
       "             'oration': 2,\n",
       "             'books-received': 3,\n",
       "             ' case-report': 1,\n",
       "             'data-paper': 8,\n",
       "             'report': 2,\n",
       "             'announcement': 2,\n",
       "             'calendar': 1})"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article_types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Look at the text of a single document matching a search result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pymongo.results.UpdateResult at 0x10d353408>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.update_many(\n",
    "    filter={},\n",
    "    update={ \"$unset\": { \"text_matches\": \"\" } }\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
