{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.display import clear_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pubcrawler.article import Article"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pymongo\n",
    "from bson.objectid import ObjectId\n",
    "from collections import defaultdict\n",
    "from random import sample, seed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "import timeit"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create a list of files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "pmc_path = \"/Volumes/Transcend/datasets/pmc/2019-04-03/oa_bulk/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_files = []\n",
    "for (dirpath, dirnames, filenames) in os.walk(pmc_path):\n",
    "    all_files += [os.path.join(dirpath, filename) for filename in filenames if filename.endswith(\"xml\")]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Counting coverage of different ID types from a sample of 5000 articles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "files = sample(all_files, 1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "id_types = defaultdict(int)\n",
    "article_types = defaultdict(int)\n",
    "\n",
    "for idx, filename in enumerate(files):\n",
    "    if idx % 25 is 0:\n",
    "        clear_output(wait=True)\n",
    "        print(\"Processed {0} articles ({1:.1f}% done).\".format(idx, idx/len(files) * 100))\n",
    "    with open(filename, \"r\") as f:\n",
    "        article = Article(f.read())\n",
    "    \n",
    "    ids = article.pub_ids().keys()\n",
    "    if len(ids) is 0:\n",
    "        id_types[\"none\"] += 1\n",
    "    else:\n",
    "        for item in ids:\n",
    "            id_types[item] += 1\n",
    "    \n",
    "    article_type = article.article_type()\n",
    "    if article_type:\n",
    "        article_types[article_type] += 1\n",
    "\n",
    "clear_output()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(int,\n",
       "            {'pmid': 923,\n",
       "             'pmc': 1000,\n",
       "             'doi': 850,\n",
       "             'publisher-id': 691,\n",
       "             'coden': 10,\n",
       "             'pii': 46,\n",
       "             'pmc-scan': 66,\n",
       "             'other': 8,\n",
       "             'art-access-id': 7,\n",
       "             'manuscript': 13,\n",
       "             'publisher-manuscript': 2})"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(int,\n",
       "            {'research-article': 716,\n",
       "             'other': 40,\n",
       "             'meeting-report': 4,\n",
       "             'case-report': 52,\n",
       "             'review-article': 59,\n",
       "             'letter': 13,\n",
       "             'brief-report': 19,\n",
       "             'abstract': 28,\n",
       "             'correction': 14,\n",
       "             'book-review': 20,\n",
       "             'editorial': 19,\n",
       "             'obituary': 2,\n",
       "             'article-commentary': 3,\n",
       "             'retraction': 3,\n",
       "             'methods-article': 2,\n",
       "             'news': 2,\n",
       "             'reply': 1,\n",
       "             'product-review': 1,\n",
       "             'protocol': 2})"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article_types"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It's clear that the PMC ID is the most broadly and consistently applied. Which makes sense! Because we're dealing with PMC articles."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "research-article\n",
      "BODY\n",
      "\n",
      "\n",
      "\n",
      "Introduction\n",
      "\n",
      "A prominent feature of schizophrenia is patients’ social exclusion. Patients have smaller social networks, less satisfactory interpersonal relationships and greater unemployment than healthy people or patients with other psychiatric disorders [1], [2]. A central and debilitating feature of schizophrenia, which may contribute to patients’ social exclusion, is patients’ difficulty interacting with others. These social deficits are poorly understood. Patients display poor performance on ‘off-line’ assessments of social cognition, which investigates the ability to discriminate facial expressions in pictures, attribute emotional states to the protagonists in short narratives and infer intentions in abstract problem solving contexts [3]. However, how patients’ social deficits manifest during their ‘on-line’ social interactions with others remains largely unexplored.\n",
      "\n",
      "Early psychiatrists described feeling an intuitive ‘lack of rapport’ when interacting with a patient with schizophrenia, which was termed the ‘Praecox feeling’ [4]. According to Rümke (1941) the praecox feeling is based on patients’ nonverbal behaviour. This may extend beyond patients’ clinical interactions and influence their everyday social interactions. Indeed, a recent study revealed that patients’ nonverbal behaviour (e.g. hand and head movement) during first encounters with unfamiliar others, influenced others’ experience of rapport with the patient [5]. This was the case even though others were unaware of the patient’s diagnosis. Thus, alongside stigma due to others’ knowledge of the disorder, patients’ behaviour during social interactions may contribute to others’ experience of the interaction.\n",
      "\n",
      "The aim of this study is to assess patients’ participation during first encounters with others. The study of three-person interactions provides an ideal interaction setting in that, unlike two-person interactions where individuals are obliged to take the role of the speaker or hearer, two hearer roles are possible: (i) a primary recipient: the individual who the speaker is addressing and directing their speech towards and (ii) a secondary recipient: a fully ratified participant who the speaker is not currently addressing [6], [7]. The speaker and primary recipient form the ‘active pair’, sharing a particular relationship with increased mutual gaze and coordinated movement, which the unaddressed, secondary recipient is not party to [7], [8].\n",
      "\n",
      "Here we investigate how, during first encounters, patients participate in interactions with two others who are unaware of their diagnosis. In line with previous studies investigating multiparty interaction, the identity of primary and secondary recipients in the current study will be approximated from speakers’ head angle, using 3-D motion capture data [5], [7], [9]. The relationship between patients’ participation, their symptoms and the rapport their partners experience with them will also be explored.\n",
      "\n",
      "\n",
      "Methods\n",
      "\n",
      "\n",
      "Ethics Statement\n",
      "\n",
      "All procedures were approved by the Charing Cross Research Ethics Committee (London, UK: 07/H0711/90) and all participants provided written informed consent.\n",
      "\n",
      "\n",
      "Study Design\n",
      "\n",
      "The current study is a secondary analysis of data collected to investigate social interaction in schizophrenia [5]. The corpus under analysis involved three-way interactions in two conditions: A patient condition, involving one patient with a diagnosis of schizophrenia and two healthy participants (i.e. patients’ partners), and a control condition, involving three healthy participants (i.e. controls). This study analysed the amount of time participants spent as a speaker, primary and secondary recipients.\n",
      "\n",
      "\n",
      "Sample\n",
      "\n",
      "Twenty patients with a diagnosis of schizophrenia (6 male, 14 female) and one hundred non-psychiatric healthy participants, forty in the patient condition (21 male, 19 female) and sixty in the control condition (34 male, 26 female), participated in the study. Recruitment and data collection occurred from July 2008 to Nov 2010. Patients were recruited at routine psychiatric outpatient clinics, on the basis of a clinical diagnosis of schizophrenia. Of all patients approached, 25% agreed to participate. Diagnosis was confirmed using the Structured Clinical Interview for DSM-IV symptoms [10]. Patients presenting with motor side effects from anti-psychotic medication (e.g. muscle stiffness and involuntary muscle spasms) were excluded from the study based on clinicians’ assessment. Patients’ antipsychotic medication was documented and converted into chlorpromazine equivalents (CPZE mg/day) according to the standard formula suggested by Woods [11].\n",
      "\n",
      "Non-fluent English speakers were also excluded. Non-psychiatric healthy participants were recruited through advertising on local community websites. Of those who responded to the advertisement, 40% participated. Participants with a diagnosis of psychosis or affective disorders in themselves, or any first-degree relatives, and those who were not fluent English speakers were excluded. Interacting participants had not met prior to the study. Healthy participants were informed that the study was an investigation of social interaction, and were not aware that a psychiatric patient was present. All interactions were conducted outside of a psychiatric department, i.e. in a non-medical university department. After complete description of the study to the subjects, written informed consent was obtained.\n",
      "\n",
      "\n",
      "Assessments\n",
      "\n",
      "Patients’ diagnosis of schizophrenia, paranoid subtype, was confirmed using the Structured Clinical Interview for Diagnostic symptoms (SCID-I) [10]. Patients’ positive, negative and general symptoms were assessed using the Positive And Negative Syndrome Scale for schizophrenia (PANSS) [12]. The Brixton Spatial Anticipation Test and the Hayling Sentence Completion Test assessed participants executive functioning [13]. Participants rated the level of rapport or connection they felt with each interacting partner on a 10 point scale, with a higher score indicating stronger rapport [14].\n",
      "\n",
      "\n",
      "3D Motion-capture Equipment\n",
      "\n",
      "The Augmented Human Interaction laboratory, where interactions were recorded, was fitted with an optical based Vicon motion-capture system, consisting of 12 infrared cameras and Vicon iQ software. Participants wore a top and cap with 27 reflective markers attached. The motion capture system detected the precise spatial and temporal coordinates of participants’ markers in 3D at each frame of the interaction, at a rate of 60 frames per second. This information was exported in numerical values and as a wire-frame representation of the interaction over time.\n",
      "\n",
      "\n",
      "Procedure\n",
      "\n",
      "Each group of three participants were seated and asked to discuss a moral dilemma called ‘The Balloon Task’ [15]. During a one-to-one interview with a researcher, all participants’ socio-demographic and executive functioning was assessed. Patients’ clinical features and social functioning were also assessed. A detailed description of the procedure is described elsewhere [5].\n",
      "\n",
      "\n",
      "Identification of speaker and hearers\n",
      "\n",
      "The identification of the speaker, primary and secondary recipients was assessed on a frame-by-frame basis. Speakers were hand coded from 2D videos using ELAN annotational software [16]. Primary recipients were identified as the person the speaker was oriented towards. Speakers’ head orientation was determined based on the precise spatial coordinates of the four optical markers attached to their caps (see figure 1) [7]. This procedure has been used in previous studies to index listener role in three-way interactions [7]. Frames with speaker head orientations falling less than two degrees from the centre line were excluded from the analysis (15% of the data set) to eliminate ambiguity.\n",
      "\n",
      "10.1371/journal.pone.0077506.g001Figure 1 Indexing primary and secondary recipient from the head angle of the speaker.Speaker’s head angle is derived from the precise positions (3D) of the four reflective markers attached to their cap.\n",
      "\n",
      "\n",
      "Statistical Analysis\n",
      "\n",
      "Statistical analyses were conducted using SPSS Version 20 (© Copyright SPSS Inc. 2010). Independent samples t-tests were used to compare interaction duration (in seconds) between conditions, and age and executive functioning between patients and healthy participants. Chi 2-tests compared: the distribution of gender across conditions (i.e. patient condition (n = 60) and control condition (n = 60) and between patients (n = 20) and healthy participants (n = 100).\n",
      "\n",
      "\n",
      "Conversation role\n",
      "\n",
      "In order to identify change in conversational roles over time, the percentage of time spent in each role was assessed at three time points during the interaction (i.e. the start, middle and end of the interaction). In order to adjust for the variations in total interaction duration between groups, each interaction was divided into ten sections based on the interaction duration (i.e. interaction duration (seconds)/10). Each section had a mean duration of approximately 30 seconds. The percentage of time participants spent in each conversation role was calculated at sections one (interaction start: approximately 0–30 seconds) five (interaction middle: approximately 120–150 seconds) and eight (interaction end: approximately 210–240 seconds). The final two sections (approx 1 minute) of the interactions were not analysed as observations revealed a trailing off during this final minute, so patterns of participation may be misrepresented in this data.\n",
      "\n",
      "Mixed models analyses compared percentage of time spent in each conversation role between the three participant types (patient, patients’ partners and controls) at the start (section 1), middle (section 5) and end (section 8) of the interaction, clustering for the three-person group participants belonged to. Participants’ age, gender and executive functioning were accounted for in the model and reported if significant.\n",
      "\n",
      "\n",
      "Patients’ conversation role, clinical features and rapport\n",
      "\n",
      "Spearman-Rho nonparametric correlation coefficients (two-tailed) assessed the relationship between percentage of time patients spend in each conversation role (speaker, primary recipient and secondary recipient) and patients’ symptoms (PANSS), antipsychotic medication dose (CPZE), and the rapport others experience with them. Significant associations were further investigated using regression analyses.\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "Interactions lasted on average 5 minutes 24 seconds (SD = 1 minute 55 seconds). The duration of interactions in the patient condition (M = 318.08 sec, SD = 113.35) did not differ from the control condition (M = 335.29 sec, SD = 118.91) (t(38) = −4.62, p = .65).\n",
      "\n",
      "The distribution of gender in the patient condition (n = 60: female = 53.33%) and control condition (n = 60: female = 43.33%) did not significantly differ (X2 (1) = 1.20, p = .27). The distribution of gender within patients (n = 20: female = 65%) and healthy participants (n = 100: female = 45%) did not significantly differ (X2 (1) = 2.67, p = .10). Participants’ age and scores on assessments of executive functioning are displayed in table 1. Compared to healthy participants, patients were older (p<.01) and displayed poorer executive functioning (p<.01).\n",
      "\n",
      "Table 1\n",
      "Schizophrenia patients (n = 20)\tHealthy participants (n = 100)\n",
      "Variables\tM\tSD\tM\tSD\tt\tdf\tp\n",
      "Age\t41.50\t8.64\t31.10\t9.60\t−4.51\t119\t<.01\n",
      "Spatial Executive functioning\t3.07\t0.51\t5.10\t0.22\t3.65\t119\t<.01\n",
      "Verbal Executive functioning\t3.79\t0.43\t4.91\t0.18\t2.52\t119\t<.01\n",
      "\n",
      "\n",
      "Patients’ Clinical Features\n",
      "\n",
      "Patients’ mean duration of illness was 15 years (SD = 10.26; range 2–46 years), with an average inpatient admissions duration of 19.95 weeks (SD = 18.45; range 0–60). Patients’ mean symptom scores were: positive 15.80 (SD = 6.76; range 7–37) negative 9.95 (SD = 3.36; range 7–19) and general symptoms 28.41 (SD = 10⋅42; range 16–59). Three patients were medication free at the time of the study and the remaining seventeen were taking typical (2 patients) or atypical (15 patients) antipsychotic medication. Of those taking antipsychotic medication, the average dose was 167.87 mg/day (SD = 109.29; range 50–400 mg/day), which fell within the low dose range (CPZE = 50–200 mg/day) [17].\n",
      "\n",
      "\n",
      "Conversation Role\n",
      "\n",
      "Mixed models comparisons of conversation role, by participant type, at each time interval are displayed in table 2. At the start of the interaction, compared to participants in the control group, patients spent less time speaking (ß = −.639, p = .02) and more time in the role of secondary recipient (ß = .349, p = .02), while patients’ partners spent less time as a secondary recipient (ß = −.357, p = .02). Compared to male participants, female participants spent more time in role of secondary recipient at the start of the interaction (p = .01), and less time as a primary recipient at the end of the interaction (p = .05). Patients or their partners did not differ from controls in the amount of time they spent in any conversation role at the middle or end of the interaction (p>.05). Figure 2 displays the mean percentage of time spent as speaker and secondary recipient at each time point, by participant type.\n",
      "\n",
      "10.1371/journal.pone.0077506.g002Figure 2 Mean percentage of time spent as speaker (A) and secondary recipient (B) at the start, middle and end of the interaction by participant type.\n",
      "\n",
      "Table 2\n",
      "95% CI\n",
      "Models\tn\tM (%)\tSE\tß\tSE\tLower\tUpper\tChi\tdf\tp\n",
      "Interaction start\n",
      "1. Speaking\n",
      "Patients\t20\t17.35\t3.81\t−.639\t.28\t−1.186\t−.093\t5.26\t1\t.02\n",
      "Patients’ partners\t40\t33.22\t1.73\t.141\t.10\t−.072\t.355\t1.68\t1\t.19\n",
      "Controls\t60\t29.97\t0.94\n",
      "2. Primary\n",
      "Patients\t20\t21.96\t3.70\t−.197\t.23\t−.657\t.264\t0.70\t1\t.40\n",
      "Patients’ partners\t40\t26.27\t2.37\t−.111\t.14\t−.377\t.155\t0.67\t1\t.41\n",
      "Controls\t60\t26.64\t0.74\n",
      "3. Secondary\n",
      "Patients\t20\t36.38\t5.36\t.349\t.15\t.054\t.645\t5.36\t1\t.02\n",
      "Patients’ partners\t40\t17.96\t2.67\t−.357\t.15\t−.659\t−.055\t5.36\t1\t.02\n",
      "Controls\t60\t25.66\t0.81\n",
      "Gender: Male\t62\t20.64\t2.20\t−.430\t.17\t−.754\t−.107\t6.79\t1\t.01\n",
      "Gender: Female\t58\t31.73\t2.45\n",
      "Interaction middle\n",
      "4. Speaking\n",
      "Patients\t20\t26.62\t5.58\t−.018\t.26\t−.533\t.498\t0.01\t1\t.94\n",
      "Patients’ partners\t40\t29.98\t2.68\t−.069\t.12\t−.165\t.303\t0.34\t1\t.56\n",
      "Controls\t60\t29.13\t2.62\n",
      "5. Primary\n",
      "Patients\t20\t20.75\t3.85\t−.305\t.21\t−.716\t.106\t2.11\t1\t.15\n",
      "Patients’ partners\t40\t27.38\t3.05\t.050\t.16\t−.261\t.361\t0.10\t1\t.75\n",
      "Controls\t60\t25.68\t2.65\n",
      "6. Secondary\n",
      "Patients\t20\t30.41\t4.06\t.189\t.27\t−.331\t.709\t0.51\t1\t.48\n",
      "Patients’ partners\t40\t22.89\t3.10\t−.002\t.14\t−.282\t.277\t0.01\t1\t.99\n",
      "Controls\t60\t25.49\t2.84\n",
      "Interaction end\n",
      "7. Speaking\n",
      "Patients\t20\t25.97\t5.22\t−.148\t.28\t−.700\t.404\t0.28\t1\t.60\n",
      "Patients’ partners\t40\t28.13\t2.62\t−.040\t.12\t−.276\t.196\t0.11\t1\t.74\n",
      "Controls\t60\t30.38\t1.11\n",
      "8. Primary\n",
      "Patients\t20\t19.97\t4.77\t−.239\t.29\t−.813\t.336\t0.66\t1\t.42\n",
      "Patients’ partners\t40\t23.98\t3.15\t−.244\t.20\t−.645\t.157\t1.43\t1\t.23\n",
      "Controls\t60\t26.96\t0.99\n",
      "Gender: Male\t62\t26.90\t2.41\t.278\t.14\t.006\t.551\t4.02\t1\t.05\n",
      "Gender: Female\t58\t20.36\t1.85\n",
      "9. Secondary\n",
      "Patients\t20\t27.52\t5.12\t−.060\t.30\t−.639\t.520\t0.04\t1\t.84\n",
      "Patients’ partners\t40\t20.55\t2.92\t−.130\t.17\t−.469\t.209\t0.56\t1\t.45\n",
      "Controls\t60\t26.23\t0.81\n",
      "\n",
      "\n",
      "Conversation role and clinical features\n",
      "\n",
      "Spearman-Rho correlations found no significant association between the time patients spent in each conversation role (start middle or end) and their symptoms (positive, negative or general) (p>.05) or antipsychotic medication dose (p>.05).\n",
      "\n",
      "\n",
      "Conversation role and rapport\n",
      "\n",
      "Spearman-Rho correlation revealed that patients’ partners experienced poorer rapport with patients who, at the start of the interaction, spent less time in the role of secondary recipient (Rho(11) = .705, p = .02) and more time in the role of primary recipient (Rho(11) = −.755, p<.01).\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "The findings revealed that, at the start of a first encounter, patients spoke less and spent more time as a secondary recipient. However, as the interaction progressed patients’ participation in the conversation did not differ from controls. Patients’ partners experienced poorer rapport with patients who, at the start of the interaction, spent more time as the focus of the speaker’s attention, in the role of primary recipient. Patients’ symptoms were not associated with their participation at any time interval.\n",
      "\n",
      "In the opening moments of the interaction, patients spent less time speaking and more time in the role of secondary recipient. This suggests that patients are taking a more peripheral role at the start of the interaction, speaking, and being spoken to, less. In line with these findings, previous studies identified that schizophrenia patients show increased ‘flight’ behaviour, signaling the avoidance of social stimuli, during the first two minutes of a clinical consultation [18]. Thus, it appears that schizophrenia patients display signs of social withdrawal at the start of an interaction. Patients may find participating in a face-to-face interaction demanding due to social anxiety. This may have implications for devising interventions to address patients’ initial difficulties settling into a conversation with others, both in and outside of clinical contexts.\n",
      "\n",
      "Differences between patients and controls were identified after adjusting for gender. However, investigation across all participants revealed differences in participation between gender. Specifically, compared to males, females were more likely to be secondary recipients at the start of the interaction and less likely to be primary recipients at the end. The size of the patient sample in the current study was insufficient to investigate any interaction effects for patient and gender. However, this should be considered in future studies.\n",
      "\n",
      "Patients’ participation at the middle or end of the interaction did not differ from healthy participants in the control condition. This suggests that over the course of an interaction, patients become more actively involved as a speaker and spend less time in the role of secondary recipient. Micro-analyses of patients’ three-way interactions has found that when patients are reluctant to speak, others actively attempt to involve them in interaction: initially using subtle nonverbal cues and, if unsuccessful, verbally requesting their input [19]. Perhaps patients’ increased involvement over the course of the interaction is a result of both others actively attempting to engage the patient and patients becoming more comfortable in the interaction. Patients’ increased participation over time suggests that, rather than patients being excluded from the speaker-primary recipient pair over the course of a first meeting, patients initially appear to participate less but become more involved as the interaction progresses.\n",
      "\n",
      "Others’ experience of rapport with the patient was associated with patients’ participation at the start of the interaction (approximately the first 30 seconds). Specifically, others experienced poorer rapport with patients who spent more time in the role of primary recipient and less time as secondary recipient. This suggests that patients’ partners are detecting something in the patient’s behaviour at the start of the interaction that is making them more likely to be the focus of their attention (i.e. primary recipient). In a previous analysis of this corpus, patients’ increased hand movement during interaction was associated with others’ poorer experience of rapport with them [5]. Thus, despite the fact that patients’ participation improves over time, it appears that the opening moments of patients’ first encounters are interpersonally important. Future research should further investigate other aspects of patients’ behaviour (e.g., what they are saying, their nonverbal behaviour) and their partners’ responses to them at this critical phase in the interaction. Moreover, understanding how patterns of participation develop over longer and repeat encounters would also be of relevance in understanding social integration of patients.\n",
      "\n",
      "The current study found no association between patients’ participation in conversation and their symptoms. It might have been expected that patients with more negative symptoms would spend less time speaking. However, the variance on patients’ symptoms in the current sample was low, which may have impeded the ability to detect an association if one existed.\n",
      "\n",
      "\n",
      "Strengths and Limitations\n",
      "\n",
      "This study benefited from the three-person experimental design, enabling the investigation of patients’ participation in contexts more typical of everyday social interaction. Furthermore, by ensuring all interacting partners were unfamiliar to each other and unaware of patients’ diagnoses, the possible confounding variables of familiarity and stigma, due to prior knowledge of the disorder, were removed. Although the interactions were unscripted conversations, they were recorded in a motion capture lab, which is a somewhat unnatural environment. Future studies could employ new motion capture techniques, such as the Kinect system, to obtain similar data from interactions outside a lab setting [20]. Examining patients’ interactions in more naturalistic environments may allow for recruitment of patients who are more symptomatic addressing the limited symptom variance seen in the current study. As the head angle of primary or secondary recipients was not investigated we cannot say whether the primary recipient is returning the gaze of the speaker. Although this was not necessary to answer the questions of the current study, future studies could investigate patients’ reciprocation of speakers’ gaze in the role of primary recipient.\n",
      "\n",
      "\n",
      "Conclusion\n",
      "\n",
      "The aim of this study was to investigate schizophrenia patients’ participation during first encounters with unfamiliar others who were unaware of their diagnosis. The findings indicate that although patients take a more peripheral role at the start of the interaction, they participate more as the interaction progresses. Despite this, it is the first 30 seconds of the encounter that influences others’ experience of rapport with them. This highlights the opening moments of patients’ conversations as interpersonally significant. Future research should identify the precise behaviours displayed, by patients and others, during this critical period in order to develop our understanding of patients’ social deficits and their impact on interpersonal relationships.\n"
     ]
    }
   ],
   "source": [
    "with open(sample(all_files, 1)[0], \"r\") as f:\n",
    "    xml = f.read()\n",
    "    article = Article(xml)\n",
    "    print(article.article_type())\n",
    "    if article.soup.body:\n",
    "        print(\"BODY\")\n",
    "        print(article.extract_text())\n",
    "    else:\n",
    "        print(\"NO BODY\")\n",
    "        print(article.extract_text())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load articles into a MongoDB collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "client = pymongo.MongoClient(\"localhost\", 27017)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "articles = client.pmc.articles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [],
   "source": [
    "seed(\"2019-04-04\")\n",
    "files = sample(all_files, 10000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [],
   "source": [
    "articles.drop()\n",
    "for file in files:\n",
    "    with open(file, \"r\") as f:\n",
    "        xml = f.read()\n",
    "        article = Article(xml)\n",
    "        articles.insert_one({\n",
    "            \"_id\": article.pub_ids().get(\"pmc\"),  # Assuming they all have them\n",
    "            \"xml\": xml,\n",
    "            \"extracted_text\": article.extract_text()\n",
    "        })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_id': '24489775',\n",
       " 'xml': '<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD v1.0 20120330//EN\" \"JATS-archivearticle1.dtd\">\\n<article xmlns:xlink=\"http://www.w3.org/1999/xlink\" xmlns:mml=\"http://www.w3.org/1998/Math/MathML\" article-type=\"research-article\"><?properties open_access?><front><journal-meta><journal-id journal-id-type=\"nlm-ta\">PLoS One</journal-id><journal-id journal-id-type=\"iso-abbrev\">PLoS ONE</journal-id><journal-id journal-id-type=\"publisher-id\">plos</journal-id><journal-id journal-id-type=\"pmc\">plosone</journal-id><journal-title-group><journal-title>PLoS ONE</journal-title></journal-title-group><issn pub-type=\"epub\">1932-6203</issn><publisher><publisher-name>Public Library of Science</publisher-name><publisher-loc>San Francisco, USA</publisher-loc></publisher></journal-meta><article-meta><article-id pub-id-type=\"pmid\">24489775</article-id><article-id pub-id-type=\"pmc\">3904948</article-id><article-id pub-id-type=\"publisher-id\">PONE-D-13-42217</article-id><article-id pub-id-type=\"doi\">10.1371/journal.pone.0086718</article-id><article-categories><subj-group subj-group-type=\"heading\"><subject>Research Article</subject></subj-group><subj-group subj-group-type=\"Discipline-v2\"><subject>Biology</subject><subj-group><subject>Computational Biology</subject><subj-group><subject>Genomics</subject><subj-group><subject>Comparative Genomics</subject></subj-group></subj-group></subj-group><subj-group><subject>Evolutionary Biology</subject><subj-group><subject>Evolutionary Processes</subject><subj-group><subject>Adaptation</subject><subject>Evolutionary Selection</subject><subject>Mutation</subject><subject>Natural Selection</subject></subj-group></subj-group><subj-group><subject>Evolutionary Systematics</subject><subj-group><subject>Phylogenetics</subject></subj-group></subj-group><subj-group><subject>Comparative Genomics</subject><subject>Evolutionary Developmental Biology</subject></subj-group></subj-group><subj-group><subject>Genomics</subject><subj-group><subject>Comparative Genomics</subject></subj-group></subj-group></subj-group><subj-group subj-group-type=\"Discipline-v2\"><subject>Medicine</subject><subj-group><subject>Hematology</subject><subj-group><subject>Heme Synthesis</subject></subj-group></subj-group></subj-group></article-categories><title-group><article-title>Molecular Evolution of Multiple-Level Control of Heme Biosynthesis Pathway in Animal Kingdom</article-title><alt-title alt-title-type=\"running-head\">Heme Biosynthesis Pathway Evolution</alt-title></title-group><contrib-group><contrib contrib-type=\"author\"><name><surname>Tzou</surname><given-names>Wen-Shyong</given-names></name><xref ref-type=\"aff\" rid=\"aff1\">\\n<sup>1</sup>\\n</xref><xref ref-type=\"corresp\" rid=\"cor1\">\\n<sup>*</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Chu</surname><given-names>Ying</given-names></name><xref ref-type=\"aff\" rid=\"aff1\">\\n<sup>1</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Lin</surname><given-names>Tzung-Yi</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Hu</surname><given-names>Chin-Hwa</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Pai</surname><given-names>Tun-Wen</given-names></name><xref ref-type=\"aff\" rid=\"aff3\">\\n<sup>3</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Liu</surname><given-names>Hsin-Fu</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref><xref ref-type=\"aff\" rid=\"aff4\">\\n<sup>4</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Lin</surname><given-names>Han-Jia</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Cases</surname><given-names>Ildeofonso</given-names></name><xref ref-type=\"aff\" rid=\"aff5\">\\n<sup>5</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Rojas</surname><given-names>Ana</given-names></name><xref ref-type=\"aff\" rid=\"aff5\">\\n<sup>5</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Sanchez</surname><given-names>Mayka</given-names></name><xref ref-type=\"aff\" rid=\"aff6\">\\n<sup>6</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>You</surname><given-names>Zong-Ye</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Hsu</surname><given-names>Ming-Wei</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref></contrib></contrib-group><aff id=\"aff1\">\\n<label>1</label>\\n<addr-line>Department of Life Sciences, National Taiwan Ocean University, Keelung, Taiwan</addr-line>\\n</aff><aff id=\"aff2\">\\n<label>2</label>\\n<addr-line>Institute of Bioscience and Biotechnology, National Taiwan Ocean University, Keelung, Taiwan</addr-line>\\n</aff><aff id=\"aff3\">\\n<label>3</label>\\n<addr-line>Department of Computer Science and Engineering, National Taiwan Ocean University, Keelung, Taiwan</addr-line>\\n</aff><aff id=\"aff4\">\\n<label>4</label>\\n<addr-line>Department of Medical Research, Mackay Memorial Hospital, Taipei, Taiwan</addr-line>\\n</aff><aff id=\"aff5\">\\n<label>5</label>\\n<addr-line>Computational Cell Biology Group, Institute of Predictive and Personalized Medicine of Cancer (IMPPC), Barcelona, Spain</addr-line>\\n</aff><aff id=\"aff6\">\\n<label>6</label>\\n<addr-line>Cancer and Iron Group, Institute of Predictive and Personalized Medicine of Cancer (IMPPC), Barcelona, Spain</addr-line>\\n</aff><contrib-group><contrib contrib-type=\"editor\"><name><surname>Missirlis</surname><given-names>Fanis</given-names></name><role>Editor</role><xref ref-type=\"aff\" rid=\"edit1\"/></contrib></contrib-group><aff id=\"edit1\">\\n<addr-line>CINVESTAV-IPN, Mexico</addr-line>\\n</aff><author-notes><corresp id=\"cor1\">* E-mail: <email>wstzou@ntou.edu.tw</email></corresp><fn fn-type=\"COI-statement\"><p><bold>Competing Interests: </bold>The authors have declared that no competing interests exist.</p></fn><fn fn-type=\"con\"><p>Conceived and designed the experiments: WST YC. Performed the experiments: YC TYL ZYY MWH. Analyzed the data: CHH TWP HFL HJL YC TYL ZYY MWH. Contributed reagents/materials/analysis tools: IC AR MS. Wrote the paper: WST YC.</p></fn></author-notes><pub-date pub-type=\"collection\"><year>2014</year></pub-date><pub-date pub-type=\"epub\"><day>28</day><month>1</month><year>2014</year></pub-date><volume>9</volume><issue>1</issue><elocation-id>e86718</elocation-id><history><date date-type=\"received\"><day>15</day><month>10</month><year>2013</year></date><date date-type=\"accepted\"><day>12</day><month>12</month><year>2013</year></date></history><permissions><copyright-statement>&#x000a9; 2014 Tzou et al</copyright-statement><copyright-year>2014</copyright-year><copyright-holder>Tzou et al</copyright-holder><license xlink:href=\"http://creativecommons.org/licenses/by/4.0/\"><license-p>This is an open-access article distributed under the terms of the Creative Commons Attribution License, which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are properly credited.</license-p></license></permissions><abstract><p>Adaptation of enzymes in a metabolic pathway can occur not only through changes in amino acid sequences but also through variations in transcriptional activation, mRNA splicing and mRNA translation. The heme biosynthesis pathway, a linear pathway comprised of eight consecutive enzymes in animals, provides researchers with ample information for multiple types of evolutionary analyses performed with respect to the position of each enzyme in the pathway. Through bioinformatics analysis, we found that the protein-coding sequences of all enzymes in this pathway are under strong purifying selection, from cnidarians to mammals. However, loose evolutionary constraints are observed for enzymes in which self-catalysis occurs. Through comparative genomics, we found that in animals, the first intron of the enzyme-encoding genes has been co-opted for transcriptional activation of the genes in this pathway. Organisms sense the cellular content of iron, and through iron-responsive elements in the 5&#x02032; untranslated regions of mRNAs and the intron-exon boundary regions of pathway genes, translational inhibition and exon choice in enzymes may be enabled, respectively. Pathway product (heme)-mediated negative feedback control can affect the transport of pathway enzymes into the mitochondria as well as the ubiquitin-mediated stability of enzymes. Remarkably, the positions of these controls on pathway activity are not ubiquitous but are biased towards the enzymes in the upstream portion of the pathway. We revealed that multiple-level controls on the activity of the heme biosynthesis pathway depend on the linear depth of the enzymes in the pathway, indicating a new strategy for discovering the molecular constraints that shape the evolution of a metabolic pathway.</p></abstract><funding-group><funding-statement>This work was supported by the National Science Council, Taiwan, R.O.C. (Grant Nos. NSC 95-2113-M-019-003, NSC 96-2627-B-019-002, NSC 97-2627-B-019-002, NSC 98-2627-B-019-002, NSC 98-2313-B-019-004-MY3, NSC 101-2311-B-019 -001, NSC 102-2627-B-019-002-, NSC 102-2633-B-019 -001 -) and the Center of Excellence for the Oceans, National Taiwan Ocean University. The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.</funding-statement></funding-group><counts><page-count count=\"15\"/></counts></article-meta></front><body><sec id=\"s1\"><title>Introduction</title><p>Molecular evolution has recently been a popular area of investigation, and through the advancement of technology and the maturation of analysis methods, this field continues to spawn important insights into the evolutionary processes affecting genes. None of these genes or their encoded proteins exists in isolation, and the products of genes construct the metabolic pathways and networks underlying the cellular and metabolic processes of organisms. As an increasing number of studies are describing the rates of protein and pathway evolution over evolutionary time, there are more opportunities to clarify the patterns and principles of natural selection acting on the pathways involved in the metabolic networks of organisms.</p><p>Research focusing on the effects of the organization of pathways on the strength of selection acting on individual proteins in these pathways has revealed various evolutionary patterns among proteins at different positions in a pathway. Important research has been conducted with respect to the plant anthocyanin biosynthetic pathway <xref rid=\"pone.0086718-Rausher1\" ref-type=\"bibr\">[1]</xref>&#x02013;<xref rid=\"pone.0086718-Rausher2\" ref-type=\"bibr\">[3]</xref>, terpenoid biosynthesis pathway <xref rid=\"pone.0086718-Ramsay1\" ref-type=\"bibr\">[4]</xref>, starch biosynthesis pathway <xref rid=\"pone.0086718-Yu1\" ref-type=\"bibr\">[5]</xref>, <xref rid=\"pone.0086718-Li1\" ref-type=\"bibr\">[6]</xref>, gibberellin biosynthesis pathway <xref rid=\"pone.0086718-Yang1\" ref-type=\"bibr\">[7]</xref> and carotenoid biosynthesis pathway <xref rid=\"pone.0086718-Clotault1\" ref-type=\"bibr\">[8]</xref> as well as lateral line innovation in teleosts <xref rid=\"pone.0086718-Philip1\" ref-type=\"bibr\">[9]</xref>, the glucosinolate pathway in <italic>Arabidopsis thaliana</italic>\\n<xref rid=\"pone.0086718-OlsonManning1\" ref-type=\"bibr\">[10]</xref> and the primate N-glycosylation pathway <xref rid=\"pone.0086718-Montanucci1\" ref-type=\"bibr\">[11]</xref>. It has been demonstrated that the pleiotropic genes in the upstream portions of pathways or those found at branch points in a network are subject to stronger selective constraints <xref rid=\"pone.0086718-Cork1\" ref-type=\"bibr\">[12]</xref>. On the other hand, selection is relaxed in the downstream enzymes, and nonsynonymous substitution rates as well as <italic>d<sub>N</sub>/d<sub>S</sub></italic> ratios are higher in these pathway components.</p><p>The heme biosynthesis pathway is an appropriate system not only for comparing the evolutionary rates of genes according to their position or pathway reticulation but also for studying functional motifs that may play a role at several levels of gene regulation. Heme acts as an essential cofactor for cytochromes, oxidases, peroxidases, catalases, hemoglobin and myoglobin in organisms. Heme acts as an iron-chelating tetrapyrrole and is composed of a complex macrocycle containing four pyrrolic rings connected by methine bridges in cyclic form. Heme also plays multiple regulatory roles, including microRNA processing, ion channel functions, circadian rhythms, mitochondrial targeting, translational regulation and protein degradation <xref rid=\"pone.0086718-Khan1\" ref-type=\"bibr\">[13]</xref>&#x02013;<xref rid=\"pone.0086718-Zhang1\" ref-type=\"bibr\">[16]</xref>. The heme biosynthesis pathway is an especially well-characterized and important pathway for erythroid production in animals, and malfunctions in heme biosynthesis result in several types of porphyrias because of the accumulation of toxic tetrapyrrole intermediates <xref rid=\"pone.0086718-Dailey1\" ref-type=\"bibr\">[17]</xref>. The heme biosynthesis pathway of animals is comprised of eight consecutive genes: 5-aminolevulinic acid synthase (<italic>ALAS</italic>), porphobilinogen synthase (<italic>PBGS</italic>), porphobilinogen deaminase (<italic>PBGD</italic>), uroporphyrinogen III synthase (<italic>UROS</italic>), uroporphyrinogen III decarboxylase (<italic>UROD</italic>), coproporphyrinogen III oxidase (<italic>CPO</italic>), protoporphyrinogen IX oxidase (<italic>PPO</italic>) and ferrochelatase (<italic>FECH</italic>), from positions one to eight (<xref ref-type=\"fig\" rid=\"pone-0086718-g001\">Figure 1</xref>). In animals, the formation of heme occurs in the mitochondria and cytosol. Heme biosynthesis begins with the formation of 5-aminolevulinic acid. In the Shemin pathway of animals, ALAS catalyzes a single-step reaction to condense glycine and succinyl-CoA into ALA, with the elimination of CO<sub>2</sub>. In vertebrates, the housekeeping isoforms of ALAS1 are expressed in all cell types, and ALAS2 is expressed in erythroid cells (eALAS) at a very high level. Subsequently, PBGS converts two 5-aminolevulinic acids into porphobilinogen, and the addition of four PBG substrates to a dipyrrole by PBGD results in the production of pre-uroporphyrinogen, also known as hydroxymethylbilane. A circularization reaction performed by UROS generates uroporphyrinogen III. The modification of side chains on the cyclic intermediates sequentially mediated by UROD, CPO and PPO results in protoporphyrin IX, which is subsequently chelated with ferrous iron by FECH (see reviews in <xref rid=\"pone.0086718-Heinemann1\" ref-type=\"bibr\">[14]</xref>, <xref rid=\"pone.0086718-Layer1\" ref-type=\"bibr\">[18]</xref>).</p><fig id=\"pone-0086718-g001\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g001</object-id><label>Figure 1</label><caption><title>Heme biosynthesis pathway in animals.</title><p>The substrate and product are indicated for each enzyme and the subcellular localization of each enzyme is also shown (cytosol or mitochondria). Each enzyme is coded from one to eight according to the linear order of the pathway. Also shown are the processes by which hydroxymethylbilane, the substrate of UROS, can be non-enzymatically cyclized to form uroporphyrinogen I, leading to uroporphyrin I or coproporphyrin I, and the process by which uroporphyrinogen III, the product of UROS, can be auto-oxidized to form uroporphyrin III. Protoporphyrinogen, the substrate of PPO, can be auto-oxidized to form protoporphyrin.</p></caption><graphic xlink:href=\"pone.0086718.g001\"/></fig><p>ALAS1 and ALAS2 are derived from gene duplication. A phylogenetic analysis of ALAS suggested that the relevant gene duplication event took place before the divergence of hagfish from the deuterostome line leading to vertebrates <xref rid=\"pone.0086718-Duncan1\" ref-type=\"bibr\">[19]</xref>. In extant vertebrate species, ALAS1 and ALAS2 are paralogs, and their amino acid sequences are highly similar to each other.</p><p>Regulation of the genes encoding the eight enzymes of the heme biosynthesis pathway can occur at the transcriptional, translational and post-translational levels. At the transcriptional level, multiple erythroid-specific factors have been observed to be involved in the transcriptional activation of several genes that participate in heme biosynthesis and erythropoiesis. Among these transcription factors, genomic DNA-binding activity and the conserved binding sites of KLF1, GATA1 and TAL1 have been studied in humans and mice <xref rid=\"pone.0086718-Kerenyi1\" ref-type=\"bibr\">[20]</xref>. The binding sites of KLF1 are located within the intergenic regions or introns (particularly the first intron) of genes encoding the components of the heme biosynthesis pathway. In a de novo motif study, GATA1 and TAL1 were hypothesized to complex with KLF1 in a small subset of erythroid <italic>cis</italic>-regulatory modules <xref rid=\"pone.0086718-Tallack1\" ref-type=\"bibr\">[21]</xref>. The majority of the GATA1-binding sites that mediate the activation of gene expression are close to the transcription start site, within either the first intron or the proximal 5&#x02032; flanking region <xref rid=\"pone.0086718-Cheng1\" ref-type=\"bibr\">[22]</xref>. TAL1-binding sites for the eight genes of the heme biosynthesis pathway are detected in either proximal promoter or intronic regions <xref rid=\"pone.0086718-Kassouf1\" ref-type=\"bibr\">[23]</xref>. A housekeeping promoter utilized in all tissue types exists in ALAS1, PBGS, PBGD, and UROS. However, for ALAS2, PBGS, PBGD and UROS, erythroid-specific promoters drive gene expression. More importantly, the alternative promoters found in PBGS, PBGD and UROS are located in intron 1, and an alternative splicing event is required for the transcript generated from the housekeeping promoters and erythroid-specific promoters <xref rid=\"pone.0086718-Ajioka1\" ref-type=\"bibr\">[24]</xref>. It has also been established that DNase-hypersensitive sites usually serve as sites for conserved and cell-type specific transcription factor binding and histone modification <xref rid=\"pone.0086718-Natarajan1\" ref-type=\"bibr\">[25]</xref>&#x02013;<xref rid=\"pone.0086718-Thurman1\" ref-type=\"bibr\">[27]</xref>. Combined analysis of intron 1 sequences and DNase-hypersensitive sites will shed light on the regulatory potential of the corresponding <italic>cis</italic>-elements and the evolution of the heme biosynthesis pathway.</p><p>At the translational level, IRE-binding protein (IRP) interacts with the iron-responsive element (IRE) located in the 5&#x02032; untranslated regions (5&#x02032;UTRs) of ALAS2 and ferritin mRNA to inhibit protein translation and in the 3&#x02032; untranslated region (3&#x02032;UTR) of TfR1 to stabilize mRNA. During iron repletion, iron-sulfur clusters can abolish the IRE-binding ability of IRP1, and the F-box protein FBXL5 recognizes IRP2 and targets it for degradation by E3 ligases. Both mechanisms offer a link between iron availability and heme synthesis <xref rid=\"pone.0086718-Evstatiev1\" ref-type=\"bibr\">[28]</xref>&#x02013;<xref rid=\"pone.0086718-Chen1\" ref-type=\"bibr\">[30]</xref>. At the post-translational level, the binding of heme to the heme-regulatory motif (HRM) in the mature ALAS1 peptide blocks mitochondrial import and results in end-product inhibition <xref rid=\"pone.0086718-Duncan1\" ref-type=\"bibr\">[19]</xref>, <xref rid=\"pone.0086718-Lathrop1\" ref-type=\"bibr\">[31]</xref>&#x02013;<xref rid=\"pone.0086718-Dailey2\" ref-type=\"bibr\">[33]</xref>.</p><p>The objectives of this study are to investigate the molecular fingerprints underlying the adaptation of the eight genes encoding the enzymes of the heme biosynthesis pathway in Kingdom Metazoa and, particularly, how these adaptations correlate with the positions of the gene products (enzymes) in the pathway. First, through the analysis of the evolutionary constraints on the protein-coding sequences, we reveal strong purifying selection on the protein sequences and clade-specific adaptations in teleosts and arthropods. We also demonstrate that the first introns of these genes in vertebrates play a role in their erythroid-specific transactivation, highlighting the emergence of erythrocytes in animals. Then, we show that pathway product (heme) feedback control is widely utilized in this pathway. The evolutionarily conserved motifs that enable this control include HRMs found not only in ALAS but also in PBGS. We also discover IREs within 5&#x02032;UTRs and exon-intron boundaries, raising the possibility of regulation of both translation and splicing choice, respectively. We conclude by summarizing the evolution of multifarious controls on the heme biosynthesis pathway.</p></sec><sec id=\"s2\"><title>Results</title><sec id=\"s2a\"><title>Relationship between Selection Pressure and Pathway Position</title><p>Based on the M0 model, assuming a constant evolutionary rate (nonsynonymous versus synonymous rate, <italic>d<sub>N</sub>/d<sub>S</sub></italic>, &#x003c9;) for all branches and all codons of the eight genes encoding the enzymes of the metazoan heme biosynthesis pathway, the &#x003c9; values vary from 0.041 (FECH) to 0.127 (UROS), providing evidence that the sequences of the coding regions of these genes are under negative selection (<xref ref-type=\"fig\" rid=\"pone-0086718-g002\">Figure 2A</xref>). We also evaluated the M1a model, which assumes that there are two groups of codons, subject to purifying selection and neutral evolution, in contrast to the M0 model. We found that the M1a model significantly improves the model (<italic>p</italic>&#x0003c;0.001) and that a large fraction of the codons are under purifying selection (&#x0003e;91% in all genes, except for PPO, where the obtained value was 79%). Notably, the &#x003c9; values at positions four (UROS) and seven (PPO) were highest (0.116 and 0.105, respectively) among the genes of the heme biosynthesis pathway. (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s004\">Table S1</xref>).</p><fig id=\"pone-0086718-g002\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g002</object-id><label>Figure 2</label><caption><title>Selection pressure on heme biosynthesis genes in animals.</title><p>&#x003c9; values (<italic>d<sub>N</sub>/d<sub>S</sub></italic>) were estimated with the M0 model for the eight heme biosynthesis genes in animals (A). The distribution of &#x003c9; values (B), the nonsynonymous substitution rate, <italic>d<sub>N</sub></italic> (C), and the synonymous substitution rate, <italic>d<sub>S</sub></italic> (D). The order of genes follows the linear order of their pathway positions (<xref ref-type=\"fig\" rid=\"pone-0086718-g001\">Figure 1</xref>).</p></caption><graphic xlink:href=\"pone.0086718.g002\"/></fig><p>To determine whether the variation in &#x003c9; values among the genes was statistically significant and showed a relationship with their positions in the pathway, we conducted several tests on the <italic>d<sub>N</sub></italic>, <italic>d<sub>S</sub></italic> and &#x003c9; values obtained for each gene (<xref ref-type=\"fig\" rid=\"pone-0086718-g002\">Figure 2B, 2C, 2D</xref>). First, we determined whether the distributions of the <italic>d<sub>N</sub></italic>, <italic>d<sub>S</sub></italic> and &#x003c9; values were correlated. We found that the distributions of the <italic>d<sub>N</sub></italic>, <italic>d<sub>S</sub></italic> and &#x003c9; values were not correlated for each gene (Kruskal-Wallis rank sum test, <italic>p</italic>&#x0003c;0.0001). Second, the &#x003c9; values of genes located at positions four (UROS) and seven (PPO) were shown to be significantly higher than for the genes located at other positions (Wilcoxon rank sum test, P&#x0003c;0.0001). Third, to determine the significance of the variations in &#x003c9; values among the genes, we conducted multiple comparisons of the &#x003c9; values for each gene pair (see Methods). Three groups were established: group 1 (low &#x003c9; values), consisting of FECH, PBGD and CPO; group 2 (intermediate &#x003c9; values), consisting of ALAS, UROD and PBGS; and group 3 (high &#x003c9; values), consisting of PPO and UROS (<xref ref-type=\"table\" rid=\"pone-0086718-t001\">Table 1</xref>). Fourth, we determined whether the <italic>d<sub>N</sub></italic>, <italic>d<sub>S</sub></italic> and &#x003c9; values were correlated with the pathway positions of the eight enzymes of the heme biosynthesis pathway. It was found that the &#x003c9; values were positively correlated with the pathway positions of the enzymes (Kendall&#x02019;s correlation test: tau&#x0200a;=&#x0200a;0.2202, <italic>P</italic>&#x0003c;0.0001), as were the <italic>d<sub>N</sub></italic> values (Kendall&#x02019;s correlation test: tau&#x0200a;=&#x0200a;0.1625, <italic>P</italic>&#x0003c;0.0001). However, the <italic>d<sub>S</sub></italic> values were not correlated with pathway position (Kendall&#x02019;s correlation test: tau&#x0200a;=&#x0200a;&#x02212;0.0372, <italic>P&#x0200a;=&#x0200a;</italic>0.0243).</p><table-wrap id=\"pone-0086718-t001\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t001</object-id><label>Table 1</label><caption><title>Comparison of &#x003c9; values among eight genes to test the significance of the &#x003c9; variations among genes.</title></caption><alternatives><graphic id=\"pone-0086718-t001-1\" xlink:href=\"pone.0086718.t001\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td colspan=\"3\" align=\"left\" rowspan=\"1\">Group 1</td><td colspan=\"3\" align=\"left\" rowspan=\"1\">Group 2</td><td colspan=\"2\" align=\"left\" rowspan=\"1\">Group 3</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Group 1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Group 2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Group 3</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt101\"><label/><p>For each comparison, a pair of genes was chosen. &#x003c9; value of the gene in the row is constrained to the average value of omega values from the genes in the rows and columns. The difference in the likelihood between the null model M0 and the constrained model was obtained. The significance level is labeled * if <italic>p</italic>&#x0003c;0.05 and ** if <italic>p</italic>&#x0003c;0.01.</p></fn></table-wrap-foot></table-wrap><p>In conclusion, the variations in &#x003c9; and <italic>d<sub>N</sub></italic> values were shown to be correlated with the pathway positions of the eight enzymes of the heme biosynthesis pathway.</p></sec><sec id=\"s2b\"><title>Amino Acid Residues under Positive Selection</title><p>When the genes of the heme biosynthesis pathway are observed to be under strong purifying selection, it is interesting to examine whether there are amino acids in certain lineages that experience positive selection. Due to limitations regarding the number of sequences collected, we focused on the protein sequences of the mammal, teleost and arthropod subgroups. Of the eight enzymes involved in heme biosynthesis, only ALAS2 from teleosts, PBGS from arthropods and UROD from teleosts showed positively selected residues in the branch-site model (<italic>p</italic> value &#x0003c;0.05) (<xref ref-type=\"table\" rid=\"pone-0086718-t002\">Table 2</xref>). In ALAS2 from teleosts, five sites were found to be positively selected (model A/A1, <italic>p</italic>&#x0003c;2.3e-10) (BEB &#x0003e;0.978) (<xref ref-type=\"table\" rid=\"pone-0086718-t003\">Table 3</xref>). Notably, while the amino acid detected in teleost ALAS2 at position 204 is I/M/T, the amino acid E is present at this position in all other vertebrate ALAS1 proteins. We mapped this amino acid in the crystal structure of the protein and found that it is positioned at the ALAS dimer interface. The amino acid R located at position 353 in teleost ALAS2 is also interesting because a K residue is found at this position in all other vertebrate ALAS1, ALAS2 and ALAS proteins. The crystal structure of ALAS indicates that this amino acid forms a hydrogen bond with the ribose-O3&#x02019; of the substrate succinyl-CoA <xref rid=\"pone.0086718-Astner1\" ref-type=\"bibr\">[34]</xref> (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s001\">Figure S1A</xref>).</p><table-wrap id=\"pone-0086718-t002\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t002</object-id><label>Table 2</label><caption><title>Selection pressure of genes of heme-biosynthesis pathway by employing the branch model.</title></caption><alternatives><graphic id=\"pone-0086718-t002-2\" xlink:href=\"pone.0086718.t002\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Model</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Model</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Estimates of parameters</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2&#x00394;L<xref ref-type=\"table-fn\" rid=\"nt102\">a</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">\\n<italic>p</italic> Value<xref ref-type=\"table-fn\" rid=\"nt103\">b</xref>\\n</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS+ALAS1+ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">M0: one ratio</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;&#x0200a;=&#x0200a;0.05356</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost ALAS2 branch</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">two ratios</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;0&#x0200a;=&#x0200a;0.05323, &#x003c9;&#x0200a;=&#x0200a;8.04979</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">10.087074</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1.49E-03</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">M0: one ratio</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;&#x0200a;=&#x0200a;0.06398</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod branch</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">two ratios</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;0&#x0200a;=&#x0200a;0.06398, &#x003c9;&#x0200a;=&#x0200a;999.000</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">4.478516</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3.43E-02</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">M0: one ratio</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;&#x0200a;=&#x0200a;0.06194</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost branch</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">two ratios</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;0&#x0200a;=&#x0200a;0.06135, &#x003c9;&#x0200a;=&#x0200a;13.69790</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">6.01671</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1.42E-02</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt102\"><label>a</label><p>Twice the difference between the log likelihood of M0 (one ratio) and two ratio model.</p></fn><fn id=\"nt103\"><label>b</label><p>\\n<italic>p</italic>(&#x003c7;2) of the likelihood ratio test.</p></fn></table-wrap-foot></table-wrap><table-wrap id=\"pone-0086718-t003\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t003</object-id><label>Table 3</label><caption><title>Positive selection in the different lineages of genes of heme biosynthesis pathway.</title></caption><alternatives><graphic id=\"pone-0086718-t003-3\" xlink:href=\"pone.0086718.t003\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Foreground branches</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2&#x00394;L<xref ref-type=\"table-fn\" rid=\"nt104\">a</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">\\n<italic>p</italic> Value<xref ref-type=\"table-fn\" rid=\"nt105\">b</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Estimates of the parameters in the modified model A<xref ref-type=\"table-fn\" rid=\"nt106\">c</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Positively selected sites<xref ref-type=\"table-fn\" rid=\"nt107\">d</xref>\\n</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">40.20213</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2.29E-10</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">p0&#x0200a;=&#x0200a;0.86464, p1&#x0200a;=&#x0200a;0.07816, p2a&#x0200a;=&#x0200a;0.05246,</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">204E, 243K, 352P, 353K,</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">p2b&#x0200a;=&#x0200a;0.00474, &#x003c9;0&#x0200a;=&#x0200a;0.04595, &#x003c9;2&#x0200a;=&#x0200a;999.0</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">431G</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">23.97615</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">9.75E-07</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">p0&#x0200a;=&#x0200a;0.86020, p1&#x0200a;=&#x0200a;0.03447, p2a&#x0200a;=&#x0200a;0.10127,</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">11Y, 106H, 162C, 195S,</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">p2b&#x0200a;=&#x0200a;0.00406, &#x003c9;0&#x0200a;=&#x0200a;0.06383, &#x003c9;2&#x0200a;=&#x0200a;999.0</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">267K, 274A, 314I</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">25.60371</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">4.19E-07</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">p0&#x0200a;=&#x0200a;0.87046, p1&#x0200a;=&#x0200a;0.07295, p2a&#x0200a;=&#x0200a;0.05222,</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">136Q, 174M, 290K, 297K,</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">p2b&#x0200a;=&#x0200a;0.00438, &#x003c9;0&#x0200a;=&#x0200a;0.06761, &#x003c9;2&#x0200a;=&#x0200a;999.0</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">300T, 316E, 349H</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt104\"><label>a</label><p>Twice the difference between the log likelihood of M0 (one ratio) and two ratio model.</p></fn><fn id=\"nt105\"><label>b</label><p>\\n<italic>p</italic>-value <italic>p</italic>(&#x003c7;2) of the likelihood ratio test.</p></fn><fn id=\"nt106\"><label>c</label><p>For the branch-site model A, the following four classes are demarcated to each amino acid: class 0 with 0&#x0003c; &#x003c9;0&#x0003c;1 in all branches; class 1 with &#x003c9;1&#x0200a;=&#x0200a;1 in all branches; class 2a with foreground &#x003c9;2&#x0003e;1 but background 0&#x0003c; &#x003c9;0&#x0003c;1; and class 2b with foreground &#x003c9;2&#x0003e;1 but background &#x003c9;1&#x0200a;=&#x0200a;1. p0 is the proportion of codons with 0&#x0003c; &#x003c9; &#x0003c;1 in class 0; p1 is the proportion of codons with &#x003c9;1&#x0200a;=&#x0200a;1 in class 1; p2a is the proportion of codons with foreground &#x003c9;2&#x0003e;1 but with background &#x003c9;0 in class 2a; p2b is the proportion of codons with foreground &#x003c9;2&#x0003e;1 but with background &#x003c9;1&#x0200a;=&#x0200a;1 in class 2b.</p></fn><fn id=\"nt107\"><label>d</label><p>Bayes empirical Bayes (BEB) is used to calculate the posterior probabilities to identify the sites (amino acid residues) under positive selection (higher than 95%). The sites are indexed by the amino acids of the site in the human sequence (ALAS1, PBGS, UROD).</p></fn></table-wrap-foot></table-wrap><p>In PBGS from arthropods, seven sites are positively selected (model A/A1, <italic>p</italic>&#x0003c;9.8e-7) (BEB &#x0003e;0.964) (<xref ref-type=\"table\" rid=\"pone-0086718-t003\">Table 3</xref>). The amino acid found in arthropod PBGS at position 11 is I/M, whereas F/Y is present at this position in all vertebrate and cnidarian PBGS proteins. We mapped this amino acid in the crystal structure and found that it is positioned at the PBGS dimer interface and is very close to the Mg<sup>2+</sup> binding site, which has been hypothesized to trigger the conversion of PBGS between its open and closed forms <xref rid=\"pone.0086718-Frankenberg1\" ref-type=\"bibr\">[35]</xref>. The amino acid A is present at position 195 in <italic>Drosophila</italic> PBGS, whereas the amino acid at this position is S in all other animal PBGS proteins. The crystal structure indicates that this amino acid forms the active site pocket <xref rid=\"pone.0086718-Frankenberg1\" ref-type=\"bibr\">[35]</xref> (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s001\">Figure S1B</xref>).</p><p>In UROD from teleosts, seven sites were observed to be positively selected (model A/A1, <italic>p</italic>&#x0003c;4.2e-7) (BEB &#x0003e;0.981) (<xref ref-type=\"table\" rid=\"pone-0086718-t003\">Table 3</xref>). The amino acid S is present at position 290 in teleost UROD, whereas D/E/K is found at this position in other vertebrate, cnidarian and arthropod UROD proteins. The amino acid at position 316 in the teleost UROD is H/R/S, whereas the amino acid at this position is D/E/K in other vertebrate, chordate, arthropod and cnidarian UROD proteins. We mapped both amino acid residues in the crystal structure and found that they are positioned at the UROD dimer interface <xref rid=\"pone.0086718-Phillips1\" ref-type=\"bibr\">[36]</xref> (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s001\">Figure S1C</xref>).</p><p>In summary, we discovered several amino acid residues under positive selection. Some of them are located at the active sites and dimer interfaces of biologically functional enzymes of the heme biosynthesis pathway.</p></sec><sec id=\"s2c\"><title>Detection of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences</title><p>It was previously reported that intron 1 sequences contain <italic>cis</italic>-elements that are necessary for the transcriptional activation of human <italic>PBGS</italic>\\n<xref rid=\"pone.0086718-Kaya1\" ref-type=\"bibr\">[37]</xref>, <italic>PBGD</italic>\\n<xref rid=\"pone.0086718-Mignotte1\" ref-type=\"bibr\">[38]</xref>&#x02013;<xref rid=\"pone.0086718-Grandchamp1\" ref-type=\"bibr\">[40]</xref> and <italic>UROS</italic>\\n<xref rid=\"pone.0086718-Aizencang1\" ref-type=\"bibr\">[41]</xref>, <xref rid=\"pone.0086718-Aizencang2\" ref-type=\"bibr\">[42]</xref>. Therefore, we investigated the selection acting on intron sequences by searching for DNA sequences that are both evolutionarily conserved across 46 vertebrates and located in DNase-hypersensitive site clusters (ENCyclopedia of DNA Elements, ENCODE, for human). We found considerable stretches of DNA longer than 40 nucleotides in intron 1 of <italic>ALAS2</italic>, <italic>PBGS</italic>, <italic>PBGD</italic>, <italic>UROS</italic>, <italic>UROD</italic> and <italic>FECH</italic> that could serve as evolutionarily conserved DNase-hypersensitive sites in host genes (<xref ref-type=\"fig\" rid=\"pone-0086718-g003\">Figure 3</xref>, <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s005\">Table S2</xref>). However, evolutionarily conserved DNase-hypersensitive sites were also found in other intron regions, including in intron 3 of <italic>ALAS1</italic>, intron 8 of <italic>ALAS2</italic>, and introns 2 and 6 of <italic>FECH</italic>. We did not find considerable lengths of evolutionarily conserved DNase-hypersensitive sites in the intron regions of <italic>CPO</italic> and <italic>PPO</italic>. In summary, we revealed evolutionarily conserved DNase-hypersensitive sites in intron regions in six of the eight genes of the heme biosynthesis pathway, implying that conserved regulatory mechanisms acting on intron sequences might be involved in the transactivation of gene expression in the vertebrate heme biosynthesis pathway.</p><fig id=\"pone-0086718-g003\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g003</object-id><label>Figure 3</label><caption><title>Three-dimensional view of the evolutionarily conserved DNase-hypersensitive sites in intron sequences.</title><p>For each gene in the biosynthesis pathway (ALAS1 and ALAS2 are treated separately because they are different genes located on different chromosomes), the length of the intersection of the DNA sequence that is evolutionarily conserved across vertebrates and DNase-hypersensitive sites is indicated on the z-axis. The intron ID is provided on the x-axis. Genes from ALAS1 to FECH are shown on the y-axis and are coded from one to eight according to the linear order of the pathway. (<xref ref-type=\"fig\" rid=\"pone-0086718-g001\">Figure 1</xref>).</p></caption><graphic xlink:href=\"pone.0086718.g003\"/></fig></sec><sec id=\"s2d\"><title>Distribution of IREs in Exon and Intron Sequences</title><p>IREs have been reported to exist in the 5&#x02032; and 3&#x02032;UTR sequences of mRNAs, through which they control the translational efficiency and stability of transcripts, respectively. The 5&#x02032;UTR of human ALAS2 has been shown to contain an IRE. We conducted a survey of the 5&#x02032;UTR sequences of genes of the heme biosynthesis pathway (<xref ref-type=\"table\" rid=\"pone-0086718-t004\">Tables 4</xref> and <xref ref-type=\"table\" rid=\"pone-0086718-t005\">5</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s006\">Table S3</xref>). Notably, the 5&#x02032;UTRs of nine of the thirteen collected vertebrate ALAS2 sequences contained IREs of high quality (see Methods). Intriguingly, we also detected IREs in the 5&#x02032;UTRs of teleost ALAS1 sequences and the ALAS sequences of one chordate (tunicate) and the purple sea urchin (echinoderm) that were of high quality as well as in a sea anemone (cnidarian) and honey bee (arthropod) that were of medium quality. These findings constitute the first demonstration that IREs can be identified in the 5&#x02032;UTRs of ALAS1 sequences from vertebrates and ALAS sequences from arthropods, echinoderms and cnidarians.</p><table-wrap id=\"pone-0086718-t004\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t004</object-id><label>Table 4</label><caption><title>IRE in 5&#x02032;UTR.</title></caption><alternatives><graphic id=\"pone-0086718-t004-4\" xlink:href=\"pone.0086718.t004\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Gene</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Mammal</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Bird,Reptile,Amphibian</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Chordate</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Echinoderm</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Cnidaria</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt108\"><label/><p>Number of species containing potential IRE in 5&#x02032;UTR region of genes involved in heme-biosynthesis pathway.</p></fn><fn id=\"nt109\"><label/><p>Number inside the parenthesis is the total number of 5&#x02032;UTR sequences under investigation.</p></fn><fn id=\"nt110\"><label/><p>NF: Not found.</p></fn></table-wrap-foot></table-wrap><table-wrap id=\"pone-0086718-t005\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t005</object-id><label>Table 5</label><caption><title>IRE in intron.</title></caption><alternatives><graphic id=\"pone-0086718-t005-5\" xlink:href=\"pone.0086718.t005\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Gene</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Mammal</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Bird,Reptile,Amphibian</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Chordate</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Echinoderm</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Cnidaria</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">4(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt111\"><label/><p>Number of species containing potential IRE in intron region of genes involved in heme biosynthesis pathway.</p></fn><fn id=\"nt112\"><label/><p>Number inside the parenthesis is the total number of intron sequences under investigation.</p></fn><fn id=\"nt113\"><label/><p>NF: Not found.</p></fn></table-wrap-foot></table-wrap><p>IREs were also found in the 5&#x02032;UTRs of the PBGD sequences of <italic>Drosophila melanogaster</italic> (fruit fly) and <italic>Apis mellifera</italic> (honey bee) (high quality) as well as that of <italic>Strongylocentrotus purpuratus</italic> (sea urchin) (medium quality) (<xref ref-type=\"table\" rid=\"pone-0086718-t004\">Tables 4</xref> and <xref ref-type=\"table\" rid=\"pone-0086718-t005\">5</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s006\">Table S3</xref>). This observation raised the possibility that the translation of genes other than ALAS in the heme biosynthesis pathway could also respond to iron availability.</p><p>We also found IREs in intron regions. In particular, several potential IREs (of medium quality) exist at the intron-exon boundaries of PBGS from <italic>Mus musculus</italic> (mouse), PBGD from <italic>Loxodonta africana</italic> (elephant) and <italic>Oryctolagus cuniculus</italic> (rabbit) and UROS from <italic>Loxodonta africana</italic> (elephant) and <italic>Mus musculus</italic> (mouse) (<xref ref-type=\"table\" rid=\"pone-0086718-t004\">Tables 4</xref> and <xref ref-type=\"table\" rid=\"pone-0086718-t005\">5</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s006\">Table S3</xref>). This class of IREs forms stem and loop regions that overlap with protein-coding and intron sequences (<xref ref-type=\"fig\" rid=\"pone-0086718-g004\">Figure 4</xref> and <xref ref-type=\"fig\" rid=\"pone-0086718-g005\">Figure 5</xref>). Based on these observations, we conducted a survey to identify potential IREs in the intron-exon boundary sequences of genes in the human and zebrafish genomes. We found 21 and 12 potential IREs of high quality at the intron-exon boundaries of genes in the human and zebrafish genomes, respectively (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s007\">Table S4</xref>).</p><fig id=\"pone-0086718-g004\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g004</object-id><label>Figure 4</label><caption><title>Potential iron-responsive elements (IREs) in the introns and intron-exon boundaries of UROS genes.</title><p>IREs depicted as stem-loop structures are shown in the corresponding intron regions. UROS exon and intron IDs from four species are indicated. The conserved splicing acceptor site AG and the unpaired nucleotide of the IRE structure are also shown.</p></caption><graphic xlink:href=\"pone.0086718.g004\"/></fig><fig id=\"pone-0086718-g005\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g005</object-id><label>Figure 5</label><caption><title>Sequence alignment of the IREs at the intron-exon boundaries of UROS from four species.</title><p>&#x0201c;&#x0003e;&#x0201d; and &#x0201c;&#x0003c;&#x0201d; represent the base pairing of the RNA secondary structure. The potential IRE consensus loop sequence, CAGUGN, and the unpaired nucleotide G are also shown with respect to the location of the IRE hairpin. The intron-exon boundary is indicated as |.</p></caption><graphic xlink:href=\"pone.0086718.g005\"/></fig><p>In summary, we detected potential IREs not only in the 5&#x02032;UTR of ALAS2 but also in the 5&#x02032;UTRs of ALAS and ALAS1. More intriguingly, several potentially conserved IREs might exist at the intron-exon boundaries of genes involved in the heme biosynthesis pathway.</p></sec><sec id=\"s2e\"><title>Distribution of HRMs in Protein Sequences</title><p>The HRM (denoted HRM_t, see Methods) has been shown to bind heme to inhibit the import of ALAS1 into mitochondria. Here, we collected recently published HRM sequences and compiled them into a new HRM consensus sequence (denoted HRM_r, see Methods). Some of the HRM_r sequences are predicted to sense the redox state of the cell and may be critical for triggering the degradation of proteins containing the HRM sequence.</p><p>Only ALAS1, ALAS2 and PBGD protein sequences contain HRM_t. We found HRM_t sequences in the ALAS sequences of chordates, echinoderms, and cnidarians. The ALAS1 genes of all vertebrate species show HRM_t sequences at the N-terminus. In ALAS2, HRM_t sequences were found in several types of mammals and in chickens and Xenopus (<xref ref-type=\"table\" rid=\"pone-0086718-t006\">Tables 6</xref> and <xref ref-type=\"table\" rid=\"pone-0086718-t007\">7</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s008\">Table S5</xref>).</p><table-wrap id=\"pone-0086718-t006\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t006</object-id><label>Table 6</label><caption><title>HRM_t in protein sequence.</title></caption><alternatives><graphic id=\"pone-0086718-t006-6\" xlink:href=\"pone.0086718.t006\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Gene</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Mammal</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Bird,Reptile,Amphibian</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Chordate</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Echinoderm</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Cnidaria</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">7(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">6(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">6(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">4(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt114\"><label/><p>Number of protein species containing potential HRM_t of genes involved in heme-biosyntheis pathway.</p></fn><fn id=\"nt115\"><label/><p>Number inside the parenthesis is the total number of protein sequences under investigation.</p></fn><fn id=\"nt116\"><label/><p>NF: Not found.</p></fn></table-wrap-foot></table-wrap><table-wrap id=\"pone-0086718-t007\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t007</object-id><label>Table 7</label><caption><title>HRM_r in protein sequence.</title></caption><alternatives><graphic id=\"pone-0086718-t007-7\" xlink:href=\"pone.0086718.t007\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Gene</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Mammal</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Bird,Reptile,Amphibian</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Chordate</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Echinoderm</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Cnidaria</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">6(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt117\"><label/><p>Number of protein species containing potential HRM_r of genes in heme biosynthesis pathway.</p></fn><fn id=\"nt118\"><label/><p>Number inside the parenthesis is the total number of protein sequences under investigation.</p></fn><fn id=\"nt119\"><label/><p>NF: Not found.</p></fn></table-wrap-foot></table-wrap><p>Notably, when we used HRM_r as the search sequence for HRMs, new HRM sequences were identified in PBGS (vertebrates, arthropods, and cnidarians) and PBGD (<italic>Drosophila</italic>). This class of HRM is not restricted to the N-terminus of the protein sequence and is conserved with respect to its position in the amino acid sequence (<xref ref-type=\"fig\" rid=\"pone-0086718-g006\">Figure 6</xref>).</p><fig id=\"pone-0086718-g006\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g006</object-id><label>Figure 6</label><caption><title>Heme-regulatory motifs (HRMs) in PBGS and PBGD.</title><p>Multiple sequence alignments of PBGS (A) and PBGD (B) are shown, with HRM_t and HRM_r colored orange and green, respectively. Amino acid numbers for HRM_t and HRM_r are also shown according to the first protein sequence in the alignment.</p></caption><graphic xlink:href=\"pone.0086718.g006\"/></fig></sec></sec><sec id=\"s3\"><title>Discussion</title><p>Elucidation of the evolutionary history of biological pathways sheds light on the principles underlying the evolutionary forces acting on organisms in the environment. The history of pathway evolution may vary among different pathways and ancestral organisms, but understanding the underlying principles helps to reveal the modifications that have taken place in the physiological processes of organisms during their evolutionary history.</p><p>Methods allowing the detection of selection pressure in the evolutionary histories of pathways and networks have made it possible to investigate the existence of fundamental principles driving selection in nature. By estimating the ratio of the nonsynonymous to the synonymous substitution rate for individual protein-encoding genes (&#x003c9;), the types of selection pressure acting on a gene can be identified. In addition to the properties of proteins themselves, the regulatory mechanisms acting on genes are important in metabolic pathways and are also exposed to selection during evolutionary processes. A novel regulatory mechanism affecting the genes in a metabolic pathway can give rise to a new gene function that may become fixed in a lineage, through which we can determine the evolutionary history of a particular biosynthesis pathway throughout different lineages of organisms. To clarify the evolutionary history of the heme biosynthetic pathway of animals, we analyzed the three previously reported regulatory mechanisms related to genes in the heme biosynthesis pathway, which involve DNase I-hypersensitive sites, IREs, and HRMs. Our <italic>in silico</italic> prediction results showed that multiple regulatory mechanisms may exist for the genes in the heme biosynthesis pathway.</p><sec id=\"s3a\"><title>Stronger Selection at the Middle and Penultimate Positions of the Pathway could Result from Self-catalysis</title><p>We conducted a phylogenetic analysis of the eight genes of the heme biosynthesis pathway found in the animal kingdom. The &#x003c9; values at positions four (UROS) and seven (PPO) were shown to be significantly higher than at the other positions in the pathway. Notably, some self-catalysis and by-products of the heme biosynthesis pathway have been reported. Hydroxymethylbilane, the substrate of UROS, can be non-enzymatically cyclized to form uroporphyrinogen I, a useless by-product that leads to uroporphyrin I or coproporphyrin I <xref rid=\"pone.0086718-Bloomer1\" ref-type=\"bibr\">[43]</xref>. The product of UROS can also be auto-oxidized to uroporphyrin III. Protoporphyrinogen, the substrate of PPO, has been shown to be auto-oxidized to protoporphyrin in air, without the need for PPO <xref rid=\"pone.0086718-Bloomer1\" ref-type=\"bibr\">[43]</xref>, <xref rid=\"pone.0086718-Brenner1\" ref-type=\"bibr\">[44]</xref>. The degree of such &#x0201c;leaking&#x0201d; among the biochemical reactions involved in the heme biosynthesis pathway is unknown, and whether this phenomenon is common in animals remains to be determined. We speculate that &#x0201c;leaking&#x0201d; in a biochemical reaction would impact selection pressure and, most likely, lead to decreased evolutionary constraint.</p></sec><sec id=\"s3b\"><title>Biological Function of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences</title><p>Based on the chromatin immunoprecipitation with high-throughput sequencing (ChIP-seq) technique, the transcription factor-binding module GATA1-KLF1 has been hypothesized to allow erythrocyte-specific activation of gene expression through binding to the intron 1 sequences of <italic>ALAS2</italic>, <italic>PBGS</italic>, <italic>PBGD</italic> and <italic>UROS</italic>\\n<xref rid=\"pone.0086718-Tallack1\" ref-type=\"bibr\">[21]</xref>, <xref rid=\"pone.0086718-Cheng1\" ref-type=\"bibr\">[22]</xref>. In this study, we identified intron 1 sequences as a type of DNase-hypersensitive site that is conserved in vertebrates among several genes, including <italic>ALAS2</italic>, <italic>PBGS</italic>, <italic>PBGD</italic>, <italic>UROS</italic> and <italic>UROD</italic>. This finding suggests that in response to the high demand for heme in the erythrocytes of the common ancestor of vertebrates, several transcription factors were recruited to bind to the first intron region to trans-activate the expression of the first five genes of the heme biosynthesis pathway. Notably, an intronic enhancer has been identified in intron 8 of <italic>ALAS2</italic> in mouse erythroleukemia cells <xref rid=\"pone.0086718-Sadlon1\" ref-type=\"bibr\">[45]</xref>&#x02013;<xref rid=\"pone.0086718-Surinya1\" ref-type=\"bibr\">[47]</xref>, which is consistent with our analysis (<xref ref-type=\"fig\" rid=\"pone-0086718-g002\">Figure 2</xref>). We also identified introns 2 and 6 of <italic>FECH</italic> as potential intronic enhancers, although two promoter regions of mouse <italic>FECH</italic> were shown to function in basic and inducible expression <xref rid=\"pone.0086718-Taketani1\" ref-type=\"bibr\">[48]</xref>.</p><p>The use of the first intron as an alternative promoter to induce erythrocyte-specific expression has been documented for several hematopoietic genes, including <italic>Abcg2</italic>\\n<xref rid=\"pone.0086718-Campbell1\" ref-type=\"bibr\">[49]</xref>, <xref rid=\"pone.0086718-Nakanishi1\" ref-type=\"bibr\">[50]</xref>, <italic>Ank1</italic>\\n<xref rid=\"pone.0086718-Birkenmeier1\" ref-type=\"bibr\">[51]</xref> and <italic>Slc11a2</italic>\\n<xref rid=\"pone.0086718-Tallack1\" ref-type=\"bibr\">[21]</xref>. We also determined whether evolutionarily conserved intron regions of these genes could be detected. Without exception, the first intron regions of all three genes contain long stretches of DNA that are both conserved across vertebrates and accessible to DNase attack, suggesting that transcription factor binding occurs (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s009\">Table S6</xref>). The high frequency of alternative first exons found in erythroid genes has been shown to be crucial for the regulation of gene function <xref rid=\"pone.0086718-Tan1\" ref-type=\"bibr\">[52]</xref>, and we propose here that the evolutionarily conserved region in the first intron can serve as either an alternative promoter or enhancer to enable alteration of the first exon during gene transcription <xref rid=\"pone.0086718-Landry1\" ref-type=\"bibr\">[53]</xref>.</p></sec><sec id=\"s3c\"><title>Elucidation of Novel Evolutionarily Conserved IREs at Intron-exon Boundaries</title><p>An IRE was previously identified in the 5&#x02032;UTR of ALAS2 <xref rid=\"pone.0086718-Cox1\" ref-type=\"bibr\">[54]</xref>. However, the detection IREs in the 5&#x02032;UTRs of PBGD sequences is reported for the first time here. The scattered appearance of IREs in enzymes other than ALAS suggests that animal genes acquired IREs at a later point through the process of convergent evolution <xref rid=\"pone.0086718-Piccinelli1\" ref-type=\"bibr\">[55]</xref>. Moreover, whereas the noted IRE found in an echinoderm ALAS (<italic>Ciona intestinalis</italic>) has been reported previously <xref rid=\"pone.0086718-Piccinelli1\" ref-type=\"bibr\">[55]</xref>, our identification of a potential IRE in the ALAS of a cnidarian (sea anemone) and an arthropod (honey bee) suggested that a more thorough search for the existence of IREs in animals is necessary.</p><p>Both the 5&#x02032;UTR and 3&#x02032;UTR are commonly regarded as sites where IREs are located. However, we found that intron sequences can also contain IREs. In particular, we identified an IRE at the intron 4-exon 5 boundary of UROS in <italic>Loxodonta africana</italic> (elephant). We also aligned the IRE in the corresponding region and found that this intron-exon IRE is conserved in humans, rhesus monkeys and rabbits, although these IREs are of low quality. If IREs found at intron-exon boundaries are functional, it is possible that the pre-mRNA splicing junction could be bound by IRP, thereby influencing splicing efficiency or choice.</p><p>In the iron-depleted state, IRP1 and IRP2 are stable and bind to IREs, either to inhibit protein translation (5&#x02032;UTR) or to prevent mRNA degradation (3&#x02032;UTR). In the iron-replete state, the degradation of IRP2 allows IREs to bind iron and eIF4E <xref rid=\"pone.0086718-Ma1\" ref-type=\"bibr\">[56]</xref> to initiate protein translation. Intriguingly, eIF4E has been demonstrated to function as a co-factor in the <italic>Sxl</italic>-dependent female-specific alternative splicing of <italic>msl</italic>-2 as well as <italic>Sxl</italic> premRNAs in <italic>Drosophila</italic>, which is required for sex determination due to the silencing of the X chromosome <xref rid=\"pone.0086718-Graham1\" ref-type=\"bibr\">[57]</xref>. This observation raised the possibility that eIF4E could be involved in splicing events in mammals. To determine how many IREs exist at intron-exon boundaries, we conducted a genome-wide survey to detect IREs of high quality with respect to the nucleotides present at the junctions of CDS-exons and introns in the human and zebrafish genomes. We found 21 and 12 high-quality IREs at human and zebrafish CDS-exon/intron junctions, respectively (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s007\">Table S4</xref>). One of human genes, ZNF446, has been found in the Friendly Alternative Splicing and Transcripts Database (FAST DB) <xref rid=\"pone.0086718-delaGrange1\" ref-type=\"bibr\">[58]</xref> that contains an alternatively spliced transcript (AY279351), which presumably depends on the corresponding junction containing an IRE. This possible phenomenon adds one more layer of complexity to the regulation of genes in the heme biosynthesis pathway in particular and to the regulation of erythropoiesis in general. Furthermore, this finding also calls for an investigation into the mechanism by which iron and IRP regulate the alternative splicing of protein-coding genes and the associated biological effects.</p></sec><sec id=\"s3d\"><title>Elucidation of Novel Evolutionarily Conserved HRMs</title><p>By defining a new HRM (HRM_r), we were able to identify HRMs in most of the studied PBGS genes (76%, 19/25) and all PBGD genes from <italic>Drosophila</italic>. Recently, HRMs have been implicated in regulatory functions other than controlling import into the mitochondria. The HRM in human IRP2 has been shown to be responsible for the ubiquitin-targeted degradation of the IRP2 protein <xref rid=\"pone.0086718-Ishikawa1\" ref-type=\"bibr\">[59]</xref>. The degradation of human circadian Factor Period 2 (hPer2) is also mediated by an HRM, suggesting that metabolic signals can modulate the circadian regulation of gene expression <xref rid=\"pone.0086718-Yang2\" ref-type=\"bibr\">[60]</xref>. Notably, mouse <italic>ALAS1</italic> and <italic>ALAS2</italic> have been shown to be under circadian control and are regulated by mPER1 and mPER2 <xref rid=\"pone.0086718-Zheng1\" ref-type=\"bibr\">[61]</xref>. It has been demonstrated that the HRM of ALAS2 is not involved in the heme-mediated control of import into the mitochondria <xref rid=\"pone.0086718-Munakata1\" ref-type=\"bibr\">[32]</xref>. Therefore, we postulate that the HRMs identified in ALAS2 and PBGS could be involved in protein degradation. This is consistent with the localization of the PBGS enzyme in the cytosol and that, unlike ALAS1 and ALAS2, PBGS does not need to enter the mitochondria to function (<xref ref-type=\"fig\" rid=\"pone-0086718-g001\">Figure 1</xref>). The HRM binds heme and senses both the concentration of heme and the oxidation/reduction state of the cell. Our findings indicate that product feedback control of protein stability could be involved in the evolution of the heme biosynthesis pathway.</p></sec><sec id=\"s3e\"><title>Evolutionary Implications of HRMs and IREs in the Teleost Lineage based on Comparison with other Species</title><p>We found HRMs in the ALAS sequences of chordates and echinoderms. ALAS1 HRM_t exists in all investigated vertebrate species as well. While we also detected ALAS2 HRM_t in certain monophyletic groups (mammals, birds, and amphibians), we did not find any HRM_t sequences in teleosts. Additionally, we identified IREs in the 5&#x02032;UTRs of ALAS sequences from chordates and echinoderms. In the 5&#x02032;UTR of ALAS2, there is an IRE in all of the examined vertebrate species. In the 5&#x02032;UTR of ALAS1, an IRE is found in teleosts, but not in mammals. Based on our phylogenetic analysis, chordate and echinoderm ALAS and vertebrate ALAS1/2 are derived from the same ancestral ALAS gene. We propose that the HRMs and IREs observed in ALAS1 and ALAS2 existed in the ALAS sequence of a common ancestor of vertebrates, chordates, and echinoderms. The unique absence of HRMs in ALAS2 proteins and the presence of IREs in ALAS1 mRNAs in teleosts are striking. We further propose that during the evolutionary branching and speciation of vertebrates, the loss of the HRM from ALAS2 and the retention of the IRE in ALAS1 in teleosts resulted from the adaptation of the heme biosynthesis pathway to the environment. Notably, we have also shown that the amino acid sequences of ALAS2 and UROD have experienced positive selection in the teleost lineage.</p></sec><sec id=\"s3f\"><title>Conclusion: An Integrated View of the Evolution of Multiple Types of Regulation vs. Pathway Position Reveals Different Depths within the Pathway</title><p>We have investigated the evolution of multiple controls modulating the heme biosynthesis pathway. It was illuminating to study the degree by which these combinatorial controls impinge on the pathway from top to bottom, if we regard ALAS, at position one, as the top and FECH, at position eight, as the bottom. In a summary table (<xref ref-type=\"table\" rid=\"pone-0086718-t008\">Table 8</xref>), with the eight genes represented from positions one to eight in rows and the six control mechanisms across the top of the table, it can clearly be observed that there are multiple degrees of regulatory potential with respect to this pathway. Notably, transcriptional control and iron-mediated splicing control infiltrate the middle and bottom of the pathway. Substrate (iron) or product (heme) feedback translational control as well as protein localization and protein stability control primarily infiltrate the top of the pathway. Purifying selection on the protein sequences is widespread but is subject to loose constraint in the middle and near the bottom of the pathway. An intriguing question from the point of view of pathway position is whether the evolvability of pathway regulation depends on the position of the genes in the pathway, thereby leading to differential adaptation. Our research suggests that the investigation of the molecular evolution of a pathway should involve the examination of different control mechanisms related to the activity of the genes that constitute the pathway under investigation. Thus, a new perspective in the field of evolutionary developmental biology could concern the roles played by the <italic>cis</italic>-regulatory regions and protein-coding regions of genes with respect to adaptive mutations <xref rid=\"pone.0086718-Carroll1\" ref-type=\"bibr\">[62]</xref>, <xref rid=\"pone.0086718-Hoekstra1\" ref-type=\"bibr\">[63]</xref>. We are aware that our key findings in this study are predictions based on the current understanding of the heme biosynthesis pathway, and these findings will require substantial experimental work to be confirmed or corrected.</p><table-wrap id=\"pone-0086718-t008\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t008</object-id><label>Table 8</label><caption><title>Multiple regulatory potentials in heme biosynthesis pathway.</title></caption><alternatives><graphic id=\"pone-0086718-t008-8\" xlink:href=\"pone.0086718.t008\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Gene</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Pathway Position</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">First Intron-mediatedTranscription Control<xref ref-type=\"table-fn\" rid=\"nt120\">a</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Splicing Control<xref ref-type=\"table-fn\" rid=\"nt121\">b</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Translational Control<xref ref-type=\"table-fn\" rid=\"nt122\">c</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Protein Localization<xref ref-type=\"table-fn\" rid=\"nt123\">d</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">ProteinStability<xref ref-type=\"table-fn\" rid=\"nt124\">e</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Selection Pressure<xref ref-type=\"table-fn\" rid=\"nt125\">f</xref>\\n</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">4</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">6</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">7</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">8</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt120\"><label>a</label><p>Evolutionarily conserved DNase-hypersensitive sites in intron sequences.</p></fn><fn id=\"nt121\"><label>b</label><p>IRE in intron-exon boundary that could potentially affect splicing.</p></fn><fn id=\"nt122\"><label>c</label><p>IRE in 5&#x02032;UTR,while binding IRP, could potentially inhibit protein translation.</p></fn><fn id=\"nt123\"><label>d</label><p>HRM_t that could potentially block the import of enzyme to mitochondria while binding heme.</p></fn><fn id=\"nt124\"><label>e</label><p>HRM_r that could potentially affect protein stability.</p></fn><fn id=\"nt125\"><label>f</label><p>Genes with two highest &#x003c9; value.</p></fn><fn id=\"nt126\"><label/><p>In <sup>a</sup>, *: &#x0003c;50 bps, **:&#x0003e;50 bps, &#x0003c;100 bps, ***:&#x0003e;100 bps. In <sup>bcde</sup>, *:&#x0003c;&#x0200a;=&#x0200a;0.33, **:&#x0003e;0.33, &#x0003c;&#x0200a;=&#x0200a;0.67, ***:&#x0003e;0.67 for the proportion of species collected.</p></fn></table-wrap-foot></table-wrap></sec></sec><sec sec-type=\"materials|methods\" id=\"s4\"><title>Materials and Methods</title><sec id=\"s4a\"><title>Sequence Collection</title><p>For genes encoding the eight enzymes of the heme biosynthesis pathway in animals, we collected the sequences of organisms whose entire genome sequences were available. Sequences were collected from mammals (seven species), amphibians (two species), birds (three species), reptiles (one species), teleosts (six species), echinoderms (two species), arthropods (eight species) and cnidarians (three species). The amino acid sequences, coding nucleotide sequences, exon sequences and intron sequences were downloaded from NCBI, UCSC <xref rid=\"pone.0086718-Kent1\" ref-type=\"bibr\">[64]</xref>, <xref rid=\"pone.0086718-Dreszer1\" ref-type=\"bibr\">[65]</xref> and Ensembl <xref rid=\"pone.0086718-Flicek1\" ref-type=\"bibr\">[66]</xref>. Some of the protein sequences of ALAS genes were extracted as suggested previously <xref rid=\"pone.0086718-Duncan1\" ref-type=\"bibr\">[19]</xref>. The 5&#x02032;UTR, exon and intron sequences of the coral <italic>Acropora digitifera</italic> came from OIST Marine Genome Unit <xref rid=\"pone.0086718-Shinzato1\" ref-type=\"bibr\">[67]</xref>. The 5&#x02032;UTR, exon and intron sequences of the sea anemone <italic>Nematostella vectensis</italic> were obtained from the DOE Joint Genome Institute <xref rid=\"pone.0086718-Putnam1\" ref-type=\"bibr\">[68]</xref>. The 5&#x02032;UTR, exon and intron sequences of the hydra <italic>Hydra magnipapillata</italic> also came from the DOE Joint Genome Institute <xref rid=\"pone.0086718-Chapman1\" ref-type=\"bibr\">[69]</xref>.</p><p>If a gene model lacked a 5&#x02032;UTR, we searched for a cDNA collection to determine whether there was any possibility of extension in the 5&#x02032; direction with respect to the direction of transcription. If such an extension was possible, a 5&#x02032;UTR was added to the gene model. Subsequently, if this extension added a new exon, which was regarded as the first exon, then a new intron, which was usually the first intron, was hypothesized to exist. The numbers of coding sequences obtained for each gene in the heme biosynthesis pathway were as follows: ALAS (46 genes from 31 species), PBGS (26 genes from 26 species), PBGD (29 genes from 28 species), UROS (28 genes from 28 species), UROD (31 genes from 31 species), CPO (27 genes from 27 species), PPO (26 genes from 26 species) and FECH (30 genes from 30 species) (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s010\">Table S7</xref>). Lists about the number of sequences and the species by taxonomic groups can be found in <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s011\">Table S8</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s012\">S9</xref>. Sequences from species whose genome had not been decoded were not included in this study.</p></sec><sec id=\"s4b\"><title>Sequence Alignment and Phylogenetic Analysis</title><p>The amino acid sequence alignment of the eight enzymes involved in the heme biosynthesis pathway was performed in MEGA5 with the MUSCLE algorithm <xref rid=\"pone.0086718-Tamura1\" ref-type=\"bibr\">[70]</xref>. After the aligned sequences were adjusted manually to confirm their accuracy, topologies of the phylogenetic trees were generated with PHYLIP <xref rid=\"pone.0086718-Felsenstein1\" ref-type=\"bibr\">[71]</xref> and PHYML <xref rid=\"pone.0086718-Guindon1\" ref-type=\"bibr\">[72]</xref> via maximum likelihood (ML) methods. The gamma rate heterogeneity model and the JTT substitution model from Tree-Puzzle <xref rid=\"pone.0086718-Schmidt1\" ref-type=\"bibr\">[73]</xref> and MEGA5 were used for the enzymes of the heme biosynthesis pathway. Branch support was provided via bootstrap analysis, involving a heuristic search with 1000 replicates. The alignments and trees are shown in <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s002\">Figures S2</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s003\">S3</xref>, respectively.</p></sec><sec id=\"s4c\"><title>Analysis of Evolutionary Constraints</title><p>The aligned nucleotide coding sequences without gaps based on the aligned protein sequences and the unrooted tree were fed into the CODEML program of the PAML program package (version V4.4e) <xref rid=\"pone.0086718-Yang3\" ref-type=\"bibr\">[74]</xref>, <xref rid=\"pone.0086718-Yang4\" ref-type=\"bibr\">[75]</xref> to analyze the evolutionary constraints on the coding sequences for the eight genes of the heme biosynthesis pathway. For the different models applied, &#x003c9; (the nonsynonymous versus synonymous rate, <italic>d<sub>N</sub>/d<sub>S</sub></italic>) for codons can be assumed to be less than one (negative selection or purifying selection), equal to one (neutral), or greater than one (positive selection).</p><p>First, we obtained &#x003c9; for each gene by applying the null model M0, assuming a constant &#x003c9; value for all codons and branches. Second, site models M1a (Nearly Neutral) and M2a (Positive Selection) were both used to allow &#x003c9; to vary among sites. In the M1a model, the codons are categorized into two types, one of which shows &#x003c9; values of less than one, while &#x003c9; is equal to one of the other type. In the M2a model, the codons are categorized into three types, one with &#x003c9; values less than one, one for which &#x003c9; is equal to one and one with &#x003c9; values greater than one. We performed the likelihood ratio test for the M1a and M2a models compared to the null model, M0 (twice the log-likelihood difference (2&#x00394;ln<italic>L</italic>) of the two models).</p><p>We also used the branch model to analyze the evolutionary constraints on the mammal, teleost and arthropod branches. We then compared the likelihood of the null model, M0, with the branch model. If the likelihood of the branch model were to be significantly higher than that of the null model M0, it was hypothesized that the branch under consideration is potentially under positive selection. Subsequently, we used the branch-site model to allow variation among sites in the proteins and across branches to determine whether any amino acid residues were under positive selection <xref rid=\"pone.0086718-Zhang2\" ref-type=\"bibr\">[76]</xref>. For branch-site model A, the following four classes were demarcated for each amino acid: class 0 with 0&#x0003c; &#x003c9;<sub>0</sub>&#x0003c;1 in all branches; class 1 with &#x003c9;<sub>1</sub>&#x0200a;=&#x0200a;1 in all branches; class 2a with foreground &#x003c9;<sub>2</sub>&#x02265;1 but background 0&#x0003c; &#x003c9;<sub>0</sub>&#x0003c;1; and class 2b with a foreground &#x003c9;<sub>2</sub>&#x02265;1 but background &#x003c9;<sub>1</sub>&#x0200a;=&#x0200a;1. Null model A1 was the same as A but with the foreground &#x003c9;<sub>2</sub> constrained to one. A likelihood ratio test is used between models A and A1. If the likelihood of model A were to be significantly higher than the likelihood of model A1 (null model) (<italic>p</italic>&#x0003c;0.05), it would indicate that there were amino acid residues under positive selection. Bayes empirical Bayes (BEB) was used to calculate the posterior probability of identifying sites under positive selection (&#x0003e;0.95) <xref rid=\"pone.0086718-Yang5\" ref-type=\"bibr\">[77]</xref>. When we describe the position of the amino acids under positive selection, we use the human sequences (for ALAS1, PBGS and UROD) as a reference to index the sequences. We mapped the positively selected amino acids onto the crystal structures of ALAS <xref rid=\"pone.0086718-Astner1\" ref-type=\"bibr\">[34]</xref>, PBGS <xref rid=\"pone.0086718-Frankenberg1\" ref-type=\"bibr\">[35]</xref> and UROD <xref rid=\"pone.0086718-Phillips1\" ref-type=\"bibr\">[36]</xref>.</p><p>We also compared the &#x003c9; values among the eight genes using a previously described method <xref rid=\"pone.0086718-Lu1\" ref-type=\"bibr\">[2]</xref>, <xref rid=\"pone.0086718-Clotault1\" ref-type=\"bibr\">[8]</xref>. When the difference of the &#x003c9; values between two genes, such as gene A, with &#x003c9;<sub>A</sub> and a likelihood of L<sub>A0</sub> in the M0 null model, and gene B, with &#x003c9;<sub>B</sub> and a likelihood of L<sub>B0</sub> in the M0 null model, was significant, we were able to find a &#x003c9;<sub>n</sub> between the two genes that was significantly different from those of the two genes being compared. To perform this test, we set &#x003c9;<sub>n</sub> as the average &#x003c9; value of the two genes being compared (&#x003c9;<sub>n</sub>&#x0200a;=&#x0200a;(&#x003c9;<sub>A</sub>+&#x003c9;<sub>B</sub>)/2). We then acquired the likelihood values (L<sub>A</sub> and L<sub>B</sub>) after we constrained the &#x003c9; of genes A and B as &#x003c9;<sub>n</sub>. Subsequently, the statistical significance (df&#x0200a;=&#x0200a;1, <italic>p</italic>&#x0003c;0.05) of the difference in &#x003c9; was determined from the difference in the likelihood between the null model M0 and the constrained model, as follows: 2*(L<sub>A</sub> &#x02013; L<sub>A0</sub>) and 2*(L<sub>B</sub> &#x02013; L<sub>B0</sub>).</p></sec><sec id=\"s4d\"><title>Detection of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences</title><p>We used the UCSC table browser <xref rid=\"pone.0086718-Rosenbloom1\" ref-type=\"bibr\">[78]</xref>, <xref rid=\"pone.0086718-Karolchik1\" ref-type=\"bibr\">[79]</xref> and Galaxy <xref rid=\"pone.0086718-Goecks1\" ref-type=\"bibr\">[80]</xref> to extract the intron sequences at the intersections of regions that are conserved in vertebrates (hg19, phastConsElements46way) and DNase-hypersensitive sites (hg19, wgEncodeRegDnaseClustered) <xref rid=\"pone.0086718-ENCODEProjectConsortium1\" ref-type=\"bibr\">[81]</xref>.</p></sec><sec id=\"s4e\"><title>Detection of IREs</title><p>We used the SIRE web service to identify potential IRE sites in the gene sequences <xref rid=\"pone.0086718-Campillos1\" ref-type=\"bibr\">[82]</xref>. SIRE takes into consideration the non-canonical sequence as indicated by SELEX. By allowing 18 motifs to be confirmed as binding to IRP1 or IRP2, SIRE accepted the input sequence and reported the motif type, the free energy of the secondary structure and the level of stringency as High, Medium, or Low. A batch version of the same algorithm was also developed for the genome-wide detection of IREs at the intron-exon boundaries of human and zebrafish genes.</p></sec><sec id=\"s4f\"><title>Detection of HRMs</title><p>The HRM sequence, N/K/R-C-P-K or a hydrophobic residue-L/M, has been commonly used to detect HRMs <xref rid=\"pone.0086718-Duncan1\" ref-type=\"bibr\">[19]</xref>, <xref rid=\"pone.0086718-Lathrop1\" ref-type=\"bibr\">[31]</xref> (denoted HRM_t). By identifying new occurrences of HRMs that have been shown to function biologically (human ALAS1; human ALAS2 <xref rid=\"pone.0086718-Munakata1\" ref-type=\"bibr\">[32]</xref>; human, mouse, rat, spalax, and zebrafish PER2 <xref rid=\"pone.0086718-Yang2\" ref-type=\"bibr\">[60]</xref>; human IRP2 <xref rid=\"pone.0086718-Ishikawa1\" ref-type=\"bibr\">[59]</xref>; human STC2 <xref rid=\"pone.0086718-Jiang1\" ref-type=\"bibr\">[83]</xref>; and human, mouse and rabbit eIF2alpha <xref rid=\"pone.0086718-Igarashi1\" ref-type=\"bibr\">[84]</xref>), we also identified a new HRM motif, denoted HRM_r, A/C/F/G/I/R/S/Q-A/C/K/H/L/N/R/S/T-C-P-A/E/F/I/K/S/V/Y-A/D/H/I/L/M/T/V-A/L/M/P/R/S. We used seven sites in HRM_r rather than five sites in HRM_t to increase the specificity of HRM_r. We note that HRM_r is similar to, but not the same as HRM_t.</p></sec></sec><sec sec-type=\"supplementary-material\" id=\"s5\"><title>Supporting Information</title><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s001\"><label>Figure S1</label><caption><p>\\n<bold>The collection of figures describing the protein structures of enzymes involved in the heme biosynthesis pathway and the positions of the positively selected residues.</bold> Homodimer structures are shown with the monomers colored in white and yellow. Positively selected amino acid residues are colored in red. Substrate analogs or prosthetic groups are colored in green. (A) ALAS2 of teleost; (B) PBGS of arthropod; (C) UROD of teleost.</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s001.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s002\"><label>Figure S2</label><caption><p>\\n<bold>The aligned protein sequences of eight genes of heme biosynthesis pathway in animals.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s002.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s003\"><label>Figure S3</label><caption><p>\\n<bold>The collection of figures for the maximum likelihood phylogeny of protein sequences for eight genes of heme biosynthesis pathway in animals.</bold> Bootstrap values &#x0003e;70% are indicated. The bootstrap values are displayed only for the branches of the main lineages. (A) ALAS, (B) PBGS, (C) PBGD, (D) UROS, (E) UROD, (F) CPO, (G) PPO, (H) FECH.</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s003.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s004\"><label>Table S1</label><caption><p>\\n<bold>Model test (M1a) for selection of genes in heme biosynthesis pathway.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s004.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s005\"><label>Table S2</label><caption><p>\\n<bold>Length of evolutionarily conserved DNase-hypersensitive sites in intron sequences.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s005.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s006\"><label>Table S3</label><caption><p>\\n<bold>Potential IRE in eight genes of heme biosynthesis pathway.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s006.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s007\"><label>Table S4</label><caption><p>\\n<bold>Genomewide detection of potential IRE in exon-intron boundary in human and zebrafish genes.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s007.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s008\"><label>Table S5</label><caption><p>\\n<bold>Potential HRM in eight genes of heme biosynthesis pathway (HRM_t and HRM_r).</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s008.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s009\"><label>Table S6</label><caption><p>\\n<bold>Length of evolutionarily conserved DNase-hypersensitive sites in intron sequences (bps) for </bold>\\n<bold><italic>Abcg2</italic></bold>\\n<bold>, </bold>\\n<bold><italic>Ank1</italic></bold>\\n<bold> and </bold>\\n<bold><italic>Slc11a2</italic></bold>\\n<bold>.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s009.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s010\"><label>Table S7</label><caption><p>\\n<bold>IDs of genes of heme biosynthesis pathway in animals.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s010.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s011\"><label>Table S8</label><caption><p>\\n<bold>Number of sequences by taxonomic groups.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s011.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s012\"><label>Table S9</label><caption><p>\\n<bold>Species by taxonomic groups.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s012.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material></sec></body><back><ref-list><title>References</title><ref id=\"pone.0086718-Rausher1\"><label>1</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Rausher</surname><given-names>MD</given-names></name>, <name><surname>Miller</surname><given-names>RE</given-names></name>, <name><surname>Tiffin</surname><given-names>P</given-names></name> (<year>1999</year>) <article-title>Patterns of evolutionary rate variation among genes of the anthocyanin biosynthetic pathway</article-title>. <source>Mol Biol Evol</source>\\n<volume>16</volume>: <fpage>266</fpage>&#x02013;<lpage>274</lpage>.<pub-id pub-id-type=\"pmid\">10028292</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Lu1\"><label>2</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Lu</surname><given-names>Y</given-names></name>, <name><surname>Rausher</surname><given-names>MD</given-names></name> (<year>2003</year>) <article-title>Evolutionary rate variation in anthocyanin pathway genes</article-title>. <source>Mol Biol Evol</source>\\n<volume>20</volume>: <fpage>1844</fpage>&#x02013;<lpage>1853</lpage>.<pub-id pub-id-type=\"pmid\">12885963</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Rausher2\"><label>3</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Rausher</surname><given-names>MD</given-names></name>, <name><surname>Lu</surname><given-names>Y</given-names></name>, <name><surname>Meyer</surname><given-names>K</given-names></name> (<year>2008</year>) <article-title>Variation in constraint versus positive selection as an explanation for evolutionary rate variation among anthocyanin genes</article-title>. <source>J Mol Evol</source>\\n<volume>67</volume>: <fpage>137</fpage>&#x02013;<lpage>144</lpage>.<pub-id pub-id-type=\"pmid\">18654810</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Ramsay1\"><label>4</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Ramsay</surname><given-names>H</given-names></name>, <name><surname>Rieseberg</surname><given-names>LH</given-names></name>, <name><surname>Ritland</surname><given-names>K</given-names></name> (<year>2009</year>) <article-title>The correlation of evolutionary rate with pathway position in plant terpenoid biosynthesis</article-title>. <source>Mol Biol Evol</source>\\n<volume>26</volume>: <fpage>1045</fpage>&#x02013;<lpage>1053</lpage>.<pub-id pub-id-type=\"pmid\">19188263</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yu1\"><label>5</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yu</surname><given-names>G</given-names></name>, <name><surname>Olsen</surname><given-names>KM</given-names></name>, <name><surname>Schaal</surname><given-names>BA</given-names></name> (<year>2011</year>) <article-title>Molecular evolution of the endosperm starch synthesis pathway genes in rice (Oryza sativa L.) and its wild ancestor, O. rufipogon L. Mol Biol Evol</article-title>. <volume>28</volume>: <fpage>659</fpage>&#x02013;<lpage>671</lpage>.</mixed-citation></ref><ref id=\"pone.0086718-Li1\"><label>6</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Li</surname><given-names>C</given-names></name>, <name><surname>Li</surname><given-names>QG</given-names></name>, <name><surname>Dunwell</surname><given-names>JM</given-names></name>, <name><surname>Zhang</surname><given-names>YM</given-names></name> (<year>2012</year>) <article-title>Divergent evolutionary pattern of starch biosynthetic pathway genes in grasses and dicots</article-title>. <source>Mol Biol Evol</source>\\n<volume>29</volume>: <fpage>3227</fpage>&#x02013;<lpage>3236</lpage>.<pub-id pub-id-type=\"pmid\">22586327</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yang1\"><label>7</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yang</surname><given-names>YH</given-names></name>, <name><surname>Zhang</surname><given-names>FM</given-names></name>, <name><surname>Ge</surname><given-names>S</given-names></name> (<year>2009</year>) <article-title>Evolutionary rate patterns of the Gibberellin pathway genes</article-title>. <source>BMC Evol Biol</source>\\n<volume>9</volume>: <fpage>206</fpage>.<pub-id pub-id-type=\"pmid\">19689796</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Clotault1\"><label>8</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Clotault</surname><given-names>J</given-names></name>, <name><surname>Peltier</surname><given-names>D</given-names></name>, <name><surname>Soufflet-Freslon</surname><given-names>V</given-names></name>, <name><surname>Briard</surname><given-names>M</given-names></name>, <name><surname>Geoffriau</surname><given-names>E</given-names></name> (<year>2012</year>) <article-title>Differential selection on carotenoid biosynthesis genes as a function of gene position in the metabolic pathway: a study on the carrot and dicots</article-title>. <source>PLoS One</source>\\n<volume>7</volume>: <fpage>e38724</fpage>.<pub-id pub-id-type=\"pmid\">22737218</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Philip1\"><label>9</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Philip</surname><given-names>S</given-names></name>, <name><surname>Machado</surname><given-names>JP</given-names></name>, <name><surname>Maldonado</surname><given-names>E</given-names></name>, <name><surname>Vasconcelos</surname><given-names>V</given-names></name>, <name><surname>O&#x02019;Brien</surname><given-names>SJ</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>Fish lateral line innovation: insights into the evolutionary genomic dynamics of a unique mechanosensory organ</article-title>. <source>Mol Biol Evol</source>\\n<volume>29</volume>: <fpage>3887</fpage>&#x02013;<lpage>3898</lpage>.<pub-id pub-id-type=\"pmid\">22844072</pub-id></mixed-citation></ref><ref id=\"pone.0086718-OlsonManning1\"><label>10</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Olson-Manning</surname><given-names>CF</given-names></name>, <name><surname>Lee</surname><given-names>CR</given-names></name>, <name><surname>Rausher</surname><given-names>MD</given-names></name>, <name><surname>Mitchell-Olds</surname><given-names>T</given-names></name> (<year>2013</year>) <article-title>Evolution of flux control in the glucosinolate pathway in Arabidopsis thaliana</article-title>. <source>Mol Biol Evol</source>\\n<volume>30</volume>: <fpage>14</fpage>&#x02013;<lpage>23</lpage>.<pub-id pub-id-type=\"pmid\">22923463</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Montanucci1\"><label>11</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Montanucci</surname><given-names>L</given-names></name>, <name><surname>Laayouni</surname><given-names>H</given-names></name>, <name><surname>Dall&#x02019;Olio</surname><given-names>GM</given-names></name>, <name><surname>Bertranpetit</surname><given-names>J</given-names></name> (<year>2011</year>) <article-title>Molecular evolution and network-level analysis of the N-glycosylation metabolic pathway across primates</article-title>. <source>Mol Biol Evol</source>\\n<volume>28</volume>: <fpage>813</fpage>&#x02013;<lpage>823</lpage>.<pub-id pub-id-type=\"pmid\">20924085</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Cork1\"><label>12</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Cork</surname><given-names>JM</given-names></name>, <name><surname>Purugganan</surname><given-names>MD</given-names></name> (<year>2004</year>) <article-title>The evolution of molecular genetic pathways and networks</article-title>. <source>Bioessays</source>\\n<volume>26</volume>: <fpage>479</fpage>&#x02013;<lpage>484</lpage>.<pub-id pub-id-type=\"pmid\">15112228</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Khan1\"><label>13</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Khan</surname><given-names>AA</given-names></name>, <name><surname>Quigley</surname><given-names>JG</given-names></name> (<year>2011</year>) <article-title>Control of intracellular heme levels: heme transporters and heme oxygenases</article-title>. <source>Biochim Biophys Acta</source>\\n<volume>1813</volume>: <fpage>668</fpage>&#x02013;<lpage>682</lpage>.<pub-id pub-id-type=\"pmid\">21238504</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Heinemann1\"><label>14</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Heinemann</surname><given-names>IU</given-names></name>, <name><surname>Jahn</surname><given-names>M</given-names></name>, <name><surname>Jahn</surname><given-names>D</given-names></name> (<year>2008</year>) <article-title>The biochemistry of heme biosynthesis</article-title>. <source>Arch Biochem Biophys</source>\\n<volume>474</volume>: <fpage>238</fpage>&#x02013;<lpage>251</lpage>.<pub-id pub-id-type=\"pmid\">18314007</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Furuyama1\"><label>15</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Furuyama</surname><given-names>K</given-names></name>, <name><surname>Kaneko</surname><given-names>K</given-names></name>, <name><surname>Vargas</surname><given-names>PD</given-names></name> (<year>2007</year>) <article-title>Heme as a magnificent molecule with multiple missions: heme determines its own fate and governs cellular homeostasis</article-title>. <source>Tohoku J Exp Med</source>\\n<volume>213</volume>: <fpage>1</fpage>&#x02013;<lpage>16</lpage>.<pub-id pub-id-type=\"pmid\">17785948</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Zhang1\"><label>16</label><mixed-citation publication-type=\"other\">Zhang L (2011) Heme biology : the secret life of heme in regulating diverse biological processes. Hackensack, NJ: World Scientific. 213 p. p.</mixed-citation></ref><ref id=\"pone.0086718-Dailey1\"><label>17</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Dailey</surname><given-names>HA</given-names></name>, <name><surname>Meissner</surname><given-names>PN</given-names></name> (<year>2013</year>) <article-title>Erythroid heme biosynthesis and its disorders</article-title>. <source>Cold Spring Harb Perspect Med</source>\\n<volume>3</volume>: <fpage>a011676</fpage>.<pub-id pub-id-type=\"pmid\">23471474</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Layer1\"><label>18</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Layer</surname><given-names>G</given-names></name>, <name><surname>Reichelt</surname><given-names>J</given-names></name>, <name><surname>Jahn</surname><given-names>D</given-names></name>, <name><surname>Heinz</surname><given-names>DW</given-names></name> (<year>2010</year>) <article-title>Structure and function of enzymes in heme biosynthesis</article-title>. <source>Protein Sci</source>\\n<volume>19</volume>: <fpage>1137</fpage>&#x02013;<lpage>1161</lpage>.<pub-id pub-id-type=\"pmid\">20506125</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Duncan1\"><label>19</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Duncan</surname><given-names>R</given-names></name>, <name><surname>Faggart</surname><given-names>MA</given-names></name>, <name><surname>Roger</surname><given-names>AJ</given-names></name>, <name><surname>Cornell</surname><given-names>NW</given-names></name> (<year>1999</year>) <article-title>Phylogenetic analysis of the 5-aminolevulinate synthase gene</article-title>. <source>Mol Biol Evol</source>\\n<volume>16</volume>: <fpage>383</fpage>&#x02013;<lpage>396</lpage>.<pub-id pub-id-type=\"pmid\">10331264</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Kerenyi1\"><label>20</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Kerenyi</surname><given-names>MA</given-names></name>, <name><surname>Orkin</surname><given-names>SH</given-names></name> (<year>2010</year>) <article-title>Networking erythropoiesis</article-title>. <source>J Exp Med</source>\\n<volume>207</volume>: <fpage>2537</fpage>&#x02013;<lpage>2541</lpage>.<pub-id pub-id-type=\"pmid\">21098097</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Tallack1\"><label>21</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Tallack</surname><given-names>MR</given-names></name>, <name><surname>Whitington</surname><given-names>T</given-names></name>, <name><surname>Yuen</surname><given-names>WS</given-names></name>, <name><surname>Wainwright</surname><given-names>EN</given-names></name>, <name><surname>Keys</surname><given-names>JR</given-names></name>, <etal>et al</etal> (<year>2010</year>) <article-title>A global role for KLF1 in erythropoiesis revealed by ChIP-seq in primary erythroid cells</article-title>. <source>Genome Res</source>\\n<volume>20</volume>: <fpage>1052</fpage>&#x02013;<lpage>1063</lpage>.<pub-id pub-id-type=\"pmid\">20508144</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Cheng1\"><label>22</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Cheng</surname><given-names>Y</given-names></name>, <name><surname>Wu</surname><given-names>W</given-names></name>, <name><surname>Kumar</surname><given-names>SA</given-names></name>, <name><surname>Yu</surname><given-names>D</given-names></name>, <name><surname>Deng</surname><given-names>W</given-names></name>, <etal>et al</etal> (<year>2009</year>) <article-title>Erythroid GATA1 function revealed by genome-wide analysis of transcription factor occupancy, histone modifications, and mRNA expression</article-title>. <source>Genome Res</source>\\n<volume>19</volume>: <fpage>2172</fpage>&#x02013;<lpage>2184</lpage>.<pub-id pub-id-type=\"pmid\">19887574</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Kassouf1\"><label>23</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Kassouf</surname><given-names>MT</given-names></name>, <name><surname>Hughes</surname><given-names>JR</given-names></name>, <name><surname>Taylor</surname><given-names>S</given-names></name>, <name><surname>McGowan</surname><given-names>SJ</given-names></name>, <name><surname>Soneji</surname><given-names>S</given-names></name>, <etal>et al</etal> (<year>2010</year>) <article-title>Genome-wide identification of TAL1&#x02019;s functional targets: insights into its mechanisms of action in primary erythroid cells</article-title>. <source>Genome Res</source>\\n<volume>20</volume>: <fpage>1064</fpage>&#x02013;<lpage>1083</lpage>.<pub-id pub-id-type=\"pmid\">20566737</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Ajioka1\"><label>24</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Ajioka</surname><given-names>RS</given-names></name>, <name><surname>Phillips</surname><given-names>JD</given-names></name>, <name><surname>Kushner</surname><given-names>JP</given-names></name> (<year>2006</year>) <article-title>Biosynthesis of heme in mammals</article-title>. <source>Biochim Biophys Acta</source>\\n<volume>1763</volume>: <fpage>723</fpage>&#x02013;<lpage>736</lpage>.<pub-id pub-id-type=\"pmid\">16839620</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Natarajan1\"><label>25</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Natarajan</surname><given-names>A</given-names></name>, <name><surname>Yardimci</surname><given-names>GG</given-names></name>, <name><surname>Sheffield</surname><given-names>NC</given-names></name>, <name><surname>Crawford</surname><given-names>GE</given-names></name>, <name><surname>Ohler</surname><given-names>U</given-names></name> (<year>2012</year>) <article-title>Predicting cell-type-specific gene expression from regions of open chromatin</article-title>. <source>Genome Res</source>\\n<volume>22</volume>: <fpage>1711</fpage>&#x02013;<lpage>1722</lpage>.<pub-id pub-id-type=\"pmid\">22955983</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Neph1\"><label>26</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Neph</surname><given-names>S</given-names></name>, <name><surname>Vierstra</surname><given-names>J</given-names></name>, <name><surname>Stergachis</surname><given-names>AB</given-names></name>, <name><surname>Reynolds</surname><given-names>AP</given-names></name>, <name><surname>Haugen</surname><given-names>E</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>An expansive human regulatory lexicon encoded in transcription factor footprints</article-title>. <source>Nature</source>\\n<volume>489</volume>: <fpage>83</fpage>&#x02013;<lpage>90</lpage>.<pub-id pub-id-type=\"pmid\">22955618</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Thurman1\"><label>27</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Thurman</surname><given-names>RE</given-names></name>, <name><surname>Rynes</surname><given-names>E</given-names></name>, <name><surname>Humbert</surname><given-names>R</given-names></name>, <name><surname>Vierstra</surname><given-names>J</given-names></name>, <name><surname>Maurano</surname><given-names>MT</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>The accessible chromatin landscape of the human genome</article-title>. <source>Nature</source>\\n<volume>489</volume>: <fpage>75</fpage>&#x02013;<lpage>82</lpage>.<pub-id pub-id-type=\"pmid\">22955617</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Evstatiev1\"><label>28</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Evstatiev</surname><given-names>R</given-names></name>, <name><surname>Gasche</surname><given-names>C</given-names></name> (<year>2012</year>) <article-title>Iron sensing and signalling</article-title>. <source>Gut</source>\\n<volume>61</volume>: <fpage>933</fpage>&#x02013;<lpage>952</lpage>.<pub-id pub-id-type=\"pmid\">22016365</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Anderson1\"><label>29</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Anderson</surname><given-names>CP</given-names></name>, <name><surname>Shen</surname><given-names>M</given-names></name>, <name><surname>Eisenstein</surname><given-names>RS</given-names></name>, <name><surname>Leibold</surname><given-names>EA</given-names></name> (<year>2012</year>) <article-title>Mammalian iron metabolism and its control by iron regulatory proteins</article-title>. <source>Biochim Biophys Acta</source>\\n<volume>1823</volume>: <fpage>1468</fpage>&#x02013;<lpage>1483</lpage>.<pub-id pub-id-type=\"pmid\">22610083</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Chen1\"><label>30</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Chen</surname><given-names>CY</given-names></name>, <name><surname>Paw</surname><given-names>BH</given-names></name> (<year>2012</year>) <article-title>Cellular and mitochondrial iron homeostasis in vertebrates</article-title>. <source>Biochimica Et Biophysica Acta-Molecular Cell Research</source>\\n<volume>1823</volume>: <fpage>1459</fpage>&#x02013;<lpage>1467</lpage>.</mixed-citation></ref><ref id=\"pone.0086718-Lathrop1\"><label>31</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Lathrop</surname><given-names>JT</given-names></name>, <name><surname>Timko</surname><given-names>MP</given-names></name> (<year>1993</year>) <article-title>Regulation by heme of mitochondrial protein transport through a conserved amino acid motif</article-title>. <source>Science</source>\\n<volume>259</volume>: <fpage>522</fpage>&#x02013;<lpage>525</lpage>.<pub-id pub-id-type=\"pmid\">8424176</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Munakata1\"><label>32</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Munakata</surname><given-names>H</given-names></name>, <name><surname>Sun</surname><given-names>JY</given-names></name>, <name><surname>Yoshida</surname><given-names>K</given-names></name>, <name><surname>Nakatani</surname><given-names>T</given-names></name>, <name><surname>Honda</surname><given-names>E</given-names></name>, <etal>et al</etal> (<year>2004</year>) <article-title>Role of the heme regulatory motif in the heme-mediated inhibition of mitochondrial import of 5-aminolevulinate synthase</article-title>. <source>J Biochem</source>\\n<volume>136</volume>: <fpage>233</fpage>&#x02013;<lpage>238</lpage>.<pub-id pub-id-type=\"pmid\">15496594</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Dailey2\"><label>33</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Dailey</surname><given-names>TA</given-names></name>, <name><surname>Woodruff</surname><given-names>JH</given-names></name>, <name><surname>Dailey</surname><given-names>HA</given-names></name> (<year>2005</year>) <article-title>Examination of mitochondrial protein targeting of haem synthetic enzymes: in vivo identification of three functional haem-responsive motifs in 5-aminolaevulinate synthase</article-title>. <source>Biochem J</source>\\n<volume>386</volume>: <fpage>381</fpage>&#x02013;<lpage>386</lpage>.<pub-id pub-id-type=\"pmid\">15482256</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Astner1\"><label>34</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Astner</surname><given-names>I</given-names></name>, <name><surname>Schulze</surname><given-names>JO</given-names></name>, <name><surname>van den Heuvel</surname><given-names>J</given-names></name>, <name><surname>Jahn</surname><given-names>D</given-names></name>, <name><surname>Schubert</surname><given-names>WD</given-names></name>, <etal>et al</etal> (<year>2005</year>) <article-title>Crystal structure of 5-aminolevulinate synthase, the first enzyme of heme biosynthesis, and its link to XLSA in humans</article-title>. <source>Embo J</source>\\n<volume>24</volume>: <fpage>3166</fpage>&#x02013;<lpage>3177</lpage>.<pub-id pub-id-type=\"pmid\">16121195</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Frankenberg1\"><label>35</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Frankenberg</surname><given-names>N</given-names></name>, <name><surname>Erskine</surname><given-names>PT</given-names></name>, <name><surname>Cooper</surname><given-names>JB</given-names></name>, <name><surname>Shoolingin-Jordan</surname><given-names>PM</given-names></name>, <name><surname>Jahn</surname><given-names>D</given-names></name>, <etal>et al</etal> (<year>1999</year>) <article-title>High resolution crystal structure of a Mg2+-dependent porphobilinogen synthase</article-title>. <source>J Mol Biol</source>\\n<volume>289</volume>: <fpage>591</fpage>&#x02013;<lpage>602</lpage>.<pub-id pub-id-type=\"pmid\">10356331</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Phillips1\"><label>36</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Phillips</surname><given-names>JD</given-names></name>, <name><surname>Whitby</surname><given-names>FG</given-names></name>, <name><surname>Kushner</surname><given-names>JP</given-names></name>, <name><surname>Hill</surname><given-names>CP</given-names></name> (<year>2003</year>) <article-title>Structural basis for tetrapyrrole coordination by uroporphyrinogen decarboxylase</article-title>. <source>Embo J</source>\\n<volume>22</volume>: <fpage>6225</fpage>&#x02013;<lpage>6233</lpage>.<pub-id pub-id-type=\"pmid\">14633982</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Kaya1\"><label>37</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Kaya</surname><given-names>AH</given-names></name>, <name><surname>Plewinska</surname><given-names>M</given-names></name>, <name><surname>Wong</surname><given-names>DM</given-names></name>, <name><surname>Desnick</surname><given-names>RJ</given-names></name>, <name><surname>Wetmur</surname><given-names>JG</given-names></name> (<year>1994</year>) <article-title>Human delta-aminolevulinate dehydratase (ALAD) gene: structure and alternative splicing of the erythroid and housekeeping mRNAs</article-title>. <source>Genomics</source>\\n<volume>19</volume>: <fpage>242</fpage>&#x02013;<lpage>248</lpage>.<pub-id pub-id-type=\"pmid\">8188255</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Mignotte1\"><label>38</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Mignotte</surname><given-names>V</given-names></name>, <name><surname>Eleouet</surname><given-names>JF</given-names></name>, <name><surname>Raich</surname><given-names>N</given-names></name>, <name><surname>Romeo</surname><given-names>PH</given-names></name> (<year>1989</year>) <article-title>Cis- and trans-acting elements involved in the regulation of the erythroid promoter of the human porphobilinogen deaminase gene</article-title>. <source>Proc Natl Acad Sci U S A</source>\\n<volume>86</volume>: <fpage>6548</fpage>&#x02013;<lpage>6552</lpage>.<pub-id pub-id-type=\"pmid\">2771941</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Chretien1\"><label>39</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Chretien</surname><given-names>S</given-names></name>, <name><surname>Dubart</surname><given-names>A</given-names></name>, <name><surname>Beaupain</surname><given-names>D</given-names></name>, <name><surname>Raich</surname><given-names>N</given-names></name>, <name><surname>Grandchamp</surname><given-names>B</given-names></name>, <etal>et al</etal> (<year>1988</year>) <article-title>Alternative transcription and splicing of the human porphobilinogen deaminase gene result either in tissue-specific or in housekeeping expression</article-title>. <source>Proc Natl Acad Sci U S A</source>\\n<volume>85</volume>: <fpage>6</fpage>&#x02013;<lpage>10</lpage>.<pub-id pub-id-type=\"pmid\">3422427</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Grandchamp1\"><label>40</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Grandchamp</surname><given-names>B</given-names></name>, <name><surname>De Verneuil</surname><given-names>H</given-names></name>, <name><surname>Beaumont</surname><given-names>C</given-names></name>, <name><surname>Chretien</surname><given-names>S</given-names></name>, <name><surname>Walter</surname><given-names>O</given-names></name>, <etal>et al</etal> (<year>1987</year>) <article-title>Tissue-specific expression of porphobilinogen deaminase. Two isoenzymes from a single gene</article-title>. <source>Eur J Biochem</source>\\n<volume>162</volume>: <fpage>105</fpage>&#x02013;<lpage>110</lpage>.<pub-id pub-id-type=\"pmid\">3816774</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Aizencang1\"><label>41</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Aizencang</surname><given-names>G</given-names></name>, <name><surname>Solis</surname><given-names>C</given-names></name>, <name><surname>Bishop</surname><given-names>DF</given-names></name>, <name><surname>Warner</surname><given-names>C</given-names></name>, <name><surname>Desnick</surname><given-names>RJ</given-names></name> (<year>2000</year>) <article-title>Human uroporphyrinogen-III synthase: genomic organization, alternative promoters, and erythroid-specific expression</article-title>. <source>Genomics</source>\\n<volume>70</volume>: <fpage>223</fpage>&#x02013;<lpage>231</lpage>.<pub-id pub-id-type=\"pmid\">11112350</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Aizencang2\"><label>42</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Aizencang</surname><given-names>GI</given-names></name>, <name><surname>Bishop</surname><given-names>DF</given-names></name>, <name><surname>Forrest</surname><given-names>D</given-names></name>, <name><surname>Astrin</surname><given-names>KH</given-names></name>, <name><surname>Desnick</surname><given-names>RJ</given-names></name> (<year>2000</year>) <article-title>Uroporphyrinogen III synthase. An alternative promoter controls erythroid-specific expression in the murine gene</article-title>. <source>J Biol Chem</source>\\n<volume>275</volume>: <fpage>2295</fpage>&#x02013;<lpage>2304</lpage>.<pub-id pub-id-type=\"pmid\">10644678</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Bloomer1\"><label>43</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Bloomer</surname><given-names>JR</given-names></name> (<year>1981</year>) <article-title>Enzyme defects in the porphyrias and their relevance to the biochemical abnormalities in these disorders</article-title>. <source>J Invest Dermatol</source>\\n<volume>77</volume>: <fpage>102</fpage>&#x02013;<lpage>106</lpage>.<pub-id pub-id-type=\"pmid\">7252240</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Brenner1\"><label>44</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Brenner</surname><given-names>DA</given-names></name>, <name><surname>Bloomer</surname><given-names>JR</given-names></name> (<year>1980</year>) <article-title>The enzymatic defect in variegate prophyria. Studies with human cultured skin fibroblasts</article-title>. <source>N Engl J Med</source>\\n<volume>302</volume>: <fpage>765</fpage>&#x02013;<lpage>769</lpage>.<pub-id pub-id-type=\"pmid\">7354807</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Sadlon1\"><label>45</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Sadlon</surname><given-names>TJ</given-names></name>, <name><surname>Dell&#x02019;Oso</surname><given-names>T</given-names></name>, <name><surname>Surinya</surname><given-names>KH</given-names></name>, <name><surname>May</surname><given-names>BK</given-names></name> (<year>1999</year>) <article-title>Regulation of erythroid 5-aminolevulinate synthase expression during erythropoiesis</article-title>. <source>Int J Biochem Cell Biol</source>\\n<volume>31</volume>: <fpage>1153</fpage>&#x02013;<lpage>1167</lpage>.<pub-id pub-id-type=\"pmid\">10582344</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Schoenhaut1\"><label>46</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Schoenhaut</surname><given-names>DS</given-names></name>, <name><surname>Curtis</surname><given-names>PJ</given-names></name> (<year>1989</year>) <article-title>Structure of a mouse erythroid 5-aminolevulinate synthase gene and mapping of erythroid-specific DNAse I hypersensitive sites</article-title>. <source>Nucleic Acids Res</source>\\n<volume>17</volume>: <fpage>7013</fpage>&#x02013;<lpage>7028</lpage>.<pub-id pub-id-type=\"pmid\">2780318</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Surinya1\"><label>47</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Surinya</surname><given-names>KH</given-names></name>, <name><surname>Cox</surname><given-names>TC</given-names></name>, <name><surname>May</surname><given-names>BK</given-names></name> (<year>1998</year>) <article-title>Identification and characterization of a conserved erythroid-specific enhancer located in intron 8 of the human 5-aminolevulinate synthase 2 gene</article-title>. <source>J Biol Chem</source>\\n<volume>273</volume>: <fpage>16798</fpage>&#x02013;<lpage>16809</lpage>.<pub-id pub-id-type=\"pmid\">9642238</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Taketani1\"><label>48</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Taketani</surname><given-names>S</given-names></name>, <name><surname>Mohri</surname><given-names>T</given-names></name>, <name><surname>Hioki</surname><given-names>K</given-names></name>, <name><surname>Tokunaga</surname><given-names>R</given-names></name>, <name><surname>Kohno</surname><given-names>H</given-names></name> (<year>1999</year>) <article-title>Structure and transcriptional regulation of the mouse ferrochelatase gene</article-title>. <source>Gene</source>\\n<volume>227</volume>: <fpage>117</fpage>&#x02013;<lpage>124</lpage>.<pub-id pub-id-type=\"pmid\">10023040</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Campbell1\"><label>49</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Campbell</surname><given-names>PK</given-names></name>, <name><surname>Zong</surname><given-names>Y</given-names></name>, <name><surname>Yang</surname><given-names>S</given-names></name>, <name><surname>Zhou</surname><given-names>S</given-names></name>, <name><surname>Rubnitz</surname><given-names>JE</given-names></name>, <etal>et al</etal> (<year>2011</year>) <article-title>Identification of a novel, tissue-specific ABCG2 promoter expressed in pediatric acute megakaryoblastic leukemia</article-title>. <source>Leuk Res</source>\\n<volume>35</volume>: <fpage>1321</fpage>&#x02013;<lpage>1329</lpage>.<pub-id pub-id-type=\"pmid\">21640380</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Nakanishi1\"><label>50</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Nakanishi</surname><given-names>T</given-names></name>, <name><surname>Bailey-Dell</surname><given-names>KJ</given-names></name>, <name><surname>Hassel</surname><given-names>BA</given-names></name>, <name><surname>Shiozawa</surname><given-names>K</given-names></name>, <name><surname>Sullivan</surname><given-names>DM</given-names></name>, <etal>et al</etal> (<year>2006</year>) <article-title>Novel 5&#x02032; untranslated region variants of BCRP mRNA are differentially expressed in drug-selected cancer cells and in normal human tissues: Implications for drug resistance, tissue-specific expression, and alternative promoter usage</article-title>. <source>Cancer Res</source>\\n<volume>66</volume>: <fpage>5007</fpage>&#x02013;<lpage>5011</lpage>.<pub-id pub-id-type=\"pmid\">16707421</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Birkenmeier1\"><label>51</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Birkenmeier</surname><given-names>CS</given-names></name>, <name><surname>White</surname><given-names>RA</given-names></name>, <name><surname>Peters</surname><given-names>LL</given-names></name>, <name><surname>Hall</surname><given-names>EJ</given-names></name>, <name><surname>Lux</surname><given-names>SE</given-names></name>, <etal>et al</etal> (<year>1993</year>) <article-title>Complex patterns of sequence variation and multiple 5&#x02032; and 3&#x02032; ends are found among transcripts of the erythroid ankyrin gene</article-title>. <source>J Biol Chem</source>\\n<volume>268</volume>: <fpage>9533</fpage>&#x02013;<lpage>9540</lpage>.<pub-id pub-id-type=\"pmid\">8486643</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Tan1\"><label>52</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Tan</surname><given-names>JS</given-names></name>, <name><surname>Mohandas</surname><given-names>N</given-names></name>, <name><surname>Conboy</surname><given-names>JG</given-names></name> (<year>2006</year>) <article-title>High frequency of alternative first exons in erythroid genes suggests a critical role in regulating gene function</article-title>. <source>Blood</source>\\n<volume>107</volume>: <fpage>2557</fpage>&#x02013;<lpage>2561</lpage>.<pub-id pub-id-type=\"pmid\">16293607</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Landry1\"><label>53</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Landry</surname><given-names>JR</given-names></name>, <name><surname>Mager</surname><given-names>DL</given-names></name>, <name><surname>Wilhelm</surname><given-names>BT</given-names></name> (<year>2003</year>) <article-title>Complex controls: the role of alternative promoters in mammalian genomes</article-title>. <source>Trends Genet</source>\\n<volume>19</volume>: <fpage>640</fpage>&#x02013;<lpage>648</lpage>.<pub-id pub-id-type=\"pmid\">14585616</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Cox1\"><label>54</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Cox</surname><given-names>TC</given-names></name>, <name><surname>Bawden</surname><given-names>MJ</given-names></name>, <name><surname>Martin</surname><given-names>A</given-names></name>, <name><surname>May</surname><given-names>BK</given-names></name> (<year>1991</year>) <article-title>Human erythroid 5-aminolevulinate synthase: promoter analysis and identification of an iron-responsive element in the mRNA</article-title>. <source>EMBO J</source>\\n<volume>10</volume>: <fpage>1891</fpage>&#x02013;<lpage>1902</lpage>.<pub-id pub-id-type=\"pmid\">2050125</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Piccinelli1\"><label>55</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Piccinelli</surname><given-names>P</given-names></name>, <name><surname>Samuelsson</surname><given-names>T</given-names></name> (<year>2007</year>) <article-title>Evolution of the iron-responsive element</article-title>. <source>RNA</source>\\n<volume>13</volume>: <fpage>952</fpage>&#x02013;<lpage>966</lpage>.<pub-id pub-id-type=\"pmid\">17513696</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Ma1\"><label>56</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Ma</surname><given-names>J</given-names></name>, <name><surname>Haldar</surname><given-names>S</given-names></name>, <name><surname>Khan</surname><given-names>MA</given-names></name>, <name><surname>Sharma</surname><given-names>SD</given-names></name>, <name><surname>Merrick</surname><given-names>WC</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>Fe2+ binds iron responsive element-RNA, selectively changing protein-binding affinities and regulating mRNA repression and activation</article-title>. <source>Proc Natl Acad Sci U S A</source>\\n<volume>109</volume>: <fpage>8417</fpage>&#x02013;<lpage>8422</lpage>.<pub-id pub-id-type=\"pmid\">22586079</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Graham1\"><label>57</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Graham</surname><given-names>PL</given-names></name>, <name><surname>Yanowitz</surname><given-names>JL</given-names></name>, <name><surname>Penn</surname><given-names>JK</given-names></name>, <name><surname>Deshpande</surname><given-names>G</given-names></name>, <name><surname>Schedl</surname><given-names>P</given-names></name> (<year>2011</year>) <article-title>The translation initiation factor eIF4E regulates the sex-specific expression of the master switch gene Sxl in Drosophila melanogaster</article-title>. <source>PLoS Genet</source>\\n<volume>7</volume>: <fpage>e1002185</fpage>.<pub-id pub-id-type=\"pmid\">21829374</pub-id></mixed-citation></ref><ref id=\"pone.0086718-delaGrange1\"><label>58</label><mixed-citation publication-type=\"journal\">\\n<name><surname>de la Grange</surname><given-names>P</given-names></name>, <name><surname>Dutertre</surname><given-names>M</given-names></name>, <name><surname>Martin</surname><given-names>N</given-names></name>, <name><surname>Auboeuf</surname><given-names>D</given-names></name> (<year>2005</year>) <article-title>FAST DB: a website resource for the study of the expression regulation of human gene products</article-title>. <source>Nucleic Acids Res</source>\\n<volume>33</volume>: <fpage>4276</fpage>&#x02013;<lpage>4284</lpage>.<pub-id pub-id-type=\"pmid\">16052034</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Ishikawa1\"><label>59</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Ishikawa</surname><given-names>H</given-names></name>, <name><surname>Kato</surname><given-names>M</given-names></name>, <name><surname>Hori</surname><given-names>H</given-names></name>, <name><surname>Ishimori</surname><given-names>K</given-names></name>, <name><surname>Kirisako</surname><given-names>T</given-names></name>, <etal>et al</etal> (<year>2005</year>) <article-title>Involvement of heme regulatory motif in heme-mediated ubiquitination and degradation of IRP2</article-title>. <source>Mol Cell</source>\\n<volume>19</volume>: <fpage>171</fpage>&#x02013;<lpage>181</lpage>.<pub-id pub-id-type=\"pmid\">16039587</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yang2\"><label>60</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yang</surname><given-names>J</given-names></name>, <name><surname>Kim</surname><given-names>KD</given-names></name>, <name><surname>Lucas</surname><given-names>A</given-names></name>, <name><surname>Drahos</surname><given-names>KE</given-names></name>, <name><surname>Santos</surname><given-names>CS</given-names></name>, <etal>et al</etal> (<year>2008</year>) <article-title>A novel heme-regulatory motif mediates heme-dependent degradation of the circadian factor period 2</article-title>. <source>Mol Cell Biol</source>\\n<volume>28</volume>: <fpage>4697</fpage>&#x02013;<lpage>4711</lpage>.<pub-id pub-id-type=\"pmid\">18505821</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Zheng1\"><label>61</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Zheng</surname><given-names>BH</given-names></name>, <name><surname>Albrecht</surname><given-names>U</given-names></name>, <name><surname>Kaasik</surname><given-names>K</given-names></name>, <name><surname>Sage</surname><given-names>M</given-names></name>, <name><surname>Lu</surname><given-names>WQ</given-names></name>, <etal>et al</etal> (<year>2001</year>) <article-title>Nonredundant roles of the mPer1 and mPer2 genes in the mammalian circadian clock</article-title>. <source>Cell</source>\\n<volume>105</volume>: <fpage>683</fpage>&#x02013;<lpage>694</lpage>.<pub-id pub-id-type=\"pmid\">11389837</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Carroll1\"><label>62</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Carroll</surname><given-names>SB</given-names></name> (<year>2005</year>) <article-title>Evolution at two levels: on genes and form</article-title>. <source>PLoS Biol</source>\\n<volume>3</volume>: <fpage>e245</fpage>.<pub-id pub-id-type=\"pmid\">16000021</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Hoekstra1\"><label>63</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Hoekstra</surname><given-names>HE</given-names></name>, <name><surname>Coyne</surname><given-names>JA</given-names></name> (<year>2007</year>) <article-title>The locus of evolution: Evo devo and the genetics of adaptation</article-title>. <source>Evolution</source>\\n<volume>61</volume>: <fpage>995</fpage>&#x02013;<lpage>1016</lpage>.<pub-id pub-id-type=\"pmid\">17492956</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Kent1\"><label>64</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Kent</surname><given-names>WJ</given-names></name>, <name><surname>Sugnet</surname><given-names>CW</given-names></name>, <name><surname>Furey</surname><given-names>TS</given-names></name>, <name><surname>Roskin</surname><given-names>KM</given-names></name>, <name><surname>Pringle</surname><given-names>TH</given-names></name>, <etal>et al</etal> (<year>2002</year>) <article-title>The human genome browser at UCSC</article-title>. <source>Genome Res</source>\\n<volume>12</volume>: <fpage>996</fpage>&#x02013;<lpage>1006</lpage>.<pub-id pub-id-type=\"pmid\">12045153</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Dreszer1\"><label>65</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Dreszer</surname><given-names>TR</given-names></name>, <name><surname>Karolchik</surname><given-names>D</given-names></name>, <name><surname>Zweig</surname><given-names>AS</given-names></name>, <name><surname>Hinrichs</surname><given-names>AS</given-names></name>, <name><surname>Raney</surname><given-names>BJ</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>The UCSC Genome Browser database: extensions and updates 2011</article-title>. <source>Nucleic Acids Res</source>\\n<volume>40</volume>: <fpage>D918</fpage>&#x02013;<lpage>D923</lpage>.<pub-id pub-id-type=\"pmid\">22086951</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Flicek1\"><label>66</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Flicek</surname><given-names>P</given-names></name>, <name><surname>Amode</surname><given-names>MR</given-names></name>, <name><surname>Barrell</surname><given-names>D</given-names></name>, <name><surname>Beal</surname><given-names>K</given-names></name>, <name><surname>Brent</surname><given-names>S</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>Ensembl 2012</article-title>. <source>Nucleic Acids Res</source>\\n<volume>40</volume>: <fpage>D84</fpage>&#x02013;<lpage>D90</lpage>.<pub-id pub-id-type=\"pmid\">22086963</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Shinzato1\"><label>67</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Shinzato</surname><given-names>C</given-names></name>, <name><surname>Shoguchi</surname><given-names>E</given-names></name>, <name><surname>Kawashima</surname><given-names>T</given-names></name>, <name><surname>Hamada</surname><given-names>M</given-names></name>, <name><surname>Hisata</surname><given-names>K</given-names></name>, <etal>et al</etal> (<year>2011</year>) <article-title>Using the Acropora digitifera genome to understand coral responses to environmental change</article-title>. <source>Nature</source>\\n<volume>476</volume>: <fpage>320</fpage>&#x02013;<lpage>U382</lpage>.<pub-id pub-id-type=\"pmid\">21785439</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Putnam1\"><label>68</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Putnam</surname><given-names>NH</given-names></name>, <name><surname>Srivastava</surname><given-names>M</given-names></name>, <name><surname>Hellsten</surname><given-names>U</given-names></name>, <name><surname>Dirks</surname><given-names>B</given-names></name>, <name><surname>Chapman</surname><given-names>J</given-names></name>, <etal>et al</etal> (<year>2007</year>) <article-title>Sea anemone genome reveals ancestral eumetazoan gene repertoire and genomic organization</article-title>. <source>Science</source>\\n<volume>317</volume>: <fpage>86</fpage>&#x02013;<lpage>94</lpage>.<pub-id pub-id-type=\"pmid\">17615350</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Chapman1\"><label>69</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Chapman</surname><given-names>JA</given-names></name>, <name><surname>Kirkness</surname><given-names>EF</given-names></name>, <name><surname>Simakov</surname><given-names>O</given-names></name>, <name><surname>Hampson</surname><given-names>SE</given-names></name>, <name><surname>Mitros</surname><given-names>T</given-names></name>, <etal>et al</etal> (<year>2010</year>) <article-title>The dynamic genome of Hydra</article-title>. <source>Nature</source>\\n<volume>464</volume>: <fpage>592</fpage>&#x02013;<lpage>596</lpage>.<pub-id pub-id-type=\"pmid\">20228792</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Tamura1\"><label>70</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Tamura</surname><given-names>K</given-names></name>, <name><surname>Peterson</surname><given-names>D</given-names></name>, <name><surname>Peterson</surname><given-names>N</given-names></name>, <name><surname>Stecher</surname><given-names>G</given-names></name>, <name><surname>Nei</surname><given-names>M</given-names></name>, <etal>et al</etal> (<year>2011</year>) <article-title>MEGA5: molecular evolutionary genetics analysis using maximum likelihood, evolutionary distance, and maximum parsimony methods</article-title>. <source>Mol Biol Evol</source>\\n<volume>28</volume>: <fpage>2731</fpage>&#x02013;<lpage>2739</lpage>.<pub-id pub-id-type=\"pmid\">21546353</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Felsenstein1\"><label>71</label><mixed-citation publication-type=\"other\">Felsenstein J (2005) PHYLIP (Phylogeny Inference Package) version 3.6. Distributed by the author Department of Genome Sciences, University of Washington, Seattle.</mixed-citation></ref><ref id=\"pone.0086718-Guindon1\"><label>72</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Guindon</surname><given-names>S</given-names></name>, <name><surname>Dufayard</surname><given-names>JF</given-names></name>, <name><surname>Lefort</surname><given-names>V</given-names></name>, <name><surname>Anisimova</surname><given-names>M</given-names></name>, <name><surname>Hordijk</surname><given-names>W</given-names></name>, <etal>et al</etal> (<year>2010</year>) <article-title>New Algorithms and Methods to Estimate Maximum-Likelihood Phylogenies: Assessing the Performance of PhyML 3.0</article-title>. <source>Syst Biol</source>\\n<volume>59</volume>: <fpage>307</fpage>&#x02013;<lpage>321</lpage>.<pub-id pub-id-type=\"pmid\">20525638</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Schmidt1\"><label>73</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Schmidt</surname><given-names>HA</given-names></name>, <name><surname>Strimmer</surname><given-names>K</given-names></name>, <name><surname>Vingron</surname><given-names>M</given-names></name>, <name><surname>von Haeseler</surname><given-names>A</given-names></name> (<year>2002</year>) <article-title>TREE-PUZZLE: maximum likelihood phylogenetic analysis using quartets and parallel computing</article-title>. <source>Bioinformatics</source>\\n<volume>18</volume>: <fpage>502</fpage>&#x02013;<lpage>504</lpage>.<pub-id pub-id-type=\"pmid\">11934758</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yang3\"><label>74</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yang</surname><given-names>Z</given-names></name> (<year>1997</year>) <article-title>PAML: a program package for phylogenetic analysis by maximum likelihood</article-title>. <source>Comput Appl Biosci</source>\\n<volume>13</volume>: <fpage>555</fpage>&#x02013;<lpage>556</lpage>.<pub-id pub-id-type=\"pmid\">9367129</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yang4\"><label>75</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yang</surname><given-names>Z</given-names></name> (<year>2007</year>) <article-title>PAML 4: phylogenetic analysis by maximum likelihood</article-title>. <source>Mol Biol Evol</source>\\n<volume>24</volume>: <fpage>1586</fpage>&#x02013;<lpage>1591</lpage>.<pub-id pub-id-type=\"pmid\">17483113</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Zhang2\"><label>76</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Zhang</surname><given-names>J</given-names></name>, <name><surname>Nielsen</surname><given-names>R</given-names></name>, <name><surname>Yang</surname><given-names>Z</given-names></name> (<year>2005</year>) <article-title>Evaluation of an improved branch-site likelihood method for detecting positive selection at the molecular level</article-title>. <source>Mol Biol Evol</source>\\n<volume>22</volume>: <fpage>2472</fpage>&#x02013;<lpage>2479</lpage>.<pub-id pub-id-type=\"pmid\">16107592</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yang5\"><label>77</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yang</surname><given-names>Z</given-names></name>, <name><surname>Wong</surname><given-names>WS</given-names></name>, <name><surname>Nielsen</surname><given-names>R</given-names></name> (<year>2005</year>) <article-title>Bayes empirical bayes inference of amino acid sites under positive selection</article-title>. <source>Mol Biol Evol</source>\\n<volume>22</volume>: <fpage>1107</fpage>&#x02013;<lpage>1118</lpage>.<pub-id pub-id-type=\"pmid\">15689528</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Rosenbloom1\"><label>78</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Rosenbloom</surname><given-names>KR</given-names></name>, <name><surname>Dreszer</surname><given-names>TR</given-names></name>, <name><surname>Long</surname><given-names>JC</given-names></name>, <name><surname>Malladi</surname><given-names>VS</given-names></name>, <name><surname>Sloan</surname><given-names>CA</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>ENCODE whole-genome data in the UCSC Genome Browser: update 2012</article-title>. <source>Nucleic Acids Res</source>\\n<volume>40</volume>: <fpage>D912</fpage>&#x02013;<lpage>D917</lpage>.<pub-id pub-id-type=\"pmid\">22075998</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Karolchik1\"><label>79</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Karolchik</surname><given-names>D</given-names></name>, <name><surname>Hinrichs</surname><given-names>AS</given-names></name>, <name><surname>Furey</surname><given-names>TS</given-names></name>, <name><surname>Roskin</surname><given-names>KM</given-names></name>, <name><surname>Sugnet</surname><given-names>CW</given-names></name>, <etal>et al</etal> (<year>2004</year>) <article-title>The UCSC Table Browser data retrieval tool</article-title>. <source>Nucleic Acids Res</source>\\n<volume>32</volume>: <fpage>D493</fpage>&#x02013;<lpage>496</lpage>.<pub-id pub-id-type=\"pmid\">14681465</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Goecks1\"><label>80</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Goecks</surname><given-names>J</given-names></name>, <name><surname>Nekrutenko</surname><given-names>A</given-names></name>, <name><surname>Taylor</surname><given-names>J</given-names></name>, <name><surname>Galaxy</surname><given-names>T</given-names></name> (<year>2010</year>) <article-title>Galaxy: a comprehensive approach for supporting accessible, reproducible, and transparent computational research in the life sciences</article-title>. <source>Genome Biol</source>\\n<volume>11</volume>: <fpage>R86</fpage>.<pub-id pub-id-type=\"pmid\">20738864</pub-id></mixed-citation></ref><ref id=\"pone.0086718-ENCODEProjectConsortium1\"><label>81</label><mixed-citation publication-type=\"journal\">\\n<collab>ENCODE-Project-Consortium</collab> (<year>2011</year>) <article-title>A user&#x02019;s guide to the encyclopedia of DNA elements (ENCODE)</article-title>. <source>PLoS Biol</source>\\n<volume>9</volume>: <fpage>e1001046</fpage>.<pub-id pub-id-type=\"pmid\">21526222</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Campillos1\"><label>82</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Campillos</surname><given-names>M</given-names></name>, <name><surname>Cases</surname><given-names>I</given-names></name>, <name><surname>Hentze</surname><given-names>MW</given-names></name>, <name><surname>Sanchez</surname><given-names>M</given-names></name> (<year>2010</year>) <article-title>SIREs: searching for iron-responsive elements</article-title>. <source>Nucleic Acids Res</source>\\n<volume>38</volume>: <fpage>W360</fpage>&#x02013;<lpage>W367</lpage>.<pub-id pub-id-type=\"pmid\">20460462</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Jiang1\"><label>83</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Jiang</surname><given-names>J</given-names></name>, <name><surname>Westberg</surname><given-names>JA</given-names></name>, <name><surname>Andersson</surname><given-names>LC</given-names></name> (<year>2012</year>) <article-title>Stanniocalcin 2, forms a complex with heme oxygenase 1, binds hemin and is a heat shock protein</article-title>. <source>Biochem Biophys Res Commun</source>\\n<volume>421</volume>: <fpage>274</fpage>&#x02013;<lpage>279</lpage>.<pub-id pub-id-type=\"pmid\">22503972</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Igarashi1\"><label>84</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Igarashi</surname><given-names>J</given-names></name>, <name><surname>Murase</surname><given-names>M</given-names></name>, <name><surname>Iizuka</surname><given-names>A</given-names></name>, <name><surname>Pichierri</surname><given-names>F</given-names></name>, <name><surname>Martinkova</surname><given-names>M</given-names></name>, <etal>et al</etal> (<year>2008</year>) <article-title>Elucidation of the heme binding site of heme-regulated eukaryotic initiation factor 2alpha kinase and the role of the regulatory motif in heme sensing by spectroscopic and catalytic studies of mutant proteins</article-title>. <source>J Biol Chem</source>\\n<volume>283</volume>: <fpage>18782</fpage>&#x02013;<lpage>18791</lpage>.<pub-id pub-id-type=\"pmid\">18450746</pub-id></mixed-citation></ref></ref-list></back></article>',\n",
       " 'extracted_text': '\\n\\n\\nIntroduction\\n\\nMolecular evolution has recently been a popular area of investigation, and through the advancement of technology and the maturation of analysis methods, this field continues to spawn important insights into the evolutionary processes affecting genes. None of these genes or their encoded proteins exists in isolation, and the products of genes construct the metabolic pathways and networks underlying the cellular and metabolic processes of organisms. As an increasing number of studies are describing the rates of protein and pathway evolution over evolutionary time, there are more opportunities to clarify the patterns and principles of natural selection acting on the pathways involved in the metabolic networks of organisms.\\n\\nResearch focusing on the effects of the organization of pathways on the strength of selection acting on individual proteins in these pathways has revealed various evolutionary patterns among proteins at different positions in a pathway. Important research has been conducted with respect to the plant anthocyanin biosynthetic pathway [1]–[3], terpenoid biosynthesis pathway [4], starch biosynthesis pathway [5], [6], gibberellin biosynthesis pathway [7] and carotenoid biosynthesis pathway [8] as well as lateral line innovation in teleosts [9], the glucosinolate pathway in Arabidopsis thaliana[10] and the primate N-glycosylation pathway [11]. It has been demonstrated that the pleiotropic genes in the upstream portions of pathways or those found at branch points in a network are subject to stronger selective constraints [12]. On the other hand, selection is relaxed in the downstream enzymes, and nonsynonymous substitution rates as well as dN/dS ratios are higher in these pathway components.\\n\\nThe heme biosynthesis pathway is an appropriate system not only for comparing the evolutionary rates of genes according to their position or pathway reticulation but also for studying functional motifs that may play a role at several levels of gene regulation. Heme acts as an essential cofactor for cytochromes, oxidases, peroxidases, catalases, hemoglobin and myoglobin in organisms. Heme acts as an iron-chelating tetrapyrrole and is composed of a complex macrocycle containing four pyrrolic rings connected by methine bridges in cyclic form. Heme also plays multiple regulatory roles, including microRNA processing, ion channel functions, circadian rhythms, mitochondrial targeting, translational regulation and protein degradation [13]–[16]. The heme biosynthesis pathway is an especially well-characterized and important pathway for erythroid production in animals, and malfunctions in heme biosynthesis result in several types of porphyrias because of the accumulation of toxic tetrapyrrole intermediates [17]. The heme biosynthesis pathway of animals is comprised of eight consecutive genes: 5-aminolevulinic acid synthase (ALAS), porphobilinogen synthase (PBGS), porphobilinogen deaminase (PBGD), uroporphyrinogen III synthase (UROS), uroporphyrinogen III decarboxylase (UROD), coproporphyrinogen III oxidase (CPO), protoporphyrinogen IX oxidase (PPO) and ferrochelatase (FECH), from positions one to eight (Figure 1). In animals, the formation of heme occurs in the mitochondria and cytosol. Heme biosynthesis begins with the formation of 5-aminolevulinic acid. In the Shemin pathway of animals, ALAS catalyzes a single-step reaction to condense glycine and succinyl-CoA into ALA, with the elimination of CO2. In vertebrates, the housekeeping isoforms of ALAS1 are expressed in all cell types, and ALAS2 is expressed in erythroid cells (eALAS) at a very high level. Subsequently, PBGS converts two 5-aminolevulinic acids into porphobilinogen, and the addition of four PBG substrates to a dipyrrole by PBGD results in the production of pre-uroporphyrinogen, also known as hydroxymethylbilane. A circularization reaction performed by UROS generates uroporphyrinogen III. The modification of side chains on the cyclic intermediates sequentially mediated by UROD, CPO and PPO results in protoporphyrin IX, which is subsequently chelated with ferrous iron by FECH (see reviews in [14], [18]).\\n\\n10.1371/journal.pone.0086718.g001Figure 1 Heme biosynthesis pathway in animals.The substrate and product are indicated for each enzyme and the subcellular localization of each enzyme is also shown (cytosol or mitochondria). Each enzyme is coded from one to eight according to the linear order of the pathway. Also shown are the processes by which hydroxymethylbilane, the substrate of UROS, can be non-enzymatically cyclized to form uroporphyrinogen I, leading to uroporphyrin I or coproporphyrin I, and the process by which uroporphyrinogen III, the product of UROS, can be auto-oxidized to form uroporphyrin III. Protoporphyrinogen, the substrate of PPO, can be auto-oxidized to form protoporphyrin.\\n\\nALAS1 and ALAS2 are derived from gene duplication. A phylogenetic analysis of ALAS suggested that the relevant gene duplication event took place before the divergence of hagfish from the deuterostome line leading to vertebrates [19]. In extant vertebrate species, ALAS1 and ALAS2 are paralogs, and their amino acid sequences are highly similar to each other.\\n\\nRegulation of the genes encoding the eight enzymes of the heme biosynthesis pathway can occur at the transcriptional, translational and post-translational levels. At the transcriptional level, multiple erythroid-specific factors have been observed to be involved in the transcriptional activation of several genes that participate in heme biosynthesis and erythropoiesis. Among these transcription factors, genomic DNA-binding activity and the conserved binding sites of KLF1, GATA1 and TAL1 have been studied in humans and mice [20]. The binding sites of KLF1 are located within the intergenic regions or introns (particularly the first intron) of genes encoding the components of the heme biosynthesis pathway. In a de novo motif study, GATA1 and TAL1 were hypothesized to complex with KLF1 in a small subset of erythroid cis-regulatory modules [21]. The majority of the GATA1-binding sites that mediate the activation of gene expression are close to the transcription start site, within either the first intron or the proximal 5′ flanking region [22]. TAL1-binding sites for the eight genes of the heme biosynthesis pathway are detected in either proximal promoter or intronic regions [23]. A housekeeping promoter utilized in all tissue types exists in ALAS1, PBGS, PBGD, and UROS. However, for ALAS2, PBGS, PBGD and UROS, erythroid-specific promoters drive gene expression. More importantly, the alternative promoters found in PBGS, PBGD and UROS are located in intron 1, and an alternative splicing event is required for the transcript generated from the housekeeping promoters and erythroid-specific promoters [24]. It has also been established that DNase-hypersensitive sites usually serve as sites for conserved and cell-type specific transcription factor binding and histone modification [25]–[27]. Combined analysis of intron 1 sequences and DNase-hypersensitive sites will shed light on the regulatory potential of the corresponding cis-elements and the evolution of the heme biosynthesis pathway.\\n\\nAt the translational level, IRE-binding protein (IRP) interacts with the iron-responsive element (IRE) located in the 5′ untranslated regions (5′UTRs) of ALAS2 and ferritin mRNA to inhibit protein translation and in the 3′ untranslated region (3′UTR) of TfR1 to stabilize mRNA. During iron repletion, iron-sulfur clusters can abolish the IRE-binding ability of IRP1, and the F-box protein FBXL5 recognizes IRP2 and targets it for degradation by E3 ligases. Both mechanisms offer a link between iron availability and heme synthesis [28]–[30]. At the post-translational level, the binding of heme to the heme-regulatory motif (HRM) in the mature ALAS1 peptide blocks mitochondrial import and results in end-product inhibition [19], [31]–[33].\\n\\nThe objectives of this study are to investigate the molecular fingerprints underlying the adaptation of the eight genes encoding the enzymes of the heme biosynthesis pathway in Kingdom Metazoa and, particularly, how these adaptations correlate with the positions of the gene products (enzymes) in the pathway. First, through the analysis of the evolutionary constraints on the protein-coding sequences, we reveal strong purifying selection on the protein sequences and clade-specific adaptations in teleosts and arthropods. We also demonstrate that the first introns of these genes in vertebrates play a role in their erythroid-specific transactivation, highlighting the emergence of erythrocytes in animals. Then, we show that pathway product (heme) feedback control is widely utilized in this pathway. The evolutionarily conserved motifs that enable this control include HRMs found not only in ALAS but also in PBGS. We also discover IREs within 5′UTRs and exon-intron boundaries, raising the possibility of regulation of both translation and splicing choice, respectively. We conclude by summarizing the evolution of multifarious controls on the heme biosynthesis pathway.\\n\\n\\nResults\\n\\n\\nRelationship between Selection Pressure and Pathway Position\\n\\nBased on the M0 model, assuming a constant evolutionary rate (nonsynonymous versus synonymous rate, dN/dS, ω) for all branches and all codons of the eight genes encoding the enzymes of the metazoan heme biosynthesis pathway, the ω values vary from 0.041 (FECH) to 0.127 (UROS), providing evidence that the sequences of the coding regions of these genes are under negative selection (Figure 2A). We also evaluated the M1a model, which assumes that there are two groups of codons, subject to purifying selection and neutral evolution, in contrast to the M0 model. We found that the M1a model significantly improves the model (p<0.001) and that a large fraction of the codons are under purifying selection (>91% in all genes, except for PPO, where the obtained value was 79%). Notably, the ω values at positions four (UROS) and seven (PPO) were highest (0.116 and 0.105, respectively) among the genes of the heme biosynthesis pathway. (Table S1).\\n\\n10.1371/journal.pone.0086718.g002Figure 2 Selection pressure on heme biosynthesis genes in animals.ω values (dN/dS) were estimated with the M0 model for the eight heme biosynthesis genes in animals (A). The distribution of ω values (B), the nonsynonymous substitution rate, dN (C), and the synonymous substitution rate, dS (D). The order of genes follows the linear order of their pathway positions (Figure 1).\\n\\nTo determine whether the variation in ω values among the genes was statistically significant and showed a relationship with their positions in the pathway, we conducted several tests on the dN, dS and ω values obtained for each gene (Figure 2B, 2C, 2D). First, we determined whether the distributions of the dN, dS and ω values were correlated. We found that the distributions of the dN, dS and ω values were not correlated for each gene (Kruskal-Wallis rank sum test, p<0.0001). Second, the ω values of genes located at positions four (UROS) and seven (PPO) were shown to be significantly higher than for the genes located at other positions (Wilcoxon rank sum test, P<0.0001). Third, to determine the significance of the variations in ω values among the genes, we conducted multiple comparisons of the ω values for each gene pair (see Methods). Three groups were established: group 1 (low ω values), consisting of FECH, PBGD and CPO; group 2 (intermediate ω values), consisting of ALAS, UROD and PBGS; and group 3 (high ω values), consisting of PPO and UROS (Table 1). Fourth, we determined whether the dN, dS and ω values were correlated with the pathway positions of the eight enzymes of the heme biosynthesis pathway. It was found that the ω values were positively correlated with the pathway positions of the enzymes (Kendall’s correlation test: tau\\u200a=\\u200a0.2202, P<0.0001), as were the dN values (Kendall’s correlation test: tau\\u200a=\\u200a0.1625, P<0.0001). However, the dS values were not correlated with pathway position (Kendall’s correlation test: tau\\u200a=\\u200a−0.0372, P\\u200a=\\u200a0.0243).\\n\\nTable 1\\nGroup 1\\tGroup 2\\tGroup 3\\nFECH\\tPBGD\\tCPO\\tALAS\\tUROD\\tPBGS\\tPPO\\tUROS\\nFECH\\t**\\t*\\t**\\nGroup 1\\tPBGD\\t*\\t**\\nCPO\\t*\\t**\\nALAS\\t**\\nGroup 2\\tUROD\\t**\\t*\\t**\\nPBGS\\t**\\t*\\t**\\nGroup 3\\tPPO\\t**\\nUROS\\t**\\n\\nIn conclusion, the variations in ω and dN values were shown to be correlated with the pathway positions of the eight enzymes of the heme biosynthesis pathway.\\n\\n\\nAmino Acid Residues under Positive Selection\\n\\nWhen the genes of the heme biosynthesis pathway are observed to be under strong purifying selection, it is interesting to examine whether there are amino acids in certain lineages that experience positive selection. Due to limitations regarding the number of sequences collected, we focused on the protein sequences of the mammal, teleost and arthropod subgroups. Of the eight enzymes involved in heme biosynthesis, only ALAS2 from teleosts, PBGS from arthropods and UROD from teleosts showed positively selected residues in the branch-site model (p value <0.05) (Table 2). In ALAS2 from teleosts, five sites were found to be positively selected (model A/A1, p<2.3e-10) (BEB >0.978) (Table 3). Notably, while the amino acid detected in teleost ALAS2 at position 204 is I/M/T, the amino acid E is present at this position in all other vertebrate ALAS1 proteins. We mapped this amino acid in the crystal structure of the protein and found that it is positioned at the ALAS dimer interface. The amino acid R located at position 353 in teleost ALAS2 is also interesting because a K residue is found at this position in all other vertebrate ALAS1, ALAS2 and ALAS proteins. The crystal structure of ALAS indicates that this amino acid forms a hydrogen bond with the ribose-O3’ of the substrate succinyl-CoA [34] (Figure S1A).\\n\\nTable 2\\nModel\\tEstimates of parameters\\t2ΔL\\tp\\nALAS+ALAS1+ALAS2\\tM0: one ratio\\tω\\u200a=\\u200a0.05356\\nTeleost ALAS2 branch\\ttwo ratios\\tω0\\u200a=\\u200a0.05323, ω\\u200a=\\u200a8.04979\\t10.087074\\t1.49E-03\\nPBGS\\tM0: one ratio\\tω\\u200a=\\u200a0.06398\\nArthropod branch\\ttwo ratios\\tω0\\u200a=\\u200a0.06398, ω\\u200a=\\u200a999.000\\t4.478516\\t3.43E-02\\nUROD\\tM0: one ratio\\tω\\u200a=\\u200a0.06194\\nTeleost branch\\ttwo ratios\\tω0\\u200a=\\u200a0.06135, ω\\u200a=\\u200a13.69790\\t6.01671\\t1.42E-02\\n\\na\\n\\nb\\n\\nTable 3\\nForeground branches\\t2ΔL\\tp\\tEstimates of the parameters in the modified model A\\tPositively selected sites\\nTeleost ALAS2\\t40.20213\\t2.29E-10\\tp0\\u200a=\\u200a0.86464, p1\\u200a=\\u200a0.07816, p2a\\u200a=\\u200a0.05246,\\t204E, 243K, 352P, 353K,\\np2b\\u200a=\\u200a0.00474, ω0\\u200a=\\u200a0.04595, ω2\\u200a=\\u200a999.0\\t431G\\nArthropod PBGS\\t23.97615\\t9.75E-07\\tp0\\u200a=\\u200a0.86020, p1\\u200a=\\u200a0.03447, p2a\\u200a=\\u200a0.10127,\\t11Y, 106H, 162C, 195S,\\np2b\\u200a=\\u200a0.00406, ω0\\u200a=\\u200a0.06383, ω2\\u200a=\\u200a999.0\\t267K, 274A, 314I\\nTeleost UROD\\t25.60371\\t4.19E-07\\tp0\\u200a=\\u200a0.87046, p1\\u200a=\\u200a0.07295, p2a\\u200a=\\u200a0.05222,\\t136Q, 174M, 290K, 297K,\\np2b\\u200a=\\u200a0.00438, ω0\\u200a=\\u200a0.06761, ω2\\u200a=\\u200a999.0\\t300T, 316E, 349H\\n\\na\\n\\nb\\n\\nc\\n\\nd\\n\\nIn PBGS from arthropods, seven sites are positively selected (model A/A1, p<9.8e-7) (BEB >0.964) (Table 3). The amino acid found in arthropod PBGS at position 11 is I/M, whereas F/Y is present at this position in all vertebrate and cnidarian PBGS proteins. We mapped this amino acid in the crystal structure and found that it is positioned at the PBGS dimer interface and is very close to the Mg2+ binding site, which has been hypothesized to trigger the conversion of PBGS between its open and closed forms [35]. The amino acid A is present at position 195 in Drosophila PBGS, whereas the amino acid at this position is S in all other animal PBGS proteins. The crystal structure indicates that this amino acid forms the active site pocket [35] (Figure S1B).\\n\\nIn UROD from teleosts, seven sites were observed to be positively selected (model A/A1, p<4.2e-7) (BEB >0.981) (Table 3). The amino acid S is present at position 290 in teleost UROD, whereas D/E/K is found at this position in other vertebrate, cnidarian and arthropod UROD proteins. The amino acid at position 316 in the teleost UROD is H/R/S, whereas the amino acid at this position is D/E/K in other vertebrate, chordate, arthropod and cnidarian UROD proteins. We mapped both amino acid residues in the crystal structure and found that they are positioned at the UROD dimer interface [36] (Figure S1C).\\n\\nIn summary, we discovered several amino acid residues under positive selection. Some of them are located at the active sites and dimer interfaces of biologically functional enzymes of the heme biosynthesis pathway.\\n\\n\\nDetection of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences\\n\\nIt was previously reported that intron 1 sequences contain cis-elements that are necessary for the transcriptional activation of human PBGS[37], PBGD[38]–[40] and UROS[41], [42]. Therefore, we investigated the selection acting on intron sequences by searching for DNA sequences that are both evolutionarily conserved across 46 vertebrates and located in DNase-hypersensitive site clusters (ENCyclopedia of DNA Elements, ENCODE, for human). We found considerable stretches of DNA longer than 40 nucleotides in intron 1 of ALAS2, PBGS, PBGD, UROS, UROD and FECH that could serve as evolutionarily conserved DNase-hypersensitive sites in host genes (Figure 3, Table S2). However, evolutionarily conserved DNase-hypersensitive sites were also found in other intron regions, including in intron 3 of ALAS1, intron 8 of ALAS2, and introns 2 and 6 of FECH. We did not find considerable lengths of evolutionarily conserved DNase-hypersensitive sites in the intron regions of CPO and PPO. In summary, we revealed evolutionarily conserved DNase-hypersensitive sites in intron regions in six of the eight genes of the heme biosynthesis pathway, implying that conserved regulatory mechanisms acting on intron sequences might be involved in the transactivation of gene expression in the vertebrate heme biosynthesis pathway.\\n\\n10.1371/journal.pone.0086718.g003Figure 3 Three-dimensional view of the evolutionarily conserved DNase-hypersensitive sites in intron sequences.For each gene in the biosynthesis pathway (ALAS1 and ALAS2 are treated separately because they are different genes located on different chromosomes), the length of the intersection of the DNA sequence that is evolutionarily conserved across vertebrates and DNase-hypersensitive sites is indicated on the z-axis. The intron ID is provided on the x-axis. Genes from ALAS1 to FECH are shown on the y-axis and are coded from one to eight according to the linear order of the pathway. (Figure 1).\\n\\n\\nDistribution of IREs in Exon and Intron Sequences\\n\\nIREs have been reported to exist in the 5′ and 3′UTR sequences of mRNAs, through which they control the translational efficiency and stability of transcripts, respectively. The 5′UTR of human ALAS2 has been shown to contain an IRE. We conducted a survey of the 5′UTR sequences of genes of the heme biosynthesis pathway (Tables 4 and 5 and Table S3). Notably, the 5′UTRs of nine of the thirteen collected vertebrate ALAS2 sequences contained IREs of high quality (see Methods). Intriguingly, we also detected IREs in the 5′UTRs of teleost ALAS1 sequences and the ALAS sequences of one chordate (tunicate) and the purple sea urchin (echinoderm) that were of high quality as well as in a sea anemone (cnidarian) and honey bee (arthropod) that were of medium quality. These findings constitute the first demonstration that IREs can be identified in the 5′UTRs of ALAS1 sequences from vertebrates and ALAS sequences from arthropods, echinoderms and cnidarians.\\n\\nTable 4\\nGene\\tMammal\\tBird,Reptile,Amphibian\\tTeleost\\tChordate\\tEchinoderm\\tArthropod\\tCnidaria\\nALAS\\tNF\\t1(2)\\t1(1)\\t1(2)\\t1(1)\\nALAS1\\t(5)\\t3(4)\\tNF\\nALAS2\\t5(5)\\t1(1)\\t3(3)\\tNF\\nPBGS\\t(6)\\t(3)\\t(4)\\t(1)\\tNF\\t(0)\\t(1)\\nPBGD\\t(5)\\t(3)\\t(1)\\t1(1)\\t2(2)\\t(1)\\nUROS\\t(5)\\t(4)\\t(3)\\t(0)\\t(1)\\nUROD\\t(5)\\t(1)\\t(3)\\t(1)\\t(2)\\nCPO\\t(4)\\t(3)\\t(4)\\t(1)\\t(0)\\nPPO\\t(5)\\t(1)\\t(4)\\t(0)\\t(1)\\t(2)\\nFECH\\t(5)\\t(0)\\t(3)\\t(0)\\t(1)\\n\\nTable 5\\nGene\\tMammal\\tBird,Reptile,Amphibian\\tTeleost\\tChordate\\tEchinoderm\\tArthropod\\tCnidaria\\nALAS\\tNF\\t2(2)\\t1(1)\\t(4)\\t(2)\\nALAS1\\t1(7)\\t1(5)\\t(6)\\tNF\\nALAS2\\t(5)\\t1(2)\\t(5)\\tNF\\nPBGS\\t4(6)\\t2(4)\\t1(5)\\t(1)\\tNF\\t(1)\\t1(3)\\nPBGD\\t3(7)\\t(5)\\t(6)\\t(1)\\t(4)\\t1(2)\\nUROS\\t5(7)\\t3(5)\\t(4)\\t(1)\\t(2)\\nUROD\\t(7)\\t(5)\\t(2)\\t(1)\\t(4)\\t(3)\\nCPO\\t(7)\\t(5)\\t(1)\\nPPO\\t(7)\\t1(2)\\t(5)\\t(2)\\t(1)\\t1(3)\\nFECH\\t5(7)\\t(4)\\t(1)\\t(5)\\t(1)\\n\\nIREs were also found in the 5′UTRs of the PBGD sequences of Drosophila melanogaster (fruit fly) and Apis mellifera (honey bee) (high quality) as well as that of Strongylocentrotus purpuratus (sea urchin) (medium quality) (Tables 4 and 5 and Table S3). This observation raised the possibility that the translation of genes other than ALAS in the heme biosynthesis pathway could also respond to iron availability.\\n\\nWe also found IREs in intron regions. In particular, several potential IREs (of medium quality) exist at the intron-exon boundaries of PBGS from Mus musculus (mouse), PBGD from Loxodonta africana (elephant) and Oryctolagus cuniculus (rabbit) and UROS from Loxodonta africana (elephant) and Mus musculus (mouse) (Tables 4 and 5 and Table S3). This class of IREs forms stem and loop regions that overlap with protein-coding and intron sequences (Figure 4 and Figure 5). Based on these observations, we conducted a survey to identify potential IREs in the intron-exon boundary sequences of genes in the human and zebrafish genomes. We found 21 and 12 potential IREs of high quality at the intron-exon boundaries of genes in the human and zebrafish genomes, respectively (Table S4).\\n\\n10.1371/journal.pone.0086718.g004Figure 4 Potential iron-responsive elements (IREs) in the introns and intron-exon boundaries of UROS genes.IREs depicted as stem-loop structures are shown in the corresponding intron regions. UROS exon and intron IDs from four species are indicated. The conserved splicing acceptor site AG and the unpaired nucleotide of the IRE structure are also shown.\\n\\n10.1371/journal.pone.0086718.g005Figure 5 Sequence alignment of the IREs at the intron-exon boundaries of UROS from four species.“>” and “<” represent the base pairing of the RNA secondary structure. The potential IRE consensus loop sequence, CAGUGN, and the unpaired nucleotide G are also shown with respect to the location of the IRE hairpin. The intron-exon boundary is indicated as |.\\n\\nIn summary, we detected potential IREs not only in the 5′UTR of ALAS2 but also in the 5′UTRs of ALAS and ALAS1. More intriguingly, several potentially conserved IREs might exist at the intron-exon boundaries of genes involved in the heme biosynthesis pathway.\\n\\n\\nDistribution of HRMs in Protein Sequences\\n\\nThe HRM (denoted HRM_t, see Methods) has been shown to bind heme to inhibit the import of ALAS1 into mitochondria. Here, we collected recently published HRM sequences and compiled them into a new HRM consensus sequence (denoted HRM_r, see Methods). Some of the HRM_r sequences are predicted to sense the redox state of the cell and may be critical for triggering the degradation of proteins containing the HRM sequence.\\n\\nOnly ALAS1, ALAS2 and PBGD protein sequences contain HRM_t. We found HRM_t sequences in the ALAS sequences of chordates, echinoderms, and cnidarians. The ALAS1 genes of all vertebrate species show HRM_t sequences at the N-terminus. In ALAS2, HRM_t sequences were found in several types of mammals and in chickens and Xenopus (Tables 6 and 7 and Table S5).\\n\\nTable 6\\nGene\\tMammal\\tBird,Reptile,Amphibian\\tTeleost\\tChordate\\tEchinoderm\\tArthropod\\tCnidaria\\nALAS\\tNF\\t3(3)\\t2(2)\\t1(6)\\t1(2)\\nALAS1\\t7(7)\\t6(6)\\tNF\\nALAS2\\t4(5)\\t3(4)\\t(5)\\tNF\\nPBGS\\t(6)\\t(5)\\t(1)\\tNF\\t(6)\\t(3)\\nPBGD\\t(7)\\t(6)\\t(5)\\t(1)\\t3(6)\\t(2)\\nUROS\\t(7)\\t(6)\\t(4)\\t(2)\\t(1)\\t(6)\\t(3)\\nUROD\\t(7)\\t(6)\\t(5)\\t(3)\\t(1)\\t(7)\\t(3)\\nCPO\\t(7)\\t(5)\\t(2)\\t(1)\\t(6)\\t(1)\\nPPO\\t(7)\\t(5)\\t(2)\\t(1)\\t(6)\\t(3)\\nFECH\\t(7)\\t(5)\\t(4)\\t(3)\\t(1)\\t(7)\\t(2)\\n\\nTable 7\\nGene\\tMammal\\tBird,Reptile,Amphibian\\tTeleost\\tChordate\\tEchinoderm\\tArthropod\\tCnidaria\\nALAS\\tNF\\t(3)\\t(2)\\t(6)\\t(2)\\nALAS1\\t(7)\\t2(6)\\t(6)\\tNF\\nALAS2\\t5(5)\\t(4)\\t1(5)\\tNF\\nPBGS\\t6(6)\\t5(5)\\t(1)\\tNF\\t2(6)\\t1(3)\\nPBGD\\t(7)\\t(6)\\t(5)\\t(1)\\t3(6)\\t(2)\\nUROS\\t(7)\\t(6)\\t(4)\\t(2)\\t(1)\\t(6)\\t(3)\\nUROD\\t(7)\\t(6)\\t(5)\\t(3)\\t(1)\\t(7)\\t(3)\\nCPO\\t(7)\\t(5)\\t(2)\\t(1)\\t(6)\\t(1)\\nPPO\\t(7)\\t(5)\\t(2)\\t(1)\\t(6)\\t(3)\\nFECH\\t(7)\\t(5)\\t(4)\\t(3)\\t(1)\\t(7)\\t(2)\\n\\nNotably, when we used HRM_r as the search sequence for HRMs, new HRM sequences were identified in PBGS (vertebrates, arthropods, and cnidarians) and PBGD (Drosophila). This class of HRM is not restricted to the N-terminus of the protein sequence and is conserved with respect to its position in the amino acid sequence (Figure 6).\\n\\n10.1371/journal.pone.0086718.g006Figure 6 Heme-regulatory motifs (HRMs) in PBGS and PBGD.Multiple sequence alignments of PBGS (A) and PBGD (B) are shown, with HRM_t and HRM_r colored orange and green, respectively. Amino acid numbers for HRM_t and HRM_r are also shown according to the first protein sequence in the alignment.\\n\\n\\nDiscussion\\n\\nElucidation of the evolutionary history of biological pathways sheds light on the principles underlying the evolutionary forces acting on organisms in the environment. The history of pathway evolution may vary among different pathways and ancestral organisms, but understanding the underlying principles helps to reveal the modifications that have taken place in the physiological processes of organisms during their evolutionary history.\\n\\nMethods allowing the detection of selection pressure in the evolutionary histories of pathways and networks have made it possible to investigate the existence of fundamental principles driving selection in nature. By estimating the ratio of the nonsynonymous to the synonymous substitution rate for individual protein-encoding genes (ω), the types of selection pressure acting on a gene can be identified. In addition to the properties of proteins themselves, the regulatory mechanisms acting on genes are important in metabolic pathways and are also exposed to selection during evolutionary processes. A novel regulatory mechanism affecting the genes in a metabolic pathway can give rise to a new gene function that may become fixed in a lineage, through which we can determine the evolutionary history of a particular biosynthesis pathway throughout different lineages of organisms. To clarify the evolutionary history of the heme biosynthetic pathway of animals, we analyzed the three previously reported regulatory mechanisms related to genes in the heme biosynthesis pathway, which involve DNase I-hypersensitive sites, IREs, and HRMs. Our in silico prediction results showed that multiple regulatory mechanisms may exist for the genes in the heme biosynthesis pathway.\\n\\n\\nStronger Selection at the Middle and Penultimate Positions of the Pathway could Result from Self-catalysis\\n\\nWe conducted a phylogenetic analysis of the eight genes of the heme biosynthesis pathway found in the animal kingdom. The ω values at positions four (UROS) and seven (PPO) were shown to be significantly higher than at the other positions in the pathway. Notably, some self-catalysis and by-products of the heme biosynthesis pathway have been reported. Hydroxymethylbilane, the substrate of UROS, can be non-enzymatically cyclized to form uroporphyrinogen I, a useless by-product that leads to uroporphyrin I or coproporphyrin I [43]. The product of UROS can also be auto-oxidized to uroporphyrin III. Protoporphyrinogen, the substrate of PPO, has been shown to be auto-oxidized to protoporphyrin in air, without the need for PPO [43], [44]. The degree of such “leaking” among the biochemical reactions involved in the heme biosynthesis pathway is unknown, and whether this phenomenon is common in animals remains to be determined. We speculate that “leaking” in a biochemical reaction would impact selection pressure and, most likely, lead to decreased evolutionary constraint.\\n\\n\\nBiological Function of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences\\n\\nBased on the chromatin immunoprecipitation with high-throughput sequencing (ChIP-seq) technique, the transcription factor-binding module GATA1-KLF1 has been hypothesized to allow erythrocyte-specific activation of gene expression through binding to the intron 1 sequences of ALAS2, PBGS, PBGD and UROS[21], [22]. In this study, we identified intron 1 sequences as a type of DNase-hypersensitive site that is conserved in vertebrates among several genes, including ALAS2, PBGS, PBGD, UROS and UROD. This finding suggests that in response to the high demand for heme in the erythrocytes of the common ancestor of vertebrates, several transcription factors were recruited to bind to the first intron region to trans-activate the expression of the first five genes of the heme biosynthesis pathway. Notably, an intronic enhancer has been identified in intron 8 of ALAS2 in mouse erythroleukemia cells [45]–[47], which is consistent with our analysis (Figure 2). We also identified introns 2 and 6 of FECH as potential intronic enhancers, although two promoter regions of mouse FECH were shown to function in basic and inducible expression [48].\\n\\nThe use of the first intron as an alternative promoter to induce erythrocyte-specific expression has been documented for several hematopoietic genes, including Abcg2[49], [50], Ank1[51] and Slc11a2[21]. We also determined whether evolutionarily conserved intron regions of these genes could be detected. Without exception, the first intron regions of all three genes contain long stretches of DNA that are both conserved across vertebrates and accessible to DNase attack, suggesting that transcription factor binding occurs (Table S6). The high frequency of alternative first exons found in erythroid genes has been shown to be crucial for the regulation of gene function [52], and we propose here that the evolutionarily conserved region in the first intron can serve as either an alternative promoter or enhancer to enable alteration of the first exon during gene transcription [53].\\n\\n\\nElucidation of Novel Evolutionarily Conserved IREs at Intron-exon Boundaries\\n\\nAn IRE was previously identified in the 5′UTR of ALAS2 [54]. However, the detection IREs in the 5′UTRs of PBGD sequences is reported for the first time here. The scattered appearance of IREs in enzymes other than ALAS suggests that animal genes acquired IREs at a later point through the process of convergent evolution [55]. Moreover, whereas the noted IRE found in an echinoderm ALAS (Ciona intestinalis) has been reported previously [55], our identification of a potential IRE in the ALAS of a cnidarian (sea anemone) and an arthropod (honey bee) suggested that a more thorough search for the existence of IREs in animals is necessary.\\n\\nBoth the 5′UTR and 3′UTR are commonly regarded as sites where IREs are located. However, we found that intron sequences can also contain IREs. In particular, we identified an IRE at the intron 4-exon 5 boundary of UROS in Loxodonta africana (elephant). We also aligned the IRE in the corresponding region and found that this intron-exon IRE is conserved in humans, rhesus monkeys and rabbits, although these IREs are of low quality. If IREs found at intron-exon boundaries are functional, it is possible that the pre-mRNA splicing junction could be bound by IRP, thereby influencing splicing efficiency or choice.\\n\\nIn the iron-depleted state, IRP1 and IRP2 are stable and bind to IREs, either to inhibit protein translation (5′UTR) or to prevent mRNA degradation (3′UTR). In the iron-replete state, the degradation of IRP2 allows IREs to bind iron and eIF4E [56] to initiate protein translation. Intriguingly, eIF4E has been demonstrated to function as a co-factor in the Sxl-dependent female-specific alternative splicing of msl-2 as well as Sxl premRNAs in Drosophila, which is required for sex determination due to the silencing of the X chromosome [57]. This observation raised the possibility that eIF4E could be involved in splicing events in mammals. To determine how many IREs exist at intron-exon boundaries, we conducted a genome-wide survey to detect IREs of high quality with respect to the nucleotides present at the junctions of CDS-exons and introns in the human and zebrafish genomes. We found 21 and 12 high-quality IREs at human and zebrafish CDS-exon/intron junctions, respectively (Table S4). One of human genes, ZNF446, has been found in the Friendly Alternative Splicing and Transcripts Database (FAST DB) [58] that contains an alternatively spliced transcript (AY279351), which presumably depends on the corresponding junction containing an IRE. This possible phenomenon adds one more layer of complexity to the regulation of genes in the heme biosynthesis pathway in particular and to the regulation of erythropoiesis in general. Furthermore, this finding also calls for an investigation into the mechanism by which iron and IRP regulate the alternative splicing of protein-coding genes and the associated biological effects.\\n\\n\\nElucidation of Novel Evolutionarily Conserved HRMs\\n\\nBy defining a new HRM (HRM_r), we were able to identify HRMs in most of the studied PBGS genes (76%, 19/25) and all PBGD genes from Drosophila. Recently, HRMs have been implicated in regulatory functions other than controlling import into the mitochondria. The HRM in human IRP2 has been shown to be responsible for the ubiquitin-targeted degradation of the IRP2 protein [59]. The degradation of human circadian Factor Period 2 (hPer2) is also mediated by an HRM, suggesting that metabolic signals can modulate the circadian regulation of gene expression [60]. Notably, mouse ALAS1 and ALAS2 have been shown to be under circadian control and are regulated by mPER1 and mPER2 [61]. It has been demonstrated that the HRM of ALAS2 is not involved in the heme-mediated control of import into the mitochondria [32]. Therefore, we postulate that the HRMs identified in ALAS2 and PBGS could be involved in protein degradation. This is consistent with the localization of the PBGS enzyme in the cytosol and that, unlike ALAS1 and ALAS2, PBGS does not need to enter the mitochondria to function (Figure 1). The HRM binds heme and senses both the concentration of heme and the oxidation/reduction state of the cell. Our findings indicate that product feedback control of protein stability could be involved in the evolution of the heme biosynthesis pathway.\\n\\n\\nEvolutionary Implications of HRMs and IREs in the Teleost Lineage based on Comparison with other Species\\n\\nWe found HRMs in the ALAS sequences of chordates and echinoderms. ALAS1 HRM_t exists in all investigated vertebrate species as well. While we also detected ALAS2 HRM_t in certain monophyletic groups (mammals, birds, and amphibians), we did not find any HRM_t sequences in teleosts. Additionally, we identified IREs in the 5′UTRs of ALAS sequences from chordates and echinoderms. In the 5′UTR of ALAS2, there is an IRE in all of the examined vertebrate species. In the 5′UTR of ALAS1, an IRE is found in teleosts, but not in mammals. Based on our phylogenetic analysis, chordate and echinoderm ALAS and vertebrate ALAS1/2 are derived from the same ancestral ALAS gene. We propose that the HRMs and IREs observed in ALAS1 and ALAS2 existed in the ALAS sequence of a common ancestor of vertebrates, chordates, and echinoderms. The unique absence of HRMs in ALAS2 proteins and the presence of IREs in ALAS1 mRNAs in teleosts are striking. We further propose that during the evolutionary branching and speciation of vertebrates, the loss of the HRM from ALAS2 and the retention of the IRE in ALAS1 in teleosts resulted from the adaptation of the heme biosynthesis pathway to the environment. Notably, we have also shown that the amino acid sequences of ALAS2 and UROD have experienced positive selection in the teleost lineage.\\n\\n\\nConclusion: An Integrated View of the Evolution of Multiple Types of Regulation vs. Pathway Position Reveals Different Depths within the Pathway\\n\\nWe have investigated the evolution of multiple controls modulating the heme biosynthesis pathway. It was illuminating to study the degree by which these combinatorial controls impinge on the pathway from top to bottom, if we regard ALAS, at position one, as the top and FECH, at position eight, as the bottom. In a summary table (Table 8), with the eight genes represented from positions one to eight in rows and the six control mechanisms across the top of the table, it can clearly be observed that there are multiple degrees of regulatory potential with respect to this pathway. Notably, transcriptional control and iron-mediated splicing control infiltrate the middle and bottom of the pathway. Substrate (iron) or product (heme) feedback translational control as well as protein localization and protein stability control primarily infiltrate the top of the pathway. Purifying selection on the protein sequences is widespread but is subject to loose constraint in the middle and near the bottom of the pathway. An intriguing question from the point of view of pathway position is whether the evolvability of pathway regulation depends on the position of the genes in the pathway, thereby leading to differential adaptation. Our research suggests that the investigation of the molecular evolution of a pathway should involve the examination of different control mechanisms related to the activity of the genes that constitute the pathway under investigation. Thus, a new perspective in the field of evolutionary developmental biology could concern the roles played by the cis-regulatory regions and protein-coding regions of genes with respect to adaptive mutations [62], [63]. We are aware that our key findings in this study are predictions based on the current understanding of the heme biosynthesis pathway, and these findings will require substantial experimental work to be confirmed or corrected.\\n\\nTable 8\\nGene\\tPathway Position\\tFirst Intron-mediatedTranscription Control\\tSplicing Control\\tTranslational Control\\tProtein Localization\\tProteinStability\\tSelection Pressure\\nALAS\\t1\\t**\\nALAS1\\t1\\t*\\t***\\t*\\nALAS2\\t1\\t**\\t***\\t**\\nPBGS\\t2\\t**\\t*\\t***\\nPBGD\\t3\\t***\\t*\\nUROS\\t4\\t*\\t***\\nUROD\\t5\\t*\\nCPO\\t6\\nPPO\\t7\\t***\\nFECH\\t8\\t**\\n\\na\\n\\nb\\n\\nc\\n\\nd\\n\\ne\\n\\nf\\n\\n\\nMaterials and Methods\\n\\n\\nSequence Collection\\n\\nFor genes encoding the eight enzymes of the heme biosynthesis pathway in animals, we collected the sequences of organisms whose entire genome sequences were available. Sequences were collected from mammals (seven species), amphibians (two species), birds (three species), reptiles (one species), teleosts (six species), echinoderms (two species), arthropods (eight species) and cnidarians (three species). The amino acid sequences, coding nucleotide sequences, exon sequences and intron sequences were downloaded from NCBI, UCSC [64], [65] and Ensembl [66]. Some of the protein sequences of ALAS genes were extracted as suggested previously [19]. The 5′UTR, exon and intron sequences of the coral Acropora digitifera came from OIST Marine Genome Unit [67]. The 5′UTR, exon and intron sequences of the sea anemone Nematostella vectensis were obtained from the DOE Joint Genome Institute [68]. The 5′UTR, exon and intron sequences of the hydra Hydra magnipapillata also came from the DOE Joint Genome Institute [69].\\n\\nIf a gene model lacked a 5′UTR, we searched for a cDNA collection to determine whether there was any possibility of extension in the 5′ direction with respect to the direction of transcription. If such an extension was possible, a 5′UTR was added to the gene model. Subsequently, if this extension added a new exon, which was regarded as the first exon, then a new intron, which was usually the first intron, was hypothesized to exist. The numbers of coding sequences obtained for each gene in the heme biosynthesis pathway were as follows: ALAS (46 genes from 31 species), PBGS (26 genes from 26 species), PBGD (29 genes from 28 species), UROS (28 genes from 28 species), UROD (31 genes from 31 species), CPO (27 genes from 27 species), PPO (26 genes from 26 species) and FECH (30 genes from 30 species) (Table S7). Lists about the number of sequences and the species by taxonomic groups can be found in Table S8 and S9. Sequences from species whose genome had not been decoded were not included in this study.\\n\\n\\nSequence Alignment and Phylogenetic Analysis\\n\\nThe amino acid sequence alignment of the eight enzymes involved in the heme biosynthesis pathway was performed in MEGA5 with the MUSCLE algorithm [70]. After the aligned sequences were adjusted manually to confirm their accuracy, topologies of the phylogenetic trees were generated with PHYLIP [71] and PHYML [72] via maximum likelihood (ML) methods. The gamma rate heterogeneity model and the JTT substitution model from Tree-Puzzle [73] and MEGA5 were used for the enzymes of the heme biosynthesis pathway. Branch support was provided via bootstrap analysis, involving a heuristic search with 1000 replicates. The alignments and trees are shown in Figures S2 and S3, respectively.\\n\\n\\nAnalysis of Evolutionary Constraints\\n\\nThe aligned nucleotide coding sequences without gaps based on the aligned protein sequences and the unrooted tree were fed into the CODEML program of the PAML program package (version V4.4e) [74], [75] to analyze the evolutionary constraints on the coding sequences for the eight genes of the heme biosynthesis pathway. For the different models applied, ω (the nonsynonymous versus synonymous rate, dN/dS) for codons can be assumed to be less than one (negative selection or purifying selection), equal to one (neutral), or greater than one (positive selection).\\n\\nFirst, we obtained ω for each gene by applying the null model M0, assuming a constant ω value for all codons and branches. Second, site models M1a (Nearly Neutral) and M2a (Positive Selection) were both used to allow ω to vary among sites. In the M1a model, the codons are categorized into two types, one of which shows ω values of less than one, while ω is equal to one of the other type. In the M2a model, the codons are categorized into three types, one with ω values less than one, one for which ω is equal to one and one with ω values greater than one. We performed the likelihood ratio test for the M1a and M2a models compared to the null model, M0 (twice the log-likelihood difference (2ΔlnL) of the two models).\\n\\nWe also used the branch model to analyze the evolutionary constraints on the mammal, teleost and arthropod branches. We then compared the likelihood of the null model, M0, with the branch model. If the likelihood of the branch model were to be significantly higher than that of the null model M0, it was hypothesized that the branch under consideration is potentially under positive selection. Subsequently, we used the branch-site model to allow variation among sites in the proteins and across branches to determine whether any amino acid residues were under positive selection [76]. For branch-site model A, the following four classes were demarcated for each amino acid: class 0 with 0< ω0<1 in all branches; class 1 with ω1\\u200a=\\u200a1 in all branches; class 2a with foreground ω2≥1 but background 0< ω0<1; and class 2b with a foreground ω2≥1 but background ω1\\u200a=\\u200a1. Null model A1 was the same as A but with the foreground ω2 constrained to one. A likelihood ratio test is used between models A and A1. If the likelihood of model A were to be significantly higher than the likelihood of model A1 (null model) (p<0.05), it would indicate that there were amino acid residues under positive selection. Bayes empirical Bayes (BEB) was used to calculate the posterior probability of identifying sites under positive selection (>0.95) [77]. When we describe the position of the amino acids under positive selection, we use the human sequences (for ALAS1, PBGS and UROD) as a reference to index the sequences. We mapped the positively selected amino acids onto the crystal structures of ALAS [34], PBGS [35] and UROD [36].\\n\\nWe also compared the ω values among the eight genes using a previously described method [2], [8]. When the difference of the ω values between two genes, such as gene A, with ωA and a likelihood of LA0 in the M0 null model, and gene B, with ωB and a likelihood of LB0 in the M0 null model, was significant, we were able to find a ωn between the two genes that was significantly different from those of the two genes being compared. To perform this test, we set ωn as the average ω value of the two genes being compared (ωn\\u200a=\\u200a(ωA+ωB)/2). We then acquired the likelihood values (LA and LB) after we constrained the ω of genes A and B as ωn. Subsequently, the statistical significance (df\\u200a=\\u200a1, p<0.05) of the difference in ω was determined from the difference in the likelihood between the null model M0 and the constrained model, as follows: 2*(LA – LA0) and 2*(LB – LB0).\\n\\n\\nDetection of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences\\n\\nWe used the UCSC table browser [78], [79] and Galaxy [80] to extract the intron sequences at the intersections of regions that are conserved in vertebrates (hg19, phastConsElements46way) and DNase-hypersensitive sites (hg19, wgEncodeRegDnaseClustered) [81].\\n\\n\\nDetection of IREs\\n\\nWe used the SIRE web service to identify potential IRE sites in the gene sequences [82]. SIRE takes into consideration the non-canonical sequence as indicated by SELEX. By allowing 18 motifs to be confirmed as binding to IRP1 or IRP2, SIRE accepted the input sequence and reported the motif type, the free energy of the secondary structure and the level of stringency as High, Medium, or Low. A batch version of the same algorithm was also developed for the genome-wide detection of IREs at the intron-exon boundaries of human and zebrafish genes.\\n\\n\\nDetection of HRMs\\n\\nThe HRM sequence, N/K/R-C-P-K or a hydrophobic residue-L/M, has been commonly used to detect HRMs [19], [31] (denoted HRM_t). By identifying new occurrences of HRMs that have been shown to function biologically (human ALAS1; human ALAS2 [32]; human, mouse, rat, spalax, and zebrafish PER2 [60]; human IRP2 [59]; human STC2 [83]; and human, mouse and rabbit eIF2alpha [84]), we also identified a new HRM motif, denoted HRM_r, A/C/F/G/I/R/S/Q-A/C/K/H/L/N/R/S/T-C-P-A/E/F/I/K/S/V/Y-A/D/H/I/L/M/T/V-A/L/M/P/R/S. We used seven sites in HRM_r rather than five sites in HRM_t to increase the specificity of HRM_r. We note that HRM_r is similar to, but not the same as HRM_t.\\n\\n\\nSupporting InformationFigure S1\\n\\n\\nThe collection of figures describing the protein structures of enzymes involved in the heme biosynthesis pathway and the positions of the positively selected residues. Homodimer structures are shown with the monomers colored in white and yellow. Positively selected amino acid residues are colored in red. Substrate analogs or prosthetic groups are colored in green. (A) ALAS2 of teleost; (B) PBGS of arthropod; (C) UROD of teleost.\\n\\n(PDF)\\n\\nClick here for additional data file.Figure S2\\n\\n\\nThe aligned protein sequences of eight genes of heme biosynthesis pathway in animals.\\n\\n(PDF)\\n\\nClick here for additional data file.Figure S3\\n\\n\\nThe collection of figures for the maximum likelihood phylogeny of protein sequences for eight genes of heme biosynthesis pathway in animals. Bootstrap values >70% are indicated. The bootstrap values are displayed only for the branches of the main lineages. (A) ALAS, (B) PBGS, (C) PBGD, (D) UROS, (E) UROD, (F) CPO, (G) PPO, (H) FECH.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S1\\n\\n\\nModel test (M1a) for selection of genes in heme biosynthesis pathway.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S2\\n\\n\\nLength of evolutionarily conserved DNase-hypersensitive sites in intron sequences.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S3\\n\\n\\nPotential IRE in eight genes of heme biosynthesis pathway.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S4\\n\\n\\nGenomewide detection of potential IRE in exon-intron boundary in human and zebrafish genes.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S5\\n\\n\\nPotential HRM in eight genes of heme biosynthesis pathway (HRM_t and HRM_r).\\n\\n(PDF)\\n\\nClick here for additional data file.Table S6\\n\\n\\nLength of evolutionarily conserved DNase-hypersensitive sites in intron sequences (bps) for Abcg2, Ank1 and Slc11a2.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S7\\n\\n\\nIDs of genes of heme biosynthesis pathway in animals.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S8\\n\\n\\nNumber of sequences by taxonomic groups.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S9\\n\\n\\nSpecies by taxonomic groups.\\n\\n(PDF)\\n\\nClick here for additional data file.'}"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.find_one()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using `collection.update_one()`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in files:\n",
    "    with open(file, \"r\") as f:\n",
    "        xml = f.read()\n",
    "        article = Article(xml)\n",
    "        articles.update_one(\n",
    "            filter = {\n",
    "                \"_id\": article.pub_ids().get(\"pmc\")\n",
    "            },\n",
    "            update = {\n",
    "                \"$set\": {\n",
    "                    \"xml\": xml,\n",
    "                    \"extracted_text\": article.extract_text()\n",
    "                }\n",
    "            },\n",
    "            upsert=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Stuff for Python"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO: Figure out how to use a boolean query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'extracted_text_text'"
      ]
     },
     "execution_count": 204,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.create_index([(\"extracted_text\", pymongo.TEXT,)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [],
   "source": [
    "study_site_query = '\"field study\"'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8404"
      ]
     },
     "execution_count": 236,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.count_documents(\n",
    "   { '$text': { '$search': study_site_query } }\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = articles.find(\n",
    "   { '$text': { '$search': study_site_query } },\n",
    "   { 'score': { '$meta': \"textScore\" } }\n",
    ").sort([('score', {'$meta': 'textScore'})])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_list = [result[\"extracted_text\"] for result in results]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "1. Introduction\n",
      "\n",
      "The current treatments for GBMs include surgery, fractionated radiotherapy (FR), and temozolomide. This approach provides a definite effect as patients receiving this multimodal treatment have a median survival of approximately 15 months [1, 2], compared with 3 months if no treatment is given. Surgical debulking reduces symptoms and provides tissue for diagnosis, but infiltrative tumor growth makes complete removal impossible. Conventional radiotherapy improves survival [3] but is associated with noteworthy toxicity due to the high doses delivered to the surrounding brain tissue. Thus, patients surviving more than 12 months often exhibit significant cognitive deficits. Due to the improved survival of GBM patients in recent years with two year survival rates of 26.5%, more people will live to experience these side effects [4]. As such, optimized radiation modalities to reduce toxicity are warranted. Future treatments may include lower dose FR combined with hypofractionated stereotactic radiosurgery and additional tumor selective radiosensitizers. To explore innovative regimens for treating malignant glioma, a representative animal model is needed.\n",
      "\n",
      "GKS enables the delivery of a high tumor dose in one single fraction by converging multiple beams to a stereotactically defined target, creating a sharp dose fall-off at the tumor margin with minimal radiation to the surrounding brain [5]. GKS is not used in the primary treatment of GBMs, but there are several reports describing its use for GBM recurrences in patients that have already received primary treatment, including FR [6–9]. Thus, little is known about the effects specifically associated with GKS.\n",
      "\n",
      "However, most GBM recurrences are local, suggesting a role for focused radiation [10]. Moreover, experimental studies suggest a therapeutic benefit with focused radiation of rats implanted with rat C6 and rat 9L glioma cells [11, 12]. These rat glioma cell lines however, do not mimic the infiltrative growth of human GBMs. Furthermore, previous animal studies have involved treatment with lateral and posterior-anterior plain X-rays [11] or additional therapies such as injection of viral vectors [12]. Thus, there remains a need to evaluate the radiobiology of GKS with clinically relevant doses in tumor models that mimic the infiltrative growth of human GBMs. We previously established highly infiltrative brain tumors in nude rats from human GBM biopsies [14]. Direct xenografting of patient GBM biopsies into nude rats has been shown to produce tumors with infiltrative growth patterns without angiogenesis, similar to low-grade glioma [13]. Serial passaging of these tumors in vivo transforms purely invasive phenotypes into mature GBM phenotypes that are also angiogenesis-dependent and necrotic [13, 14]. In this study, we used this highly representative GBM model to evaluate the effect of GKS treatment on survival, local tumor control, and radiotoxicity.\n",
      "\n",
      "\n",
      "2. Material and Methods\n",
      "\n",
      "\n",
      "2.1. Tumor Spheroids\n",
      "\n",
      "GBM biopsies were obtained during surgical resections performed at the Department of Neurosurgery at Haukeland University Hospital (Bergen, Norway), with informed consent from the patients and approval by the regional ethical committee. The biopsy spheroids were prepared as described previously [15] and passaged for several generations in immunodeficient rats [13]. Tumor spheroids were prepared from two GBM patients, pA and pB.\n",
      "\n",
      "\n",
      "2.2. Animal Experiments\n",
      "\n",
      "In total, 39 male and female athymic, homozygous nude rats (rnu/rnu, Rowett) between 200 and 250 g were used in 3 different experiments (Table 1). They were kept on a standard pellet diet, in a pathogen free environment at a constant temperature and humidity, on a standard 12/12 h light and dark cycle. The animals were provided a standard pellet diet and tap water ad libitum. All experimental procedures were approved by the Norwegian Animal Research Authority (Oslo, Norway). Prior to implantation, all animals were anesthetized with Isoflurane gas (1.5% mixed with 50% air and 50% O2) and 0.5% Marcaine subcutaneously. Intracranial tumor implantations were conducted with a stereotactic frame (Kopf instruments, Tujunga, CA, USA) as previously described [14].\n",
      "\n",
      "In the first experiment, animals with comparable tumor volumes derived from a patient GBM xenograft, pA, were randomly assigned to 3 different groups: (1) GKS with 12 Gy margin dose; (2) GKS with 18 Gy margin dose; and (3) untreated controls. In the second experiment, we used another GBM xenograft, pB, and the animals were randomized to two groups: (1) GKS with 12 Gy margin dose and (2) untreated controls. Finally, 6 animals were grafted with pA tumor material in a separate experiment to assess radiotoxicity (see Section 2.6 describing the open field study). Animals were inspected daily by the staff at the animal facility that were blinded to the treatments the animals had undergone. Animals were sacrificed with CO2 at the onset of symptoms such as weight loss, neurological deficits, passiveness, or other signs of illness. All animals were observed until the onset of symptoms or until 165 days after implantation. The brains were removed and fixed in 4% formalin. Survival was used as primary endpoint and histological changes and volume changes on MRI follow-up were used as secondary endpoints.\n",
      "\n",
      "\n",
      "2.3. Magnetic Resonance Imaging\n",
      "\n",
      "Animals implanted with pA and pB tumor spheroids underwent MRI scanning 6 and 7 weeks after implantation, respectively, using a Bruker Pharmascan 7 T small animal MRI (Bruker Biospin MRI GmnH, Ettingen, Germany). Animals implanted with pB GBM spheroids in the second experiment also underwent a repeated scanning one week later. Axial T1 and T2 weighted images were acquired as previously described [14]. The tumor volumes at treatment and on follow-up MRI were measured in Gamma Plan (Elekta Instrument AB, Stockholm, Sweden).\n",
      "\n",
      "\n",
      "2.4. Gamma Knife Treatment\n",
      "\n",
      "Animals were irradiated with the Leksell Gamma Knife Perfexion (Elekta Instrument AB, Stockholm, Sweden) anesthetized with intramuscular Hypnorm-Dormicum (0, 4 mL/kg). One day after MRI, the rats were immobilized in a Regis-Valliccioni stereotactic frame 5 (Neuropace, Neuilly, France (Figure 1(a))) [16] and underwent CT scanning with the frame attached. The MRI images of the tumors were coaligned with the CT scans in the Gammaplan, aided by anatomical landmarks and the visible trajectory from the tumor implantation (Figure 1(b)). The rats were randomized to different treatment groups and treated with a tumor margin dose of 12 Gy or 18 Gy to the 50–88% (mean 74.6%) isodose, using collimator size 4 (Figure 1(c)), or they were randomized to no treatment. The mean treatment time was 5.0 minutes. In the radiotoxicity experiment, the rats received early GKS only 9 days after implantation to allow for a long follow-up time. These animals received 12 Gy to the 80% isodose, centered at the site of the implantation (Figure 1(c)).\n",
      "\n",
      "\n",
      "2.5. Histology and Immunohistochemical Analyses\n",
      "\n",
      "Tissue blocks were paraffin embedded and 5 μm sections were obtained. These were stained with haematoxylin and eosin and examined by a neuropathologist. Sections were prepared as described previously [13] and immunostained for Nestin (1 : 1000, Chemicon, Temecula, CA, USA), Vimentin (1 : 500, DAKO, Glostrup, Denmark), and Ki-67 (MIB-1, 1 : 500, Dako). Cell death was determined using the terminal transferase-mediated dUTP nick-end-labeling (TUNEL) method according to the manufacturer's instructions (Boehringer Mannheim, Mannheim, Germany) using DAB as a chromogen. TUNEL staining was quantitated as % immunopositive area fraction at 400x magnification, using a Nikon light microscope (THP Eclipse E600, Nikon Corporation, Tokyo, Japan) and the NIS-Elements BR 4.0 software (Nikon Corporation). Four hotspot fields of view were analyzed per animal from each treatment group and control group and quantified by a person blinded to the groups.\n",
      "\n",
      "\n",
      "2.6. Open Field Experiment\n",
      "\n",
      "In order to evaluate locomotor and exploratory behavior deficits as a measure of radiotoxicity, we performed an open field test 2 days and 4 weeks after randomization to treatment or no treatment: (1) GKS after tumor implantation (3 animals); (2) no treatment after tumor implantation (3 animals); and (3) controls with no tumor nor treatment (3 animals). The open field behavioral test was developed by Calvin S. Hall [17], and measures general locomotor activity and willingness to explore in rodents. The field consists of a square black box, 1 m × 1 m with 40 cm high walls. All animals were placed in the centre of the open field arena and the animals' spontaneous activities for 9 minutes were digitally recorded. Latency to enter the periphery (indicative of anxiety) and central zone, angular velocity, which is the animals' ability to turn in a specific direction, general mobility, and time spent in left or right part of the box were measured.\n",
      "\n",
      "\n",
      "2.7. Statistical Methods\n",
      "\n",
      "Animal survival was registered from the time of implantation and from the time of treatment. Survival was plotted on a Kaplan-Meyer survival curve, and median survival times were compared using the log-rank test in SPSS version 18.0. Univariate analyses of categorical variables according to treatment group (12 Gy versus 18 Gy), 12 Gy volume outside the tumor (<30 mm³ versus >70 mm³), maximum dose (<20 Gy versus >20 Gy), tumor volume (<5 mm³ versus 5–15 mm³ versus >15 mm³), and the percentage isodose used during treatment (<80% isodose curve versus >80% isodose curve) were conducted. Tumor volumes on T1 weighted MRI images on the day of the treatment and follow-up images were measured based on the contrast-enhancing areas to detect differences in tumor volumes, and the treatment groups were compared using Student's t-test. Ki-67 labeling and apoptotic indices were analyzed using ANOVA. The open field experiments were analyzed using ANOVA for repeated measures and the post hoc Fisher LSD test. A probability value ≤0.05 was regarded significant.\n",
      "\n",
      "\n",
      "3. Results\n",
      "\n",
      "\n",
      "3.1. Survival\n",
      "\n",
      "In the first experiment using pA GBM xenografts, 8 rats received a 12 Gy margin dose (mean volume 12.0 ± 7.9 mm³), 7 rats received an 18 Gy margin dose (mean volume 8.6 ± 5.9 mm³), and 6 rats served as controls (mean volume 10.0 ± 6.7 mm³, P < 0.65). The median survival from the time of implantation was 96 days (95% CI 78.3–113.7) for the two treatment groups collectively, compared to 72 days (95% CI 68.6–75.4) for the untreated rats (P < 0.0001, Figure 2(a)). The median survival from the time of GKS was 41 days (95% CI 23.3–58.6) for the GKS treated rats compared to 10 days (95% CI 6.6–11.4) for the untreated rats (P < 0.0001, Figure 2(b)). Notably, the median survival time both from implantation and from the time of treatment was significantly longer for animals treated with 12 Gy (131 days from implantation, 72 days from GKS), compared to animals treated with a 18 Gy margin dose (84 days and 22 days, respectively, P < 0.001, Figures 2(c) and 2(d).\n",
      "\n",
      "The rats treated with an 18 Gy margin dose had a mean maximum dose of 23.0 ± 1.7 Gy, whereas the rats treated with a 12 Gy margin dose had a mean maximum dose of 15.5 ± 1.9 Gy (P < 0.001). Furthermore, for the rats receiving 18 Gy the mean 12 Gy volume outside the tumor was 77.9 ± 10.7 mm³ versus 20.7 ± 7.7 mm³ for the rats receiving 12 Gy (P < 0.001). A multivariate analysis according to margin dose and adjusted for maximum dose, tumor volume, and volume of normal brain receiving >12 Gy confirmed a significantly longer survival in the 12 Gy group (P < 0.02).\n",
      "\n",
      "In the second series, spheroids from the pB GBM xenograft produced a more rapid disease course than the pA tumors (Figures 2(e) and 2(f)). At the time of GKS 7 weeks after implantation, the tumors had reached a bigger volume than in the first series. The tumors were treated with 12 Gy to the periphery. The mean volume for the pB tumors was 120 mm³ (versus 10.3 mm³ for the pA tumors), with no significant differences between the 4 treated animals and the 5 controls. However, even for these late stage tumors, median survival was significantly longer for the GKS treatment group both from the time of implantation, 72 days versus 54 days (P = 0.006, Figure 2(e)), and from GKS, 12 days versus 5 days (P = 0.006, Figure 2(f)).\n",
      "\n",
      "\n",
      "3.2. MRI Monitoring of Tumor Growth\n",
      "\n",
      "In the second experiment, with pB tumors, MRI performed 8 days after the initial MRI showed a larger increase in tumor volume for the untreated controls compared to rats subjected to GKS (Figure 3). The mean difference in tumor volumes between baseline and follow-up was 111.8 mm³ for the 12 Gy group versus 252.8 mm³ for the controls (P = 0.032). The tumors increased in volume by a factor of 2.9 for the controls versus a factor of 2.1 for the treated rats.\n",
      "\n",
      "\n",
      "3.3. Tumor Histology\n",
      "\n",
      "Both the pA and pB GBM xenografts produced tumors in all rats that were implanted. The pA tumors showed predominantly infiltrative growth with extensive cellular infiltration throughout the brain, whereas the pB tumors predominantly displayed microvascular proliferation with necrosis (Figures 4(a) and 4(b)). Brains from the treatment groups also showed histological changes suggesting granulomatous cerebral angiitis with polynuclear macrophages in the perivascular space, tumor cells in the Virchow-Robin space, and fibrosis (Figure 4(a), upper right panel). Notably, the number of microscopic fields (×10) with the presence of granulomatous angiitis were significantly higher in the group receiving 18 Gy (49%) compared to the group receiving 12 Gy (8%, P = 0.02).\n",
      "\n",
      "Recently, several studies have suggested that cancer stem cells mediate radioresistance in gliomas [18–20]. Thus, we investigated the expression of stem cell markers Nestin and Vimentin (Figure 4(b)). We observed Nestin and Vimentin expression in the treatment groups as well as the controls, both in the tumor bulk and the invasion zone. Labeling index for the proliferation marker Ki-67 did not differ significantly between any of the groups. In the first experiment, control tumors displayed 18 ± 2.0% Ki-67 positive cells, tumors treated with 12 Gy contained 19 ± 3.9% positive cells, whereas 18 Gy treated tumors had 18 ± 0.7% Ki-67 positive cells (P = 0.99). Apoptosis estimated by % area fractions of TUNEL-positive staining did not show an increase with radiation dose and were 0.87% for the controls, 0.204 for the 12 Gy group, and 2.210 for the 18 Gy group, (Figure 4(b), right panels).\n",
      "\n",
      "\n",
      "3.4. Open Field Test\n",
      "\n",
      "In order to evaluate radiotoxic side effects, we performed an open field test two days and four weeks following treatment (Figure 5). We compared the activity and exploratory behavior of rats with GBM receiving GKS with the behavior of both untreated rats harboring GBM tumors and untreated rats without tumors (controls). Two days after GKS, the treated rats had a longer latency to enter the periphery zone (P = 0.005) and seemed less active in the open field (P = 0.09) than untreated rats with tumors. After 4 weeks, all rats with tumors were less mobile than the controls. Notably, the latency for moving to the outer ring was longer in the untreated rats with tumors than the treated rats with brain tumors (P = 0.04) and the controls with no tumor (P = 0.002).\n",
      "\n",
      "\n",
      "4. Discussion\n",
      "\n",
      "The potential for GSK in GBM management has so far not been fully elucidated, whereas the Stupp regime is widely adapted as a first line of treatment. However, GSK can be useful for small, deep seated, or recurrent tumors. Herein we assessed the efficacy of GSK treatment and its potential toxicity in a relevant model of GBM in nude rats. We established that GSK treatment prolonged survival of animals and that this was associated with minimal toxicity as indicated by mild histological and behavioral changes.\n",
      "\n",
      "The accuracy of the Regis-Valliccioni frame has been documented previously [21, 22]. Moreover, the precision of 12 and 18 Gy dose delivery in our study was verified firstly by the presence of radiation-induced changes in the isodose center on histological examination in the rats receiving GKS and secondly by reduced tumor growth on follow-up MRI.\n",
      "\n",
      "Previously, Kondziolka et al. reported prolonged survival after GKS for C6 glioma in rats receiving margin doses in the range of 30–100 Gy [11], whereas the survival seemed independent of the dose given. Khil et al. found a dose-dependent improvement in survival using margin doses of 25 Gy, 35 Gy, and 44 Gy in rats with 9L glioma cells [23]. However, these experiments included non-human, chemically induced glioma cell lines treated with high doses which cannot be administered clinically in single fraction stereotactic radiosurgery to previously irradiated patients. Accordingly, there remains a need to evaluate the radiobiological effects of GKS with doses relevant to the clinical setting of tumor relapse in a patient that has previously received FR.\n",
      "\n",
      "In this study, GBM xenografts derived from patient tumors displayed the typical morphology of human GBMs, including infiltrative growth, florid angiogenesis, and necrotic regions. Moreover, the tumors were visible on MRI at the time of treatment, resembling the treatment setting in patients. In clinical studies validating the use of GKS for tumor recurrences in GBM patients, the doses have varied between 12 and 18 Gy [6–8]. Therefore, we selected 12 Gy and 18 Gy as representative clinical doses for our experiments. Our data demonstrate that monotherapy with GKS using clinically relevant doses prolongs survival. With a similar efficacy in patients, GKS may be an alternative to surgery in cases with deep seated small GBMs, when a diagnostic biopsy has been performed in patients who are otherwise unable to undergo a full craniotomy under narcosis due to severe comorbidity. Although most GBMs are too large for GKS at the time of diagnosis, GKS could be preferable to conventional radiotherapy in the initial treatment of small tumors due to a sharper dose fall and less toxicity [24]. Moreover, GKS in combination with lower, less toxic doses of conventional radiotherapy may improve future results.\n",
      "\n",
      "Interestingly, rats treated with 12 Gy to the tumor margin lived significantly longer survival than rats treated with an 18 Gy margin dose. The reduced survival in the 18 Gy group could be caused by edema and radiation damage to adjacent brain tissue as a considerable volume of normal brain received more than 12 Gy in this group, which is seen as the limit for radiation damage of the normal human brain [25]. Polynuclear macrophage clusters in the perivascular space and fibrosis as seen in radiation induced granulomatous cerebral angiitis were found in the 18 Gy rats. Moreover, increased radiation injury with the 18 Gy treatments may trigger immune-mediated damage of healthy nervous tissue. However, it should be pointed out that even the 4 mm collimator becomes a crude tool for dose delivery, given the small rat brain volumes. Therefore, the extratumoral 12 Gy volumes in these animals are relatively much larger than in patients, and the observed toxicity may therefore have a very limited relevance to the clinical setting. In open field studies, animals with untreated tumors, animals treated with 12 Gy, and controls were tested. The rats treated with 12 Gy fared better than rats with untreated tumors. In addition, rats treated with 12 Gy improved from 2 days to 4 weeks after GKS, showing a shorter latency for moving to the outer ring of the open field arena after the start of the test. The data indicate a positive long-lasting effect of the treatment on rats' locomotor behavior. These animals readily started moving around, whereas animals receiving no treatment responded with more immobility or freezing in the start of the test. The latency to the outer zone showed large variation between individuals; however, the effects of the treatment on this parameter were robust compared to animals with untreated tumors. Thus, GKS using the 12 Gy dose seemed to have low toxicity with little impact on behavior. Notably, controls without tumor and no treatment also performed significantly better than untreated rats with tumors. Thus, the impact of the location of tumors on the locomotor behavior of the animal is likely as important as GKS.\n",
      "\n",
      "\n",
      "5. Conclusion\n",
      "\n",
      "We conclude that GKS administered as monotherapy with clinically relevant doses prolongs survival with low associated toxicity in rats harboring GBM xenografts. This in vivo nude rat model displays tumor implants highly representative of human GBMs. Moreover, the precision of dose delivery was verified by reduced tumor growth and the presence of radiation-induced histological changes in the rats subjected to GKS. The model described represents a valuable tool for future small animal studies. Additional studies on potential drug modification of the radiosurgery response in human GBM are warranted.\n"
     ]
    }
   ],
   "source": [
    "print(result_list[2])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Basic code for walking with a limit\n",
    "\n",
    "However, this isn't usable, because we actually need to *randomly* sample the files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "Stopped.\n"
     ]
    }
   ],
   "source": [
    "i = 10\n",
    "for path, dirs, files in os.walk(pmc_path):\n",
    "    if i <= 0:\n",
    "        break\n",
    "    for file in files[:i]:\n",
    "        if file.endswith(\"xml\"):\n",
    "            i -= 1\n",
    "            with open(os.path.join(path, file), \"r\") as file:\n",
    "                article = Article(file.read())\n",
    "            print(article.pub_ids().get(\"pmid\"))\n",
    "print(\"Stopped.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Counting how many have PMIDs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "foo = dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(foo.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2500\n",
      "2475\n",
      "2450\n",
      "2425\n",
      "2400\n",
      "2375\n",
      "2350\n",
      "2325\n",
      "2300\n",
      "2275\n",
      "2250\n",
      "2225\n",
      "2200\n",
      "2175\n",
      "2150\n",
      "2125\n",
      "2100\n",
      "2075\n",
      "2050\n",
      "2025\n",
      "2000\n",
      "1975\n",
      "1950\n",
      "1925\n",
      "1900\n",
      "1875\n",
      "1850\n",
      "1825\n",
      "1800\n",
      "1775\n",
      "1750\n",
      "1725\n",
      "1700\n",
      "1675\n",
      "1650\n",
      "1625\n",
      "1600\n",
      "1575\n",
      "1550\n",
      "1525\n",
      "1500\n",
      "1475\n",
      "1450\n",
      "1425\n",
      "1400\n",
      "1375\n",
      "1350\n",
      "1325\n",
      "1300\n",
      "1275\n",
      "1250\n",
      "1225\n",
      "1200\n",
      "1175\n",
      "1150\n",
      "1125\n",
      "1100\n",
      "1075\n",
      "1050\n",
      "1025\n",
      "1000\n",
      "975\n",
      "950\n",
      "925\n",
      "900\n",
      "875\n",
      "850\n",
      "825\n",
      "800\n",
      "775\n",
      "750\n",
      "725\n",
      "700\n",
      "675\n",
      "650\n",
      "625\n",
      "600\n",
      "575\n",
      "550\n",
      "525\n",
      "500\n",
      "475\n",
      "450\n",
      "425\n",
      "400\n",
      "375\n",
      "350\n",
      "325\n",
      "300\n",
      "275\n",
      "250\n",
      "225\n",
      "200\n",
      "175\n",
      "150\n",
      "125\n",
      "100\n",
      "75\n",
      "50\n",
      "25\n",
      "defaultdict(<class 'int'>, {'pmid': 2484, 'pmc': 2500, 'doi': 2500, 'publisher-id': 2476, 'art-access-id': 4, 'manuscript': 11, 'coden': 2002, 'pii': 1995})\n"
     ]
    }
   ],
   "source": [
    "id_types = defaultdict(int)\n",
    "\n",
    "i = 2500\n",
    "for path, dirs, files in os.walk(pmc_path):\n",
    "    if i <= 0:\n",
    "        break\n",
    "    for file in files[:i]:\n",
    "        if file.endswith(\"xml\"):\n",
    "            if i % 25 is 0:\n",
    "                print(i)\n",
    "            i -= 1\n",
    "            with open(os.path.join(path, file), \"r\") as file:\n",
    "                article = Article(file.read())\n",
    "            ids = article.pub_ids().keys()\n",
    "            if len(ids) is 0:\n",
    "                id_types[\"none\"] += 1\n",
    "            else:\n",
    "                for item in ids:\n",
    "                    id_types[item] += 1\n",
    "            \n",
    "print(id_types)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
