{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pubcrawler.article import Article"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.display import clear_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pymongo\n",
    "from bson.objectid import ObjectId\n",
    "from collections import defaultdict\n",
    "from random import sample, seed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "import time"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create a list of files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "pmc_path = \"/Volumes/Transcend/datasets/pmc/2019-04-03/oa_bulk/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_files = []\n",
    "for (dirpath, dirnames, filenames) in os.walk(pmc_path):\n",
    "    all_files += [os.path.join(dirpath, filename) for filename in filenames if filename.endswith(\"xml\")]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Counting coverage of different ID types from a sample of 5000 articles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "files = sample(all_files, 500)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.display import clear_output\n",
    "class Reporter:\n",
    "    def __init__(self, interval, total):\n",
    "        self.start = time.time()\n",
    "        self.this_time = self.start\n",
    "        self.interval = interval\n",
    "        self.total = total\n",
    "        print(self.start, self.this_time, self.interval, self.total)\n",
    "    \n",
    "    def report(self, idx):\n",
    "        self.last_time = self.this_time\n",
    "        self.this_time = time.time()\n",
    "        elapsed = self.this_time - self.start\n",
    "        mean_time_per = elapsed / idx\n",
    "        mean_time_left = (self.total - idx) * mean_time_per\n",
    "        this_time_per = (self.this_time - self.last_time) / self.interval\n",
    "        this_time_left = (self.total - idx) * this_time_per\n",
    "        output = \"\"\n",
    "        output += \"Processed {0} articles ({1:.1f}% done).\\n\".format(idx, idx/self.total * 100)\n",
    "        output += \"Elapsed: {0:.1f} sec\\n\".format(elapsed)\n",
    "        output += \"Time Per All {0}: {1:.3f} sec    Est Remaining: {2:.1f} sec\\n\".format(idx, mean_time_per, mean_time_left)\n",
    "        output += \"Time Per Last {0}: {1:.3f} sec    Est Remaining: {2:.1f} sec\\n\".format(report_every, this_time_per, this_time_left)\n",
    "        output += \"Est Total (Mean): {0:.1f} sec    Est Total (Last {1}): {2:.1f} sec\".format(elapsed + mean_time_left, self.interval, elapsed + this_time_left)\n",
    "        print(output, flush=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1554496751.104486 1554496751.104486 50 10000\n",
      "Processed 50 articles (0.5% done).\n",
      "Elapsed: 2.2 sec\n",
      "Time Per All 50: 0.044 sec    Est Remaining: 435.4 sec\n",
      "Time Per Last 50: 0.044 sec    Est Remaining: 435.4 sec\n",
      "Est Total (Mean): 437.6 sec    Est Total (Last 50): 437.6 sec\n",
      "Processed 100 articles (1.0% done).\n",
      "Elapsed: 4.5 sec\n",
      "Time Per All 100: 0.045 sec    Est Remaining: 445.8 sec\n",
      "Time Per Last 50: 0.046 sec    Est Remaining: 458.4 sec\n",
      "Est Total (Mean): 450.3 sec    Est Total (Last 50): 462.9 sec\n",
      "Processed 150 articles (1.5% done).\n",
      "Elapsed: 6.4 sec\n",
      "Time Per All 150: 0.043 sec    Est Remaining: 422.2 sec\n",
      "Time Per Last 50: 0.039 sec    Est Remaining: 379.6 sec\n",
      "Est Total (Mean): 428.7 sec    Est Total (Last 50): 386.0 sec\n",
      "Processed 200 articles (2.0% done).\n",
      "Elapsed: 8.8 sec\n",
      "Time Per All 200: 0.044 sec    Est Remaining: 432.4 sec\n",
      "Time Per Last 50: 0.048 sec    Est Remaining: 469.5 sec\n",
      "Est Total (Mean): 441.2 sec    Est Total (Last 50): 478.3 sec\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-261-6a8c1c76b910>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      7\u001b[0m         \u001b[0mreporter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreport\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0midx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"r\"\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m         \u001b[0marticle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mArticle\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     10\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m     \u001b[0mids\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marticle\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpub_ids\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Dropbox (EHA)/repositories/pubcrawler/pubcrawler/article.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, xml)\u001b[0m\n\u001b[1;32m     13\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mxml\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mxml\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mxml\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 15\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msoup\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mBeautifulSoup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mxml\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'lxml-xml'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     16\u001b[0m         \u001b[0;31m# self._id = article_dict['_id']\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     17\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, markup, features, builder, parse_only, from_encoding, exclude_encodings, **kwargs)\u001b[0m\n\u001b[1;32m    279\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    280\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 281\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_feed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    282\u001b[0m                 \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    283\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mParserRejectedMarkup\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36m_feed\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    340\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuilder\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    341\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 342\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuilder\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmarkup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    343\u001b[0m         \u001b[0;31m# Close out any unfinished strings and close all the open tags.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    344\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mendData\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/builder/_lxml.py\u001b[0m in \u001b[0;36mfeed\u001b[0;34m(self, markup)\u001b[0m\n\u001b[1;32m    166\u001b[0m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmarkup\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mCHUNK_SIZE\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    167\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 168\u001b[0;31m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    169\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    170\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mUnicodeDecodeError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mLookupError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0metree\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mParserError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parser.pxi\u001b[0m in \u001b[0;36mlxml.etree._FeedParser.feed\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parser.pxi\u001b[0m in \u001b[0;36mlxml.etree._FeedParser.feed\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._TargetParserContext._handleParseResult\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._TargetParserContext._handleParseResult\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/etree.pyx\u001b[0m in \u001b[0;36mlxml.etree._ExceptionContext._raise_if_stored\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/saxparser.pxi\u001b[0m in \u001b[0;36mlxml.etree._handleSaxTargetStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/saxparser.pxi\u001b[0m in \u001b[0;36mlxml.etree._callTargetSaxStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32msrc/lxml/parsertarget.pxi\u001b[0m in \u001b[0;36mlxml.etree._PythonSaxParserTarget._handleSaxStart\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/builder/_lxml.py\u001b[0m in \u001b[0;36mstart\u001b[0;34m(self, name, attrs, nsmap)\u001b[0m\n\u001b[1;32m    218\u001b[0m         \u001b[0mnamespace\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getNsTag\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    219\u001b[0m         \u001b[0mnsprefix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_prefix_for_namespace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnamespace\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 220\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msoup\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhandle_starttag\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnamespace\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnsprefix\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mattrs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    221\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    222\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_prefix_for_namespace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnamespace\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36mhandle_starttag\u001b[0;34m(self, name, namespace, nsprefix, attrs)\u001b[0m\n\u001b[1;32m    520\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    521\u001b[0m         \u001b[0;31m# print \"Start tag %s: %s\" % (name, attrs)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 522\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mendData\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    523\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    524\u001b[0m         if (self.parse_only and len(self.tagStack) <= 1\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36mendData\u001b[0;34m(self, containerClass)\u001b[0m\n\u001b[1;32m    417\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    418\u001b[0m             \u001b[0mo\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcontainerClass\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcurrent_data\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 419\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobject_was_parsed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mo\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    420\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    421\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mobject_was_parsed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mo\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparent\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmost_recent_element\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/__init__.py\u001b[0m in \u001b[0;36mobject_was_parsed\u001b[0;34m(self, o, parent, most_recent_element)\u001b[0m\n\u001b[1;32m    438\u001b[0m         \u001b[0mfix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparent\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnext_element\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    439\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 440\u001b[0;31m         \u001b[0mo\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mparent\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprevious_element\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnext_element\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprevious_sibling\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnext_sibling\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    441\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    442\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_most_recent_element\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mo\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.virtualenvs/ai4e-data-labeling/lib/python3.7/site-packages/bs4/element.py\u001b[0m in \u001b[0;36msetup\u001b[0;34m(self, parent, previous_element, next_element, previous_sibling, next_sibling)\u001b[0m\n\u001b[1;32m    267\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    268\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnext_element\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnext_element\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 269\u001b[0;31m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnext_element\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    270\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnext_element\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprevious_element\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    271\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "id_types = defaultdict(int)\n",
    "article_types = defaultdict(int)\n",
    "\n",
    "reporter = Reporter(50, len(files))\n",
    "for idx, filename in enumerate(files):\n",
    "    if idx is not 0 and idx % reporter.interval is 0:\n",
    "        reporter.report(idx)\n",
    "    with open(filename, \"r\") as f:\n",
    "        article = Article(f.read())\n",
    "    \n",
    "    ids = article.pub_ids().keys()\n",
    "    if len(ids) is 0:\n",
    "        id_types[\"none\"] += 1\n",
    "    else:\n",
    "        for item in ids:\n",
    "            id_types[item] += 1\n",
    "    \n",
    "    article_type = article.article_type()\n",
    "    if article_type:\n",
    "        article_types[article_type] += 1\n",
    "\n",
    "clear_output()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(int,\n",
       "            {'pmc': 5000,\n",
       "             'publisher-id': 3494,\n",
       "             'doi': 4347,\n",
       "             'pmid': 4721,\n",
       "             'other': 33,\n",
       "             'pii': 285,\n",
       "             'pmc-scan': 283,\n",
       "             'coden': 61,\n",
       "             'art-access-id': 55,\n",
       "             'manuscript': 50,\n",
       "             'publisher-manuscript': 3,\n",
       "             'sici': 7})"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(int,\n",
       "            {'abstract': 108,\n",
       "             'research-article': 3615,\n",
       "             'case-report': 238,\n",
       "             'product-review': 8,\n",
       "             'review-article': 338,\n",
       "             'other': 178,\n",
       "             'book-review': 76,\n",
       "             'brief-report': 77,\n",
       "             'letter': 86,\n",
       "             'correction': 67,\n",
       "             'news': 29,\n",
       "             'editorial': 82,\n",
       "             'article-commentary': 27,\n",
       "             'meeting-report': 12,\n",
       "             'discussion': 11,\n",
       "             'protocol': 10,\n",
       "             'addendum': 2,\n",
       "             'announcement': 4,\n",
       "             'retraction': 3,\n",
       "             'introduction': 1,\n",
       "             'expression-of-concern': 1,\n",
       "             'methods-article': 8,\n",
       "             'obituary': 7,\n",
       "             'rapid-communication': 4,\n",
       "             'report': 3,\n",
       "             'reply': 1,\n",
       "             'systematic-review': 3,\n",
       "             'data-paper': 1})"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article_types"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It's clear that the PMC ID is the most broadly and consistently applied. Which makes sense! Because we're dealing with PMC articles."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "research-article\n",
      "BODY\n",
      "\n",
      "\n",
      "\n",
      "Background\n",
      "\n",
      "The range of malaria has contracted through a century of economic development and disease control\n",
      "[1-3]. During an era of renewed interest in elimination and eradication there is a need to better understand and quantify the forces behind this recession. A variety of direct control efforts were likely a major factor behind the global contraction in malaria transmission\n",
      "[2,3], although these gains were often coincident with rapid economic and social development and land use changes\n",
      "[4-6]. One major aspect of this development that has been shown to significantly impact malaria transmission is urbanization\n",
      "[7-9].\n",
      "\n",
      "The process of urbanization includes physical landscape modification and transformation of environs through demand for resources and improved communications. Moreover, urbanization involves significant socio-economic change; generally improved health, housing and increased wealth\n",
      "[10,11]. These factors, common to urban areas, cause marked entomological, parasitological and behavioural effects that result in reduced malaria transmission both within the urban core and surrounding peri-urban areas\n",
      "[7,9,12]. A century of rapid urbanization has therefore likely had an impact on malaria transmission globally, but the size and importance of this impact has never been examined.\n",
      "\n",
      "The recent construction of 20th Century time series of global urban extent data\n",
      "[13], and a contemporary malaria transmission map\n",
      "[14] that is comparable to historic data\n",
      "[15], enables a more detailed exploration of this relationship. Here, these datasets were utilized to examine for the first time the association at global, national and subnational scales between changes in malaria transmission and urban growth across the last century.\n",
      "\n",
      "\n",
      "Methods\n",
      "\n",
      "\n",
      "Data\n",
      "\n",
      "\n",
      "Urbanization\n",
      "\n",
      "HYDE 3.1\n",
      "[13,16] is a dynamic modelling effort of long-term historical population growth, providing a consistent database of 20th-century population and urban distribution at 5-minute spatial resolution (equivalent to approximately 10 km at the Equator). Full details of dataset construction are provided in Goldwijk et al.[13,16]. In brief, urban-rural population numbers and fractions for each country and for each decade were obtained from a variety of sources, including the UN\n",
      "[17] and published databases and reviews\n",
      "[18]. Urban areas were then mapped by combining different spatial datasets\n",
      "[19-21] and contemporary urban population datasets, which were derived from existing databases\n",
      "[21,22]. Finally, urban population densities over time were estimated based on the fitting of asymmetric Gaussian probability density function models to the population data\n",
      "[13]. The range of input datasets, assumptions and modeling methods used in construction of the HYDE 3.1 datasets mean that significant uncertainties are inherent in the outputs, and these are likely higher in the mapping covering both the early 20th Century and lower income areas of the World, where data scarcity meant that more assumptions had to be made\n",
      "[13,16]. Nevertheless, at the broad spatial and temporal scales of analysis undertaken here, the effects of such assumptions and uncertainties on results are likely small. Figure \n",
      "1 shows changes in urban population sizes across the malaria endemic world from 1900 to 2000. Additional file\n",
      "1 shows the mapped urban areas for 1900 and 2000, while Additional file\n",
      "2 maps the change in urban extent over the 100-year period, and Additional file\n",
      "3 shows urban extents in 1900 and 2000 for Brazil.\n",
      "\n",
      "Figure 1 Maps showing global changes in urban population size 1900-2000 and change in malaria endemicity. The bar height is proportional to the size of urban population change. Areas that saw no change or an increase in endemicity are coloured dark red, those that saw a reduction by one endemicity class are in red, two classes in orange and three or more classes in yellow.\n",
      "\n",
      "\n",
      "Malaria endemicity\n",
      "\n",
      "The only global map of pre-intervention malaria endemicity dates from a 1968 study\n",
      "[15,23] in which a major synthesis of historical records, documents and maps of a variety of malariometric indices for all four Plasmodium species was used to map parasite rate (PR—the proportion of individuals with malaria parasites detectable in their peripheral blood) and stratified into four endemic classes associated with Plasmodium falciparum endemicity\n",
      "[24] (hypo-endemic, PR < 10%; meso-endemic, PR > 10% and <50%; hyperendemic, PR > 50% and < 75%; holo-endemic, PR > 75%). This map is the only reconstruction of historical malaria at its assumed historical peak around the start of the 20th Century and triangulates well with the plethora of national level malaria maps published throughout the last century\n",
      "[25]. The historical malaria endemicity map was scanned from the original publication, digitized on-screen and rasterized to a 5 × 5 km grid. The map for an area of Brazil is shown in Additional file\n",
      "3(a) – the full map can be seen in Gething et al.[3].\n",
      "\n",
      "The publication of an evidence-based map of contemporary malaria endemicity\n",
      "[14] and its conversion to classes that match the c.1900 map\n",
      "[3] described above, allows an audit of changes in the global epidemiology of malaria since the start of the 20th Century. With just two timepoints of malaria endemicity data, precise information on the timing and progression of changes is absent, placing a limitation on the conclusions that can be drawn from analyses. However, the datasets do provide a unique and valuable quantitative picture of the spatial changes in malaria epidemiology that have occurred over the last 100 years. The map of contemporary malaria endemicity was generated from a recently defined model of age-standardized P. falciparum parasite rate, PfPR2-10\n",
      "[26]. Using a model-based geostatistical framework, the underlying value of PfPR2-10 at each location was modelled for the year 2007 as a transformation of a space-time Gaussian process (GP), with the number of P. falciparum-positive individuals in each survey modelled as a binomial variate given the unobserved age-standardized prevalence surface\n",
      "[14]. The GP was parameterized by a mean component and a space-time covariance function which was spatially anisotropic, used great-circle distance to incorporate the curvature of Earth, and included a periodic temporal component to capture seasonality. Bayesian inference was implemented using Markov chain Monte Carlo and direct simulation to generate posterior predictive samples of the 2007 annual mean prevalence surface and to assign each pixel to the endemicity class with the highest posterior probability of membership. This dataset for an area of Brazil is shown in Additional file\n",
      "3(b). The classes matched those of the c.1900 endemicity map, enabling a class change dataset to be produced, which is shown in Figure \n",
      "1 and Additional file\n",
      "4.\n",
      "\n",
      "In order to compare the observed changes in endemicity between these two time periods with levels of urbanization, it was necessary to translate both the historical and contemporary maps into approximate units of PfRC, the P. falciparum basic reproductive number under the levels of control that existed at the time. This enabled comparisons of the effect sizes of changes in endemicity, following previous studies\n",
      "[3], and was undertaken using a simple P. falciparum transmission model\n",
      "[27] to estimate a value of PfRC corresponding to the mid-value of each endemicity class. Using these conversion values, maps of PfRC were made corresponding to both historical and contemporary endemicity. These two maps were overlaid in a geographical information system (GIS) (ArcGIS 9.2, ESRI Inc, Redlands CA, USA), and the relative change in PfRC between the historic and contemporary map was calculated for each 5 × 5 km pixel. These relative changes were summarized as areas of increase, areas of no change, and areas of decrease of between zero and one, one and two and greater than two orders of magnitude (see Additional file\n",
      "5).\n",
      "\n",
      "Finally, the recent construction of an evidence-based map of the limits of Plasmodium vivax transmission\n",
      "[28] meant that analyses could be repeated to examine the similarity in results between the two parasite species. Additional file\n",
      "6 shows the P. vivax transmission limits map, constructed using data on the presence of P. vivax infection and spatial information on climatic conditions that impede transmission (low ambient temperature and extremely arid environments) in order to delineate areas where transmission was unlikely to take place.\n",
      "\n",
      "\n",
      "Analyses\n",
      "\n",
      "The data analyses focused on statistical explorations of the link between urbanization and changes in malaria endemicity. They covered three areas, examining in each case different spatial scales and factors:\n",
      "\n",
      "(i) National scale analyses of differences in urbanization between countries that eliminated malaria during the past century and those that remain endemic today. Here, urban area percentages between 1900 and 2000 for those countries that remain endemic today, and urban area percentages between 1900 and the date of elimination (Additional file\n",
      "7) for those countries that achieved it were compared.\n",
      "\n",
      "(ii) Within country analyses of subnational differences in urbanization trends between regions that underwent malaria elimination 1900-2007, and regions within the same country that remain endemic today. Here, countries for which >20% of their land area became malaria-free over the past century and for which >20% of their land area remains endemic today were identified, and differences in urban areas, populations and rates of change between them were examined.\n",
      "\n",
      "(iii)  Within country analyses of changes in transmission intensity over the past century and their relationship to urbanization trends. Here, per-country mean changes in transmission class in areas that are urban today, versus those that have remained rural, for malaria-endemic areas in 1900 were examined.\n",
      "\n",
      "For (ii) and (iii), by undertaking analyses at the subnational scale and treating each country independently, the differences in wealth, governance and latitude between countries that are often confounding factors in assessing the drivers of long-term global changes in malaria endemicity are controlled for. For each set of analyses and each time period, the total urban areas and urban population counts were extracted using ArcGIS 9.3 (ArcGIS 9.2, ESRI Inc, Redlands CA, USA) and analysed using R2.10\n",
      "[29].\n",
      "\n",
      "\n",
      "Results\n",
      "\n",
      "A consistent pattern of greater rates of urbanization in areas that either eliminated malaria or displayed the greatest transmission reductions was found. At the time of elimination certification of those countries that achieved malaria elimination, those areas that were originally malarious had significantly higher proportions (Mann-Whitney test: area, z = -3.207, p < 0.01; populations, z = -5.432, p < 0.01) of their populations living in urban areas and higher percentages of urban land area (Figure \n",
      "2) than the contemporary situation in those areas of the world that remain endemic today. Moreover, the rate of urbanization from 1900 onwards was significantly larger in those countries that achieved elimination than those that remain endemic today (see Additional file\n",
      "8). Those countries that achieved elimination exhibited significantly greater increases in the proportions of their populations (Mann-Whitney test: z = -1.985, p < 0.05) and land area (Mann-Whitney test: z = -3.232, p < 0.01) that became urban during the 20 years prior to elimination, compared to the recent rates of urbanization in currently endemic countries over the last 20 years. Such findings confirm the presence of a correlation between urbanization and the achievement of malaria elimination at national scales, but this relationship is likely to be intimately tied to wealth. Countries with greater wealth have greater malaria control resources at their disposal, and it is those wealthier countries that have succeeded in eliminating malaria, as confirmed by undertaking similar analyses based on average income (average income data from\n",
      "[30], Mann-Whitney test: z = -6.491, p < 0.01). Moreover, those higher income countries that succeeded in elimination are located in latitudinal regions of lower climatic and environmental receptivity to malaria transmission, making elimination likely an easier task with a fixed set of intervention tools. Therefore, untangling the effects of urbanization upon changes in malaria transmission required analysis independent of these economic or latitudinal biases, and these are described below.\n",
      "\n",
      "Figure 2 Box plots of the percentage urban area in 1900 and in 2000 for countries that are still endemic and in the decade that elimination was achieved for those countries that achieved elimination.\n",
      "\n",
      "Twenty-nine countries were identified for which at least a fifth of their land area became malaria-free over the past century and for which at least a fifth of their land area remains endemic today. Over 75% of these countries had greater proportions of land area urbanized in the malaria-free areas, and showed a greater percentage increase in urban extent (Figure \n",
      "3) than the areas that remain endemic today. Both of these differences are significant (Wilcoxen test: urban area: z = -2.505 p < 0.05, urban extent change: z = -2.001, p < 0.05). However, these relationships show regional variations, (Figure \n",
      "3). Almost all countries analysed in the Americas (Figure \n",
      "3a) and Asia (Figure \n",
      "3c) showed greater urbanization in their malaria-free versus still-endemic areas. In the Africa and Arabian Peninsula region (Figure \n",
      "3b), where both levels of urbanization and transmission reductions were smaller, the results were more mixed. Many of those falling below the one-to-one line (Figure \n",
      "3a-\n",
      "3c) are countries where extremely arid or mountainous conditions have influenced human settlement to occur principally in areas most suitable for malaria transmission (e.g., Saudi Arabia, Botswana, Swaziland). A full set of national level results can be found in the Additional file\n",
      "9. In addition, the results of a repeat analysis for P. vivax yielding very similar results are shown in Additional file\n",
      "10.\n",
      "\n",
      "Figure 3 Plots showing urban area changes 1900-2000 between areas of countries that remain malaria endemic today, and those that have undergone elimination for (a) the Americas, (b) Africa plus Arabian peninsula and (c) Asia. In each case, scatterplots of urban area increases in endemic versus eliminated areas with one-to-one lines overlaid are shown at the top, and example plots of trends in urban area percentages between areas that eliminated malaria (blue) and that remain endemic (red) are shown at the bottom (the full set of these plots is provided in Additional file\n",
      "9). The ISO country abbreviation for country name is used on the scatterplots (\n",
      "http://www.iso.org/iso/english_country_names_and_code_elements).\n",
      "\n",
      "Urban areas have been shown to exhibit lower levels of transmission than surrounding rural ones\n",
      "[8,9]. If the process of urbanization is a causal factor in malaria declines, a consistent pattern of greater transmission reduction in those areas that have undergone urbanization should be seen. Comparing mean changes in transmission class in areas that are urban today, versus those that have remained rural, for malaria-endemic areas in 1900, was possible in 158 countries. It shows that 82.3% underwent a greater transmission reduction in their urban areas than rural ones (Wilcoxen test: p < 0.001, see Additional file\n",
      "10). Of the 28 countries that displayed a greater malaria reduction in rural areas, half of these were in sub-Saharan Africa, where the smallest levels of urbanization and reductions in transmission occurred. This confirms that greater transmission reductions occurred in areas that are urban today, but does not indicate if the largest declines were coincident with greater rates of urbanization.\n",
      "\n",
      "Finally, examination was undertaken of whether the rate of urbanization was higher in regions of countries that saw the greatest reductions in PfRC. Sixty-three countries were identified that had two or more PfRC reduction classes, each covering at least 20% of their land area, and the rates of urbanization 1900-2000 in the areas exhibiting the smallest reduction in transmission were compared to those in the areas displaying the greatest reduction in transmission. Some 84.1% of countries displayed a greater urban increase in the areas that showed the greater PfRC reduction than those that showed the smallest transmission reductions (Wilcoxen test: p < 0.001, Figure \n",
      "4). Further supporting analyses and more detail on these analyses are presented in Additional file\n",
      "11.\n",
      "\n",
      "Figure 4 Plasmodium falciparum basic reproductive number (PfRC) changes and urbanization. Countries in red show a greater percentage increase in urban area over the past century in areas of the country with the greatest magnitude of PfRC decreases than areas that showed the smallest changes. Countries in yellow show the reverse. Countries in light grey had either insufficient variation in PfRC changes, or no transmission. Countries in dark grey have either always been malaria free, or only exhibited unstable transmission.\n",
      "\n",
      "\n",
      "Discussion\n",
      "\n",
      "The process of urbanization results in a variety of changes that reduce receptivity to malaria transmission\n",
      "[9-11]. Here, a clear picture of increased urbanization associated with greater malaria transmission reductions across countries and continents is documented for the first time. Whilst it remains challenging to untangle whether this urbanization resulted in decreased transmission, or that malaria reductions promoted development, a close relationship is evident, irrespective of national wealth and latitude.\n",
      "\n",
      "Other local evidence\n",
      "[7,12,31,32] hints that changes commensurate with urbanization play a substantive role in driving malaria transmission declines. Whether measured by proportion of land area or population urbanized, the majority of nations that remain malaria endemic today exhibit substantially lower levels of urbanization compared to that at the time of elimination for those countries that have achieved it (Figure \n",
      "2 and Additional file\n",
      "8). Moreover, the magnitude of change in urban extent from 1900 is also correlated with malaria declines, with continental differences, notably sub-Saharan Africa showing lower levels of urbanization today, and smaller changes in urban extent and endemicity over the past century (Figure \n",
      "3).\n",
      "\n",
      "Recent malaria declines in sub-Saharan Africa point towards the success of interventions, however, in several cases the decline began before specific interventions were deployed, suggesting the contribution of alternative factors\n",
      "[33]. While malaria declines due to urbanization and its effects are likely to be more gradual than some of the sudden drops seen, it remains a possibility that the rapid urbanization ongoing in sub-Saharan Africa\n",
      "[34] is at least a contributory driver to these changes.\n",
      "\n",
      "While over three-quarters of countries show decreasing transmission in areas that have undergone urbanization over the past century, a handful of countries go against this trend. A possible reason for this is the likely multi-factorial complexity behind both changes in transmission and human settlement dynamics, and the difficulty of attributing changes to a single cause. Recent analyses have indicated that vector-borne and parasitic diseases have systematically impacted economic growth\n",
      "[35], but more detailed studies of these types of relationships across disease types and ecozones are required to gain a fuller understanding. Almost all of the countries that show greater transmission reductions in rural areas are those where human settlement was constrained to the more malarious areas of the country, due to uninhabitable arid or mountainous areas elsewhere. Equally however, the reverse is true for some countries – i.e. uninhabitable areas with intense transmission forced human settlement in the lower transmission regions.\n",
      "\n",
      "It is clear that various sources of uncertainty exist in the inputs and methodologies used in this study. Uncertainties are inherent in the urbanization datasets through the lack of data for many regions and time periods, and the assumptions made to fill these gaps\n",
      "[13]. The proposed levels of historical endemicity\n",
      "[15,23] are plausible when triangulated against other values reported from the pre-intervention era (for example,\n",
      "[26,36]), but the relatively crude categorization of all-cause malaria endemicity strata and the cartographic approach used preclude a more formal quantification of the global P. falciparum endemicity declines and their link to urbanization beyond the broad relationships presented here. Nevertheless, recent mapping of the limits of P. vivax transmission\n",
      "[28] and analyses of contemporary impacts of urbanization\n",
      "[8], indicate very similar effects and contractions as that seen for P. falciparum, with results repeated, where possible, for P. vivax showing almost identical results (Additional file\n",
      "10), and transmission rarely more intense than meso-endemic\n",
      "[37,38].\n",
      "\n",
      "Much of the low-income world, where malaria burden is highest and levels of urbanization are lowest, is set to undergo an urban and demographic transition in the coming decades\n",
      "[17,34], ultimately likely arriving at levels of urbanization similar to those exhibited by countries that eliminated malaria (Additional file\n",
      "8). Significant efforts towards modelling future malaria scenarios have been completed or are underway, focused principally on the effects of a variety of interventions\n",
      "[39-41] or on climate change scenarios\n",
      "[42-45], but the impacts of urbanization are rarely considered\n",
      "[46]. Yet, if the past century of malaria declines are indicative, the study of its impacts should receive more attention as nations start to monitor their progress toward elimination\n",
      "[47]. There exist multiple data gaps and uncertainties in obtaining suitable data to build urbanization into scenario models, however. Firstly, simple consistent definitions of what constitutes an urban area in general are difficult to outline. Beyond this, multiple approaches to mapping urban extents in a consistent fashion have been attempted (e.g.,\n",
      "[48-50]), but the spatial modelling of their future growth is lacking. Secondly, definitions and measures of urbanization that are relevant to understanding transmission patterns are poorly quantified, with only occasional attempts at empirical definitions made\n",
      "[8,9], and little consideration of the urban preferences of the dominant Anopheles species\n",
      "[51-53], or adaptation of them to urban environments\n",
      "[54,55]. Finally, the treatment of urban areas as single homogenous entities ignores the great variations in demographic, socio-economic and land uses within cities with, for example, urban agricultural practices often maintaining transmission within urban areas\n",
      "[56-58].\n",
      "\n",
      "The quantification of a global recession in the range and intensity of malaria over the 20th century has allowed us to review the impact that urbanization has had on these declines, and gauge its importance as a driver of future changes in malaria epidemiology. Results highlight for the first time the consistent relationship between urbanization and malaria declines over the past century globally, and point towards continuing declines as urbanization permanently alters the receptivity of many areas to malaria transmission. The findings presented here suggest that these trends will likely continue to catalyze malaria declines on the path to the goal of a malaria-free future.\n",
      "\n",
      "\n",
      "Competing interests\n",
      "\n",
      "The authors declare that they have no competing interests.\n",
      "\n",
      "\n",
      "Authors’ contributions\n",
      "\n",
      "AJT conceived the analyses, developed the study design and conducted the analyses. PWG, DLS and SIH provided the malaria endemicity data and mathematical models for dataset conversions. All authors contributed to the writing of the manuscript. All authors read and approved the final manuscript.\n",
      "\n",
      "\n",
      "Supplementary Material\n",
      "\n",
      "\n",
      "Additional file 1\n",
      "\n",
      "\n",
      "Urban extents and malaria transmission mapped for 1900 and 2000. Description: Maps of urban extents in 1900 and 2000 overlaid onto mapped areas of where malaria was eliminated over the past century and where it remains endemic today.\n",
      "\n",
      "Click here for file\n",
      "\n",
      "\n",
      "Additional file 2\n",
      "\n",
      "\n",
      "Change in urban extent between 1900 and 2000. Description: Map of estimated changes in urban extent globally between 1900 and 2000.\n",
      "\n",
      "Click here for file\n",
      "\n",
      "\n",
      "Additional file 3\n",
      "\n",
      "\n",
      "Malaria endemicity classes and urban areas in Brazil. Description: Maps of malaria endemicity classes and urban areas in Brazil for 1900 and 2000.\n",
      "\n",
      "Click here for file\n",
      "\n",
      "\n",
      "Additional file 4\n",
      "\n",
      "\n",
      "Change in malaria endemicity class between 1900 and 2007. Description: Map of changes in malaria endemicity class between 1900 and 2007.\n",
      "\n",
      "Click here for file\n",
      "\n",
      "\n",
      "Additional file 5\n",
      "\n",
      "\n",
      "The magnitude of decrease in P. falciparum basic reproductive number 1900-2007. Description: Map of the magnitude of decrease in P. falciparum basic reproductive number for 1900-2007.\n",
      "\n",
      "Click here for file\n",
      "\n",
      "\n",
      "Additional file 6\n",
      "\n",
      "\n",
      "The global spatial limits of Plasmodium vivax malaria transmission in 2009. Description: Map of the global spatial limits of Plasmodium vivax malaria transmission in 2009.\n",
      "\n",
      "Click here for file\n",
      "\n",
      "\n",
      "Additional file 7\n",
      "\n",
      "\n",
      "Dates of malaria elimination for those countries that achieved it. Description: Table showing the estimated and official dates of malaria elimination for those countries that achieved it.\n",
      "\n",
      "Click here for file\n",
      "\n",
      "\n",
      "Additional file 8\n",
      "\n",
      "\n",
      "Urbanization and countries that have eliminated malaria. Description: Description and results of statistical analyses of urbanization levels between countries that achieved elimination and those that remain endemic today.\n",
      "\n",
      "Click here for file\n",
      "\n",
      "\n",
      "Additional file 9\n",
      "\n",
      "\n",
      "Urbanization and sub-national malaria elimination. Description: Description and results of per-country statistical analyses of urbanization levels between regions that became malaria free and those that remain endemic today.\n",
      "\n",
      "Click here for file\n",
      "\n",
      "\n",
      "Additional file 10\n",
      "\n",
      "\n",
      "Results using the contemporary limits of Plasmodium vivax transmission. Description: Description and results of analyses repeated using Plasmodium vivax transmission limits data.\n",
      "\n",
      "Click here for file\n",
      "\n",
      "\n",
      "Additional file 11\n",
      "\n",
      "\n",
      "Urbanization and changes in malaria transmission. Description: Description and results of per-country statistical analyses of changes in malaria transmission intensity and urbanization levels.\n",
      "\n",
      "Click here for file\n"
     ]
    }
   ],
   "source": [
    "with open(sample(all_files, 1)[0], \"r\") as f:\n",
    "    xml = f.read()\n",
    "    article = Article(xml)\n",
    "    print(article.article_type())\n",
    "    if article.soup.body:\n",
    "        print(\"BODY\")\n",
    "        print(article.extract_text())\n",
    "    else:\n",
    "        print(\"NO BODY\")\n",
    "        print(article.extract_text())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load articles into a MongoDB collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "client = pymongo.MongoClient(\"localhost\", 27017)\n",
    "articles = client.pmc.articles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "seed(\"2019-04-04\")\n",
    "files = sample(all_files, 10000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processed 9950 articles (99.5% done).\n",
      "Elapsed: 519.1 sec\n",
      "Time Per All 9950: 0.052 sec    Est Remaining: 2.6 sec\n",
      "Time Per Last 50: 0.054 sec    Est Remaining: 2.7 sec\n",
      "Est Total (Mean): 521.7 sec    Est Total (This): 521.8 sec\n"
     ]
    }
   ],
   "source": [
    "articles.drop()\n",
    "reporter = Reporter(50, len(files))\n",
    "for idx, file in enumerate(files):\n",
    "    if idx is not 0 and idx % reporter.interval is 0:\n",
    "        reporter.report(idx)\n",
    "    with open(file, \"r\") as f:\n",
    "        xml = f.read()\n",
    "        article = Article(xml)\n",
    "        articles.insert_one({\n",
    "            \"_id\": article.pub_ids().get(\"pmc\"),  # Assuming they all have them\n",
    "            \"xml\": xml,\n",
    "            \"extracted_text\": article.extract_text()\n",
    "        })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_id': '3904948',\n",
       " 'xml': '<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD v1.0 20120330//EN\" \"JATS-archivearticle1.dtd\">\\n<article xmlns:xlink=\"http://www.w3.org/1999/xlink\" xmlns:mml=\"http://www.w3.org/1998/Math/MathML\" article-type=\"research-article\"><?properties open_access?><front><journal-meta><journal-id journal-id-type=\"nlm-ta\">PLoS One</journal-id><journal-id journal-id-type=\"iso-abbrev\">PLoS ONE</journal-id><journal-id journal-id-type=\"publisher-id\">plos</journal-id><journal-id journal-id-type=\"pmc\">plosone</journal-id><journal-title-group><journal-title>PLoS ONE</journal-title></journal-title-group><issn pub-type=\"epub\">1932-6203</issn><publisher><publisher-name>Public Library of Science</publisher-name><publisher-loc>San Francisco, USA</publisher-loc></publisher></journal-meta><article-meta><article-id pub-id-type=\"pmid\">24489775</article-id><article-id pub-id-type=\"pmc\">3904948</article-id><article-id pub-id-type=\"publisher-id\">PONE-D-13-42217</article-id><article-id pub-id-type=\"doi\">10.1371/journal.pone.0086718</article-id><article-categories><subj-group subj-group-type=\"heading\"><subject>Research Article</subject></subj-group><subj-group subj-group-type=\"Discipline-v2\"><subject>Biology</subject><subj-group><subject>Computational Biology</subject><subj-group><subject>Genomics</subject><subj-group><subject>Comparative Genomics</subject></subj-group></subj-group></subj-group><subj-group><subject>Evolutionary Biology</subject><subj-group><subject>Evolutionary Processes</subject><subj-group><subject>Adaptation</subject><subject>Evolutionary Selection</subject><subject>Mutation</subject><subject>Natural Selection</subject></subj-group></subj-group><subj-group><subject>Evolutionary Systematics</subject><subj-group><subject>Phylogenetics</subject></subj-group></subj-group><subj-group><subject>Comparative Genomics</subject><subject>Evolutionary Developmental Biology</subject></subj-group></subj-group><subj-group><subject>Genomics</subject><subj-group><subject>Comparative Genomics</subject></subj-group></subj-group></subj-group><subj-group subj-group-type=\"Discipline-v2\"><subject>Medicine</subject><subj-group><subject>Hematology</subject><subj-group><subject>Heme Synthesis</subject></subj-group></subj-group></subj-group></article-categories><title-group><article-title>Molecular Evolution of Multiple-Level Control of Heme Biosynthesis Pathway in Animal Kingdom</article-title><alt-title alt-title-type=\"running-head\">Heme Biosynthesis Pathway Evolution</alt-title></title-group><contrib-group><contrib contrib-type=\"author\"><name><surname>Tzou</surname><given-names>Wen-Shyong</given-names></name><xref ref-type=\"aff\" rid=\"aff1\">\\n<sup>1</sup>\\n</xref><xref ref-type=\"corresp\" rid=\"cor1\">\\n<sup>*</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Chu</surname><given-names>Ying</given-names></name><xref ref-type=\"aff\" rid=\"aff1\">\\n<sup>1</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Lin</surname><given-names>Tzung-Yi</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Hu</surname><given-names>Chin-Hwa</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Pai</surname><given-names>Tun-Wen</given-names></name><xref ref-type=\"aff\" rid=\"aff3\">\\n<sup>3</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Liu</surname><given-names>Hsin-Fu</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref><xref ref-type=\"aff\" rid=\"aff4\">\\n<sup>4</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Lin</surname><given-names>Han-Jia</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Cases</surname><given-names>Ildeofonso</given-names></name><xref ref-type=\"aff\" rid=\"aff5\">\\n<sup>5</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Rojas</surname><given-names>Ana</given-names></name><xref ref-type=\"aff\" rid=\"aff5\">\\n<sup>5</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Sanchez</surname><given-names>Mayka</given-names></name><xref ref-type=\"aff\" rid=\"aff6\">\\n<sup>6</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>You</surname><given-names>Zong-Ye</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref></contrib><contrib contrib-type=\"author\"><name><surname>Hsu</surname><given-names>Ming-Wei</given-names></name><xref ref-type=\"aff\" rid=\"aff2\">\\n<sup>2</sup>\\n</xref></contrib></contrib-group><aff id=\"aff1\">\\n<label>1</label>\\n<addr-line>Department of Life Sciences, National Taiwan Ocean University, Keelung, Taiwan</addr-line>\\n</aff><aff id=\"aff2\">\\n<label>2</label>\\n<addr-line>Institute of Bioscience and Biotechnology, National Taiwan Ocean University, Keelung, Taiwan</addr-line>\\n</aff><aff id=\"aff3\">\\n<label>3</label>\\n<addr-line>Department of Computer Science and Engineering, National Taiwan Ocean University, Keelung, Taiwan</addr-line>\\n</aff><aff id=\"aff4\">\\n<label>4</label>\\n<addr-line>Department of Medical Research, Mackay Memorial Hospital, Taipei, Taiwan</addr-line>\\n</aff><aff id=\"aff5\">\\n<label>5</label>\\n<addr-line>Computational Cell Biology Group, Institute of Predictive and Personalized Medicine of Cancer (IMPPC), Barcelona, Spain</addr-line>\\n</aff><aff id=\"aff6\">\\n<label>6</label>\\n<addr-line>Cancer and Iron Group, Institute of Predictive and Personalized Medicine of Cancer (IMPPC), Barcelona, Spain</addr-line>\\n</aff><contrib-group><contrib contrib-type=\"editor\"><name><surname>Missirlis</surname><given-names>Fanis</given-names></name><role>Editor</role><xref ref-type=\"aff\" rid=\"edit1\"/></contrib></contrib-group><aff id=\"edit1\">\\n<addr-line>CINVESTAV-IPN, Mexico</addr-line>\\n</aff><author-notes><corresp id=\"cor1\">* E-mail: <email>wstzou@ntou.edu.tw</email></corresp><fn fn-type=\"COI-statement\"><p><bold>Competing Interests: </bold>The authors have declared that no competing interests exist.</p></fn><fn fn-type=\"con\"><p>Conceived and designed the experiments: WST YC. Performed the experiments: YC TYL ZYY MWH. Analyzed the data: CHH TWP HFL HJL YC TYL ZYY MWH. Contributed reagents/materials/analysis tools: IC AR MS. Wrote the paper: WST YC.</p></fn></author-notes><pub-date pub-type=\"collection\"><year>2014</year></pub-date><pub-date pub-type=\"epub\"><day>28</day><month>1</month><year>2014</year></pub-date><volume>9</volume><issue>1</issue><elocation-id>e86718</elocation-id><history><date date-type=\"received\"><day>15</day><month>10</month><year>2013</year></date><date date-type=\"accepted\"><day>12</day><month>12</month><year>2013</year></date></history><permissions><copyright-statement>&#x000a9; 2014 Tzou et al</copyright-statement><copyright-year>2014</copyright-year><copyright-holder>Tzou et al</copyright-holder><license xlink:href=\"http://creativecommons.org/licenses/by/4.0/\"><license-p>This is an open-access article distributed under the terms of the Creative Commons Attribution License, which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are properly credited.</license-p></license></permissions><abstract><p>Adaptation of enzymes in a metabolic pathway can occur not only through changes in amino acid sequences but also through variations in transcriptional activation, mRNA splicing and mRNA translation. The heme biosynthesis pathway, a linear pathway comprised of eight consecutive enzymes in animals, provides researchers with ample information for multiple types of evolutionary analyses performed with respect to the position of each enzyme in the pathway. Through bioinformatics analysis, we found that the protein-coding sequences of all enzymes in this pathway are under strong purifying selection, from cnidarians to mammals. However, loose evolutionary constraints are observed for enzymes in which self-catalysis occurs. Through comparative genomics, we found that in animals, the first intron of the enzyme-encoding genes has been co-opted for transcriptional activation of the genes in this pathway. Organisms sense the cellular content of iron, and through iron-responsive elements in the 5&#x02032; untranslated regions of mRNAs and the intron-exon boundary regions of pathway genes, translational inhibition and exon choice in enzymes may be enabled, respectively. Pathway product (heme)-mediated negative feedback control can affect the transport of pathway enzymes into the mitochondria as well as the ubiquitin-mediated stability of enzymes. Remarkably, the positions of these controls on pathway activity are not ubiquitous but are biased towards the enzymes in the upstream portion of the pathway. We revealed that multiple-level controls on the activity of the heme biosynthesis pathway depend on the linear depth of the enzymes in the pathway, indicating a new strategy for discovering the molecular constraints that shape the evolution of a metabolic pathway.</p></abstract><funding-group><funding-statement>This work was supported by the National Science Council, Taiwan, R.O.C. (Grant Nos. NSC 95-2113-M-019-003, NSC 96-2627-B-019-002, NSC 97-2627-B-019-002, NSC 98-2627-B-019-002, NSC 98-2313-B-019-004-MY3, NSC 101-2311-B-019 -001, NSC 102-2627-B-019-002-, NSC 102-2633-B-019 -001 -) and the Center of Excellence for the Oceans, National Taiwan Ocean University. The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.</funding-statement></funding-group><counts><page-count count=\"15\"/></counts></article-meta></front><body><sec id=\"s1\"><title>Introduction</title><p>Molecular evolution has recently been a popular area of investigation, and through the advancement of technology and the maturation of analysis methods, this field continues to spawn important insights into the evolutionary processes affecting genes. None of these genes or their encoded proteins exists in isolation, and the products of genes construct the metabolic pathways and networks underlying the cellular and metabolic processes of organisms. As an increasing number of studies are describing the rates of protein and pathway evolution over evolutionary time, there are more opportunities to clarify the patterns and principles of natural selection acting on the pathways involved in the metabolic networks of organisms.</p><p>Research focusing on the effects of the organization of pathways on the strength of selection acting on individual proteins in these pathways has revealed various evolutionary patterns among proteins at different positions in a pathway. Important research has been conducted with respect to the plant anthocyanin biosynthetic pathway <xref rid=\"pone.0086718-Rausher1\" ref-type=\"bibr\">[1]</xref>&#x02013;<xref rid=\"pone.0086718-Rausher2\" ref-type=\"bibr\">[3]</xref>, terpenoid biosynthesis pathway <xref rid=\"pone.0086718-Ramsay1\" ref-type=\"bibr\">[4]</xref>, starch biosynthesis pathway <xref rid=\"pone.0086718-Yu1\" ref-type=\"bibr\">[5]</xref>, <xref rid=\"pone.0086718-Li1\" ref-type=\"bibr\">[6]</xref>, gibberellin biosynthesis pathway <xref rid=\"pone.0086718-Yang1\" ref-type=\"bibr\">[7]</xref> and carotenoid biosynthesis pathway <xref rid=\"pone.0086718-Clotault1\" ref-type=\"bibr\">[8]</xref> as well as lateral line innovation in teleosts <xref rid=\"pone.0086718-Philip1\" ref-type=\"bibr\">[9]</xref>, the glucosinolate pathway in <italic>Arabidopsis thaliana</italic>\\n<xref rid=\"pone.0086718-OlsonManning1\" ref-type=\"bibr\">[10]</xref> and the primate N-glycosylation pathway <xref rid=\"pone.0086718-Montanucci1\" ref-type=\"bibr\">[11]</xref>. It has been demonstrated that the pleiotropic genes in the upstream portions of pathways or those found at branch points in a network are subject to stronger selective constraints <xref rid=\"pone.0086718-Cork1\" ref-type=\"bibr\">[12]</xref>. On the other hand, selection is relaxed in the downstream enzymes, and nonsynonymous substitution rates as well as <italic>d<sub>N</sub>/d<sub>S</sub></italic> ratios are higher in these pathway components.</p><p>The heme biosynthesis pathway is an appropriate system not only for comparing the evolutionary rates of genes according to their position or pathway reticulation but also for studying functional motifs that may play a role at several levels of gene regulation. Heme acts as an essential cofactor for cytochromes, oxidases, peroxidases, catalases, hemoglobin and myoglobin in organisms. Heme acts as an iron-chelating tetrapyrrole and is composed of a complex macrocycle containing four pyrrolic rings connected by methine bridges in cyclic form. Heme also plays multiple regulatory roles, including microRNA processing, ion channel functions, circadian rhythms, mitochondrial targeting, translational regulation and protein degradation <xref rid=\"pone.0086718-Khan1\" ref-type=\"bibr\">[13]</xref>&#x02013;<xref rid=\"pone.0086718-Zhang1\" ref-type=\"bibr\">[16]</xref>. The heme biosynthesis pathway is an especially well-characterized and important pathway for erythroid production in animals, and malfunctions in heme biosynthesis result in several types of porphyrias because of the accumulation of toxic tetrapyrrole intermediates <xref rid=\"pone.0086718-Dailey1\" ref-type=\"bibr\">[17]</xref>. The heme biosynthesis pathway of animals is comprised of eight consecutive genes: 5-aminolevulinic acid synthase (<italic>ALAS</italic>), porphobilinogen synthase (<italic>PBGS</italic>), porphobilinogen deaminase (<italic>PBGD</italic>), uroporphyrinogen III synthase (<italic>UROS</italic>), uroporphyrinogen III decarboxylase (<italic>UROD</italic>), coproporphyrinogen III oxidase (<italic>CPO</italic>), protoporphyrinogen IX oxidase (<italic>PPO</italic>) and ferrochelatase (<italic>FECH</italic>), from positions one to eight (<xref ref-type=\"fig\" rid=\"pone-0086718-g001\">Figure 1</xref>). In animals, the formation of heme occurs in the mitochondria and cytosol. Heme biosynthesis begins with the formation of 5-aminolevulinic acid. In the Shemin pathway of animals, ALAS catalyzes a single-step reaction to condense glycine and succinyl-CoA into ALA, with the elimination of CO<sub>2</sub>. In vertebrates, the housekeeping isoforms of ALAS1 are expressed in all cell types, and ALAS2 is expressed in erythroid cells (eALAS) at a very high level. Subsequently, PBGS converts two 5-aminolevulinic acids into porphobilinogen, and the addition of four PBG substrates to a dipyrrole by PBGD results in the production of pre-uroporphyrinogen, also known as hydroxymethylbilane. A circularization reaction performed by UROS generates uroporphyrinogen III. The modification of side chains on the cyclic intermediates sequentially mediated by UROD, CPO and PPO results in protoporphyrin IX, which is subsequently chelated with ferrous iron by FECH (see reviews in <xref rid=\"pone.0086718-Heinemann1\" ref-type=\"bibr\">[14]</xref>, <xref rid=\"pone.0086718-Layer1\" ref-type=\"bibr\">[18]</xref>).</p><fig id=\"pone-0086718-g001\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g001</object-id><label>Figure 1</label><caption><title>Heme biosynthesis pathway in animals.</title><p>The substrate and product are indicated for each enzyme and the subcellular localization of each enzyme is also shown (cytosol or mitochondria). Each enzyme is coded from one to eight according to the linear order of the pathway. Also shown are the processes by which hydroxymethylbilane, the substrate of UROS, can be non-enzymatically cyclized to form uroporphyrinogen I, leading to uroporphyrin I or coproporphyrin I, and the process by which uroporphyrinogen III, the product of UROS, can be auto-oxidized to form uroporphyrin III. Protoporphyrinogen, the substrate of PPO, can be auto-oxidized to form protoporphyrin.</p></caption><graphic xlink:href=\"pone.0086718.g001\"/></fig><p>ALAS1 and ALAS2 are derived from gene duplication. A phylogenetic analysis of ALAS suggested that the relevant gene duplication event took place before the divergence of hagfish from the deuterostome line leading to vertebrates <xref rid=\"pone.0086718-Duncan1\" ref-type=\"bibr\">[19]</xref>. In extant vertebrate species, ALAS1 and ALAS2 are paralogs, and their amino acid sequences are highly similar to each other.</p><p>Regulation of the genes encoding the eight enzymes of the heme biosynthesis pathway can occur at the transcriptional, translational and post-translational levels. At the transcriptional level, multiple erythroid-specific factors have been observed to be involved in the transcriptional activation of several genes that participate in heme biosynthesis and erythropoiesis. Among these transcription factors, genomic DNA-binding activity and the conserved binding sites of KLF1, GATA1 and TAL1 have been studied in humans and mice <xref rid=\"pone.0086718-Kerenyi1\" ref-type=\"bibr\">[20]</xref>. The binding sites of KLF1 are located within the intergenic regions or introns (particularly the first intron) of genes encoding the components of the heme biosynthesis pathway. In a de novo motif study, GATA1 and TAL1 were hypothesized to complex with KLF1 in a small subset of erythroid <italic>cis</italic>-regulatory modules <xref rid=\"pone.0086718-Tallack1\" ref-type=\"bibr\">[21]</xref>. The majority of the GATA1-binding sites that mediate the activation of gene expression are close to the transcription start site, within either the first intron or the proximal 5&#x02032; flanking region <xref rid=\"pone.0086718-Cheng1\" ref-type=\"bibr\">[22]</xref>. TAL1-binding sites for the eight genes of the heme biosynthesis pathway are detected in either proximal promoter or intronic regions <xref rid=\"pone.0086718-Kassouf1\" ref-type=\"bibr\">[23]</xref>. A housekeeping promoter utilized in all tissue types exists in ALAS1, PBGS, PBGD, and UROS. However, for ALAS2, PBGS, PBGD and UROS, erythroid-specific promoters drive gene expression. More importantly, the alternative promoters found in PBGS, PBGD and UROS are located in intron 1, and an alternative splicing event is required for the transcript generated from the housekeeping promoters and erythroid-specific promoters <xref rid=\"pone.0086718-Ajioka1\" ref-type=\"bibr\">[24]</xref>. It has also been established that DNase-hypersensitive sites usually serve as sites for conserved and cell-type specific transcription factor binding and histone modification <xref rid=\"pone.0086718-Natarajan1\" ref-type=\"bibr\">[25]</xref>&#x02013;<xref rid=\"pone.0086718-Thurman1\" ref-type=\"bibr\">[27]</xref>. Combined analysis of intron 1 sequences and DNase-hypersensitive sites will shed light on the regulatory potential of the corresponding <italic>cis</italic>-elements and the evolution of the heme biosynthesis pathway.</p><p>At the translational level, IRE-binding protein (IRP) interacts with the iron-responsive element (IRE) located in the 5&#x02032; untranslated regions (5&#x02032;UTRs) of ALAS2 and ferritin mRNA to inhibit protein translation and in the 3&#x02032; untranslated region (3&#x02032;UTR) of TfR1 to stabilize mRNA. During iron repletion, iron-sulfur clusters can abolish the IRE-binding ability of IRP1, and the F-box protein FBXL5 recognizes IRP2 and targets it for degradation by E3 ligases. Both mechanisms offer a link between iron availability and heme synthesis <xref rid=\"pone.0086718-Evstatiev1\" ref-type=\"bibr\">[28]</xref>&#x02013;<xref rid=\"pone.0086718-Chen1\" ref-type=\"bibr\">[30]</xref>. At the post-translational level, the binding of heme to the heme-regulatory motif (HRM) in the mature ALAS1 peptide blocks mitochondrial import and results in end-product inhibition <xref rid=\"pone.0086718-Duncan1\" ref-type=\"bibr\">[19]</xref>, <xref rid=\"pone.0086718-Lathrop1\" ref-type=\"bibr\">[31]</xref>&#x02013;<xref rid=\"pone.0086718-Dailey2\" ref-type=\"bibr\">[33]</xref>.</p><p>The objectives of this study are to investigate the molecular fingerprints underlying the adaptation of the eight genes encoding the enzymes of the heme biosynthesis pathway in Kingdom Metazoa and, particularly, how these adaptations correlate with the positions of the gene products (enzymes) in the pathway. First, through the analysis of the evolutionary constraints on the protein-coding sequences, we reveal strong purifying selection on the protein sequences and clade-specific adaptations in teleosts and arthropods. We also demonstrate that the first introns of these genes in vertebrates play a role in their erythroid-specific transactivation, highlighting the emergence of erythrocytes in animals. Then, we show that pathway product (heme) feedback control is widely utilized in this pathway. The evolutionarily conserved motifs that enable this control include HRMs found not only in ALAS but also in PBGS. We also discover IREs within 5&#x02032;UTRs and exon-intron boundaries, raising the possibility of regulation of both translation and splicing choice, respectively. We conclude by summarizing the evolution of multifarious controls on the heme biosynthesis pathway.</p></sec><sec id=\"s2\"><title>Results</title><sec id=\"s2a\"><title>Relationship between Selection Pressure and Pathway Position</title><p>Based on the M0 model, assuming a constant evolutionary rate (nonsynonymous versus synonymous rate, <italic>d<sub>N</sub>/d<sub>S</sub></italic>, &#x003c9;) for all branches and all codons of the eight genes encoding the enzymes of the metazoan heme biosynthesis pathway, the &#x003c9; values vary from 0.041 (FECH) to 0.127 (UROS), providing evidence that the sequences of the coding regions of these genes are under negative selection (<xref ref-type=\"fig\" rid=\"pone-0086718-g002\">Figure 2A</xref>). We also evaluated the M1a model, which assumes that there are two groups of codons, subject to purifying selection and neutral evolution, in contrast to the M0 model. We found that the M1a model significantly improves the model (<italic>p</italic>&#x0003c;0.001) and that a large fraction of the codons are under purifying selection (&#x0003e;91% in all genes, except for PPO, where the obtained value was 79%). Notably, the &#x003c9; values at positions four (UROS) and seven (PPO) were highest (0.116 and 0.105, respectively) among the genes of the heme biosynthesis pathway. (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s004\">Table S1</xref>).</p><fig id=\"pone-0086718-g002\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g002</object-id><label>Figure 2</label><caption><title>Selection pressure on heme biosynthesis genes in animals.</title><p>&#x003c9; values (<italic>d<sub>N</sub>/d<sub>S</sub></italic>) were estimated with the M0 model for the eight heme biosynthesis genes in animals (A). The distribution of &#x003c9; values (B), the nonsynonymous substitution rate, <italic>d<sub>N</sub></italic> (C), and the synonymous substitution rate, <italic>d<sub>S</sub></italic> (D). The order of genes follows the linear order of their pathway positions (<xref ref-type=\"fig\" rid=\"pone-0086718-g001\">Figure 1</xref>).</p></caption><graphic xlink:href=\"pone.0086718.g002\"/></fig><p>To determine whether the variation in &#x003c9; values among the genes was statistically significant and showed a relationship with their positions in the pathway, we conducted several tests on the <italic>d<sub>N</sub></italic>, <italic>d<sub>S</sub></italic> and &#x003c9; values obtained for each gene (<xref ref-type=\"fig\" rid=\"pone-0086718-g002\">Figure 2B, 2C, 2D</xref>). First, we determined whether the distributions of the <italic>d<sub>N</sub></italic>, <italic>d<sub>S</sub></italic> and &#x003c9; values were correlated. We found that the distributions of the <italic>d<sub>N</sub></italic>, <italic>d<sub>S</sub></italic> and &#x003c9; values were not correlated for each gene (Kruskal-Wallis rank sum test, <italic>p</italic>&#x0003c;0.0001). Second, the &#x003c9; values of genes located at positions four (UROS) and seven (PPO) were shown to be significantly higher than for the genes located at other positions (Wilcoxon rank sum test, P&#x0003c;0.0001). Third, to determine the significance of the variations in &#x003c9; values among the genes, we conducted multiple comparisons of the &#x003c9; values for each gene pair (see Methods). Three groups were established: group 1 (low &#x003c9; values), consisting of FECH, PBGD and CPO; group 2 (intermediate &#x003c9; values), consisting of ALAS, UROD and PBGS; and group 3 (high &#x003c9; values), consisting of PPO and UROS (<xref ref-type=\"table\" rid=\"pone-0086718-t001\">Table 1</xref>). Fourth, we determined whether the <italic>d<sub>N</sub></italic>, <italic>d<sub>S</sub></italic> and &#x003c9; values were correlated with the pathway positions of the eight enzymes of the heme biosynthesis pathway. It was found that the &#x003c9; values were positively correlated with the pathway positions of the enzymes (Kendall&#x02019;s correlation test: tau&#x0200a;=&#x0200a;0.2202, <italic>P</italic>&#x0003c;0.0001), as were the <italic>d<sub>N</sub></italic> values (Kendall&#x02019;s correlation test: tau&#x0200a;=&#x0200a;0.1625, <italic>P</italic>&#x0003c;0.0001). However, the <italic>d<sub>S</sub></italic> values were not correlated with pathway position (Kendall&#x02019;s correlation test: tau&#x0200a;=&#x0200a;&#x02212;0.0372, <italic>P&#x0200a;=&#x0200a;</italic>0.0243).</p><table-wrap id=\"pone-0086718-t001\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t001</object-id><label>Table 1</label><caption><title>Comparison of &#x003c9; values among eight genes to test the significance of the &#x003c9; variations among genes.</title></caption><alternatives><graphic id=\"pone-0086718-t001-1\" xlink:href=\"pone.0086718.t001\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td colspan=\"3\" align=\"left\" rowspan=\"1\">Group 1</td><td colspan=\"3\" align=\"left\" rowspan=\"1\">Group 2</td><td colspan=\"2\" align=\"left\" rowspan=\"1\">Group 3</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Group 1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Group 2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Group 3</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt101\"><label/><p>For each comparison, a pair of genes was chosen. &#x003c9; value of the gene in the row is constrained to the average value of omega values from the genes in the rows and columns. The difference in the likelihood between the null model M0 and the constrained model was obtained. The significance level is labeled * if <italic>p</italic>&#x0003c;0.05 and ** if <italic>p</italic>&#x0003c;0.01.</p></fn></table-wrap-foot></table-wrap><p>In conclusion, the variations in &#x003c9; and <italic>d<sub>N</sub></italic> values were shown to be correlated with the pathway positions of the eight enzymes of the heme biosynthesis pathway.</p></sec><sec id=\"s2b\"><title>Amino Acid Residues under Positive Selection</title><p>When the genes of the heme biosynthesis pathway are observed to be under strong purifying selection, it is interesting to examine whether there are amino acids in certain lineages that experience positive selection. Due to limitations regarding the number of sequences collected, we focused on the protein sequences of the mammal, teleost and arthropod subgroups. Of the eight enzymes involved in heme biosynthesis, only ALAS2 from teleosts, PBGS from arthropods and UROD from teleosts showed positively selected residues in the branch-site model (<italic>p</italic> value &#x0003c;0.05) (<xref ref-type=\"table\" rid=\"pone-0086718-t002\">Table 2</xref>). In ALAS2 from teleosts, five sites were found to be positively selected (model A/A1, <italic>p</italic>&#x0003c;2.3e-10) (BEB &#x0003e;0.978) (<xref ref-type=\"table\" rid=\"pone-0086718-t003\">Table 3</xref>). Notably, while the amino acid detected in teleost ALAS2 at position 204 is I/M/T, the amino acid E is present at this position in all other vertebrate ALAS1 proteins. We mapped this amino acid in the crystal structure of the protein and found that it is positioned at the ALAS dimer interface. The amino acid R located at position 353 in teleost ALAS2 is also interesting because a K residue is found at this position in all other vertebrate ALAS1, ALAS2 and ALAS proteins. The crystal structure of ALAS indicates that this amino acid forms a hydrogen bond with the ribose-O3&#x02019; of the substrate succinyl-CoA <xref rid=\"pone.0086718-Astner1\" ref-type=\"bibr\">[34]</xref> (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s001\">Figure S1A</xref>).</p><table-wrap id=\"pone-0086718-t002\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t002</object-id><label>Table 2</label><caption><title>Selection pressure of genes of heme-biosynthesis pathway by employing the branch model.</title></caption><alternatives><graphic id=\"pone-0086718-t002-2\" xlink:href=\"pone.0086718.t002\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Model</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Model</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Estimates of parameters</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2&#x00394;L<xref ref-type=\"table-fn\" rid=\"nt102\">a</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">\\n<italic>p</italic> Value<xref ref-type=\"table-fn\" rid=\"nt103\">b</xref>\\n</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS+ALAS1+ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">M0: one ratio</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;&#x0200a;=&#x0200a;0.05356</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost ALAS2 branch</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">two ratios</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;0&#x0200a;=&#x0200a;0.05323, &#x003c9;&#x0200a;=&#x0200a;8.04979</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">10.087074</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1.49E-03</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">M0: one ratio</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;&#x0200a;=&#x0200a;0.06398</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod branch</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">two ratios</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;0&#x0200a;=&#x0200a;0.06398, &#x003c9;&#x0200a;=&#x0200a;999.000</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">4.478516</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3.43E-02</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">M0: one ratio</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;&#x0200a;=&#x0200a;0.06194</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost branch</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">two ratios</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">&#x003c9;0&#x0200a;=&#x0200a;0.06135, &#x003c9;&#x0200a;=&#x0200a;13.69790</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">6.01671</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1.42E-02</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt102\"><label>a</label><p>Twice the difference between the log likelihood of M0 (one ratio) and two ratio model.</p></fn><fn id=\"nt103\"><label>b</label><p>\\n<italic>p</italic>(&#x003c7;2) of the likelihood ratio test.</p></fn></table-wrap-foot></table-wrap><table-wrap id=\"pone-0086718-t003\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t003</object-id><label>Table 3</label><caption><title>Positive selection in the different lineages of genes of heme biosynthesis pathway.</title></caption><alternatives><graphic id=\"pone-0086718-t003-3\" xlink:href=\"pone.0086718.t003\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Foreground branches</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2&#x00394;L<xref ref-type=\"table-fn\" rid=\"nt104\">a</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">\\n<italic>p</italic> Value<xref ref-type=\"table-fn\" rid=\"nt105\">b</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Estimates of the parameters in the modified model A<xref ref-type=\"table-fn\" rid=\"nt106\">c</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Positively selected sites<xref ref-type=\"table-fn\" rid=\"nt107\">d</xref>\\n</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">40.20213</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2.29E-10</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">p0&#x0200a;=&#x0200a;0.86464, p1&#x0200a;=&#x0200a;0.07816, p2a&#x0200a;=&#x0200a;0.05246,</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">204E, 243K, 352P, 353K,</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">p2b&#x0200a;=&#x0200a;0.00474, &#x003c9;0&#x0200a;=&#x0200a;0.04595, &#x003c9;2&#x0200a;=&#x0200a;999.0</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">431G</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">23.97615</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">9.75E-07</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">p0&#x0200a;=&#x0200a;0.86020, p1&#x0200a;=&#x0200a;0.03447, p2a&#x0200a;=&#x0200a;0.10127,</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">11Y, 106H, 162C, 195S,</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">p2b&#x0200a;=&#x0200a;0.00406, &#x003c9;0&#x0200a;=&#x0200a;0.06383, &#x003c9;2&#x0200a;=&#x0200a;999.0</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">267K, 274A, 314I</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">25.60371</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">4.19E-07</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">p0&#x0200a;=&#x0200a;0.87046, p1&#x0200a;=&#x0200a;0.07295, p2a&#x0200a;=&#x0200a;0.05222,</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">136Q, 174M, 290K, 297K,</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">p2b&#x0200a;=&#x0200a;0.00438, &#x003c9;0&#x0200a;=&#x0200a;0.06761, &#x003c9;2&#x0200a;=&#x0200a;999.0</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">300T, 316E, 349H</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt104\"><label>a</label><p>Twice the difference between the log likelihood of M0 (one ratio) and two ratio model.</p></fn><fn id=\"nt105\"><label>b</label><p>\\n<italic>p</italic>-value <italic>p</italic>(&#x003c7;2) of the likelihood ratio test.</p></fn><fn id=\"nt106\"><label>c</label><p>For the branch-site model A, the following four classes are demarcated to each amino acid: class 0 with 0&#x0003c; &#x003c9;0&#x0003c;1 in all branches; class 1 with &#x003c9;1&#x0200a;=&#x0200a;1 in all branches; class 2a with foreground &#x003c9;2&#x0003e;1 but background 0&#x0003c; &#x003c9;0&#x0003c;1; and class 2b with foreground &#x003c9;2&#x0003e;1 but background &#x003c9;1&#x0200a;=&#x0200a;1. p0 is the proportion of codons with 0&#x0003c; &#x003c9; &#x0003c;1 in class 0; p1 is the proportion of codons with &#x003c9;1&#x0200a;=&#x0200a;1 in class 1; p2a is the proportion of codons with foreground &#x003c9;2&#x0003e;1 but with background &#x003c9;0 in class 2a; p2b is the proportion of codons with foreground &#x003c9;2&#x0003e;1 but with background &#x003c9;1&#x0200a;=&#x0200a;1 in class 2b.</p></fn><fn id=\"nt107\"><label>d</label><p>Bayes empirical Bayes (BEB) is used to calculate the posterior probabilities to identify the sites (amino acid residues) under positive selection (higher than 95%). The sites are indexed by the amino acids of the site in the human sequence (ALAS1, PBGS, UROD).</p></fn></table-wrap-foot></table-wrap><p>In PBGS from arthropods, seven sites are positively selected (model A/A1, <italic>p</italic>&#x0003c;9.8e-7) (BEB &#x0003e;0.964) (<xref ref-type=\"table\" rid=\"pone-0086718-t003\">Table 3</xref>). The amino acid found in arthropod PBGS at position 11 is I/M, whereas F/Y is present at this position in all vertebrate and cnidarian PBGS proteins. We mapped this amino acid in the crystal structure and found that it is positioned at the PBGS dimer interface and is very close to the Mg<sup>2+</sup> binding site, which has been hypothesized to trigger the conversion of PBGS between its open and closed forms <xref rid=\"pone.0086718-Frankenberg1\" ref-type=\"bibr\">[35]</xref>. The amino acid A is present at position 195 in <italic>Drosophila</italic> PBGS, whereas the amino acid at this position is S in all other animal PBGS proteins. The crystal structure indicates that this amino acid forms the active site pocket <xref rid=\"pone.0086718-Frankenberg1\" ref-type=\"bibr\">[35]</xref> (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s001\">Figure S1B</xref>).</p><p>In UROD from teleosts, seven sites were observed to be positively selected (model A/A1, <italic>p</italic>&#x0003c;4.2e-7) (BEB &#x0003e;0.981) (<xref ref-type=\"table\" rid=\"pone-0086718-t003\">Table 3</xref>). The amino acid S is present at position 290 in teleost UROD, whereas D/E/K is found at this position in other vertebrate, cnidarian and arthropod UROD proteins. The amino acid at position 316 in the teleost UROD is H/R/S, whereas the amino acid at this position is D/E/K in other vertebrate, chordate, arthropod and cnidarian UROD proteins. We mapped both amino acid residues in the crystal structure and found that they are positioned at the UROD dimer interface <xref rid=\"pone.0086718-Phillips1\" ref-type=\"bibr\">[36]</xref> (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s001\">Figure S1C</xref>).</p><p>In summary, we discovered several amino acid residues under positive selection. Some of them are located at the active sites and dimer interfaces of biologically functional enzymes of the heme biosynthesis pathway.</p></sec><sec id=\"s2c\"><title>Detection of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences</title><p>It was previously reported that intron 1 sequences contain <italic>cis</italic>-elements that are necessary for the transcriptional activation of human <italic>PBGS</italic>\\n<xref rid=\"pone.0086718-Kaya1\" ref-type=\"bibr\">[37]</xref>, <italic>PBGD</italic>\\n<xref rid=\"pone.0086718-Mignotte1\" ref-type=\"bibr\">[38]</xref>&#x02013;<xref rid=\"pone.0086718-Grandchamp1\" ref-type=\"bibr\">[40]</xref> and <italic>UROS</italic>\\n<xref rid=\"pone.0086718-Aizencang1\" ref-type=\"bibr\">[41]</xref>, <xref rid=\"pone.0086718-Aizencang2\" ref-type=\"bibr\">[42]</xref>. Therefore, we investigated the selection acting on intron sequences by searching for DNA sequences that are both evolutionarily conserved across 46 vertebrates and located in DNase-hypersensitive site clusters (ENCyclopedia of DNA Elements, ENCODE, for human). We found considerable stretches of DNA longer than 40 nucleotides in intron 1 of <italic>ALAS2</italic>, <italic>PBGS</italic>, <italic>PBGD</italic>, <italic>UROS</italic>, <italic>UROD</italic> and <italic>FECH</italic> that could serve as evolutionarily conserved DNase-hypersensitive sites in host genes (<xref ref-type=\"fig\" rid=\"pone-0086718-g003\">Figure 3</xref>, <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s005\">Table S2</xref>). However, evolutionarily conserved DNase-hypersensitive sites were also found in other intron regions, including in intron 3 of <italic>ALAS1</italic>, intron 8 of <italic>ALAS2</italic>, and introns 2 and 6 of <italic>FECH</italic>. We did not find considerable lengths of evolutionarily conserved DNase-hypersensitive sites in the intron regions of <italic>CPO</italic> and <italic>PPO</italic>. In summary, we revealed evolutionarily conserved DNase-hypersensitive sites in intron regions in six of the eight genes of the heme biosynthesis pathway, implying that conserved regulatory mechanisms acting on intron sequences might be involved in the transactivation of gene expression in the vertebrate heme biosynthesis pathway.</p><fig id=\"pone-0086718-g003\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g003</object-id><label>Figure 3</label><caption><title>Three-dimensional view of the evolutionarily conserved DNase-hypersensitive sites in intron sequences.</title><p>For each gene in the biosynthesis pathway (ALAS1 and ALAS2 are treated separately because they are different genes located on different chromosomes), the length of the intersection of the DNA sequence that is evolutionarily conserved across vertebrates and DNase-hypersensitive sites is indicated on the z-axis. The intron ID is provided on the x-axis. Genes from ALAS1 to FECH are shown on the y-axis and are coded from one to eight according to the linear order of the pathway. (<xref ref-type=\"fig\" rid=\"pone-0086718-g001\">Figure 1</xref>).</p></caption><graphic xlink:href=\"pone.0086718.g003\"/></fig></sec><sec id=\"s2d\"><title>Distribution of IREs in Exon and Intron Sequences</title><p>IREs have been reported to exist in the 5&#x02032; and 3&#x02032;UTR sequences of mRNAs, through which they control the translational efficiency and stability of transcripts, respectively. The 5&#x02032;UTR of human ALAS2 has been shown to contain an IRE. We conducted a survey of the 5&#x02032;UTR sequences of genes of the heme biosynthesis pathway (<xref ref-type=\"table\" rid=\"pone-0086718-t004\">Tables 4</xref> and <xref ref-type=\"table\" rid=\"pone-0086718-t005\">5</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s006\">Table S3</xref>). Notably, the 5&#x02032;UTRs of nine of the thirteen collected vertebrate ALAS2 sequences contained IREs of high quality (see Methods). Intriguingly, we also detected IREs in the 5&#x02032;UTRs of teleost ALAS1 sequences and the ALAS sequences of one chordate (tunicate) and the purple sea urchin (echinoderm) that were of high quality as well as in a sea anemone (cnidarian) and honey bee (arthropod) that were of medium quality. These findings constitute the first demonstration that IREs can be identified in the 5&#x02032;UTRs of ALAS1 sequences from vertebrates and ALAS sequences from arthropods, echinoderms and cnidarians.</p><table-wrap id=\"pone-0086718-t004\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t004</object-id><label>Table 4</label><caption><title>IRE in 5&#x02032;UTR.</title></caption><alternatives><graphic id=\"pone-0086718-t004-4\" xlink:href=\"pone.0086718.t004\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Gene</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Mammal</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Bird,Reptile,Amphibian</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Chordate</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Echinoderm</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Cnidaria</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(0)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt108\"><label/><p>Number of species containing potential IRE in 5&#x02032;UTR region of genes involved in heme-biosynthesis pathway.</p></fn><fn id=\"nt109\"><label/><p>Number inside the parenthesis is the total number of 5&#x02032;UTR sequences under investigation.</p></fn><fn id=\"nt110\"><label/><p>NF: Not found.</p></fn></table-wrap-foot></table-wrap><table-wrap id=\"pone-0086718-t005\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t005</object-id><label>Table 5</label><caption><title>IRE in intron.</title></caption><alternatives><graphic id=\"pone-0086718-t005-5\" xlink:href=\"pone.0086718.t005\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Gene</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Mammal</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Bird,Reptile,Amphibian</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Chordate</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Echinoderm</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Cnidaria</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">4(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt111\"><label/><p>Number of species containing potential IRE in intron region of genes involved in heme biosynthesis pathway.</p></fn><fn id=\"nt112\"><label/><p>Number inside the parenthesis is the total number of intron sequences under investigation.</p></fn><fn id=\"nt113\"><label/><p>NF: Not found.</p></fn></table-wrap-foot></table-wrap><p>IREs were also found in the 5&#x02032;UTRs of the PBGD sequences of <italic>Drosophila melanogaster</italic> (fruit fly) and <italic>Apis mellifera</italic> (honey bee) (high quality) as well as that of <italic>Strongylocentrotus purpuratus</italic> (sea urchin) (medium quality) (<xref ref-type=\"table\" rid=\"pone-0086718-t004\">Tables 4</xref> and <xref ref-type=\"table\" rid=\"pone-0086718-t005\">5</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s006\">Table S3</xref>). This observation raised the possibility that the translation of genes other than ALAS in the heme biosynthesis pathway could also respond to iron availability.</p><p>We also found IREs in intron regions. In particular, several potential IREs (of medium quality) exist at the intron-exon boundaries of PBGS from <italic>Mus musculus</italic> (mouse), PBGD from <italic>Loxodonta africana</italic> (elephant) and <italic>Oryctolagus cuniculus</italic> (rabbit) and UROS from <italic>Loxodonta africana</italic> (elephant) and <italic>Mus musculus</italic> (mouse) (<xref ref-type=\"table\" rid=\"pone-0086718-t004\">Tables 4</xref> and <xref ref-type=\"table\" rid=\"pone-0086718-t005\">5</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s006\">Table S3</xref>). This class of IREs forms stem and loop regions that overlap with protein-coding and intron sequences (<xref ref-type=\"fig\" rid=\"pone-0086718-g004\">Figure 4</xref> and <xref ref-type=\"fig\" rid=\"pone-0086718-g005\">Figure 5</xref>). Based on these observations, we conducted a survey to identify potential IREs in the intron-exon boundary sequences of genes in the human and zebrafish genomes. We found 21 and 12 potential IREs of high quality at the intron-exon boundaries of genes in the human and zebrafish genomes, respectively (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s007\">Table S4</xref>).</p><fig id=\"pone-0086718-g004\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g004</object-id><label>Figure 4</label><caption><title>Potential iron-responsive elements (IREs) in the introns and intron-exon boundaries of UROS genes.</title><p>IREs depicted as stem-loop structures are shown in the corresponding intron regions. UROS exon and intron IDs from four species are indicated. The conserved splicing acceptor site AG and the unpaired nucleotide of the IRE structure are also shown.</p></caption><graphic xlink:href=\"pone.0086718.g004\"/></fig><fig id=\"pone-0086718-g005\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g005</object-id><label>Figure 5</label><caption><title>Sequence alignment of the IREs at the intron-exon boundaries of UROS from four species.</title><p>&#x0201c;&#x0003e;&#x0201d; and &#x0201c;&#x0003c;&#x0201d; represent the base pairing of the RNA secondary structure. The potential IRE consensus loop sequence, CAGUGN, and the unpaired nucleotide G are also shown with respect to the location of the IRE hairpin. The intron-exon boundary is indicated as |.</p></caption><graphic xlink:href=\"pone.0086718.g005\"/></fig><p>In summary, we detected potential IREs not only in the 5&#x02032;UTR of ALAS2 but also in the 5&#x02032;UTRs of ALAS and ALAS1. More intriguingly, several potentially conserved IREs might exist at the intron-exon boundaries of genes involved in the heme biosynthesis pathway.</p></sec><sec id=\"s2e\"><title>Distribution of HRMs in Protein Sequences</title><p>The HRM (denoted HRM_t, see Methods) has been shown to bind heme to inhibit the import of ALAS1 into mitochondria. Here, we collected recently published HRM sequences and compiled them into a new HRM consensus sequence (denoted HRM_r, see Methods). Some of the HRM_r sequences are predicted to sense the redox state of the cell and may be critical for triggering the degradation of proteins containing the HRM sequence.</p><p>Only ALAS1, ALAS2 and PBGD protein sequences contain HRM_t. We found HRM_t sequences in the ALAS sequences of chordates, echinoderms, and cnidarians. The ALAS1 genes of all vertebrate species show HRM_t sequences at the N-terminus. In ALAS2, HRM_t sequences were found in several types of mammals and in chickens and Xenopus (<xref ref-type=\"table\" rid=\"pone-0086718-t006\">Tables 6</xref> and <xref ref-type=\"table\" rid=\"pone-0086718-t007\">7</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s008\">Table S5</xref>).</p><table-wrap id=\"pone-0086718-t006\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t006</object-id><label>Table 6</label><caption><title>HRM_t in protein sequence.</title></caption><alternatives><graphic id=\"pone-0086718-t006-6\" xlink:href=\"pone.0086718.t006\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Gene</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Mammal</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Bird,Reptile,Amphibian</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Chordate</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Echinoderm</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Cnidaria</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">7(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">6(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">6(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">4(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt114\"><label/><p>Number of protein species containing potential HRM_t of genes involved in heme-biosyntheis pathway.</p></fn><fn id=\"nt115\"><label/><p>Number inside the parenthesis is the total number of protein sequences under investigation.</p></fn><fn id=\"nt116\"><label/><p>NF: Not found.</p></fn></table-wrap-foot></table-wrap><table-wrap id=\"pone-0086718-t007\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t007</object-id><label>Table 7</label><caption><title>HRM_r in protein sequence.</title></caption><alternatives><graphic id=\"pone-0086718-t007-7\" xlink:href=\"pone.0086718.t007\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Gene</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Mammal</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Bird,Reptile,Amphibian</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Teleost</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Chordate</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Echinoderm</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Arthropod</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Cnidaria</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">6(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">NF</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(6)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(5)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(4)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(3)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(1)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(7)</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">(2)</td></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt117\"><label/><p>Number of protein species containing potential HRM_r of genes in heme biosynthesis pathway.</p></fn><fn id=\"nt118\"><label/><p>Number inside the parenthesis is the total number of protein sequences under investigation.</p></fn><fn id=\"nt119\"><label/><p>NF: Not found.</p></fn></table-wrap-foot></table-wrap><p>Notably, when we used HRM_r as the search sequence for HRMs, new HRM sequences were identified in PBGS (vertebrates, arthropods, and cnidarians) and PBGD (<italic>Drosophila</italic>). This class of HRM is not restricted to the N-terminus of the protein sequence and is conserved with respect to its position in the amino acid sequence (<xref ref-type=\"fig\" rid=\"pone-0086718-g006\">Figure 6</xref>).</p><fig id=\"pone-0086718-g006\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.g006</object-id><label>Figure 6</label><caption><title>Heme-regulatory motifs (HRMs) in PBGS and PBGD.</title><p>Multiple sequence alignments of PBGS (A) and PBGD (B) are shown, with HRM_t and HRM_r colored orange and green, respectively. Amino acid numbers for HRM_t and HRM_r are also shown according to the first protein sequence in the alignment.</p></caption><graphic xlink:href=\"pone.0086718.g006\"/></fig></sec></sec><sec id=\"s3\"><title>Discussion</title><p>Elucidation of the evolutionary history of biological pathways sheds light on the principles underlying the evolutionary forces acting on organisms in the environment. The history of pathway evolution may vary among different pathways and ancestral organisms, but understanding the underlying principles helps to reveal the modifications that have taken place in the physiological processes of organisms during their evolutionary history.</p><p>Methods allowing the detection of selection pressure in the evolutionary histories of pathways and networks have made it possible to investigate the existence of fundamental principles driving selection in nature. By estimating the ratio of the nonsynonymous to the synonymous substitution rate for individual protein-encoding genes (&#x003c9;), the types of selection pressure acting on a gene can be identified. In addition to the properties of proteins themselves, the regulatory mechanisms acting on genes are important in metabolic pathways and are also exposed to selection during evolutionary processes. A novel regulatory mechanism affecting the genes in a metabolic pathway can give rise to a new gene function that may become fixed in a lineage, through which we can determine the evolutionary history of a particular biosynthesis pathway throughout different lineages of organisms. To clarify the evolutionary history of the heme biosynthetic pathway of animals, we analyzed the three previously reported regulatory mechanisms related to genes in the heme biosynthesis pathway, which involve DNase I-hypersensitive sites, IREs, and HRMs. Our <italic>in silico</italic> prediction results showed that multiple regulatory mechanisms may exist for the genes in the heme biosynthesis pathway.</p><sec id=\"s3a\"><title>Stronger Selection at the Middle and Penultimate Positions of the Pathway could Result from Self-catalysis</title><p>We conducted a phylogenetic analysis of the eight genes of the heme biosynthesis pathway found in the animal kingdom. The &#x003c9; values at positions four (UROS) and seven (PPO) were shown to be significantly higher than at the other positions in the pathway. Notably, some self-catalysis and by-products of the heme biosynthesis pathway have been reported. Hydroxymethylbilane, the substrate of UROS, can be non-enzymatically cyclized to form uroporphyrinogen I, a useless by-product that leads to uroporphyrin I or coproporphyrin I <xref rid=\"pone.0086718-Bloomer1\" ref-type=\"bibr\">[43]</xref>. The product of UROS can also be auto-oxidized to uroporphyrin III. Protoporphyrinogen, the substrate of PPO, has been shown to be auto-oxidized to protoporphyrin in air, without the need for PPO <xref rid=\"pone.0086718-Bloomer1\" ref-type=\"bibr\">[43]</xref>, <xref rid=\"pone.0086718-Brenner1\" ref-type=\"bibr\">[44]</xref>. The degree of such &#x0201c;leaking&#x0201d; among the biochemical reactions involved in the heme biosynthesis pathway is unknown, and whether this phenomenon is common in animals remains to be determined. We speculate that &#x0201c;leaking&#x0201d; in a biochemical reaction would impact selection pressure and, most likely, lead to decreased evolutionary constraint.</p></sec><sec id=\"s3b\"><title>Biological Function of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences</title><p>Based on the chromatin immunoprecipitation with high-throughput sequencing (ChIP-seq) technique, the transcription factor-binding module GATA1-KLF1 has been hypothesized to allow erythrocyte-specific activation of gene expression through binding to the intron 1 sequences of <italic>ALAS2</italic>, <italic>PBGS</italic>, <italic>PBGD</italic> and <italic>UROS</italic>\\n<xref rid=\"pone.0086718-Tallack1\" ref-type=\"bibr\">[21]</xref>, <xref rid=\"pone.0086718-Cheng1\" ref-type=\"bibr\">[22]</xref>. In this study, we identified intron 1 sequences as a type of DNase-hypersensitive site that is conserved in vertebrates among several genes, including <italic>ALAS2</italic>, <italic>PBGS</italic>, <italic>PBGD</italic>, <italic>UROS</italic> and <italic>UROD</italic>. This finding suggests that in response to the high demand for heme in the erythrocytes of the common ancestor of vertebrates, several transcription factors were recruited to bind to the first intron region to trans-activate the expression of the first five genes of the heme biosynthesis pathway. Notably, an intronic enhancer has been identified in intron 8 of <italic>ALAS2</italic> in mouse erythroleukemia cells <xref rid=\"pone.0086718-Sadlon1\" ref-type=\"bibr\">[45]</xref>&#x02013;<xref rid=\"pone.0086718-Surinya1\" ref-type=\"bibr\">[47]</xref>, which is consistent with our analysis (<xref ref-type=\"fig\" rid=\"pone-0086718-g002\">Figure 2</xref>). We also identified introns 2 and 6 of <italic>FECH</italic> as potential intronic enhancers, although two promoter regions of mouse <italic>FECH</italic> were shown to function in basic and inducible expression <xref rid=\"pone.0086718-Taketani1\" ref-type=\"bibr\">[48]</xref>.</p><p>The use of the first intron as an alternative promoter to induce erythrocyte-specific expression has been documented for several hematopoietic genes, including <italic>Abcg2</italic>\\n<xref rid=\"pone.0086718-Campbell1\" ref-type=\"bibr\">[49]</xref>, <xref rid=\"pone.0086718-Nakanishi1\" ref-type=\"bibr\">[50]</xref>, <italic>Ank1</italic>\\n<xref rid=\"pone.0086718-Birkenmeier1\" ref-type=\"bibr\">[51]</xref> and <italic>Slc11a2</italic>\\n<xref rid=\"pone.0086718-Tallack1\" ref-type=\"bibr\">[21]</xref>. We also determined whether evolutionarily conserved intron regions of these genes could be detected. Without exception, the first intron regions of all three genes contain long stretches of DNA that are both conserved across vertebrates and accessible to DNase attack, suggesting that transcription factor binding occurs (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s009\">Table S6</xref>). The high frequency of alternative first exons found in erythroid genes has been shown to be crucial for the regulation of gene function <xref rid=\"pone.0086718-Tan1\" ref-type=\"bibr\">[52]</xref>, and we propose here that the evolutionarily conserved region in the first intron can serve as either an alternative promoter or enhancer to enable alteration of the first exon during gene transcription <xref rid=\"pone.0086718-Landry1\" ref-type=\"bibr\">[53]</xref>.</p></sec><sec id=\"s3c\"><title>Elucidation of Novel Evolutionarily Conserved IREs at Intron-exon Boundaries</title><p>An IRE was previously identified in the 5&#x02032;UTR of ALAS2 <xref rid=\"pone.0086718-Cox1\" ref-type=\"bibr\">[54]</xref>. However, the detection IREs in the 5&#x02032;UTRs of PBGD sequences is reported for the first time here. The scattered appearance of IREs in enzymes other than ALAS suggests that animal genes acquired IREs at a later point through the process of convergent evolution <xref rid=\"pone.0086718-Piccinelli1\" ref-type=\"bibr\">[55]</xref>. Moreover, whereas the noted IRE found in an echinoderm ALAS (<italic>Ciona intestinalis</italic>) has been reported previously <xref rid=\"pone.0086718-Piccinelli1\" ref-type=\"bibr\">[55]</xref>, our identification of a potential IRE in the ALAS of a cnidarian (sea anemone) and an arthropod (honey bee) suggested that a more thorough search for the existence of IREs in animals is necessary.</p><p>Both the 5&#x02032;UTR and 3&#x02032;UTR are commonly regarded as sites where IREs are located. However, we found that intron sequences can also contain IREs. In particular, we identified an IRE at the intron 4-exon 5 boundary of UROS in <italic>Loxodonta africana</italic> (elephant). We also aligned the IRE in the corresponding region and found that this intron-exon IRE is conserved in humans, rhesus monkeys and rabbits, although these IREs are of low quality. If IREs found at intron-exon boundaries are functional, it is possible that the pre-mRNA splicing junction could be bound by IRP, thereby influencing splicing efficiency or choice.</p><p>In the iron-depleted state, IRP1 and IRP2 are stable and bind to IREs, either to inhibit protein translation (5&#x02032;UTR) or to prevent mRNA degradation (3&#x02032;UTR). In the iron-replete state, the degradation of IRP2 allows IREs to bind iron and eIF4E <xref rid=\"pone.0086718-Ma1\" ref-type=\"bibr\">[56]</xref> to initiate protein translation. Intriguingly, eIF4E has been demonstrated to function as a co-factor in the <italic>Sxl</italic>-dependent female-specific alternative splicing of <italic>msl</italic>-2 as well as <italic>Sxl</italic> premRNAs in <italic>Drosophila</italic>, which is required for sex determination due to the silencing of the X chromosome <xref rid=\"pone.0086718-Graham1\" ref-type=\"bibr\">[57]</xref>. This observation raised the possibility that eIF4E could be involved in splicing events in mammals. To determine how many IREs exist at intron-exon boundaries, we conducted a genome-wide survey to detect IREs of high quality with respect to the nucleotides present at the junctions of CDS-exons and introns in the human and zebrafish genomes. We found 21 and 12 high-quality IREs at human and zebrafish CDS-exon/intron junctions, respectively (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s007\">Table S4</xref>). One of human genes, ZNF446, has been found in the Friendly Alternative Splicing and Transcripts Database (FAST DB) <xref rid=\"pone.0086718-delaGrange1\" ref-type=\"bibr\">[58]</xref> that contains an alternatively spliced transcript (AY279351), which presumably depends on the corresponding junction containing an IRE. This possible phenomenon adds one more layer of complexity to the regulation of genes in the heme biosynthesis pathway in particular and to the regulation of erythropoiesis in general. Furthermore, this finding also calls for an investigation into the mechanism by which iron and IRP regulate the alternative splicing of protein-coding genes and the associated biological effects.</p></sec><sec id=\"s3d\"><title>Elucidation of Novel Evolutionarily Conserved HRMs</title><p>By defining a new HRM (HRM_r), we were able to identify HRMs in most of the studied PBGS genes (76%, 19/25) and all PBGD genes from <italic>Drosophila</italic>. Recently, HRMs have been implicated in regulatory functions other than controlling import into the mitochondria. The HRM in human IRP2 has been shown to be responsible for the ubiquitin-targeted degradation of the IRP2 protein <xref rid=\"pone.0086718-Ishikawa1\" ref-type=\"bibr\">[59]</xref>. The degradation of human circadian Factor Period 2 (hPer2) is also mediated by an HRM, suggesting that metabolic signals can modulate the circadian regulation of gene expression <xref rid=\"pone.0086718-Yang2\" ref-type=\"bibr\">[60]</xref>. Notably, mouse <italic>ALAS1</italic> and <italic>ALAS2</italic> have been shown to be under circadian control and are regulated by mPER1 and mPER2 <xref rid=\"pone.0086718-Zheng1\" ref-type=\"bibr\">[61]</xref>. It has been demonstrated that the HRM of ALAS2 is not involved in the heme-mediated control of import into the mitochondria <xref rid=\"pone.0086718-Munakata1\" ref-type=\"bibr\">[32]</xref>. Therefore, we postulate that the HRMs identified in ALAS2 and PBGS could be involved in protein degradation. This is consistent with the localization of the PBGS enzyme in the cytosol and that, unlike ALAS1 and ALAS2, PBGS does not need to enter the mitochondria to function (<xref ref-type=\"fig\" rid=\"pone-0086718-g001\">Figure 1</xref>). The HRM binds heme and senses both the concentration of heme and the oxidation/reduction state of the cell. Our findings indicate that product feedback control of protein stability could be involved in the evolution of the heme biosynthesis pathway.</p></sec><sec id=\"s3e\"><title>Evolutionary Implications of HRMs and IREs in the Teleost Lineage based on Comparison with other Species</title><p>We found HRMs in the ALAS sequences of chordates and echinoderms. ALAS1 HRM_t exists in all investigated vertebrate species as well. While we also detected ALAS2 HRM_t in certain monophyletic groups (mammals, birds, and amphibians), we did not find any HRM_t sequences in teleosts. Additionally, we identified IREs in the 5&#x02032;UTRs of ALAS sequences from chordates and echinoderms. In the 5&#x02032;UTR of ALAS2, there is an IRE in all of the examined vertebrate species. In the 5&#x02032;UTR of ALAS1, an IRE is found in teleosts, but not in mammals. Based on our phylogenetic analysis, chordate and echinoderm ALAS and vertebrate ALAS1/2 are derived from the same ancestral ALAS gene. We propose that the HRMs and IREs observed in ALAS1 and ALAS2 existed in the ALAS sequence of a common ancestor of vertebrates, chordates, and echinoderms. The unique absence of HRMs in ALAS2 proteins and the presence of IREs in ALAS1 mRNAs in teleosts are striking. We further propose that during the evolutionary branching and speciation of vertebrates, the loss of the HRM from ALAS2 and the retention of the IRE in ALAS1 in teleosts resulted from the adaptation of the heme biosynthesis pathway to the environment. Notably, we have also shown that the amino acid sequences of ALAS2 and UROD have experienced positive selection in the teleost lineage.</p></sec><sec id=\"s3f\"><title>Conclusion: An Integrated View of the Evolution of Multiple Types of Regulation vs. Pathway Position Reveals Different Depths within the Pathway</title><p>We have investigated the evolution of multiple controls modulating the heme biosynthesis pathway. It was illuminating to study the degree by which these combinatorial controls impinge on the pathway from top to bottom, if we regard ALAS, at position one, as the top and FECH, at position eight, as the bottom. In a summary table (<xref ref-type=\"table\" rid=\"pone-0086718-t008\">Table 8</xref>), with the eight genes represented from positions one to eight in rows and the six control mechanisms across the top of the table, it can clearly be observed that there are multiple degrees of regulatory potential with respect to this pathway. Notably, transcriptional control and iron-mediated splicing control infiltrate the middle and bottom of the pathway. Substrate (iron) or product (heme) feedback translational control as well as protein localization and protein stability control primarily infiltrate the top of the pathway. Purifying selection on the protein sequences is widespread but is subject to loose constraint in the middle and near the bottom of the pathway. An intriguing question from the point of view of pathway position is whether the evolvability of pathway regulation depends on the position of the genes in the pathway, thereby leading to differential adaptation. Our research suggests that the investigation of the molecular evolution of a pathway should involve the examination of different control mechanisms related to the activity of the genes that constitute the pathway under investigation. Thus, a new perspective in the field of evolutionary developmental biology could concern the roles played by the <italic>cis</italic>-regulatory regions and protein-coding regions of genes with respect to adaptive mutations <xref rid=\"pone.0086718-Carroll1\" ref-type=\"bibr\">[62]</xref>, <xref rid=\"pone.0086718-Hoekstra1\" ref-type=\"bibr\">[63]</xref>. We are aware that our key findings in this study are predictions based on the current understanding of the heme biosynthesis pathway, and these findings will require substantial experimental work to be confirmed or corrected.</p><table-wrap id=\"pone-0086718-t008\" orientation=\"portrait\" position=\"float\"><object-id pub-id-type=\"doi\">10.1371/journal.pone.0086718.t008</object-id><label>Table 8</label><caption><title>Multiple regulatory potentials in heme biosynthesis pathway.</title></caption><alternatives><graphic id=\"pone-0086718-t008-8\" xlink:href=\"pone.0086718.t008\"/><table frame=\"hsides\" rules=\"groups\"><colgroup span=\"1\"><col align=\"left\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/><col align=\"center\" span=\"1\"/></colgroup><thead><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">Gene</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Pathway Position</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">First Intron-mediatedTranscription Control<xref ref-type=\"table-fn\" rid=\"nt120\">a</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Splicing Control<xref ref-type=\"table-fn\" rid=\"nt121\">b</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Translational Control<xref ref-type=\"table-fn\" rid=\"nt122\">c</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Protein Localization<xref ref-type=\"table-fn\" rid=\"nt123\">d</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">ProteinStability<xref ref-type=\"table-fn\" rid=\"nt124\">e</xref>\\n</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">Selection Pressure<xref ref-type=\"table-fn\" rid=\"nt125\">f</xref>\\n</td></tr></thead><tbody><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">ALAS2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">1</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">2</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PBGD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">3</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROS</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">4</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">UROD</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">5</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">*</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">CPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">6</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">PPO</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">7</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\">***</td></tr><tr><td align=\"left\" rowspan=\"1\" colspan=\"1\">FECH</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">8</td><td align=\"left\" rowspan=\"1\" colspan=\"1\">**</td><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/><td align=\"left\" rowspan=\"1\" colspan=\"1\"/></tr></tbody></table></alternatives><table-wrap-foot><fn id=\"nt120\"><label>a</label><p>Evolutionarily conserved DNase-hypersensitive sites in intron sequences.</p></fn><fn id=\"nt121\"><label>b</label><p>IRE in intron-exon boundary that could potentially affect splicing.</p></fn><fn id=\"nt122\"><label>c</label><p>IRE in 5&#x02032;UTR,while binding IRP, could potentially inhibit protein translation.</p></fn><fn id=\"nt123\"><label>d</label><p>HRM_t that could potentially block the import of enzyme to mitochondria while binding heme.</p></fn><fn id=\"nt124\"><label>e</label><p>HRM_r that could potentially affect protein stability.</p></fn><fn id=\"nt125\"><label>f</label><p>Genes with two highest &#x003c9; value.</p></fn><fn id=\"nt126\"><label/><p>In <sup>a</sup>, *: &#x0003c;50 bps, **:&#x0003e;50 bps, &#x0003c;100 bps, ***:&#x0003e;100 bps. In <sup>bcde</sup>, *:&#x0003c;&#x0200a;=&#x0200a;0.33, **:&#x0003e;0.33, &#x0003c;&#x0200a;=&#x0200a;0.67, ***:&#x0003e;0.67 for the proportion of species collected.</p></fn></table-wrap-foot></table-wrap></sec></sec><sec sec-type=\"materials|methods\" id=\"s4\"><title>Materials and Methods</title><sec id=\"s4a\"><title>Sequence Collection</title><p>For genes encoding the eight enzymes of the heme biosynthesis pathway in animals, we collected the sequences of organisms whose entire genome sequences were available. Sequences were collected from mammals (seven species), amphibians (two species), birds (three species), reptiles (one species), teleosts (six species), echinoderms (two species), arthropods (eight species) and cnidarians (three species). The amino acid sequences, coding nucleotide sequences, exon sequences and intron sequences were downloaded from NCBI, UCSC <xref rid=\"pone.0086718-Kent1\" ref-type=\"bibr\">[64]</xref>, <xref rid=\"pone.0086718-Dreszer1\" ref-type=\"bibr\">[65]</xref> and Ensembl <xref rid=\"pone.0086718-Flicek1\" ref-type=\"bibr\">[66]</xref>. Some of the protein sequences of ALAS genes were extracted as suggested previously <xref rid=\"pone.0086718-Duncan1\" ref-type=\"bibr\">[19]</xref>. The 5&#x02032;UTR, exon and intron sequences of the coral <italic>Acropora digitifera</italic> came from OIST Marine Genome Unit <xref rid=\"pone.0086718-Shinzato1\" ref-type=\"bibr\">[67]</xref>. The 5&#x02032;UTR, exon and intron sequences of the sea anemone <italic>Nematostella vectensis</italic> were obtained from the DOE Joint Genome Institute <xref rid=\"pone.0086718-Putnam1\" ref-type=\"bibr\">[68]</xref>. The 5&#x02032;UTR, exon and intron sequences of the hydra <italic>Hydra magnipapillata</italic> also came from the DOE Joint Genome Institute <xref rid=\"pone.0086718-Chapman1\" ref-type=\"bibr\">[69]</xref>.</p><p>If a gene model lacked a 5&#x02032;UTR, we searched for a cDNA collection to determine whether there was any possibility of extension in the 5&#x02032; direction with respect to the direction of transcription. If such an extension was possible, a 5&#x02032;UTR was added to the gene model. Subsequently, if this extension added a new exon, which was regarded as the first exon, then a new intron, which was usually the first intron, was hypothesized to exist. The numbers of coding sequences obtained for each gene in the heme biosynthesis pathway were as follows: ALAS (46 genes from 31 species), PBGS (26 genes from 26 species), PBGD (29 genes from 28 species), UROS (28 genes from 28 species), UROD (31 genes from 31 species), CPO (27 genes from 27 species), PPO (26 genes from 26 species) and FECH (30 genes from 30 species) (<xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s010\">Table S7</xref>). Lists about the number of sequences and the species by taxonomic groups can be found in <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s011\">Table S8</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s012\">S9</xref>. Sequences from species whose genome had not been decoded were not included in this study.</p></sec><sec id=\"s4b\"><title>Sequence Alignment and Phylogenetic Analysis</title><p>The amino acid sequence alignment of the eight enzymes involved in the heme biosynthesis pathway was performed in MEGA5 with the MUSCLE algorithm <xref rid=\"pone.0086718-Tamura1\" ref-type=\"bibr\">[70]</xref>. After the aligned sequences were adjusted manually to confirm their accuracy, topologies of the phylogenetic trees were generated with PHYLIP <xref rid=\"pone.0086718-Felsenstein1\" ref-type=\"bibr\">[71]</xref> and PHYML <xref rid=\"pone.0086718-Guindon1\" ref-type=\"bibr\">[72]</xref> via maximum likelihood (ML) methods. The gamma rate heterogeneity model and the JTT substitution model from Tree-Puzzle <xref rid=\"pone.0086718-Schmidt1\" ref-type=\"bibr\">[73]</xref> and MEGA5 were used for the enzymes of the heme biosynthesis pathway. Branch support was provided via bootstrap analysis, involving a heuristic search with 1000 replicates. The alignments and trees are shown in <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s002\">Figures S2</xref> and <xref ref-type=\"supplementary-material\" rid=\"pone.0086718.s003\">S3</xref>, respectively.</p></sec><sec id=\"s4c\"><title>Analysis of Evolutionary Constraints</title><p>The aligned nucleotide coding sequences without gaps based on the aligned protein sequences and the unrooted tree were fed into the CODEML program of the PAML program package (version V4.4e) <xref rid=\"pone.0086718-Yang3\" ref-type=\"bibr\">[74]</xref>, <xref rid=\"pone.0086718-Yang4\" ref-type=\"bibr\">[75]</xref> to analyze the evolutionary constraints on the coding sequences for the eight genes of the heme biosynthesis pathway. For the different models applied, &#x003c9; (the nonsynonymous versus synonymous rate, <italic>d<sub>N</sub>/d<sub>S</sub></italic>) for codons can be assumed to be less than one (negative selection or purifying selection), equal to one (neutral), or greater than one (positive selection).</p><p>First, we obtained &#x003c9; for each gene by applying the null model M0, assuming a constant &#x003c9; value for all codons and branches. Second, site models M1a (Nearly Neutral) and M2a (Positive Selection) were both used to allow &#x003c9; to vary among sites. In the M1a model, the codons are categorized into two types, one of which shows &#x003c9; values of less than one, while &#x003c9; is equal to one of the other type. In the M2a model, the codons are categorized into three types, one with &#x003c9; values less than one, one for which &#x003c9; is equal to one and one with &#x003c9; values greater than one. We performed the likelihood ratio test for the M1a and M2a models compared to the null model, M0 (twice the log-likelihood difference (2&#x00394;ln<italic>L</italic>) of the two models).</p><p>We also used the branch model to analyze the evolutionary constraints on the mammal, teleost and arthropod branches. We then compared the likelihood of the null model, M0, with the branch model. If the likelihood of the branch model were to be significantly higher than that of the null model M0, it was hypothesized that the branch under consideration is potentially under positive selection. Subsequently, we used the branch-site model to allow variation among sites in the proteins and across branches to determine whether any amino acid residues were under positive selection <xref rid=\"pone.0086718-Zhang2\" ref-type=\"bibr\">[76]</xref>. For branch-site model A, the following four classes were demarcated for each amino acid: class 0 with 0&#x0003c; &#x003c9;<sub>0</sub>&#x0003c;1 in all branches; class 1 with &#x003c9;<sub>1</sub>&#x0200a;=&#x0200a;1 in all branches; class 2a with foreground &#x003c9;<sub>2</sub>&#x02265;1 but background 0&#x0003c; &#x003c9;<sub>0</sub>&#x0003c;1; and class 2b with a foreground &#x003c9;<sub>2</sub>&#x02265;1 but background &#x003c9;<sub>1</sub>&#x0200a;=&#x0200a;1. Null model A1 was the same as A but with the foreground &#x003c9;<sub>2</sub> constrained to one. A likelihood ratio test is used between models A and A1. If the likelihood of model A were to be significantly higher than the likelihood of model A1 (null model) (<italic>p</italic>&#x0003c;0.05), it would indicate that there were amino acid residues under positive selection. Bayes empirical Bayes (BEB) was used to calculate the posterior probability of identifying sites under positive selection (&#x0003e;0.95) <xref rid=\"pone.0086718-Yang5\" ref-type=\"bibr\">[77]</xref>. When we describe the position of the amino acids under positive selection, we use the human sequences (for ALAS1, PBGS and UROD) as a reference to index the sequences. We mapped the positively selected amino acids onto the crystal structures of ALAS <xref rid=\"pone.0086718-Astner1\" ref-type=\"bibr\">[34]</xref>, PBGS <xref rid=\"pone.0086718-Frankenberg1\" ref-type=\"bibr\">[35]</xref> and UROD <xref rid=\"pone.0086718-Phillips1\" ref-type=\"bibr\">[36]</xref>.</p><p>We also compared the &#x003c9; values among the eight genes using a previously described method <xref rid=\"pone.0086718-Lu1\" ref-type=\"bibr\">[2]</xref>, <xref rid=\"pone.0086718-Clotault1\" ref-type=\"bibr\">[8]</xref>. When the difference of the &#x003c9; values between two genes, such as gene A, with &#x003c9;<sub>A</sub> and a likelihood of L<sub>A0</sub> in the M0 null model, and gene B, with &#x003c9;<sub>B</sub> and a likelihood of L<sub>B0</sub> in the M0 null model, was significant, we were able to find a &#x003c9;<sub>n</sub> between the two genes that was significantly different from those of the two genes being compared. To perform this test, we set &#x003c9;<sub>n</sub> as the average &#x003c9; value of the two genes being compared (&#x003c9;<sub>n</sub>&#x0200a;=&#x0200a;(&#x003c9;<sub>A</sub>+&#x003c9;<sub>B</sub>)/2). We then acquired the likelihood values (L<sub>A</sub> and L<sub>B</sub>) after we constrained the &#x003c9; of genes A and B as &#x003c9;<sub>n</sub>. Subsequently, the statistical significance (df&#x0200a;=&#x0200a;1, <italic>p</italic>&#x0003c;0.05) of the difference in &#x003c9; was determined from the difference in the likelihood between the null model M0 and the constrained model, as follows: 2*(L<sub>A</sub> &#x02013; L<sub>A0</sub>) and 2*(L<sub>B</sub> &#x02013; L<sub>B0</sub>).</p></sec><sec id=\"s4d\"><title>Detection of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences</title><p>We used the UCSC table browser <xref rid=\"pone.0086718-Rosenbloom1\" ref-type=\"bibr\">[78]</xref>, <xref rid=\"pone.0086718-Karolchik1\" ref-type=\"bibr\">[79]</xref> and Galaxy <xref rid=\"pone.0086718-Goecks1\" ref-type=\"bibr\">[80]</xref> to extract the intron sequences at the intersections of regions that are conserved in vertebrates (hg19, phastConsElements46way) and DNase-hypersensitive sites (hg19, wgEncodeRegDnaseClustered) <xref rid=\"pone.0086718-ENCODEProjectConsortium1\" ref-type=\"bibr\">[81]</xref>.</p></sec><sec id=\"s4e\"><title>Detection of IREs</title><p>We used the SIRE web service to identify potential IRE sites in the gene sequences <xref rid=\"pone.0086718-Campillos1\" ref-type=\"bibr\">[82]</xref>. SIRE takes into consideration the non-canonical sequence as indicated by SELEX. By allowing 18 motifs to be confirmed as binding to IRP1 or IRP2, SIRE accepted the input sequence and reported the motif type, the free energy of the secondary structure and the level of stringency as High, Medium, or Low. A batch version of the same algorithm was also developed for the genome-wide detection of IREs at the intron-exon boundaries of human and zebrafish genes.</p></sec><sec id=\"s4f\"><title>Detection of HRMs</title><p>The HRM sequence, N/K/R-C-P-K or a hydrophobic residue-L/M, has been commonly used to detect HRMs <xref rid=\"pone.0086718-Duncan1\" ref-type=\"bibr\">[19]</xref>, <xref rid=\"pone.0086718-Lathrop1\" ref-type=\"bibr\">[31]</xref> (denoted HRM_t). By identifying new occurrences of HRMs that have been shown to function biologically (human ALAS1; human ALAS2 <xref rid=\"pone.0086718-Munakata1\" ref-type=\"bibr\">[32]</xref>; human, mouse, rat, spalax, and zebrafish PER2 <xref rid=\"pone.0086718-Yang2\" ref-type=\"bibr\">[60]</xref>; human IRP2 <xref rid=\"pone.0086718-Ishikawa1\" ref-type=\"bibr\">[59]</xref>; human STC2 <xref rid=\"pone.0086718-Jiang1\" ref-type=\"bibr\">[83]</xref>; and human, mouse and rabbit eIF2alpha <xref rid=\"pone.0086718-Igarashi1\" ref-type=\"bibr\">[84]</xref>), we also identified a new HRM motif, denoted HRM_r, A/C/F/G/I/R/S/Q-A/C/K/H/L/N/R/S/T-C-P-A/E/F/I/K/S/V/Y-A/D/H/I/L/M/T/V-A/L/M/P/R/S. We used seven sites in HRM_r rather than five sites in HRM_t to increase the specificity of HRM_r. We note that HRM_r is similar to, but not the same as HRM_t.</p></sec></sec><sec sec-type=\"supplementary-material\" id=\"s5\"><title>Supporting Information</title><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s001\"><label>Figure S1</label><caption><p>\\n<bold>The collection of figures describing the protein structures of enzymes involved in the heme biosynthesis pathway and the positions of the positively selected residues.</bold> Homodimer structures are shown with the monomers colored in white and yellow. Positively selected amino acid residues are colored in red. Substrate analogs or prosthetic groups are colored in green. (A) ALAS2 of teleost; (B) PBGS of arthropod; (C) UROD of teleost.</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s001.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s002\"><label>Figure S2</label><caption><p>\\n<bold>The aligned protein sequences of eight genes of heme biosynthesis pathway in animals.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s002.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s003\"><label>Figure S3</label><caption><p>\\n<bold>The collection of figures for the maximum likelihood phylogeny of protein sequences for eight genes of heme biosynthesis pathway in animals.</bold> Bootstrap values &#x0003e;70% are indicated. The bootstrap values are displayed only for the branches of the main lineages. (A) ALAS, (B) PBGS, (C) PBGD, (D) UROS, (E) UROD, (F) CPO, (G) PPO, (H) FECH.</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s003.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s004\"><label>Table S1</label><caption><p>\\n<bold>Model test (M1a) for selection of genes in heme biosynthesis pathway.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s004.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s005\"><label>Table S2</label><caption><p>\\n<bold>Length of evolutionarily conserved DNase-hypersensitive sites in intron sequences.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s005.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s006\"><label>Table S3</label><caption><p>\\n<bold>Potential IRE in eight genes of heme biosynthesis pathway.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s006.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s007\"><label>Table S4</label><caption><p>\\n<bold>Genomewide detection of potential IRE in exon-intron boundary in human and zebrafish genes.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s007.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s008\"><label>Table S5</label><caption><p>\\n<bold>Potential HRM in eight genes of heme biosynthesis pathway (HRM_t and HRM_r).</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s008.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s009\"><label>Table S6</label><caption><p>\\n<bold>Length of evolutionarily conserved DNase-hypersensitive sites in intron sequences (bps) for </bold>\\n<bold><italic>Abcg2</italic></bold>\\n<bold>, </bold>\\n<bold><italic>Ank1</italic></bold>\\n<bold> and </bold>\\n<bold><italic>Slc11a2</italic></bold>\\n<bold>.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s009.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s010\"><label>Table S7</label><caption><p>\\n<bold>IDs of genes of heme biosynthesis pathway in animals.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s010.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s011\"><label>Table S8</label><caption><p>\\n<bold>Number of sequences by taxonomic groups.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s011.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material><supplementary-material content-type=\"local-data\" id=\"pone.0086718.s012\"><label>Table S9</label><caption><p>\\n<bold>Species by taxonomic groups.</bold>\\n</p><p>(PDF)</p></caption><media xlink:href=\"pone.0086718.s012.pdf\"><caption><p>Click here for additional data file.</p></caption></media></supplementary-material></sec></body><back><ref-list><title>References</title><ref id=\"pone.0086718-Rausher1\"><label>1</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Rausher</surname><given-names>MD</given-names></name>, <name><surname>Miller</surname><given-names>RE</given-names></name>, <name><surname>Tiffin</surname><given-names>P</given-names></name> (<year>1999</year>) <article-title>Patterns of evolutionary rate variation among genes of the anthocyanin biosynthetic pathway</article-title>. <source>Mol Biol Evol</source>\\n<volume>16</volume>: <fpage>266</fpage>&#x02013;<lpage>274</lpage>.<pub-id pub-id-type=\"pmid\">10028292</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Lu1\"><label>2</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Lu</surname><given-names>Y</given-names></name>, <name><surname>Rausher</surname><given-names>MD</given-names></name> (<year>2003</year>) <article-title>Evolutionary rate variation in anthocyanin pathway genes</article-title>. <source>Mol Biol Evol</source>\\n<volume>20</volume>: <fpage>1844</fpage>&#x02013;<lpage>1853</lpage>.<pub-id pub-id-type=\"pmid\">12885963</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Rausher2\"><label>3</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Rausher</surname><given-names>MD</given-names></name>, <name><surname>Lu</surname><given-names>Y</given-names></name>, <name><surname>Meyer</surname><given-names>K</given-names></name> (<year>2008</year>) <article-title>Variation in constraint versus positive selection as an explanation for evolutionary rate variation among anthocyanin genes</article-title>. <source>J Mol Evol</source>\\n<volume>67</volume>: <fpage>137</fpage>&#x02013;<lpage>144</lpage>.<pub-id pub-id-type=\"pmid\">18654810</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Ramsay1\"><label>4</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Ramsay</surname><given-names>H</given-names></name>, <name><surname>Rieseberg</surname><given-names>LH</given-names></name>, <name><surname>Ritland</surname><given-names>K</given-names></name> (<year>2009</year>) <article-title>The correlation of evolutionary rate with pathway position in plant terpenoid biosynthesis</article-title>. <source>Mol Biol Evol</source>\\n<volume>26</volume>: <fpage>1045</fpage>&#x02013;<lpage>1053</lpage>.<pub-id pub-id-type=\"pmid\">19188263</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yu1\"><label>5</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yu</surname><given-names>G</given-names></name>, <name><surname>Olsen</surname><given-names>KM</given-names></name>, <name><surname>Schaal</surname><given-names>BA</given-names></name> (<year>2011</year>) <article-title>Molecular evolution of the endosperm starch synthesis pathway genes in rice (Oryza sativa L.) and its wild ancestor, O. rufipogon L. Mol Biol Evol</article-title>. <volume>28</volume>: <fpage>659</fpage>&#x02013;<lpage>671</lpage>.</mixed-citation></ref><ref id=\"pone.0086718-Li1\"><label>6</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Li</surname><given-names>C</given-names></name>, <name><surname>Li</surname><given-names>QG</given-names></name>, <name><surname>Dunwell</surname><given-names>JM</given-names></name>, <name><surname>Zhang</surname><given-names>YM</given-names></name> (<year>2012</year>) <article-title>Divergent evolutionary pattern of starch biosynthetic pathway genes in grasses and dicots</article-title>. <source>Mol Biol Evol</source>\\n<volume>29</volume>: <fpage>3227</fpage>&#x02013;<lpage>3236</lpage>.<pub-id pub-id-type=\"pmid\">22586327</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yang1\"><label>7</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yang</surname><given-names>YH</given-names></name>, <name><surname>Zhang</surname><given-names>FM</given-names></name>, <name><surname>Ge</surname><given-names>S</given-names></name> (<year>2009</year>) <article-title>Evolutionary rate patterns of the Gibberellin pathway genes</article-title>. <source>BMC Evol Biol</source>\\n<volume>9</volume>: <fpage>206</fpage>.<pub-id pub-id-type=\"pmid\">19689796</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Clotault1\"><label>8</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Clotault</surname><given-names>J</given-names></name>, <name><surname>Peltier</surname><given-names>D</given-names></name>, <name><surname>Soufflet-Freslon</surname><given-names>V</given-names></name>, <name><surname>Briard</surname><given-names>M</given-names></name>, <name><surname>Geoffriau</surname><given-names>E</given-names></name> (<year>2012</year>) <article-title>Differential selection on carotenoid biosynthesis genes as a function of gene position in the metabolic pathway: a study on the carrot and dicots</article-title>. <source>PLoS One</source>\\n<volume>7</volume>: <fpage>e38724</fpage>.<pub-id pub-id-type=\"pmid\">22737218</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Philip1\"><label>9</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Philip</surname><given-names>S</given-names></name>, <name><surname>Machado</surname><given-names>JP</given-names></name>, <name><surname>Maldonado</surname><given-names>E</given-names></name>, <name><surname>Vasconcelos</surname><given-names>V</given-names></name>, <name><surname>O&#x02019;Brien</surname><given-names>SJ</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>Fish lateral line innovation: insights into the evolutionary genomic dynamics of a unique mechanosensory organ</article-title>. <source>Mol Biol Evol</source>\\n<volume>29</volume>: <fpage>3887</fpage>&#x02013;<lpage>3898</lpage>.<pub-id pub-id-type=\"pmid\">22844072</pub-id></mixed-citation></ref><ref id=\"pone.0086718-OlsonManning1\"><label>10</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Olson-Manning</surname><given-names>CF</given-names></name>, <name><surname>Lee</surname><given-names>CR</given-names></name>, <name><surname>Rausher</surname><given-names>MD</given-names></name>, <name><surname>Mitchell-Olds</surname><given-names>T</given-names></name> (<year>2013</year>) <article-title>Evolution of flux control in the glucosinolate pathway in Arabidopsis thaliana</article-title>. <source>Mol Biol Evol</source>\\n<volume>30</volume>: <fpage>14</fpage>&#x02013;<lpage>23</lpage>.<pub-id pub-id-type=\"pmid\">22923463</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Montanucci1\"><label>11</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Montanucci</surname><given-names>L</given-names></name>, <name><surname>Laayouni</surname><given-names>H</given-names></name>, <name><surname>Dall&#x02019;Olio</surname><given-names>GM</given-names></name>, <name><surname>Bertranpetit</surname><given-names>J</given-names></name> (<year>2011</year>) <article-title>Molecular evolution and network-level analysis of the N-glycosylation metabolic pathway across primates</article-title>. <source>Mol Biol Evol</source>\\n<volume>28</volume>: <fpage>813</fpage>&#x02013;<lpage>823</lpage>.<pub-id pub-id-type=\"pmid\">20924085</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Cork1\"><label>12</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Cork</surname><given-names>JM</given-names></name>, <name><surname>Purugganan</surname><given-names>MD</given-names></name> (<year>2004</year>) <article-title>The evolution of molecular genetic pathways and networks</article-title>. <source>Bioessays</source>\\n<volume>26</volume>: <fpage>479</fpage>&#x02013;<lpage>484</lpage>.<pub-id pub-id-type=\"pmid\">15112228</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Khan1\"><label>13</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Khan</surname><given-names>AA</given-names></name>, <name><surname>Quigley</surname><given-names>JG</given-names></name> (<year>2011</year>) <article-title>Control of intracellular heme levels: heme transporters and heme oxygenases</article-title>. <source>Biochim Biophys Acta</source>\\n<volume>1813</volume>: <fpage>668</fpage>&#x02013;<lpage>682</lpage>.<pub-id pub-id-type=\"pmid\">21238504</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Heinemann1\"><label>14</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Heinemann</surname><given-names>IU</given-names></name>, <name><surname>Jahn</surname><given-names>M</given-names></name>, <name><surname>Jahn</surname><given-names>D</given-names></name> (<year>2008</year>) <article-title>The biochemistry of heme biosynthesis</article-title>. <source>Arch Biochem Biophys</source>\\n<volume>474</volume>: <fpage>238</fpage>&#x02013;<lpage>251</lpage>.<pub-id pub-id-type=\"pmid\">18314007</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Furuyama1\"><label>15</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Furuyama</surname><given-names>K</given-names></name>, <name><surname>Kaneko</surname><given-names>K</given-names></name>, <name><surname>Vargas</surname><given-names>PD</given-names></name> (<year>2007</year>) <article-title>Heme as a magnificent molecule with multiple missions: heme determines its own fate and governs cellular homeostasis</article-title>. <source>Tohoku J Exp Med</source>\\n<volume>213</volume>: <fpage>1</fpage>&#x02013;<lpage>16</lpage>.<pub-id pub-id-type=\"pmid\">17785948</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Zhang1\"><label>16</label><mixed-citation publication-type=\"other\">Zhang L (2011) Heme biology : the secret life of heme in regulating diverse biological processes. Hackensack, NJ: World Scientific. 213 p. p.</mixed-citation></ref><ref id=\"pone.0086718-Dailey1\"><label>17</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Dailey</surname><given-names>HA</given-names></name>, <name><surname>Meissner</surname><given-names>PN</given-names></name> (<year>2013</year>) <article-title>Erythroid heme biosynthesis and its disorders</article-title>. <source>Cold Spring Harb Perspect Med</source>\\n<volume>3</volume>: <fpage>a011676</fpage>.<pub-id pub-id-type=\"pmid\">23471474</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Layer1\"><label>18</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Layer</surname><given-names>G</given-names></name>, <name><surname>Reichelt</surname><given-names>J</given-names></name>, <name><surname>Jahn</surname><given-names>D</given-names></name>, <name><surname>Heinz</surname><given-names>DW</given-names></name> (<year>2010</year>) <article-title>Structure and function of enzymes in heme biosynthesis</article-title>. <source>Protein Sci</source>\\n<volume>19</volume>: <fpage>1137</fpage>&#x02013;<lpage>1161</lpage>.<pub-id pub-id-type=\"pmid\">20506125</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Duncan1\"><label>19</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Duncan</surname><given-names>R</given-names></name>, <name><surname>Faggart</surname><given-names>MA</given-names></name>, <name><surname>Roger</surname><given-names>AJ</given-names></name>, <name><surname>Cornell</surname><given-names>NW</given-names></name> (<year>1999</year>) <article-title>Phylogenetic analysis of the 5-aminolevulinate synthase gene</article-title>. <source>Mol Biol Evol</source>\\n<volume>16</volume>: <fpage>383</fpage>&#x02013;<lpage>396</lpage>.<pub-id pub-id-type=\"pmid\">10331264</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Kerenyi1\"><label>20</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Kerenyi</surname><given-names>MA</given-names></name>, <name><surname>Orkin</surname><given-names>SH</given-names></name> (<year>2010</year>) <article-title>Networking erythropoiesis</article-title>. <source>J Exp Med</source>\\n<volume>207</volume>: <fpage>2537</fpage>&#x02013;<lpage>2541</lpage>.<pub-id pub-id-type=\"pmid\">21098097</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Tallack1\"><label>21</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Tallack</surname><given-names>MR</given-names></name>, <name><surname>Whitington</surname><given-names>T</given-names></name>, <name><surname>Yuen</surname><given-names>WS</given-names></name>, <name><surname>Wainwright</surname><given-names>EN</given-names></name>, <name><surname>Keys</surname><given-names>JR</given-names></name>, <etal>et al</etal> (<year>2010</year>) <article-title>A global role for KLF1 in erythropoiesis revealed by ChIP-seq in primary erythroid cells</article-title>. <source>Genome Res</source>\\n<volume>20</volume>: <fpage>1052</fpage>&#x02013;<lpage>1063</lpage>.<pub-id pub-id-type=\"pmid\">20508144</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Cheng1\"><label>22</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Cheng</surname><given-names>Y</given-names></name>, <name><surname>Wu</surname><given-names>W</given-names></name>, <name><surname>Kumar</surname><given-names>SA</given-names></name>, <name><surname>Yu</surname><given-names>D</given-names></name>, <name><surname>Deng</surname><given-names>W</given-names></name>, <etal>et al</etal> (<year>2009</year>) <article-title>Erythroid GATA1 function revealed by genome-wide analysis of transcription factor occupancy, histone modifications, and mRNA expression</article-title>. <source>Genome Res</source>\\n<volume>19</volume>: <fpage>2172</fpage>&#x02013;<lpage>2184</lpage>.<pub-id pub-id-type=\"pmid\">19887574</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Kassouf1\"><label>23</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Kassouf</surname><given-names>MT</given-names></name>, <name><surname>Hughes</surname><given-names>JR</given-names></name>, <name><surname>Taylor</surname><given-names>S</given-names></name>, <name><surname>McGowan</surname><given-names>SJ</given-names></name>, <name><surname>Soneji</surname><given-names>S</given-names></name>, <etal>et al</etal> (<year>2010</year>) <article-title>Genome-wide identification of TAL1&#x02019;s functional targets: insights into its mechanisms of action in primary erythroid cells</article-title>. <source>Genome Res</source>\\n<volume>20</volume>: <fpage>1064</fpage>&#x02013;<lpage>1083</lpage>.<pub-id pub-id-type=\"pmid\">20566737</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Ajioka1\"><label>24</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Ajioka</surname><given-names>RS</given-names></name>, <name><surname>Phillips</surname><given-names>JD</given-names></name>, <name><surname>Kushner</surname><given-names>JP</given-names></name> (<year>2006</year>) <article-title>Biosynthesis of heme in mammals</article-title>. <source>Biochim Biophys Acta</source>\\n<volume>1763</volume>: <fpage>723</fpage>&#x02013;<lpage>736</lpage>.<pub-id pub-id-type=\"pmid\">16839620</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Natarajan1\"><label>25</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Natarajan</surname><given-names>A</given-names></name>, <name><surname>Yardimci</surname><given-names>GG</given-names></name>, <name><surname>Sheffield</surname><given-names>NC</given-names></name>, <name><surname>Crawford</surname><given-names>GE</given-names></name>, <name><surname>Ohler</surname><given-names>U</given-names></name> (<year>2012</year>) <article-title>Predicting cell-type-specific gene expression from regions of open chromatin</article-title>. <source>Genome Res</source>\\n<volume>22</volume>: <fpage>1711</fpage>&#x02013;<lpage>1722</lpage>.<pub-id pub-id-type=\"pmid\">22955983</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Neph1\"><label>26</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Neph</surname><given-names>S</given-names></name>, <name><surname>Vierstra</surname><given-names>J</given-names></name>, <name><surname>Stergachis</surname><given-names>AB</given-names></name>, <name><surname>Reynolds</surname><given-names>AP</given-names></name>, <name><surname>Haugen</surname><given-names>E</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>An expansive human regulatory lexicon encoded in transcription factor footprints</article-title>. <source>Nature</source>\\n<volume>489</volume>: <fpage>83</fpage>&#x02013;<lpage>90</lpage>.<pub-id pub-id-type=\"pmid\">22955618</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Thurman1\"><label>27</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Thurman</surname><given-names>RE</given-names></name>, <name><surname>Rynes</surname><given-names>E</given-names></name>, <name><surname>Humbert</surname><given-names>R</given-names></name>, <name><surname>Vierstra</surname><given-names>J</given-names></name>, <name><surname>Maurano</surname><given-names>MT</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>The accessible chromatin landscape of the human genome</article-title>. <source>Nature</source>\\n<volume>489</volume>: <fpage>75</fpage>&#x02013;<lpage>82</lpage>.<pub-id pub-id-type=\"pmid\">22955617</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Evstatiev1\"><label>28</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Evstatiev</surname><given-names>R</given-names></name>, <name><surname>Gasche</surname><given-names>C</given-names></name> (<year>2012</year>) <article-title>Iron sensing and signalling</article-title>. <source>Gut</source>\\n<volume>61</volume>: <fpage>933</fpage>&#x02013;<lpage>952</lpage>.<pub-id pub-id-type=\"pmid\">22016365</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Anderson1\"><label>29</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Anderson</surname><given-names>CP</given-names></name>, <name><surname>Shen</surname><given-names>M</given-names></name>, <name><surname>Eisenstein</surname><given-names>RS</given-names></name>, <name><surname>Leibold</surname><given-names>EA</given-names></name> (<year>2012</year>) <article-title>Mammalian iron metabolism and its control by iron regulatory proteins</article-title>. <source>Biochim Biophys Acta</source>\\n<volume>1823</volume>: <fpage>1468</fpage>&#x02013;<lpage>1483</lpage>.<pub-id pub-id-type=\"pmid\">22610083</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Chen1\"><label>30</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Chen</surname><given-names>CY</given-names></name>, <name><surname>Paw</surname><given-names>BH</given-names></name> (<year>2012</year>) <article-title>Cellular and mitochondrial iron homeostasis in vertebrates</article-title>. <source>Biochimica Et Biophysica Acta-Molecular Cell Research</source>\\n<volume>1823</volume>: <fpage>1459</fpage>&#x02013;<lpage>1467</lpage>.</mixed-citation></ref><ref id=\"pone.0086718-Lathrop1\"><label>31</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Lathrop</surname><given-names>JT</given-names></name>, <name><surname>Timko</surname><given-names>MP</given-names></name> (<year>1993</year>) <article-title>Regulation by heme of mitochondrial protein transport through a conserved amino acid motif</article-title>. <source>Science</source>\\n<volume>259</volume>: <fpage>522</fpage>&#x02013;<lpage>525</lpage>.<pub-id pub-id-type=\"pmid\">8424176</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Munakata1\"><label>32</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Munakata</surname><given-names>H</given-names></name>, <name><surname>Sun</surname><given-names>JY</given-names></name>, <name><surname>Yoshida</surname><given-names>K</given-names></name>, <name><surname>Nakatani</surname><given-names>T</given-names></name>, <name><surname>Honda</surname><given-names>E</given-names></name>, <etal>et al</etal> (<year>2004</year>) <article-title>Role of the heme regulatory motif in the heme-mediated inhibition of mitochondrial import of 5-aminolevulinate synthase</article-title>. <source>J Biochem</source>\\n<volume>136</volume>: <fpage>233</fpage>&#x02013;<lpage>238</lpage>.<pub-id pub-id-type=\"pmid\">15496594</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Dailey2\"><label>33</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Dailey</surname><given-names>TA</given-names></name>, <name><surname>Woodruff</surname><given-names>JH</given-names></name>, <name><surname>Dailey</surname><given-names>HA</given-names></name> (<year>2005</year>) <article-title>Examination of mitochondrial protein targeting of haem synthetic enzymes: in vivo identification of three functional haem-responsive motifs in 5-aminolaevulinate synthase</article-title>. <source>Biochem J</source>\\n<volume>386</volume>: <fpage>381</fpage>&#x02013;<lpage>386</lpage>.<pub-id pub-id-type=\"pmid\">15482256</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Astner1\"><label>34</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Astner</surname><given-names>I</given-names></name>, <name><surname>Schulze</surname><given-names>JO</given-names></name>, <name><surname>van den Heuvel</surname><given-names>J</given-names></name>, <name><surname>Jahn</surname><given-names>D</given-names></name>, <name><surname>Schubert</surname><given-names>WD</given-names></name>, <etal>et al</etal> (<year>2005</year>) <article-title>Crystal structure of 5-aminolevulinate synthase, the first enzyme of heme biosynthesis, and its link to XLSA in humans</article-title>. <source>Embo J</source>\\n<volume>24</volume>: <fpage>3166</fpage>&#x02013;<lpage>3177</lpage>.<pub-id pub-id-type=\"pmid\">16121195</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Frankenberg1\"><label>35</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Frankenberg</surname><given-names>N</given-names></name>, <name><surname>Erskine</surname><given-names>PT</given-names></name>, <name><surname>Cooper</surname><given-names>JB</given-names></name>, <name><surname>Shoolingin-Jordan</surname><given-names>PM</given-names></name>, <name><surname>Jahn</surname><given-names>D</given-names></name>, <etal>et al</etal> (<year>1999</year>) <article-title>High resolution crystal structure of a Mg2+-dependent porphobilinogen synthase</article-title>. <source>J Mol Biol</source>\\n<volume>289</volume>: <fpage>591</fpage>&#x02013;<lpage>602</lpage>.<pub-id pub-id-type=\"pmid\">10356331</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Phillips1\"><label>36</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Phillips</surname><given-names>JD</given-names></name>, <name><surname>Whitby</surname><given-names>FG</given-names></name>, <name><surname>Kushner</surname><given-names>JP</given-names></name>, <name><surname>Hill</surname><given-names>CP</given-names></name> (<year>2003</year>) <article-title>Structural basis for tetrapyrrole coordination by uroporphyrinogen decarboxylase</article-title>. <source>Embo J</source>\\n<volume>22</volume>: <fpage>6225</fpage>&#x02013;<lpage>6233</lpage>.<pub-id pub-id-type=\"pmid\">14633982</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Kaya1\"><label>37</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Kaya</surname><given-names>AH</given-names></name>, <name><surname>Plewinska</surname><given-names>M</given-names></name>, <name><surname>Wong</surname><given-names>DM</given-names></name>, <name><surname>Desnick</surname><given-names>RJ</given-names></name>, <name><surname>Wetmur</surname><given-names>JG</given-names></name> (<year>1994</year>) <article-title>Human delta-aminolevulinate dehydratase (ALAD) gene: structure and alternative splicing of the erythroid and housekeeping mRNAs</article-title>. <source>Genomics</source>\\n<volume>19</volume>: <fpage>242</fpage>&#x02013;<lpage>248</lpage>.<pub-id pub-id-type=\"pmid\">8188255</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Mignotte1\"><label>38</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Mignotte</surname><given-names>V</given-names></name>, <name><surname>Eleouet</surname><given-names>JF</given-names></name>, <name><surname>Raich</surname><given-names>N</given-names></name>, <name><surname>Romeo</surname><given-names>PH</given-names></name> (<year>1989</year>) <article-title>Cis- and trans-acting elements involved in the regulation of the erythroid promoter of the human porphobilinogen deaminase gene</article-title>. <source>Proc Natl Acad Sci U S A</source>\\n<volume>86</volume>: <fpage>6548</fpage>&#x02013;<lpage>6552</lpage>.<pub-id pub-id-type=\"pmid\">2771941</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Chretien1\"><label>39</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Chretien</surname><given-names>S</given-names></name>, <name><surname>Dubart</surname><given-names>A</given-names></name>, <name><surname>Beaupain</surname><given-names>D</given-names></name>, <name><surname>Raich</surname><given-names>N</given-names></name>, <name><surname>Grandchamp</surname><given-names>B</given-names></name>, <etal>et al</etal> (<year>1988</year>) <article-title>Alternative transcription and splicing of the human porphobilinogen deaminase gene result either in tissue-specific or in housekeeping expression</article-title>. <source>Proc Natl Acad Sci U S A</source>\\n<volume>85</volume>: <fpage>6</fpage>&#x02013;<lpage>10</lpage>.<pub-id pub-id-type=\"pmid\">3422427</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Grandchamp1\"><label>40</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Grandchamp</surname><given-names>B</given-names></name>, <name><surname>De Verneuil</surname><given-names>H</given-names></name>, <name><surname>Beaumont</surname><given-names>C</given-names></name>, <name><surname>Chretien</surname><given-names>S</given-names></name>, <name><surname>Walter</surname><given-names>O</given-names></name>, <etal>et al</etal> (<year>1987</year>) <article-title>Tissue-specific expression of porphobilinogen deaminase. Two isoenzymes from a single gene</article-title>. <source>Eur J Biochem</source>\\n<volume>162</volume>: <fpage>105</fpage>&#x02013;<lpage>110</lpage>.<pub-id pub-id-type=\"pmid\">3816774</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Aizencang1\"><label>41</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Aizencang</surname><given-names>G</given-names></name>, <name><surname>Solis</surname><given-names>C</given-names></name>, <name><surname>Bishop</surname><given-names>DF</given-names></name>, <name><surname>Warner</surname><given-names>C</given-names></name>, <name><surname>Desnick</surname><given-names>RJ</given-names></name> (<year>2000</year>) <article-title>Human uroporphyrinogen-III synthase: genomic organization, alternative promoters, and erythroid-specific expression</article-title>. <source>Genomics</source>\\n<volume>70</volume>: <fpage>223</fpage>&#x02013;<lpage>231</lpage>.<pub-id pub-id-type=\"pmid\">11112350</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Aizencang2\"><label>42</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Aizencang</surname><given-names>GI</given-names></name>, <name><surname>Bishop</surname><given-names>DF</given-names></name>, <name><surname>Forrest</surname><given-names>D</given-names></name>, <name><surname>Astrin</surname><given-names>KH</given-names></name>, <name><surname>Desnick</surname><given-names>RJ</given-names></name> (<year>2000</year>) <article-title>Uroporphyrinogen III synthase. An alternative promoter controls erythroid-specific expression in the murine gene</article-title>. <source>J Biol Chem</source>\\n<volume>275</volume>: <fpage>2295</fpage>&#x02013;<lpage>2304</lpage>.<pub-id pub-id-type=\"pmid\">10644678</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Bloomer1\"><label>43</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Bloomer</surname><given-names>JR</given-names></name> (<year>1981</year>) <article-title>Enzyme defects in the porphyrias and their relevance to the biochemical abnormalities in these disorders</article-title>. <source>J Invest Dermatol</source>\\n<volume>77</volume>: <fpage>102</fpage>&#x02013;<lpage>106</lpage>.<pub-id pub-id-type=\"pmid\">7252240</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Brenner1\"><label>44</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Brenner</surname><given-names>DA</given-names></name>, <name><surname>Bloomer</surname><given-names>JR</given-names></name> (<year>1980</year>) <article-title>The enzymatic defect in variegate prophyria. Studies with human cultured skin fibroblasts</article-title>. <source>N Engl J Med</source>\\n<volume>302</volume>: <fpage>765</fpage>&#x02013;<lpage>769</lpage>.<pub-id pub-id-type=\"pmid\">7354807</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Sadlon1\"><label>45</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Sadlon</surname><given-names>TJ</given-names></name>, <name><surname>Dell&#x02019;Oso</surname><given-names>T</given-names></name>, <name><surname>Surinya</surname><given-names>KH</given-names></name>, <name><surname>May</surname><given-names>BK</given-names></name> (<year>1999</year>) <article-title>Regulation of erythroid 5-aminolevulinate synthase expression during erythropoiesis</article-title>. <source>Int J Biochem Cell Biol</source>\\n<volume>31</volume>: <fpage>1153</fpage>&#x02013;<lpage>1167</lpage>.<pub-id pub-id-type=\"pmid\">10582344</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Schoenhaut1\"><label>46</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Schoenhaut</surname><given-names>DS</given-names></name>, <name><surname>Curtis</surname><given-names>PJ</given-names></name> (<year>1989</year>) <article-title>Structure of a mouse erythroid 5-aminolevulinate synthase gene and mapping of erythroid-specific DNAse I hypersensitive sites</article-title>. <source>Nucleic Acids Res</source>\\n<volume>17</volume>: <fpage>7013</fpage>&#x02013;<lpage>7028</lpage>.<pub-id pub-id-type=\"pmid\">2780318</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Surinya1\"><label>47</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Surinya</surname><given-names>KH</given-names></name>, <name><surname>Cox</surname><given-names>TC</given-names></name>, <name><surname>May</surname><given-names>BK</given-names></name> (<year>1998</year>) <article-title>Identification and characterization of a conserved erythroid-specific enhancer located in intron 8 of the human 5-aminolevulinate synthase 2 gene</article-title>. <source>J Biol Chem</source>\\n<volume>273</volume>: <fpage>16798</fpage>&#x02013;<lpage>16809</lpage>.<pub-id pub-id-type=\"pmid\">9642238</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Taketani1\"><label>48</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Taketani</surname><given-names>S</given-names></name>, <name><surname>Mohri</surname><given-names>T</given-names></name>, <name><surname>Hioki</surname><given-names>K</given-names></name>, <name><surname>Tokunaga</surname><given-names>R</given-names></name>, <name><surname>Kohno</surname><given-names>H</given-names></name> (<year>1999</year>) <article-title>Structure and transcriptional regulation of the mouse ferrochelatase gene</article-title>. <source>Gene</source>\\n<volume>227</volume>: <fpage>117</fpage>&#x02013;<lpage>124</lpage>.<pub-id pub-id-type=\"pmid\">10023040</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Campbell1\"><label>49</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Campbell</surname><given-names>PK</given-names></name>, <name><surname>Zong</surname><given-names>Y</given-names></name>, <name><surname>Yang</surname><given-names>S</given-names></name>, <name><surname>Zhou</surname><given-names>S</given-names></name>, <name><surname>Rubnitz</surname><given-names>JE</given-names></name>, <etal>et al</etal> (<year>2011</year>) <article-title>Identification of a novel, tissue-specific ABCG2 promoter expressed in pediatric acute megakaryoblastic leukemia</article-title>. <source>Leuk Res</source>\\n<volume>35</volume>: <fpage>1321</fpage>&#x02013;<lpage>1329</lpage>.<pub-id pub-id-type=\"pmid\">21640380</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Nakanishi1\"><label>50</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Nakanishi</surname><given-names>T</given-names></name>, <name><surname>Bailey-Dell</surname><given-names>KJ</given-names></name>, <name><surname>Hassel</surname><given-names>BA</given-names></name>, <name><surname>Shiozawa</surname><given-names>K</given-names></name>, <name><surname>Sullivan</surname><given-names>DM</given-names></name>, <etal>et al</etal> (<year>2006</year>) <article-title>Novel 5&#x02032; untranslated region variants of BCRP mRNA are differentially expressed in drug-selected cancer cells and in normal human tissues: Implications for drug resistance, tissue-specific expression, and alternative promoter usage</article-title>. <source>Cancer Res</source>\\n<volume>66</volume>: <fpage>5007</fpage>&#x02013;<lpage>5011</lpage>.<pub-id pub-id-type=\"pmid\">16707421</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Birkenmeier1\"><label>51</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Birkenmeier</surname><given-names>CS</given-names></name>, <name><surname>White</surname><given-names>RA</given-names></name>, <name><surname>Peters</surname><given-names>LL</given-names></name>, <name><surname>Hall</surname><given-names>EJ</given-names></name>, <name><surname>Lux</surname><given-names>SE</given-names></name>, <etal>et al</etal> (<year>1993</year>) <article-title>Complex patterns of sequence variation and multiple 5&#x02032; and 3&#x02032; ends are found among transcripts of the erythroid ankyrin gene</article-title>. <source>J Biol Chem</source>\\n<volume>268</volume>: <fpage>9533</fpage>&#x02013;<lpage>9540</lpage>.<pub-id pub-id-type=\"pmid\">8486643</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Tan1\"><label>52</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Tan</surname><given-names>JS</given-names></name>, <name><surname>Mohandas</surname><given-names>N</given-names></name>, <name><surname>Conboy</surname><given-names>JG</given-names></name> (<year>2006</year>) <article-title>High frequency of alternative first exons in erythroid genes suggests a critical role in regulating gene function</article-title>. <source>Blood</source>\\n<volume>107</volume>: <fpage>2557</fpage>&#x02013;<lpage>2561</lpage>.<pub-id pub-id-type=\"pmid\">16293607</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Landry1\"><label>53</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Landry</surname><given-names>JR</given-names></name>, <name><surname>Mager</surname><given-names>DL</given-names></name>, <name><surname>Wilhelm</surname><given-names>BT</given-names></name> (<year>2003</year>) <article-title>Complex controls: the role of alternative promoters in mammalian genomes</article-title>. <source>Trends Genet</source>\\n<volume>19</volume>: <fpage>640</fpage>&#x02013;<lpage>648</lpage>.<pub-id pub-id-type=\"pmid\">14585616</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Cox1\"><label>54</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Cox</surname><given-names>TC</given-names></name>, <name><surname>Bawden</surname><given-names>MJ</given-names></name>, <name><surname>Martin</surname><given-names>A</given-names></name>, <name><surname>May</surname><given-names>BK</given-names></name> (<year>1991</year>) <article-title>Human erythroid 5-aminolevulinate synthase: promoter analysis and identification of an iron-responsive element in the mRNA</article-title>. <source>EMBO J</source>\\n<volume>10</volume>: <fpage>1891</fpage>&#x02013;<lpage>1902</lpage>.<pub-id pub-id-type=\"pmid\">2050125</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Piccinelli1\"><label>55</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Piccinelli</surname><given-names>P</given-names></name>, <name><surname>Samuelsson</surname><given-names>T</given-names></name> (<year>2007</year>) <article-title>Evolution of the iron-responsive element</article-title>. <source>RNA</source>\\n<volume>13</volume>: <fpage>952</fpage>&#x02013;<lpage>966</lpage>.<pub-id pub-id-type=\"pmid\">17513696</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Ma1\"><label>56</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Ma</surname><given-names>J</given-names></name>, <name><surname>Haldar</surname><given-names>S</given-names></name>, <name><surname>Khan</surname><given-names>MA</given-names></name>, <name><surname>Sharma</surname><given-names>SD</given-names></name>, <name><surname>Merrick</surname><given-names>WC</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>Fe2+ binds iron responsive element-RNA, selectively changing protein-binding affinities and regulating mRNA repression and activation</article-title>. <source>Proc Natl Acad Sci U S A</source>\\n<volume>109</volume>: <fpage>8417</fpage>&#x02013;<lpage>8422</lpage>.<pub-id pub-id-type=\"pmid\">22586079</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Graham1\"><label>57</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Graham</surname><given-names>PL</given-names></name>, <name><surname>Yanowitz</surname><given-names>JL</given-names></name>, <name><surname>Penn</surname><given-names>JK</given-names></name>, <name><surname>Deshpande</surname><given-names>G</given-names></name>, <name><surname>Schedl</surname><given-names>P</given-names></name> (<year>2011</year>) <article-title>The translation initiation factor eIF4E regulates the sex-specific expression of the master switch gene Sxl in Drosophila melanogaster</article-title>. <source>PLoS Genet</source>\\n<volume>7</volume>: <fpage>e1002185</fpage>.<pub-id pub-id-type=\"pmid\">21829374</pub-id></mixed-citation></ref><ref id=\"pone.0086718-delaGrange1\"><label>58</label><mixed-citation publication-type=\"journal\">\\n<name><surname>de la Grange</surname><given-names>P</given-names></name>, <name><surname>Dutertre</surname><given-names>M</given-names></name>, <name><surname>Martin</surname><given-names>N</given-names></name>, <name><surname>Auboeuf</surname><given-names>D</given-names></name> (<year>2005</year>) <article-title>FAST DB: a website resource for the study of the expression regulation of human gene products</article-title>. <source>Nucleic Acids Res</source>\\n<volume>33</volume>: <fpage>4276</fpage>&#x02013;<lpage>4284</lpage>.<pub-id pub-id-type=\"pmid\">16052034</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Ishikawa1\"><label>59</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Ishikawa</surname><given-names>H</given-names></name>, <name><surname>Kato</surname><given-names>M</given-names></name>, <name><surname>Hori</surname><given-names>H</given-names></name>, <name><surname>Ishimori</surname><given-names>K</given-names></name>, <name><surname>Kirisako</surname><given-names>T</given-names></name>, <etal>et al</etal> (<year>2005</year>) <article-title>Involvement of heme regulatory motif in heme-mediated ubiquitination and degradation of IRP2</article-title>. <source>Mol Cell</source>\\n<volume>19</volume>: <fpage>171</fpage>&#x02013;<lpage>181</lpage>.<pub-id pub-id-type=\"pmid\">16039587</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yang2\"><label>60</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yang</surname><given-names>J</given-names></name>, <name><surname>Kim</surname><given-names>KD</given-names></name>, <name><surname>Lucas</surname><given-names>A</given-names></name>, <name><surname>Drahos</surname><given-names>KE</given-names></name>, <name><surname>Santos</surname><given-names>CS</given-names></name>, <etal>et al</etal> (<year>2008</year>) <article-title>A novel heme-regulatory motif mediates heme-dependent degradation of the circadian factor period 2</article-title>. <source>Mol Cell Biol</source>\\n<volume>28</volume>: <fpage>4697</fpage>&#x02013;<lpage>4711</lpage>.<pub-id pub-id-type=\"pmid\">18505821</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Zheng1\"><label>61</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Zheng</surname><given-names>BH</given-names></name>, <name><surname>Albrecht</surname><given-names>U</given-names></name>, <name><surname>Kaasik</surname><given-names>K</given-names></name>, <name><surname>Sage</surname><given-names>M</given-names></name>, <name><surname>Lu</surname><given-names>WQ</given-names></name>, <etal>et al</etal> (<year>2001</year>) <article-title>Nonredundant roles of the mPer1 and mPer2 genes in the mammalian circadian clock</article-title>. <source>Cell</source>\\n<volume>105</volume>: <fpage>683</fpage>&#x02013;<lpage>694</lpage>.<pub-id pub-id-type=\"pmid\">11389837</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Carroll1\"><label>62</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Carroll</surname><given-names>SB</given-names></name> (<year>2005</year>) <article-title>Evolution at two levels: on genes and form</article-title>. <source>PLoS Biol</source>\\n<volume>3</volume>: <fpage>e245</fpage>.<pub-id pub-id-type=\"pmid\">16000021</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Hoekstra1\"><label>63</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Hoekstra</surname><given-names>HE</given-names></name>, <name><surname>Coyne</surname><given-names>JA</given-names></name> (<year>2007</year>) <article-title>The locus of evolution: Evo devo and the genetics of adaptation</article-title>. <source>Evolution</source>\\n<volume>61</volume>: <fpage>995</fpage>&#x02013;<lpage>1016</lpage>.<pub-id pub-id-type=\"pmid\">17492956</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Kent1\"><label>64</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Kent</surname><given-names>WJ</given-names></name>, <name><surname>Sugnet</surname><given-names>CW</given-names></name>, <name><surname>Furey</surname><given-names>TS</given-names></name>, <name><surname>Roskin</surname><given-names>KM</given-names></name>, <name><surname>Pringle</surname><given-names>TH</given-names></name>, <etal>et al</etal> (<year>2002</year>) <article-title>The human genome browser at UCSC</article-title>. <source>Genome Res</source>\\n<volume>12</volume>: <fpage>996</fpage>&#x02013;<lpage>1006</lpage>.<pub-id pub-id-type=\"pmid\">12045153</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Dreszer1\"><label>65</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Dreszer</surname><given-names>TR</given-names></name>, <name><surname>Karolchik</surname><given-names>D</given-names></name>, <name><surname>Zweig</surname><given-names>AS</given-names></name>, <name><surname>Hinrichs</surname><given-names>AS</given-names></name>, <name><surname>Raney</surname><given-names>BJ</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>The UCSC Genome Browser database: extensions and updates 2011</article-title>. <source>Nucleic Acids Res</source>\\n<volume>40</volume>: <fpage>D918</fpage>&#x02013;<lpage>D923</lpage>.<pub-id pub-id-type=\"pmid\">22086951</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Flicek1\"><label>66</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Flicek</surname><given-names>P</given-names></name>, <name><surname>Amode</surname><given-names>MR</given-names></name>, <name><surname>Barrell</surname><given-names>D</given-names></name>, <name><surname>Beal</surname><given-names>K</given-names></name>, <name><surname>Brent</surname><given-names>S</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>Ensembl 2012</article-title>. <source>Nucleic Acids Res</source>\\n<volume>40</volume>: <fpage>D84</fpage>&#x02013;<lpage>D90</lpage>.<pub-id pub-id-type=\"pmid\">22086963</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Shinzato1\"><label>67</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Shinzato</surname><given-names>C</given-names></name>, <name><surname>Shoguchi</surname><given-names>E</given-names></name>, <name><surname>Kawashima</surname><given-names>T</given-names></name>, <name><surname>Hamada</surname><given-names>M</given-names></name>, <name><surname>Hisata</surname><given-names>K</given-names></name>, <etal>et al</etal> (<year>2011</year>) <article-title>Using the Acropora digitifera genome to understand coral responses to environmental change</article-title>. <source>Nature</source>\\n<volume>476</volume>: <fpage>320</fpage>&#x02013;<lpage>U382</lpage>.<pub-id pub-id-type=\"pmid\">21785439</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Putnam1\"><label>68</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Putnam</surname><given-names>NH</given-names></name>, <name><surname>Srivastava</surname><given-names>M</given-names></name>, <name><surname>Hellsten</surname><given-names>U</given-names></name>, <name><surname>Dirks</surname><given-names>B</given-names></name>, <name><surname>Chapman</surname><given-names>J</given-names></name>, <etal>et al</etal> (<year>2007</year>) <article-title>Sea anemone genome reveals ancestral eumetazoan gene repertoire and genomic organization</article-title>. <source>Science</source>\\n<volume>317</volume>: <fpage>86</fpage>&#x02013;<lpage>94</lpage>.<pub-id pub-id-type=\"pmid\">17615350</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Chapman1\"><label>69</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Chapman</surname><given-names>JA</given-names></name>, <name><surname>Kirkness</surname><given-names>EF</given-names></name>, <name><surname>Simakov</surname><given-names>O</given-names></name>, <name><surname>Hampson</surname><given-names>SE</given-names></name>, <name><surname>Mitros</surname><given-names>T</given-names></name>, <etal>et al</etal> (<year>2010</year>) <article-title>The dynamic genome of Hydra</article-title>. <source>Nature</source>\\n<volume>464</volume>: <fpage>592</fpage>&#x02013;<lpage>596</lpage>.<pub-id pub-id-type=\"pmid\">20228792</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Tamura1\"><label>70</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Tamura</surname><given-names>K</given-names></name>, <name><surname>Peterson</surname><given-names>D</given-names></name>, <name><surname>Peterson</surname><given-names>N</given-names></name>, <name><surname>Stecher</surname><given-names>G</given-names></name>, <name><surname>Nei</surname><given-names>M</given-names></name>, <etal>et al</etal> (<year>2011</year>) <article-title>MEGA5: molecular evolutionary genetics analysis using maximum likelihood, evolutionary distance, and maximum parsimony methods</article-title>. <source>Mol Biol Evol</source>\\n<volume>28</volume>: <fpage>2731</fpage>&#x02013;<lpage>2739</lpage>.<pub-id pub-id-type=\"pmid\">21546353</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Felsenstein1\"><label>71</label><mixed-citation publication-type=\"other\">Felsenstein J (2005) PHYLIP (Phylogeny Inference Package) version 3.6. Distributed by the author Department of Genome Sciences, University of Washington, Seattle.</mixed-citation></ref><ref id=\"pone.0086718-Guindon1\"><label>72</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Guindon</surname><given-names>S</given-names></name>, <name><surname>Dufayard</surname><given-names>JF</given-names></name>, <name><surname>Lefort</surname><given-names>V</given-names></name>, <name><surname>Anisimova</surname><given-names>M</given-names></name>, <name><surname>Hordijk</surname><given-names>W</given-names></name>, <etal>et al</etal> (<year>2010</year>) <article-title>New Algorithms and Methods to Estimate Maximum-Likelihood Phylogenies: Assessing the Performance of PhyML 3.0</article-title>. <source>Syst Biol</source>\\n<volume>59</volume>: <fpage>307</fpage>&#x02013;<lpage>321</lpage>.<pub-id pub-id-type=\"pmid\">20525638</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Schmidt1\"><label>73</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Schmidt</surname><given-names>HA</given-names></name>, <name><surname>Strimmer</surname><given-names>K</given-names></name>, <name><surname>Vingron</surname><given-names>M</given-names></name>, <name><surname>von Haeseler</surname><given-names>A</given-names></name> (<year>2002</year>) <article-title>TREE-PUZZLE: maximum likelihood phylogenetic analysis using quartets and parallel computing</article-title>. <source>Bioinformatics</source>\\n<volume>18</volume>: <fpage>502</fpage>&#x02013;<lpage>504</lpage>.<pub-id pub-id-type=\"pmid\">11934758</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yang3\"><label>74</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yang</surname><given-names>Z</given-names></name> (<year>1997</year>) <article-title>PAML: a program package for phylogenetic analysis by maximum likelihood</article-title>. <source>Comput Appl Biosci</source>\\n<volume>13</volume>: <fpage>555</fpage>&#x02013;<lpage>556</lpage>.<pub-id pub-id-type=\"pmid\">9367129</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yang4\"><label>75</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yang</surname><given-names>Z</given-names></name> (<year>2007</year>) <article-title>PAML 4: phylogenetic analysis by maximum likelihood</article-title>. <source>Mol Biol Evol</source>\\n<volume>24</volume>: <fpage>1586</fpage>&#x02013;<lpage>1591</lpage>.<pub-id pub-id-type=\"pmid\">17483113</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Zhang2\"><label>76</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Zhang</surname><given-names>J</given-names></name>, <name><surname>Nielsen</surname><given-names>R</given-names></name>, <name><surname>Yang</surname><given-names>Z</given-names></name> (<year>2005</year>) <article-title>Evaluation of an improved branch-site likelihood method for detecting positive selection at the molecular level</article-title>. <source>Mol Biol Evol</source>\\n<volume>22</volume>: <fpage>2472</fpage>&#x02013;<lpage>2479</lpage>.<pub-id pub-id-type=\"pmid\">16107592</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Yang5\"><label>77</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Yang</surname><given-names>Z</given-names></name>, <name><surname>Wong</surname><given-names>WS</given-names></name>, <name><surname>Nielsen</surname><given-names>R</given-names></name> (<year>2005</year>) <article-title>Bayes empirical bayes inference of amino acid sites under positive selection</article-title>. <source>Mol Biol Evol</source>\\n<volume>22</volume>: <fpage>1107</fpage>&#x02013;<lpage>1118</lpage>.<pub-id pub-id-type=\"pmid\">15689528</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Rosenbloom1\"><label>78</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Rosenbloom</surname><given-names>KR</given-names></name>, <name><surname>Dreszer</surname><given-names>TR</given-names></name>, <name><surname>Long</surname><given-names>JC</given-names></name>, <name><surname>Malladi</surname><given-names>VS</given-names></name>, <name><surname>Sloan</surname><given-names>CA</given-names></name>, <etal>et al</etal> (<year>2012</year>) <article-title>ENCODE whole-genome data in the UCSC Genome Browser: update 2012</article-title>. <source>Nucleic Acids Res</source>\\n<volume>40</volume>: <fpage>D912</fpage>&#x02013;<lpage>D917</lpage>.<pub-id pub-id-type=\"pmid\">22075998</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Karolchik1\"><label>79</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Karolchik</surname><given-names>D</given-names></name>, <name><surname>Hinrichs</surname><given-names>AS</given-names></name>, <name><surname>Furey</surname><given-names>TS</given-names></name>, <name><surname>Roskin</surname><given-names>KM</given-names></name>, <name><surname>Sugnet</surname><given-names>CW</given-names></name>, <etal>et al</etal> (<year>2004</year>) <article-title>The UCSC Table Browser data retrieval tool</article-title>. <source>Nucleic Acids Res</source>\\n<volume>32</volume>: <fpage>D493</fpage>&#x02013;<lpage>496</lpage>.<pub-id pub-id-type=\"pmid\">14681465</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Goecks1\"><label>80</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Goecks</surname><given-names>J</given-names></name>, <name><surname>Nekrutenko</surname><given-names>A</given-names></name>, <name><surname>Taylor</surname><given-names>J</given-names></name>, <name><surname>Galaxy</surname><given-names>T</given-names></name> (<year>2010</year>) <article-title>Galaxy: a comprehensive approach for supporting accessible, reproducible, and transparent computational research in the life sciences</article-title>. <source>Genome Biol</source>\\n<volume>11</volume>: <fpage>R86</fpage>.<pub-id pub-id-type=\"pmid\">20738864</pub-id></mixed-citation></ref><ref id=\"pone.0086718-ENCODEProjectConsortium1\"><label>81</label><mixed-citation publication-type=\"journal\">\\n<collab>ENCODE-Project-Consortium</collab> (<year>2011</year>) <article-title>A user&#x02019;s guide to the encyclopedia of DNA elements (ENCODE)</article-title>. <source>PLoS Biol</source>\\n<volume>9</volume>: <fpage>e1001046</fpage>.<pub-id pub-id-type=\"pmid\">21526222</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Campillos1\"><label>82</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Campillos</surname><given-names>M</given-names></name>, <name><surname>Cases</surname><given-names>I</given-names></name>, <name><surname>Hentze</surname><given-names>MW</given-names></name>, <name><surname>Sanchez</surname><given-names>M</given-names></name> (<year>2010</year>) <article-title>SIREs: searching for iron-responsive elements</article-title>. <source>Nucleic Acids Res</source>\\n<volume>38</volume>: <fpage>W360</fpage>&#x02013;<lpage>W367</lpage>.<pub-id pub-id-type=\"pmid\">20460462</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Jiang1\"><label>83</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Jiang</surname><given-names>J</given-names></name>, <name><surname>Westberg</surname><given-names>JA</given-names></name>, <name><surname>Andersson</surname><given-names>LC</given-names></name> (<year>2012</year>) <article-title>Stanniocalcin 2, forms a complex with heme oxygenase 1, binds hemin and is a heat shock protein</article-title>. <source>Biochem Biophys Res Commun</source>\\n<volume>421</volume>: <fpage>274</fpage>&#x02013;<lpage>279</lpage>.<pub-id pub-id-type=\"pmid\">22503972</pub-id></mixed-citation></ref><ref id=\"pone.0086718-Igarashi1\"><label>84</label><mixed-citation publication-type=\"journal\">\\n<name><surname>Igarashi</surname><given-names>J</given-names></name>, <name><surname>Murase</surname><given-names>M</given-names></name>, <name><surname>Iizuka</surname><given-names>A</given-names></name>, <name><surname>Pichierri</surname><given-names>F</given-names></name>, <name><surname>Martinkova</surname><given-names>M</given-names></name>, <etal>et al</etal> (<year>2008</year>) <article-title>Elucidation of the heme binding site of heme-regulated eukaryotic initiation factor 2alpha kinase and the role of the regulatory motif in heme sensing by spectroscopic and catalytic studies of mutant proteins</article-title>. <source>J Biol Chem</source>\\n<volume>283</volume>: <fpage>18782</fpage>&#x02013;<lpage>18791</lpage>.<pub-id pub-id-type=\"pmid\">18450746</pub-id></mixed-citation></ref></ref-list></back></article>',\n",
       " 'extracted_text': '\\n\\n\\nIntroduction\\n\\nMolecular evolution has recently been a popular area of investigation, and through the advancement of technology and the maturation of analysis methods, this field continues to spawn important insights into the evolutionary processes affecting genes. None of these genes or their encoded proteins exists in isolation, and the products of genes construct the metabolic pathways and networks underlying the cellular and metabolic processes of organisms. As an increasing number of studies are describing the rates of protein and pathway evolution over evolutionary time, there are more opportunities to clarify the patterns and principles of natural selection acting on the pathways involved in the metabolic networks of organisms.\\n\\nResearch focusing on the effects of the organization of pathways on the strength of selection acting on individual proteins in these pathways has revealed various evolutionary patterns among proteins at different positions in a pathway. Important research has been conducted with respect to the plant anthocyanin biosynthetic pathway [1]–[3], terpenoid biosynthesis pathway [4], starch biosynthesis pathway [5], [6], gibberellin biosynthesis pathway [7] and carotenoid biosynthesis pathway [8] as well as lateral line innovation in teleosts [9], the glucosinolate pathway in Arabidopsis thaliana[10] and the primate N-glycosylation pathway [11]. It has been demonstrated that the pleiotropic genes in the upstream portions of pathways or those found at branch points in a network are subject to stronger selective constraints [12]. On the other hand, selection is relaxed in the downstream enzymes, and nonsynonymous substitution rates as well as dN/dS ratios are higher in these pathway components.\\n\\nThe heme biosynthesis pathway is an appropriate system not only for comparing the evolutionary rates of genes according to their position or pathway reticulation but also for studying functional motifs that may play a role at several levels of gene regulation. Heme acts as an essential cofactor for cytochromes, oxidases, peroxidases, catalases, hemoglobin and myoglobin in organisms. Heme acts as an iron-chelating tetrapyrrole and is composed of a complex macrocycle containing four pyrrolic rings connected by methine bridges in cyclic form. Heme also plays multiple regulatory roles, including microRNA processing, ion channel functions, circadian rhythms, mitochondrial targeting, translational regulation and protein degradation [13]–[16]. The heme biosynthesis pathway is an especially well-characterized and important pathway for erythroid production in animals, and malfunctions in heme biosynthesis result in several types of porphyrias because of the accumulation of toxic tetrapyrrole intermediates [17]. The heme biosynthesis pathway of animals is comprised of eight consecutive genes: 5-aminolevulinic acid synthase (ALAS), porphobilinogen synthase (PBGS), porphobilinogen deaminase (PBGD), uroporphyrinogen III synthase (UROS), uroporphyrinogen III decarboxylase (UROD), coproporphyrinogen III oxidase (CPO), protoporphyrinogen IX oxidase (PPO) and ferrochelatase (FECH), from positions one to eight (Figure 1). In animals, the formation of heme occurs in the mitochondria and cytosol. Heme biosynthesis begins with the formation of 5-aminolevulinic acid. In the Shemin pathway of animals, ALAS catalyzes a single-step reaction to condense glycine and succinyl-CoA into ALA, with the elimination of CO2. In vertebrates, the housekeeping isoforms of ALAS1 are expressed in all cell types, and ALAS2 is expressed in erythroid cells (eALAS) at a very high level. Subsequently, PBGS converts two 5-aminolevulinic acids into porphobilinogen, and the addition of four PBG substrates to a dipyrrole by PBGD results in the production of pre-uroporphyrinogen, also known as hydroxymethylbilane. A circularization reaction performed by UROS generates uroporphyrinogen III. The modification of side chains on the cyclic intermediates sequentially mediated by UROD, CPO and PPO results in protoporphyrin IX, which is subsequently chelated with ferrous iron by FECH (see reviews in [14], [18]).\\n\\n10.1371/journal.pone.0086718.g001Figure 1 Heme biosynthesis pathway in animals.The substrate and product are indicated for each enzyme and the subcellular localization of each enzyme is also shown (cytosol or mitochondria). Each enzyme is coded from one to eight according to the linear order of the pathway. Also shown are the processes by which hydroxymethylbilane, the substrate of UROS, can be non-enzymatically cyclized to form uroporphyrinogen I, leading to uroporphyrin I or coproporphyrin I, and the process by which uroporphyrinogen III, the product of UROS, can be auto-oxidized to form uroporphyrin III. Protoporphyrinogen, the substrate of PPO, can be auto-oxidized to form protoporphyrin.\\n\\nALAS1 and ALAS2 are derived from gene duplication. A phylogenetic analysis of ALAS suggested that the relevant gene duplication event took place before the divergence of hagfish from the deuterostome line leading to vertebrates [19]. In extant vertebrate species, ALAS1 and ALAS2 are paralogs, and their amino acid sequences are highly similar to each other.\\n\\nRegulation of the genes encoding the eight enzymes of the heme biosynthesis pathway can occur at the transcriptional, translational and post-translational levels. At the transcriptional level, multiple erythroid-specific factors have been observed to be involved in the transcriptional activation of several genes that participate in heme biosynthesis and erythropoiesis. Among these transcription factors, genomic DNA-binding activity and the conserved binding sites of KLF1, GATA1 and TAL1 have been studied in humans and mice [20]. The binding sites of KLF1 are located within the intergenic regions or introns (particularly the first intron) of genes encoding the components of the heme biosynthesis pathway. In a de novo motif study, GATA1 and TAL1 were hypothesized to complex with KLF1 in a small subset of erythroid cis-regulatory modules [21]. The majority of the GATA1-binding sites that mediate the activation of gene expression are close to the transcription start site, within either the first intron or the proximal 5′ flanking region [22]. TAL1-binding sites for the eight genes of the heme biosynthesis pathway are detected in either proximal promoter or intronic regions [23]. A housekeeping promoter utilized in all tissue types exists in ALAS1, PBGS, PBGD, and UROS. However, for ALAS2, PBGS, PBGD and UROS, erythroid-specific promoters drive gene expression. More importantly, the alternative promoters found in PBGS, PBGD and UROS are located in intron 1, and an alternative splicing event is required for the transcript generated from the housekeeping promoters and erythroid-specific promoters [24]. It has also been established that DNase-hypersensitive sites usually serve as sites for conserved and cell-type specific transcription factor binding and histone modification [25]–[27]. Combined analysis of intron 1 sequences and DNase-hypersensitive sites will shed light on the regulatory potential of the corresponding cis-elements and the evolution of the heme biosynthesis pathway.\\n\\nAt the translational level, IRE-binding protein (IRP) interacts with the iron-responsive element (IRE) located in the 5′ untranslated regions (5′UTRs) of ALAS2 and ferritin mRNA to inhibit protein translation and in the 3′ untranslated region (3′UTR) of TfR1 to stabilize mRNA. During iron repletion, iron-sulfur clusters can abolish the IRE-binding ability of IRP1, and the F-box protein FBXL5 recognizes IRP2 and targets it for degradation by E3 ligases. Both mechanisms offer a link between iron availability and heme synthesis [28]–[30]. At the post-translational level, the binding of heme to the heme-regulatory motif (HRM) in the mature ALAS1 peptide blocks mitochondrial import and results in end-product inhibition [19], [31]–[33].\\n\\nThe objectives of this study are to investigate the molecular fingerprints underlying the adaptation of the eight genes encoding the enzymes of the heme biosynthesis pathway in Kingdom Metazoa and, particularly, how these adaptations correlate with the positions of the gene products (enzymes) in the pathway. First, through the analysis of the evolutionary constraints on the protein-coding sequences, we reveal strong purifying selection on the protein sequences and clade-specific adaptations in teleosts and arthropods. We also demonstrate that the first introns of these genes in vertebrates play a role in their erythroid-specific transactivation, highlighting the emergence of erythrocytes in animals. Then, we show that pathway product (heme) feedback control is widely utilized in this pathway. The evolutionarily conserved motifs that enable this control include HRMs found not only in ALAS but also in PBGS. We also discover IREs within 5′UTRs and exon-intron boundaries, raising the possibility of regulation of both translation and splicing choice, respectively. We conclude by summarizing the evolution of multifarious controls on the heme biosynthesis pathway.\\n\\n\\nResults\\n\\n\\nRelationship between Selection Pressure and Pathway Position\\n\\nBased on the M0 model, assuming a constant evolutionary rate (nonsynonymous versus synonymous rate, dN/dS, ω) for all branches and all codons of the eight genes encoding the enzymes of the metazoan heme biosynthesis pathway, the ω values vary from 0.041 (FECH) to 0.127 (UROS), providing evidence that the sequences of the coding regions of these genes are under negative selection (Figure 2A). We also evaluated the M1a model, which assumes that there are two groups of codons, subject to purifying selection and neutral evolution, in contrast to the M0 model. We found that the M1a model significantly improves the model (p<0.001) and that a large fraction of the codons are under purifying selection (>91% in all genes, except for PPO, where the obtained value was 79%). Notably, the ω values at positions four (UROS) and seven (PPO) were highest (0.116 and 0.105, respectively) among the genes of the heme biosynthesis pathway. (Table S1).\\n\\n10.1371/journal.pone.0086718.g002Figure 2 Selection pressure on heme biosynthesis genes in animals.ω values (dN/dS) were estimated with the M0 model for the eight heme biosynthesis genes in animals (A). The distribution of ω values (B), the nonsynonymous substitution rate, dN (C), and the synonymous substitution rate, dS (D). The order of genes follows the linear order of their pathway positions (Figure 1).\\n\\nTo determine whether the variation in ω values among the genes was statistically significant and showed a relationship with their positions in the pathway, we conducted several tests on the dN, dS and ω values obtained for each gene (Figure 2B, 2C, 2D). First, we determined whether the distributions of the dN, dS and ω values were correlated. We found that the distributions of the dN, dS and ω values were not correlated for each gene (Kruskal-Wallis rank sum test, p<0.0001). Second, the ω values of genes located at positions four (UROS) and seven (PPO) were shown to be significantly higher than for the genes located at other positions (Wilcoxon rank sum test, P<0.0001). Third, to determine the significance of the variations in ω values among the genes, we conducted multiple comparisons of the ω values for each gene pair (see Methods). Three groups were established: group 1 (low ω values), consisting of FECH, PBGD and CPO; group 2 (intermediate ω values), consisting of ALAS, UROD and PBGS; and group 3 (high ω values), consisting of PPO and UROS (Table 1). Fourth, we determined whether the dN, dS and ω values were correlated with the pathway positions of the eight enzymes of the heme biosynthesis pathway. It was found that the ω values were positively correlated with the pathway positions of the enzymes (Kendall’s correlation test: tau\\u200a=\\u200a0.2202, P<0.0001), as were the dN values (Kendall’s correlation test: tau\\u200a=\\u200a0.1625, P<0.0001). However, the dS values were not correlated with pathway position (Kendall’s correlation test: tau\\u200a=\\u200a−0.0372, P\\u200a=\\u200a0.0243).\\n\\nTable 1\\nGroup 1\\tGroup 2\\tGroup 3\\nFECH\\tPBGD\\tCPO\\tALAS\\tUROD\\tPBGS\\tPPO\\tUROS\\nFECH\\t**\\t*\\t**\\nGroup 1\\tPBGD\\t*\\t**\\nCPO\\t*\\t**\\nALAS\\t**\\nGroup 2\\tUROD\\t**\\t*\\t**\\nPBGS\\t**\\t*\\t**\\nGroup 3\\tPPO\\t**\\nUROS\\t**\\n\\nIn conclusion, the variations in ω and dN values were shown to be correlated with the pathway positions of the eight enzymes of the heme biosynthesis pathway.\\n\\n\\nAmino Acid Residues under Positive Selection\\n\\nWhen the genes of the heme biosynthesis pathway are observed to be under strong purifying selection, it is interesting to examine whether there are amino acids in certain lineages that experience positive selection. Due to limitations regarding the number of sequences collected, we focused on the protein sequences of the mammal, teleost and arthropod subgroups. Of the eight enzymes involved in heme biosynthesis, only ALAS2 from teleosts, PBGS from arthropods and UROD from teleosts showed positively selected residues in the branch-site model (p value <0.05) (Table 2). In ALAS2 from teleosts, five sites were found to be positively selected (model A/A1, p<2.3e-10) (BEB >0.978) (Table 3). Notably, while the amino acid detected in teleost ALAS2 at position 204 is I/M/T, the amino acid E is present at this position in all other vertebrate ALAS1 proteins. We mapped this amino acid in the crystal structure of the protein and found that it is positioned at the ALAS dimer interface. The amino acid R located at position 353 in teleost ALAS2 is also interesting because a K residue is found at this position in all other vertebrate ALAS1, ALAS2 and ALAS proteins. The crystal structure of ALAS indicates that this amino acid forms a hydrogen bond with the ribose-O3’ of the substrate succinyl-CoA [34] (Figure S1A).\\n\\nTable 2\\nModel\\tEstimates of parameters\\t2ΔL\\tp\\nALAS+ALAS1+ALAS2\\tM0: one ratio\\tω\\u200a=\\u200a0.05356\\nTeleost ALAS2 branch\\ttwo ratios\\tω0\\u200a=\\u200a0.05323, ω\\u200a=\\u200a8.04979\\t10.087074\\t1.49E-03\\nPBGS\\tM0: one ratio\\tω\\u200a=\\u200a0.06398\\nArthropod branch\\ttwo ratios\\tω0\\u200a=\\u200a0.06398, ω\\u200a=\\u200a999.000\\t4.478516\\t3.43E-02\\nUROD\\tM0: one ratio\\tω\\u200a=\\u200a0.06194\\nTeleost branch\\ttwo ratios\\tω0\\u200a=\\u200a0.06135, ω\\u200a=\\u200a13.69790\\t6.01671\\t1.42E-02\\n\\na\\n\\nb\\n\\nTable 3\\nForeground branches\\t2ΔL\\tp\\tEstimates of the parameters in the modified model A\\tPositively selected sites\\nTeleost ALAS2\\t40.20213\\t2.29E-10\\tp0\\u200a=\\u200a0.86464, p1\\u200a=\\u200a0.07816, p2a\\u200a=\\u200a0.05246,\\t204E, 243K, 352P, 353K,\\np2b\\u200a=\\u200a0.00474, ω0\\u200a=\\u200a0.04595, ω2\\u200a=\\u200a999.0\\t431G\\nArthropod PBGS\\t23.97615\\t9.75E-07\\tp0\\u200a=\\u200a0.86020, p1\\u200a=\\u200a0.03447, p2a\\u200a=\\u200a0.10127,\\t11Y, 106H, 162C, 195S,\\np2b\\u200a=\\u200a0.00406, ω0\\u200a=\\u200a0.06383, ω2\\u200a=\\u200a999.0\\t267K, 274A, 314I\\nTeleost UROD\\t25.60371\\t4.19E-07\\tp0\\u200a=\\u200a0.87046, p1\\u200a=\\u200a0.07295, p2a\\u200a=\\u200a0.05222,\\t136Q, 174M, 290K, 297K,\\np2b\\u200a=\\u200a0.00438, ω0\\u200a=\\u200a0.06761, ω2\\u200a=\\u200a999.0\\t300T, 316E, 349H\\n\\na\\n\\nb\\n\\nc\\n\\nd\\n\\nIn PBGS from arthropods, seven sites are positively selected (model A/A1, p<9.8e-7) (BEB >0.964) (Table 3). The amino acid found in arthropod PBGS at position 11 is I/M, whereas F/Y is present at this position in all vertebrate and cnidarian PBGS proteins. We mapped this amino acid in the crystal structure and found that it is positioned at the PBGS dimer interface and is very close to the Mg2+ binding site, which has been hypothesized to trigger the conversion of PBGS between its open and closed forms [35]. The amino acid A is present at position 195 in Drosophila PBGS, whereas the amino acid at this position is S in all other animal PBGS proteins. The crystal structure indicates that this amino acid forms the active site pocket [35] (Figure S1B).\\n\\nIn UROD from teleosts, seven sites were observed to be positively selected (model A/A1, p<4.2e-7) (BEB >0.981) (Table 3). The amino acid S is present at position 290 in teleost UROD, whereas D/E/K is found at this position in other vertebrate, cnidarian and arthropod UROD proteins. The amino acid at position 316 in the teleost UROD is H/R/S, whereas the amino acid at this position is D/E/K in other vertebrate, chordate, arthropod and cnidarian UROD proteins. We mapped both amino acid residues in the crystal structure and found that they are positioned at the UROD dimer interface [36] (Figure S1C).\\n\\nIn summary, we discovered several amino acid residues under positive selection. Some of them are located at the active sites and dimer interfaces of biologically functional enzymes of the heme biosynthesis pathway.\\n\\n\\nDetection of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences\\n\\nIt was previously reported that intron 1 sequences contain cis-elements that are necessary for the transcriptional activation of human PBGS[37], PBGD[38]–[40] and UROS[41], [42]. Therefore, we investigated the selection acting on intron sequences by searching for DNA sequences that are both evolutionarily conserved across 46 vertebrates and located in DNase-hypersensitive site clusters (ENCyclopedia of DNA Elements, ENCODE, for human). We found considerable stretches of DNA longer than 40 nucleotides in intron 1 of ALAS2, PBGS, PBGD, UROS, UROD and FECH that could serve as evolutionarily conserved DNase-hypersensitive sites in host genes (Figure 3, Table S2). However, evolutionarily conserved DNase-hypersensitive sites were also found in other intron regions, including in intron 3 of ALAS1, intron 8 of ALAS2, and introns 2 and 6 of FECH. We did not find considerable lengths of evolutionarily conserved DNase-hypersensitive sites in the intron regions of CPO and PPO. In summary, we revealed evolutionarily conserved DNase-hypersensitive sites in intron regions in six of the eight genes of the heme biosynthesis pathway, implying that conserved regulatory mechanisms acting on intron sequences might be involved in the transactivation of gene expression in the vertebrate heme biosynthesis pathway.\\n\\n10.1371/journal.pone.0086718.g003Figure 3 Three-dimensional view of the evolutionarily conserved DNase-hypersensitive sites in intron sequences.For each gene in the biosynthesis pathway (ALAS1 and ALAS2 are treated separately because they are different genes located on different chromosomes), the length of the intersection of the DNA sequence that is evolutionarily conserved across vertebrates and DNase-hypersensitive sites is indicated on the z-axis. The intron ID is provided on the x-axis. Genes from ALAS1 to FECH are shown on the y-axis and are coded from one to eight according to the linear order of the pathway. (Figure 1).\\n\\n\\nDistribution of IREs in Exon and Intron Sequences\\n\\nIREs have been reported to exist in the 5′ and 3′UTR sequences of mRNAs, through which they control the translational efficiency and stability of transcripts, respectively. The 5′UTR of human ALAS2 has been shown to contain an IRE. We conducted a survey of the 5′UTR sequences of genes of the heme biosynthesis pathway (Tables 4 and 5 and Table S3). Notably, the 5′UTRs of nine of the thirteen collected vertebrate ALAS2 sequences contained IREs of high quality (see Methods). Intriguingly, we also detected IREs in the 5′UTRs of teleost ALAS1 sequences and the ALAS sequences of one chordate (tunicate) and the purple sea urchin (echinoderm) that were of high quality as well as in a sea anemone (cnidarian) and honey bee (arthropod) that were of medium quality. These findings constitute the first demonstration that IREs can be identified in the 5′UTRs of ALAS1 sequences from vertebrates and ALAS sequences from arthropods, echinoderms and cnidarians.\\n\\nTable 4\\nGene\\tMammal\\tBird,Reptile,Amphibian\\tTeleost\\tChordate\\tEchinoderm\\tArthropod\\tCnidaria\\nALAS\\tNF\\t1(2)\\t1(1)\\t1(2)\\t1(1)\\nALAS1\\t(5)\\t3(4)\\tNF\\nALAS2\\t5(5)\\t1(1)\\t3(3)\\tNF\\nPBGS\\t(6)\\t(3)\\t(4)\\t(1)\\tNF\\t(0)\\t(1)\\nPBGD\\t(5)\\t(3)\\t(1)\\t1(1)\\t2(2)\\t(1)\\nUROS\\t(5)\\t(4)\\t(3)\\t(0)\\t(1)\\nUROD\\t(5)\\t(1)\\t(3)\\t(1)\\t(2)\\nCPO\\t(4)\\t(3)\\t(4)\\t(1)\\t(0)\\nPPO\\t(5)\\t(1)\\t(4)\\t(0)\\t(1)\\t(2)\\nFECH\\t(5)\\t(0)\\t(3)\\t(0)\\t(1)\\n\\nTable 5\\nGene\\tMammal\\tBird,Reptile,Amphibian\\tTeleost\\tChordate\\tEchinoderm\\tArthropod\\tCnidaria\\nALAS\\tNF\\t2(2)\\t1(1)\\t(4)\\t(2)\\nALAS1\\t1(7)\\t1(5)\\t(6)\\tNF\\nALAS2\\t(5)\\t1(2)\\t(5)\\tNF\\nPBGS\\t4(6)\\t2(4)\\t1(5)\\t(1)\\tNF\\t(1)\\t1(3)\\nPBGD\\t3(7)\\t(5)\\t(6)\\t(1)\\t(4)\\t1(2)\\nUROS\\t5(7)\\t3(5)\\t(4)\\t(1)\\t(2)\\nUROD\\t(7)\\t(5)\\t(2)\\t(1)\\t(4)\\t(3)\\nCPO\\t(7)\\t(5)\\t(1)\\nPPO\\t(7)\\t1(2)\\t(5)\\t(2)\\t(1)\\t1(3)\\nFECH\\t5(7)\\t(4)\\t(1)\\t(5)\\t(1)\\n\\nIREs were also found in the 5′UTRs of the PBGD sequences of Drosophila melanogaster (fruit fly) and Apis mellifera (honey bee) (high quality) as well as that of Strongylocentrotus purpuratus (sea urchin) (medium quality) (Tables 4 and 5 and Table S3). This observation raised the possibility that the translation of genes other than ALAS in the heme biosynthesis pathway could also respond to iron availability.\\n\\nWe also found IREs in intron regions. In particular, several potential IREs (of medium quality) exist at the intron-exon boundaries of PBGS from Mus musculus (mouse), PBGD from Loxodonta africana (elephant) and Oryctolagus cuniculus (rabbit) and UROS from Loxodonta africana (elephant) and Mus musculus (mouse) (Tables 4 and 5 and Table S3). This class of IREs forms stem and loop regions that overlap with protein-coding and intron sequences (Figure 4 and Figure 5). Based on these observations, we conducted a survey to identify potential IREs in the intron-exon boundary sequences of genes in the human and zebrafish genomes. We found 21 and 12 potential IREs of high quality at the intron-exon boundaries of genes in the human and zebrafish genomes, respectively (Table S4).\\n\\n10.1371/journal.pone.0086718.g004Figure 4 Potential iron-responsive elements (IREs) in the introns and intron-exon boundaries of UROS genes.IREs depicted as stem-loop structures are shown in the corresponding intron regions. UROS exon and intron IDs from four species are indicated. The conserved splicing acceptor site AG and the unpaired nucleotide of the IRE structure are also shown.\\n\\n10.1371/journal.pone.0086718.g005Figure 5 Sequence alignment of the IREs at the intron-exon boundaries of UROS from four species.“>” and “<” represent the base pairing of the RNA secondary structure. The potential IRE consensus loop sequence, CAGUGN, and the unpaired nucleotide G are also shown with respect to the location of the IRE hairpin. The intron-exon boundary is indicated as |.\\n\\nIn summary, we detected potential IREs not only in the 5′UTR of ALAS2 but also in the 5′UTRs of ALAS and ALAS1. More intriguingly, several potentially conserved IREs might exist at the intron-exon boundaries of genes involved in the heme biosynthesis pathway.\\n\\n\\nDistribution of HRMs in Protein Sequences\\n\\nThe HRM (denoted HRM_t, see Methods) has been shown to bind heme to inhibit the import of ALAS1 into mitochondria. Here, we collected recently published HRM sequences and compiled them into a new HRM consensus sequence (denoted HRM_r, see Methods). Some of the HRM_r sequences are predicted to sense the redox state of the cell and may be critical for triggering the degradation of proteins containing the HRM sequence.\\n\\nOnly ALAS1, ALAS2 and PBGD protein sequences contain HRM_t. We found HRM_t sequences in the ALAS sequences of chordates, echinoderms, and cnidarians. The ALAS1 genes of all vertebrate species show HRM_t sequences at the N-terminus. In ALAS2, HRM_t sequences were found in several types of mammals and in chickens and Xenopus (Tables 6 and 7 and Table S5).\\n\\nTable 6\\nGene\\tMammal\\tBird,Reptile,Amphibian\\tTeleost\\tChordate\\tEchinoderm\\tArthropod\\tCnidaria\\nALAS\\tNF\\t3(3)\\t2(2)\\t1(6)\\t1(2)\\nALAS1\\t7(7)\\t6(6)\\tNF\\nALAS2\\t4(5)\\t3(4)\\t(5)\\tNF\\nPBGS\\t(6)\\t(5)\\t(1)\\tNF\\t(6)\\t(3)\\nPBGD\\t(7)\\t(6)\\t(5)\\t(1)\\t3(6)\\t(2)\\nUROS\\t(7)\\t(6)\\t(4)\\t(2)\\t(1)\\t(6)\\t(3)\\nUROD\\t(7)\\t(6)\\t(5)\\t(3)\\t(1)\\t(7)\\t(3)\\nCPO\\t(7)\\t(5)\\t(2)\\t(1)\\t(6)\\t(1)\\nPPO\\t(7)\\t(5)\\t(2)\\t(1)\\t(6)\\t(3)\\nFECH\\t(7)\\t(5)\\t(4)\\t(3)\\t(1)\\t(7)\\t(2)\\n\\nTable 7\\nGene\\tMammal\\tBird,Reptile,Amphibian\\tTeleost\\tChordate\\tEchinoderm\\tArthropod\\tCnidaria\\nALAS\\tNF\\t(3)\\t(2)\\t(6)\\t(2)\\nALAS1\\t(7)\\t2(6)\\t(6)\\tNF\\nALAS2\\t5(5)\\t(4)\\t1(5)\\tNF\\nPBGS\\t6(6)\\t5(5)\\t(1)\\tNF\\t2(6)\\t1(3)\\nPBGD\\t(7)\\t(6)\\t(5)\\t(1)\\t3(6)\\t(2)\\nUROS\\t(7)\\t(6)\\t(4)\\t(2)\\t(1)\\t(6)\\t(3)\\nUROD\\t(7)\\t(6)\\t(5)\\t(3)\\t(1)\\t(7)\\t(3)\\nCPO\\t(7)\\t(5)\\t(2)\\t(1)\\t(6)\\t(1)\\nPPO\\t(7)\\t(5)\\t(2)\\t(1)\\t(6)\\t(3)\\nFECH\\t(7)\\t(5)\\t(4)\\t(3)\\t(1)\\t(7)\\t(2)\\n\\nNotably, when we used HRM_r as the search sequence for HRMs, new HRM sequences were identified in PBGS (vertebrates, arthropods, and cnidarians) and PBGD (Drosophila). This class of HRM is not restricted to the N-terminus of the protein sequence and is conserved with respect to its position in the amino acid sequence (Figure 6).\\n\\n10.1371/journal.pone.0086718.g006Figure 6 Heme-regulatory motifs (HRMs) in PBGS and PBGD.Multiple sequence alignments of PBGS (A) and PBGD (B) are shown, with HRM_t and HRM_r colored orange and green, respectively. Amino acid numbers for HRM_t and HRM_r are also shown according to the first protein sequence in the alignment.\\n\\n\\nDiscussion\\n\\nElucidation of the evolutionary history of biological pathways sheds light on the principles underlying the evolutionary forces acting on organisms in the environment. The history of pathway evolution may vary among different pathways and ancestral organisms, but understanding the underlying principles helps to reveal the modifications that have taken place in the physiological processes of organisms during their evolutionary history.\\n\\nMethods allowing the detection of selection pressure in the evolutionary histories of pathways and networks have made it possible to investigate the existence of fundamental principles driving selection in nature. By estimating the ratio of the nonsynonymous to the synonymous substitution rate for individual protein-encoding genes (ω), the types of selection pressure acting on a gene can be identified. In addition to the properties of proteins themselves, the regulatory mechanisms acting on genes are important in metabolic pathways and are also exposed to selection during evolutionary processes. A novel regulatory mechanism affecting the genes in a metabolic pathway can give rise to a new gene function that may become fixed in a lineage, through which we can determine the evolutionary history of a particular biosynthesis pathway throughout different lineages of organisms. To clarify the evolutionary history of the heme biosynthetic pathway of animals, we analyzed the three previously reported regulatory mechanisms related to genes in the heme biosynthesis pathway, which involve DNase I-hypersensitive sites, IREs, and HRMs. Our in silico prediction results showed that multiple regulatory mechanisms may exist for the genes in the heme biosynthesis pathway.\\n\\n\\nStronger Selection at the Middle and Penultimate Positions of the Pathway could Result from Self-catalysis\\n\\nWe conducted a phylogenetic analysis of the eight genes of the heme biosynthesis pathway found in the animal kingdom. The ω values at positions four (UROS) and seven (PPO) were shown to be significantly higher than at the other positions in the pathway. Notably, some self-catalysis and by-products of the heme biosynthesis pathway have been reported. Hydroxymethylbilane, the substrate of UROS, can be non-enzymatically cyclized to form uroporphyrinogen I, a useless by-product that leads to uroporphyrin I or coproporphyrin I [43]. The product of UROS can also be auto-oxidized to uroporphyrin III. Protoporphyrinogen, the substrate of PPO, has been shown to be auto-oxidized to protoporphyrin in air, without the need for PPO [43], [44]. The degree of such “leaking” among the biochemical reactions involved in the heme biosynthesis pathway is unknown, and whether this phenomenon is common in animals remains to be determined. We speculate that “leaking” in a biochemical reaction would impact selection pressure and, most likely, lead to decreased evolutionary constraint.\\n\\n\\nBiological Function of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences\\n\\nBased on the chromatin immunoprecipitation with high-throughput sequencing (ChIP-seq) technique, the transcription factor-binding module GATA1-KLF1 has been hypothesized to allow erythrocyte-specific activation of gene expression through binding to the intron 1 sequences of ALAS2, PBGS, PBGD and UROS[21], [22]. In this study, we identified intron 1 sequences as a type of DNase-hypersensitive site that is conserved in vertebrates among several genes, including ALAS2, PBGS, PBGD, UROS and UROD. This finding suggests that in response to the high demand for heme in the erythrocytes of the common ancestor of vertebrates, several transcription factors were recruited to bind to the first intron region to trans-activate the expression of the first five genes of the heme biosynthesis pathway. Notably, an intronic enhancer has been identified in intron 8 of ALAS2 in mouse erythroleukemia cells [45]–[47], which is consistent with our analysis (Figure 2). We also identified introns 2 and 6 of FECH as potential intronic enhancers, although two promoter regions of mouse FECH were shown to function in basic and inducible expression [48].\\n\\nThe use of the first intron as an alternative promoter to induce erythrocyte-specific expression has been documented for several hematopoietic genes, including Abcg2[49], [50], Ank1[51] and Slc11a2[21]. We also determined whether evolutionarily conserved intron regions of these genes could be detected. Without exception, the first intron regions of all three genes contain long stretches of DNA that are both conserved across vertebrates and accessible to DNase attack, suggesting that transcription factor binding occurs (Table S6). The high frequency of alternative first exons found in erythroid genes has been shown to be crucial for the regulation of gene function [52], and we propose here that the evolutionarily conserved region in the first intron can serve as either an alternative promoter or enhancer to enable alteration of the first exon during gene transcription [53].\\n\\n\\nElucidation of Novel Evolutionarily Conserved IREs at Intron-exon Boundaries\\n\\nAn IRE was previously identified in the 5′UTR of ALAS2 [54]. However, the detection IREs in the 5′UTRs of PBGD sequences is reported for the first time here. The scattered appearance of IREs in enzymes other than ALAS suggests that animal genes acquired IREs at a later point through the process of convergent evolution [55]. Moreover, whereas the noted IRE found in an echinoderm ALAS (Ciona intestinalis) has been reported previously [55], our identification of a potential IRE in the ALAS of a cnidarian (sea anemone) and an arthropod (honey bee) suggested that a more thorough search for the existence of IREs in animals is necessary.\\n\\nBoth the 5′UTR and 3′UTR are commonly regarded as sites where IREs are located. However, we found that intron sequences can also contain IREs. In particular, we identified an IRE at the intron 4-exon 5 boundary of UROS in Loxodonta africana (elephant). We also aligned the IRE in the corresponding region and found that this intron-exon IRE is conserved in humans, rhesus monkeys and rabbits, although these IREs are of low quality. If IREs found at intron-exon boundaries are functional, it is possible that the pre-mRNA splicing junction could be bound by IRP, thereby influencing splicing efficiency or choice.\\n\\nIn the iron-depleted state, IRP1 and IRP2 are stable and bind to IREs, either to inhibit protein translation (5′UTR) or to prevent mRNA degradation (3′UTR). In the iron-replete state, the degradation of IRP2 allows IREs to bind iron and eIF4E [56] to initiate protein translation. Intriguingly, eIF4E has been demonstrated to function as a co-factor in the Sxl-dependent female-specific alternative splicing of msl-2 as well as Sxl premRNAs in Drosophila, which is required for sex determination due to the silencing of the X chromosome [57]. This observation raised the possibility that eIF4E could be involved in splicing events in mammals. To determine how many IREs exist at intron-exon boundaries, we conducted a genome-wide survey to detect IREs of high quality with respect to the nucleotides present at the junctions of CDS-exons and introns in the human and zebrafish genomes. We found 21 and 12 high-quality IREs at human and zebrafish CDS-exon/intron junctions, respectively (Table S4). One of human genes, ZNF446, has been found in the Friendly Alternative Splicing and Transcripts Database (FAST DB) [58] that contains an alternatively spliced transcript (AY279351), which presumably depends on the corresponding junction containing an IRE. This possible phenomenon adds one more layer of complexity to the regulation of genes in the heme biosynthesis pathway in particular and to the regulation of erythropoiesis in general. Furthermore, this finding also calls for an investigation into the mechanism by which iron and IRP regulate the alternative splicing of protein-coding genes and the associated biological effects.\\n\\n\\nElucidation of Novel Evolutionarily Conserved HRMs\\n\\nBy defining a new HRM (HRM_r), we were able to identify HRMs in most of the studied PBGS genes (76%, 19/25) and all PBGD genes from Drosophila. Recently, HRMs have been implicated in regulatory functions other than controlling import into the mitochondria. The HRM in human IRP2 has been shown to be responsible for the ubiquitin-targeted degradation of the IRP2 protein [59]. The degradation of human circadian Factor Period 2 (hPer2) is also mediated by an HRM, suggesting that metabolic signals can modulate the circadian regulation of gene expression [60]. Notably, mouse ALAS1 and ALAS2 have been shown to be under circadian control and are regulated by mPER1 and mPER2 [61]. It has been demonstrated that the HRM of ALAS2 is not involved in the heme-mediated control of import into the mitochondria [32]. Therefore, we postulate that the HRMs identified in ALAS2 and PBGS could be involved in protein degradation. This is consistent with the localization of the PBGS enzyme in the cytosol and that, unlike ALAS1 and ALAS2, PBGS does not need to enter the mitochondria to function (Figure 1). The HRM binds heme and senses both the concentration of heme and the oxidation/reduction state of the cell. Our findings indicate that product feedback control of protein stability could be involved in the evolution of the heme biosynthesis pathway.\\n\\n\\nEvolutionary Implications of HRMs and IREs in the Teleost Lineage based on Comparison with other Species\\n\\nWe found HRMs in the ALAS sequences of chordates and echinoderms. ALAS1 HRM_t exists in all investigated vertebrate species as well. While we also detected ALAS2 HRM_t in certain monophyletic groups (mammals, birds, and amphibians), we did not find any HRM_t sequences in teleosts. Additionally, we identified IREs in the 5′UTRs of ALAS sequences from chordates and echinoderms. In the 5′UTR of ALAS2, there is an IRE in all of the examined vertebrate species. In the 5′UTR of ALAS1, an IRE is found in teleosts, but not in mammals. Based on our phylogenetic analysis, chordate and echinoderm ALAS and vertebrate ALAS1/2 are derived from the same ancestral ALAS gene. We propose that the HRMs and IREs observed in ALAS1 and ALAS2 existed in the ALAS sequence of a common ancestor of vertebrates, chordates, and echinoderms. The unique absence of HRMs in ALAS2 proteins and the presence of IREs in ALAS1 mRNAs in teleosts are striking. We further propose that during the evolutionary branching and speciation of vertebrates, the loss of the HRM from ALAS2 and the retention of the IRE in ALAS1 in teleosts resulted from the adaptation of the heme biosynthesis pathway to the environment. Notably, we have also shown that the amino acid sequences of ALAS2 and UROD have experienced positive selection in the teleost lineage.\\n\\n\\nConclusion: An Integrated View of the Evolution of Multiple Types of Regulation vs. Pathway Position Reveals Different Depths within the Pathway\\n\\nWe have investigated the evolution of multiple controls modulating the heme biosynthesis pathway. It was illuminating to study the degree by which these combinatorial controls impinge on the pathway from top to bottom, if we regard ALAS, at position one, as the top and FECH, at position eight, as the bottom. In a summary table (Table 8), with the eight genes represented from positions one to eight in rows and the six control mechanisms across the top of the table, it can clearly be observed that there are multiple degrees of regulatory potential with respect to this pathway. Notably, transcriptional control and iron-mediated splicing control infiltrate the middle and bottom of the pathway. Substrate (iron) or product (heme) feedback translational control as well as protein localization and protein stability control primarily infiltrate the top of the pathway. Purifying selection on the protein sequences is widespread but is subject to loose constraint in the middle and near the bottom of the pathway. An intriguing question from the point of view of pathway position is whether the evolvability of pathway regulation depends on the position of the genes in the pathway, thereby leading to differential adaptation. Our research suggests that the investigation of the molecular evolution of a pathway should involve the examination of different control mechanisms related to the activity of the genes that constitute the pathway under investigation. Thus, a new perspective in the field of evolutionary developmental biology could concern the roles played by the cis-regulatory regions and protein-coding regions of genes with respect to adaptive mutations [62], [63]. We are aware that our key findings in this study are predictions based on the current understanding of the heme biosynthesis pathway, and these findings will require substantial experimental work to be confirmed or corrected.\\n\\nTable 8\\nGene\\tPathway Position\\tFirst Intron-mediatedTranscription Control\\tSplicing Control\\tTranslational Control\\tProtein Localization\\tProteinStability\\tSelection Pressure\\nALAS\\t1\\t**\\nALAS1\\t1\\t*\\t***\\t*\\nALAS2\\t1\\t**\\t***\\t**\\nPBGS\\t2\\t**\\t*\\t***\\nPBGD\\t3\\t***\\t*\\nUROS\\t4\\t*\\t***\\nUROD\\t5\\t*\\nCPO\\t6\\nPPO\\t7\\t***\\nFECH\\t8\\t**\\n\\na\\n\\nb\\n\\nc\\n\\nd\\n\\ne\\n\\nf\\n\\n\\nMaterials and Methods\\n\\n\\nSequence Collection\\n\\nFor genes encoding the eight enzymes of the heme biosynthesis pathway in animals, we collected the sequences of organisms whose entire genome sequences were available. Sequences were collected from mammals (seven species), amphibians (two species), birds (three species), reptiles (one species), teleosts (six species), echinoderms (two species), arthropods (eight species) and cnidarians (three species). The amino acid sequences, coding nucleotide sequences, exon sequences and intron sequences were downloaded from NCBI, UCSC [64], [65] and Ensembl [66]. Some of the protein sequences of ALAS genes were extracted as suggested previously [19]. The 5′UTR, exon and intron sequences of the coral Acropora digitifera came from OIST Marine Genome Unit [67]. The 5′UTR, exon and intron sequences of the sea anemone Nematostella vectensis were obtained from the DOE Joint Genome Institute [68]. The 5′UTR, exon and intron sequences of the hydra Hydra magnipapillata also came from the DOE Joint Genome Institute [69].\\n\\nIf a gene model lacked a 5′UTR, we searched for a cDNA collection to determine whether there was any possibility of extension in the 5′ direction with respect to the direction of transcription. If such an extension was possible, a 5′UTR was added to the gene model. Subsequently, if this extension added a new exon, which was regarded as the first exon, then a new intron, which was usually the first intron, was hypothesized to exist. The numbers of coding sequences obtained for each gene in the heme biosynthesis pathway were as follows: ALAS (46 genes from 31 species), PBGS (26 genes from 26 species), PBGD (29 genes from 28 species), UROS (28 genes from 28 species), UROD (31 genes from 31 species), CPO (27 genes from 27 species), PPO (26 genes from 26 species) and FECH (30 genes from 30 species) (Table S7). Lists about the number of sequences and the species by taxonomic groups can be found in Table S8 and S9. Sequences from species whose genome had not been decoded were not included in this study.\\n\\n\\nSequence Alignment and Phylogenetic Analysis\\n\\nThe amino acid sequence alignment of the eight enzymes involved in the heme biosynthesis pathway was performed in MEGA5 with the MUSCLE algorithm [70]. After the aligned sequences were adjusted manually to confirm their accuracy, topologies of the phylogenetic trees were generated with PHYLIP [71] and PHYML [72] via maximum likelihood (ML) methods. The gamma rate heterogeneity model and the JTT substitution model from Tree-Puzzle [73] and MEGA5 were used for the enzymes of the heme biosynthesis pathway. Branch support was provided via bootstrap analysis, involving a heuristic search with 1000 replicates. The alignments and trees are shown in Figures S2 and S3, respectively.\\n\\n\\nAnalysis of Evolutionary Constraints\\n\\nThe aligned nucleotide coding sequences without gaps based on the aligned protein sequences and the unrooted tree were fed into the CODEML program of the PAML program package (version V4.4e) [74], [75] to analyze the evolutionary constraints on the coding sequences for the eight genes of the heme biosynthesis pathway. For the different models applied, ω (the nonsynonymous versus synonymous rate, dN/dS) for codons can be assumed to be less than one (negative selection or purifying selection), equal to one (neutral), or greater than one (positive selection).\\n\\nFirst, we obtained ω for each gene by applying the null model M0, assuming a constant ω value for all codons and branches. Second, site models M1a (Nearly Neutral) and M2a (Positive Selection) were both used to allow ω to vary among sites. In the M1a model, the codons are categorized into two types, one of which shows ω values of less than one, while ω is equal to one of the other type. In the M2a model, the codons are categorized into three types, one with ω values less than one, one for which ω is equal to one and one with ω values greater than one. We performed the likelihood ratio test for the M1a and M2a models compared to the null model, M0 (twice the log-likelihood difference (2ΔlnL) of the two models).\\n\\nWe also used the branch model to analyze the evolutionary constraints on the mammal, teleost and arthropod branches. We then compared the likelihood of the null model, M0, with the branch model. If the likelihood of the branch model were to be significantly higher than that of the null model M0, it was hypothesized that the branch under consideration is potentially under positive selection. Subsequently, we used the branch-site model to allow variation among sites in the proteins and across branches to determine whether any amino acid residues were under positive selection [76]. For branch-site model A, the following four classes were demarcated for each amino acid: class 0 with 0< ω0<1 in all branches; class 1 with ω1\\u200a=\\u200a1 in all branches; class 2a with foreground ω2≥1 but background 0< ω0<1; and class 2b with a foreground ω2≥1 but background ω1\\u200a=\\u200a1. Null model A1 was the same as A but with the foreground ω2 constrained to one. A likelihood ratio test is used between models A and A1. If the likelihood of model A were to be significantly higher than the likelihood of model A1 (null model) (p<0.05), it would indicate that there were amino acid residues under positive selection. Bayes empirical Bayes (BEB) was used to calculate the posterior probability of identifying sites under positive selection (>0.95) [77]. When we describe the position of the amino acids under positive selection, we use the human sequences (for ALAS1, PBGS and UROD) as a reference to index the sequences. We mapped the positively selected amino acids onto the crystal structures of ALAS [34], PBGS [35] and UROD [36].\\n\\nWe also compared the ω values among the eight genes using a previously described method [2], [8]. When the difference of the ω values between two genes, such as gene A, with ωA and a likelihood of LA0 in the M0 null model, and gene B, with ωB and a likelihood of LB0 in the M0 null model, was significant, we were able to find a ωn between the two genes that was significantly different from those of the two genes being compared. To perform this test, we set ωn as the average ω value of the two genes being compared (ωn\\u200a=\\u200a(ωA+ωB)/2). We then acquired the likelihood values (LA and LB) after we constrained the ω of genes A and B as ωn. Subsequently, the statistical significance (df\\u200a=\\u200a1, p<0.05) of the difference in ω was determined from the difference in the likelihood between the null model M0 and the constrained model, as follows: 2*(LA – LA0) and 2*(LB – LB0).\\n\\n\\nDetection of Evolutionarily Conserved DNase-hypersensitive Sites in Intron Sequences\\n\\nWe used the UCSC table browser [78], [79] and Galaxy [80] to extract the intron sequences at the intersections of regions that are conserved in vertebrates (hg19, phastConsElements46way) and DNase-hypersensitive sites (hg19, wgEncodeRegDnaseClustered) [81].\\n\\n\\nDetection of IREs\\n\\nWe used the SIRE web service to identify potential IRE sites in the gene sequences [82]. SIRE takes into consideration the non-canonical sequence as indicated by SELEX. By allowing 18 motifs to be confirmed as binding to IRP1 or IRP2, SIRE accepted the input sequence and reported the motif type, the free energy of the secondary structure and the level of stringency as High, Medium, or Low. A batch version of the same algorithm was also developed for the genome-wide detection of IREs at the intron-exon boundaries of human and zebrafish genes.\\n\\n\\nDetection of HRMs\\n\\nThe HRM sequence, N/K/R-C-P-K or a hydrophobic residue-L/M, has been commonly used to detect HRMs [19], [31] (denoted HRM_t). By identifying new occurrences of HRMs that have been shown to function biologically (human ALAS1; human ALAS2 [32]; human, mouse, rat, spalax, and zebrafish PER2 [60]; human IRP2 [59]; human STC2 [83]; and human, mouse and rabbit eIF2alpha [84]), we also identified a new HRM motif, denoted HRM_r, A/C/F/G/I/R/S/Q-A/C/K/H/L/N/R/S/T-C-P-A/E/F/I/K/S/V/Y-A/D/H/I/L/M/T/V-A/L/M/P/R/S. We used seven sites in HRM_r rather than five sites in HRM_t to increase the specificity of HRM_r. We note that HRM_r is similar to, but not the same as HRM_t.\\n\\n\\nSupporting InformationFigure S1\\n\\n\\nThe collection of figures describing the protein structures of enzymes involved in the heme biosynthesis pathway and the positions of the positively selected residues. Homodimer structures are shown with the monomers colored in white and yellow. Positively selected amino acid residues are colored in red. Substrate analogs or prosthetic groups are colored in green. (A) ALAS2 of teleost; (B) PBGS of arthropod; (C) UROD of teleost.\\n\\n(PDF)\\n\\nClick here for additional data file.Figure S2\\n\\n\\nThe aligned protein sequences of eight genes of heme biosynthesis pathway in animals.\\n\\n(PDF)\\n\\nClick here for additional data file.Figure S3\\n\\n\\nThe collection of figures for the maximum likelihood phylogeny of protein sequences for eight genes of heme biosynthesis pathway in animals. Bootstrap values >70% are indicated. The bootstrap values are displayed only for the branches of the main lineages. (A) ALAS, (B) PBGS, (C) PBGD, (D) UROS, (E) UROD, (F) CPO, (G) PPO, (H) FECH.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S1\\n\\n\\nModel test (M1a) for selection of genes in heme biosynthesis pathway.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S2\\n\\n\\nLength of evolutionarily conserved DNase-hypersensitive sites in intron sequences.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S3\\n\\n\\nPotential IRE in eight genes of heme biosynthesis pathway.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S4\\n\\n\\nGenomewide detection of potential IRE in exon-intron boundary in human and zebrafish genes.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S5\\n\\n\\nPotential HRM in eight genes of heme biosynthesis pathway (HRM_t and HRM_r).\\n\\n(PDF)\\n\\nClick here for additional data file.Table S6\\n\\n\\nLength of evolutionarily conserved DNase-hypersensitive sites in intron sequences (bps) for Abcg2, Ank1 and Slc11a2.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S7\\n\\n\\nIDs of genes of heme biosynthesis pathway in animals.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S8\\n\\n\\nNumber of sequences by taxonomic groups.\\n\\n(PDF)\\n\\nClick here for additional data file.Table S9\\n\\n\\nSpecies by taxonomic groups.\\n\\n(PDF)\\n\\nClick here for additional data file.'}"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.find_one()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using `collection.update_one()`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in files:\n",
    "    with open(file, \"r\") as f:\n",
    "        xml = f.read()\n",
    "        article = Article(xml)\n",
    "        articles.update_one(\n",
    "            filter = {\n",
    "                \"_id\": article.pub_ids().get(\"pmc\")\n",
    "            },\n",
    "            update = {\n",
    "                \"$set\": {\n",
    "                    \"xml\": xml,\n",
    "                    \"extracted_text\": article.extract_text()\n",
    "                }\n",
    "            },\n",
    "            upsert=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Stuff for Python"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO: Figure out how to use a boolean query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'extracted_text_text'"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "articles.create_index([(\"extracted_text\", pymongo.TEXT,)])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Searching on multiple phrases"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Using sets to record the IDs of matching documents"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "metadata": {},
   "outputs": [],
   "source": [
    "matching_ids = {}\n",
    "for term in terms:\n",
    "    results = articles.find(\n",
    "        { '$text': { '$search': term } },\n",
    "        { '_id': '_id', 'score': { '$meta': \"textScore\" } }\n",
    "    )\n",
    "    result_set = {result[\"_id\"] for result in results}\n",
    "    matching_ids[term] = result_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'1247408',\n",
       " '1481558',\n",
       " '1564427',\n",
       " '1779612',\n",
       " '1890282',\n",
       " '2234391',\n",
       " '2453293',\n",
       " '2528962',\n",
       " '2605341',\n",
       " '2606823',\n",
       " '2667394',\n",
       " '2688363',\n",
       " '2689396',\n",
       " '2732522',\n",
       " '2736310',\n",
       " '2738531',\n",
       " '2748713',\n",
       " '2779852',\n",
       " '2789409',\n",
       " '2811100',\n",
       " '2830968',\n",
       " '2882574',\n",
       " '2886090',\n",
       " '2896373',\n",
       " '2908054',\n",
       " '2965324',\n",
       " '2974751',\n",
       " '2995795',\n",
       " '3037366',\n",
       " '3083667',\n",
       " '3111462',\n",
       " '3112453',\n",
       " '3135997',\n",
       " '3163512',\n",
       " '3227807',\n",
       " '3236786',\n",
       " '3249557',\n",
       " '3266242',\n",
       " '3281017',\n",
       " '3287448',\n",
       " '3313868',\n",
       " '3317851',\n",
       " '3320900',\n",
       " '3366669',\n",
       " '3388975',\n",
       " '3426517',\n",
       " '3457864',\n",
       " '3484178',\n",
       " '3494555',\n",
       " '3497848',\n",
       " '3526392',\n",
       " '3536940',\n",
       " '3556712',\n",
       " '3557901',\n",
       " '3562581',\n",
       " '3565895',\n",
       " '3577466',\n",
       " '3580738',\n",
       " '3597257',\n",
       " '3612753',\n",
       " '3638147',\n",
       " '3646733',\n",
       " '3663739',\n",
       " '3670493',\n",
       " '3694090',\n",
       " '3784500',\n",
       " '3808113',\n",
       " '3826076',\n",
       " '3842058',\n",
       " '3846669',\n",
       " '3849780',\n",
       " '3850638',\n",
       " '3853547',\n",
       " '3886783',\n",
       " '3890265',\n",
       " '3922777',\n",
       " '3922792',\n",
       " '3994642',\n",
       " '4014550',\n",
       " '4023239',\n",
       " '4079231',\n",
       " '4109881',\n",
       " '4117438',\n",
       " '4154735',\n",
       " '4185335',\n",
       " '4185895',\n",
       " '4191159',\n",
       " '4220626',\n",
       " '4259558',\n",
       " '4262227',\n",
       " '4263169',\n",
       " '4284477',\n",
       " '4299528',\n",
       " '4328079',\n",
       " '4340618',\n",
       " '4361704',\n",
       " '4381858',\n",
       " '4407983',\n",
       " '4418331',\n",
       " '4478264',\n",
       " '4514805',\n",
       " '4591604',\n",
       " '4606121',\n",
       " '4607726',\n",
       " '4627380',\n",
       " '4650156',\n",
       " '4677441',\n",
       " '4682827',\n",
       " '4689723',\n",
       " '4710461',\n",
       " '4729434',\n",
       " '4737873',\n",
       " '4744409',\n",
       " '4747220',\n",
       " '4756622',\n",
       " '4813400',\n",
       " '4837512',\n",
       " '4901396',\n",
       " '4919788',\n",
       " '4931437',\n",
       " '4942914',\n",
       " '4955207',\n",
       " '4978428',\n",
       " '4983589',\n",
       " '4998946',\n",
       " '5030537',\n",
       " '5087500',\n",
       " '5120930',\n",
       " '5125606',\n",
       " '5215488',\n",
       " '5226610',\n",
       " '5293235',\n",
       " '5333444',\n",
       " '5333626',\n",
       " '5379547',\n",
       " '5388494',\n",
       " '5388882',\n",
       " '5404402',\n",
       " '5430074',\n",
       " '5435308',\n",
       " '5437650',\n",
       " '5437860',\n",
       " '5451426',\n",
       " '5493353',\n",
       " '5524359',\n",
       " '5544251',\n",
       " '5547140',\n",
       " '5572604',\n",
       " '5586202',\n",
       " '5600973',\n",
       " '5646764',\n",
       " '5650168',\n",
       " '5658195',\n",
       " '5660108',\n",
       " '5660281',\n",
       " '5664348',\n",
       " '5664651',\n",
       " '5664821',\n",
       " '5724700',\n",
       " '5727347',\n",
       " '5738111',\n",
       " '5738821',\n",
       " '5750227',\n",
       " '5773159',\n",
       " '5774785',\n",
       " '5777222',\n",
       " '5781586',\n",
       " '5809952',\n",
       " '5812211',\n",
       " '5815329',\n",
       " '5820427',\n",
       " '5836856',\n",
       " '5842876',\n",
       " '5856350',\n",
       " '5953934',\n",
       " '5978865',\n",
       " '5988275',\n",
       " '6014127',\n",
       " '6028742',\n",
       " '6029123',\n",
       " '6080975',\n",
       " '6086862',\n",
       " '6145019',\n",
       " '6154931',\n",
       " '6168496',\n",
       " '6175676',\n",
       " '6202706',\n",
       " '6203989',\n",
       " '6239056',\n",
       " '6288267',\n",
       " '6291107',\n",
       " '6292131',\n",
       " '6293958',\n",
       " '6351983',\n",
       " '6377993',\n",
       " '6378561',\n",
       " '6393904',\n",
       " '6425011'}"
      ]
     },
     "execution_count": 244,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matching_any = set.union(*matching_ids.values())\n",
    "matching_any"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Using a defaultdict() to record which IDs a thing as matched\n",
    "\n",
    "**USE THIS ONE**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(list,\n",
       "            {'3236786': ['\"field study\"', '\"study site\"'],\n",
       "             '3994642': ['\"field study\"'],\n",
       "             '4837512': ['\"field study\"'],\n",
       "             '5773159': ['\"field study\"'],\n",
       "             '4185895': ['\"field study\"'],\n",
       "             '5600973': ['\"field study\"'],\n",
       "             '3426517': ['\"field study\"', '\"study site\"'],\n",
       "             '5660281': ['\"field study\"'],\n",
       "             '3842058': ['\"field study\"'],\n",
       "             '3320900': ['\"field study\"'],\n",
       "             '5660108': ['\"field study\"'],\n",
       "             '3922777': ['\"field study\"'],\n",
       "             '4650156': ['\"field study\"'],\n",
       "             '4284477': ['\"field study\"'],\n",
       "             '5664821': ['\"field study\"', '\"study site\"'],\n",
       "             '4259558': ['\"field study\"'],\n",
       "             '4682827': ['\"field study\"'],\n",
       "             '4361704': ['\"field study\"'],\n",
       "             '4931437': ['\"field study\"'],\n",
       "             '3638147': ['\"field study\"'],\n",
       "             '3037366': ['\"field study\"'],\n",
       "             '6425011': ['\"field study\"'],\n",
       "             '1564427': ['\"study site\"'],\n",
       "             '3163512': ['\"study site\"', '\"field work\"'],\n",
       "             '4677441': ['\"study site\"'],\n",
       "             '5493353': ['\"study site\"'],\n",
       "             '3281017': ['\"study site\"'],\n",
       "             '6377993': ['\"study site\"'],\n",
       "             '6168496': ['\"study site\"'],\n",
       "             '5215488': ['\"study site\"'],\n",
       "             '5451426': ['\"study site\"'],\n",
       "             '4744409': ['\"study site\"'],\n",
       "             '2779852': ['\"study site\"'],\n",
       "             '3366669': ['\"study site\"'],\n",
       "             '6080975': ['\"study site\"'],\n",
       "             '2830968': ['\"study site\"'],\n",
       "             '5524359': ['\"study site\"'],\n",
       "             '5836856': ['\"study site\"'],\n",
       "             '4191159': ['\"study site\"'],\n",
       "             '6028742': ['\"study site\"'],\n",
       "             '3670493': ['\"study site\"'],\n",
       "             '2453293': ['\"study site\"'],\n",
       "             '2689396': ['\"study site\"', '\"field work\"'],\n",
       "             '4079231': ['\"study site\"'],\n",
       "             '3497848': ['\"study site\"'],\n",
       "             '5572604': ['\"study site\"'],\n",
       "             '4263169': ['\"study site\"'],\n",
       "             '3556712': ['\"study site\"'],\n",
       "             '5781586': ['\"study site\"'],\n",
       "             '6202706': ['\"study site\"'],\n",
       "             '4729434': ['\"study site\"'],\n",
       "             '3784500': ['\"study site\"', '\"field work\"'],\n",
       "             '5978865': ['\"study site\"'],\n",
       "             '2886090': ['\"study site\"'],\n",
       "             '3886783': ['\"study site\"'],\n",
       "             '5226610': ['\"study site\"'],\n",
       "             '6393904': ['\"study site\"'],\n",
       "             '2738531': ['\"study site\"'],\n",
       "             '4955207': ['\"study site\"'],\n",
       "             '6014127': ['\"study site\"'],\n",
       "             '3388975': ['\"study site\"'],\n",
       "             '4737873': ['\"study site\"'],\n",
       "             '2736310': ['\"study site\"'],\n",
       "             '2606823': ['\"study site\"'],\n",
       "             '4756622': ['\"study site\"'],\n",
       "             '6154931': ['\"study site\"'],\n",
       "             '4747220': ['\"study site\"'],\n",
       "             '5333626': ['\"study site\"'],\n",
       "             '3850638': ['\"study site\"', '\"field work\"'],\n",
       "             '6239056': ['\"study site\"'],\n",
       "             '5646764': ['\"study site\"'],\n",
       "             '5664651': ['\"study site\"'],\n",
       "             '5586202': ['\"study site\"'],\n",
       "             '3922792': ['\"study site\"'],\n",
       "             '5812211': ['\"study site\"'],\n",
       "             '1481558': ['\"study site\"'],\n",
       "             '4627380': ['\"study site\"'],\n",
       "             '5430074': ['\"study site\"'],\n",
       "             '3313868': ['\"study site\"', '\"field work\"'],\n",
       "             '4407983': ['\"study site\"'],\n",
       "             '5404402': ['\"study site\"'],\n",
       "             '5388494': ['\"study site\"'],\n",
       "             '2995795': ['\"study site\"'],\n",
       "             '3612753': ['\"study site\"'],\n",
       "             '6378561': ['\"study site\"'],\n",
       "             '5809952': ['\"study site\"'],\n",
       "             '4689723': ['\"study site\"'],\n",
       "             '2896373': ['\"study site\"'],\n",
       "             '4381858': ['\"study site\"'],\n",
       "             '6203989': ['\"study site\"'],\n",
       "             '6175676': ['\"study site\"'],\n",
       "             '4607726': ['\"study site\"'],\n",
       "             '4606121': ['\"study site\"'],\n",
       "             '5437860': ['\"study site\"'],\n",
       "             '6029123': ['\"study site\"'],\n",
       "             '4262227': ['\"study site\"'],\n",
       "             '6291107': ['\"study site\"'],\n",
       "             '4299528': ['\"study site\"'],\n",
       "             '5379547': ['\"study site\"'],\n",
       "             '4328079': ['\"study site\"'],\n",
       "             '3808113': ['\"study site\"'],\n",
       "             '2811100': ['\"study site\"'],\n",
       "             '3111462': ['\"study site\"'],\n",
       "             '2908054': ['\"study site\"'],\n",
       "             '2688363': ['\"study site\"'],\n",
       "             '3646733': ['\"study site\"'],\n",
       "             '3849780': ['\"study site\"', '\"field work\"'],\n",
       "             '5658195': ['\"study site\"'],\n",
       "             '3112453': ['\"study site\"'],\n",
       "             '5333444': ['\"study site\"'],\n",
       "             '5664348': ['\"study site\"'],\n",
       "             '4813400': ['\"study site\"'],\n",
       "             '3562581': ['\"study site\"'],\n",
       "             '5087500': ['\"study site\"'],\n",
       "             '6288267': ['\"study site\"'],\n",
       "             '3266242': ['\"study site\"'],\n",
       "             '5125606': ['\"study site\"'],\n",
       "             '3890265': ['\"study site\"'],\n",
       "             '2732522': ['\"study site\"'],\n",
       "             '4418331': ['\"study site\"'],\n",
       "             '3484178': ['\"study site\"'],\n",
       "             '5738111': ['\"study site\"'],\n",
       "             '3597257': ['\"study site\"'],\n",
       "             '4942914': ['\"study site\"'],\n",
       "             '3846669': ['\"study site\"'],\n",
       "             '4998946': ['\"study site\"'],\n",
       "             '3694090': ['\"study site\"'],\n",
       "             '5544251': ['\"study site\"'],\n",
       "             '4340618': ['\"study site\"'],\n",
       "             '5650168': ['\"study site\"'],\n",
       "             '4591604': ['\"study site\"'],\n",
       "             '5030537': ['\"study site\"'],\n",
       "             '4978428': ['\"study site\"'],\n",
       "             '3135997': ['\"study site\"'],\n",
       "             '5750227': ['\"study site\"'],\n",
       "             '5293235': ['\"study site\"'],\n",
       "             '4014550': ['\"study site\"'],\n",
       "             '5437650': ['\"study site\"'],\n",
       "             '5856350': ['\"study site\"'],\n",
       "             '3565895': ['\"study site\"'],\n",
       "             '2528962': ['\"study site\"'],\n",
       "             '5988275': ['\"study site\"'],\n",
       "             '5120930': ['\"study site\"'],\n",
       "             '4154735': ['\"study site\"'],\n",
       "             '2234391': ['\"study site\"'],\n",
       "             '4901396': ['\"study site\"'],\n",
       "             '6351983': ['\"study site\"'],\n",
       "             '5727347': ['\"study site\"'],\n",
       "             '3580738': ['\"study site\"'],\n",
       "             '5388882': ['\"study site\"'],\n",
       "             '6145019': ['\"study site\"'],\n",
       "             '3536940': ['\"study site\"'],\n",
       "             '3249557': ['\"study site\"'],\n",
       "             '4185335': ['\"study site\"'],\n",
       "             '4983589': ['\"study site\"'],\n",
       "             '3227807': ['\"study site\"'],\n",
       "             '5724700': ['\"study site\"'],\n",
       "             '4117438': ['\"study site\"'],\n",
       "             '1779612': ['\"study site\"'],\n",
       "             '3853547': ['\"study site\"'],\n",
       "             '5820427': ['\"study site\"'],\n",
       "             '5777222': ['\"study site\"'],\n",
       "             '3494555': ['\"study site\"'],\n",
       "             '5953934': ['\"study site\"'],\n",
       "             '5815329': ['\"study site\"'],\n",
       "             '5842876': ['\"study site\"'],\n",
       "             '4220626': ['\"study site\"'],\n",
       "             '3577466': ['\"study site\"'],\n",
       "             '4710461': ['\"study site\"'],\n",
       "             '3557901': ['\"study site\"'],\n",
       "             '6086862': ['\"study site\"'],\n",
       "             '3083667': ['\"study site\"'],\n",
       "             '6292131': ['\"study site\"'],\n",
       "             '4478264': ['\"study site\"'],\n",
       "             '3826076': ['\"field work\"'],\n",
       "             '5738821': ['\"field work\"'],\n",
       "             '3457864': ['\"field work\"'],\n",
       "             '2789409': ['\"field work\"'],\n",
       "             '5435308': ['\"field work\"'],\n",
       "             '2605341': ['\"field work\"'],\n",
       "             '1247408': ['\"field work\"'],\n",
       "             '3317851': ['\"field work\"'],\n",
       "             '3287448': ['\"field work\"'],\n",
       "             '5547140': ['\"field work\"'],\n",
       "             '2748713': ['\"field work\"'],\n",
       "             '1890282': ['\"field work\"'],\n",
       "             '4514805': ['\"field work\"'],\n",
       "             '3526392': ['\"field work\"'],\n",
       "             '5774785': ['\"field work\"'],\n",
       "             '2667394': ['\"field work\"'],\n",
       "             '2974751': ['\"field work\"'],\n",
       "             '6293958': ['\"field work\"'],\n",
       "             '3663739': ['\"field work\"'],\n",
       "             '2965324': ['\"field work\"'],\n",
       "             '4109881': ['\"field work\"'],\n",
       "             '4919788': ['\"field work\"'],\n",
       "             '4023239': ['\"field work\"'],\n",
       "             '2882574': ['\"field work\"']})"
      ]
     },
     "execution_count": 248,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results = defaultdict(list)\n",
    "for term in terms:\n",
    "    term_results = articles.find(\n",
    "        { '$text': { '$search': term } },\n",
    "        { '_id': '_id', 'score': { '$meta': \"textScore\" } }\n",
    "    )\n",
    "    for r in term_results:\n",
    "        results[r[\"_id\"]].append(term)\n",
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_list = [result for result in results]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_id': '3826076', 'score': 1.438532305076512}"
      ]
     },
     "execution_count": 227,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result_list[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_set = {result[\"_id\"] for result in results}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'1247408',\n",
       " '1890282',\n",
       " '2605341',\n",
       " '2667394',\n",
       " '2689396',\n",
       " '2748713',\n",
       " '2789409',\n",
       " '2882574',\n",
       " '2965324',\n",
       " '2974751',\n",
       " '3163512',\n",
       " '3287448',\n",
       " '3313868',\n",
       " '3317851',\n",
       " '3457864',\n",
       " '3526392',\n",
       " '3663739',\n",
       " '3784500',\n",
       " '3826076',\n",
       " '3849780',\n",
       " '3850638',\n",
       " '4023239',\n",
       " '4109881',\n",
       " '4514805',\n",
       " '4919788',\n",
       " '5435308',\n",
       " '5547140',\n",
       " '5738821',\n",
       " '5774785',\n",
       " '6293958'}"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result_set"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Basic code for walking with a limit\n",
    "\n",
    "However, this isn't usable, because we actually need to *randomly* sample the files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "<class 'str'>\n",
      "Stopped.\n"
     ]
    }
   ],
   "source": [
    "i = 10\n",
    "for path, dirs, files in os.walk(pmc_path):\n",
    "    if i <= 0:\n",
    "        break\n",
    "    for file in files[:i]:\n",
    "        if file.endswith(\"xml\"):\n",
    "            i -= 1\n",
    "            with open(os.path.join(path, file), \"r\") as file:\n",
    "                article = Article(file.read())\n",
    "            print(article.pub_ids().get(\"pmid\"))\n",
    "print(\"Stopped.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Counting how many have PMIDs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "foo = dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(foo.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2500\n",
      "2475\n",
      "2450\n",
      "2425\n",
      "2400\n",
      "2375\n",
      "2350\n",
      "2325\n",
      "2300\n",
      "2275\n",
      "2250\n",
      "2225\n",
      "2200\n",
      "2175\n",
      "2150\n",
      "2125\n",
      "2100\n",
      "2075\n",
      "2050\n",
      "2025\n",
      "2000\n",
      "1975\n",
      "1950\n",
      "1925\n",
      "1900\n",
      "1875\n",
      "1850\n",
      "1825\n",
      "1800\n",
      "1775\n",
      "1750\n",
      "1725\n",
      "1700\n",
      "1675\n",
      "1650\n",
      "1625\n",
      "1600\n",
      "1575\n",
      "1550\n",
      "1525\n",
      "1500\n",
      "1475\n",
      "1450\n",
      "1425\n",
      "1400\n",
      "1375\n",
      "1350\n",
      "1325\n",
      "1300\n",
      "1275\n",
      "1250\n",
      "1225\n",
      "1200\n",
      "1175\n",
      "1150\n",
      "1125\n",
      "1100\n",
      "1075\n",
      "1050\n",
      "1025\n",
      "1000\n",
      "975\n",
      "950\n",
      "925\n",
      "900\n",
      "875\n",
      "850\n",
      "825\n",
      "800\n",
      "775\n",
      "750\n",
      "725\n",
      "700\n",
      "675\n",
      "650\n",
      "625\n",
      "600\n",
      "575\n",
      "550\n",
      "525\n",
      "500\n",
      "475\n",
      "450\n",
      "425\n",
      "400\n",
      "375\n",
      "350\n",
      "325\n",
      "300\n",
      "275\n",
      "250\n",
      "225\n",
      "200\n",
      "175\n",
      "150\n",
      "125\n",
      "100\n",
      "75\n",
      "50\n",
      "25\n",
      "defaultdict(<class 'int'>, {'pmid': 2484, 'pmc': 2500, 'doi': 2500, 'publisher-id': 2476, 'art-access-id': 4, 'manuscript': 11, 'coden': 2002, 'pii': 1995})\n"
     ]
    }
   ],
   "source": [
    "id_types = defaultdict(int)\n",
    "\n",
    "i = 2500\n",
    "for path, dirs, files in os.walk(pmc_path):\n",
    "    if i <= 0:\n",
    "        break\n",
    "    for file in files[:i]:\n",
    "        if file.endswith(\"xml\"):\n",
    "            if i % 25 is 0:\n",
    "                print(i)\n",
    "            i -= 1\n",
    "            with open(os.path.join(path, file), \"r\") as file:\n",
    "                article = Article(file.read())\n",
    "            ids = article.pub_ids().keys()\n",
    "            if len(ids) is 0:\n",
    "                id_types[\"none\"] += 1\n",
    "            else:\n",
    "                for item in ids:\n",
    "                    id_types[item] += 1\n",
    "            \n",
    "print(id_types)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
